{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7d39dbd5",
   "metadata": {},
   "source": [
    "## Prostate methylation panel - 5/2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7ce53b5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(stringr)\n",
    "library(openxlsx)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2cba478a",
   "metadata": {},
   "source": [
    "# make a file that can be used to annotate literature regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8010437f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# HG38\n",
    "# gencode gene annotations: https://www.gencodegenes.org/human/\n",
    "# load gene exon annotations\n",
    "\n",
    "# df_gencode = read.csv(file=\"../Translational_Science/gencode.v40.annotation.tsv\", sep=\"\\t\", header=FALSE) #HG38 file\n",
    "# colnames(df_gencode) = c(\"chr\",\"database\",\"type\",\"start\",\"end\",\"dot1\",\"strand\",\"dot2\",\"annotation\")\n",
    "# head(df_gencode, 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "96d543ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>database</th><th scope=col>type</th><th scope=col>start</th><th scope=col>end</th><th scope=col>dot1</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>HAVANA</td><td>gene      </td><td>11869</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                                                                                                                                                                                                                                                                       </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>HAVANA</td><td>transcript</td><td>11869</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>ID=ENST00000456328.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap                                                             </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>HAVANA</td><td>exon      </td><td>11869</td><td>12227</td><td>.</td><td>+</td><td>.</td><td>ID=exon:ENST00000456328.2:1;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=1;exon_id=ENSE00002234944.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:11869-12227;remap_status=full_contig                  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>HAVANA</td><td>exon      </td><td>12613</td><td>12721</td><td>.</td><td>+</td><td>.</td><td>ID=exon:ENST00000456328.2:2;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=2;exon_id=ENSE00003582793.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:12613-12721;remap_status=full_contig                  </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>HAVANA</td><td>exon      </td><td>13221</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>ID=exon:ENST00000456328.2:3;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=3;exon_id=ENSE00002312635.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:13221-14409;remap_status=full_contig                  </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>HAVANA</td><td>transcript</td><td>12010</td><td>13670</td><td>.</td><td>+</td><td>.</td><td>ID=ENST00000450305.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000450305.2_2;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=transcribed_unprocessed_pseudogene;transcript_name=DDX11L1-201;level=2;transcript_support_level=NA;hgnc_id=HGNC:37102;ont=PGO:0000005,PGO:0000019;tag=basic,Ensembl_canonical;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000002844.2_2;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & chr & database & type & start & end & dot1 & strand & dot2 & annotation\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & HAVANA & gene       & 11869 & 14409 & . & + & . & ID=ENSG00000223972.5;gene\\_id=ENSG00000223972.5\\_5;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;level=2;hgnc\\_id=HGNC:37102;havana\\_gene=OTTHUMG00000000961.2\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                                                                                                                                                                                                                                                                       \\\\\n",
       "\t2 & chr1 & HAVANA & transcript & 11869 & 14409 & . & + & . & ID=ENST00000456328.2;Parent=ENSG00000223972.5;gene\\_id=ENSG00000223972.5\\_5;transcript\\_id=ENST00000456328.2\\_1;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;transcript\\_type=processed\\_transcript;transcript\\_name=DDX11L1-202;level=2;transcript\\_support\\_level=1;hgnc\\_id=HGNC:37102;tag=basic;havana\\_gene=OTTHUMG00000000961.2\\_5;havana\\_transcript=OTTHUMT00000362751.1\\_1;remap\\_num\\_mappings=1;remap\\_status=full\\_contig;remap\\_target\\_status=overlap                                                             \\\\\n",
       "\t3 & chr1 & HAVANA & exon       & 11869 & 12227 & . & + & . & ID=exon:ENST00000456328.2:1;Parent=ENST00000456328.2;gene\\_id=ENSG00000223972.5\\_5;transcript\\_id=ENST00000456328.2\\_1;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;transcript\\_type=processed\\_transcript;transcript\\_name=DDX11L1-202;exon\\_number=1;exon\\_id=ENSE00002234944.1\\_1;level=2;transcript\\_support\\_level=1;hgnc\\_id=HGNC:37102;tag=basic;havana\\_gene=OTTHUMG00000000961.2\\_5;havana\\_transcript=OTTHUMT00000362751.1\\_1;remap\\_original\\_location=chr1:+:11869-12227;remap\\_status=full\\_contig                  \\\\\n",
       "\t4 & chr1 & HAVANA & exon       & 12613 & 12721 & . & + & . & ID=exon:ENST00000456328.2:2;Parent=ENST00000456328.2;gene\\_id=ENSG00000223972.5\\_5;transcript\\_id=ENST00000456328.2\\_1;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;transcript\\_type=processed\\_transcript;transcript\\_name=DDX11L1-202;exon\\_number=2;exon\\_id=ENSE00003582793.1\\_1;level=2;transcript\\_support\\_level=1;hgnc\\_id=HGNC:37102;tag=basic;havana\\_gene=OTTHUMG00000000961.2\\_5;havana\\_transcript=OTTHUMT00000362751.1\\_1;remap\\_original\\_location=chr1:+:12613-12721;remap\\_status=full\\_contig                  \\\\\n",
       "\t5 & chr1 & HAVANA & exon       & 13221 & 14409 & . & + & . & ID=exon:ENST00000456328.2:3;Parent=ENST00000456328.2;gene\\_id=ENSG00000223972.5\\_5;transcript\\_id=ENST00000456328.2\\_1;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;transcript\\_type=processed\\_transcript;transcript\\_name=DDX11L1-202;exon\\_number=3;exon\\_id=ENSE00002312635.1\\_1;level=2;transcript\\_support\\_level=1;hgnc\\_id=HGNC:37102;tag=basic;havana\\_gene=OTTHUMG00000000961.2\\_5;havana\\_transcript=OTTHUMT00000362751.1\\_1;remap\\_original\\_location=chr1:+:13221-14409;remap\\_status=full\\_contig                  \\\\\n",
       "\t6 & chr1 & HAVANA & transcript & 12010 & 13670 & . & + & . & ID=ENST00000450305.2;Parent=ENSG00000223972.5;gene\\_id=ENSG00000223972.5\\_5;transcript\\_id=ENST00000450305.2\\_2;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;transcript\\_type=transcribed\\_unprocessed\\_pseudogene;transcript\\_name=DDX11L1-201;level=2;transcript\\_support\\_level=NA;hgnc\\_id=HGNC:37102;ont=PGO:0000005,PGO:0000019;tag=basic,Ensembl\\_canonical;havana\\_gene=OTTHUMG00000000961.2\\_5;havana\\_transcript=OTTHUMT00000002844.2\\_2;remap\\_num\\_mappings=1;remap\\_status=full\\_contig;remap\\_target\\_status=overlap\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | database &lt;chr&gt; | type &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | dot1 &lt;chr&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | annotation &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | HAVANA | gene       | 11869 | 14409 | . | + | . | ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                                                                                                                                                                                                                                                                        |\n",
       "| 2 | chr1 | HAVANA | transcript | 11869 | 14409 | . | + | . | ID=ENST00000456328.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap                                                              |\n",
       "| 3 | chr1 | HAVANA | exon       | 11869 | 12227 | . | + | . | ID=exon:ENST00000456328.2:1;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=1;exon_id=ENSE00002234944.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:11869-12227;remap_status=full_contig                   |\n",
       "| 4 | chr1 | HAVANA | exon       | 12613 | 12721 | . | + | . | ID=exon:ENST00000456328.2:2;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=2;exon_id=ENSE00003582793.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:12613-12721;remap_status=full_contig                   |\n",
       "| 5 | chr1 | HAVANA | exon       | 13221 | 14409 | . | + | . | ID=exon:ENST00000456328.2:3;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=3;exon_id=ENSE00002312635.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:13221-14409;remap_status=full_contig                   |\n",
       "| 6 | chr1 | HAVANA | transcript | 12010 | 13670 | . | + | . | ID=ENST00000450305.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000450305.2_2;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=transcribed_unprocessed_pseudogene;transcript_name=DDX11L1-201;level=2;transcript_support_level=NA;hgnc_id=HGNC:37102;ont=PGO:0000005,PGO:0000019;tag=basic,Ensembl_canonical;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000002844.2_2;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  database type       start end   dot1 strand dot2\n",
       "1 chr1 HAVANA   gene       11869 14409 .    +      .   \n",
       "2 chr1 HAVANA   transcript 11869 14409 .    +      .   \n",
       "3 chr1 HAVANA   exon       11869 12227 .    +      .   \n",
       "4 chr1 HAVANA   exon       12613 12721 .    +      .   \n",
       "5 chr1 HAVANA   exon       13221 14409 .    +      .   \n",
       "6 chr1 HAVANA   transcript 12010 13670 .    +      .   \n",
       "  annotation                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
       "1 ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                                                                                                                                                                                                                                                                       \n",
       "2 ID=ENST00000456328.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap                                                             \n",
       "3 ID=exon:ENST00000456328.2:1;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=1;exon_id=ENSE00002234944.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:11869-12227;remap_status=full_contig                  \n",
       "4 ID=exon:ENST00000456328.2:2;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=2;exon_id=ENSE00003582793.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:12613-12721;remap_status=full_contig                  \n",
       "5 ID=exon:ENST00000456328.2:3;Parent=ENST00000456328.2;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000456328.2_1;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;exon_number=3;exon_id=ENSE00002312635.1_1;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000362751.1_1;remap_original_location=chr1:+:13221-14409;remap_status=full_contig                  \n",
       "6 ID=ENST00000450305.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;transcript_id=ENST00000450305.2_2;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=transcribed_unprocessed_pseudogene;transcript_name=DDX11L1-201;level=2;transcript_support_level=NA;hgnc_id=HGNC:37102;ont=PGO:0000005,PGO:0000019;tag=basic,Ensembl_canonical;havana_gene=OTTHUMG00000000961.2_5;havana_transcript=OTTHUMT00000002844.2_2;remap_num_mappings=1;remap_status=full_contig;remap_target_status=overlap"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# hg19/GR37 gencode file\n",
    "df_gencode_hg19 = read.table(file=\"../Translational_Science/gencode.v40lift37.annotation_hg19..gff3\", sep=\"\\t\", header=FALSE)\n",
    "colnames(df_gencode_hg19) = c(\"chr\",\"database\",\"type\",\"start\",\"end\",\"dot1\",\"strand\",\"dot2\",\"annotation\")\n",
    "head(df_gencode_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "0a872e98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>database</th><th scope=col>type</th><th scope=col>start</th><th scope=col>end</th><th scope=col>dot1</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>annotation</th><th scope=col>tmp</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>HAVANA</td><td>gene</td><td>11869</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>DDX11L1    </td></tr>\n",
       "\t<tr><th scope=row>13</th><td>1</td><td>HAVANA</td><td>gene</td><td>14404</td><td>29570</td><td>.</td><td>-</td><td>.</td><td>ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             </td><td>WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     </td><td>WASH7P     </td></tr>\n",
       "\t<tr><th scope=row>26</th><td>1</td><td>HAVANA</td><td>gene</td><td>29554</td><td>31109</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap       </td><td>MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>MIR1302-2HG</td></tr>\n",
       "\t<tr><th scope=row>34</th><td>1</td><td>HAVANA</td><td>gene</td><td>34554</td><td>36081</td><td>.</td><td>-</td><td>.</td><td>ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                            </td><td>FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>FAM138A    </td></tr>\n",
       "\t<tr><th scope=row>42</th><td>1</td><td>HAVANA</td><td>gene</td><td>52473</td><td>53312</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             </td><td>OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     </td><td>OR4G4P     </td></tr>\n",
       "\t<tr><th scope=row>45</th><td>1</td><td>HAVANA</td><td>gene</td><td>57598</td><td>64116</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>OR4G11P    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & chr & database & type & start & end & dot1 & strand & dot2 & annotation & tmp & gene\\_name\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & HAVANA & gene & 11869 & 14409 & . & + & . & ID=ENSG00000223972.5;gene\\_id=ENSG00000223972.5\\_5;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;level=2;hgnc\\_id=HGNC:37102;havana\\_gene=OTTHUMG00000000961.2\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & DDX11L1;level=2;hgnc\\_id=HGNC:37102;havana\\_gene=OTTHUMG00000000961.2\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & DDX11L1    \\\\\n",
       "\t13 & 1 & HAVANA & gene & 14404 & 29570 & . & - & . & ID=ENSG00000227232.5;gene\\_id=ENSG00000227232.5\\_5;gene\\_type=unprocessed\\_pseudogene;gene\\_name=WASH7P;level=2;hgnc\\_id=HGNC:38034;havana\\_gene=OTTHUMG00000000958.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap              & WASH7P;level=2;hgnc\\_id=HGNC:38034;havana\\_gene=OTTHUMG00000000958.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                      & WASH7P     \\\\\n",
       "\t26 & 1 & HAVANA & gene & 29554 & 31109 & . & + & . & ID=ENSG00000243485.5;gene\\_id=ENSG00000243485.5\\_11;gene\\_type=lncRNA;gene\\_name=MIR1302-2HG;level=2;hgnc\\_id=HGNC:52482;tag=ncRNA\\_host;havana\\_gene=OTTHUMG00000000959.2\\_11;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap        & MIR1302-2HG;level=2;hgnc\\_id=HGNC:52482;tag=ncRNA\\_host;havana\\_gene=OTTHUMG00000000959.2\\_11;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & MIR1302-2HG\\\\\n",
       "\t34 & 1 & HAVANA & gene & 34554 & 36081 & . & - & . & ID=ENSG00000237613.2;gene\\_id=ENSG00000237613.2\\_6;gene\\_type=lncRNA;gene\\_name=FAM138A;level=2;hgnc\\_id=HGNC:32334;havana\\_gene=OTTHUMG00000000960.1\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                             & FAM138A;level=2;hgnc\\_id=HGNC:32334;havana\\_gene=OTTHUMG00000000960.1\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & FAM138A    \\\\\n",
       "\t42 & 1 & HAVANA & gene & 52473 & 53312 & . & + & . & ID=ENSG00000268020.3;gene\\_id=ENSG00000268020.3\\_5;gene\\_type=unprocessed\\_pseudogene;gene\\_name=OR4G4P;level=2;hgnc\\_id=HGNC:14822;havana\\_gene=OTTHUMG00000185779.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap              & OR4G4P;level=2;hgnc\\_id=HGNC:14822;havana\\_gene=OTTHUMG00000185779.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                      & OR4G4P     \\\\\n",
       "\t45 & 1 & HAVANA & gene & 57598 & 64116 & . & + & . & ID=ENSG00000240361.2;gene\\_id=ENSG00000240361.2\\_6;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=OR4G11P;level=2;hgnc\\_id=HGNC:31276;havana\\_gene=OTTHUMG00000001095.3\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & OR4G11P;level=2;hgnc\\_id=HGNC:31276;havana\\_gene=OTTHUMG00000001095.3\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & OR4G11P    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | database &lt;chr&gt; | type &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | dot1 &lt;chr&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | annotation &lt;chr&gt; | tmp &lt;chr&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | HAVANA | gene | 11869 | 14409 | . | + | . | ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | DDX11L1     |\n",
       "| 13 | 1 | HAVANA | gene | 14404 | 29570 | . | - | . | ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap              | WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                      | WASH7P      |\n",
       "| 26 | 1 | HAVANA | gene | 29554 | 31109 | . | + | . | ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap        | MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | MIR1302-2HG |\n",
       "| 34 | 1 | HAVANA | gene | 34554 | 36081 | . | - | . | ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                             | FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | FAM138A     |\n",
       "| 42 | 1 | HAVANA | gene | 52473 | 53312 | . | + | . | ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap              | OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                      | OR4G4P      |\n",
       "| 45 | 1 | HAVANA | gene | 57598 | 64116 | . | + | . | ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | OR4G11P     |\n",
       "\n"
      ],
      "text/plain": [
       "   chr database type start end   dot1 strand dot2\n",
       "1  1   HAVANA   gene 11869 14409 .    +      .   \n",
       "13 1   HAVANA   gene 14404 29570 .    -      .   \n",
       "26 1   HAVANA   gene 29554 31109 .    +      .   \n",
       "34 1   HAVANA   gene 34554 36081 .    -      .   \n",
       "42 1   HAVANA   gene 52473 53312 .    +      .   \n",
       "45 1   HAVANA   gene 57598 64116 .    +      .   \n",
       "   annotation                                                                                                                                                                                                                                             \n",
       "1  ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "13 ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             \n",
       "26 ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap       \n",
       "34 ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                            \n",
       "42 ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             \n",
       "45 ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "   tmp                                                                                                                                                                \n",
       "1  DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "13 WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     \n",
       "26 MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "34 FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "42 OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     \n",
       "45 OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "   gene_name  \n",
       "1  DDX11L1    \n",
       "13 WASH7P     \n",
       "26 MIR1302-2HG\n",
       "34 FAM138A    \n",
       "42 OR4G4P     \n",
       "45 OR4G11P    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_gencode_hg19_genes = df_gencode_hg19[df_gencode_hg19$type==\"gene\",]\n",
    "df_gencode_hg19_genes$tmp = sapply(strsplit(df_gencode_hg19_genes$annotation, \"gene_name=\"), \"[\", 2)\n",
    "df_gencode_hg19_genes$gene_name = sapply(strsplit(df_gencode_hg19_genes$tmp, \";\"), \"[\", 1)\n",
    "\n",
    "df_gencode_hg19_genes$chr <- sapply(strsplit(df_gencode_hg19_genes$chr, \"chr\"), \"[\", 2)\n",
    "head(df_gencode_hg19_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "555048a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>database</th><th scope=col>type</th><th scope=col>start</th><th scope=col>end</th><th scope=col>dot1</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>annotation</th><th scope=col>tmp</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>HAVANA</td><td>gene</td><td>11869</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>DDX11L1    </td></tr>\n",
       "\t<tr><th scope=row>13</th><td>1</td><td>HAVANA</td><td>gene</td><td>14404</td><td>29570</td><td>.</td><td>-</td><td>.</td><td>ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             </td><td>WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     </td><td>WASH7P     </td></tr>\n",
       "\t<tr><th scope=row>26</th><td>1</td><td>HAVANA</td><td>gene</td><td>29554</td><td>31109</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap       </td><td>MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>MIR1302-2HG</td></tr>\n",
       "\t<tr><th scope=row>34</th><td>1</td><td>HAVANA</td><td>gene</td><td>34554</td><td>36081</td><td>.</td><td>-</td><td>.</td><td>ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                            </td><td>FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>FAM138A    </td></tr>\n",
       "\t<tr><th scope=row>42</th><td>1</td><td>HAVANA</td><td>gene</td><td>52473</td><td>53312</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             </td><td>OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     </td><td>OR4G4P     </td></tr>\n",
       "\t<tr><th scope=row>45</th><td>1</td><td>HAVANA</td><td>gene</td><td>57598</td><td>64116</td><td>.</td><td>+</td><td>.</td><td>ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap</td><td>OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    </td><td>OR4G11P    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & chr & database & type & start & end & dot1 & strand & dot2 & annotation & tmp & gene\\_name\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & HAVANA & gene & 11869 & 14409 & . & + & . & ID=ENSG00000223972.5;gene\\_id=ENSG00000223972.5\\_5;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=DDX11L1;level=2;hgnc\\_id=HGNC:37102;havana\\_gene=OTTHUMG00000000961.2\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & DDX11L1;level=2;hgnc\\_id=HGNC:37102;havana\\_gene=OTTHUMG00000000961.2\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & DDX11L1    \\\\\n",
       "\t13 & 1 & HAVANA & gene & 14404 & 29570 & . & - & . & ID=ENSG00000227232.5;gene\\_id=ENSG00000227232.5\\_5;gene\\_type=unprocessed\\_pseudogene;gene\\_name=WASH7P;level=2;hgnc\\_id=HGNC:38034;havana\\_gene=OTTHUMG00000000958.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap              & WASH7P;level=2;hgnc\\_id=HGNC:38034;havana\\_gene=OTTHUMG00000000958.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                      & WASH7P     \\\\\n",
       "\t26 & 1 & HAVANA & gene & 29554 & 31109 & . & + & . & ID=ENSG00000243485.5;gene\\_id=ENSG00000243485.5\\_11;gene\\_type=lncRNA;gene\\_name=MIR1302-2HG;level=2;hgnc\\_id=HGNC:52482;tag=ncRNA\\_host;havana\\_gene=OTTHUMG00000000959.2\\_11;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap        & MIR1302-2HG;level=2;hgnc\\_id=HGNC:52482;tag=ncRNA\\_host;havana\\_gene=OTTHUMG00000000959.2\\_11;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & MIR1302-2HG\\\\\n",
       "\t34 & 1 & HAVANA & gene & 34554 & 36081 & . & - & . & ID=ENSG00000237613.2;gene\\_id=ENSG00000237613.2\\_6;gene\\_type=lncRNA;gene\\_name=FAM138A;level=2;hgnc\\_id=HGNC:32334;havana\\_gene=OTTHUMG00000000960.1\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                             & FAM138A;level=2;hgnc\\_id=HGNC:32334;havana\\_gene=OTTHUMG00000000960.1\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & FAM138A    \\\\\n",
       "\t42 & 1 & HAVANA & gene & 52473 & 53312 & . & + & . & ID=ENSG00000268020.3;gene\\_id=ENSG00000268020.3\\_5;gene\\_type=unprocessed\\_pseudogene;gene\\_name=OR4G4P;level=2;hgnc\\_id=HGNC:14822;havana\\_gene=OTTHUMG00000185779.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap              & OR4G4P;level=2;hgnc\\_id=HGNC:14822;havana\\_gene=OTTHUMG00000185779.1\\_5;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                      & OR4G4P     \\\\\n",
       "\t45 & 1 & HAVANA & gene & 57598 & 64116 & . & + & . & ID=ENSG00000240361.2;gene\\_id=ENSG00000240361.2\\_6;gene\\_type=transcribed\\_unprocessed\\_pseudogene;gene\\_name=OR4G11P;level=2;hgnc\\_id=HGNC:31276;havana\\_gene=OTTHUMG00000001095.3\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap & OR4G11P;level=2;hgnc\\_id=HGNC:31276;havana\\_gene=OTTHUMG00000001095.3\\_6;remap\\_status=full\\_contig;remap\\_num\\_mappings=1;remap\\_target\\_status=overlap                     & OR4G11P    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | database &lt;chr&gt; | type &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | dot1 &lt;chr&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | annotation &lt;chr&gt; | tmp &lt;chr&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | HAVANA | gene | 11869 | 14409 | . | + | . | ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | DDX11L1     |\n",
       "| 13 | 1 | HAVANA | gene | 14404 | 29570 | . | - | . | ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap              | WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                      | WASH7P      |\n",
       "| 26 | 1 | HAVANA | gene | 29554 | 31109 | . | + | . | ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap        | MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | MIR1302-2HG |\n",
       "| 34 | 1 | HAVANA | gene | 34554 | 36081 | . | - | . | ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                             | FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | FAM138A     |\n",
       "| 42 | 1 | HAVANA | gene | 52473 | 53312 | . | + | . | ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap              | OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                      | OR4G4P      |\n",
       "| 45 | 1 | HAVANA | gene | 57598 | 64116 | . | + | . | ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap | OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     | OR4G11P     |\n",
       "\n"
      ],
      "text/plain": [
       "   chr database type start end   dot1 strand dot2\n",
       "1  1   HAVANA   gene 11869 14409 .    +      .   \n",
       "13 1   HAVANA   gene 14404 29570 .    -      .   \n",
       "26 1   HAVANA   gene 29554 31109 .    +      .   \n",
       "34 1   HAVANA   gene 34554 36081 .    -      .   \n",
       "42 1   HAVANA   gene 52473 53312 .    +      .   \n",
       "45 1   HAVANA   gene 57598 64116 .    +      .   \n",
       "   annotation                                                                                                                                                                                                                                             \n",
       "1  ID=ENSG00000223972.5;gene_id=ENSG00000223972.5_5;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "13 ID=ENSG00000227232.5;gene_id=ENSG00000227232.5_5;gene_type=unprocessed_pseudogene;gene_name=WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             \n",
       "26 ID=ENSG00000243485.5;gene_id=ENSG00000243485.5_11;gene_type=lncRNA;gene_name=MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap       \n",
       "34 ID=ENSG00000237613.2;gene_id=ENSG00000237613.2_6;gene_type=lncRNA;gene_name=FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                            \n",
       "42 ID=ENSG00000268020.3;gene_id=ENSG00000268020.3_5;gene_type=unprocessed_pseudogene;gene_name=OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap             \n",
       "45 ID=ENSG00000240361.2;gene_id=ENSG00000240361.2_6;gene_type=transcribed_unprocessed_pseudogene;gene_name=OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "   tmp                                                                                                                                                                \n",
       "1  DDX11L1;level=2;hgnc_id=HGNC:37102;havana_gene=OTTHUMG00000000961.2_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "13 WASH7P;level=2;hgnc_id=HGNC:38034;havana_gene=OTTHUMG00000000958.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     \n",
       "26 MIR1302-2HG;level=2;hgnc_id=HGNC:52482;tag=ncRNA_host;havana_gene=OTTHUMG00000000959.2_11;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap\n",
       "34 FAM138A;level=2;hgnc_id=HGNC:32334;havana_gene=OTTHUMG00000000960.1_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "42 OR4G4P;level=2;hgnc_id=HGNC:14822;havana_gene=OTTHUMG00000185779.1_5;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                     \n",
       "45 OR4G11P;level=2;hgnc_id=HGNC:31276;havana_gene=OTTHUMG00000001095.3_6;remap_status=full_contig;remap_num_mappings=1;remap_target_status=overlap                    \n",
       "   gene_name  \n",
       "1  DDX11L1    \n",
       "13 WASH7P     \n",
       "26 MIR1302-2HG\n",
       "34 FAM138A    \n",
       "42 OR4G4P     \n",
       "45 OR4G11P    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_gencode_hg19_genes$end <- as.integer(format(df_gencode_hg19_genes$end, scientific=FALSE))\n",
    "df_gencode_hg19_genes$start <- as.integer(format(df_gencode_hg19_genes$start, scientific=FALSE))\n",
    "df_gencode_hg19_genes = df_gencode_hg19_genes[order(df_gencode_hg19_genes$chr, df_gencode_hg19_genes$start,df_gencode_hg19_genes$end),]\n",
    "\n",
    "df_gencode_hg19_genes = df_gencode_hg19_genes[df_gencode_hg19_genes!=\"M\",]\n",
    "df_gencode_hg19_genes = df_gencode_hg19_genes[complete.cases(df_gencode_hg19_genes),]\n",
    "head(df_gencode_hg19_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "11930497",
   "metadata": {},
   "outputs": [],
   "source": [
    "return_promoter_start <- function(bed_row){\n",
    "    if (!(bed_row[7] %in% c(\"+\",\"-\"))){\n",
    "        print(bed_row[7])\n",
    "        stop()}\n",
    "    if (bed_row[7]==\"+\"){\n",
    "        return(as.numeric(bed_row[4])-1000)\n",
    "    }else if(bed_row[7]==\"-\"){\n",
    "        return(bed_row[5])\n",
    "    }}\n",
    "return_promoter_end <- function(bed_row){\n",
    "    if (!(bed_row[7] %in% c(\"+\",\"-\"))){\n",
    "        print(bed_row[7])\n",
    "        stop()}\n",
    "    if (bed_row[7]==\"+\"){\n",
    "        return(bed_row[4])\n",
    "    }else if(bed_row[7]==\"-\"){\n",
    "        return(as.numeric(bed_row[5])+1000)\n",
    "    }}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "7ea5eb99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>10869</td><td>11869</td><td>DDX11L1    </td><td>0</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>1</td><td>29570</td><td>30570</td><td>WASH7P     </td><td>0</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>1</td><td>28554</td><td>29554</td><td>MIR1302-2HG</td><td>0</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>34</th><td>1</td><td>36081</td><td>37081</td><td>FAM138A    </td><td>0</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>42</th><td>1</td><td>51473</td><td>52473</td><td>OR4G4P     </td><td>0</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>45</th><td>1</td><td>56598</td><td>57598</td><td>OR4G11P    </td><td>0</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & chr & start & end & gene\\_name & score & strand & dot2 & type\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 10869 & 11869 & DDX11L1     & 0 & + & . & promoter\\\\\n",
       "\t13 & 1 & 29570 & 30570 & WASH7P      & 0 & - & . & promoter\\\\\n",
       "\t26 & 1 & 28554 & 29554 & MIR1302-2HG & 0 & + & . & promoter\\\\\n",
       "\t34 & 1 & 36081 & 37081 & FAM138A     & 0 & - & . & promoter\\\\\n",
       "\t42 & 1 & 51473 & 52473 & OR4G4P      & 0 & + & . & promoter\\\\\n",
       "\t45 & 1 & 56598 & 57598 & OR4G11P     & 0 & + & . & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;dbl&gt; | end &lt;dbl&gt; | gene_name &lt;chr&gt; | score &lt;dbl&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 10869 | 11869 | DDX11L1     | 0 | + | . | promoter |\n",
       "| 13 | 1 | 29570 | 30570 | WASH7P      | 0 | - | . | promoter |\n",
       "| 26 | 1 | 28554 | 29554 | MIR1302-2HG | 0 | + | . | promoter |\n",
       "| 34 | 1 | 36081 | 37081 | FAM138A     | 0 | - | . | promoter |\n",
       "| 42 | 1 | 51473 | 52473 | OR4G4P      | 0 | + | . | promoter |\n",
       "| 45 | 1 | 56598 | 57598 | OR4G11P     | 0 | + | . | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "   chr start end   gene_name   score strand dot2 type    \n",
       "1  1   10869 11869 DDX11L1     0     +      .    promoter\n",
       "13 1   29570 30570 WASH7P      0     -      .    promoter\n",
       "26 1   28554 29554 MIR1302-2HG 0     +      .    promoter\n",
       "34 1   36081 37081 FAM138A     0     -      .    promoter\n",
       "42 1   51473 52473 OR4G4P      0     +      .    promoter\n",
       "45 1   56598 57598 OR4G11P     0     +      .    promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_promoters_hg19 = df_gencode_hg19_genes \n",
    "df_promoters_hg19$start = as.numeric(apply(df_gencode_hg19_genes, 1, return_promoter_start))\n",
    "df_promoters_hg19$end = as.numeric(apply(df_gencode_hg19_genes, 1, return_promoter_end))\n",
    "\n",
    "df_promoters_hg19 = df_promoters_hg19 %>% mutate(type=\"promoter\", score=0) %>%\n",
    "    select(chr, start, end, gene_name, score, strand, dot2, type)\n",
    "head(df_promoters_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "9548191d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_promoters_hg19 = apply(df_promoters_hg19, 2, as.character)\n",
    "# df_promoters_hg19$start <- as.numeric(df_promoters_hg19$start)\n",
    "# df_promoters_hg19$end <- as.numeric(df_promoters_hg19$end)\n",
    "\n",
    "cols = c('chr','start','end','gene_name','score','strand','type')\n",
    "write.table(df_promoters_hg19[,cols], file=\"../Translational_Science/gencode.v40lift37.promoters_hg19.bed\",\n",
    "           row.names=FALSE, sep=\"\\t\", quote=FALSE, col.names=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "8e156492",
   "metadata": {},
   "outputs": [],
   "source": [
    "expand_start <- function(bed_row){\n",
    "    if (!(bed_row[7] %in% c(\"+\",\"-\"))){\n",
    "        print(bed_row[7])\n",
    "        stop()}\n",
    "    if (bed_row[7]==\"+\"){\n",
    "        return(as.numeric(bed_row[4])-1000)\n",
    "    }else{\n",
    "        return(bed_row[4])\n",
    "    }}\n",
    "expand_end <- function(bed_row){\n",
    "    if (!(bed_row[7] %in% c(\"+\",\"-\"))){\n",
    "        print(bed_row[7])\n",
    "        stop()}\n",
    "    if (bed_row[7]==\"-\"){\n",
    "        return(as.numeric(bed_row[5])+1000)\n",
    "    }else if(bed_row[7]==\"+\"){\n",
    "        return(bed_row[5])\n",
    "    }}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "d3764f2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>10869    </td><td>    14409</td><td>DDX11L1    </td><td>0</td><td>+</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>    14404</td><td>30570    </td><td>WASH7P     </td><td>0</td><td>-</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>28554    </td><td>    31109</td><td>MIR1302-2HG</td><td>0</td><td>+</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>34</th><td>chr1</td><td>    34554</td><td>37081    </td><td>FAM138A    </td><td>0</td><td>-</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>42</th><td>chr1</td><td>51473    </td><td>    53312</td><td>OR4G4P     </td><td>0</td><td>+</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>45</th><td>chr1</td><td>56598    </td><td>    64116</td><td>OR4G11P    </td><td>0</td><td>+</td><td>.</td><td>gene_plus_promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & chr & start & end & gene\\_name & score & strand & dot2 & type\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 10869     &     14409 & DDX11L1     & 0 & + & . & gene\\_plus\\_promoter\\\\\n",
       "\t13 & chr1 &     14404 & 30570     & WASH7P      & 0 & - & . & gene\\_plus\\_promoter\\\\\n",
       "\t26 & chr1 & 28554     &     31109 & MIR1302-2HG & 0 & + & . & gene\\_plus\\_promoter\\\\\n",
       "\t34 & chr1 &     34554 & 37081     & FAM138A     & 0 & - & . & gene\\_plus\\_promoter\\\\\n",
       "\t42 & chr1 & 51473     &     53312 & OR4G4P      & 0 & + & . & gene\\_plus\\_promoter\\\\\n",
       "\t45 & chr1 & 56598     &     64116 & OR4G11P     & 0 & + & . & gene\\_plus\\_promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;chr&gt; | end &lt;chr&gt; | gene_name &lt;chr&gt; | score &lt;dbl&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 10869     |     14409 | DDX11L1     | 0 | + | . | gene_plus_promoter |\n",
       "| 13 | chr1 |     14404 | 30570     | WASH7P      | 0 | - | . | gene_plus_promoter |\n",
       "| 26 | chr1 | 28554     |     31109 | MIR1302-2HG | 0 | + | . | gene_plus_promoter |\n",
       "| 34 | chr1 |     34554 | 37081     | FAM138A     | 0 | - | . | gene_plus_promoter |\n",
       "| 42 | chr1 | 51473     |     53312 | OR4G4P      | 0 | + | . | gene_plus_promoter |\n",
       "| 45 | chr1 | 56598     |     64116 | OR4G11P     | 0 | + | . | gene_plus_promoter |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  start     end       gene_name   score strand dot2 type              \n",
       "1  chr1 10869         14409 DDX11L1     0     +      .    gene_plus_promoter\n",
       "13 chr1     14404 30570     WASH7P      0     -      .    gene_plus_promoter\n",
       "26 chr1 28554         31109 MIR1302-2HG 0     +      .    gene_plus_promoter\n",
       "34 chr1     34554 37081     FAM138A     0     -      .    gene_plus_promoter\n",
       "42 chr1 51473         53312 OR4G4P      0     +      .    gene_plus_promoter\n",
       "45 chr1 56598         64116 OR4G11P     0     +      .    gene_plus_promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# full genes (body+prom)\n",
    "df_full_genes_hg19 = df_gencode_hg19_genes\n",
    "df_full_genes_hg19$start = apply(df_gencode_hg19_genes, 1, expand_start)\n",
    "df_full_genes_hg19$end = apply(df_gencode_hg19_genes, 1, expand_end)\n",
    "\n",
    "df_full_genes_hg19 = df_full_genes_hg19 %>% mutate(type=\"gene_plus_promoter\", score=0, chr=paste0(\"chr\",chr, sep=\"\")) %>%\n",
    "    select(chr, start, end, gene_name, score, strand, dot2, type)\n",
    "head(df_full_genes_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "83cef072",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'chr'</li><li>'start'</li><li>'end'</li><li>'gene_name'</li><li>'strand'</li><li>'dot2'</li><li>'type'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'gene\\_name'\n",
       "\\item 'strand'\n",
       "\\item 'dot2'\n",
       "\\item 'type'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'chr'\n",
       "2. 'start'\n",
       "3. 'end'\n",
       "4. 'gene_name'\n",
       "5. 'strand'\n",
       "6. 'dot2'\n",
       "7. 'type'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"chr\"       \"start\"     \"end\"       \"gene_name\" \"strand\"    \"dot2\"     \n",
       "[7] \"type\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(df_full_genes_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "4203ab5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"A\"\n"
     ]
    }
   ],
   "source": [
    "cols = c('chr','start','end','gene_name','score','strand','type')\n",
    "df_full_genes_hg19$start <- as.numeric(df_full_genes_hg19$start)\n",
    "df_full_genes_hg19$end <- as.numeric(df_full_genes_hg19$end)\n",
    "df_full_genes_hg19$chr <- gsub(\"chr\",\"\",df_full_genes_hg19$chr)\n",
    "\n",
    "write.table(df_full_genes_hg19[,cols], file=\"../Translational_Science/gencode.v40lift37.full_genes_hg19.bed\",\n",
    "          quote=FALSE, row.names=FALSE,\n",
    "          sep=\"\\t\", col.names=FALSE\n",
    "         )\n",
    "print(\"A\")\n",
    "\n",
    "# check bed file format by upload here: https://genome.ucsc.edu/cgi-bin/hgCustom"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8000ec34",
   "metadata": {},
   "source": [
    "## annotate regions in Zhao et all list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "fb37317d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "105982"
      ],
      "text/latex": [
       "105982"
      ],
      "text/markdown": [
       "105982"
      ],
      "text/plain": [
       "[1] 105982"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start38</th><th scope=col>end38</th><th scope=col>stat1</th><th scope=col>stat2</th><th scope=col>methyl.status</th><th scope=col>region38</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 29194</td><td> 29436</td><td>-0.3396937</td><td>-289.58630</td><td>hypomethylated </td><td>chr1:29194-29436  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 51588</td><td> 51682</td><td>-0.1208693</td><td> -94.00083</td><td>hypomethylated </td><td>chr1:51588-51682  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>600448</td><td>600499</td><td>-0.1665657</td><td> -25.01982</td><td>hypomethylated </td><td>chr1:600448-600499</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>631079</td><td>631359</td><td> 0.2422400</td><td> 177.16574</td><td>hypermethylated</td><td>chr1:631079-631359</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>631433</td><td>631500</td><td> 0.2247558</td><td>  35.21353</td><td>hypermethylated</td><td>chr1:631433-631500</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>631733</td><td>631859</td><td> 0.1340444</td><td>  34.67026</td><td>hypermethylated</td><td>chr1:631733-631859</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start38 & end38 & stat1 & stat2 & methyl.status & region38\\\\\n",
       "  & <int> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  29194 &  29436 & -0.3396937 & -289.58630 & hypomethylated  & chr1:29194-29436  \\\\\n",
       "\t2 & 1 &  51588 &  51682 & -0.1208693 &  -94.00083 & hypomethylated  & chr1:51588-51682  \\\\\n",
       "\t3 & 1 & 600448 & 600499 & -0.1665657 &  -25.01982 & hypomethylated  & chr1:600448-600499\\\\\n",
       "\t4 & 1 & 631079 & 631359 &  0.2422400 &  177.16574 & hypermethylated & chr1:631079-631359\\\\\n",
       "\t5 & 1 & 631433 & 631500 &  0.2247558 &   35.21353 & hypermethylated & chr1:631433-631500\\\\\n",
       "\t6 & 1 & 631733 & 631859 &  0.1340444 &   34.67026 & hypermethylated & chr1:631733-631859\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;int&gt; | start38 &lt;int&gt; | end38 &lt;int&gt; | stat1 &lt;dbl&gt; | stat2 &lt;dbl&gt; | methyl.status &lt;chr&gt; | region38 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  29194 |  29436 | -0.3396937 | -289.58630 | hypomethylated  | chr1:29194-29436   |\n",
       "| 2 | 1 |  51588 |  51682 | -0.1208693 |  -94.00083 | hypomethylated  | chr1:51588-51682   |\n",
       "| 3 | 1 | 600448 | 600499 | -0.1665657 |  -25.01982 | hypomethylated  | chr1:600448-600499 |\n",
       "| 4 | 1 | 631079 | 631359 |  0.2422400 |  177.16574 | hypermethylated | chr1:631079-631359 |\n",
       "| 5 | 1 | 631433 | 631500 |  0.2247558 |   35.21353 | hypermethylated | chr1:631433-631500 |\n",
       "| 6 | 1 | 631733 | 631859 |  0.1340444 |   34.67026 | hypermethylated | chr1:631733-631859 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start38 end38  stat1      stat2      methyl.status   region38          \n",
       "1 1    29194   29436 -0.3396937 -289.58630 hypomethylated  chr1:29194-29436  \n",
       "2 1    51588   51682 -0.1208693  -94.00083 hypomethylated  chr1:51588-51682  \n",
       "3 1   600448  600499 -0.1665657  -25.01982 hypomethylated  chr1:600448-600499\n",
       "4 1   631079  631359  0.2422400  177.16574 hypermethylated chr1:631079-631359\n",
       "5 1   631433  631500  0.2247558   35.21353 hypermethylated chr1:631433-631500\n",
       "6 1   631733  631859  0.1340444   34.67026 hypermethylated chr1:631733-631859"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# hg38 coords\n",
    "df_zhao = read.csv(file=\"./marker_collation/Zhao_2020_DMRs_coords.csv\", header=FALSE)\n",
    "colnames(df_zhao) <- c(\"chr\", \"start38\", \"end38\", \"stat1\", \"stat2\", \"methyl.status\")\n",
    "df_zhao$region38 = paste0(\"chr\",df_zhao$chr,\":\",df_zhao$start,\"-\",df_zhao$end)\n",
    "nrow(df_zhao)\n",
    "head(df_zhao)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "cb73a80f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "103318"
      ],
      "text/latex": [
       "103318"
      ],
      "text/markdown": [
       "103318"
      ],
      "text/plain": [
       "[1] 103318"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start19</th><th scope=col>end19</th><th scope=col>region19</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td> 29194</td><td> 29436</td><td>chr1:29194-29436  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td> 51588</td><td> 51682</td><td>chr1:51588-51682  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>535828</td><td>535879</td><td>chr1:535828-535879</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>566459</td><td>566739</td><td>chr1:566459-566739</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>566813</td><td>566880</td><td>chr1:566813-566880</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>567113</td><td>567239</td><td>chr1:567113-567239</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & chr & start19 & end19 & region19\\\\\n",
       "  & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 &  29194 &  29436 & chr1:29194-29436  \\\\\n",
       "\t2 & chr1 &  51588 &  51682 & chr1:51588-51682  \\\\\n",
       "\t3 & chr1 & 535828 & 535879 & chr1:535828-535879\\\\\n",
       "\t4 & chr1 & 566459 & 566739 & chr1:566459-566739\\\\\n",
       "\t5 & chr1 & 566813 & 566880 & chr1:566813-566880\\\\\n",
       "\t6 & chr1 & 567113 & 567239 & chr1:567113-567239\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start19 &lt;int&gt; | end19 &lt;int&gt; | region19 &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | chr1 |  29194 |  29436 | chr1:29194-29436   |\n",
       "| 2 | chr1 |  51588 |  51682 | chr1:51588-51682   |\n",
       "| 3 | chr1 | 535828 | 535879 | chr1:535828-535879 |\n",
       "| 4 | chr1 | 566459 | 566739 | chr1:566459-566739 |\n",
       "| 5 | chr1 | 566813 | 566880 | chr1:566813-566880 |\n",
       "| 6 | chr1 | 567113 | 567239 | chr1:567113-567239 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  start19 end19  region19          \n",
       "1 chr1  29194   29436 chr1:29194-29436  \n",
       "2 chr1  51588   51682 chr1:51588-51682  \n",
       "3 chr1 535828  535879 chr1:535828-535879\n",
       "4 chr1 566459  566739 chr1:566459-566739\n",
       "5 chr1 566813  566880 chr1:566813-566880\n",
       "6 chr1 567113  567239 chr1:567113-567239"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2664"
      ],
      "text/latex": [
       "2664"
      ],
      "text/markdown": [
       "2664"
      ],
      "text/plain": [
       "[1] 2664"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "103318"
      ],
      "text/latex": [
       "103318"
      ],
      "text/markdown": [
       "103318"
      ],
      "text/plain": [
       "[1] 103318"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start38</th><th scope=col>end38</th><th scope=col>stat1</th><th scope=col>stat2</th><th scope=col>methyl.status</th><th scope=col>region38</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 29194</td><td> 29436</td><td>-0.3396937</td><td>-289.58630</td><td>hypomethylated </td><td>chr1:29194-29436  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 51588</td><td> 51682</td><td>-0.1208693</td><td> -94.00083</td><td>hypomethylated </td><td>chr1:51588-51682  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>600448</td><td>600499</td><td>-0.1665657</td><td> -25.01982</td><td>hypomethylated </td><td>chr1:600448-600499</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>631079</td><td>631359</td><td> 0.2422400</td><td> 177.16574</td><td>hypermethylated</td><td>chr1:631079-631359</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>631433</td><td>631500</td><td> 0.2247558</td><td>  35.21353</td><td>hypermethylated</td><td>chr1:631433-631500</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>631733</td><td>631859</td><td> 0.1340444</td><td>  34.67026</td><td>hypermethylated</td><td>chr1:631733-631859</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start38 & end38 & stat1 & stat2 & methyl.status & region38\\\\\n",
       "  & <int> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  29194 &  29436 & -0.3396937 & -289.58630 & hypomethylated  & chr1:29194-29436  \\\\\n",
       "\t2 & 1 &  51588 &  51682 & -0.1208693 &  -94.00083 & hypomethylated  & chr1:51588-51682  \\\\\n",
       "\t3 & 1 & 600448 & 600499 & -0.1665657 &  -25.01982 & hypomethylated  & chr1:600448-600499\\\\\n",
       "\t4 & 1 & 631079 & 631359 &  0.2422400 &  177.16574 & hypermethylated & chr1:631079-631359\\\\\n",
       "\t5 & 1 & 631433 & 631500 &  0.2247558 &   35.21353 & hypermethylated & chr1:631433-631500\\\\\n",
       "\t6 & 1 & 631733 & 631859 &  0.1340444 &   34.67026 & hypermethylated & chr1:631733-631859\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;int&gt; | start38 &lt;int&gt; | end38 &lt;int&gt; | stat1 &lt;dbl&gt; | stat2 &lt;dbl&gt; | methyl.status &lt;chr&gt; | region38 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  29194 |  29436 | -0.3396937 | -289.58630 | hypomethylated  | chr1:29194-29436   |\n",
       "| 2 | 1 |  51588 |  51682 | -0.1208693 |  -94.00083 | hypomethylated  | chr1:51588-51682   |\n",
       "| 3 | 1 | 600448 | 600499 | -0.1665657 |  -25.01982 | hypomethylated  | chr1:600448-600499 |\n",
       "| 4 | 1 | 631079 | 631359 |  0.2422400 |  177.16574 | hypermethylated | chr1:631079-631359 |\n",
       "| 5 | 1 | 631433 | 631500 |  0.2247558 |   35.21353 | hypermethylated | chr1:631433-631500 |\n",
       "| 6 | 1 | 631733 | 631859 |  0.1340444 |   34.67026 | hypermethylated | chr1:631733-631859 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start38 end38  stat1      stat2      methyl.status   region38          \n",
       "1 1    29194   29436 -0.3396937 -289.58630 hypomethylated  chr1:29194-29436  \n",
       "2 1    51588   51682 -0.1208693  -94.00083 hypomethylated  chr1:51588-51682  \n",
       "3 1   600448  600499 -0.1665657  -25.01982 hypomethylated  chr1:600448-600499\n",
       "4 1   631079  631359  0.2422400  177.16574 hypermethylated chr1:631079-631359\n",
       "5 1   631433  631500  0.2247558   35.21353 hypermethylated chr1:631433-631500\n",
       "6 1   631733  631859  0.1340444   34.67026 hypermethylated chr1:631733-631859"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge web lift over to hg19 ** make sure to select output only 1 match per region **\n",
    "# output hg19 regions\n",
    "df_hg19_LO = read.table(file=\"./marker_collation/Zhao_hglft_toHG19_genome_bee7_df2600.bed\", header=FALSE, sep=\"\\t\")\n",
    "colnames(df_hg19_LO) <- c(\"chr\",\"start19\",\"end19\")\n",
    "df_hg19_LO$region19 = paste0(df_hg19_LO$chr,\":\",df_hg19_LO$start19,\"-\",df_hg19_LO$end19)\n",
    "nrow(df_hg19_LO)\n",
    "head(df_hg19_LO)\n",
    "\n",
    "# original hg38 regionsn that dont lift over\n",
    "df_hg19_LO_err = read.table(file=\"./marker_collation/Zhao_hglft_toHG19_genome_bee7_df2600.err.txt\", header=FALSE, sep=\"\\t\")\n",
    "colnames(df_hg19_LO_err) <- c(\"chr38\",\"start38\",\"end38\")\n",
    "df_hg19_LO_err$region38 = paste0(df_hg19_LO_err$chr38,\":\",df_hg19_LO_err$start38,\"-\",df_hg19_LO_err$end38)\n",
    "nrow(df_hg19_LO_err)\n",
    "stopifnot(nrow(df_hg19_LO)+nrow(df_hg19_LO_err)==nrow(df_zhao)) #106248\n",
    "\n",
    "# drop regions without lift over coords\n",
    "df_zhao_hg19 = df_zhao\n",
    "for (reg in df_hg19_LO_err$region38){\n",
    "    df_zhao_hg19 = df_zhao_hg19[(df_zhao_hg19$region38 != reg),]\n",
    "}\n",
    "\n",
    "nrow(df_zhao_hg19)\n",
    "head(df_zhao_hg19)\n",
    "\n",
    "if (nrow(df_zhao_hg19)!=nrow(df_hg19_LO)){\n",
    "    print(nrow(df_zhao_hg19))\n",
    "    print(nrow(df_hg19_LO))\n",
    "    stop()\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "a82956d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start38</th><th scope=col>end38</th><th scope=col>stat1</th><th scope=col>stat2</th><th scope=col>methyl.status</th><th scope=col>region38</th><th scope=col>region19</th><th scope=col>chr19</th><th scope=col>start19</th><th scope=col>end19</th><th scope=col>source</th><th scope=col>start</th><th scope=col>end</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 29194</td><td> 29436</td><td>-0.3396937</td><td>-289.58630</td><td>hypomethylated </td><td>chr1:29194-29436  </td><td>chr1:29194-29436  </td><td>chr1</td><td>29194 </td><td>29436 </td><td>Zhao et al 2020</td><td> 29194</td><td> 29436</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 51588</td><td> 51682</td><td>-0.1208693</td><td> -94.00083</td><td>hypomethylated </td><td>chr1:51588-51682  </td><td>chr1:51588-51682  </td><td>chr1</td><td>51588 </td><td>51682 </td><td>Zhao et al 2020</td><td> 51588</td><td> 51682</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>600448</td><td>600499</td><td>-0.1665657</td><td> -25.01982</td><td>hypomethylated </td><td>chr1:600448-600499</td><td>chr1:535828-535879</td><td>chr1</td><td>535828</td><td>535879</td><td>Zhao et al 2020</td><td>535828</td><td>535879</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>631079</td><td>631359</td><td> 0.2422400</td><td> 177.16574</td><td>hypermethylated</td><td>chr1:631079-631359</td><td>chr1:566459-566739</td><td>chr1</td><td>566459</td><td>566739</td><td>Zhao et al 2020</td><td>566459</td><td>566739</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>631433</td><td>631500</td><td> 0.2247558</td><td>  35.21353</td><td>hypermethylated</td><td>chr1:631433-631500</td><td>chr1:566813-566880</td><td>chr1</td><td>566813</td><td>566880</td><td>Zhao et al 2020</td><td>566813</td><td>566880</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>631733</td><td>631859</td><td> 0.1340444</td><td>  34.67026</td><td>hypermethylated</td><td>chr1:631733-631859</td><td>chr1:567113-567239</td><td>chr1</td><td>567113</td><td>567239</td><td>Zhao et al 2020</td><td>567113</td><td>567239</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & chr & start38 & end38 & stat1 & stat2 & methyl.status & region38 & region19 & chr19 & start19 & end19 & source & start & end\\\\\n",
       "  & <chr> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  29194 &  29436 & -0.3396937 & -289.58630 & hypomethylated  & chr1:29194-29436   & chr1:29194-29436   & chr1 & 29194  & 29436  & Zhao et al 2020 &  29194 &  29436\\\\\n",
       "\t2 & 1 &  51588 &  51682 & -0.1208693 &  -94.00083 & hypomethylated  & chr1:51588-51682   & chr1:51588-51682   & chr1 & 51588  & 51682  & Zhao et al 2020 &  51588 &  51682\\\\\n",
       "\t3 & 1 & 600448 & 600499 & -0.1665657 &  -25.01982 & hypomethylated  & chr1:600448-600499 & chr1:535828-535879 & chr1 & 535828 & 535879 & Zhao et al 2020 & 535828 & 535879\\\\\n",
       "\t4 & 1 & 631079 & 631359 &  0.2422400 &  177.16574 & hypermethylated & chr1:631079-631359 & chr1:566459-566739 & chr1 & 566459 & 566739 & Zhao et al 2020 & 566459 & 566739\\\\\n",
       "\t5 & 1 & 631433 & 631500 &  0.2247558 &   35.21353 & hypermethylated & chr1:631433-631500 & chr1:566813-566880 & chr1 & 566813 & 566880 & Zhao et al 2020 & 566813 & 566880\\\\\n",
       "\t6 & 1 & 631733 & 631859 &  0.1340444 &   34.67026 & hypermethylated & chr1:631733-631859 & chr1:567113-567239 & chr1 & 567113 & 567239 & Zhao et al 2020 & 567113 & 567239\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start38 &lt;int&gt; | end38 &lt;int&gt; | stat1 &lt;dbl&gt; | stat2 &lt;dbl&gt; | methyl.status &lt;chr&gt; | region38 &lt;chr&gt; | region19 &lt;chr&gt; | chr19 &lt;chr&gt; | start19 &lt;chr&gt; | end19 &lt;chr&gt; | source &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  29194 |  29436 | -0.3396937 | -289.58630 | hypomethylated  | chr1:29194-29436   | chr1:29194-29436   | chr1 | 29194  | 29436  | Zhao et al 2020 |  29194 |  29436 |\n",
       "| 2 | 1 |  51588 |  51682 | -0.1208693 |  -94.00083 | hypomethylated  | chr1:51588-51682   | chr1:51588-51682   | chr1 | 51588  | 51682  | Zhao et al 2020 |  51588 |  51682 |\n",
       "| 3 | 1 | 600448 | 600499 | -0.1665657 |  -25.01982 | hypomethylated  | chr1:600448-600499 | chr1:535828-535879 | chr1 | 535828 | 535879 | Zhao et al 2020 | 535828 | 535879 |\n",
       "| 4 | 1 | 631079 | 631359 |  0.2422400 |  177.16574 | hypermethylated | chr1:631079-631359 | chr1:566459-566739 | chr1 | 566459 | 566739 | Zhao et al 2020 | 566459 | 566739 |\n",
       "| 5 | 1 | 631433 | 631500 |  0.2247558 |   35.21353 | hypermethylated | chr1:631433-631500 | chr1:566813-566880 | chr1 | 566813 | 566880 | Zhao et al 2020 | 566813 | 566880 |\n",
       "| 6 | 1 | 631733 | 631859 |  0.1340444 |   34.67026 | hypermethylated | chr1:631733-631859 | chr1:567113-567239 | chr1 | 567113 | 567239 | Zhao et al 2020 | 567113 | 567239 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start38 end38  stat1      stat2      methyl.status   region38          \n",
       "1 1    29194   29436 -0.3396937 -289.58630 hypomethylated  chr1:29194-29436  \n",
       "2 1    51588   51682 -0.1208693  -94.00083 hypomethylated  chr1:51588-51682  \n",
       "3 1   600448  600499 -0.1665657  -25.01982 hypomethylated  chr1:600448-600499\n",
       "4 1   631079  631359  0.2422400  177.16574 hypermethylated chr1:631079-631359\n",
       "5 1   631433  631500  0.2247558   35.21353 hypermethylated chr1:631433-631500\n",
       "6 1   631733  631859  0.1340444   34.67026 hypermethylated chr1:631733-631859\n",
       "  region19           chr19 start19 end19  source          start  end   \n",
       "1 chr1:29194-29436   chr1  29194   29436  Zhao et al 2020  29194  29436\n",
       "2 chr1:51588-51682   chr1  51588   51682  Zhao et al 2020  51588  51682\n",
       "3 chr1:535828-535879 chr1  535828  535879 Zhao et al 2020 535828 535879\n",
       "4 chr1:566459-566739 chr1  566459  566739 Zhao et al 2020 566459 566739\n",
       "5 chr1:566813-566880 chr1  566813  566880 Zhao et al 2020 566813 566880\n",
       "6 chr1:567113-567239 chr1  567113  567239 Zhao et al 2020 567113 567239"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start38</th><th scope=col>end38</th><th scope=col>stat1</th><th scope=col>stat2</th><th scope=col>methyl.status</th><th scope=col>region38</th><th scope=col>region19</th><th scope=col>chr19</th><th scope=col>start19</th><th scope=col>end19</th><th scope=col>source</th><th scope=col>start</th><th scope=col>end</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 29194</td><td> 29436</td><td>-0.3396937</td><td>-289.58630</td><td>hypomethylated </td><td>chr1:29194-29436  </td><td>chr1:29194-29436  </td><td>chr1</td><td>29194 </td><td>29436 </td><td>Zhao et al 2020</td><td> 29194</td><td> 29436</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 51588</td><td> 51682</td><td>-0.1208693</td><td> -94.00083</td><td>hypomethylated </td><td>chr1:51588-51682  </td><td>chr1:51588-51682  </td><td>chr1</td><td>51588 </td><td>51682 </td><td>Zhao et al 2020</td><td> 51588</td><td> 51682</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>600448</td><td>600499</td><td>-0.1665657</td><td> -25.01982</td><td>hypomethylated </td><td>chr1:600448-600499</td><td>chr1:535828-535879</td><td>chr1</td><td>535828</td><td>535879</td><td>Zhao et al 2020</td><td>535828</td><td>535879</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>631079</td><td>631359</td><td> 0.2422400</td><td> 177.16574</td><td>hypermethylated</td><td>chr1:631079-631359</td><td>chr1:566459-566739</td><td>chr1</td><td>566459</td><td>566739</td><td>Zhao et al 2020</td><td>566459</td><td>566739</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>631433</td><td>631500</td><td> 0.2247558</td><td>  35.21353</td><td>hypermethylated</td><td>chr1:631433-631500</td><td>chr1:566813-566880</td><td>chr1</td><td>566813</td><td>566880</td><td>Zhao et al 2020</td><td>566813</td><td>566880</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>631733</td><td>631859</td><td> 0.1340444</td><td>  34.67026</td><td>hypermethylated</td><td>chr1:631733-631859</td><td>chr1:567113-567239</td><td>chr1</td><td>567113</td><td>567239</td><td>Zhao et al 2020</td><td>567113</td><td>567239</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & chr & start38 & end38 & stat1 & stat2 & methyl.status & region38 & region19 & chr19 & start19 & end19 & source & start & end\\\\\n",
       "  & <chr> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  29194 &  29436 & -0.3396937 & -289.58630 & hypomethylated  & chr1:29194-29436   & chr1:29194-29436   & chr1 & 29194  & 29436  & Zhao et al 2020 &  29194 &  29436\\\\\n",
       "\t2 & 1 &  51588 &  51682 & -0.1208693 &  -94.00083 & hypomethylated  & chr1:51588-51682   & chr1:51588-51682   & chr1 & 51588  & 51682  & Zhao et al 2020 &  51588 &  51682\\\\\n",
       "\t3 & 1 & 600448 & 600499 & -0.1665657 &  -25.01982 & hypomethylated  & chr1:600448-600499 & chr1:535828-535879 & chr1 & 535828 & 535879 & Zhao et al 2020 & 535828 & 535879\\\\\n",
       "\t4 & 1 & 631079 & 631359 &  0.2422400 &  177.16574 & hypermethylated & chr1:631079-631359 & chr1:566459-566739 & chr1 & 566459 & 566739 & Zhao et al 2020 & 566459 & 566739\\\\\n",
       "\t5 & 1 & 631433 & 631500 &  0.2247558 &   35.21353 & hypermethylated & chr1:631433-631500 & chr1:566813-566880 & chr1 & 566813 & 566880 & Zhao et al 2020 & 566813 & 566880\\\\\n",
       "\t6 & 1 & 631733 & 631859 &  0.1340444 &   34.67026 & hypermethylated & chr1:631733-631859 & chr1:567113-567239 & chr1 & 567113 & 567239 & Zhao et al 2020 & 567113 & 567239\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start38 &lt;int&gt; | end38 &lt;int&gt; | stat1 &lt;dbl&gt; | stat2 &lt;dbl&gt; | methyl.status &lt;chr&gt; | region38 &lt;chr&gt; | region19 &lt;chr&gt; | chr19 &lt;chr&gt; | start19 &lt;chr&gt; | end19 &lt;chr&gt; | source &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  29194 |  29436 | -0.3396937 | -289.58630 | hypomethylated  | chr1:29194-29436   | chr1:29194-29436   | chr1 | 29194  | 29436  | Zhao et al 2020 |  29194 |  29436 |\n",
       "| 2 | 1 |  51588 |  51682 | -0.1208693 |  -94.00083 | hypomethylated  | chr1:51588-51682   | chr1:51588-51682   | chr1 | 51588  | 51682  | Zhao et al 2020 |  51588 |  51682 |\n",
       "| 3 | 1 | 600448 | 600499 | -0.1665657 |  -25.01982 | hypomethylated  | chr1:600448-600499 | chr1:535828-535879 | chr1 | 535828 | 535879 | Zhao et al 2020 | 535828 | 535879 |\n",
       "| 4 | 1 | 631079 | 631359 |  0.2422400 |  177.16574 | hypermethylated | chr1:631079-631359 | chr1:566459-566739 | chr1 | 566459 | 566739 | Zhao et al 2020 | 566459 | 566739 |\n",
       "| 5 | 1 | 631433 | 631500 |  0.2247558 |   35.21353 | hypermethylated | chr1:631433-631500 | chr1:566813-566880 | chr1 | 566813 | 566880 | Zhao et al 2020 | 566813 | 566880 |\n",
       "| 6 | 1 | 631733 | 631859 |  0.1340444 |   34.67026 | hypermethylated | chr1:631733-631859 | chr1:567113-567239 | chr1 | 567113 | 567239 | Zhao et al 2020 | 567113 | 567239 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start38 end38  stat1      stat2      methyl.status   region38          \n",
       "1 1    29194   29436 -0.3396937 -289.58630 hypomethylated  chr1:29194-29436  \n",
       "2 1    51588   51682 -0.1208693  -94.00083 hypomethylated  chr1:51588-51682  \n",
       "3 1   600448  600499 -0.1665657  -25.01982 hypomethylated  chr1:600448-600499\n",
       "4 1   631079  631359  0.2422400  177.16574 hypermethylated chr1:631079-631359\n",
       "5 1   631433  631500  0.2247558   35.21353 hypermethylated chr1:631433-631500\n",
       "6 1   631733  631859  0.1340444   34.67026 hypermethylated chr1:631733-631859\n",
       "  region19           chr19 start19 end19  source          start  end   \n",
       "1 chr1:29194-29436   chr1  29194   29436  Zhao et al 2020  29194  29436\n",
       "2 chr1:51588-51682   chr1  51588   51682  Zhao et al 2020  51588  51682\n",
       "3 chr1:535828-535879 chr1  535828  535879 Zhao et al 2020 535828 535879\n",
       "4 chr1:566459-566739 chr1  566459  566739 Zhao et al 2020 566459 566739\n",
       "5 chr1:566813-566880 chr1  566813  566880 Zhao et al 2020 566813 566880\n",
       "6 chr1:567113-567239 chr1  567113  567239 Zhao et al 2020 567113 567239"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_zhao_hg19$region19 <- df_hg19_LO$region19\n",
    "df_zhao_hg19$chr19 <- sapply(strsplit(df_zhao_hg19$region19, \":|-\"),\"[\", 1)\n",
    "df_zhao_hg19$start19 <- sapply(strsplit(df_zhao_hg19$region19, \":|-\"),\"[\", 2)\n",
    "df_zhao_hg19$end19 <- sapply(strsplit(df_zhao_hg19$region19, \":|-\"),\"[\", 3)\n",
    "df_zhao_hg19$source <- \"Zhao et al 2020\"\n",
    "head(df_zhao_hg19)\n",
    "\n",
    "df_zhao_hg19$chr <- sapply(strsplit(df_zhao_hg19$chr19, \"chr\"), \"[\", 2)\n",
    "df_zhao_hg19$start <- as.integer(df_zhao_hg19$start19)\n",
    "df_zhao_hg19$end <- as.integer(df_zhao_hg19$end19)\n",
    "\n",
    "head(df_zhao_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "c50d3120",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_zhao_hg19[,c(\"chr\", \"start\", \"end\", \"source\",\"methyl.status\")], \n",
    "            sep=\"\\t\",row.names=FALSE, col.names=FALSE, quote=FALSE, file=\"./marker_collation/Zhao_2020_DMRs_hg19.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7cb425e",
   "metadata": {},
   "source": [
    "## annotate Beltran regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "78707547",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>X</th><th scope=col>end</th><th scope=col>rockermeth.q_value</th><th scope=col>beta_difference</th><th scope=col>meth.state</th><th scope=col>DMR_set</th><th scope=col>region</th><th scope=col>hg38.coord</th><th scope=col>X.1</th><th scope=col>X.2</th><th scope=col>X.3</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>833806</td><td>834358</td><td>0.0038710350</td><td> 30.95833</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:833806-834358</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>841565</td><td>843221</td><td>0.0002034520</td><td> 48.13401</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:841565-843221</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>861132</td><td>861612</td><td>0.0000000012</td><td> 27.45051</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:861132-861612</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>863044</td><td>863496</td><td>0.0006457100</td><td>-46.40476</td><td>Hypo-methylated </td><td>PCa|NT DMRs</td><td>chr1:863044-863496</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>868475</td><td>869290</td><td>0.0005530510</td><td>-36.01151</td><td>Hypo-methylated </td><td>PCa|NT DMRs</td><td>chr1:868475-869290</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>906156</td><td>907273</td><td>0.0000205000</td><td> 24.38440</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:906156-907273</td><td>NA</td><td>NA</td><td>NA</td><td></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & chr & X & end & rockermeth.q\\_value & beta\\_difference & meth.state & DMR\\_set & region & hg38.coord & X.1 & X.2 & X.3\\\\\n",
       "  & <chr> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 833806 & 834358 & 0.0038710350 &  30.95833 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:833806-834358 & NA & NA & NA & \\\\\n",
       "\t2 & chr1 & 841565 & 843221 & 0.0002034520 &  48.13401 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:841565-843221 & NA & NA & NA & \\\\\n",
       "\t3 & chr1 & 861132 & 861612 & 0.0000000012 &  27.45051 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:861132-861612 & NA & NA & NA & \\\\\n",
       "\t4 & chr1 & 863044 & 863496 & 0.0006457100 & -46.40476 & Hypo-methylated  & PCa\\textbar{}NT DMRs & chr1:863044-863496 & NA & NA & NA & \\\\\n",
       "\t5 & chr1 & 868475 & 869290 & 0.0005530510 & -36.01151 & Hypo-methylated  & PCa\\textbar{}NT DMRs & chr1:868475-869290 & NA & NA & NA & \\\\\n",
       "\t6 & chr1 & 906156 & 907273 & 0.0000205000 &  24.38440 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:906156-907273 & NA & NA & NA & \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | X &lt;int&gt; | end &lt;int&gt; | rockermeth.q_value &lt;dbl&gt; | beta_difference &lt;dbl&gt; | meth.state &lt;chr&gt; | DMR_set &lt;chr&gt; | region &lt;chr&gt; | hg38.coord &lt;lgl&gt; | X.1 &lt;lgl&gt; | X.2 &lt;lgl&gt; | X.3 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 833806 | 834358 | 0.0038710350 |  30.95833 | Hyper-methylated | PCa|NT DMRs | chr1:833806-834358 | NA | NA | NA | <!----> |\n",
       "| 2 | chr1 | 841565 | 843221 | 0.0002034520 |  48.13401 | Hyper-methylated | PCa|NT DMRs | chr1:841565-843221 | NA | NA | NA | <!----> |\n",
       "| 3 | chr1 | 861132 | 861612 | 0.0000000012 |  27.45051 | Hyper-methylated | PCa|NT DMRs | chr1:861132-861612 | NA | NA | NA | <!----> |\n",
       "| 4 | chr1 | 863044 | 863496 | 0.0006457100 | -46.40476 | Hypo-methylated  | PCa|NT DMRs | chr1:863044-863496 | NA | NA | NA | <!----> |\n",
       "| 5 | chr1 | 868475 | 869290 | 0.0005530510 | -36.01151 | Hypo-methylated  | PCa|NT DMRs | chr1:868475-869290 | NA | NA | NA | <!----> |\n",
       "| 6 | chr1 | 906156 | 907273 | 0.0000205000 |  24.38440 | Hyper-methylated | PCa|NT DMRs | chr1:906156-907273 | NA | NA | NA | <!----> |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  X      end    rockermeth.q_value beta_difference meth.state      \n",
       "1 chr1 833806 834358 0.0038710350        30.95833       Hyper-methylated\n",
       "2 chr1 841565 843221 0.0002034520        48.13401       Hyper-methylated\n",
       "3 chr1 861132 861612 0.0000000012        27.45051       Hyper-methylated\n",
       "4 chr1 863044 863496 0.0006457100       -46.40476       Hypo-methylated \n",
       "5 chr1 868475 869290 0.0005530510       -36.01151       Hypo-methylated \n",
       "6 chr1 906156 907273 0.0000205000        24.38440       Hyper-methylated\n",
       "  DMR_set     region             hg38.coord X.1 X.2 X.3\n",
       "1 PCa|NT DMRs chr1:833806-834358 NA         NA  NA     \n",
       "2 PCa|NT DMRs chr1:841565-843221 NA         NA  NA     \n",
       "3 PCa|NT DMRs chr1:861132-861612 NA         NA  NA     \n",
       "4 PCa|NT DMRs chr1:863044-863496 NA         NA  NA     \n",
       "5 PCa|NT DMRs chr1:868475-869290 NA         NA  NA     \n",
       "6 PCa|NT DMRs chr1:906156-907273 NA         NA  NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#hg19/gr37\n",
    "df_beltran_orig = read.csv(file=\"./marker_collation/Beltran_2020_DMRs.csv\", header=TRUE)\n",
    "df_beltran_orig = df_beltran_orig %>% rename(\"region\"=\"region..hg19.\")\n",
    "head(df_beltran_orig)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "910750eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge web lift over to hg38\n",
    "# df_hg38_LO = read.table(file=\"./marker_collation//Beltran_hg38_LO_result.txt\", header=FALSE, sep=\"\\t\")\n",
    "# df_hg38_LO$region = paste0(df_hg38_LO$V1,\":\",df_hg38_LO$V2,\"-\",df_hg38_LO$V3)\n",
    "# head(df_hg38_LO)\n",
    "# df_hg38_LO_err = read.table(file=\"./marker_collation//Beltran_hg38_LO_errors.txt\", header=FALSE, sep=\"\\t\")\n",
    "# df_hg38_LO_err$region = paste0(df_hg38_LO_err$V1,\":\",df_hg38_LO_err$V2,\"-\",df_hg38_LO_err$V3)\n",
    "# head(df_hg38_LO_err)\n",
    "\n",
    "# drop regions without lift over coords\n",
    "# for (reg in df_hg38_LO_err$region){\n",
    "#     df_beltran_orig = df_beltran_orig[(df_beltran_orig$region != reg),]\n",
    "# }\n",
    "# dim(df_beltran_orig)\n",
    "\n",
    "# df_beltran_orig$region_hg38 <- df_hg38_LO$region\n",
    "# df_beltran_orig$chr_hg38 <- sapply(strsplit(df_beltran_orig$region_hg38, \":|-\"),\"[\", 1)\n",
    "# df_beltran_orig$start_hg38 <- sapply(strsplit(df_beltran_orig$region_hg38, \":|-\"),\"[\", 2)\n",
    "# df_beltran_orig$end_hg38 <- sapply(strsplit(df_beltran_orig$region_hg38, \":|-\"),\"[\", 3)\n",
    "# df_beltran_orig$source <- \"Beltran et al 2020\"\n",
    "# head(df_beltran_orig)\n",
    "\n",
    "# df_beltran_orig$chr_hg38 <- sapply(strsplit(df_beltran_orig$chr_hg38, \"chr\"), \"[\", 2)\n",
    "# df_beltran_orig$start_hg38 <- as.integer(df_beltran_orig$start_hg38)\n",
    "# df_beltran_orig$end_hg38 <- as.integer(df_beltran_orig$end_hg38)\n",
    "\n",
    "# head(df_beltran_orig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "a40903f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>X</th><th scope=col>end</th><th scope=col>rockermeth.q_value</th><th scope=col>beta_difference</th><th scope=col>meth.state</th><th scope=col>DMR_set</th><th scope=col>region</th><th scope=col>hg38.coord</th><th scope=col>X.1</th><th scope=col>X.2</th><th scope=col>X.3</th><th scope=col>name</th><th scope=col>score</th><th scope=col>strand</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>833806</td><td>834358</td><td>0.0038710350</td><td> 30.95833</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:833806-834358</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>841565</td><td>843221</td><td>0.0002034520</td><td> 48.13401</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:841565-843221</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>861132</td><td>861612</td><td>0.0000000012</td><td> 27.45051</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:861132-861612</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>863044</td><td>863496</td><td>0.0006457100</td><td>-46.40476</td><td>Hypo-methylated </td><td>PCa|NT DMRs</td><td>chr1:863044-863496</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>868475</td><td>869290</td><td>0.0005530510</td><td>-36.01151</td><td>Hypo-methylated </td><td>PCa|NT DMRs</td><td>chr1:868475-869290</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>906156</td><td>907273</td><td>0.0000205000</td><td> 24.38440</td><td>Hyper-methylated</td><td>PCa|NT DMRs</td><td>chr1:906156-907273</td><td>NA</td><td>NA</td><td>NA</td><td></td><td>.</td><td>.</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & chr & X & end & rockermeth.q\\_value & beta\\_difference & meth.state & DMR\\_set & region & hg38.coord & X.1 & X.2 & X.3 & name & score & strand\\\\\n",
       "  & <chr> & <int> & <int> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 833806 & 834358 & 0.0038710350 &  30.95833 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:833806-834358 & NA & NA & NA &  & . & . & .\\\\\n",
       "\t2 & chr1 & 841565 & 843221 & 0.0002034520 &  48.13401 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:841565-843221 & NA & NA & NA &  & . & . & .\\\\\n",
       "\t3 & chr1 & 861132 & 861612 & 0.0000000012 &  27.45051 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:861132-861612 & NA & NA & NA &  & . & . & .\\\\\n",
       "\t4 & chr1 & 863044 & 863496 & 0.0006457100 & -46.40476 & Hypo-methylated  & PCa\\textbar{}NT DMRs & chr1:863044-863496 & NA & NA & NA &  & . & . & .\\\\\n",
       "\t5 & chr1 & 868475 & 869290 & 0.0005530510 & -36.01151 & Hypo-methylated  & PCa\\textbar{}NT DMRs & chr1:868475-869290 & NA & NA & NA &  & . & . & .\\\\\n",
       "\t6 & chr1 & 906156 & 907273 & 0.0000205000 &  24.38440 & Hyper-methylated & PCa\\textbar{}NT DMRs & chr1:906156-907273 & NA & NA & NA &  & . & . & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | X &lt;int&gt; | end &lt;int&gt; | rockermeth.q_value &lt;dbl&gt; | beta_difference &lt;dbl&gt; | meth.state &lt;chr&gt; | DMR_set &lt;chr&gt; | region &lt;chr&gt; | hg38.coord &lt;lgl&gt; | X.1 &lt;lgl&gt; | X.2 &lt;lgl&gt; | X.3 &lt;chr&gt; | name &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 833806 | 834358 | 0.0038710350 |  30.95833 | Hyper-methylated | PCa|NT DMRs | chr1:833806-834358 | NA | NA | NA | <!----> | . | . | . |\n",
       "| 2 | chr1 | 841565 | 843221 | 0.0002034520 |  48.13401 | Hyper-methylated | PCa|NT DMRs | chr1:841565-843221 | NA | NA | NA | <!----> | . | . | . |\n",
       "| 3 | chr1 | 861132 | 861612 | 0.0000000012 |  27.45051 | Hyper-methylated | PCa|NT DMRs | chr1:861132-861612 | NA | NA | NA | <!----> | . | . | . |\n",
       "| 4 | chr1 | 863044 | 863496 | 0.0006457100 | -46.40476 | Hypo-methylated  | PCa|NT DMRs | chr1:863044-863496 | NA | NA | NA | <!----> | . | . | . |\n",
       "| 5 | chr1 | 868475 | 869290 | 0.0005530510 | -36.01151 | Hypo-methylated  | PCa|NT DMRs | chr1:868475-869290 | NA | NA | NA | <!----> | . | . | . |\n",
       "| 6 | chr1 | 906156 | 907273 | 0.0000205000 |  24.38440 | Hyper-methylated | PCa|NT DMRs | chr1:906156-907273 | NA | NA | NA | <!----> | . | . | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  X      end    rockermeth.q_value beta_difference meth.state      \n",
       "1 chr1 833806 834358 0.0038710350        30.95833       Hyper-methylated\n",
       "2 chr1 841565 843221 0.0002034520        48.13401       Hyper-methylated\n",
       "3 chr1 861132 861612 0.0000000012        27.45051       Hyper-methylated\n",
       "4 chr1 863044 863496 0.0006457100       -46.40476       Hypo-methylated \n",
       "5 chr1 868475 869290 0.0005530510       -36.01151       Hypo-methylated \n",
       "6 chr1 906156 907273 0.0000205000        24.38440       Hyper-methylated\n",
       "  DMR_set     region             hg38.coord X.1 X.2 X.3 name score strand\n",
       "1 PCa|NT DMRs chr1:833806-834358 NA         NA  NA      .    .     .     \n",
       "2 PCa|NT DMRs chr1:841565-843221 NA         NA  NA      .    .     .     \n",
       "3 PCa|NT DMRs chr1:861132-861612 NA         NA  NA      .    .     .     \n",
       "4 PCa|NT DMRs chr1:863044-863496 NA         NA  NA      .    .     .     \n",
       "5 PCa|NT DMRs chr1:868475-869290 NA         NA  NA      .    .     .     \n",
       "6 PCa|NT DMRs chr1:906156-907273 NA         NA  NA      .    .     .     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_beltran_orig$name <- \".\"\n",
    "df_beltran_orig$score <- \".\"\n",
    "df_beltran_orig$strand <- \".\"\n",
    "head(df_beltran_orig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "aa5e89f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_beltran_orig$source <- \"Beltran et al 2020\"\n",
    "df_beltran_orig$start = df_beltran_orig$X\n",
    "df_beltran_orig <- df_beltran_orig[order(df_beltran_orig$chr, df_beltran_orig$start, df_beltran_orig$end),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "568c3190",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_beltran_orig[,c(\"chr\", \"start\", \"end\", \"name\",\"score\",\"strand\",\"source\",\"meth.state\")], \n",
    "            sep=\"\\t\",row.names=FALSE, col.names=FALSE, quote=FALSE, file=\"./marker_collation/Beltran_2020_DMRs_hg19.bed\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4313d8a1",
   "metadata": {},
   "source": [
    "## retrieve coords for genes summary list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "07811906",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td></td><td>CD24  </td><td></td><td>NA</td><td>Lu et al 2022       </td><td>hypermethylated                    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td></td><td>TET2  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>DNMT3B</td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>IDH1  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>BRAF  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>AR    </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated intergenic regions </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 &  & CD24   &  & NA & Lu et al 2022        & hypermethylated                    \\\\\n",
       "\t2 &  & TET2   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype            \\\\\n",
       "\t3 &  & DNMT3B &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype            \\\\\n",
       "\t4 &  & IDH1   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype            \\\\\n",
       "\t5 &  & BRAF   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype            \\\\\n",
       "\t6 &  & AR     &  &  1 & Kukkonen et al. 2021 & hypermethylated intergenic regions \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | <!----> | CD24   | <!----> | NA | Lu et al 2022        | hypermethylated                     |\n",
       "| 2 | <!----> | TET2   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             |\n",
       "| 3 | <!----> | DNMT3B | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             |\n",
       "| 4 | <!----> | IDH1   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             |\n",
       "| 5 | <!----> | BRAF   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             |\n",
       "| 6 | <!----> | AR     | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated intergenic regions  |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe No..pubs Source              \n",
       "1        CD24         NA       Lu et al 2022       \n",
       "2        TET2          1       Kukkonen et al. 2021\n",
       "3        DNMT3B        1       Kukkonen et al. 2021\n",
       "4        IDH1          1       Kukkonen et al. 2021\n",
       "5        BRAF          1       Kukkonen et al. 2021\n",
       "6        AR            1       Kukkonen et al. 2021\n",
       "  note                               \n",
       "1 hypermethylated                    \n",
       "2 hypermethylated subtype            \n",
       "3 hypermethylated subtype            \n",
       "4 hypermethylated subtype            \n",
       "5 hypermethylated subtype            \n",
       "6 hypermethylated intergenic regions "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pubs = read.csv(file=\"./marker_collation/study_gene_regions.csv\", stringsAsFactors=FALSE)\n",
    "head(df_pubs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "92ffa6b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>7010</li><li>6</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7010\n",
       "\\item 6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7010\n",
       "2. 6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7010    6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5642</li><li>6</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5642\n",
       "\\item 6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5642\n",
       "2. 6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5642    6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>27</li><li>6</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 27\n",
       "\\item 6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 27\n",
       "2. 6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 27  6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5615</li><li>6</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5615\n",
       "\\item 6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5615\n",
       "2. 6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5615    6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# isolate probe entries\n",
    "df_pub_probes = df_pubs[df_pubs$probe != \"\",]\n",
    "dim(df_pub_probes)\n",
    "df_pub_genes = df_pubs[df_pubs$probe == \"\",]\n",
    "dim(df_pub_genes)\n",
    "df_pub_prom = df_pub_genes[df_pub_genes$region==\"promoter\",]\n",
    "dim(df_pub_prom)\n",
    "df_pub_genes = df_pub_genes[df_pub_genes$region==\"\",]\n",
    "dim(df_pub_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "f8f9a81a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 52</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>IlmnID</th><th scope=col>probe</th><th scope=col>AddressA_ID</th><th scope=col>AlleleA_ProbeSeq</th><th scope=col>AddressB_ID</th><th scope=col>AlleleB_ProbeSeq</th><th scope=col>Infinium_Design_Type</th><th scope=col>Next_Base</th><th scope=col>Color_Channel</th><th scope=col>Forward_Sequence</th><th scope=col>⋯</th><th scope=col>Coordinate_36</th><th scope=col>SNP_ID</th><th scope=col>SNP_DISTANCE</th><th scope=col>SNP_MinorAlleleFrequency</th><th scope=col>Random_Loci</th><th scope=col>MFG_Change_Flagged</th><th scope=col>CHR_hg38</th><th scope=col>Start_hg38</th><th scope=col>End_hg38</th><th scope=col>Strand_hg38</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>cg07881041</td><td>cg07881041</td><td>0085713262</td><td>CTACAAATACAACACCCTCAACCCATATTTCATATATTATCTCATTTAAC</td><td>          </td><td>                                                  </td><td>II</td><td> </td><td>   </td><td>CTGCACGCCTACTGCAGGTGCAGCACCCTCAGCCCATGTTTCATGTATTATCTCATTTAA[CG]CATGTATCATCTCATTTAATGCATGCATTATCTCATTTAATTCTCACAACCCCTCAGGTG</td><td>⋯</td><td>5187016  </td><td>rs187313142            </td><td>18   </td><td>0.000200         </td><td>NA</td><td>FALSE</td><td>chr19</td><td>  5236004</td><td>  5236006</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>cg18478105</td><td>cg18478105</td><td>0046761277</td><td>AAATAAATTTCACTCTCAAATCCCAATCTCATACAACAAAACAAAAACCA</td><td>0086644198</td><td>AAATAAATTTCGCTCTCAAATCCCAATCTCGTACGACGAAACGAAAACCG</td><td>I </td><td>C</td><td>Grn</td><td>TCCCGTCTTACGGGATGGATTTCGCTCTCAGGTCCCAGTCTCGTGCGGCGGGGCGGGGAC[CG]CAGCCGGCTGGGCGGGGAAGCCCTGAGCCGGGGAAGTCACGTGGGGCGTGTCCGGAGGCG</td><td>⋯</td><td>61318095 </td><td>rs549944121            </td><td>5    </td><td>0.001797         </td><td>NA</td><td>FALSE</td><td>chr20</td><td> 63216297</td><td> 63216299</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>cg23229610</td><td>cg23229610</td><td>0021717843</td><td>ATAAAATTCTTTCCTTAAAAAACATTAAAACCAAAATAAACAAAAATTCC</td><td>          </td><td>                                                  </td><td>II</td><td> </td><td>   </td><td>GTTTCTGGACAGTAAAATTCTTTCCTTGAAGGACATTAGGGCCAAAATGGGCAAGGATTC[CG]AGATTGGTACATCGAGCGTTATCTTCCAACTCTCTTTTCTAAATGGGCTCATTTAGTAAT</td><td>⋯</td><td>6763712  </td><td>rs545824288;rs527255711</td><td>40;12</td><td>0.000200;0.001198</td><td>NA</td><td>FALSE</td><td>chr1 </td><td>  6781064</td><td>  6781066</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>cg03513874</td><td>cg03513874</td><td>0029622133</td><td>ACAATAAAATAATAAAATCCCATCACTACTTACCCTCCTTAAATAATATC</td><td>          </td><td>                                                  </td><td>II</td><td> </td><td>   </td><td>ATTGTGCCCACCTTGCTGCTGACAGTTAAGCATCACTAAAGTAGGAAATAGGGTCCAAAC[CG]ACACTACTTAAGGAGGGCAAGTAGTGATGGGACCTCATCATCCCATTGCTATCATGGAGC</td><td>⋯</td><td>198011711</td><td>                       </td><td>     </td><td>                 </td><td>NA</td><td>FALSE</td><td>chr2 </td><td>197438741</td><td>197438743</td><td>-</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>cg09835024</td><td>cg09835024</td><td>0016745152</td><td>AATAAACACCAACCCCAAACCAATCTCACTTTATTAAATTACAAAAATCA</td><td>0081631976</td><td>AATAAACGCCGACCCCGAACCGATCTCGCTTTATTAAATTACAAAAATCG</td><td>I </td><td>A</td><td>Red</td><td>AGCCCCGTCATAGGTGGGCGCCGACCCCGAGCCGATCTCGCTTTATTAAATTACAGAAAT[CG]GTATTCAAAAAAAAAAAAAAAAAAGGGCGGGGAGGACACTCCCTCTTCTCTGTTCCCACA</td><td>⋯</td><td>23982561 </td><td>                       </td><td>     </td><td>                 </td><td>NA</td><td>FALSE</td><td>chrX </td><td> 24054522</td><td> 24054524</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>cg05451842</td><td>cg05451842</td><td>0016681196</td><td>CRTTCAAATACACTATAACCCRACTAAAAAAACCCCCAACAACCCAAAAC</td><td>          </td><td>                                                  </td><td>II</td><td> </td><td>   </td><td>CACAGCGTGGATGCCCCGATTTCCCAGGTCCCTCCGCAACCCTCAGTAGAACTCCCACCG[CG]CCCTGGGCTGCTGGGGGCCTCCCCAGCCGGGTCACAGTGCACCTGAACGCCCCGGTCCGT</td><td>⋯</td><td>92650892 </td><td>rs550745821            </td><td>22   </td><td>0.000200         </td><td>NA</td><td>FALSE</td><td>chr14</td><td> 93114793</td><td> 93114795</td><td>-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 52\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & IlmnID & probe & AddressA\\_ID & AlleleA\\_ProbeSeq & AddressB\\_ID & AlleleB\\_ProbeSeq & Infinium\\_Design\\_Type & Next\\_Base & Color\\_Channel & Forward\\_Sequence & ⋯ & Coordinate\\_36 & SNP\\_ID & SNP\\_DISTANCE & SNP\\_MinorAlleleFrequency & Random\\_Loci & MFG\\_Change\\_Flagged & CHR\\_hg38 & Start\\_hg38 & End\\_hg38 & Strand\\_hg38\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & cg07881041 & cg07881041 & 0085713262 & CTACAAATACAACACCCTCAACCCATATTTCATATATTATCTCATTTAAC &            &                                                    & II &   &     & CTGCACGCCTACTGCAGGTGCAGCACCCTCAGCCCATGTTTCATGTATTATCTCATTTAA{[}CG{]}CATGTATCATCTCATTTAATGCATGCATTATCTCATTTAATTCTCACAACCCCTCAGGTG & ⋯ & 5187016   & rs187313142             & 18    & 0.000200          & NA & FALSE & chr19 &   5236004 &   5236006 & +\\\\\n",
       "\t2 & cg18478105 & cg18478105 & 0046761277 & AAATAAATTTCACTCTCAAATCCCAATCTCATACAACAAAACAAAAACCA & 0086644198 & AAATAAATTTCGCTCTCAAATCCCAATCTCGTACGACGAAACGAAAACCG & I  & C & Grn & TCCCGTCTTACGGGATGGATTTCGCTCTCAGGTCCCAGTCTCGTGCGGCGGGGCGGGGAC{[}CG{]}CAGCCGGCTGGGCGGGGAAGCCCTGAGCCGGGGAAGTCACGTGGGGCGTGTCCGGAGGCG & ⋯ & 61318095  & rs549944121             & 5     & 0.001797          & NA & FALSE & chr20 &  63216297 &  63216299 & +\\\\\n",
       "\t3 & cg23229610 & cg23229610 & 0021717843 & ATAAAATTCTTTCCTTAAAAAACATTAAAACCAAAATAAACAAAAATTCC &            &                                                    & II &   &     & GTTTCTGGACAGTAAAATTCTTTCCTTGAAGGACATTAGGGCCAAAATGGGCAAGGATTC{[}CG{]}AGATTGGTACATCGAGCGTTATCTTCCAACTCTCTTTTCTAAATGGGCTCATTTAGTAAT & ⋯ & 6763712   & rs545824288;rs527255711 & 40;12 & 0.000200;0.001198 & NA & FALSE & chr1  &   6781064 &   6781066 & +\\\\\n",
       "\t4 & cg03513874 & cg03513874 & 0029622133 & ACAATAAAATAATAAAATCCCATCACTACTTACCCTCCTTAAATAATATC &            &                                                    & II &   &     & ATTGTGCCCACCTTGCTGCTGACAGTTAAGCATCACTAAAGTAGGAAATAGGGTCCAAAC{[}CG{]}ACACTACTTAAGGAGGGCAAGTAGTGATGGGACCTCATCATCCCATTGCTATCATGGAGC & ⋯ & 198011711 &                         &       &                   & NA & FALSE & chr2  & 197438741 & 197438743 & -\\\\\n",
       "\t5 & cg09835024 & cg09835024 & 0016745152 & AATAAACACCAACCCCAAACCAATCTCACTTTATTAAATTACAAAAATCA & 0081631976 & AATAAACGCCGACCCCGAACCGATCTCGCTTTATTAAATTACAAAAATCG & I  & A & Red & AGCCCCGTCATAGGTGGGCGCCGACCCCGAGCCGATCTCGCTTTATTAAATTACAGAAAT{[}CG{]}GTATTCAAAAAAAAAAAAAAAAAAGGGCGGGGAGGACACTCCCTCTTCTCTGTTCCCACA & ⋯ & 23982561  &                         &       &                   & NA & FALSE & chrX  &  24054522 &  24054524 & +\\\\\n",
       "\t6 & cg05451842 & cg05451842 & 0016681196 & CRTTCAAATACACTATAACCCRACTAAAAAAACCCCCAACAACCCAAAAC &            &                                                    & II &   &     & CACAGCGTGGATGCCCCGATTTCCCAGGTCCCTCCGCAACCCTCAGTAGAACTCCCACCG{[}CG{]}CCCTGGGCTGCTGGGGGCCTCCCCAGCCGGGTCACAGTGCACCTGAACGCCCCGGTCCGT & ⋯ & 92650892  & rs550745821             & 22    & 0.000200          & NA & FALSE & chr14 &  93114793 &  93114795 & -\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 52\n",
       "\n",
       "| <!--/--> | IlmnID &lt;chr&gt; | probe &lt;chr&gt; | AddressA_ID &lt;chr&gt; | AlleleA_ProbeSeq &lt;chr&gt; | AddressB_ID &lt;chr&gt; | AlleleB_ProbeSeq &lt;chr&gt; | Infinium_Design_Type &lt;chr&gt; | Next_Base &lt;chr&gt; | Color_Channel &lt;chr&gt; | Forward_Sequence &lt;chr&gt; | ⋯ ⋯ | Coordinate_36 &lt;chr&gt; | SNP_ID &lt;chr&gt; | SNP_DISTANCE &lt;chr&gt; | SNP_MinorAlleleFrequency &lt;chr&gt; | Random_Loci &lt;lgl&gt; | MFG_Change_Flagged &lt;lgl&gt; | CHR_hg38 &lt;chr&gt; | Start_hg38 &lt;int&gt; | End_hg38 &lt;int&gt; | Strand_hg38 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | cg07881041 | cg07881041 | 0085713262 | CTACAAATACAACACCCTCAACCCATATTTCATATATTATCTCATTTAAC | <!----> | <!----> | II | <!----> | <!----> | CTGCACGCCTACTGCAGGTGCAGCACCCTCAGCCCATGTTTCATGTATTATCTCATTTAA[CG]CATGTATCATCTCATTTAATGCATGCATTATCTCATTTAATTCTCACAACCCCTCAGGTG | ⋯ | 5187016   | rs187313142             | 18    | 0.000200          | NA | FALSE | chr19 |   5236004 |   5236006 | + |\n",
       "| 2 | cg18478105 | cg18478105 | 0046761277 | AAATAAATTTCACTCTCAAATCCCAATCTCATACAACAAAACAAAAACCA | 0086644198 | AAATAAATTTCGCTCTCAAATCCCAATCTCGTACGACGAAACGAAAACCG | I  | C | Grn | TCCCGTCTTACGGGATGGATTTCGCTCTCAGGTCCCAGTCTCGTGCGGCGGGGCGGGGAC[CG]CAGCCGGCTGGGCGGGGAAGCCCTGAGCCGGGGAAGTCACGTGGGGCGTGTCCGGAGGCG | ⋯ | 61318095  | rs549944121             | 5     | 0.001797          | NA | FALSE | chr20 |  63216297 |  63216299 | + |\n",
       "| 3 | cg23229610 | cg23229610 | 0021717843 | ATAAAATTCTTTCCTTAAAAAACATTAAAACCAAAATAAACAAAAATTCC | <!----> | <!----> | II | <!----> | <!----> | GTTTCTGGACAGTAAAATTCTTTCCTTGAAGGACATTAGGGCCAAAATGGGCAAGGATTC[CG]AGATTGGTACATCGAGCGTTATCTTCCAACTCTCTTTTCTAAATGGGCTCATTTAGTAAT | ⋯ | 6763712   | rs545824288;rs527255711 | 40;12 | 0.000200;0.001198 | NA | FALSE | chr1  |   6781064 |   6781066 | + |\n",
       "| 4 | cg03513874 | cg03513874 | 0029622133 | ACAATAAAATAATAAAATCCCATCACTACTTACCCTCCTTAAATAATATC | <!----> | <!----> | II | <!----> | <!----> | ATTGTGCCCACCTTGCTGCTGACAGTTAAGCATCACTAAAGTAGGAAATAGGGTCCAAAC[CG]ACACTACTTAAGGAGGGCAAGTAGTGATGGGACCTCATCATCCCATTGCTATCATGGAGC | ⋯ | 198011711 | <!----> | <!----> | <!----> | NA | FALSE | chr2  | 197438741 | 197438743 | - |\n",
       "| 5 | cg09835024 | cg09835024 | 0016745152 | AATAAACACCAACCCCAAACCAATCTCACTTTATTAAATTACAAAAATCA | 0081631976 | AATAAACGCCGACCCCGAACCGATCTCGCTTTATTAAATTACAAAAATCG | I  | A | Red | AGCCCCGTCATAGGTGGGCGCCGACCCCGAGCCGATCTCGCTTTATTAAATTACAGAAAT[CG]GTATTCAAAAAAAAAAAAAAAAAAGGGCGGGGAGGACACTCCCTCTTCTCTGTTCCCACA | ⋯ | 23982561  | <!----> | <!----> | <!----> | NA | FALSE | chrX  |  24054522 |  24054524 | + |\n",
       "| 6 | cg05451842 | cg05451842 | 0016681196 | CRTTCAAATACACTATAACCCRACTAAAAAAACCCCCAACAACCCAAAAC | <!----> | <!----> | II | <!----> | <!----> | CACAGCGTGGATGCCCCGATTTCCCAGGTCCCTCCGCAACCCTCAGTAGAACTCCCACCG[CG]CCCTGGGCTGCTGGGGGCCTCCCCAGCCGGGTCACAGTGCACCTGAACGCCCCGGTCCGT | ⋯ | 92650892  | rs550745821             | 22    | 0.000200          | NA | FALSE | chr14 |  93114793 |  93114795 | - |\n",
       "\n"
      ],
      "text/plain": [
       "  IlmnID     probe      AddressA_ID\n",
       "1 cg07881041 cg07881041 0085713262 \n",
       "2 cg18478105 cg18478105 0046761277 \n",
       "3 cg23229610 cg23229610 0021717843 \n",
       "4 cg03513874 cg03513874 0029622133 \n",
       "5 cg09835024 cg09835024 0016745152 \n",
       "6 cg05451842 cg05451842 0016681196 \n",
       "  AlleleA_ProbeSeq                                   AddressB_ID\n",
       "1 CTACAAATACAACACCCTCAACCCATATTTCATATATTATCTCATTTAAC            \n",
       "2 AAATAAATTTCACTCTCAAATCCCAATCTCATACAACAAAACAAAAACCA 0086644198 \n",
       "3 ATAAAATTCTTTCCTTAAAAAACATTAAAACCAAAATAAACAAAAATTCC            \n",
       "4 ACAATAAAATAATAAAATCCCATCACTACTTACCCTCCTTAAATAATATC            \n",
       "5 AATAAACACCAACCCCAAACCAATCTCACTTTATTAAATTACAAAAATCA 0081631976 \n",
       "6 CRTTCAAATACACTATAACCCRACTAAAAAAACCCCCAACAACCCAAAAC            \n",
       "  AlleleB_ProbeSeq                                   Infinium_Design_Type\n",
       "1                                                    II                  \n",
       "2 AAATAAATTTCGCTCTCAAATCCCAATCTCGTACGACGAAACGAAAACCG I                   \n",
       "3                                                    II                  \n",
       "4                                                    II                  \n",
       "5 AATAAACGCCGACCCCGAACCGATCTCGCTTTATTAAATTACAAAAATCG I                   \n",
       "6                                                    II                  \n",
       "  Next_Base Color_Channel\n",
       "1                        \n",
       "2 C         Grn          \n",
       "3                        \n",
       "4                        \n",
       "5 A         Red          \n",
       "6                        \n",
       "  Forward_Sequence                                                                                                            \n",
       "1 CTGCACGCCTACTGCAGGTGCAGCACCCTCAGCCCATGTTTCATGTATTATCTCATTTAA[CG]CATGTATCATCTCATTTAATGCATGCATTATCTCATTTAATTCTCACAACCCCTCAGGTG\n",
       "2 TCCCGTCTTACGGGATGGATTTCGCTCTCAGGTCCCAGTCTCGTGCGGCGGGGCGGGGAC[CG]CAGCCGGCTGGGCGGGGAAGCCCTGAGCCGGGGAAGTCACGTGGGGCGTGTCCGGAGGCG\n",
       "3 GTTTCTGGACAGTAAAATTCTTTCCTTGAAGGACATTAGGGCCAAAATGGGCAAGGATTC[CG]AGATTGGTACATCGAGCGTTATCTTCCAACTCTCTTTTCTAAATGGGCTCATTTAGTAAT\n",
       "4 ATTGTGCCCACCTTGCTGCTGACAGTTAAGCATCACTAAAGTAGGAAATAGGGTCCAAAC[CG]ACACTACTTAAGGAGGGCAAGTAGTGATGGGACCTCATCATCCCATTGCTATCATGGAGC\n",
       "5 AGCCCCGTCATAGGTGGGCGCCGACCCCGAGCCGATCTCGCTTTATTAAATTACAGAAAT[CG]GTATTCAAAAAAAAAAAAAAAAAAGGGCGGGGAGGACACTCCCTCTTCTCTGTTCCCACA\n",
       "6 CACAGCGTGGATGCCCCGATTTCCCAGGTCCCTCCGCAACCCTCAGTAGAACTCCCACCG[CG]CCCTGGGCTGCTGGGGGCCTCCCCAGCCGGGTCACAGTGCACCTGAACGCCCCGGTCCGT\n",
       "  ⋯ Coordinate_36 SNP_ID                  SNP_DISTANCE SNP_MinorAlleleFrequency\n",
       "1 ⋯ 5187016       rs187313142             18           0.000200                \n",
       "2 ⋯ 61318095      rs549944121             5            0.001797                \n",
       "3 ⋯ 6763712       rs545824288;rs527255711 40;12        0.000200;0.001198       \n",
       "4 ⋯ 198011711                                                                  \n",
       "5 ⋯ 23982561                                                                   \n",
       "6 ⋯ 92650892      rs550745821             22           0.000200                \n",
       "  Random_Loci MFG_Change_Flagged CHR_hg38 Start_hg38 End_hg38  Strand_hg38\n",
       "1 NA          FALSE              chr19      5236004    5236006 +          \n",
       "2 NA          FALSE              chr20     63216297   63216299 +          \n",
       "3 NA          FALSE              chr1       6781064    6781066 +          \n",
       "4 NA          FALSE              chr2     197438741  197438743 -          \n",
       "5 NA          FALSE              chrX      24054522   24054524 +          \n",
       "6 NA          FALSE              chr14     93114793   93114795 -          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_probes = read.csv(file=\"./marker_collation//EPIC_array_manifest_no_head.csv\",header=TRUE)\n",
    "df_probes = df_probes %>% rename(probe=Name)\n",
    "head(df_probes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "cae51830",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'IlmnID'</li><li>'probe'</li><li>'AddressA_ID'</li><li>'AlleleA_ProbeSeq'</li><li>'AddressB_ID'</li><li>'AlleleB_ProbeSeq'</li><li>'Infinium_Design_Type'</li><li>'Next_Base'</li><li>'Color_Channel'</li><li>'Forward_Sequence'</li><li>'Genome_Build'</li><li>'CHR'</li><li>'MAPINFO'</li><li>'SourceSeq'</li><li>'Strand'</li><li>'UCSC_RefGene_Name'</li><li>'UCSC_RefGene_Accession'</li><li>'UCSC_RefGene_Group'</li><li>'UCSC_CpG_Islands_Name'</li><li>'Relation_to_UCSC_CpG_Island'</li><li>'Phantom4_Enhancers'</li><li>'Phantom5_Enhancers'</li><li>'DMR'</li><li>'X450k_Enhancer'</li><li>'HMM_Island'</li><li>'Regulatory_Feature_Name'</li><li>'Regulatory_Feature_Group'</li><li>'GencodeBasicV12_NAME'</li><li>'GencodeBasicV12_Accession'</li><li>'GencodeBasicV12_Group'</li><li>'GencodeCompV12_NAME'</li><li>'GencodeCompV12_Accession'</li><li>'GencodeCompV12_Group'</li><li>'DNase_Hypersensitivity_NAME'</li><li>'DNase_Hypersensitivity_Evidence_Count'</li><li>'OpenChromatin_NAME'</li><li>'OpenChromatin_Evidence_Count'</li><li>'TFBS_NAME'</li><li>'TFBS_Evidence_Count'</li><li>'Methyl27_Loci'</li><li>'Methyl450_Loci'</li><li>'Chromosome_36'</li><li>'Coordinate_36'</li><li>'SNP_ID'</li><li>'SNP_DISTANCE'</li><li>'SNP_MinorAlleleFrequency'</li><li>'Random_Loci'</li><li>'MFG_Change_Flagged'</li><li>'CHR_hg38'</li><li>'Start_hg38'</li><li>'End_hg38'</li><li>'Strand_hg38'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'IlmnID'\n",
       "\\item 'probe'\n",
       "\\item 'AddressA\\_ID'\n",
       "\\item 'AlleleA\\_ProbeSeq'\n",
       "\\item 'AddressB\\_ID'\n",
       "\\item 'AlleleB\\_ProbeSeq'\n",
       "\\item 'Infinium\\_Design\\_Type'\n",
       "\\item 'Next\\_Base'\n",
       "\\item 'Color\\_Channel'\n",
       "\\item 'Forward\\_Sequence'\n",
       "\\item 'Genome\\_Build'\n",
       "\\item 'CHR'\n",
       "\\item 'MAPINFO'\n",
       "\\item 'SourceSeq'\n",
       "\\item 'Strand'\n",
       "\\item 'UCSC\\_RefGene\\_Name'\n",
       "\\item 'UCSC\\_RefGene\\_Accession'\n",
       "\\item 'UCSC\\_RefGene\\_Group'\n",
       "\\item 'UCSC\\_CpG\\_Islands\\_Name'\n",
       "\\item 'Relation\\_to\\_UCSC\\_CpG\\_Island'\n",
       "\\item 'Phantom4\\_Enhancers'\n",
       "\\item 'Phantom5\\_Enhancers'\n",
       "\\item 'DMR'\n",
       "\\item 'X450k\\_Enhancer'\n",
       "\\item 'HMM\\_Island'\n",
       "\\item 'Regulatory\\_Feature\\_Name'\n",
       "\\item 'Regulatory\\_Feature\\_Group'\n",
       "\\item 'GencodeBasicV12\\_NAME'\n",
       "\\item 'GencodeBasicV12\\_Accession'\n",
       "\\item 'GencodeBasicV12\\_Group'\n",
       "\\item 'GencodeCompV12\\_NAME'\n",
       "\\item 'GencodeCompV12\\_Accession'\n",
       "\\item 'GencodeCompV12\\_Group'\n",
       "\\item 'DNase\\_Hypersensitivity\\_NAME'\n",
       "\\item 'DNase\\_Hypersensitivity\\_Evidence\\_Count'\n",
       "\\item 'OpenChromatin\\_NAME'\n",
       "\\item 'OpenChromatin\\_Evidence\\_Count'\n",
       "\\item 'TFBS\\_NAME'\n",
       "\\item 'TFBS\\_Evidence\\_Count'\n",
       "\\item 'Methyl27\\_Loci'\n",
       "\\item 'Methyl450\\_Loci'\n",
       "\\item 'Chromosome\\_36'\n",
       "\\item 'Coordinate\\_36'\n",
       "\\item 'SNP\\_ID'\n",
       "\\item 'SNP\\_DISTANCE'\n",
       "\\item 'SNP\\_MinorAlleleFrequency'\n",
       "\\item 'Random\\_Loci'\n",
       "\\item 'MFG\\_Change\\_Flagged'\n",
       "\\item 'CHR\\_hg38'\n",
       "\\item 'Start\\_hg38'\n",
       "\\item 'End\\_hg38'\n",
       "\\item 'Strand\\_hg38'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'IlmnID'\n",
       "2. 'probe'\n",
       "3. 'AddressA_ID'\n",
       "4. 'AlleleA_ProbeSeq'\n",
       "5. 'AddressB_ID'\n",
       "6. 'AlleleB_ProbeSeq'\n",
       "7. 'Infinium_Design_Type'\n",
       "8. 'Next_Base'\n",
       "9. 'Color_Channel'\n",
       "10. 'Forward_Sequence'\n",
       "11. 'Genome_Build'\n",
       "12. 'CHR'\n",
       "13. 'MAPINFO'\n",
       "14. 'SourceSeq'\n",
       "15. 'Strand'\n",
       "16. 'UCSC_RefGene_Name'\n",
       "17. 'UCSC_RefGene_Accession'\n",
       "18. 'UCSC_RefGene_Group'\n",
       "19. 'UCSC_CpG_Islands_Name'\n",
       "20. 'Relation_to_UCSC_CpG_Island'\n",
       "21. 'Phantom4_Enhancers'\n",
       "22. 'Phantom5_Enhancers'\n",
       "23. 'DMR'\n",
       "24. 'X450k_Enhancer'\n",
       "25. 'HMM_Island'\n",
       "26. 'Regulatory_Feature_Name'\n",
       "27. 'Regulatory_Feature_Group'\n",
       "28. 'GencodeBasicV12_NAME'\n",
       "29. 'GencodeBasicV12_Accession'\n",
       "30. 'GencodeBasicV12_Group'\n",
       "31. 'GencodeCompV12_NAME'\n",
       "32. 'GencodeCompV12_Accession'\n",
       "33. 'GencodeCompV12_Group'\n",
       "34. 'DNase_Hypersensitivity_NAME'\n",
       "35. 'DNase_Hypersensitivity_Evidence_Count'\n",
       "36. 'OpenChromatin_NAME'\n",
       "37. 'OpenChromatin_Evidence_Count'\n",
       "38. 'TFBS_NAME'\n",
       "39. 'TFBS_Evidence_Count'\n",
       "40. 'Methyl27_Loci'\n",
       "41. 'Methyl450_Loci'\n",
       "42. 'Chromosome_36'\n",
       "43. 'Coordinate_36'\n",
       "44. 'SNP_ID'\n",
       "45. 'SNP_DISTANCE'\n",
       "46. 'SNP_MinorAlleleFrequency'\n",
       "47. 'Random_Loci'\n",
       "48. 'MFG_Change_Flagged'\n",
       "49. 'CHR_hg38'\n",
       "50. 'Start_hg38'\n",
       "51. 'End_hg38'\n",
       "52. 'Strand_hg38'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"IlmnID\"                               \n",
       " [2] \"probe\"                                \n",
       " [3] \"AddressA_ID\"                          \n",
       " [4] \"AlleleA_ProbeSeq\"                     \n",
       " [5] \"AddressB_ID\"                          \n",
       " [6] \"AlleleB_ProbeSeq\"                     \n",
       " [7] \"Infinium_Design_Type\"                 \n",
       " [8] \"Next_Base\"                            \n",
       " [9] \"Color_Channel\"                        \n",
       "[10] \"Forward_Sequence\"                     \n",
       "[11] \"Genome_Build\"                         \n",
       "[12] \"CHR\"                                  \n",
       "[13] \"MAPINFO\"                              \n",
       "[14] \"SourceSeq\"                            \n",
       "[15] \"Strand\"                               \n",
       "[16] \"UCSC_RefGene_Name\"                    \n",
       "[17] \"UCSC_RefGene_Accession\"               \n",
       "[18] \"UCSC_RefGene_Group\"                   \n",
       "[19] \"UCSC_CpG_Islands_Name\"                \n",
       "[20] \"Relation_to_UCSC_CpG_Island\"          \n",
       "[21] \"Phantom4_Enhancers\"                   \n",
       "[22] \"Phantom5_Enhancers\"                   \n",
       "[23] \"DMR\"                                  \n",
       "[24] \"X450k_Enhancer\"                       \n",
       "[25] \"HMM_Island\"                           \n",
       "[26] \"Regulatory_Feature_Name\"              \n",
       "[27] \"Regulatory_Feature_Group\"             \n",
       "[28] \"GencodeBasicV12_NAME\"                 \n",
       "[29] \"GencodeBasicV12_Accession\"            \n",
       "[30] \"GencodeBasicV12_Group\"                \n",
       "[31] \"GencodeCompV12_NAME\"                  \n",
       "[32] \"GencodeCompV12_Accession\"             \n",
       "[33] \"GencodeCompV12_Group\"                 \n",
       "[34] \"DNase_Hypersensitivity_NAME\"          \n",
       "[35] \"DNase_Hypersensitivity_Evidence_Count\"\n",
       "[36] \"OpenChromatin_NAME\"                   \n",
       "[37] \"OpenChromatin_Evidence_Count\"         \n",
       "[38] \"TFBS_NAME\"                            \n",
       "[39] \"TFBS_Evidence_Count\"                  \n",
       "[40] \"Methyl27_Loci\"                        \n",
       "[41] \"Methyl450_Loci\"                       \n",
       "[42] \"Chromosome_36\"                        \n",
       "[43] \"Coordinate_36\"                        \n",
       "[44] \"SNP_ID\"                               \n",
       "[45] \"SNP_DISTANCE\"                         \n",
       "[46] \"SNP_MinorAlleleFrequency\"             \n",
       "[47] \"Random_Loci\"                          \n",
       "[48] \"MFG_Change_Flagged\"                   \n",
       "[49] \"CHR_hg38\"                             \n",
       "[50] \"Start_hg38\"                           \n",
       "[51] \"End_hg38\"                             \n",
       "[52] \"Strand_hg38\"                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(df_probes)\n",
    "# https://support.illumina.com/content/dam/illumina-support/documents/downloads/productfiles/methylationepic/infinium-methylationepic-manifest-column-headings.pdf\n",
    "# CHR: Chromosome containing the CpG (Build 37).\n",
    "# MAPINFO: Chromosomal coordinates of the CpG (Build 37)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "5a9bf460",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>CHR</th><th scope=col>MAPINFO</th><th scope=col>Strand</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td></td><td>KLF8  </td><td>cg03834574</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>X </td><td> 56259094</td><td>R</td><td>epic_probe</td></tr>\n",
       "\t<tr><th scope=row>2</th><td></td><td>HOPX  </td><td>cg21899596</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>4 </td><td> 57522493</td><td>R</td><td>epic_probe</td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>RARB  </td><td>cg24396624</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>3 </td><td> 25469860</td><td>R</td><td>epic_probe</td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>HIF3A </td><td>cg21544503</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>19</td><td> 46800131</td><td>R</td><td>epic_probe</td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>HIF3A </td><td>cg14117138</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>19</td><td> 46800085</td><td>R</td><td>epic_probe</td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>CXCL14</td><td>cg10493994</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>5 </td><td>134914932</td><td>F</td><td>epic_probe</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & CHR & MAPINFO & Strand & type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 &  & KLF8   & cg03834574 & 1 & Ylitalo et al 2021 & Hypermethylated & X  &  56259094 & R & epic\\_probe\\\\\n",
       "\t2 &  & HOPX   & cg21899596 & 1 & Ylitalo et al 2021 & Hypermethylated & 4  &  57522493 & R & epic\\_probe\\\\\n",
       "\t3 &  & RARB   & cg24396624 & 1 & Ylitalo et al 2021 & Hypermethylated & 3  &  25469860 & R & epic\\_probe\\\\\n",
       "\t4 &  & HIF3A  & cg21544503 & 1 & Ylitalo et al 2021 & Hypermethylated & 19 &  46800131 & R & epic\\_probe\\\\\n",
       "\t5 &  & HIF3A  & cg14117138 & 1 & Ylitalo et al 2021 & Hypermethylated & 19 &  46800085 & R & epic\\_probe\\\\\n",
       "\t6 &  & CXCL14 & cg10493994 & 1 & Ylitalo et al 2021 & Hypermethylated & 5  & 134914932 & F & epic\\_probe\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | CHR &lt;chr&gt; | MAPINFO &lt;int&gt; | Strand &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | <!----> | KLF8   | cg03834574 | 1 | Ylitalo et al 2021 | Hypermethylated | X  |  56259094 | R | epic_probe |\n",
       "| 2 | <!----> | HOPX   | cg21899596 | 1 | Ylitalo et al 2021 | Hypermethylated | 4  |  57522493 | R | epic_probe |\n",
       "| 3 | <!----> | RARB   | cg24396624 | 1 | Ylitalo et al 2021 | Hypermethylated | 3  |  25469860 | R | epic_probe |\n",
       "| 4 | <!----> | HIF3A  | cg21544503 | 1 | Ylitalo et al 2021 | Hypermethylated | 19 |  46800131 | R | epic_probe |\n",
       "| 5 | <!----> | HIF3A  | cg14117138 | 1 | Ylitalo et al 2021 | Hypermethylated | 19 |  46800085 | R | epic_probe |\n",
       "| 6 | <!----> | CXCL14 | cg10493994 | 1 | Ylitalo et al 2021 | Hypermethylated | 5  | 134914932 | F | epic_probe |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe      No..pubs Source             note            CHR\n",
       "1        KLF8   cg03834574 1        Ylitalo et al 2021 Hypermethylated X  \n",
       "2        HOPX   cg21899596 1        Ylitalo et al 2021 Hypermethylated 4  \n",
       "3        RARB   cg24396624 1        Ylitalo et al 2021 Hypermethylated 3  \n",
       "4        HIF3A  cg21544503 1        Ylitalo et al 2021 Hypermethylated 19 \n",
       "5        HIF3A  cg14117138 1        Ylitalo et al 2021 Hypermethylated 19 \n",
       "6        CXCL14 cg10493994 1        Ylitalo et al 2021 Hypermethylated 5  \n",
       "  MAPINFO   Strand type      \n",
       "1  56259094 R      epic_probe\n",
       "2  57522493 R      epic_probe\n",
       "3  25469860 R      epic_probe\n",
       "4  46800131 R      epic_probe\n",
       "5  46800085 R      epic_probe\n",
       "6 134914932 F      epic_probe"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# hg19/gr37\n",
    "df_probes = df_probes[, c(\"probe\",'CHR','MAPINFO',\"Strand\")]\n",
    "df_probes$type = \"epic_probe\"\n",
    "# most cg probes have more than one illumina probeID (why?) see orig sheet. retain only first entry\n",
    "df_pub_probes_annot = df_pub_probes %>% left_join(df_probes, by=\"probe\") %>% distinct(probe, .keep_all=TRUE)\n",
    "head(df_pub_probes_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "e23318c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>10869</td><td>14409</td><td>DDX11L1    </td><td>0</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14404</td><td>30570</td><td>WASH7P     </td><td>0</td><td>-</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>28554</td><td>31109</td><td>MIR1302-2HG</td><td>0</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>34554</td><td>37081</td><td>FAM138A    </td><td>0</td><td>-</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>51473</td><td>53312</td><td>OR4G4P     </td><td>0</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>56598</td><td>64116</td><td>OR4G11P    </td><td>0</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start & end & gene\\_name & score & strand & gencode.type\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 10869 & 14409 & DDX11L1     & 0 & + & gene\\_plus\\_promoter\\\\\n",
       "\t2 & chr1 & 14404 & 30570 & WASH7P      & 0 & - & gene\\_plus\\_promoter\\\\\n",
       "\t3 & chr1 & 28554 & 31109 & MIR1302-2HG & 0 & + & gene\\_plus\\_promoter\\\\\n",
       "\t4 & chr1 & 34554 & 37081 & FAM138A     & 0 & - & gene\\_plus\\_promoter\\\\\n",
       "\t5 & chr1 & 51473 & 53312 & OR4G4P      & 0 & + & gene\\_plus\\_promoter\\\\\n",
       "\t6 & chr1 & 56598 & 64116 & OR4G11P     & 0 & + & gene\\_plus\\_promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | gene_name &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 10869 | 14409 | DDX11L1     | 0 | + | gene_plus_promoter |\n",
       "| 2 | chr1 | 14404 | 30570 | WASH7P      | 0 | - | gene_plus_promoter |\n",
       "| 3 | chr1 | 28554 | 31109 | MIR1302-2HG | 0 | + | gene_plus_promoter |\n",
       "| 4 | chr1 | 34554 | 37081 | FAM138A     | 0 | - | gene_plus_promoter |\n",
       "| 5 | chr1 | 51473 | 53312 | OR4G4P      | 0 | + | gene_plus_promoter |\n",
       "| 6 | chr1 | 56598 | 64116 | OR4G11P     | 0 | + | gene_plus_promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  start end   gene_name   score strand gencode.type      \n",
       "1 chr1 10869 14409 DDX11L1     0     +      gene_plus_promoter\n",
       "2 chr1 14404 30570 WASH7P      0     -      gene_plus_promoter\n",
       "3 chr1 28554 31109 MIR1302-2HG 0     +      gene_plus_promoter\n",
       "4 chr1 34554 37081 FAM138A     0     -      gene_plus_promoter\n",
       "5 chr1 51473 53312 OR4G4P      0     +      gene_plus_promoter\n",
       "6 chr1 56598 64116 OR4G11P     0     +      gene_plus_promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>10869</td><td>11869</td><td>DDX11L1    </td><td>0</td><td>+</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>29570</td><td>30570</td><td>WASH7P     </td><td>0</td><td>-</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>28554</td><td>29554</td><td>MIR1302-2HG</td><td>0</td><td>+</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>36081</td><td>37081</td><td>FAM138A    </td><td>0</td><td>-</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>51473</td><td>52473</td><td>OR4G4P     </td><td>0</td><td>+</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>56598</td><td>57598</td><td>OR4G11P    </td><td>0</td><td>+</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start & end & gene\\_name & score & strand & gencode.type\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 10869 & 11869 & DDX11L1     & 0 & + & promoter\\\\\n",
       "\t2 & 1 & 29570 & 30570 & WASH7P      & 0 & - & promoter\\\\\n",
       "\t3 & 1 & 28554 & 29554 & MIR1302-2HG & 0 & + & promoter\\\\\n",
       "\t4 & 1 & 36081 & 37081 & FAM138A     & 0 & - & promoter\\\\\n",
       "\t5 & 1 & 51473 & 52473 & OR4G4P      & 0 & + & promoter\\\\\n",
       "\t6 & 1 & 56598 & 57598 & OR4G11P     & 0 & + & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | gene_name &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 10869 | 11869 | DDX11L1     | 0 | + | promoter |\n",
       "| 2 | 1 | 29570 | 30570 | WASH7P      | 0 | - | promoter |\n",
       "| 3 | 1 | 28554 | 29554 | MIR1302-2HG | 0 | + | promoter |\n",
       "| 4 | 1 | 36081 | 37081 | FAM138A     | 0 | - | promoter |\n",
       "| 5 | 1 | 51473 | 52473 | OR4G4P      | 0 | + | promoter |\n",
       "| 6 | 1 | 56598 | 57598 | OR4G11P     | 0 | + | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start end   gene_name   score strand gencode.type\n",
       "1 1   10869 11869 DDX11L1     0     +      promoter    \n",
       "2 1   29570 30570 WASH7P      0     -      promoter    \n",
       "3 1   28554 29554 MIR1302-2HG 0     +      promoter    \n",
       "4 1   36081 37081 FAM138A     0     -      promoter    \n",
       "5 1   51473 52473 OR4G4P      0     +      promoter    \n",
       "6 1   56598 57598 OR4G11P     0     +      promoter    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# retrieve gene region coords hg19/gr37\n",
    "df_gene_coords = read.table(file=\"../Translational_Science/gencode.v40lift37.full_genes_hg19.bed\", sep=\"\\t\", header=FALSE)\n",
    "colnames(df_gene_coords) <- c(\"chr\",\"start\",\"end\",\"gene_name\", \"score\",\"strand\",\"gencode.type\")\n",
    "df_promoter_coords = read.table(file=\"../Translational_Science/gencode.v40lift37.promoters_hg19.bed\", sep=\"\\t\", header=FALSE)\n",
    "colnames(df_promoter_coords) <- c(\"chr\",\"start\",\"end\",\"gene_name\", \"score\",\"strand\",\"gencode.type\")\n",
    "head(df_gene_coords)\n",
    "head(df_promoter_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "56146897",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td></td><td>CD24  </td><td></td><td>NA</td><td>Lu et al 2022       </td><td>hypermethylated                    </td><td>NA   </td><td>       NA</td><td>       NA</td><td>NA</td><td>NA</td><td>NA                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td></td><td>TET2  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr4 </td><td>106066032</td><td>106200973</td><td> 0</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>DNMT3B</td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr20</td><td> 31349191</td><td> 31397165</td><td> 0</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>IDH1  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr2 </td><td>209100953</td><td>209131798</td><td> 0</td><td>- </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>BRAF  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr7 </td><td>140419127</td><td>140625729</td><td> 0</td><td>- </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>AR    </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated intergenic regions </td><td>chrX </td><td> 66762863</td><td> 66950461</td><td> 0</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & start & end & score & strand & gencode.type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 &  & CD24   &  & NA & Lu et al 2022        & hypermethylated                     & NA    &        NA &        NA & NA & NA & NA                \\\\\n",
       "\t2 &  & TET2   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr4  & 106066032 & 106200973 &  0 & +  & gene\\_plus\\_promoter\\\\\n",
       "\t3 &  & DNMT3B &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr20 &  31349191 &  31397165 &  0 & +  & gene\\_plus\\_promoter\\\\\n",
       "\t4 &  & IDH1   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr2  & 209100953 & 209131798 &  0 & -  & gene\\_plus\\_promoter\\\\\n",
       "\t5 &  & BRAF   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr7  & 140419127 & 140625729 &  0 & -  & gene\\_plus\\_promoter\\\\\n",
       "\t6 &  & AR     &  &  1 & Kukkonen et al. 2021 & hypermethylated intergenic regions  & chrX  &  66762863 &  66950461 &  0 & +  & gene\\_plus\\_promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | <!----> | CD24   | <!----> | NA | Lu et al 2022        | hypermethylated                     | NA    |        NA |        NA | NA | NA | NA                 |\n",
       "| 2 | <!----> | TET2   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr4  | 106066032 | 106200973 |  0 | +  | gene_plus_promoter |\n",
       "| 3 | <!----> | DNMT3B | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr20 |  31349191 |  31397165 |  0 | +  | gene_plus_promoter |\n",
       "| 4 | <!----> | IDH1   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr2  | 209100953 | 209131798 |  0 | -  | gene_plus_promoter |\n",
       "| 5 | <!----> | BRAF   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr7  | 140419127 | 140625729 |  0 | -  | gene_plus_promoter |\n",
       "| 6 | <!----> | AR     | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated intergenic regions  | chrX  |  66762863 |  66950461 |  0 | +  | gene_plus_promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe No..pubs Source              \n",
       "1        CD24         NA       Lu et al 2022       \n",
       "2        TET2          1       Kukkonen et al. 2021\n",
       "3        DNMT3B        1       Kukkonen et al. 2021\n",
       "4        IDH1          1       Kukkonen et al. 2021\n",
       "5        BRAF          1       Kukkonen et al. 2021\n",
       "6        AR            1       Kukkonen et al. 2021\n",
       "  note                                chr   start     end       score strand\n",
       "1 hypermethylated                     NA           NA        NA NA    NA    \n",
       "2 hypermethylated subtype             chr4  106066032 106200973  0    +     \n",
       "3 hypermethylated subtype             chr20  31349191  31397165  0    +     \n",
       "4 hypermethylated subtype             chr2  209100953 209131798  0    -     \n",
       "5 hypermethylated subtype             chr7  140419127 140625729  0    -     \n",
       "6 hypermethylated intergenic regions  chrX   66762863  66950461  0    +     \n",
       "  gencode.type      \n",
       "1 NA                \n",
       "2 gene_plus_promoter\n",
       "3 gene_plus_promoter\n",
       "4 gene_plus_promoter\n",
       "5 gene_plus_promoter\n",
       "6 gene_plus_promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# annotate genes using only the first match from the gene table (should only be one match per gene anyway)\n",
    "df_pub_genes_annot = df_pub_genes %>% left_join(df_gene_coords %>% distinct(gene_name, .keep_all=TRUE), by=c(\"Gene\"=\"gene_name\"))\n",
    "head(df_pub_genes_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "21d5864f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>promoter</td><td>CD24 </td><td></td><td>1</td><td>Tolkach et al . 2021</td><td>hypermethylated</td><td>NA</td><td>       NA</td><td>       NA</td><td>NA</td><td>NA</td><td>NA      </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>promoter</td><td>HLA-A</td><td></td><td>1</td><td>Rodems et al 2021   </td><td>hypermethylated</td><td>6 </td><td> 29908037</td><td> 29909037</td><td> 0</td><td>+ </td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>promoter</td><td>HLA-B</td><td></td><td>1</td><td>Rodems et al 2021   </td><td>hypermethylated</td><td>6 </td><td> 31334844</td><td> 31335844</td><td> 0</td><td>- </td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>promoter</td><td>HLA-C</td><td></td><td>1</td><td>Rodems et al 2021   </td><td>hypermethylated</td><td>6 </td><td> 31239907</td><td> 31240907</td><td> 0</td><td>- </td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>promoter</td><td>HIF3A</td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated</td><td>19</td><td> 46799299</td><td> 46800299</td><td> 0</td><td>+ </td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>promoter</td><td>HOXD3</td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated</td><td>2 </td><td>177000340</td><td>177001340</td><td> 0</td><td>+ </td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & start & end & score & strand & gencode.type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & promoter & CD24  &  & 1 & Tolkach et al . 2021 & hypermethylated & NA &        NA &        NA & NA & NA & NA      \\\\\n",
       "\t2 & promoter & HLA-A &  & 1 & Rodems et al 2021    & hypermethylated & 6  &  29908037 &  29909037 &  0 & +  & promoter\\\\\n",
       "\t3 & promoter & HLA-B &  & 1 & Rodems et al 2021    & hypermethylated & 6  &  31334844 &  31335844 &  0 & -  & promoter\\\\\n",
       "\t4 & promoter & HLA-C &  & 1 & Rodems et al 2021    & hypermethylated & 6  &  31239907 &  31240907 &  0 & -  & promoter\\\\\n",
       "\t5 & promoter & HIF3A &  & 1 & Kukkonen et al. 2021 & hypermethylated & 19 &  46799299 &  46800299 &  0 & +  & promoter\\\\\n",
       "\t6 & promoter & HOXD3 &  & 1 & Kukkonen et al. 2021 & hypermethylated & 2  & 177000340 & 177001340 &  0 & +  & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | promoter | CD24  | <!----> | 1 | Tolkach et al . 2021 | hypermethylated | NA |        NA |        NA | NA | NA | NA       |\n",
       "| 2 | promoter | HLA-A | <!----> | 1 | Rodems et al 2021    | hypermethylated | 6  |  29908037 |  29909037 |  0 | +  | promoter |\n",
       "| 3 | promoter | HLA-B | <!----> | 1 | Rodems et al 2021    | hypermethylated | 6  |  31334844 |  31335844 |  0 | -  | promoter |\n",
       "| 4 | promoter | HLA-C | <!----> | 1 | Rodems et al 2021    | hypermethylated | 6  |  31239907 |  31240907 |  0 | -  | promoter |\n",
       "| 5 | promoter | HIF3A | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated | 19 |  46799299 |  46800299 |  0 | +  | promoter |\n",
       "| 6 | promoter | HOXD3 | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated | 2  | 177000340 | 177001340 |  0 | +  | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  region   Gene  probe No..pubs Source               note            chr\n",
       "1 promoter CD24        1        Tolkach et al . 2021 hypermethylated NA \n",
       "2 promoter HLA-A       1        Rodems et al 2021    hypermethylated 6  \n",
       "3 promoter HLA-B       1        Rodems et al 2021    hypermethylated 6  \n",
       "4 promoter HLA-C       1        Rodems et al 2021    hypermethylated 6  \n",
       "5 promoter HIF3A       1        Kukkonen et al. 2021 hypermethylated 19 \n",
       "6 promoter HOXD3       1        Kukkonen et al. 2021 hypermethylated 2  \n",
       "  start     end       score strand gencode.type\n",
       "1        NA        NA NA    NA     NA          \n",
       "2  29908037  29909037  0    +      promoter    \n",
       "3  31334844  31335844  0    -      promoter    \n",
       "4  31239907  31240907  0    -      promoter    \n",
       "5  46799299  46800299  0    +      promoter    \n",
       "6 177000340 177001340  0    +      promoter    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#retieve promoter coords\n",
    "df_pub_prom_annot = df_pub_prom %>% left_join(df_promoter_coords %>% distinct(gene_name, .keep_all=TRUE), by=c(\"Gene\"=\"gene_name\"))\n",
    "head(df_pub_prom_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "b42adb09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>start</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td></td><td>KLF8  </td><td>cg03834574</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>X </td><td> 56259094</td><td>R</td><td>EPIC_probe</td><td> 56259093</td></tr>\n",
       "\t<tr><th scope=row>2</th><td></td><td>HOPX  </td><td>cg21899596</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>4 </td><td> 57522493</td><td>R</td><td>EPIC_probe</td><td> 57522492</td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>RARB  </td><td>cg24396624</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>3 </td><td> 25469860</td><td>R</td><td>EPIC_probe</td><td> 25469859</td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>HIF3A </td><td>cg21544503</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>19</td><td> 46800131</td><td>R</td><td>EPIC_probe</td><td> 46800130</td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>HIF3A </td><td>cg14117138</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>19</td><td> 46800085</td><td>R</td><td>EPIC_probe</td><td> 46800084</td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>CXCL14</td><td>cg10493994</td><td>1</td><td>Ylitalo et al 2021</td><td>Hypermethylated</td><td>5 </td><td>134914932</td><td>F</td><td>EPIC_probe</td><td>134914931</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & end & strand & type & start\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 &  & KLF8   & cg03834574 & 1 & Ylitalo et al 2021 & Hypermethylated & X  &  56259094 & R & EPIC\\_probe &  56259093\\\\\n",
       "\t2 &  & HOPX   & cg21899596 & 1 & Ylitalo et al 2021 & Hypermethylated & 4  &  57522493 & R & EPIC\\_probe &  57522492\\\\\n",
       "\t3 &  & RARB   & cg24396624 & 1 & Ylitalo et al 2021 & Hypermethylated & 3  &  25469860 & R & EPIC\\_probe &  25469859\\\\\n",
       "\t4 &  & HIF3A  & cg21544503 & 1 & Ylitalo et al 2021 & Hypermethylated & 19 &  46800131 & R & EPIC\\_probe &  46800130\\\\\n",
       "\t5 &  & HIF3A  & cg14117138 & 1 & Ylitalo et al 2021 & Hypermethylated & 19 &  46800085 & R & EPIC\\_probe &  46800084\\\\\n",
       "\t6 &  & CXCL14 & cg10493994 & 1 & Ylitalo et al 2021 & Hypermethylated & 5  & 134914932 & F & EPIC\\_probe & 134914931\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | end &lt;int&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | start &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | <!----> | KLF8   | cg03834574 | 1 | Ylitalo et al 2021 | Hypermethylated | X  |  56259094 | R | EPIC_probe |  56259093 |\n",
       "| 2 | <!----> | HOPX   | cg21899596 | 1 | Ylitalo et al 2021 | Hypermethylated | 4  |  57522493 | R | EPIC_probe |  57522492 |\n",
       "| 3 | <!----> | RARB   | cg24396624 | 1 | Ylitalo et al 2021 | Hypermethylated | 3  |  25469860 | R | EPIC_probe |  25469859 |\n",
       "| 4 | <!----> | HIF3A  | cg21544503 | 1 | Ylitalo et al 2021 | Hypermethylated | 19 |  46800131 | R | EPIC_probe |  46800130 |\n",
       "| 5 | <!----> | HIF3A  | cg14117138 | 1 | Ylitalo et al 2021 | Hypermethylated | 19 |  46800085 | R | EPIC_probe |  46800084 |\n",
       "| 6 | <!----> | CXCL14 | cg10493994 | 1 | Ylitalo et al 2021 | Hypermethylated | 5  | 134914932 | F | EPIC_probe | 134914931 |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe      No..pubs Source             note            chr\n",
       "1        KLF8   cg03834574 1        Ylitalo et al 2021 Hypermethylated X  \n",
       "2        HOPX   cg21899596 1        Ylitalo et al 2021 Hypermethylated 4  \n",
       "3        RARB   cg24396624 1        Ylitalo et al 2021 Hypermethylated 3  \n",
       "4        HIF3A  cg21544503 1        Ylitalo et al 2021 Hypermethylated 19 \n",
       "5        HIF3A  cg14117138 1        Ylitalo et al 2021 Hypermethylated 19 \n",
       "6        CXCL14 cg10493994 1        Ylitalo et al 2021 Hypermethylated 5  \n",
       "  end       strand type       start    \n",
       "1  56259094 R      EPIC_probe  56259093\n",
       "2  57522493 R      EPIC_probe  57522492\n",
       "3  25469860 R      EPIC_probe  25469859\n",
       "4  46800131 R      EPIC_probe  46800130\n",
       "5  46800085 R      EPIC_probe  46800084\n",
       "6 134914932 F      EPIC_probe 134914931"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df_pub_probes_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "e21dd800",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'region'</li><li>'Gene'</li><li>'probe'</li><li>'No..pubs'</li><li>'Source'</li><li>'note'</li><li>'chr'</li><li>'start'</li><li>'end'</li><li>'score'</li><li>'strand'</li><li>'gencode.type'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'region'\n",
       "\\item 'Gene'\n",
       "\\item 'probe'\n",
       "\\item 'No..pubs'\n",
       "\\item 'Source'\n",
       "\\item 'note'\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'score'\n",
       "\\item 'strand'\n",
       "\\item 'gencode.type'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'region'\n",
       "2. 'Gene'\n",
       "3. 'probe'\n",
       "4. 'No..pubs'\n",
       "5. 'Source'\n",
       "6. 'note'\n",
       "7. 'chr'\n",
       "8. 'start'\n",
       "9. 'end'\n",
       "10. 'score'\n",
       "11. 'strand'\n",
       "12. 'gencode.type'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"region\"       \"Gene\"         \"probe\"        \"No..pubs\"     \"Source\"      \n",
       " [6] \"note\"         \"chr\"          \"start\"        \"end\"          \"score\"       \n",
       "[11] \"strand\"       \"gencode.type\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'region'</li><li>'Gene'</li><li>'probe'</li><li>'No..pubs'</li><li>'Source'</li><li>'note'</li><li>'chr'</li><li>'end'</li><li>'strand'</li><li>'gencode.type'</li><li>NA</li><li>NA</li><li>'start'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'region'\n",
       "\\item 'Gene'\n",
       "\\item 'probe'\n",
       "\\item 'No..pubs'\n",
       "\\item 'Source'\n",
       "\\item 'note'\n",
       "\\item 'chr'\n",
       "\\item 'end'\n",
       "\\item 'strand'\n",
       "\\item 'gencode.type'\n",
       "\\item NA\n",
       "\\item NA\n",
       "\\item 'start'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'region'\n",
       "2. 'Gene'\n",
       "3. 'probe'\n",
       "4. 'No..pubs'\n",
       "5. 'Source'\n",
       "6. 'note'\n",
       "7. 'chr'\n",
       "8. 'end'\n",
       "9. 'strand'\n",
       "10. 'gencode.type'\n",
       "11. NA\n",
       "12. NA\n",
       "13. 'start'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"region\"       \"Gene\"         \"probe\"        \"No..pubs\"     \"Source\"      \n",
       " [6] \"note\"         \"chr\"          \"end\"          \"strand\"       \"gencode.type\"\n",
       "[11] NA             NA             \"start\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'region'</li><li>'Gene'</li><li>'probe'</li><li>'No..pubs'</li><li>'Source'</li><li>'note'</li><li>'chr'</li><li>'start'</li><li>'end'</li><li>'score'</li><li>'strand'</li><li>'gencode.type'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'region'\n",
       "\\item 'Gene'\n",
       "\\item 'probe'\n",
       "\\item 'No..pubs'\n",
       "\\item 'Source'\n",
       "\\item 'note'\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'score'\n",
       "\\item 'strand'\n",
       "\\item 'gencode.type'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'region'\n",
       "2. 'Gene'\n",
       "3. 'probe'\n",
       "4. 'No..pubs'\n",
       "5. 'Source'\n",
       "6. 'note'\n",
       "7. 'chr'\n",
       "8. 'start'\n",
       "9. 'end'\n",
       "10. 'score'\n",
       "11. 'strand'\n",
       "12. 'gencode.type'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"region\"       \"Gene\"         \"probe\"        \"No..pubs\"     \"Source\"      \n",
       " [6] \"note\"         \"chr\"          \"start\"        \"end\"          \"score\"       \n",
       "[11] \"strand\"       \"gencode.type\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pub_probes_annot$gencode.type = \"EPIC_probe\"\n",
    "colnames(df_pub_probes_annot) <- c(\"region\",\"Gene\",\"probe\",\"No..pubs\",\"Source\",\"note\",\n",
    "                             \"chr\",\"end\",\"strand\",\"gencode.type\")\n",
    "df_pub_probes_annot$start = as.numeric(df_pub_probes_annot$end) - 1\n",
    "\n",
    "colnames(df_pub_genes_annot)\n",
    "colnames(df_pub_probes_annot)\n",
    "colnames(df_pub_prom_annot)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "1159324d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td></td><td>CD24  </td><td></td><td>NA</td><td>Lu et al 2022       </td><td>hypermethylated                    </td><td>NA   </td><td>       NA</td><td>       NA</td><td>NA</td><td>NA                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td></td><td>TET2  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr4 </td><td>106066032</td><td>106200973</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>DNMT3B</td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr20</td><td> 31349191</td><td> 31397165</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>IDH1  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr2 </td><td>209100953</td><td>209131798</td><td>- </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>BRAF  </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr7 </td><td>140419127</td><td>140625729</td><td>- </td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>AR    </td><td></td><td> 1</td><td>Kukkonen et al. 2021</td><td>hypermethylated intergenic regions </td><td>chrX </td><td> 66762863</td><td> 66950461</td><td>+ </td><td>gene_plus_promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & start & end & strand & gencode.type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 &  & CD24   &  & NA & Lu et al 2022        & hypermethylated                     & NA    &        NA &        NA & NA & NA                \\\\\n",
       "\t2 &  & TET2   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr4  & 106066032 & 106200973 & +  & gene\\_plus\\_promoter\\\\\n",
       "\t3 &  & DNMT3B &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr20 &  31349191 &  31397165 & +  & gene\\_plus\\_promoter\\\\\n",
       "\t4 &  & IDH1   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr2  & 209100953 & 209131798 & -  & gene\\_plus\\_promoter\\\\\n",
       "\t5 &  & BRAF   &  &  1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr7  & 140419127 & 140625729 & -  & gene\\_plus\\_promoter\\\\\n",
       "\t6 &  & AR     &  &  1 & Kukkonen et al. 2021 & hypermethylated intergenic regions  & chrX  &  66762863 &  66950461 & +  & gene\\_plus\\_promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | start &lt;dbl&gt; | end &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | <!----> | CD24   | <!----> | NA | Lu et al 2022        | hypermethylated                     | NA    |        NA |        NA | NA | NA                 |\n",
       "| 2 | <!----> | TET2   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr4  | 106066032 | 106200973 | +  | gene_plus_promoter |\n",
       "| 3 | <!----> | DNMT3B | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr20 |  31349191 |  31397165 | +  | gene_plus_promoter |\n",
       "| 4 | <!----> | IDH1   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr2  | 209100953 | 209131798 | -  | gene_plus_promoter |\n",
       "| 5 | <!----> | BRAF   | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr7  | 140419127 | 140625729 | -  | gene_plus_promoter |\n",
       "| 6 | <!----> | AR     | <!----> |  1 | Kukkonen et al. 2021 | hypermethylated intergenic regions  | chrX  |  66762863 |  66950461 | +  | gene_plus_promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe No..pubs Source              \n",
       "1        CD24         NA       Lu et al 2022       \n",
       "2        TET2          1       Kukkonen et al. 2021\n",
       "3        DNMT3B        1       Kukkonen et al. 2021\n",
       "4        IDH1          1       Kukkonen et al. 2021\n",
       "5        BRAF          1       Kukkonen et al. 2021\n",
       "6        AR            1       Kukkonen et al. 2021\n",
       "  note                                chr   start     end       strand\n",
       "1 hypermethylated                     NA           NA        NA NA    \n",
       "2 hypermethylated subtype             chr4  106066032 106200973 +     \n",
       "3 hypermethylated subtype             chr20  31349191  31397165 +     \n",
       "4 hypermethylated subtype             chr2  209100953 209131798 -     \n",
       "5 hypermethylated subtype             chr7  140419127 140625729 -     \n",
       "6 hypermethylated intergenic regions  chrX   66762863  66950461 +     \n",
       "  gencode.type      \n",
       "1 NA                \n",
       "2 gene_plus_promoter\n",
       "3 gene_plus_promoter\n",
       "4 gene_plus_promoter\n",
       "5 gene_plus_promoter\n",
       "6 gene_plus_promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# put the full pub table back together\n",
    "cols = c(\"region\",\"Gene\",\"probe\",\"No..pubs\",\"Source\",\"note\",\n",
    "                             \"chr\",\"start\",\"end\",\"strand\",\"gencode.type\")\n",
    "df_pubs_annot = rbind(df_pub_genes_annot[,cols], df_pub_probes_annot[,cols], df_pub_prom_annot[,cols])\n",
    "head(df_pubs_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "7bf6c8d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>gencode.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>9</th><td>        </td><td>GSTP1</td><td>          </td><td> 1</td><td>Constancio et al. 2019</td><td>early detection</td><td>chr11</td><td>67350213</td><td>67354127</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>        </td><td>GSTP1</td><td>          </td><td>12</td><td>Massie et al. 2017    </td><td>               </td><td>chr11</td><td>67350213</td><td>67354127</td><td>+</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>5772</th><td>        </td><td>GSTP1</td><td>cg02659086</td><td> 1</td><td>Ylitalo et al 2021    </td><td>Hypermethylated</td><td>11   </td><td>67350975</td><td>67350976</td><td>R</td><td>EPIC_probe        </td></tr>\n",
       "\t<tr><th scope=row>7118</th><td>        </td><td>GSTP1</td><td>cg04920951</td><td> 1</td><td>Ylitalo et al 2021    </td><td>Hypermethylated</td><td>11   </td><td>67351270</td><td>67351271</td><td>R</td><td>EPIC_probe        </td></tr>\n",
       "\t<tr><th scope=row>7557</th><td>        </td><td>GSTP1</td><td>cg26250609</td><td> 1</td><td>Ylitalo et al 2021    </td><td>Hypermethylated</td><td>11   </td><td>67351272</td><td>67351273</td><td>F</td><td>EPIC_probe        </td></tr>\n",
       "\t<tr><th scope=row>9990</th><td>promoter</td><td>GSTP1</td><td>          </td><td> 1</td><td>Sugiura et al 2021    </td><td>hypermethylated</td><td>11   </td><td>67350213</td><td>67351213</td><td>+</td><td>promoter          </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & start & end & strand & gencode.type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t9 &          & GSTP1 &            &  1 & Constancio et al. 2019 & early detection & chr11 & 67350213 & 67354127 & + & gene\\_plus\\_promoter\\\\\n",
       "\t23 &          & GSTP1 &            & 12 & Massie et al. 2017     &                 & chr11 & 67350213 & 67354127 & + & gene\\_plus\\_promoter\\\\\n",
       "\t5772 &          & GSTP1 & cg02659086 &  1 & Ylitalo et al 2021     & Hypermethylated & 11    & 67350975 & 67350976 & R & EPIC\\_probe        \\\\\n",
       "\t7118 &          & GSTP1 & cg04920951 &  1 & Ylitalo et al 2021     & Hypermethylated & 11    & 67351270 & 67351271 & R & EPIC\\_probe        \\\\\n",
       "\t7557 &          & GSTP1 & cg26250609 &  1 & Ylitalo et al 2021     & Hypermethylated & 11    & 67351272 & 67351273 & F & EPIC\\_probe        \\\\\n",
       "\t9990 & promoter & GSTP1 &            &  1 & Sugiura et al 2021     & hypermethylated & 11    & 67350213 & 67351213 & + & promoter          \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | start &lt;dbl&gt; | end &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 9 | <!----> | GSTP1 | <!----> |  1 | Constancio et al. 2019 | early detection | chr11 | 67350213 | 67354127 | + | gene_plus_promoter |\n",
       "| 23 | <!----> | GSTP1 | <!----> | 12 | Massie et al. 2017     | <!----> | chr11 | 67350213 | 67354127 | + | gene_plus_promoter |\n",
       "| 5772 | <!----> | GSTP1 | cg02659086 |  1 | Ylitalo et al 2021     | Hypermethylated | 11    | 67350975 | 67350976 | R | EPIC_probe         |\n",
       "| 7118 | <!----> | GSTP1 | cg04920951 |  1 | Ylitalo et al 2021     | Hypermethylated | 11    | 67351270 | 67351271 | R | EPIC_probe         |\n",
       "| 7557 | <!----> | GSTP1 | cg26250609 |  1 | Ylitalo et al 2021     | Hypermethylated | 11    | 67351272 | 67351273 | F | EPIC_probe         |\n",
       "| 9990 | promoter | GSTP1 | <!----> |  1 | Sugiura et al 2021     | hypermethylated | 11    | 67350213 | 67351213 | + | promoter           |\n",
       "\n"
      ],
      "text/plain": [
       "     region   Gene  probe      No..pubs Source                 note           \n",
       "9             GSTP1             1       Constancio et al. 2019 early detection\n",
       "23            GSTP1            12       Massie et al. 2017                    \n",
       "5772          GSTP1 cg02659086  1       Ylitalo et al 2021     Hypermethylated\n",
       "7118          GSTP1 cg04920951  1       Ylitalo et al 2021     Hypermethylated\n",
       "7557          GSTP1 cg26250609  1       Ylitalo et al 2021     Hypermethylated\n",
       "9990 promoter GSTP1             1       Sugiura et al 2021     hypermethylated\n",
       "     chr   start    end      strand gencode.type      \n",
       "9    chr11 67350213 67354127 +      gene_plus_promoter\n",
       "23   chr11 67350213 67354127 +      gene_plus_promoter\n",
       "5772 11    67350975 67350976 R      EPIC_probe        \n",
       "7118 11    67351270 67351271 R      EPIC_probe        \n",
       "7557 11    67351272 67351273 F      EPIC_probe        \n",
       "9990 11    67350213 67351213 +      promoter          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pubs_annot[df_pubs_annot$Gene==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "974c2b89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "9058"
      ],
      "text/latex": [
       "9058"
      ],
      "text/markdown": [
       "9058"
      ],
      "text/plain": [
       "[1] 9058"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pubs_annot$No..pubs[is.na(df_pubs_annot$No..pubs)] <- 1\n",
    "df_pubs_annot <- df_pubs_annot[!is.na(df_pubs_annot$chr),]\n",
    "df_pubs_annot$pub.reg.type <- df_pubs_annot$gencode.type\n",
    "nrow(df_pubs_annot)\n",
    "                             \n",
    "#df_pubs_annot = df_pubs_annot[,c(\"chr\",\"start\",\"end\",\"Gene\",\"probe\",\"strand\",\"Source\",\"type\",\"note\",\"No..pubs\")]\n",
    "write.table(df_pubs_annot, file=\"./marker_collation/study_gene_regions_wcoords_hg19.bed\", sep=\"\\t\",\n",
    "           quote=FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d040a745",
   "metadata": {},
   "source": [
    "## join all data, reannotate gene names, count pub no. per gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "7b686b94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'region'</li><li>'Gene'</li><li>'probe'</li><li>'No..pubs'</li><li>'Source'</li><li>'note'</li><li>'chr'</li><li>'start'</li><li>'end'</li><li>'strand'</li><li>'gencode.type'</li><li>'pub.reg.type'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'region'\n",
       "\\item 'Gene'\n",
       "\\item 'probe'\n",
       "\\item 'No..pubs'\n",
       "\\item 'Source'\n",
       "\\item 'note'\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'strand'\n",
       "\\item 'gencode.type'\n",
       "\\item 'pub.reg.type'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'region'\n",
       "2. 'Gene'\n",
       "3. 'probe'\n",
       "4. 'No..pubs'\n",
       "5. 'Source'\n",
       "6. 'note'\n",
       "7. 'chr'\n",
       "8. 'start'\n",
       "9. 'end'\n",
       "10. 'strand'\n",
       "11. 'gencode.type'\n",
       "12. 'pub.reg.type'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"region\"       \"Gene\"         \"probe\"        \"No..pubs\"     \"Source\"      \n",
       " [6] \"note\"         \"chr\"          \"start\"        \"end\"          \"strand\"      \n",
       "[11] \"gencode.type\" \"pub.reg.type\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>Gene</th><th scope=col>probe</th><th scope=col>No..pubs</th><th scope=col>Source</th><th scope=col>note</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>gencode.type</th><th scope=col>pub.reg.type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>2</th><td></td><td>TET2  </td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr4 </td><td>106066032</td><td>106200973</td><td>+</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td></td><td>DNMT3B</td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr20</td><td> 31349191</td><td> 31397165</td><td>+</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td></td><td>IDH1  </td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr2 </td><td>209100953</td><td>209131798</td><td>-</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td></td><td>BRAF  </td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated subtype            </td><td>chr7 </td><td>140419127</td><td>140625729</td><td>-</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td></td><td>AR    </td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated intergenic regions </td><td>chrX </td><td> 66762863</td><td> 66950461</td><td>+</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "\t<tr><th scope=row>7</th><td></td><td>MYC   </td><td></td><td>1</td><td>Kukkonen et al. 2021</td><td>hypermethylated intergenic regions </td><td>chr8 </td><td>128746680</td><td>128755197</td><td>+</td><td>gene_plus_promoter</td><td>gene_plus_promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & region & Gene & probe & No..pubs & Source & note & chr & start & end & strand & gencode.type & pub.reg.type\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t2 &  & TET2   &  & 1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr4  & 106066032 & 106200973 & + & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\t3 &  & DNMT3B &  & 1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr20 &  31349191 &  31397165 & + & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\t4 &  & IDH1   &  & 1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr2  & 209100953 & 209131798 & - & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\t5 &  & BRAF   &  & 1 & Kukkonen et al. 2021 & hypermethylated subtype             & chr7  & 140419127 & 140625729 & - & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\t6 &  & AR     &  & 1 & Kukkonen et al. 2021 & hypermethylated intergenic regions  & chrX  &  66762863 &  66950461 & + & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\t7 &  & MYC    &  & 1 & Kukkonen et al. 2021 & hypermethylated intergenic regions  & chr8  & 128746680 & 128755197 & + & gene\\_plus\\_promoter & gene\\_plus\\_promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | region &lt;chr&gt; | Gene &lt;chr&gt; | probe &lt;chr&gt; | No..pubs &lt;int&gt; | Source &lt;chr&gt; | note &lt;chr&gt; | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | gencode.type &lt;chr&gt; | pub.reg.type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 2 | <!----> | TET2   | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr4  | 106066032 | 106200973 | + | gene_plus_promoter | gene_plus_promoter |\n",
       "| 3 | <!----> | DNMT3B | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr20 |  31349191 |  31397165 | + | gene_plus_promoter | gene_plus_promoter |\n",
       "| 4 | <!----> | IDH1   | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr2  | 209100953 | 209131798 | - | gene_plus_promoter | gene_plus_promoter |\n",
       "| 5 | <!----> | BRAF   | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated subtype             | chr7  | 140419127 | 140625729 | - | gene_plus_promoter | gene_plus_promoter |\n",
       "| 6 | <!----> | AR     | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated intergenic regions  | chrX  |  66762863 |  66950461 | + | gene_plus_promoter | gene_plus_promoter |\n",
       "| 7 | <!----> | MYC    | <!----> | 1 | Kukkonen et al. 2021 | hypermethylated intergenic regions  | chr8  | 128746680 | 128755197 | + | gene_plus_promoter | gene_plus_promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  region Gene   probe No..pubs Source              \n",
       "2        TET2         1        Kukkonen et al. 2021\n",
       "3        DNMT3B       1        Kukkonen et al. 2021\n",
       "4        IDH1         1        Kukkonen et al. 2021\n",
       "5        BRAF         1        Kukkonen et al. 2021\n",
       "6        AR           1        Kukkonen et al. 2021\n",
       "7        MYC          1        Kukkonen et al. 2021\n",
       "  note                                chr   start     end       strand\n",
       "2 hypermethylated subtype             chr4  106066032 106200973 +     \n",
       "3 hypermethylated subtype             chr20  31349191  31397165 +     \n",
       "4 hypermethylated subtype             chr2  209100953 209131798 -     \n",
       "5 hypermethylated subtype             chr7  140419127 140625729 -     \n",
       "6 hypermethylated intergenic regions  chrX   66762863  66950461 +     \n",
       "7 hypermethylated intergenic regions  chr8  128746680 128755197 +     \n",
       "  gencode.type       pub.reg.type      \n",
       "2 gene_plus_promoter gene_plus_promoter\n",
       "3 gene_plus_promoter gene_plus_promoter\n",
       "4 gene_plus_promoter gene_plus_promoter\n",
       "5 gene_plus_promoter gene_plus_promoter\n",
       "6 gene_plus_promoter gene_plus_promoter\n",
       "7 gene_plus_promoter gene_plus_promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>9058</li><li>12</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 9058\n",
       "\\item 12\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 9058\n",
       "2. 12\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 9058   12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pubs = read.table(file=\"./marker_collation//study_gene_regions_wcoords_hg19.bed\",sep=\"\\t\",header=TRUE)\n",
    "names(df_pubs)\n",
    "#colnames(df_pubs) <- c(\"chr\",\"start\",\"end\",\"pub.gene.name\",\"probe\",\"strand\",\"source\",\"pub.reg.type\",\"notes\",\"no.pubs\")\n",
    "head(df_pubs)\n",
    "dim(df_pubs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "415cbbcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>dot1</th><th scope=col>dot2</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>notes</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>no.pubs</th><th scope=col>probe</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>833806</td><td>834358</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hyper-methylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>841565</td><td>843221</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hyper-methylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>861132</td><td>861612</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hyper-methylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>863044</td><td>863496</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hypo-methylated </td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>868475</td><td>869290</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hypo-methylated </td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>906156</td><td>907273</td><td>.</td><td>.</td><td>.</td><td>Beltran et al 2020</td><td>Hyper-methylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & chr & start & end & dot1 & dot2 & strand & source & notes & pub.gene.name & pub.reg.type & no.pubs & probe\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <chr> & <dbl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 833806 & 834358 & . & . & . & Beltran et al 2020 & Hyper-methylated & NA & DMR & 1 & NA\\\\\n",
       "\t2 & chr1 & 841565 & 843221 & . & . & . & Beltran et al 2020 & Hyper-methylated & NA & DMR & 1 & NA\\\\\n",
       "\t3 & chr1 & 861132 & 861612 & . & . & . & Beltran et al 2020 & Hyper-methylated & NA & DMR & 1 & NA\\\\\n",
       "\t4 & chr1 & 863044 & 863496 & . & . & . & Beltran et al 2020 & Hypo-methylated  & NA & DMR & 1 & NA\\\\\n",
       "\t5 & chr1 & 868475 & 869290 & . & . & . & Beltran et al 2020 & Hypo-methylated  & NA & DMR & 1 & NA\\\\\n",
       "\t6 & chr1 & 906156 & 907273 & . & . & . & Beltran et al 2020 & Hyper-methylated & NA & DMR & 1 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | dot1 &lt;chr&gt; | dot2 &lt;chr&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | notes &lt;chr&gt; | pub.gene.name &lt;lgl&gt; | pub.reg.type &lt;chr&gt; | no.pubs &lt;dbl&gt; | probe &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 833806 | 834358 | . | . | . | Beltran et al 2020 | Hyper-methylated | NA | DMR | 1 | NA |\n",
       "| 2 | chr1 | 841565 | 843221 | . | . | . | Beltran et al 2020 | Hyper-methylated | NA | DMR | 1 | NA |\n",
       "| 3 | chr1 | 861132 | 861612 | . | . | . | Beltran et al 2020 | Hyper-methylated | NA | DMR | 1 | NA |\n",
       "| 4 | chr1 | 863044 | 863496 | . | . | . | Beltran et al 2020 | Hypo-methylated  | NA | DMR | 1 | NA |\n",
       "| 5 | chr1 | 868475 | 869290 | . | . | . | Beltran et al 2020 | Hypo-methylated  | NA | DMR | 1 | NA |\n",
       "| 6 | chr1 | 906156 | 907273 | . | . | . | Beltran et al 2020 | Hyper-methylated | NA | DMR | 1 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  start  end    dot1 dot2 strand source             notes           \n",
       "1 chr1 833806 834358 .    .    .      Beltran et al 2020 Hyper-methylated\n",
       "2 chr1 841565 843221 .    .    .      Beltran et al 2020 Hyper-methylated\n",
       "3 chr1 861132 861612 .    .    .      Beltran et al 2020 Hyper-methylated\n",
       "4 chr1 863044 863496 .    .    .      Beltran et al 2020 Hypo-methylated \n",
       "5 chr1 868475 869290 .    .    .      Beltran et al 2020 Hypo-methylated \n",
       "6 chr1 906156 907273 .    .    .      Beltran et al 2020 Hyper-methylated\n",
       "  pub.gene.name pub.reg.type no.pubs probe\n",
       "1 NA            DMR          1       NA   \n",
       "2 NA            DMR          1       NA   \n",
       "3 NA            DMR          1       NA   \n",
       "4 NA            DMR          1       NA   \n",
       "5 NA            DMR          1       NA   \n",
       "6 NA            DMR          1       NA   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_beltran = read.table(file=\"./marker_collation//Beltran_2020_DMRs_hg19.bed\", sep=\"\\t\", header=FALSE)\n",
    "colnames(df_beltran) <- c(\"chr\",\"start\",\"end\",\"dot1\",\"dot2\",\"strand\",\"source\",\"notes\")\n",
    "df_beltran$pub.gene.name <- NA\n",
    "df_beltran$pub.reg.type <- \"DMR\"\n",
    "df_beltran$no.pubs <- 1\n",
    "df_beltran$probe <- NA\n",
    "head(df_beltran)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "0066fa8b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>source</th><th scope=col>status</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>no.pubs</th><th scope=col>probe</th><th scope=col>strand</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td> 29194</td><td> 29436</td><td>Zhao et al 2020</td><td>hypomethylated </td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td> 51588</td><td> 51682</td><td>Zhao et al 2020</td><td>hypomethylated </td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>535828</td><td>535879</td><td>Zhao et al 2020</td><td>hypomethylated </td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>566459</td><td>566739</td><td>Zhao et al 2020</td><td>hypermethylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>566813</td><td>566880</td><td>Zhao et al 2020</td><td>hypermethylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>567113</td><td>567239</td><td>Zhao et al 2020</td><td>hypermethylated</td><td>NA</td><td>DMR</td><td>1</td><td>NA</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & chr & start & end & source & status & pub.gene.name & pub.reg.type & no.pubs & probe & strand\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <lgl> & <chr> & <dbl> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 &  29194 &  29436 & Zhao et al 2020 & hypomethylated  & NA & DMR & 1 & NA & .\\\\\n",
       "\t2 & chr1 &  51588 &  51682 & Zhao et al 2020 & hypomethylated  & NA & DMR & 1 & NA & .\\\\\n",
       "\t3 & chr1 & 535828 & 535879 & Zhao et al 2020 & hypomethylated  & NA & DMR & 1 & NA & .\\\\\n",
       "\t4 & chr1 & 566459 & 566739 & Zhao et al 2020 & hypermethylated & NA & DMR & 1 & NA & .\\\\\n",
       "\t5 & chr1 & 566813 & 566880 & Zhao et al 2020 & hypermethylated & NA & DMR & 1 & NA & .\\\\\n",
       "\t6 & chr1 & 567113 & 567239 & Zhao et al 2020 & hypermethylated & NA & DMR & 1 & NA & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | source &lt;chr&gt; | status &lt;chr&gt; | pub.gene.name &lt;lgl&gt; | pub.reg.type &lt;chr&gt; | no.pubs &lt;dbl&gt; | probe &lt;lgl&gt; | strand &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 |  29194 |  29436 | Zhao et al 2020 | hypomethylated  | NA | DMR | 1 | NA | . |\n",
       "| 2 | chr1 |  51588 |  51682 | Zhao et al 2020 | hypomethylated  | NA | DMR | 1 | NA | . |\n",
       "| 3 | chr1 | 535828 | 535879 | Zhao et al 2020 | hypomethylated  | NA | DMR | 1 | NA | . |\n",
       "| 4 | chr1 | 566459 | 566739 | Zhao et al 2020 | hypermethylated | NA | DMR | 1 | NA | . |\n",
       "| 5 | chr1 | 566813 | 566880 | Zhao et al 2020 | hypermethylated | NA | DMR | 1 | NA | . |\n",
       "| 6 | chr1 | 567113 | 567239 | Zhao et al 2020 | hypermethylated | NA | DMR | 1 | NA | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  start  end    source          status          pub.gene.name pub.reg.type\n",
       "1 chr1  29194  29436 Zhao et al 2020 hypomethylated  NA            DMR         \n",
       "2 chr1  51588  51682 Zhao et al 2020 hypomethylated  NA            DMR         \n",
       "3 chr1 535828 535879 Zhao et al 2020 hypomethylated  NA            DMR         \n",
       "4 chr1 566459 566739 Zhao et al 2020 hypermethylated NA            DMR         \n",
       "5 chr1 566813 566880 Zhao et al 2020 hypermethylated NA            DMR         \n",
       "6 chr1 567113 567239 Zhao et al 2020 hypermethylated NA            DMR         \n",
       "  no.pubs probe strand\n",
       "1 1       NA    .     \n",
       "2 1       NA    .     \n",
       "3 1       NA    .     \n",
       "4 1       NA    .     \n",
       "5 1       NA    .     \n",
       "6 1       NA    .     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_zhao = read.table(file=\"./marker_collation//Zhao_2020_DMRs_hg19.tsv\", sep=\"\\t\", header=FALSE)\n",
    "colnames(df_zhao) = c(\"chr\",\"start\",\"end\",\"source\",\"status\")\n",
    "df_zhao$pub.gene.name <- NA\n",
    "df_zhao$pub.reg.type <- \"DMR\"\n",
    "df_zhao$no.pubs <- 1\n",
    "df_zhao$probe <- NA\n",
    "df_zhao$strand <- \".\"\n",
    "df_zhao$chr = paste0(\"chr\",df_zhao$chr, sep=\"\")\n",
    "head(df_zhao)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "46fb8e8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'region'</li><li>'Gene'</li><li>'probe'</li><li>'No..pubs'</li><li>'Source'</li><li>'note'</li><li>'chr'</li><li>'start'</li><li>'end'</li><li>'strand'</li><li>'gencode.type'</li><li>'pub.reg.type'</li><li>'source'</li><li>'pub.gene.name'</li><li>'no.pubs'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'region'\n",
       "\\item 'Gene'\n",
       "\\item 'probe'\n",
       "\\item 'No..pubs'\n",
       "\\item 'Source'\n",
       "\\item 'note'\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'strand'\n",
       "\\item 'gencode.type'\n",
       "\\item 'pub.reg.type'\n",
       "\\item 'source'\n",
       "\\item 'pub.gene.name'\n",
       "\\item 'no.pubs'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'region'\n",
       "2. 'Gene'\n",
       "3. 'probe'\n",
       "4. 'No..pubs'\n",
       "5. 'Source'\n",
       "6. 'note'\n",
       "7. 'chr'\n",
       "8. 'start'\n",
       "9. 'end'\n",
       "10. 'strand'\n",
       "11. 'gencode.type'\n",
       "12. 'pub.reg.type'\n",
       "13. 'source'\n",
       "14. 'pub.gene.name'\n",
       "15. 'no.pubs'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"region\"        \"Gene\"          \"probe\"         \"No..pubs\"     \n",
       " [5] \"Source\"        \"note\"          \"chr\"           \"start\"        \n",
       " [9] \"end\"           \"strand\"        \"gencode.type\"  \"pub.reg.type\" \n",
       "[13] \"source\"        \"pub.gene.name\" \"no.pubs\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'chr'</li><li>'start'</li><li>'end'</li><li>'dot1'</li><li>'dot2'</li><li>'strand'</li><li>'source'</li><li>'notes'</li><li>'pub.gene.name'</li><li>'pub.reg.type'</li><li>'no.pubs'</li><li>'probe'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'dot1'\n",
       "\\item 'dot2'\n",
       "\\item 'strand'\n",
       "\\item 'source'\n",
       "\\item 'notes'\n",
       "\\item 'pub.gene.name'\n",
       "\\item 'pub.reg.type'\n",
       "\\item 'no.pubs'\n",
       "\\item 'probe'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'chr'\n",
       "2. 'start'\n",
       "3. 'end'\n",
       "4. 'dot1'\n",
       "5. 'dot2'\n",
       "6. 'strand'\n",
       "7. 'source'\n",
       "8. 'notes'\n",
       "9. 'pub.gene.name'\n",
       "10. 'pub.reg.type'\n",
       "11. 'no.pubs'\n",
       "12. 'probe'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"chr\"           \"start\"         \"end\"           \"dot1\"         \n",
       " [5] \"dot2\"          \"strand\"        \"source\"        \"notes\"        \n",
       " [9] \"pub.gene.name\" \"pub.reg.type\"  \"no.pubs\"       \"probe\"        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'chr'</li><li>'start'</li><li>'end'</li><li>'source'</li><li>'status'</li><li>'pub.gene.name'</li><li>'pub.reg.type'</li><li>'no.pubs'</li><li>'probe'</li><li>'strand'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'source'\n",
       "\\item 'status'\n",
       "\\item 'pub.gene.name'\n",
       "\\item 'pub.reg.type'\n",
       "\\item 'no.pubs'\n",
       "\\item 'probe'\n",
       "\\item 'strand'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'chr'\n",
       "2. 'start'\n",
       "3. 'end'\n",
       "4. 'source'\n",
       "5. 'status'\n",
       "6. 'pub.gene.name'\n",
       "7. 'pub.reg.type'\n",
       "8. 'no.pubs'\n",
       "9. 'probe'\n",
       "10. 'strand'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"chr\"           \"start\"         \"end\"           \"source\"       \n",
       " [5] \"status\"        \"pub.gene.name\" \"pub.reg.type\"  \"no.pubs\"      \n",
       " [9] \"probe\"         \"strand\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(df_pubs)\n",
    "colnames(df_beltran)\n",
    "colnames(df_zhao)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "2199597d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_pubs$source = df_pubs$Source\n",
    "df_pubs$pub.gene.name = df_pubs$Gene\n",
    "df_pubs$notes = df_pubs$notes\n",
    "df_pubs$no.pubs = df_pubs$No..pubs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "086ab816",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>2</th><td>4 </td><td>106066032</td><td>106200973</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>20</td><td> 31349191</td><td> 31397165</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>2 </td><td>209100953</td><td>209131798</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>7 </td><td>140419127</td><td>140625729</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>X </td><td> 66762863</td><td> 66950461</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>8 </td><td>128746680</td><td>128755197</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t2 & 4  & 106066032 & 106200973 & + & Kukkonen et al. 2021 & TET2   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t3 & 20 &  31349191 &  31397165 & + & Kukkonen et al. 2021 & DNMT3B & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t4 & 2  & 209100953 & 209131798 & - & Kukkonen et al. 2021 & IDH1   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t5 & 7  & 140419127 & 140625729 & - & Kukkonen et al. 2021 & BRAF   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t6 & X  &  66762863 &  66950461 & + & Kukkonen et al. 2021 & AR     & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t7 & 8  & 128746680 & 128755197 & + & Kukkonen et al. 2021 & MYC    & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 2 | 4  | 106066032 | 106200973 | + | Kukkonen et al. 2021 | TET2   | gene_plus_promoter | <!----> | 1 |\n",
       "| 3 | 20 |  31349191 |  31397165 | + | Kukkonen et al. 2021 | DNMT3B | gene_plus_promoter | <!----> | 1 |\n",
       "| 4 | 2  | 209100953 | 209131798 | - | Kukkonen et al. 2021 | IDH1   | gene_plus_promoter | <!----> | 1 |\n",
       "| 5 | 7  | 140419127 | 140625729 | - | Kukkonen et al. 2021 | BRAF   | gene_plus_promoter | <!----> | 1 |\n",
       "| 6 | X  |  66762863 |  66950461 | + | Kukkonen et al. 2021 | AR     | gene_plus_promoter | <!----> | 1 |\n",
       "| 7 | 8  | 128746680 | 128755197 | + | Kukkonen et al. 2021 | MYC    | gene_plus_promoter | <!----> | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start     end       strand source               pub.gene.name\n",
       "2 4   106066032 106200973 +      Kukkonen et al. 2021 TET2         \n",
       "3 20   31349191  31397165 +      Kukkonen et al. 2021 DNMT3B       \n",
       "4 2   209100953 209131798 -      Kukkonen et al. 2021 IDH1         \n",
       "5 7   140419127 140625729 -      Kukkonen et al. 2021 BRAF         \n",
       "6 X    66762863  66950461 +      Kukkonen et al. 2021 AR           \n",
       "7 8   128746680 128755197 +      Kukkonen et al. 2021 MYC          \n",
       "  pub.reg.type       probe no.pubs\n",
       "2 gene_plus_promoter       1      \n",
       "3 gene_plus_promoter       1      \n",
       "4 gene_plus_promoter       1      \n",
       "5 gene_plus_promoter       1      \n",
       "6 gene_plus_promoter       1      \n",
       "7 gene_plus_promoter       1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>120461</li><li>9</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 120461\n",
       "\\item 9\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 120461\n",
       "2. 9\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 120461      9"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols = c(\"chr\",\"start\",\"end\",\"strand\",\"source\",\"pub.gene.name\",\"pub.reg.type\",\"probe\",\"no.pubs\")\n",
    "df_all = rbind(df_pubs[,cols], df_beltran[,cols], df_zhao[,cols])\n",
    "df_all$chr = gsub(\"chr\",\"\",df_all$chr)\n",
    "head(df_all)\n",
    "dim(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "508369c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106066032</td><td>106200973</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31349191</td><td> 31397165</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209131798</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140625729</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66762863</td><td> 66950461</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128746680</td><td>128755197</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene_plus_promoter</td><td></td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106066032 & 106200973 & + & Kukkonen et al. 2021 & TET2   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t2 & 20 &  31349191 &  31397165 & + & Kukkonen et al. 2021 & DNMT3B & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t3 & 2  & 209100953 & 209131798 & - & Kukkonen et al. 2021 & IDH1   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t4 & 7  & 140419127 & 140625729 & - & Kukkonen et al. 2021 & BRAF   & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t5 & X  &  66762863 &  66950461 & + & Kukkonen et al. 2021 & AR     & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\t6 & 8  & 128746680 & 128755197 & + & Kukkonen et al. 2021 & MYC    & gene\\_plus\\_promoter &  & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106066032 | 106200973 | + | Kukkonen et al. 2021 | TET2   | gene_plus_promoter | <!----> | 1 |\n",
       "| 2 | 20 |  31349191 |  31397165 | + | Kukkonen et al. 2021 | DNMT3B | gene_plus_promoter | <!----> | 1 |\n",
       "| 3 | 2  | 209100953 | 209131798 | - | Kukkonen et al. 2021 | IDH1   | gene_plus_promoter | <!----> | 1 |\n",
       "| 4 | 7  | 140419127 | 140625729 | - | Kukkonen et al. 2021 | BRAF   | gene_plus_promoter | <!----> | 1 |\n",
       "| 5 | X  |  66762863 |  66950461 | + | Kukkonen et al. 2021 | AR     | gene_plus_promoter | <!----> | 1 |\n",
       "| 6 | 8  | 128746680 | 128755197 | + | Kukkonen et al. 2021 | MYC    | gene_plus_promoter | <!----> | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start     end       strand source               pub.gene.name\n",
       "1 4   106066032 106200973 +      Kukkonen et al. 2021 TET2         \n",
       "2 20   31349191  31397165 +      Kukkonen et al. 2021 DNMT3B       \n",
       "3 2   209100953 209131798 -      Kukkonen et al. 2021 IDH1         \n",
       "4 7   140419127 140625729 -      Kukkonen et al. 2021 BRAF         \n",
       "5 X    66762863  66950461 +      Kukkonen et al. 2021 AR           \n",
       "6 8   128746680 128755197 +      Kukkonen et al. 2021 MYC          \n",
       "  pub.reg.type       probe no.pubs\n",
       "1 gene_plus_promoter       1      \n",
       "2 gene_plus_promoter       1      \n",
       "3 gene_plus_promoter       1      \n",
       "4 gene_plus_promoter       1      \n",
       "5 gene_plus_promoter       1      \n",
       "6 gene_plus_promoter       1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all$start <- as.integer(format(df_all$start, scientific=FALSE))\n",
    "df_all$end <- as.integer(format(df_all$end, scientific=FALSE))\n",
    "head(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "ba4bcadf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# many genes have no coords in the genecode file, exclude those.\n",
    "write.table(df_all[!is.na(df_all$chr),], file=\"./marker_collation/all_lit_regions_wcoords_10052022.bed\", sep=\"\\t\",\n",
    "            row.names=FALSE, col.names=TRUE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "89d7b3e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in system(\"bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.full_genes_hg19.bed -wao > all_lit_regions_wcoords_gene_annot_10052022.bed\", :\n",
      "“running command 'bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.full_genes_hg19.bed -wao > all_lit_regions_wcoords_gene_annot_10052022.bed' had status 1”\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)\n",
       "attr(,\"status\")\n",
       "[1] 1\n",
       "attr(,\"errmsg\")\n",
       "[1] \"Resource temporarily unavailable\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in system(\"bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.promoters_hg19.bed -wao > all_lit_regions_wcoords_promoter_annot_10052022.bed\", :\n",
      "“running command 'bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.promoters_hg19.bed -wao > all_lit_regions_wcoords_promoter_annot_10052022.bed' had status 1”\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)\n",
       "attr(,\"status\")\n",
       "[1] 1\n",
       "attr(,\"errmsg\")\n",
       "[1] \"Resource temporarily unavailable\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# re annotate all regions with genes and promoters\n",
    "#system(\"bash\", intern=TRUE)\n",
    "#system(\"sort -k1,1 -k2,2 -k3,3 -n all_lit_regions_wcoords_10052022.bed >all_lit_regions_wcoords_10052022_sorted.bed\", intern=TRUE)                                                                               \n",
    "#system(\"bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.full_genes_hg19.bed -wao > all_lit_regions_wcoords_gene_annot_10052022.bed\", intern=TRUE)\n",
    "#system(\"bedtools intersect -a all_lit_regions_wcoords_10052022_sorted.bed -b ../../Translational_Science/gencode.v40lift37.promoters_hg19.bed -wao > all_lit_regions_wcoords_promoter_annot_10052022.bed\", intern=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "id": "97138851",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "120461"
      ],
      "text/latex": [
       "120461"
      ],
      "text/markdown": [
       "120461"
      ],
      "text/plain": [
       "[1] 120461"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 111409 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>8</th><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td></tr>\n",
       "\t<tr><th scope=row>4834</th><td>11</td><td>67350975</td><td>67350976</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>6180</th><td>11</td><td>67351270</td><td>67351271</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>6619</th><td>11</td><td>67351272</td><td>67351273</td><td>F </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>9051</th><td>11</td><td>67350213</td><td>67351213</td><td>+ </td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>NA</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.1</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.2</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.3</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.4</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.5</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.6</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.7</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.8</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.9</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.10</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.11</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.12</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.13</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.14</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.15</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.16</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.17</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.18</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.19</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.20</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.21</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.22</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.23</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>NA.111373</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111374</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111375</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111376</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111377</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111378</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111379</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111380</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111381</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111382</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111383</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111384</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111385</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111386</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111387</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111388</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111389</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111390</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111391</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111392</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111393</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111394</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111395</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111396</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111397</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111398</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111399</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111400</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111401</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111402</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 111409 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t8 & 11 & 67350213 & 67354127 & +  & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1\\\\\n",
       "\t15 & 11 & 67350213 & 67354127 & +  & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12\\\\\n",
       "\t4834 & 11 & 67350975 & 67350976 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1\\\\\n",
       "\t6180 & 11 & 67351270 & 67351271 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1\\\\\n",
       "\t6619 & 11 & 67351272 & 67351273 & F  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1\\\\\n",
       "\t9051 & 11 & 67350213 & 67351213 & +  & Sugiura et al 2021     & GSTP1 & promoter           &            &  1\\\\\n",
       "\tNA & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.1 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.2 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.3 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.4 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.5 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.6 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.7 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.8 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.9 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.10 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.11 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.12 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.13 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.14 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.15 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.16 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.17 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.18 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.19 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.20 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.21 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.22 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\tNA.23 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tNA.111373 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111374 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111375 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111376 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111377 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111378 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111379 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111380 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111381 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111382 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111383 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111384 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111385 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111386 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111387 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111388 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111389 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111390 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111391 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111392 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111393 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111394 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111395 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111396 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111397 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111398 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111399 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111400 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111401 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111402 & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 111409 × 9\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 8 | 11 | 67350213 | 67354127 | +  | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 |\n",
       "| 15 | 11 | 67350213 | 67354127 | +  | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 |\n",
       "| 4834 | 11 | 67350975 | 67350976 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 |\n",
       "| 6180 | 11 | 67351270 | 67351271 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 |\n",
       "| 6619 | 11 | 67351272 | 67351273 | F  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 |\n",
       "| 9051 | 11 | 67350213 | 67351213 | +  | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 |\n",
       "| NA | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.1 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.2 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.3 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.4 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.5 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.6 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.7 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.8 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.9 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.10 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.11 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.12 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.13 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.14 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.15 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.16 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.17 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.18 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.19 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.20 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.21 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.22 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| NA.23 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| NA.111373 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111374 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111375 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111376 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111377 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111378 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111379 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111380 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111381 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111382 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111383 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111384 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111385 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111386 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111387 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111388 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111389 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111390 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111391 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111392 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111393 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111394 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111395 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111396 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111397 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111398 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111399 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111400 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111401 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111402 | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "          chr start    end      strand source                 pub.gene.name\n",
       "8         11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "15        11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "4834      11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "6180      11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "6619      11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "9051      11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "NA        NA        NA       NA NA     NA                     NA           \n",
       "NA.1      NA        NA       NA NA     NA                     NA           \n",
       "NA.2      NA        NA       NA NA     NA                     NA           \n",
       "NA.3      NA        NA       NA NA     NA                     NA           \n",
       "NA.4      NA        NA       NA NA     NA                     NA           \n",
       "NA.5      NA        NA       NA NA     NA                     NA           \n",
       "NA.6      NA        NA       NA NA     NA                     NA           \n",
       "NA.7      NA        NA       NA NA     NA                     NA           \n",
       "NA.8      NA        NA       NA NA     NA                     NA           \n",
       "NA.9      NA        NA       NA NA     NA                     NA           \n",
       "NA.10     NA        NA       NA NA     NA                     NA           \n",
       "NA.11     NA        NA       NA NA     NA                     NA           \n",
       "NA.12     NA        NA       NA NA     NA                     NA           \n",
       "NA.13     NA        NA       NA NA     NA                     NA           \n",
       "NA.14     NA        NA       NA NA     NA                     NA           \n",
       "NA.15     NA        NA       NA NA     NA                     NA           \n",
       "NA.16     NA        NA       NA NA     NA                     NA           \n",
       "NA.17     NA        NA       NA NA     NA                     NA           \n",
       "NA.18     NA        NA       NA NA     NA                     NA           \n",
       "NA.19     NA        NA       NA NA     NA                     NA           \n",
       "NA.20     NA        NA       NA NA     NA                     NA           \n",
       "NA.21     NA        NA       NA NA     NA                     NA           \n",
       "NA.22     NA        NA       NA NA     NA                     NA           \n",
       "NA.23     NA        NA       NA NA     NA                     NA           \n",
       "⋮         ⋮   ⋮        ⋮        ⋮      ⋮                      ⋮            \n",
       "NA.111373 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111374 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111375 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111376 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111377 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111378 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111379 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111380 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111381 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111382 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111383 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111384 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111385 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111386 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111387 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111388 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111389 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111390 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111391 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111392 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111393 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111394 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111395 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111396 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111397 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111398 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111399 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111400 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111401 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111402 NA  NA       NA       NA     NA                     NA           \n",
       "          pub.reg.type       probe      no.pubs\n",
       "8         gene_plus_promoter             1     \n",
       "15        gene_plus_promoter            12     \n",
       "4834      EPIC_probe         cg02659086  1     \n",
       "6180      EPIC_probe         cg04920951  1     \n",
       "6619      EPIC_probe         cg26250609  1     \n",
       "9051      promoter                       1     \n",
       "NA        NA                 NA         NA     \n",
       "NA.1      NA                 NA         NA     \n",
       "NA.2      NA                 NA         NA     \n",
       "NA.3      NA                 NA         NA     \n",
       "NA.4      NA                 NA         NA     \n",
       "NA.5      NA                 NA         NA     \n",
       "NA.6      NA                 NA         NA     \n",
       "NA.7      NA                 NA         NA     \n",
       "NA.8      NA                 NA         NA     \n",
       "NA.9      NA                 NA         NA     \n",
       "NA.10     NA                 NA         NA     \n",
       "NA.11     NA                 NA         NA     \n",
       "NA.12     NA                 NA         NA     \n",
       "NA.13     NA                 NA         NA     \n",
       "NA.14     NA                 NA         NA     \n",
       "NA.15     NA                 NA         NA     \n",
       "NA.16     NA                 NA         NA     \n",
       "NA.17     NA                 NA         NA     \n",
       "NA.18     NA                 NA         NA     \n",
       "NA.19     NA                 NA         NA     \n",
       "NA.20     NA                 NA         NA     \n",
       "NA.21     NA                 NA         NA     \n",
       "NA.22     NA                 NA         NA     \n",
       "NA.23     NA                 NA         NA     \n",
       "⋮         ⋮                  ⋮          ⋮      \n",
       "NA.111373 NA                 NA         NA     \n",
       "NA.111374 NA                 NA         NA     \n",
       "NA.111375 NA                 NA         NA     \n",
       "NA.111376 NA                 NA         NA     \n",
       "NA.111377 NA                 NA         NA     \n",
       "NA.111378 NA                 NA         NA     \n",
       "NA.111379 NA                 NA         NA     \n",
       "NA.111380 NA                 NA         NA     \n",
       "NA.111381 NA                 NA         NA     \n",
       "NA.111382 NA                 NA         NA     \n",
       "NA.111383 NA                 NA         NA     \n",
       "NA.111384 NA                 NA         NA     \n",
       "NA.111385 NA                 NA         NA     \n",
       "NA.111386 NA                 NA         NA     \n",
       "NA.111387 NA                 NA         NA     \n",
       "NA.111388 NA                 NA         NA     \n",
       "NA.111389 NA                 NA         NA     \n",
       "NA.111390 NA                 NA         NA     \n",
       "NA.111391 NA                 NA         NA     \n",
       "NA.111392 NA                 NA         NA     \n",
       "NA.111393 NA                 NA         NA     \n",
       "NA.111394 NA                 NA         NA     \n",
       "NA.111395 NA                 NA         NA     \n",
       "NA.111396 NA                 NA         NA     \n",
       "NA.111397 NA                 NA         NA     \n",
       "NA.111398 NA                 NA         NA     \n",
       "NA.111399 NA                 NA         NA     \n",
       "NA.111400 NA                 NA         NA     \n",
       "NA.111401 NA                 NA         NA     \n",
       "NA.111402 NA                 NA         NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all = read.table(file=\"./marker_collation/all_lit_regions_wcoords_10052022.bed\", header=TRUE,sep=\"\\t\")\n",
    "nrow(df_all[!is.na(df_all$chr),])\n",
    "df_all[df_all$pub.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "3060f5af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>144404</th><td>22</td><td>51110120</td><td>51112461</td><td>.</td><td>Beltran et al 2020</td><td>NA    </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51171640</td><td>SHANK3         </td><td>0</td><td>+</td><td>gene_plus_promoter</td><td> 618</td></tr>\n",
       "\t<tr><th scope=row>144405</th><td>22</td><td>51111713</td><td>51111714</td><td>R</td><td>Ylitalo et al 2021</td><td>SHANK3</td><td>EPIC_probe</td><td>cg26201811</td><td>1</td><td>22</td><td>51110142</td><td>51112602</td><td>ENSG00000286508</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "\t<tr><th scope=row>144406</th><td>22</td><td>51135020</td><td>51136679</td><td>.</td><td>Beltran et al 2020</td><td>NA    </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51171640</td><td>SHANK3         </td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>1659</td></tr>\n",
       "\t<tr><th scope=row>144407</th><td>22</td><td>51135139</td><td>51135320</td><td>.</td><td>Zhao et al 2020   </td><td>NA    </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51171640</td><td>SHANK3         </td><td>0</td><td>+</td><td>gene_plus_promoter</td><td> 181</td></tr>\n",
       "\t<tr><th scope=row>144408</th><td>22</td><td>51135380</td><td>51135451</td><td>.</td><td>Zhao et al 2020   </td><td>NA    </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51171640</td><td>SHANK3         </td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>  71</td></tr>\n",
       "\t<tr><th scope=row>144409</th><td>22</td><td>51142482</td><td>51143093</td><td>.</td><td>Beltran et al 2020</td><td>NA    </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51171640</td><td>SHANK3         </td><td>0</td><td>+</td><td>gene_plus_promoter</td><td> 611</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13 & V14 & V15 & V16 & V17\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t144404 & 22 & 51110120 & 51112461 & . & Beltran et al 2020 & NA     & DMR        & NA         & 1 & 22 & 51111843 & 51171640 & SHANK3          & 0 & + & gene\\_plus\\_promoter &  618\\\\\n",
       "\t144405 & 22 & 51111713 & 51111714 & R & Ylitalo et al 2021 & SHANK3 & EPIC\\_probe & cg26201811 & 1 & 22 & 51110142 & 51112602 & ENSG00000286508 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\t144406 & 22 & 51135020 & 51136679 & . & Beltran et al 2020 & NA     & DMR        & NA         & 1 & 22 & 51111843 & 51171640 & SHANK3          & 0 & + & gene\\_plus\\_promoter & 1659\\\\\n",
       "\t144407 & 22 & 51135139 & 51135320 & . & Zhao et al 2020    & NA     & DMR        & NA         & 1 & 22 & 51111843 & 51171640 & SHANK3          & 0 & + & gene\\_plus\\_promoter &  181\\\\\n",
       "\t144408 & 22 & 51135380 & 51135451 & . & Zhao et al 2020    & NA     & DMR        & NA         & 1 & 22 & 51111843 & 51171640 & SHANK3          & 0 & + & gene\\_plus\\_promoter &   71\\\\\n",
       "\t144409 & 22 & 51142482 & 51143093 & . & Beltran et al 2020 & NA     & DMR        & NA         & 1 & 22 & 51111843 & 51171640 & SHANK3          & 0 & + & gene\\_plus\\_promoter &  611\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | V11 &lt;int&gt; | V12 &lt;int&gt; | V13 &lt;chr&gt; | V14 &lt;chr&gt; | V15 &lt;chr&gt; | V16 &lt;chr&gt; | V17 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 144404 | 22 | 51110120 | 51112461 | . | Beltran et al 2020 | NA     | DMR        | NA         | 1 | 22 | 51111843 | 51171640 | SHANK3          | 0 | + | gene_plus_promoter |  618 |\n",
       "| 144405 | 22 | 51111713 | 51111714 | R | Ylitalo et al 2021 | SHANK3 | EPIC_probe | cg26201811 | 1 | 22 | 51110142 | 51112602 | ENSG00000286508 | 0 | + | gene_plus_promoter |    1 |\n",
       "| 144406 | 22 | 51135020 | 51136679 | . | Beltran et al 2020 | NA     | DMR        | NA         | 1 | 22 | 51111843 | 51171640 | SHANK3          | 0 | + | gene_plus_promoter | 1659 |\n",
       "| 144407 | 22 | 51135139 | 51135320 | . | Zhao et al 2020    | NA     | DMR        | NA         | 1 | 22 | 51111843 | 51171640 | SHANK3          | 0 | + | gene_plus_promoter |  181 |\n",
       "| 144408 | 22 | 51135380 | 51135451 | . | Zhao et al 2020    | NA     | DMR        | NA         | 1 | 22 | 51111843 | 51171640 | SHANK3          | 0 | + | gene_plus_promoter |   71 |\n",
       "| 144409 | 22 | 51142482 | 51143093 | . | Beltran et al 2020 | NA     | DMR        | NA         | 1 | 22 | 51111843 | 51171640 | SHANK3          | 0 | + | gene_plus_promoter |  611 |\n",
       "\n"
      ],
      "text/plain": [
       "       V1 V2       V3       V4 V5                 V6     V7         V8        \n",
       "144404 22 51110120 51112461 .  Beltran et al 2020 NA     DMR        NA        \n",
       "144405 22 51111713 51111714 R  Ylitalo et al 2021 SHANK3 EPIC_probe cg26201811\n",
       "144406 22 51135020 51136679 .  Beltran et al 2020 NA     DMR        NA        \n",
       "144407 22 51135139 51135320 .  Zhao et al 2020    NA     DMR        NA        \n",
       "144408 22 51135380 51135451 .  Zhao et al 2020    NA     DMR        NA        \n",
       "144409 22 51142482 51143093 .  Beltran et al 2020 NA     DMR        NA        \n",
       "       V9 V10 V11      V12      V13             V14 V15 V16                V17 \n",
       "144404 1  22  51111843 51171640 SHANK3          0   +   gene_plus_promoter  618\n",
       "144405 1  22  51110142 51112602 ENSG00000286508 0   +   gene_plus_promoter    1\n",
       "144406 1  22  51111843 51171640 SHANK3          0   +   gene_plus_promoter 1659\n",
       "144407 1  22  51111843 51171640 SHANK3          0   +   gene_plus_promoter  181\n",
       "144408 1  22  51111843 51171640 SHANK3          0   +   gene_plus_promoter   71\n",
       "144409 1  22  51111843 51171640 SHANK3          0   +   gene_plus_promoter  611"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>chr.g</th><th scope=col>start.g</th><th scope=col>end.g</th><th scope=col>gencode.gene.name</th><th scope=col>zero</th><th scope=col>gencode.gene.strand</th><th scope=col>gencode.reg.type</th><th scope=col>bp.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>128841</th><td>22</td><td>51110120</td><td>51112461</td><td>.</td><td>Beltran et al 2020</td><td>none  </td><td>DMR       </td><td>NA        </td><td>1</td><td>22</td><td>51111843</td><td>51112843</td><td>SHANK3</td><td>0</td><td>+</td><td>promoter</td><td>618</td></tr>\n",
       "\t<tr><th scope=row>128842</th><td>22</td><td>51111713</td><td>51111714</td><td>R</td><td>Ylitalo et al 2021</td><td>SHANK3</td><td>EPIC_probe</td><td>cg26201811</td><td>1</td><td>. </td><td>      -1</td><td>      -1</td><td>.     </td><td>.</td><td>.</td><td>.       </td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>128843</th><td>22</td><td>51135020</td><td>51136679</td><td>.</td><td>Beltran et al 2020</td><td>none  </td><td>DMR       </td><td>NA        </td><td>1</td><td>. </td><td>      -1</td><td>      -1</td><td>.     </td><td>.</td><td>.</td><td>.       </td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>128844</th><td>22</td><td>51135139</td><td>51135320</td><td>.</td><td>Zhao et al 2020   </td><td>none  </td><td>DMR       </td><td>NA        </td><td>1</td><td>. </td><td>      -1</td><td>      -1</td><td>.     </td><td>.</td><td>.</td><td>.       </td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>128845</th><td>22</td><td>51135380</td><td>51135451</td><td>.</td><td>Zhao et al 2020   </td><td>none  </td><td>DMR       </td><td>NA        </td><td>1</td><td>. </td><td>      -1</td><td>      -1</td><td>.     </td><td>.</td><td>.</td><td>.       </td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>128846</th><td>22</td><td>51142482</td><td>51143093</td><td>.</td><td>Beltran et al 2020</td><td>none  </td><td>DMR       </td><td>NA        </td><td>1</td><td>. </td><td>      -1</td><td>      -1</td><td>.     </td><td>.</td><td>.</td><td>.       </td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & chr.g & start.g & end.g & gencode.gene.name & zero & gencode.gene.strand & gencode.reg.type & bp.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t128841 & 22 & 51110120 & 51112461 & . & Beltran et al 2020 & none   & DMR        & NA         & 1 & 22 & 51111843 & 51112843 & SHANK3 & 0 & + & promoter & 618\\\\\n",
       "\t128842 & 22 & 51111713 & 51111714 & R & Ylitalo et al 2021 & SHANK3 & EPIC\\_probe & cg26201811 & 1 & .  &       -1 &       -1 & .      & . & . & .        &   0\\\\\n",
       "\t128843 & 22 & 51135020 & 51136679 & . & Beltran et al 2020 & none   & DMR        & NA         & 1 & .  &       -1 &       -1 & .      & . & . & .        &   0\\\\\n",
       "\t128844 & 22 & 51135139 & 51135320 & . & Zhao et al 2020    & none   & DMR        & NA         & 1 & .  &       -1 &       -1 & .      & . & . & .        &   0\\\\\n",
       "\t128845 & 22 & 51135380 & 51135451 & . & Zhao et al 2020    & none   & DMR        & NA         & 1 & .  &       -1 &       -1 & .      & . & . & .        &   0\\\\\n",
       "\t128846 & 22 & 51142482 & 51143093 & . & Beltran et al 2020 & none   & DMR        & NA         & 1 & .  &       -1 &       -1 & .      & . & . & .        &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | chr.g &lt;chr&gt; | start.g &lt;int&gt; | end.g &lt;int&gt; | gencode.gene.name &lt;chr&gt; | zero &lt;chr&gt; | gencode.gene.strand &lt;chr&gt; | gencode.reg.type &lt;chr&gt; | bp.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 128841 | 22 | 51110120 | 51112461 | . | Beltran et al 2020 | none   | DMR        | NA         | 1 | 22 | 51111843 | 51112843 | SHANK3 | 0 | + | promoter | 618 |\n",
       "| 128842 | 22 | 51111713 | 51111714 | R | Ylitalo et al 2021 | SHANK3 | EPIC_probe | cg26201811 | 1 | .  |       -1 |       -1 | .      | . | . | .        |   0 |\n",
       "| 128843 | 22 | 51135020 | 51136679 | . | Beltran et al 2020 | none   | DMR        | NA         | 1 | .  |       -1 |       -1 | .      | . | . | .        |   0 |\n",
       "| 128844 | 22 | 51135139 | 51135320 | . | Zhao et al 2020    | none   | DMR        | NA         | 1 | .  |       -1 |       -1 | .      | . | . | .        |   0 |\n",
       "| 128845 | 22 | 51135380 | 51135451 | . | Zhao et al 2020    | none   | DMR        | NA         | 1 | .  |       -1 |       -1 | .      | . | . | .        |   0 |\n",
       "| 128846 | 22 | 51142482 | 51143093 | . | Beltran et al 2020 | none   | DMR        | NA         | 1 | .  |       -1 |       -1 | .      | . | . | .        |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "       chr start    end      strand source             pub.gene.name\n",
       "128841 22  51110120 51112461 .      Beltran et al 2020 none         \n",
       "128842 22  51111713 51111714 R      Ylitalo et al 2021 SHANK3       \n",
       "128843 22  51135020 51136679 .      Beltran et al 2020 none         \n",
       "128844 22  51135139 51135320 .      Zhao et al 2020    none         \n",
       "128845 22  51135380 51135451 .      Zhao et al 2020    none         \n",
       "128846 22  51142482 51143093 .      Beltran et al 2020 none         \n",
       "       pub.reg.type probe      no.pubs chr.g start.g  end.g   \n",
       "128841 DMR          NA         1       22    51111843 51112843\n",
       "128842 EPIC_probe   cg26201811 1       .           -1       -1\n",
       "128843 DMR          NA         1       .           -1       -1\n",
       "128844 DMR          NA         1       .           -1       -1\n",
       "128845 DMR          NA         1       .           -1       -1\n",
       "128846 DMR          NA         1       .           -1       -1\n",
       "       gencode.gene.name zero gencode.gene.strand gencode.reg.type bp.overlap\n",
       "128841 SHANK3            0    +                   promoter         618       \n",
       "128842 .                 .    .                   .                  0       \n",
       "128843 .                 .    .                   .                  0       \n",
       "128844 .                 .    .                   .                  0       \n",
       "128845 .                 .    .                   .                  0       \n",
       "128846 .                 .    .                   .                  0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>chr.g</th><th scope=col>start.g</th><th scope=col>end.g</th><th scope=col>gencode.gene.name</th><th scope=col>zero</th><th scope=col>gencode.gene.strand</th><th scope=col>gencode.reg.type</th><th scope=col>bp.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>93941</th><td>11</td><td>67350213</td><td>67351213</td><td>+</td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>1000</td></tr>\n",
       "\t<tr><th scope=row>93942</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>3914</td></tr>\n",
       "\t<tr><th scope=row>93943</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>3914</td></tr>\n",
       "\t<tr><th scope=row>93944</th><td>11</td><td>67350975</td><td>67350976</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "\t<tr><th scope=row>93945</th><td>11</td><td>67351270</td><td>67351271</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "\t<tr><th scope=row>93946</th><td>11</td><td>67351272</td><td>67351273</td><td>F</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & chr.g & start.g & end.g & gencode.gene.name & zero & gencode.gene.strand & gencode.reg.type & bp.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t93941 & 11 & 67350213 & 67351213 & + & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 1000\\\\\n",
       "\t93942 & 11 & 67350213 & 67354127 & + & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 3914\\\\\n",
       "\t93943 & 11 & 67350213 & 67354127 & + & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 3914\\\\\n",
       "\t93944 & 11 & 67350975 & 67350976 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\t93945 & 11 & 67351270 & 67351271 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\t93946 & 11 & 67351272 & 67351273 & F & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | chr.g &lt;chr&gt; | start.g &lt;int&gt; | end.g &lt;int&gt; | gencode.gene.name &lt;chr&gt; | zero &lt;chr&gt; | gencode.gene.strand &lt;chr&gt; | gencode.reg.type &lt;chr&gt; | bp.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 93941 | 11 | 67350213 | 67351213 | + | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 1000 |\n",
       "| 93942 | 11 | 67350213 | 67354127 | + | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 3914 |\n",
       "| 93943 | 11 | 67350213 | 67354127 | + | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 3914 |\n",
       "| 93944 | 11 | 67350975 | 67350976 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "| 93945 | 11 | 67351270 | 67351271 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "| 93946 | 11 | 67351272 | 67351273 | F | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "\n"
      ],
      "text/plain": [
       "      chr start    end      strand source                 pub.gene.name\n",
       "93941 11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "93942 11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "93943 11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "93944 11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "93945 11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "93946 11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "      pub.reg.type       probe      no.pubs chr.g start.g  end.g   \n",
       "93941 promoter                       1      11    67350213 67354127\n",
       "93942 gene_plus_promoter             1      11    67350213 67354127\n",
       "93943 gene_plus_promoter            12      11    67350213 67354127\n",
       "93944 EPIC_probe         cg02659086  1      11    67350213 67354127\n",
       "93945 EPIC_probe         cg04920951  1      11    67350213 67354127\n",
       "93946 EPIC_probe         cg26250609  1      11    67350213 67354127\n",
       "      gencode.gene.name zero gencode.gene.strand gencode.reg.type   bp.overlap\n",
       "93941 GSTP1             0    +                   gene_plus_promoter 1000      \n",
       "93942 GSTP1             0    +                   gene_plus_promoter 3914      \n",
       "93943 GSTP1             0    +                   gene_plus_promoter 3914      \n",
       "93944 GSTP1             0    +                   gene_plus_promoter    1      \n",
       "93945 GSTP1             0    +                   gene_plus_promoter    1      \n",
       "93946 GSTP1             0    +                   gene_plus_promoter    1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_gene_annot = read.table(file=\"./marker_collation/all_lit_regions_wcoords_gene_annot_10052022.bed\", header=FALSE, sep=\"\\t\")\n",
    "tail(df_gene_annot)\n",
    "cols = c(\"chr\",\"start\",\"end\",\"strand\",\"source\",\"pub.gene.name\",\"pub.reg.type\",\n",
    "                          \"probe\",\"no.pubs\",\"chr.g\",\"start.g\",\"end.g\",\"gencode.gene.name\",\"zero\",\n",
    "                           \"gencode.gene.strand\",\"gencode.reg.type\",\"bp.overlap\")\n",
    "df_prom_annot = read.table(file=\"./marker_collation/all_lit_regions_wcoords_promoter_annot_10052022.bed\", header=FALSE, sep=\"\\t\")\n",
    "colnames(df_gene_annot) <- cols\n",
    "colnames(df_prom_annot) <- cols\n",
    "\n",
    "df_gene_annot = df_gene_annot[!is.na(df_gene_annot$start),]\n",
    "df_gene_annot$gencode.gene.name[is.na(df_gene_annot$gencode.gene.name)] <- \"none\"\n",
    "df_gene_annot$pub.gene.name[is.na(df_gene_annot$pub.gene.name)] <- \"none\"\n",
    "df_gene_annot$start = as.integer(df_gene_annot$start)\n",
    "df_gene_annot$end = as.integer(df_gene_annot$end)\n",
    "\n",
    "df_prom_annot = df_prom_annot[!is.na(df_prom_annot$start),]\n",
    "df_prom_annot$gencode.gene.name[is.na(df_prom_annot$gencode.gene.name)] <- \"none\"\n",
    "df_prom_annot$pub.gene.name[is.na(df_prom_annot$pub.gene.name)] <- \"none\"\n",
    "df_prom_annot$start = as.integer(df_prom_annot$start)\n",
    "df_prom_annot$end = as.integer(df_prom_annot$end)\n",
    "\n",
    "tail(df_prom_annot)\n",
    "df_gene_annot[df_gene_annot$pub.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "b44814db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>chr.g</th><th scope=col>start.g</th><th scope=col>end.g</th><th scope=col>gencode.gene.name</th><th scope=col>zero</th><th scope=col>gencode.gene.strand</th><th scope=col>gencode.reg.type</th><th scope=col>bp.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>79324</th><td>11</td><td>67350213</td><td>67351213</td><td>+</td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>1000</td></tr>\n",
       "\t<tr><th scope=row>79325</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>3914</td></tr>\n",
       "\t<tr><th scope=row>79326</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>3914</td></tr>\n",
       "\t<tr><th scope=row>79327</th><td>11</td><td>67350975</td><td>67350976</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "\t<tr><th scope=row>79328</th><td>11</td><td>67351270</td><td>67351271</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "\t<tr><th scope=row>79329</th><td>11</td><td>67351272</td><td>67351273</td><td>F</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>11</td><td>67350213</td><td>67354127</td><td>GSTP1</td><td>0</td><td>+</td><td>gene_plus_promoter</td><td>   1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & chr.g & start.g & end.g & gencode.gene.name & zero & gencode.gene.strand & gencode.reg.type & bp.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t79324 & 11 & 67350213 & 67351213 & + & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 1000\\\\\n",
       "\t79325 & 11 & 67350213 & 67354127 & + & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 3914\\\\\n",
       "\t79326 & 11 & 67350213 & 67354127 & + & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter & 3914\\\\\n",
       "\t79327 & 11 & 67350975 & 67350976 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\t79328 & 11 & 67351270 & 67351271 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\t79329 & 11 & 67351272 & 67351273 & F & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & 11 & 67350213 & 67354127 & GSTP1 & 0 & + & gene\\_plus\\_promoter &    1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | chr.g &lt;chr&gt; | start.g &lt;int&gt; | end.g &lt;int&gt; | gencode.gene.name &lt;chr&gt; | zero &lt;chr&gt; | gencode.gene.strand &lt;chr&gt; | gencode.reg.type &lt;chr&gt; | bp.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 79324 | 11 | 67350213 | 67351213 | + | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 1000 |\n",
       "| 79325 | 11 | 67350213 | 67354127 | + | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 3914 |\n",
       "| 79326 | 11 | 67350213 | 67354127 | + | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter | 3914 |\n",
       "| 79327 | 11 | 67350975 | 67350976 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "| 79328 | 11 | 67351270 | 67351271 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "| 79329 | 11 | 67351272 | 67351273 | F | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | 11 | 67350213 | 67354127 | GSTP1 | 0 | + | gene_plus_promoter |    1 |\n",
       "\n"
      ],
      "text/plain": [
       "      chr start    end      strand source                 pub.gene.name\n",
       "79324 11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "79325 11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "79326 11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "79327 11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "79328 11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "79329 11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "      pub.reg.type       probe      no.pubs chr.g start.g  end.g   \n",
       "79324 promoter                       1      11    67350213 67354127\n",
       "79325 gene_plus_promoter             1      11    67350213 67354127\n",
       "79326 gene_plus_promoter            12      11    67350213 67354127\n",
       "79327 EPIC_probe         cg02659086  1      11    67350213 67354127\n",
       "79328 EPIC_probe         cg04920951  1      11    67350213 67354127\n",
       "79329 EPIC_probe         cg26250609  1      11    67350213 67354127\n",
       "      gencode.gene.name zero gencode.gene.strand gencode.reg.type   bp.overlap\n",
       "79324 GSTP1             0    +                   gene_plus_promoter 1000      \n",
       "79325 GSTP1             0    +                   gene_plus_promoter 3914      \n",
       "79326 GSTP1             0    +                   gene_plus_promoter 3914      \n",
       "79327 GSTP1             0    +                   gene_plus_promoter    1      \n",
       "79328 GSTP1             0    +                   gene_plus_promoter    1      \n",
       "79329 GSTP1             0    +                   gene_plus_promoter    1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# pub regions could be duplicated if they overlap with more than one gencode gene\n",
    "#keep just the first gencode hit\n",
    "df_gene_annot = df_gene_annot %>% distinct(chr,start,end,source,pub.gene.name, .keep_all=TRUE)\n",
    "df_gene_annot[df_gene_annot$pub.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "648dba1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_prom_annot = df_prom_annot %>% distinct(chr,start,end,source,pub.gene.name, .keep_all=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "14fd5705",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'chr'</li><li>'start'</li><li>'end'</li><li>'strand'</li><li>'source'</li><li>'pub.gene.name'</li><li>'pub.reg.type'</li><li>'probe'</li><li>'no.pubs'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'chr'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'strand'\n",
       "\\item 'source'\n",
       "\\item 'pub.gene.name'\n",
       "\\item 'pub.reg.type'\n",
       "\\item 'probe'\n",
       "\\item 'no.pubs'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'chr'\n",
       "2. 'start'\n",
       "3. 'end'\n",
       "4. 'strand'\n",
       "5. 'source'\n",
       "6. 'pub.gene.name'\n",
       "7. 'pub.reg.type'\n",
       "8. 'probe'\n",
       "9. 'no.pubs'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"chr\"           \"start\"         \"end\"           \"strand\"       \n",
       "[5] \"source\"        \"pub.gene.name\" \"pub.reg.type\"  \"probe\"        \n",
       "[9] \"no.pubs\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "id": "fe0c1466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 111409 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type</th><th scope=col>gencode.gene.name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>8</th><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>4834</th><td>11</td><td>67350975</td><td>67350976</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>6180</th><td>11</td><td>67351270</td><td>67351271</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>6619</th><td>11</td><td>67351272</td><td>67351273</td><td>F </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>9051</th><td>11</td><td>67350213</td><td>67351213</td><td>+ </td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td></tr>\n",
       "\t<tr><th scope=row>NA</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.1</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.2</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.3</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.4</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.5</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.6</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.7</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.8</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.9</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.10</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.11</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.12</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.13</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.14</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.15</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.16</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.17</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.18</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.19</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.20</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.21</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.22</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>NA.23</th><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>NA.111373</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111374</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111375</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111376</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111377</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111378</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111379</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111380</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111381</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111382</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111383</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111384</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111385</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111386</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111387</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111388</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111389</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111390</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111391</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111392</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111393</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111394</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111395</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111396</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111397</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111398</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111399</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111400</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111401</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>NA.111402</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 111409 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type & gencode.gene.name\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t8 & 11 & 67350213 & 67354127 & +  & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\t15 & 11 & 67350213 & 67354127 & +  & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\t4834 & 11 & 67350975 & 67350976 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\t6180 & 11 & 67351270 & 67351271 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\t6619 & 11 & 67351272 & 67351273 & F  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\t9051 & 11 & 67350213 & 67351213 & +  & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & gene\\_plus\\_promoter & GSTP1\\\\\n",
       "\tNA & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.1 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.2 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.3 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.4 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.5 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.6 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.7 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.8 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.9 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.10 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.11 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.12 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.13 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.14 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.15 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.16 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.17 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.18 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.19 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.20 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.21 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.22 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\tNA.23 & NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA   \\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tNA.111373 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111374 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111375 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111376 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111377 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111378 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111379 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111380 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111381 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111382 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111383 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111384 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111385 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111386 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111387 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111388 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111389 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111390 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111391 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111392 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111393 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111394 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111395 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111396 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111397 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111398 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111399 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111400 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111401 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tNA.111402 & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 111409 × 11\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type &lt;chr&gt; | gencode.gene.name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 8 | 11 | 67350213 | 67354127 | +  | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | gene_plus_promoter | GSTP1 |\n",
       "| 15 | 11 | 67350213 | 67354127 | +  | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | gene_plus_promoter | GSTP1 |\n",
       "| 4834 | 11 | 67350975 | 67350976 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | gene_plus_promoter | GSTP1 |\n",
       "| 6180 | 11 | 67351270 | 67351271 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | gene_plus_promoter | GSTP1 |\n",
       "| 6619 | 11 | 67351272 | 67351273 | F  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | gene_plus_promoter | GSTP1 |\n",
       "| 9051 | 11 | 67350213 | 67351213 | +  | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | gene_plus_promoter | GSTP1 |\n",
       "| NA | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.1 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.2 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.3 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.4 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.5 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.6 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.7 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.8 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.9 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.10 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.11 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.12 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.13 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.14 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.15 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.16 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.17 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.18 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.19 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.20 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.21 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.22 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| NA.23 | NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| NA.111373 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111374 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111375 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111376 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111377 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111378 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111379 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111380 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111381 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111382 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111383 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111384 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111385 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111386 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111387 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111388 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111389 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111390 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111391 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111392 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111393 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111394 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111395 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111396 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111397 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111398 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111399 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111400 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111401 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA.111402 | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "          chr start    end      strand source                 pub.gene.name\n",
       "8         11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "15        11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "4834      11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "6180      11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "6619      11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "9051      11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "NA        NA        NA       NA NA     NA                     NA           \n",
       "NA.1      NA        NA       NA NA     NA                     NA           \n",
       "NA.2      NA        NA       NA NA     NA                     NA           \n",
       "NA.3      NA        NA       NA NA     NA                     NA           \n",
       "NA.4      NA        NA       NA NA     NA                     NA           \n",
       "NA.5      NA        NA       NA NA     NA                     NA           \n",
       "NA.6      NA        NA       NA NA     NA                     NA           \n",
       "NA.7      NA        NA       NA NA     NA                     NA           \n",
       "NA.8      NA        NA       NA NA     NA                     NA           \n",
       "NA.9      NA        NA       NA NA     NA                     NA           \n",
       "NA.10     NA        NA       NA NA     NA                     NA           \n",
       "NA.11     NA        NA       NA NA     NA                     NA           \n",
       "NA.12     NA        NA       NA NA     NA                     NA           \n",
       "NA.13     NA        NA       NA NA     NA                     NA           \n",
       "NA.14     NA        NA       NA NA     NA                     NA           \n",
       "NA.15     NA        NA       NA NA     NA                     NA           \n",
       "NA.16     NA        NA       NA NA     NA                     NA           \n",
       "NA.17     NA        NA       NA NA     NA                     NA           \n",
       "NA.18     NA        NA       NA NA     NA                     NA           \n",
       "NA.19     NA        NA       NA NA     NA                     NA           \n",
       "NA.20     NA        NA       NA NA     NA                     NA           \n",
       "NA.21     NA        NA       NA NA     NA                     NA           \n",
       "NA.22     NA        NA       NA NA     NA                     NA           \n",
       "NA.23     NA        NA       NA NA     NA                     NA           \n",
       "⋮         ⋮   ⋮        ⋮        ⋮      ⋮                      ⋮            \n",
       "NA.111373 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111374 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111375 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111376 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111377 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111378 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111379 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111380 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111381 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111382 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111383 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111384 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111385 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111386 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111387 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111388 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111389 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111390 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111391 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111392 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111393 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111394 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111395 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111396 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111397 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111398 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111399 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111400 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111401 NA  NA       NA       NA     NA                     NA           \n",
       "NA.111402 NA  NA       NA       NA     NA                     NA           \n",
       "          pub.reg.type       probe      no.pubs gencode.reg.type  \n",
       "8         gene_plus_promoter             1      gene_plus_promoter\n",
       "15        gene_plus_promoter            12      gene_plus_promoter\n",
       "4834      EPIC_probe         cg02659086  1      gene_plus_promoter\n",
       "6180      EPIC_probe         cg04920951  1      gene_plus_promoter\n",
       "6619      EPIC_probe         cg26250609  1      gene_plus_promoter\n",
       "9051      promoter                       1      gene_plus_promoter\n",
       "NA        NA                 NA         NA      NA                \n",
       "NA.1      NA                 NA         NA      NA                \n",
       "NA.2      NA                 NA         NA      NA                \n",
       "NA.3      NA                 NA         NA      NA                \n",
       "NA.4      NA                 NA         NA      NA                \n",
       "NA.5      NA                 NA         NA      NA                \n",
       "NA.6      NA                 NA         NA      NA                \n",
       "NA.7      NA                 NA         NA      NA                \n",
       "NA.8      NA                 NA         NA      NA                \n",
       "NA.9      NA                 NA         NA      NA                \n",
       "NA.10     NA                 NA         NA      NA                \n",
       "NA.11     NA                 NA         NA      NA                \n",
       "NA.12     NA                 NA         NA      NA                \n",
       "NA.13     NA                 NA         NA      NA                \n",
       "NA.14     NA                 NA         NA      NA                \n",
       "NA.15     NA                 NA         NA      NA                \n",
       "NA.16     NA                 NA         NA      NA                \n",
       "NA.17     NA                 NA         NA      NA                \n",
       "NA.18     NA                 NA         NA      NA                \n",
       "NA.19     NA                 NA         NA      NA                \n",
       "NA.20     NA                 NA         NA      NA                \n",
       "NA.21     NA                 NA         NA      NA                \n",
       "NA.22     NA                 NA         NA      NA                \n",
       "NA.23     NA                 NA         NA      NA                \n",
       "⋮         ⋮                  ⋮          ⋮       ⋮                 \n",
       "NA.111373 NA                 NA         NA      NA                \n",
       "NA.111374 NA                 NA         NA      NA                \n",
       "NA.111375 NA                 NA         NA      NA                \n",
       "NA.111376 NA                 NA         NA      NA                \n",
       "NA.111377 NA                 NA         NA      NA                \n",
       "NA.111378 NA                 NA         NA      NA                \n",
       "NA.111379 NA                 NA         NA      NA                \n",
       "NA.111380 NA                 NA         NA      NA                \n",
       "NA.111381 NA                 NA         NA      NA                \n",
       "NA.111382 NA                 NA         NA      NA                \n",
       "NA.111383 NA                 NA         NA      NA                \n",
       "NA.111384 NA                 NA         NA      NA                \n",
       "NA.111385 NA                 NA         NA      NA                \n",
       "NA.111386 NA                 NA         NA      NA                \n",
       "NA.111387 NA                 NA         NA      NA                \n",
       "NA.111388 NA                 NA         NA      NA                \n",
       "NA.111389 NA                 NA         NA      NA                \n",
       "NA.111390 NA                 NA         NA      NA                \n",
       "NA.111391 NA                 NA         NA      NA                \n",
       "NA.111392 NA                 NA         NA      NA                \n",
       "NA.111393 NA                 NA         NA      NA                \n",
       "NA.111394 NA                 NA         NA      NA                \n",
       "NA.111395 NA                 NA         NA      NA                \n",
       "NA.111396 NA                 NA         NA      NA                \n",
       "NA.111397 NA                 NA         NA      NA                \n",
       "NA.111398 NA                 NA         NA      NA                \n",
       "NA.111399 NA                 NA         NA      NA                \n",
       "NA.111400 NA                 NA         NA      NA                \n",
       "NA.111401 NA                 NA         NA      NA                \n",
       "NA.111402 NA                 NA         NA      NA                \n",
       "          gencode.gene.name\n",
       "8         GSTP1            \n",
       "15        GSTP1            \n",
       "4834      GSTP1            \n",
       "6180      GSTP1            \n",
       "6619      GSTP1            \n",
       "9051      GSTP1            \n",
       "NA        NA               \n",
       "NA.1      NA               \n",
       "NA.2      NA               \n",
       "NA.3      NA               \n",
       "NA.4      NA               \n",
       "NA.5      NA               \n",
       "NA.6      NA               \n",
       "NA.7      NA               \n",
       "NA.8      NA               \n",
       "NA.9      NA               \n",
       "NA.10     NA               \n",
       "NA.11     NA               \n",
       "NA.12     NA               \n",
       "NA.13     NA               \n",
       "NA.14     NA               \n",
       "NA.15     NA               \n",
       "NA.16     NA               \n",
       "NA.17     NA               \n",
       "NA.18     NA               \n",
       "NA.19     NA               \n",
       "NA.20     NA               \n",
       "NA.21     NA               \n",
       "NA.22     NA               \n",
       "NA.23     NA               \n",
       "⋮         ⋮                \n",
       "NA.111373 NA               \n",
       "NA.111374 NA               \n",
       "NA.111375 NA               \n",
       "NA.111376 NA               \n",
       "NA.111377 NA               \n",
       "NA.111378 NA               \n",
       "NA.111379 NA               \n",
       "NA.111380 NA               \n",
       "NA.111381 NA               \n",
       "NA.111382 NA               \n",
       "NA.111383 NA               \n",
       "NA.111384 NA               \n",
       "NA.111385 NA               \n",
       "NA.111386 NA               \n",
       "NA.111387 NA               \n",
       "NA.111388 NA               \n",
       "NA.111389 NA               \n",
       "NA.111390 NA               \n",
       "NA.111391 NA               \n",
       "NA.111392 NA               \n",
       "NA.111393 NA               \n",
       "NA.111394 NA               \n",
       "NA.111395 NA               \n",
       "NA.111396 NA               \n",
       "NA.111397 NA               \n",
       "NA.111398 NA               \n",
       "NA.111399 NA               \n",
       "NA.111400 NA               \n",
       "NA.111401 NA               \n",
       "NA.111402 NA               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# extract gencode name and region type\n",
    "df_all_annot = df_all %>% left_join(df_gene_annot[,c(\"chr\",\"start\",\"end\",\"source\",\"gencode.reg.type\",\"gencode.gene.name\")], by=c(\"chr\",\"start\",\"end\",\"source\"))\n",
    "df_all_annot[df_all_annot$pub.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "id": "8b1695b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>gencode.gene.name.x</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106066032</td><td>106200973</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>TET2-AS1</td><td>promoter</td><td>TET2    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31349191</td><td> 31397165</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>DNMT3B  </td><td>promoter</td><td>DNMT3B  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209131798</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>IDH1-AS1</td><td>promoter</td><td>IDH1-AS1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140625729</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>U6      </td><td>promoter</td><td>U6      </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66762863</td><td> 66950461</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>AR      </td><td>promoter</td><td>AR      </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128746680</td><td>128755197</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>MYC     </td><td>promoter</td><td>CASC11  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & gencode.gene.name.x & gencode.reg.type.y & gencode.gene.name.y\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106066032 & 106200973 & + & Kukkonen et al. 2021 & TET2   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & TET2-AS1 & promoter & TET2    \\\\\n",
       "\t2 & 20 &  31349191 &  31397165 & + & Kukkonen et al. 2021 & DNMT3B & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & DNMT3B   & promoter & DNMT3B  \\\\\n",
       "\t3 & 2  & 209100953 & 209131798 & - & Kukkonen et al. 2021 & IDH1   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & IDH1-AS1 & promoter & IDH1-AS1\\\\\n",
       "\t4 & 7  & 140419127 & 140625729 & - & Kukkonen et al. 2021 & BRAF   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & U6       & promoter & U6      \\\\\n",
       "\t5 & X  &  66762863 &  66950461 & + & Kukkonen et al. 2021 & AR     & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & AR       & promoter & AR      \\\\\n",
       "\t6 & 8  & 128746680 & 128755197 & + & Kukkonen et al. 2021 & MYC    & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & MYC      & promoter & CASC11  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | gencode.gene.name.x &lt;chr&gt; | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106066032 | 106200973 | + | Kukkonen et al. 2021 | TET2   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | TET2-AS1 | promoter | TET2     |\n",
       "| 2 | 20 |  31349191 |  31397165 | + | Kukkonen et al. 2021 | DNMT3B | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | DNMT3B   | promoter | DNMT3B   |\n",
       "| 3 | 2  | 209100953 | 209131798 | - | Kukkonen et al. 2021 | IDH1   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | IDH1-AS1 | promoter | IDH1-AS1 |\n",
       "| 4 | 7  | 140419127 | 140625729 | - | Kukkonen et al. 2021 | BRAF   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | U6       | promoter | U6       |\n",
       "| 5 | X  |  66762863 |  66950461 | + | Kukkonen et al. 2021 | AR     | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | AR       | promoter | AR       |\n",
       "| 6 | 8  | 128746680 | 128755197 | + | Kukkonen et al. 2021 | MYC    | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | MYC      | promoter | CASC11   |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start     end       strand source               pub.gene.name\n",
       "1 4   106066032 106200973 +      Kukkonen et al. 2021 TET2         \n",
       "2 20   31349191  31397165 +      Kukkonen et al. 2021 DNMT3B       \n",
       "3 2   209100953 209131798 -      Kukkonen et al. 2021 IDH1         \n",
       "4 7   140419127 140625729 -      Kukkonen et al. 2021 BRAF         \n",
       "5 X    66762863  66950461 +      Kukkonen et al. 2021 AR           \n",
       "6 8   128746680 128755197 +      Kukkonen et al. 2021 MYC          \n",
       "  pub.reg.type       probe no.pubs gencode.reg.type.x gencode.gene.name.x\n",
       "1 gene_plus_promoter       1       gene_plus_promoter TET2-AS1           \n",
       "2 gene_plus_promoter       1       gene_plus_promoter DNMT3B             \n",
       "3 gene_plus_promoter       1       gene_plus_promoter IDH1-AS1           \n",
       "4 gene_plus_promoter       1       gene_plus_promoter U6                 \n",
       "5 gene_plus_promoter       1       gene_plus_promoter AR                 \n",
       "6 gene_plus_promoter       1       gene_plus_promoter MYC                \n",
       "  gencode.reg.type.y gencode.gene.name.y\n",
       "1 promoter           TET2               \n",
       "2 promoter           DNMT3B             \n",
       "3 promoter           IDH1-AS1           \n",
       "4 promoter           U6                 \n",
       "5 promoter           AR                 \n",
       "6 promoter           CASC11             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all_annot = df_all_annot %>% left_join(df_prom_annot[,c(\"chr\",\"start\",\"end\",\"source\",\"gencode.reg.type\",\"gencode.gene.name\")], by=c(\"chr\",\"start\",\"end\",\"source\"))\n",
    "head(df_all_annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "id": "0aab37ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all_annot = df_all_annot[!is.na(df_all_annot$chr),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "07ff578c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 111409 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>gencode.gene.name.x</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67354127</td><td>+ </td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>11</td><td>67350975</td><td>67350976</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>11</td><td>67351270</td><td>67351271</td><td>R </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>11</td><td>67351272</td><td>67351273</td><td>F </td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67351213</td><td>+ </td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>NA                    </td><td>NA   </td><td>NA                </td><td>NA        </td><td>NA</td><td>NA                </td><td>NA   </td><td>NA      </td><td>NA   </td><td>NA   </td><td>NA</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 111409 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & gencode.gene.name.x & gencode.reg.type.y & gencode.gene.name.y & hybrid.gene.name & no.total.pubs\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 11 & 67350213 & 67354127 & +  & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15\\\\\n",
       "\t 11 & 67350213 & 67354127 & +  & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15\\\\\n",
       "\t 11 & 67350975 & 67350976 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15\\\\\n",
       "\t 11 & 67351270 & 67351271 & R  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15\\\\\n",
       "\t 11 & 67351272 & 67351273 & F  & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15\\\\\n",
       "\t 11 & 67350213 & 67351213 & +  & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t NA &       NA &       NA & NA & NA                     & NA    & NA                 & NA         & NA & NA                 & NA    & NA       & NA    & NA    & NA\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 111409 × 15\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | gencode.gene.name.x &lt;chr&gt; | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 67350213 | 67354127 | +  | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |\n",
       "| 11 | 67350213 | 67354127 | +  | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |\n",
       "| 11 | 67350975 | 67350976 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |\n",
       "| 11 | 67351270 | 67351271 | R  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 |\n",
       "| 11 | 67351272 | 67351273 | F  | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 |\n",
       "| 11 | 67350213 | 67351213 | +  | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| NA |       NA |       NA | NA | NA                     | NA    | NA                 | NA         | NA | NA                 | NA    | NA       | NA    | NA    | NA |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "       chr start    end      strand source                 pub.gene.name\n",
       "1      11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "2      11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "3      11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "4      11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "5      11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "6      11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "7      NA        NA       NA NA     NA                     NA           \n",
       "8      NA        NA       NA NA     NA                     NA           \n",
       "9      NA        NA       NA NA     NA                     NA           \n",
       "10     NA        NA       NA NA     NA                     NA           \n",
       "11     NA        NA       NA NA     NA                     NA           \n",
       "12     NA        NA       NA NA     NA                     NA           \n",
       "13     NA        NA       NA NA     NA                     NA           \n",
       "14     NA        NA       NA NA     NA                     NA           \n",
       "15     NA        NA       NA NA     NA                     NA           \n",
       "16     NA        NA       NA NA     NA                     NA           \n",
       "17     NA        NA       NA NA     NA                     NA           \n",
       "18     NA        NA       NA NA     NA                     NA           \n",
       "19     NA        NA       NA NA     NA                     NA           \n",
       "20     NA        NA       NA NA     NA                     NA           \n",
       "21     NA        NA       NA NA     NA                     NA           \n",
       "22     NA        NA       NA NA     NA                     NA           \n",
       "23     NA        NA       NA NA     NA                     NA           \n",
       "24     NA        NA       NA NA     NA                     NA           \n",
       "25     NA        NA       NA NA     NA                     NA           \n",
       "26     NA        NA       NA NA     NA                     NA           \n",
       "27     NA        NA       NA NA     NA                     NA           \n",
       "28     NA        NA       NA NA     NA                     NA           \n",
       "29     NA        NA       NA NA     NA                     NA           \n",
       "30     NA        NA       NA NA     NA                     NA           \n",
       "⋮      ⋮   ⋮        ⋮        ⋮      ⋮                      ⋮            \n",
       "111380 NA  NA       NA       NA     NA                     NA           \n",
       "111381 NA  NA       NA       NA     NA                     NA           \n",
       "111382 NA  NA       NA       NA     NA                     NA           \n",
       "111383 NA  NA       NA       NA     NA                     NA           \n",
       "111384 NA  NA       NA       NA     NA                     NA           \n",
       "111385 NA  NA       NA       NA     NA                     NA           \n",
       "111386 NA  NA       NA       NA     NA                     NA           \n",
       "111387 NA  NA       NA       NA     NA                     NA           \n",
       "111388 NA  NA       NA       NA     NA                     NA           \n",
       "111389 NA  NA       NA       NA     NA                     NA           \n",
       "111390 NA  NA       NA       NA     NA                     NA           \n",
       "111391 NA  NA       NA       NA     NA                     NA           \n",
       "111392 NA  NA       NA       NA     NA                     NA           \n",
       "111393 NA  NA       NA       NA     NA                     NA           \n",
       "111394 NA  NA       NA       NA     NA                     NA           \n",
       "111395 NA  NA       NA       NA     NA                     NA           \n",
       "111396 NA  NA       NA       NA     NA                     NA           \n",
       "111397 NA  NA       NA       NA     NA                     NA           \n",
       "111398 NA  NA       NA       NA     NA                     NA           \n",
       "111399 NA  NA       NA       NA     NA                     NA           \n",
       "111400 NA  NA       NA       NA     NA                     NA           \n",
       "111401 NA  NA       NA       NA     NA                     NA           \n",
       "111402 NA  NA       NA       NA     NA                     NA           \n",
       "111403 NA  NA       NA       NA     NA                     NA           \n",
       "111404 NA  NA       NA       NA     NA                     NA           \n",
       "111405 NA  NA       NA       NA     NA                     NA           \n",
       "111406 NA  NA       NA       NA     NA                     NA           \n",
       "111407 NA  NA       NA       NA     NA                     NA           \n",
       "111408 NA  NA       NA       NA     NA                     NA           \n",
       "111409 NA  NA       NA       NA     NA                     NA           \n",
       "       pub.reg.type       probe      no.pubs gencode.reg.type.x\n",
       "1      gene_plus_promoter             1      gene_plus_promoter\n",
       "2      gene_plus_promoter            12      gene_plus_promoter\n",
       "3      EPIC_probe         cg02659086  1      gene_plus_promoter\n",
       "4      EPIC_probe         cg04920951  1      gene_plus_promoter\n",
       "5      EPIC_probe         cg26250609  1      gene_plus_promoter\n",
       "6      promoter                       1      gene_plus_promoter\n",
       "7      NA                 NA         NA      NA                \n",
       "8      NA                 NA         NA      NA                \n",
       "9      NA                 NA         NA      NA                \n",
       "10     NA                 NA         NA      NA                \n",
       "11     NA                 NA         NA      NA                \n",
       "12     NA                 NA         NA      NA                \n",
       "13     NA                 NA         NA      NA                \n",
       "14     NA                 NA         NA      NA                \n",
       "15     NA                 NA         NA      NA                \n",
       "16     NA                 NA         NA      NA                \n",
       "17     NA                 NA         NA      NA                \n",
       "18     NA                 NA         NA      NA                \n",
       "19     NA                 NA         NA      NA                \n",
       "20     NA                 NA         NA      NA                \n",
       "21     NA                 NA         NA      NA                \n",
       "22     NA                 NA         NA      NA                \n",
       "23     NA                 NA         NA      NA                \n",
       "24     NA                 NA         NA      NA                \n",
       "25     NA                 NA         NA      NA                \n",
       "26     NA                 NA         NA      NA                \n",
       "27     NA                 NA         NA      NA                \n",
       "28     NA                 NA         NA      NA                \n",
       "29     NA                 NA         NA      NA                \n",
       "30     NA                 NA         NA      NA                \n",
       "⋮      ⋮                  ⋮          ⋮       ⋮                 \n",
       "111380 NA                 NA         NA      NA                \n",
       "111381 NA                 NA         NA      NA                \n",
       "111382 NA                 NA         NA      NA                \n",
       "111383 NA                 NA         NA      NA                \n",
       "111384 NA                 NA         NA      NA                \n",
       "111385 NA                 NA         NA      NA                \n",
       "111386 NA                 NA         NA      NA                \n",
       "111387 NA                 NA         NA      NA                \n",
       "111388 NA                 NA         NA      NA                \n",
       "111389 NA                 NA         NA      NA                \n",
       "111390 NA                 NA         NA      NA                \n",
       "111391 NA                 NA         NA      NA                \n",
       "111392 NA                 NA         NA      NA                \n",
       "111393 NA                 NA         NA      NA                \n",
       "111394 NA                 NA         NA      NA                \n",
       "111395 NA                 NA         NA      NA                \n",
       "111396 NA                 NA         NA      NA                \n",
       "111397 NA                 NA         NA      NA                \n",
       "111398 NA                 NA         NA      NA                \n",
       "111399 NA                 NA         NA      NA                \n",
       "111400 NA                 NA         NA      NA                \n",
       "111401 NA                 NA         NA      NA                \n",
       "111402 NA                 NA         NA      NA                \n",
       "111403 NA                 NA         NA      NA                \n",
       "111404 NA                 NA         NA      NA                \n",
       "111405 NA                 NA         NA      NA                \n",
       "111406 NA                 NA         NA      NA                \n",
       "111407 NA                 NA         NA      NA                \n",
       "111408 NA                 NA         NA      NA                \n",
       "111409 NA                 NA         NA      NA                \n",
       "       gencode.gene.name.x gencode.reg.type.y gencode.gene.name.y\n",
       "1      GSTP1               promoter           GSTP1              \n",
       "2      GSTP1               promoter           GSTP1              \n",
       "3      GSTP1               promoter           GSTP1              \n",
       "4      GSTP1               .                  .                  \n",
       "5      GSTP1               .                  .                  \n",
       "6      GSTP1               promoter           GSTP1              \n",
       "7      NA                  NA                 NA                 \n",
       "8      NA                  NA                 NA                 \n",
       "9      NA                  NA                 NA                 \n",
       "10     NA                  NA                 NA                 \n",
       "11     NA                  NA                 NA                 \n",
       "12     NA                  NA                 NA                 \n",
       "13     NA                  NA                 NA                 \n",
       "14     NA                  NA                 NA                 \n",
       "15     NA                  NA                 NA                 \n",
       "16     NA                  NA                 NA                 \n",
       "17     NA                  NA                 NA                 \n",
       "18     NA                  NA                 NA                 \n",
       "19     NA                  NA                 NA                 \n",
       "20     NA                  NA                 NA                 \n",
       "21     NA                  NA                 NA                 \n",
       "22     NA                  NA                 NA                 \n",
       "23     NA                  NA                 NA                 \n",
       "24     NA                  NA                 NA                 \n",
       "25     NA                  NA                 NA                 \n",
       "26     NA                  NA                 NA                 \n",
       "27     NA                  NA                 NA                 \n",
       "28     NA                  NA                 NA                 \n",
       "29     NA                  NA                 NA                 \n",
       "30     NA                  NA                 NA                 \n",
       "⋮      ⋮                   ⋮                  ⋮                  \n",
       "111380 NA                  NA                 NA                 \n",
       "111381 NA                  NA                 NA                 \n",
       "111382 NA                  NA                 NA                 \n",
       "111383 NA                  NA                 NA                 \n",
       "111384 NA                  NA                 NA                 \n",
       "111385 NA                  NA                 NA                 \n",
       "111386 NA                  NA                 NA                 \n",
       "111387 NA                  NA                 NA                 \n",
       "111388 NA                  NA                 NA                 \n",
       "111389 NA                  NA                 NA                 \n",
       "111390 NA                  NA                 NA                 \n",
       "111391 NA                  NA                 NA                 \n",
       "111392 NA                  NA                 NA                 \n",
       "111393 NA                  NA                 NA                 \n",
       "111394 NA                  NA                 NA                 \n",
       "111395 NA                  NA                 NA                 \n",
       "111396 NA                  NA                 NA                 \n",
       "111397 NA                  NA                 NA                 \n",
       "111398 NA                  NA                 NA                 \n",
       "111399 NA                  NA                 NA                 \n",
       "111400 NA                  NA                 NA                 \n",
       "111401 NA                  NA                 NA                 \n",
       "111402 NA                  NA                 NA                 \n",
       "111403 NA                  NA                 NA                 \n",
       "111404 NA                  NA                 NA                 \n",
       "111405 NA                  NA                 NA                 \n",
       "111406 NA                  NA                 NA                 \n",
       "111407 NA                  NA                 NA                 \n",
       "111408 NA                  NA                 NA                 \n",
       "111409 NA                  NA                 NA                 \n",
       "       hybrid.gene.name no.total.pubs\n",
       "1      GSTP1            15           \n",
       "2      GSTP1            15           \n",
       "3      GSTP1            15           \n",
       "4      GSTP1            15           \n",
       "5      GSTP1            15           \n",
       "6      GSTP1            15           \n",
       "7      NA               NA           \n",
       "8      NA               NA           \n",
       "9      NA               NA           \n",
       "10     NA               NA           \n",
       "11     NA               NA           \n",
       "12     NA               NA           \n",
       "13     NA               NA           \n",
       "14     NA               NA           \n",
       "15     NA               NA           \n",
       "16     NA               NA           \n",
       "17     NA               NA           \n",
       "18     NA               NA           \n",
       "19     NA               NA           \n",
       "20     NA               NA           \n",
       "21     NA               NA           \n",
       "22     NA               NA           \n",
       "23     NA               NA           \n",
       "24     NA               NA           \n",
       "25     NA               NA           \n",
       "26     NA               NA           \n",
       "27     NA               NA           \n",
       "28     NA               NA           \n",
       "29     NA               NA           \n",
       "30     NA               NA           \n",
       "⋮      ⋮                ⋮            \n",
       "111380 NA               NA           \n",
       "111381 NA               NA           \n",
       "111382 NA               NA           \n",
       "111383 NA               NA           \n",
       "111384 NA               NA           \n",
       "111385 NA               NA           \n",
       "111386 NA               NA           \n",
       "111387 NA               NA           \n",
       "111388 NA               NA           \n",
       "111389 NA               NA           \n",
       "111390 NA               NA           \n",
       "111391 NA               NA           \n",
       "111392 NA               NA           \n",
       "111393 NA               NA           \n",
       "111394 NA               NA           \n",
       "111395 NA               NA           \n",
       "111396 NA               NA           \n",
       "111397 NA               NA           \n",
       "111398 NA               NA           \n",
       "111399 NA               NA           \n",
       "111400 NA               NA           \n",
       "111401 NA               NA           \n",
       "111402 NA               NA           \n",
       "111403 NA               NA           \n",
       "111404 NA               NA           \n",
       "111405 NA               NA           \n",
       "111406 NA               NA           \n",
       "111407 NA               NA           \n",
       "111408 NA               NA           \n",
       "111409 NA               NA           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all_annot$hybrid.gene.name = ifelse(df_all_annot$pub.gene.name==\"none\", \n",
    "                                        df_all_annot$gencode.gene.name.x,\n",
    "                                        df_all_annot$pub.gene.name)\n",
    "df_all_annot2 = df_all_annot %>% group_by(hybrid.gene.name) %>% \n",
    "   mutate(no.total.pubs= sum(no.pubs) - sum(duplicated(source))) #note there are no duplicate gene names for source Massie\n",
    "\n",
    "df_all_annot2[df_all_annot2$pub.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "109621c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "     1      2      3      4      5      6      7      8     15 \n",
       "  5442 113753    789    284     88     34     21     44      6 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#df = df_all_annot2[df_all_annot2$gencode.gene.name!=\"Y_RNA\",]\n",
    "table(df_all_annot2$no.total.pubs)\n",
    "#df_all_annot2[df_all_annot2$no.total.pubs==15,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "61128507",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>gencode.gene.name.x</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>in.gencode.promoter</th><th scope=col>gencode.gene.name</th><th scope=col>pub.strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><td>11</td><td>67350975</td><td>67350976</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>-</td></tr>\n",
       "\t<tr><td>11</td><td>67351270</td><td>67351271</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>-</td></tr>\n",
       "\t<tr><td>11</td><td>67351272</td><td>67351273</td><td>F</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67351213</td><td>+</td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 18\n",
       "\\begin{tabular}{llllllllllllllllll}\n",
       " chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & gencode.gene.name.x & gencode.reg.type.y & gencode.gene.name.y & hybrid.gene.name & no.total.pubs & in.gencode.promoter & gencode.gene.name & pub.strand\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 11 & 67350213 & 67354127 & + & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\t 11 & 67350213 & 67354127 & + & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\t 11 & 67350975 & 67350976 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & -\\\\\n",
       "\t 11 & 67351270 & 67351271 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & -\\\\\n",
       "\t 11 & 67351272 & 67351273 & F & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & +\\\\\n",
       "\t 11 & 67350213 & 67351213 & + & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 18\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | gencode.gene.name.x &lt;chr&gt; | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | in.gencode.promoter &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | pub.strand &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 67350213 | 67354127 | + | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "| 11 | 67350213 | 67354127 | + | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "| 11 | 67350975 | 67350976 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | - |\n",
       "| 11 | 67351270 | 67351271 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | - |\n",
       "| 11 | 67351272 | 67351273 | F | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | + |\n",
       "| 11 | 67350213 | 67351213 | + | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start    end      strand source                 pub.gene.name\n",
       "1 11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "2 11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "3 11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "4 11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "5 11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "6 11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "  pub.reg.type       probe      no.pubs gencode.reg.type.x gencode.gene.name.x\n",
       "1 gene_plus_promoter             1      gene_plus_promoter GSTP1              \n",
       "2 gene_plus_promoter            12      gene_plus_promoter GSTP1              \n",
       "3 EPIC_probe         cg02659086  1      gene_plus_promoter GSTP1              \n",
       "4 EPIC_probe         cg04920951  1      gene_plus_promoter GSTP1              \n",
       "5 EPIC_probe         cg26250609  1      gene_plus_promoter GSTP1              \n",
       "6 promoter                       1      gene_plus_promoter GSTP1              \n",
       "  gencode.reg.type.y gencode.gene.name.y hybrid.gene.name no.total.pubs\n",
       "1 promoter           GSTP1               GSTP1            15           \n",
       "2 promoter           GSTP1               GSTP1            15           \n",
       "3 promoter           GSTP1               GSTP1            15           \n",
       "4 .                  .                   GSTP1            15           \n",
       "5 .                  .                   GSTP1            15           \n",
       "6 promoter           GSTP1               GSTP1            15           \n",
       "  in.gencode.promoter gencode.gene.name pub.strand\n",
       "1  TRUE               GSTP1             +         \n",
       "2  TRUE               GSTP1             +         \n",
       "3  TRUE               GSTP1             -         \n",
       "4 FALSE               GSTP1             -         \n",
       "5 FALSE               GSTP1             +         \n",
       "6  TRUE               GSTP1             +         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all_annot2$in.gencode.promoter = ifelse(df_all_annot2$gencode.reg.type.y==\"promoter\", TRUE, FALSE)\n",
    "df_all_annot2$gencode.gene.name <- df_all_annot2$gencode.gene.name.x\n",
    "df_all_annot2$pub.strand <- ifelse(df_all_annot2$strand==\"R\",\"-\",\n",
    "                                     ifelse(df_all_annot2$strand==\"F\",\"+\",df_all_annot2$strand))\n",
    "df_all_annot2[df_all_annot2$gencode.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "id": "f30289f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "     1      2      3      4      5      6      7      8     15 \n",
       "  5442 113753    789    284     88     34     21     44      6 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols=c(\"chr\",\"start\",\"end\",\"hybrid.gene.name\",\"pub.strand\",\"pub.gene.name\",\"pub.reg.type\",\"gencode.gene.name\",\"in.gencode.promoter\",\"probe\",\"source\",\"no.pubs\",\"no.total.pubs\")\n",
    "write.table(df_all_annot2, file=\"./marker_collation/all_lit_regions_wcoords_annot_hg19_10052022.bed\",\n",
    "            row.names=FALSE,col.names=TRUE,sep=\"\\t\",quote=FALSE)\n",
    "table(df_all_annot2$no.total.pubs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "186f2912",
   "metadata": {},
   "source": [
    "## TCGA and ECM overlaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d491e83",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort -k1,1 -k2,2 -k3,3 -n all_lit_regions_wcoords_annot_hg19_10052022.bed >all_lit_regions_wcoords_annot_hg19_10052022_sorted.bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "affcd3d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# bedtools intersect -a all_lit_regions_wcoords_annot_hg19_10052022_sorted.bed   -b ECM_DMRs_raw_Hang_05172022_sorted.bed -wao > all_regions_ECM_overlap.out\n",
    "# bedtools intersect -a all_lit_regions_wcoords_annot_hg19_10052022_sorted.bed  -b ../TCGA_PRAD_DMRS_Hang_05172022_sorted.bed -wao > all_regions_TCGA_overlap.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "8a164dcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 38</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>⋯</th><th scope=col>V29</th><th scope=col>V30</th><th scope=col>V31</th><th scope=col>V32</th><th scope=col>V33</th><th scope=col>V34</th><th scope=col>V35</th><th scope=col>V36</th><th scope=col>V37</th><th scope=col>V38</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Un_gl000241</td><td> 613</td><td> 773</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Un_gl000235</td><td>1719</td><td>1794</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Un_gl000237</td><td>2192</td><td>2271</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Un_gl000229</td><td>2437</td><td>2503</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Un_gl000232</td><td>2702</td><td>3175</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Un_gl000229</td><td>2796</td><td>2914</td><td>.</td><td>Zhao et al 2020</td><td>NA</td><td>DMR</td><td>NA</td><td>1</td><td>.</td><td>⋯</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 38\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & ⋯ & V29 & V30 & V31 & V32 & V33 & V34 & V35 & V36 & V37 & V38\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Un\\_gl000241 &  613 &  773 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\t2 & Un\\_gl000235 & 1719 & 1794 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\t3 & Un\\_gl000237 & 2192 & 2271 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\t4 & Un\\_gl000229 & 2437 & 2503 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\t5 & Un\\_gl000232 & 2702 & 3175 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\t6 & Un\\_gl000229 & 2796 & 2914 & . & Zhao et al 2020 & NA & DMR & NA & 1 & . & ⋯ & . & . & . & . & . & . & . & . & . & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 38\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | ⋯ ⋯ | V29 &lt;chr&gt; | V30 &lt;chr&gt; | V31 &lt;chr&gt; | V32 &lt;chr&gt; | V33 &lt;chr&gt; | V34 &lt;chr&gt; | V35 &lt;chr&gt; | V36 &lt;chr&gt; | V37 &lt;chr&gt; | V38 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | Un_gl000241 |  613 |  773 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "| 2 | Un_gl000235 | 1719 | 1794 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "| 3 | Un_gl000237 | 2192 | 2271 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "| 4 | Un_gl000229 | 2437 | 2503 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "| 5 | Un_gl000232 | 2702 | 3175 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "| 6 | Un_gl000229 | 2796 | 2914 | . | Zhao et al 2020 | NA | DMR | NA | 1 | . | ⋯ | . | . | . | . | . | . | . | . | . | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1          V2   V3   V4 V5              V6 V7  V8 V9 V10 ⋯ V29 V30 V31 V32\n",
       "1 Un_gl000241  613  773 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "2 Un_gl000235 1719 1794 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "3 Un_gl000237 2192 2271 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "4 Un_gl000229 2437 2503 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "5 Un_gl000232 2702 3175 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "6 Un_gl000229 2796 2914 .  Zhao et al 2020 NA DMR NA 1  .   ⋯ .   .   .   .  \n",
       "  V33 V34 V35 V36 V37 V38\n",
       "1 .   .   .   .   .   0  \n",
       "2 .   .   .   .   .   0  \n",
       "3 .   .   .   .   .   0  \n",
       "4 .   .   .   .   .   0  \n",
       "5 .   .   .   .   .   0  \n",
       "6 .   .   .   .   .   0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>source</th><th scope=col>gencode.gene</th><th scope=col>bp.tcga.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Un_gl000241</td><td> 613</td><td> 773</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Un_gl000235</td><td>1719</td><td>1794</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Un_gl000237</td><td>2192</td><td>2271</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Un_gl000229</td><td>2437</td><td>2503</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Un_gl000232</td><td>2702</td><td>3175</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Un_gl000229</td><td>2796</td><td>2914</td><td>Zhao et al 2020</td><td>none</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & start & end & source & gencode.gene & bp.tcga.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Un\\_gl000241 &  613 &  773 & Zhao et al 2020 & none & 0\\\\\n",
       "\t2 & Un\\_gl000235 & 1719 & 1794 & Zhao et al 2020 & none & 0\\\\\n",
       "\t3 & Un\\_gl000237 & 2192 & 2271 & Zhao et al 2020 & none & 0\\\\\n",
       "\t4 & Un\\_gl000229 & 2437 & 2503 & Zhao et al 2020 & none & 0\\\\\n",
       "\t5 & Un\\_gl000232 & 2702 & 3175 & Zhao et al 2020 & none & 0\\\\\n",
       "\t6 & Un\\_gl000229 & 2796 & 2914 & Zhao et al 2020 & none & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | source &lt;chr&gt; | gencode.gene &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | Un_gl000241 |  613 |  773 | Zhao et al 2020 | none | 0 |\n",
       "| 2 | Un_gl000235 | 1719 | 1794 | Zhao et al 2020 | none | 0 |\n",
       "| 3 | Un_gl000237 | 2192 | 2271 | Zhao et al 2020 | none | 0 |\n",
       "| 4 | Un_gl000229 | 2437 | 2503 | Zhao et al 2020 | none | 0 |\n",
       "| 5 | Un_gl000232 | 2702 | 3175 | Zhao et al 2020 | none | 0 |\n",
       "| 6 | Un_gl000229 | 2796 | 2914 | Zhao et al 2020 | none | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr         start end  source          gencode.gene bp.tcga.overlap\n",
       "1 Un_gl000241  613   773 Zhao et al 2020 none         0              \n",
       "2 Un_gl000235 1719  1794 Zhao et al 2020 none         0              \n",
       "3 Un_gl000237 2192  2271 Zhao et al 2020 none         0              \n",
       "4 Un_gl000229 2437  2503 Zhao et al 2020 none         0              \n",
       "5 Un_gl000232 2702  3175 Zhao et al 2020 none         0              \n",
       "6 Un_gl000229 2796  2914 Zhao et al 2020 none         0              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_tcga_ol = read.table(file=\"./marker_collation/all_regions_TCGA_overlap.out\", sep=\"\\t\",header=FALSE)\n",
    "head(df_tcga_ol)\n",
    "df_tcga_ol = df_tcga_ol[,c(\"V1\",\"V2\",\"V3\",\"V5\",\"V6\",\"V38\")]\n",
    "colnames(df_tcga_ol) <- c(\"chr\",\"start\",\"end\",\"source\",\"gencode.gene\",\"bp.tcga.overlap\")\n",
    "df_tcga_ol$gencode.gene[is.na(df_tcga_ol$gencode.gene)] <- \"none\"\n",
    "head(df_tcga_ol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "c70f9db7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'chr', 'start', 'end'. You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "df_tcga_ol = df_tcga_ol %>% group_by(chr,start,end,source) %>% \n",
    "    summarize(bp.tcga.overlap=sum(bp.tcga.overlap)) %>%\n",
    "    distinct(chr,start,end,source,.keep_all=TRUE) #assumes tcga dmrs are nonoverlapping\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "id": "026d712b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>source</th><th scope=col>ecm.methyl.status</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Un_gl000241</td><td> 613</td><td> 773</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Un_gl000235</td><td>1719</td><td>1794</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Un_gl000237</td><td>2192</td><td>2271</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Un_gl000229</td><td>2437</td><td>2503</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Un_gl000232</td><td>2702</td><td>3175</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Un_gl000229</td><td>2796</td><td>2914</td><td>Zhao et al 2020</td><td>.</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & start & end & source & ecm.methyl.status & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Un\\_gl000241 &  613 &  773 & Zhao et al 2020 & . & 0\\\\\n",
       "\t2 & Un\\_gl000235 & 1719 & 1794 & Zhao et al 2020 & . & 0\\\\\n",
       "\t3 & Un\\_gl000237 & 2192 & 2271 & Zhao et al 2020 & . & 0\\\\\n",
       "\t4 & Un\\_gl000229 & 2437 & 2503 & Zhao et al 2020 & . & 0\\\\\n",
       "\t5 & Un\\_gl000232 & 2702 & 3175 & Zhao et al 2020 & . & 0\\\\\n",
       "\t6 & Un\\_gl000229 & 2796 & 2914 & Zhao et al 2020 & . & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | source &lt;chr&gt; | ecm.methyl.status &lt;chr&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | Un_gl000241 |  613 |  773 | Zhao et al 2020 | . | 0 |\n",
       "| 2 | Un_gl000235 | 1719 | 1794 | Zhao et al 2020 | . | 0 |\n",
       "| 3 | Un_gl000237 | 2192 | 2271 | Zhao et al 2020 | . | 0 |\n",
       "| 4 | Un_gl000229 | 2437 | 2503 | Zhao et al 2020 | . | 0 |\n",
       "| 5 | Un_gl000232 | 2702 | 3175 | Zhao et al 2020 | . | 0 |\n",
       "| 6 | Un_gl000229 | 2796 | 2914 | Zhao et al 2020 | . | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr         start end  source          ecm.methyl.status bp.ecm.overlap\n",
       "1 Un_gl000241  613   773 Zhao et al 2020 .                 0             \n",
       "2 Un_gl000235 1719  1794 Zhao et al 2020 .                 0             \n",
       "3 Un_gl000237 2192  2271 Zhao et al 2020 .                 0             \n",
       "4 Un_gl000229 2437  2503 Zhao et al 2020 .                 0             \n",
       "5 Un_gl000232 2702  3175 Zhao et al 2020 .                 0             \n",
       "6 Un_gl000229 2796  2914 Zhao et al 2020 .                 0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_ecm_ol = read.table(file=\"./marker_collation/all_regions_ECM_overlap.out\", sep=\"\\t\", header=FALSE)\n",
    "#head(df_ecm_ol)\n",
    "df_ecm_ol = df_ecm_ol[,c(\"V1\",\"V2\",\"V3\",\"V5\",'V25',\"V26\")]\n",
    "colnames(df_ecm_ol) <- c(\"chr\",\"start\",\"end\",\"source\",\"ecm.methyl.status\",\"bp.ecm.overlap\")\n",
    "head(df_ecm_ol)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "id": "5c9f5a9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'chr', 'start', 'end', 'source'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>source</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 29194</td><td> 29436</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td> 51588</td><td> 51682</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td>535828</td><td>535879</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td>566459</td><td>566739</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td>566813</td><td>566880</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td>567113</td><td>567239</td><td>Zhao et al 2020</td><td>0</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chr & start & end & source & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 &  29194 &  29436 & Zhao et al 2020 & 0 & .\\\\\n",
       "\t 1 &  51588 &  51682 & Zhao et al 2020 & 0 & .\\\\\n",
       "\t 1 & 535828 & 535879 & Zhao et al 2020 & 0 & .\\\\\n",
       "\t 1 & 566459 & 566739 & Zhao et al 2020 & 0 & .\\\\\n",
       "\t 1 & 566813 & 566880 & Zhao et al 2020 & 0 & .\\\\\n",
       "\t 1 & 567113 & 567239 & Zhao et al 2020 & 0 & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 6\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | source &lt;chr&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 |  29194 |  29436 | Zhao et al 2020 | 0 | . |\n",
       "| 1 |  51588 |  51682 | Zhao et al 2020 | 0 | . |\n",
       "| 1 | 535828 | 535879 | Zhao et al 2020 | 0 | . |\n",
       "| 1 | 566459 | 566739 | Zhao et al 2020 | 0 | . |\n",
       "| 1 | 566813 | 566880 | Zhao et al 2020 | 0 | . |\n",
       "| 1 | 567113 | 567239 | Zhao et al 2020 | 0 | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start  end    source          bp.ecm.overlap ecm.methyl.status\n",
       "1 1    29194  29436 Zhao et al 2020 0              .                \n",
       "2 1    51588  51682 Zhao et al 2020 0              .                \n",
       "3 1   535828 535879 Zhao et al 2020 0              .                \n",
       "4 1   566459 566739 Zhao et al 2020 0              .                \n",
       "5 1   566813 566880 Zhao et al 2020 0              .                \n",
       "6 1   567113 567239 Zhao et al 2020 0              .                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_ecm_ol = df_ecm_ol %>% group_by(chr,start,end,source) %>% \n",
    "    summarize(bp.ecm.overlap=sum(bp.ecm.overlap),ecm.methyl.status=ecm.methyl.status) %>%\n",
    "    distinct(chr,start,end,source,.keep_all=TRUE)#assumes ecm dmrs are nonoverlapping\n",
    "head(df_ecm_ol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "30ad0bea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>gencode.gene.name.x</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>in.gencode.promoter</th><th scope=col>gencode.gene.name</th><th scope=col>pub.strand</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>8</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>4834</th><td>11</td><td>67350975</td><td>67350976</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>-</td></tr>\n",
       "\t<tr><th scope=row>6180</th><td>11</td><td>67351270</td><td>67351271</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>-</td></tr>\n",
       "\t<tr><th scope=row>6619</th><td>11</td><td>67351272</td><td>67351273</td><td>F</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>+</td></tr>\n",
       "\t<tr><th scope=row>9051</th><td>11</td><td>67350213</td><td>67351213</td><td>+</td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>GSTP1</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & gencode.gene.name.x & gencode.reg.type.y & gencode.gene.name.y & hybrid.gene.name & no.total.pubs & in.gencode.promoter & gencode.gene.name & pub.strand\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t8 & 11 & 67350213 & 67354127 & + & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\t15 & 11 & 67350213 & 67354127 & + & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\t4834 & 11 & 67350975 & 67350976 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & -\\\\\n",
       "\t6180 & 11 & 67351270 & 67351271 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & -\\\\\n",
       "\t6619 & 11 & 67351272 & 67351273 & F & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & gene\\_plus\\_promoter & GSTP1 & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & +\\\\\n",
       "\t9051 & 11 & 67350213 & 67351213 & + & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & gene\\_plus\\_promoter & GSTP1 & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & +\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | gencode.gene.name.x &lt;chr&gt; | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | in.gencode.promoter &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | pub.strand &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 8 | 11 | 67350213 | 67354127 | + | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "| 15 | 11 | 67350213 | 67354127 | + | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "| 4834 | 11 | 67350975 | 67350976 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | - |\n",
       "| 6180 | 11 | 67351270 | 67351271 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | - |\n",
       "| 6619 | 11 | 67351272 | 67351273 | F | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | gene_plus_promoter | GSTP1 | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | + |\n",
       "| 9051 | 11 | 67350213 | 67351213 | + | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | gene_plus_promoter | GSTP1 | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + |\n",
       "\n"
      ],
      "text/plain": [
       "     chr start    end      strand source                 pub.gene.name\n",
       "8    11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "15   11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "4834 11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "6180 11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "6619 11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "9051 11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "     pub.reg.type       probe      no.pubs gencode.reg.type.x\n",
       "8    gene_plus_promoter             1      gene_plus_promoter\n",
       "15   gene_plus_promoter            12      gene_plus_promoter\n",
       "4834 EPIC_probe         cg02659086  1      gene_plus_promoter\n",
       "6180 EPIC_probe         cg04920951  1      gene_plus_promoter\n",
       "6619 EPIC_probe         cg26250609  1      gene_plus_promoter\n",
       "9051 promoter                       1      gene_plus_promoter\n",
       "     gencode.gene.name.x gencode.reg.type.y gencode.gene.name.y\n",
       "8    GSTP1               promoter           GSTP1              \n",
       "15   GSTP1               promoter           GSTP1              \n",
       "4834 GSTP1               promoter           GSTP1              \n",
       "6180 GSTP1               .                  .                  \n",
       "6619 GSTP1               .                  .                  \n",
       "9051 GSTP1               promoter           GSTP1              \n",
       "     hybrid.gene.name no.total.pubs in.gencode.promoter gencode.gene.name\n",
       "8    GSTP1            15             TRUE               GSTP1            \n",
       "15   GSTP1            15             TRUE               GSTP1            \n",
       "4834 GSTP1            15             TRUE               GSTP1            \n",
       "6180 GSTP1            15            FALSE               GSTP1            \n",
       "6619 GSTP1            15            FALSE               GSTP1            \n",
       "9051 GSTP1            15             TRUE               GSTP1            \n",
       "     pub.strand\n",
       "8    +         \n",
       "15   +         \n",
       "4834 -         \n",
       "6180 -         \n",
       "6619 +         \n",
       "9051 +         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all = read.table(file=\"./marker_collation/all_lit_regions_wcoords_annot_hg19_10052022.bed\", sep=\"\\t\",header=TRUE)\n",
    "#head(df_all)\n",
    "#df_all = df_all %>% rename(\"chr\"=\"V1\",\"start\"=\"V2\",\"end\"=\"V3\")\n",
    "df_all$gencode.gene.name[is.na(df_all$gencode.gene.name)] <- \"none\"\n",
    "df_all[df_all$gencode.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "2f0ccb79",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all_wtcga = df_all %>% left_join(df_tcga_ol[,c(\"chr\",\"start\",\"end\",\"source\",\"bp.tcga.overlap\")], by=c(\"chr\",\"start\",\"end\",\"source\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "65115503",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>⋯</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>in.gencode.promoter</th><th scope=col>gencode.gene.name</th><th scope=col>pub.strand</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106066032</td><td>106200973</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>TET2    </td><td>TET2  </td><td>4</td><td>TRUE</td><td>TET2-AS1</td><td>+</td><td>373</td><td> 630</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31349191</td><td> 31397165</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>DNMT3B  </td><td>DNMT3B</td><td>2</td><td>TRUE</td><td>DNMT3B  </td><td>+</td><td>  0</td><td> 806</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209131798</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>IDH1-AS1</td><td>IDH1  </td><td>1</td><td>TRUE</td><td>IDH1-AS1</td><td>-</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140625729</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>U6      </td><td>BRAF  </td><td>1</td><td>TRUE</td><td>U6      </td><td>-</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66762863</td><td> 66950461</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>AR      </td><td>AR    </td><td>2</td><td>TRUE</td><td>AR      </td><td>+</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128746680</td><td>128755197</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene_plus_promoter</td><td></td><td>1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>CASC11  </td><td>MYC   </td><td>3</td><td>TRUE</td><td>MYC     </td><td>+</td><td>  0</td><td>1954</td><td>hypo </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 21\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & ⋯ & gencode.reg.type.y & gencode.gene.name.y & hybrid.gene.name & no.total.pubs & in.gencode.promoter & gencode.gene.name & pub.strand & bp.tcga.overlap & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & ⋯ & <chr> & <chr> & <chr> & <int> & <lgl> & <chr> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106066032 & 106200973 & + & Kukkonen et al. 2021 & TET2   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & TET2     & TET2   & 4 & TRUE & TET2-AS1 & + & 373 &  630 & hyper\\\\\n",
       "\t2 & 20 &  31349191 &  31397165 & + & Kukkonen et al. 2021 & DNMT3B & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & DNMT3B   & DNMT3B & 2 & TRUE & DNMT3B   & + &   0 &  806 & hyper\\\\\n",
       "\t3 & 2  & 209100953 & 209131798 & - & Kukkonen et al. 2021 & IDH1   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & IDH1-AS1 & IDH1   & 1 & TRUE & IDH1-AS1 & - &   0 &    0 & .    \\\\\n",
       "\t4 & 7  & 140419127 & 140625729 & - & Kukkonen et al. 2021 & BRAF   & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & U6       & BRAF   & 1 & TRUE & U6       & - &   0 &    0 & .    \\\\\n",
       "\t5 & X  &  66762863 &  66950461 & + & Kukkonen et al. 2021 & AR     & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & AR       & AR     & 2 & TRUE & AR       & + &   0 &    0 & .    \\\\\n",
       "\t6 & 8  & 128746680 & 128755197 & + & Kukkonen et al. 2021 & MYC    & gene\\_plus\\_promoter &  & 1 & gene\\_plus\\_promoter & ⋯ & promoter & CASC11   & MYC    & 3 & TRUE & MYC      & + &   0 & 1954 & hypo \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 21\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | ⋯ ⋯ | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | in.gencode.promoter &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | pub.strand &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106066032 | 106200973 | + | Kukkonen et al. 2021 | TET2   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | TET2     | TET2   | 4 | TRUE | TET2-AS1 | + | 373 |  630 | hyper |\n",
       "| 2 | 20 |  31349191 |  31397165 | + | Kukkonen et al. 2021 | DNMT3B | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | DNMT3B   | DNMT3B | 2 | TRUE | DNMT3B   | + |   0 |  806 | hyper |\n",
       "| 3 | 2  | 209100953 | 209131798 | - | Kukkonen et al. 2021 | IDH1   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | IDH1-AS1 | IDH1   | 1 | TRUE | IDH1-AS1 | - |   0 |    0 | .     |\n",
       "| 4 | 7  | 140419127 | 140625729 | - | Kukkonen et al. 2021 | BRAF   | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | U6       | BRAF   | 1 | TRUE | U6       | - |   0 |    0 | .     |\n",
       "| 5 | X  |  66762863 |  66950461 | + | Kukkonen et al. 2021 | AR     | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | AR       | AR     | 2 | TRUE | AR       | + |   0 |    0 | .     |\n",
       "| 6 | 8  | 128746680 | 128755197 | + | Kukkonen et al. 2021 | MYC    | gene_plus_promoter | <!----> | 1 | gene_plus_promoter | ⋯ | promoter | CASC11   | MYC    | 3 | TRUE | MYC      | + |   0 | 1954 | hypo  |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start     end       strand source               pub.gene.name\n",
       "1 4   106066032 106200973 +      Kukkonen et al. 2021 TET2         \n",
       "2 20   31349191  31397165 +      Kukkonen et al. 2021 DNMT3B       \n",
       "3 2   209100953 209131798 -      Kukkonen et al. 2021 IDH1         \n",
       "4 7   140419127 140625729 -      Kukkonen et al. 2021 BRAF         \n",
       "5 X    66762863  66950461 +      Kukkonen et al. 2021 AR           \n",
       "6 8   128746680 128755197 +      Kukkonen et al. 2021 MYC          \n",
       "  pub.reg.type       probe no.pubs gencode.reg.type.x ⋯ gencode.reg.type.y\n",
       "1 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "2 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "3 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "4 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "5 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "6 gene_plus_promoter       1       gene_plus_promoter ⋯ promoter          \n",
       "  gencode.gene.name.y hybrid.gene.name no.total.pubs in.gencode.promoter\n",
       "1 TET2                TET2             4             TRUE               \n",
       "2 DNMT3B              DNMT3B           2             TRUE               \n",
       "3 IDH1-AS1            IDH1             1             TRUE               \n",
       "4 U6                  BRAF             1             TRUE               \n",
       "5 AR                  AR               2             TRUE               \n",
       "6 CASC11              MYC              3             TRUE               \n",
       "  gencode.gene.name pub.strand bp.tcga.overlap bp.ecm.overlap ecm.methyl.status\n",
       "1 TET2-AS1          +          373              630           hyper            \n",
       "2 DNMT3B            +            0              806           hyper            \n",
       "3 IDH1-AS1          -            0                0           .                \n",
       "4 U6                -            0                0           .                \n",
       "5 AR                +            0                0           .                \n",
       "6 MYC               +            0             1954           hypo             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all_w_tcga_ecm = df_all_wtcga %>% left_join(df_ecm_ol, \n",
    "                                               by=c(\"chr\",\"start\",\"end\",\"source\"))\n",
    "head(df_all_w_tcga_ecm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "5e8802fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>no.pubs</th><th scope=col>gencode.reg.type.x</th><th scope=col>⋯</th><th scope=col>gencode.reg.type.y</th><th scope=col>gencode.gene.name.y</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>in.gencode.promoter</th><th scope=col>gencode.gene.name</th><th scope=col>pub.strand</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>8</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Constancio et al. 2019</td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td><td>0</td><td>849</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>11</td><td>67350213</td><td>67354127</td><td>+</td><td>Massie et al. 2017    </td><td>GSTP1</td><td>gene_plus_promoter</td><td>          </td><td>12</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td><td>0</td><td>849</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>4834</th><td>11</td><td>67350975</td><td>67350976</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg02659086</td><td> 1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>-</td><td>0</td><td>  0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>6180</th><td>11</td><td>67351270</td><td>67351271</td><td>R</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg04920951</td><td> 1</td><td>gene_plus_promoter</td><td>⋯</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>-</td><td>0</td><td>  0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>6619</th><td>11</td><td>67351272</td><td>67351273</td><td>F</td><td>Ylitalo et al 2021    </td><td>GSTP1</td><td>EPIC_probe        </td><td>cg26250609</td><td> 1</td><td>gene_plus_promoter</td><td>⋯</td><td>.       </td><td>.    </td><td>GSTP1</td><td>15</td><td>FALSE</td><td>GSTP1</td><td>+</td><td>0</td><td>  0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>9051</th><td>11</td><td>67350213</td><td>67351213</td><td>+</td><td>Sugiura et al 2021    </td><td>GSTP1</td><td>promoter          </td><td>          </td><td> 1</td><td>gene_plus_promoter</td><td>⋯</td><td>promoter</td><td>GSTP1</td><td>GSTP1</td><td>15</td><td> TRUE</td><td>GSTP1</td><td>+</td><td>0</td><td>  0</td><td>.    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 21\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr & start & end & strand & source & pub.gene.name & pub.reg.type & probe & no.pubs & gencode.reg.type.x & ⋯ & gencode.reg.type.y & gencode.gene.name.y & hybrid.gene.name & no.total.pubs & in.gencode.promoter & gencode.gene.name & pub.strand & bp.tcga.overlap & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & ⋯ & <chr> & <chr> & <chr> & <int> & <lgl> & <chr> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t8 & 11 & 67350213 & 67354127 & + & Constancio et al. 2019 & GSTP1 & gene\\_plus\\_promoter &            &  1 & gene\\_plus\\_promoter & ⋯ & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & + & 0 & 849 & hyper\\\\\n",
       "\t15 & 11 & 67350213 & 67354127 & + & Massie et al. 2017     & GSTP1 & gene\\_plus\\_promoter &            & 12 & gene\\_plus\\_promoter & ⋯ & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & + & 0 & 849 & hyper\\\\\n",
       "\t4834 & 11 & 67350975 & 67350976 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg02659086 &  1 & gene\\_plus\\_promoter & ⋯ & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & - & 0 &   0 & .    \\\\\n",
       "\t6180 & 11 & 67351270 & 67351271 & R & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg04920951 &  1 & gene\\_plus\\_promoter & ⋯ & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & - & 0 &   0 & .    \\\\\n",
       "\t6619 & 11 & 67351272 & 67351273 & F & Ylitalo et al 2021     & GSTP1 & EPIC\\_probe         & cg26250609 &  1 & gene\\_plus\\_promoter & ⋯ & .        & .     & GSTP1 & 15 & FALSE & GSTP1 & + & 0 &   0 & .    \\\\\n",
       "\t9051 & 11 & 67350213 & 67351213 & + & Sugiura et al 2021     & GSTP1 & promoter           &            &  1 & gene\\_plus\\_promoter & ⋯ & promoter & GSTP1 & GSTP1 & 15 &  TRUE & GSTP1 & + & 0 &   0 & .    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 21\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | no.pubs &lt;int&gt; | gencode.reg.type.x &lt;chr&gt; | ⋯ ⋯ | gencode.reg.type.y &lt;chr&gt; | gencode.gene.name.y &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | in.gencode.promoter &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | pub.strand &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 8 | 11 | 67350213 | 67354127 | + | Constancio et al. 2019 | GSTP1 | gene_plus_promoter | <!----> |  1 | gene_plus_promoter | ⋯ | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + | 0 | 849 | hyper |\n",
       "| 15 | 11 | 67350213 | 67354127 | + | Massie et al. 2017     | GSTP1 | gene_plus_promoter | <!----> | 12 | gene_plus_promoter | ⋯ | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + | 0 | 849 | hyper |\n",
       "| 4834 | 11 | 67350975 | 67350976 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg02659086 |  1 | gene_plus_promoter | ⋯ | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | - | 0 |   0 | .     |\n",
       "| 6180 | 11 | 67351270 | 67351271 | R | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg04920951 |  1 | gene_plus_promoter | ⋯ | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | - | 0 |   0 | .     |\n",
       "| 6619 | 11 | 67351272 | 67351273 | F | Ylitalo et al 2021     | GSTP1 | EPIC_probe         | cg26250609 |  1 | gene_plus_promoter | ⋯ | .        | .     | GSTP1 | 15 | FALSE | GSTP1 | + | 0 |   0 | .     |\n",
       "| 9051 | 11 | 67350213 | 67351213 | + | Sugiura et al 2021     | GSTP1 | promoter           | <!----> |  1 | gene_plus_promoter | ⋯ | promoter | GSTP1 | GSTP1 | 15 |  TRUE | GSTP1 | + | 0 |   0 | .     |\n",
       "\n"
      ],
      "text/plain": [
       "     chr start    end      strand source                 pub.gene.name\n",
       "8    11  67350213 67354127 +      Constancio et al. 2019 GSTP1        \n",
       "15   11  67350213 67354127 +      Massie et al. 2017     GSTP1        \n",
       "4834 11  67350975 67350976 R      Ylitalo et al 2021     GSTP1        \n",
       "6180 11  67351270 67351271 R      Ylitalo et al 2021     GSTP1        \n",
       "6619 11  67351272 67351273 F      Ylitalo et al 2021     GSTP1        \n",
       "9051 11  67350213 67351213 +      Sugiura et al 2021     GSTP1        \n",
       "     pub.reg.type       probe      no.pubs gencode.reg.type.x ⋯\n",
       "8    gene_plus_promoter             1      gene_plus_promoter ⋯\n",
       "15   gene_plus_promoter            12      gene_plus_promoter ⋯\n",
       "4834 EPIC_probe         cg02659086  1      gene_plus_promoter ⋯\n",
       "6180 EPIC_probe         cg04920951  1      gene_plus_promoter ⋯\n",
       "6619 EPIC_probe         cg26250609  1      gene_plus_promoter ⋯\n",
       "9051 promoter                       1      gene_plus_promoter ⋯\n",
       "     gencode.reg.type.y gencode.gene.name.y hybrid.gene.name no.total.pubs\n",
       "8    promoter           GSTP1               GSTP1            15           \n",
       "15   promoter           GSTP1               GSTP1            15           \n",
       "4834 promoter           GSTP1               GSTP1            15           \n",
       "6180 .                  .                   GSTP1            15           \n",
       "6619 .                  .                   GSTP1            15           \n",
       "9051 promoter           GSTP1               GSTP1            15           \n",
       "     in.gencode.promoter gencode.gene.name pub.strand bp.tcga.overlap\n",
       "8     TRUE               GSTP1             +          0              \n",
       "15    TRUE               GSTP1             +          0              \n",
       "4834  TRUE               GSTP1             -          0              \n",
       "6180 FALSE               GSTP1             -          0              \n",
       "6619 FALSE               GSTP1             +          0              \n",
       "9051  TRUE               GSTP1             +          0              \n",
       "     bp.ecm.overlap ecm.methyl.status\n",
       "8    849            hyper            \n",
       "15   849            hyper            \n",
       "4834   0            .                \n",
       "6180   0            .                \n",
       "6619   0            .                \n",
       "9051   0            .                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all_w_tcga_ecm[df_all_w_tcga_ecm$gencode.gene.name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "64398543",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all_w_tcga_ecm = df_all_w_tcga_ecm %>% select(-gencode.reg.type.x, -gencode.reg.type.y,-gencode.gene.name.x,-gencode.gene.name.y)\n",
    "write.table(df_all_w_tcga_ecm, col.names=TRUE, row.names=FALSE, quote=FALSE, sep=\"\\t\",\n",
    "           file=\"./marker_collation/all_lit_regions_annotated_overlaps_10052022.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "720e20fe",
   "metadata": {},
   "source": [
    "## Summarize (UPDATE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52b001d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# pie chart number of publications\n",
    "# pie chart number genes vs intergenic\n",
    "# pie chart intron vs exon etc. \n",
    "# pie chart yes/no overlap with TCGA and ECM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "17efc318",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 23</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr_hg19</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>chr_hg38</th><th scope=col>start_hg38</th><th scope=col>end_hg38</th><th scope=col>pub.strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>⋯</th><th scope=col>in.gencode.gene</th><th scope=col>gencode.gene.name</th><th scope=col>in.gencode.promoter</th><th scope=col>is.ingergenic</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>region</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106067032</td><td>106200973</td><td>chr4 </td><td>105145875</td><td>105279816</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>TET2-AS1</td><td>TRUE</td><td>FALSE</td><td>TET2  </td><td>4</td><td>4 : 105145875 - 105279816</td><td>373</td><td> 174</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31350191</td><td> 31397165</td><td>chr20</td><td> 32762385</td><td> 32809359</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene</td><td>⋯</td><td>TRUE</td><td>DNMT3B  </td><td>TRUE</td><td>FALSE</td><td>DNMT3B</td><td>2</td><td>20 : 32762385 - 32809359 </td><td>  0</td><td> 806</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209130798</td><td>chr2 </td><td>208236229</td><td>208266074</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>IDH1-AS1</td><td>TRUE</td><td>FALSE</td><td>IDH1  </td><td>1</td><td>2 : 208236229 - 208266074</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140624729</td><td>chr7 </td><td>140719327</td><td>140924929</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>RNU6-85P</td><td>TRUE</td><td>FALSE</td><td>BRAF  </td><td>1</td><td>7 : 140719327 - 140924929</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66763863</td><td> 66950461</td><td>chrX </td><td> 67544021</td><td> 67730619</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene</td><td>⋯</td><td>TRUE</td><td>AR      </td><td>TRUE</td><td>FALSE</td><td>AR    </td><td>2</td><td>X : 67544021 - 67730619  </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128747680</td><td>128755197</td><td>chr8 </td><td>127735434</td><td>127742951</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene</td><td>⋯</td><td>TRUE</td><td>MYC     </td><td>TRUE</td><td>FALSE</td><td>MYC   </td><td>3</td><td>8 : 127735434 - 127742951</td><td>  0</td><td>1954</td><td>hypo </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 23\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr\\_hg19 & start\\_hg19 & end\\_hg19 & chr\\_hg38 & start\\_hg38 & end\\_hg38 & pub.strand & source & pub.gene.name & pub.reg.type & ⋯ & in.gencode.gene & gencode.gene.name & in.gencode.promoter & is.ingergenic & hybrid.gene.name & no.total.pubs & region & bp.tcga.overlap & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <chr> & <lgl> & <lgl> & <chr> & <int> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106067032 & 106200973 & chr4  & 105145875 & 105279816 & + & Kukkonen et al. 2021 & TET2   & gene & ⋯ & TRUE & TET2-AS1 & TRUE & FALSE & TET2   & 4 & 4 : 105145875 - 105279816 & 373 &  174 & hyper\\\\\n",
       "\t2 & 20 &  31350191 &  31397165 & chr20 &  32762385 &  32809359 & + & Kukkonen et al. 2021 & DNMT3B & gene & ⋯ & TRUE & DNMT3B   & TRUE & FALSE & DNMT3B & 2 & 20 : 32762385 - 32809359  &   0 &  806 & hyper\\\\\n",
       "\t3 & 2  & 209100953 & 209130798 & chr2  & 208236229 & 208266074 & - & Kukkonen et al. 2021 & IDH1   & gene & ⋯ & TRUE & IDH1-AS1 & TRUE & FALSE & IDH1   & 1 & 2 : 208236229 - 208266074 &   0 &    0 & .    \\\\\n",
       "\t4 & 7  & 140419127 & 140624729 & chr7  & 140719327 & 140924929 & - & Kukkonen et al. 2021 & BRAF   & gene & ⋯ & TRUE & RNU6-85P & TRUE & FALSE & BRAF   & 1 & 7 : 140719327 - 140924929 &   0 &    0 & .    \\\\\n",
       "\t5 & X  &  66763863 &  66950461 & chrX  &  67544021 &  67730619 & + & Kukkonen et al. 2021 & AR     & gene & ⋯ & TRUE & AR       & TRUE & FALSE & AR     & 2 & X : 67544021 - 67730619   &   0 &    0 & .    \\\\\n",
       "\t6 & 8  & 128747680 & 128755197 & chr8  & 127735434 & 127742951 & + & Kukkonen et al. 2021 & MYC    & gene & ⋯ & TRUE & MYC      & TRUE & FALSE & MYC    & 3 & 8 : 127735434 - 127742951 &   0 & 1954 & hypo \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 23\n",
       "\n",
       "| <!--/--> | chr_hg19 &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | chr_hg38 &lt;chr&gt; | start_hg38 &lt;int&gt; | end_hg38 &lt;int&gt; | pub.strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | ⋯ ⋯ | in.gencode.gene &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | in.gencode.promoter &lt;lgl&gt; | is.ingergenic &lt;lgl&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | region &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106067032 | 106200973 | chr4  | 105145875 | 105279816 | + | Kukkonen et al. 2021 | TET2   | gene | ⋯ | TRUE | TET2-AS1 | TRUE | FALSE | TET2   | 4 | 4 : 105145875 - 105279816 | 373 |  174 | hyper |\n",
       "| 2 | 20 |  31350191 |  31397165 | chr20 |  32762385 |  32809359 | + | Kukkonen et al. 2021 | DNMT3B | gene | ⋯ | TRUE | DNMT3B   | TRUE | FALSE | DNMT3B | 2 | 20 : 32762385 - 32809359  |   0 |  806 | hyper |\n",
       "| 3 | 2  | 209100953 | 209130798 | chr2  | 208236229 | 208266074 | - | Kukkonen et al. 2021 | IDH1   | gene | ⋯ | TRUE | IDH1-AS1 | TRUE | FALSE | IDH1   | 1 | 2 : 208236229 - 208266074 |   0 |    0 | .     |\n",
       "| 4 | 7  | 140419127 | 140624729 | chr7  | 140719327 | 140924929 | - | Kukkonen et al. 2021 | BRAF   | gene | ⋯ | TRUE | RNU6-85P | TRUE | FALSE | BRAF   | 1 | 7 : 140719327 - 140924929 |   0 |    0 | .     |\n",
       "| 5 | X  |  66763863 |  66950461 | chrX  |  67544021 |  67730619 | + | Kukkonen et al. 2021 | AR     | gene | ⋯ | TRUE | AR       | TRUE | FALSE | AR     | 2 | X : 67544021 - 67730619   |   0 |    0 | .     |\n",
       "| 6 | 8  | 128747680 | 128755197 | chr8  | 127735434 | 127742951 | + | Kukkonen et al. 2021 | MYC    | gene | ⋯ | TRUE | MYC      | TRUE | FALSE | MYC    | 3 | 8 : 127735434 - 127742951 |   0 | 1954 | hypo  |\n",
       "\n"
      ],
      "text/plain": [
       "  chr_hg19 start_hg19 end_hg19  chr_hg38 start_hg38 end_hg38  pub.strand\n",
       "1 4        106067032  106200973 chr4     105145875  105279816 +         \n",
       "2 20        31350191   31397165 chr20     32762385   32809359 +         \n",
       "3 2        209100953  209130798 chr2     208236229  208266074 -         \n",
       "4 7        140419127  140624729 chr7     140719327  140924929 -         \n",
       "5 X         66763863   66950461 chrX      67544021   67730619 +         \n",
       "6 8        128747680  128755197 chr8     127735434  127742951 +         \n",
       "  source               pub.gene.name pub.reg.type ⋯ in.gencode.gene\n",
       "1 Kukkonen et al. 2021 TET2          gene         ⋯ TRUE           \n",
       "2 Kukkonen et al. 2021 DNMT3B        gene         ⋯ TRUE           \n",
       "3 Kukkonen et al. 2021 IDH1          gene         ⋯ TRUE           \n",
       "4 Kukkonen et al. 2021 BRAF          gene         ⋯ TRUE           \n",
       "5 Kukkonen et al. 2021 AR            gene         ⋯ TRUE           \n",
       "6 Kukkonen et al. 2021 MYC           gene         ⋯ TRUE           \n",
       "  gencode.gene.name in.gencode.promoter is.ingergenic hybrid.gene.name\n",
       "1 TET2-AS1          TRUE                FALSE         TET2            \n",
       "2 DNMT3B            TRUE                FALSE         DNMT3B          \n",
       "3 IDH1-AS1          TRUE                FALSE         IDH1            \n",
       "4 RNU6-85P          TRUE                FALSE         BRAF            \n",
       "5 AR                TRUE                FALSE         AR              \n",
       "6 MYC               TRUE                FALSE         MYC             \n",
       "  no.total.pubs region                    bp.tcga.overlap bp.ecm.overlap\n",
       "1 4             4 : 105145875 - 105279816 373              174          \n",
       "2 2             20 : 32762385 - 32809359    0              806          \n",
       "3 1             2 : 208236229 - 208266074   0                0          \n",
       "4 1             7 : 140719327 - 140924929   0                0          \n",
       "5 2             X : 67544021 - 67730619     0                0          \n",
       "6 3             8 : 127735434 - 127742951   0             1954          \n",
       "  ecm.methyl.status\n",
       "1 hyper            \n",
       "2 hyper            \n",
       "3 .                \n",
       "4 .                \n",
       "5 .                \n",
       "6 hypo             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all = read.table(file=\"./marker_collation/all_lit_regions_annotated_overlaps_05202022.tsv\", header=TRUE, sep=\"\\t\")\n",
    "head(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "fe1a2eb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 23</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr_hg19</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>chr_hg38</th><th scope=col>start_hg38</th><th scope=col>end_hg38</th><th scope=col>pub.strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>⋯</th><th scope=col>in.gencode.gene</th><th scope=col>gencode.gene.name</th><th scope=col>in.gencode.promoter</th><th scope=col>is.ingergenic</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>region</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>22</td><td> 43088127</td><td> 43117304</td><td>chr22</td><td> 42692121</td><td> 42721298</td><td>-</td><td>Massie et al. 2017</td><td>A4GALT</td><td>gene      </td><td>⋯</td><td>TRUE</td><td>CYB5R3         </td><td> TRUE</td><td>FALSE</td><td>A4GALT</td><td>2</td><td>22 : 42692121 - 42721298  </td><td>  0</td><td>1420</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>16</td><td>  2325882</td><td>  2390747</td><td>chr16</td><td>  2275881</td><td>  2340746</td><td>-</td><td>Massie et al. 2017</td><td>ABCA3 </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>ENSG00000243445</td><td> TRUE</td><td>FALSE</td><td>ABCA3 </td><td>2</td><td>16 : 2275881 - 2340746    </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>220074494</td><td>220083716</td><td>chr2 </td><td>219209772</td><td>219218994</td><td>-</td><td>Massie et al. 2017</td><td>ABCB6 </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>ABCB6          </td><td> TRUE</td><td>FALSE</td><td>ABCB6 </td><td>2</td><td>2 : 219209772 - 219218994 </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>21</td><td> 43619799</td><td> 43717354</td><td>chr21</td><td> 42199689</td><td> 42297244</td><td>+</td><td>Massie et al. 2017</td><td>ABCG1 </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>RNA5SP492      </td><td> TRUE</td><td>FALSE</td><td>ABCG1 </td><td>2</td><td>21 : 42199689 - 42297244  </td><td>  0</td><td> 761</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>10</td><td>116190871</td><td>116527820</td><td>chr10</td><td>114431112</td><td>114768061</td><td>-</td><td>Massie et al. 2017</td><td>ABLIM1</td><td>gene      </td><td>⋯</td><td>TRUE</td><td>ENSG00000229668</td><td> TRUE</td><td>FALSE</td><td>ABLIM1</td><td>2</td><td>10 : 114431112 - 114768061</td><td>  0</td><td> 633</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>10</td><td>116264861</td><td>116264863</td><td>chr10</td><td>114505102</td><td>114505104</td><td>+</td><td>Ylitalo et al 2021</td><td>ABLIM1</td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ABLIM1         </td><td>FALSE</td><td>FALSE</td><td>ABLIM1</td><td>2</td><td>10 : 114505102 - 114505104</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>7</th><td>4 </td><td>  7967037</td><td>  8160559</td><td>chr4 </td><td>  7965310</td><td>  8158832</td><td>-</td><td>Massie et al. 2017</td><td>ABLIM2</td><td>gene      </td><td>⋯</td><td>TRUE</td><td>MIR95          </td><td> TRUE</td><td>FALSE</td><td>ABLIM2</td><td>2</td><td>4 : 7965310 - 8158832     </td><td>  0</td><td>2143</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>4 </td><td>  8073800</td><td>  8073802</td><td>chr4 </td><td>  8072073</td><td>  8072075</td><td>-</td><td>Ylitalo et al 2021</td><td>ABLIM2</td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ABLIM2         </td><td>FALSE</td><td>FALSE</td><td>ABLIM2</td><td>2</td><td>4 : 8072073 - 8072075     </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>9</th><td>17</td><td>   906759</td><td>  1133032</td><td>chr17</td><td>  1003519</td><td>  1229738</td><td>-</td><td>Massie et al. 2017</td><td>ABR   </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>MIR3183        </td><td> TRUE</td><td>FALSE</td><td>ABR   </td><td>2</td><td>17 : 1003519 - 1229738    </td><td>  0</td><td>4101</td><td>hypo </td></tr>\n",
       "\t<tr><th scope=row>10</th><td>17</td><td>  1058209</td><td>  1058211</td><td>chr17</td><td>  1154916</td><td>  1154918</td><td>-</td><td>Ylitalo et al 2021</td><td>ABR   </td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ABR            </td><td>FALSE</td><td>FALSE</td><td>ABR   </td><td>2</td><td>17 : 1154916 - 1154918    </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>11</th><td>16</td><td> 67691415</td><td> 67694713</td><td>chr16</td><td> 67657512</td><td> 67660810</td><td>-</td><td>Massie et al. 2017</td><td>ACD   </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>ACD            </td><td> TRUE</td><td>FALSE</td><td>ACD   </td><td>2</td><td>16 : 67657512 - 67660810  </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>12</th><td>1 </td><td> 55007930</td><td> 55104865</td><td>chr1 </td><td> 54542257</td><td> 54639192</td><td>+</td><td>Massie et al. 2017</td><td>ACOT11</td><td>gene      </td><td>⋯</td><td>TRUE</td><td>FAM151A        </td><td> TRUE</td><td>FALSE</td><td>ACOT11</td><td>2</td><td>1 : 54542257 - 54639192   </td><td>488</td><td>2554</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>1 </td><td> 55013708</td><td> 55013710</td><td>chr1 </td><td> 54548035</td><td> 54548037</td><td>+</td><td>Ylitalo et al 2021</td><td>ACOT11</td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ACOT11         </td><td>FALSE</td><td>FALSE</td><td>ACOT11</td><td>2</td><td>1 : 54548035 - 54548037   </td><td>  0</td><td>   2</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>1 </td><td> 55013418</td><td> 55013420</td><td>chr1 </td><td> 54547745</td><td> 54547747</td><td>+</td><td>Ylitalo et al 2021</td><td>ACOT11</td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ACOT11         </td><td>FALSE</td><td>FALSE</td><td>ACOT11</td><td>2</td><td>1 : 54547745 - 54547747   </td><td>  0</td><td>   2</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>1 </td><td> 55013734</td><td> 55013736</td><td>chr1 </td><td> 54548061</td><td> 54548063</td><td>+</td><td>Ylitalo et al 2021</td><td>ACOT11</td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ACOT11         </td><td>FALSE</td><td>FALSE</td><td>ACOT11</td><td>2</td><td>1 : 54548061 - 54548063   </td><td>  0</td><td>   2</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>2 </td><td>111490116</td><td>111878187</td><td>chr2 </td><td>110732539</td><td>111120610</td><td>+</td><td>Massie et al. 2017</td><td>ACOXL </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>RPL5P9         </td><td> TRUE</td><td>FALSE</td><td>ACOXL </td><td>2</td><td>2 : 110732539 - 111120610 </td><td>  0</td><td>1977</td><td>hypo </td></tr>\n",
       "\t<tr><th scope=row>17</th><td>20</td><td> 24986866</td><td> 25039616</td><td>chr20</td><td> 25006230</td><td> 25058980</td><td>-</td><td>Massie et al. 2017</td><td>ACSS1 </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>ACSS1          </td><td> TRUE</td><td>FALSE</td><td>ACSS1 </td><td>2</td><td>20 : 25006230 - 25058980  </td><td>  0</td><td> 958</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>20</td><td> 25013228</td><td> 25013230</td><td>chr20</td><td> 25032592</td><td> 25032594</td><td>-</td><td>Ylitalo et al 2021</td><td>ACSS1 </td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ACSS1          </td><td>FALSE</td><td>FALSE</td><td>ACSS1 </td><td>2</td><td>20 : 25032592 - 25032594  </td><td>  0</td><td>   2</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>20</td><td> 25013522</td><td> 25013524</td><td>chr20</td><td> 25032886</td><td> 25032888</td><td>-</td><td>Ylitalo et al 2021</td><td>ACSS1 </td><td>EPIC_probe</td><td>⋯</td><td>TRUE</td><td>ACSS1          </td><td>FALSE</td><td>FALSE</td><td>ACSS1 </td><td>2</td><td>20 : 25032886 - 25032888  </td><td>  0</td><td>   2</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>12</td><td> 81330193</td><td> 81654989</td><td>chr12</td><td> 80936414</td><td> 81261210</td><td>+</td><td>Massie et al. 2017</td><td>ACSS3 </td><td>gene      </td><td>⋯</td><td>TRUE</td><td>MIR4699        </td><td> TRUE</td><td>FALSE</td><td>ACSS3 </td><td>2</td><td>12 : 80936414 - 81261210  </td><td>866</td><td> 356</td><td>hyper</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 23\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr\\_hg19 & start\\_hg19 & end\\_hg19 & chr\\_hg38 & start\\_hg38 & end\\_hg38 & pub.strand & source & pub.gene.name & pub.reg.type & ⋯ & in.gencode.gene & gencode.gene.name & in.gencode.promoter & is.ingergenic & hybrid.gene.name & no.total.pubs & region & bp.tcga.overlap & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <chr> & <lgl> & <lgl> & <chr> & <int> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 22 &  43088127 &  43117304 & chr22 &  42692121 &  42721298 & - & Massie et al. 2017 & A4GALT & gene       & ⋯ & TRUE & CYB5R3          &  TRUE & FALSE & A4GALT & 2 & 22 : 42692121 - 42721298   &   0 & 1420 & hyper\\\\\n",
       "\t2 & 16 &   2325882 &   2390747 & chr16 &   2275881 &   2340746 & - & Massie et al. 2017 & ABCA3  & gene       & ⋯ & TRUE & ENSG00000243445 &  TRUE & FALSE & ABCA3  & 2 & 16 : 2275881 - 2340746     &   0 &    0 & .    \\\\\n",
       "\t3 & 2  & 220074494 & 220083716 & chr2  & 219209772 & 219218994 & - & Massie et al. 2017 & ABCB6  & gene       & ⋯ & TRUE & ABCB6           &  TRUE & FALSE & ABCB6  & 2 & 2 : 219209772 - 219218994  &   0 &    0 & .    \\\\\n",
       "\t4 & 21 &  43619799 &  43717354 & chr21 &  42199689 &  42297244 & + & Massie et al. 2017 & ABCG1  & gene       & ⋯ & TRUE & RNA5SP492       &  TRUE & FALSE & ABCG1  & 2 & 21 : 42199689 - 42297244   &   0 &  761 & hyper\\\\\n",
       "\t5 & 10 & 116190871 & 116527820 & chr10 & 114431112 & 114768061 & - & Massie et al. 2017 & ABLIM1 & gene       & ⋯ & TRUE & ENSG00000229668 &  TRUE & FALSE & ABLIM1 & 2 & 10 : 114431112 - 114768061 &   0 &  633 & hyper\\\\\n",
       "\t6 & 10 & 116264861 & 116264863 & chr10 & 114505102 & 114505104 & + & Ylitalo et al 2021 & ABLIM1 & EPIC\\_probe & ⋯ & TRUE & ABLIM1          & FALSE & FALSE & ABLIM1 & 2 & 10 : 114505102 - 114505104 &   0 &    0 & .    \\\\\n",
       "\t7 & 4  &   7967037 &   8160559 & chr4  &   7965310 &   8158832 & - & Massie et al. 2017 & ABLIM2 & gene       & ⋯ & TRUE & MIR95           &  TRUE & FALSE & ABLIM2 & 2 & 4 : 7965310 - 8158832      &   0 & 2143 & hyper\\\\\n",
       "\t8 & 4  &   8073800 &   8073802 & chr4  &   8072073 &   8072075 & - & Ylitalo et al 2021 & ABLIM2 & EPIC\\_probe & ⋯ & TRUE & ABLIM2          & FALSE & FALSE & ABLIM2 & 2 & 4 : 8072073 - 8072075      &   0 &    0 & .    \\\\\n",
       "\t9 & 17 &    906759 &   1133032 & chr17 &   1003519 &   1229738 & - & Massie et al. 2017 & ABR    & gene       & ⋯ & TRUE & MIR3183         &  TRUE & FALSE & ABR    & 2 & 17 : 1003519 - 1229738     &   0 & 4101 & hypo \\\\\n",
       "\t10 & 17 &   1058209 &   1058211 & chr17 &   1154916 &   1154918 & - & Ylitalo et al 2021 & ABR    & EPIC\\_probe & ⋯ & TRUE & ABR             & FALSE & FALSE & ABR    & 2 & 17 : 1154916 - 1154918     &   0 &    0 & .    \\\\\n",
       "\t11 & 16 &  67691415 &  67694713 & chr16 &  67657512 &  67660810 & - & Massie et al. 2017 & ACD    & gene       & ⋯ & TRUE & ACD             &  TRUE & FALSE & ACD    & 2 & 16 : 67657512 - 67660810   &   0 &    0 & .    \\\\\n",
       "\t12 & 1  &  55007930 &  55104865 & chr1  &  54542257 &  54639192 & + & Massie et al. 2017 & ACOT11 & gene       & ⋯ & TRUE & FAM151A         &  TRUE & FALSE & ACOT11 & 2 & 1 : 54542257 - 54639192    & 488 & 2554 & hyper\\\\\n",
       "\t13 & 1  &  55013708 &  55013710 & chr1  &  54548035 &  54548037 & + & Ylitalo et al 2021 & ACOT11 & EPIC\\_probe & ⋯ & TRUE & ACOT11          & FALSE & FALSE & ACOT11 & 2 & 1 : 54548035 - 54548037    &   0 &    2 & hyper\\\\\n",
       "\t14 & 1  &  55013418 &  55013420 & chr1  &  54547745 &  54547747 & + & Ylitalo et al 2021 & ACOT11 & EPIC\\_probe & ⋯ & TRUE & ACOT11          & FALSE & FALSE & ACOT11 & 2 & 1 : 54547745 - 54547747    &   0 &    2 & hyper\\\\\n",
       "\t15 & 1  &  55013734 &  55013736 & chr1  &  54548061 &  54548063 & + & Ylitalo et al 2021 & ACOT11 & EPIC\\_probe & ⋯ & TRUE & ACOT11          & FALSE & FALSE & ACOT11 & 2 & 1 : 54548061 - 54548063    &   0 &    2 & hyper\\\\\n",
       "\t16 & 2  & 111490116 & 111878187 & chr2  & 110732539 & 111120610 & + & Massie et al. 2017 & ACOXL  & gene       & ⋯ & TRUE & RPL5P9          &  TRUE & FALSE & ACOXL  & 2 & 2 : 110732539 - 111120610  &   0 & 1977 & hypo \\\\\n",
       "\t17 & 20 &  24986866 &  25039616 & chr20 &  25006230 &  25058980 & - & Massie et al. 2017 & ACSS1  & gene       & ⋯ & TRUE & ACSS1           &  TRUE & FALSE & ACSS1  & 2 & 20 : 25006230 - 25058980   &   0 &  958 & hyper\\\\\n",
       "\t18 & 20 &  25013228 &  25013230 & chr20 &  25032592 &  25032594 & - & Ylitalo et al 2021 & ACSS1  & EPIC\\_probe & ⋯ & TRUE & ACSS1           & FALSE & FALSE & ACSS1  & 2 & 20 : 25032592 - 25032594   &   0 &    2 & hyper\\\\\n",
       "\t19 & 20 &  25013522 &  25013524 & chr20 &  25032886 &  25032888 & - & Ylitalo et al 2021 & ACSS1  & EPIC\\_probe & ⋯ & TRUE & ACSS1           & FALSE & FALSE & ACSS1  & 2 & 20 : 25032886 - 25032888   &   0 &    2 & hyper\\\\\n",
       "\t20 & 12 &  81330193 &  81654989 & chr12 &  80936414 &  81261210 & + & Massie et al. 2017 & ACSS3  & gene       & ⋯ & TRUE & MIR4699         &  TRUE & FALSE & ACSS3  & 2 & 12 : 80936414 - 81261210   & 866 &  356 & hyper\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 23\n",
       "\n",
       "| <!--/--> | chr_hg19 &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | chr_hg38 &lt;chr&gt; | start_hg38 &lt;int&gt; | end_hg38 &lt;int&gt; | pub.strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | ⋯ ⋯ | in.gencode.gene &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | in.gencode.promoter &lt;lgl&gt; | is.ingergenic &lt;lgl&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | region &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 22 |  43088127 |  43117304 | chr22 |  42692121 |  42721298 | - | Massie et al. 2017 | A4GALT | gene       | ⋯ | TRUE | CYB5R3          |  TRUE | FALSE | A4GALT | 2 | 22 : 42692121 - 42721298   |   0 | 1420 | hyper |\n",
       "| 2 | 16 |   2325882 |   2390747 | chr16 |   2275881 |   2340746 | - | Massie et al. 2017 | ABCA3  | gene       | ⋯ | TRUE | ENSG00000243445 |  TRUE | FALSE | ABCA3  | 2 | 16 : 2275881 - 2340746     |   0 |    0 | .     |\n",
       "| 3 | 2  | 220074494 | 220083716 | chr2  | 219209772 | 219218994 | - | Massie et al. 2017 | ABCB6  | gene       | ⋯ | TRUE | ABCB6           |  TRUE | FALSE | ABCB6  | 2 | 2 : 219209772 - 219218994  |   0 |    0 | .     |\n",
       "| 4 | 21 |  43619799 |  43717354 | chr21 |  42199689 |  42297244 | + | Massie et al. 2017 | ABCG1  | gene       | ⋯ | TRUE | RNA5SP492       |  TRUE | FALSE | ABCG1  | 2 | 21 : 42199689 - 42297244   |   0 |  761 | hyper |\n",
       "| 5 | 10 | 116190871 | 116527820 | chr10 | 114431112 | 114768061 | - | Massie et al. 2017 | ABLIM1 | gene       | ⋯ | TRUE | ENSG00000229668 |  TRUE | FALSE | ABLIM1 | 2 | 10 : 114431112 - 114768061 |   0 |  633 | hyper |\n",
       "| 6 | 10 | 116264861 | 116264863 | chr10 | 114505102 | 114505104 | + | Ylitalo et al 2021 | ABLIM1 | EPIC_probe | ⋯ | TRUE | ABLIM1          | FALSE | FALSE | ABLIM1 | 2 | 10 : 114505102 - 114505104 |   0 |    0 | .     |\n",
       "| 7 | 4  |   7967037 |   8160559 | chr4  |   7965310 |   8158832 | - | Massie et al. 2017 | ABLIM2 | gene       | ⋯ | TRUE | MIR95           |  TRUE | FALSE | ABLIM2 | 2 | 4 : 7965310 - 8158832      |   0 | 2143 | hyper |\n",
       "| 8 | 4  |   8073800 |   8073802 | chr4  |   8072073 |   8072075 | - | Ylitalo et al 2021 | ABLIM2 | EPIC_probe | ⋯ | TRUE | ABLIM2          | FALSE | FALSE | ABLIM2 | 2 | 4 : 8072073 - 8072075      |   0 |    0 | .     |\n",
       "| 9 | 17 |    906759 |   1133032 | chr17 |   1003519 |   1229738 | - | Massie et al. 2017 | ABR    | gene       | ⋯ | TRUE | MIR3183         |  TRUE | FALSE | ABR    | 2 | 17 : 1003519 - 1229738     |   0 | 4101 | hypo  |\n",
       "| 10 | 17 |   1058209 |   1058211 | chr17 |   1154916 |   1154918 | - | Ylitalo et al 2021 | ABR    | EPIC_probe | ⋯ | TRUE | ABR             | FALSE | FALSE | ABR    | 2 | 17 : 1154916 - 1154918     |   0 |    0 | .     |\n",
       "| 11 | 16 |  67691415 |  67694713 | chr16 |  67657512 |  67660810 | - | Massie et al. 2017 | ACD    | gene       | ⋯ | TRUE | ACD             |  TRUE | FALSE | ACD    | 2 | 16 : 67657512 - 67660810   |   0 |    0 | .     |\n",
       "| 12 | 1  |  55007930 |  55104865 | chr1  |  54542257 |  54639192 | + | Massie et al. 2017 | ACOT11 | gene       | ⋯ | TRUE | FAM151A         |  TRUE | FALSE | ACOT11 | 2 | 1 : 54542257 - 54639192    | 488 | 2554 | hyper |\n",
       "| 13 | 1  |  55013708 |  55013710 | chr1  |  54548035 |  54548037 | + | Ylitalo et al 2021 | ACOT11 | EPIC_probe | ⋯ | TRUE | ACOT11          | FALSE | FALSE | ACOT11 | 2 | 1 : 54548035 - 54548037    |   0 |    2 | hyper |\n",
       "| 14 | 1  |  55013418 |  55013420 | chr1  |  54547745 |  54547747 | + | Ylitalo et al 2021 | ACOT11 | EPIC_probe | ⋯ | TRUE | ACOT11          | FALSE | FALSE | ACOT11 | 2 | 1 : 54547745 - 54547747    |   0 |    2 | hyper |\n",
       "| 15 | 1  |  55013734 |  55013736 | chr1  |  54548061 |  54548063 | + | Ylitalo et al 2021 | ACOT11 | EPIC_probe | ⋯ | TRUE | ACOT11          | FALSE | FALSE | ACOT11 | 2 | 1 : 54548061 - 54548063    |   0 |    2 | hyper |\n",
       "| 16 | 2  | 111490116 | 111878187 | chr2  | 110732539 | 111120610 | + | Massie et al. 2017 | ACOXL  | gene       | ⋯ | TRUE | RPL5P9          |  TRUE | FALSE | ACOXL  | 2 | 2 : 110732539 - 111120610  |   0 | 1977 | hypo  |\n",
       "| 17 | 20 |  24986866 |  25039616 | chr20 |  25006230 |  25058980 | - | Massie et al. 2017 | ACSS1  | gene       | ⋯ | TRUE | ACSS1           |  TRUE | FALSE | ACSS1  | 2 | 20 : 25006230 - 25058980   |   0 |  958 | hyper |\n",
       "| 18 | 20 |  25013228 |  25013230 | chr20 |  25032592 |  25032594 | - | Ylitalo et al 2021 | ACSS1  | EPIC_probe | ⋯ | TRUE | ACSS1           | FALSE | FALSE | ACSS1  | 2 | 20 : 25032592 - 25032594   |   0 |    2 | hyper |\n",
       "| 19 | 20 |  25013522 |  25013524 | chr20 |  25032886 |  25032888 | - | Ylitalo et al 2021 | ACSS1  | EPIC_probe | ⋯ | TRUE | ACSS1           | FALSE | FALSE | ACSS1  | 2 | 20 : 25032886 - 25032888   |   0 |    2 | hyper |\n",
       "| 20 | 12 |  81330193 |  81654989 | chr12 |  80936414 |  81261210 | + | Massie et al. 2017 | ACSS3  | gene       | ⋯ | TRUE | MIR4699         |  TRUE | FALSE | ACSS3  | 2 | 12 : 80936414 - 81261210   | 866 |  356 | hyper |\n",
       "\n"
      ],
      "text/plain": [
       "   chr_hg19 start_hg19 end_hg19  chr_hg38 start_hg38 end_hg38  pub.strand\n",
       "1  22        43088127   43117304 chr22     42692121   42721298 -         \n",
       "2  16         2325882    2390747 chr16      2275881    2340746 -         \n",
       "3  2        220074494  220083716 chr2     219209772  219218994 -         \n",
       "4  21        43619799   43717354 chr21     42199689   42297244 +         \n",
       "5  10       116190871  116527820 chr10    114431112  114768061 -         \n",
       "6  10       116264861  116264863 chr10    114505102  114505104 +         \n",
       "7  4          7967037    8160559 chr4       7965310    8158832 -         \n",
       "8  4          8073800    8073802 chr4       8072073    8072075 -         \n",
       "9  17          906759    1133032 chr17      1003519    1229738 -         \n",
       "10 17         1058209    1058211 chr17      1154916    1154918 -         \n",
       "11 16        67691415   67694713 chr16     67657512   67660810 -         \n",
       "12 1         55007930   55104865 chr1      54542257   54639192 +         \n",
       "13 1         55013708   55013710 chr1      54548035   54548037 +         \n",
       "14 1         55013418   55013420 chr1      54547745   54547747 +         \n",
       "15 1         55013734   55013736 chr1      54548061   54548063 +         \n",
       "16 2        111490116  111878187 chr2     110732539  111120610 +         \n",
       "17 20        24986866   25039616 chr20     25006230   25058980 -         \n",
       "18 20        25013228   25013230 chr20     25032592   25032594 -         \n",
       "19 20        25013522   25013524 chr20     25032886   25032888 -         \n",
       "20 12        81330193   81654989 chr12     80936414   81261210 +         \n",
       "   source             pub.gene.name pub.reg.type ⋯ in.gencode.gene\n",
       "1  Massie et al. 2017 A4GALT        gene         ⋯ TRUE           \n",
       "2  Massie et al. 2017 ABCA3         gene         ⋯ TRUE           \n",
       "3  Massie et al. 2017 ABCB6         gene         ⋯ TRUE           \n",
       "4  Massie et al. 2017 ABCG1         gene         ⋯ TRUE           \n",
       "5  Massie et al. 2017 ABLIM1        gene         ⋯ TRUE           \n",
       "6  Ylitalo et al 2021 ABLIM1        EPIC_probe   ⋯ TRUE           \n",
       "7  Massie et al. 2017 ABLIM2        gene         ⋯ TRUE           \n",
       "8  Ylitalo et al 2021 ABLIM2        EPIC_probe   ⋯ TRUE           \n",
       "9  Massie et al. 2017 ABR           gene         ⋯ TRUE           \n",
       "10 Ylitalo et al 2021 ABR           EPIC_probe   ⋯ TRUE           \n",
       "11 Massie et al. 2017 ACD           gene         ⋯ TRUE           \n",
       "12 Massie et al. 2017 ACOT11        gene         ⋯ TRUE           \n",
       "13 Ylitalo et al 2021 ACOT11        EPIC_probe   ⋯ TRUE           \n",
       "14 Ylitalo et al 2021 ACOT11        EPIC_probe   ⋯ TRUE           \n",
       "15 Ylitalo et al 2021 ACOT11        EPIC_probe   ⋯ TRUE           \n",
       "16 Massie et al. 2017 ACOXL         gene         ⋯ TRUE           \n",
       "17 Massie et al. 2017 ACSS1         gene         ⋯ TRUE           \n",
       "18 Ylitalo et al 2021 ACSS1         EPIC_probe   ⋯ TRUE           \n",
       "19 Ylitalo et al 2021 ACSS1         EPIC_probe   ⋯ TRUE           \n",
       "20 Massie et al. 2017 ACSS3         gene         ⋯ TRUE           \n",
       "   gencode.gene.name in.gencode.promoter is.ingergenic hybrid.gene.name\n",
       "1  CYB5R3             TRUE               FALSE         A4GALT          \n",
       "2  ENSG00000243445    TRUE               FALSE         ABCA3           \n",
       "3  ABCB6              TRUE               FALSE         ABCB6           \n",
       "4  RNA5SP492          TRUE               FALSE         ABCG1           \n",
       "5  ENSG00000229668    TRUE               FALSE         ABLIM1          \n",
       "6  ABLIM1            FALSE               FALSE         ABLIM1          \n",
       "7  MIR95              TRUE               FALSE         ABLIM2          \n",
       "8  ABLIM2            FALSE               FALSE         ABLIM2          \n",
       "9  MIR3183            TRUE               FALSE         ABR             \n",
       "10 ABR               FALSE               FALSE         ABR             \n",
       "11 ACD                TRUE               FALSE         ACD             \n",
       "12 FAM151A            TRUE               FALSE         ACOT11          \n",
       "13 ACOT11            FALSE               FALSE         ACOT11          \n",
       "14 ACOT11            FALSE               FALSE         ACOT11          \n",
       "15 ACOT11            FALSE               FALSE         ACOT11          \n",
       "16 RPL5P9             TRUE               FALSE         ACOXL           \n",
       "17 ACSS1              TRUE               FALSE         ACSS1           \n",
       "18 ACSS1             FALSE               FALSE         ACSS1           \n",
       "19 ACSS1             FALSE               FALSE         ACSS1           \n",
       "20 MIR4699            TRUE               FALSE         ACSS3           \n",
       "   no.total.pubs region                     bp.tcga.overlap bp.ecm.overlap\n",
       "1  2             22 : 42692121 - 42721298     0             1420          \n",
       "2  2             16 : 2275881 - 2340746       0                0          \n",
       "3  2             2 : 219209772 - 219218994    0                0          \n",
       "4  2             21 : 42199689 - 42297244     0              761          \n",
       "5  2             10 : 114431112 - 114768061   0              633          \n",
       "6  2             10 : 114505102 - 114505104   0                0          \n",
       "7  2             4 : 7965310 - 8158832        0             2143          \n",
       "8  2             4 : 8072073 - 8072075        0                0          \n",
       "9  2             17 : 1003519 - 1229738       0             4101          \n",
       "10 2             17 : 1154916 - 1154918       0                0          \n",
       "11 2             16 : 67657512 - 67660810     0                0          \n",
       "12 2             1 : 54542257 - 54639192    488             2554          \n",
       "13 2             1 : 54548035 - 54548037      0                2          \n",
       "14 2             1 : 54547745 - 54547747      0                2          \n",
       "15 2             1 : 54548061 - 54548063      0                2          \n",
       "16 2             2 : 110732539 - 111120610    0             1977          \n",
       "17 2             20 : 25006230 - 25058980     0              958          \n",
       "18 2             20 : 25032592 - 25032594     0                2          \n",
       "19 2             20 : 25032886 - 25032888     0                2          \n",
       "20 2             12 : 80936414 - 81261210   866              356          \n",
       "   ecm.methyl.status\n",
       "1  hyper            \n",
       "2  .                \n",
       "3  .                \n",
       "4  hyper            \n",
       "5  hyper            \n",
       "6  .                \n",
       "7  hyper            \n",
       "8  .                \n",
       "9  hypo             \n",
       "10 .                \n",
       "11 .                \n",
       "12 hyper            \n",
       "13 hyper            \n",
       "14 hyper            \n",
       "15 hyper            \n",
       "16 hypo             \n",
       "17 hyper            \n",
       "18 hyper            \n",
       "19 hyper            \n",
       "20 hyper            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df_all[df_all$no.total.pubs == 2,] %>% arrange(pub.gene.name), 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "a690381e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "     1      2      3      4      5      6      7      8     15 \n",
       "  5429 113722    789    284     88     34     21     44      6 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "&lt;NA&gt;"
      ],
      "text/latex": [
       "<NA>"
      ],
      "text/markdown": [
       "&lt;NA&gt;"
      ],
      "text/plain": [
       "[1] NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts = table(df_all$no.total.pubs)\n",
    "counts\n",
    "sum(counts[c(3,4,5,6,7,8,15)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c448c13d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "d5a748b1",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in seq.int(x[i], x[i + 1], length.out = n): 'length.out' must be a non-negative number\n",
     "output_type": "error",
     "traceback": [
      "Error in seq.int(x[i], x[i + 1], length.out = n): 'length.out' must be a non-negative number\nTraceback:\n",
      "1. pie(c(intergenic, in_genes), labels = paste(c(\"intergenic\", \"genic\"), \n .     c(intergenic_pct, genic_pct), c(\"%\", \"%\")), main = \"Region types\", \n .     cex = 2)",
      "2. t2xy(seq.int(x[i], x[i + 1], length.out = n))"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAYAAAD958/bAAAEDmlDQ1BrQ0dDb2xvclNwYWNl\nR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRB\nkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4\na73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PC\nv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UA\nVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXd\na8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8\nHOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojL\njVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0\nyDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5Pt\nXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEw\nQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXH\nliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vW\nc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUt\nVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJf\ncl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdd\nuwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqv\ngcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCg\nKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8A\nrD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvF\nY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAANIoAMA\nBAAAAAEAAANIAAAAAN/ryxkAADAuSURBVHgB7dABDQAAAMKg909tDwcRKAwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwY\nMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAwM/AABbNAAHmfQCNAAAA\nAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "intergenic = nrow(df_all[df_all$type==\"intergenic\",])\n",
    "in_genes = nrow(df_all[df_all$type!=\"intergenic\",])\n",
    "intergenic_pct = round((intergenic/nrow(df_all))*100)\n",
    "genic_pct = round((in_genes/nrow(df_all))*100)\n",
    "pie(c(intergenic, in_genes), labels = paste(c(\"intergenic\",\"genic\"),c(intergenic_pct,genic_pct),c(\"%\",\"%\")), \n",
    "    main=\"Region types\", cex=2)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 828,
   "id": "5359f2aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 20 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene2</th><th scope=col>bp.tcga.overlap</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1-Dec      </td><td>0</td></tr>\n",
       "\t<tr><td>5_8S_rRNA  </td><td>0</td></tr>\n",
       "\t<tr><td>5S_rRNA    </td><td>0</td></tr>\n",
       "\t<tr><td>6-Sep      </td><td>0</td></tr>\n",
       "\t<tr><td>9-Sep      </td><td>0</td></tr>\n",
       "\t<tr><td>A1BG       </td><td>0</td></tr>\n",
       "\t<tr><td>A1BG-AS1   </td><td>0</td></tr>\n",
       "\t<tr><td>A1CF       </td><td>0</td></tr>\n",
       "\t<tr><td>A2M        </td><td>0</td></tr>\n",
       "\t<tr><td>A2MP1      </td><td>0</td></tr>\n",
       "\t<tr><td>A3GALT2    </td><td>0</td></tr>\n",
       "\t<tr><td>A4GALT     </td><td>0</td></tr>\n",
       "\t<tr><td>A4GNT      </td><td>0</td></tr>\n",
       "\t<tr><td>AACS       </td><td>0</td></tr>\n",
       "\t<tr><td>AACSP1     </td><td>0</td></tr>\n",
       "\t<tr><td>AADAC      </td><td>0</td></tr>\n",
       "\t<tr><td>AADACL2-AS1</td><td>0</td></tr>\n",
       "\t<tr><td>AADACL4    </td><td>0</td></tr>\n",
       "\t<tr><td>AADAT      </td><td>0</td></tr>\n",
       "\t<tr><td>AAK1       </td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 20 × 2\n",
       "\\begin{tabular}{ll}\n",
       " gene2 & bp.tcga.overlap\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 1-Dec       & 0\\\\\n",
       "\t 5\\_8S\\_rRNA   & 0\\\\\n",
       "\t 5S\\_rRNA     & 0\\\\\n",
       "\t 6-Sep       & 0\\\\\n",
       "\t 9-Sep       & 0\\\\\n",
       "\t A1BG        & 0\\\\\n",
       "\t A1BG-AS1    & 0\\\\\n",
       "\t A1CF        & 0\\\\\n",
       "\t A2M         & 0\\\\\n",
       "\t A2MP1       & 0\\\\\n",
       "\t A3GALT2     & 0\\\\\n",
       "\t A4GALT      & 0\\\\\n",
       "\t A4GNT       & 0\\\\\n",
       "\t AACS        & 0\\\\\n",
       "\t AACSP1      & 0\\\\\n",
       "\t AADAC       & 0\\\\\n",
       "\t AADACL2-AS1 & 0\\\\\n",
       "\t AADACL4     & 0\\\\\n",
       "\t AADAT       & 0\\\\\n",
       "\t AAK1        & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 20 × 2\n",
       "\n",
       "| gene2 &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; |\n",
       "|---|---|\n",
       "| 1-Dec       | 0 |\n",
       "| 5_8S_rRNA   | 0 |\n",
       "| 5S_rRNA     | 0 |\n",
       "| 6-Sep       | 0 |\n",
       "| 9-Sep       | 0 |\n",
       "| A1BG        | 0 |\n",
       "| A1BG-AS1    | 0 |\n",
       "| A1CF        | 0 |\n",
       "| A2M         | 0 |\n",
       "| A2MP1       | 0 |\n",
       "| A3GALT2     | 0 |\n",
       "| A4GALT      | 0 |\n",
       "| A4GNT       | 0 |\n",
       "| AACS        | 0 |\n",
       "| AACSP1      | 0 |\n",
       "| AADAC       | 0 |\n",
       "| AADACL2-AS1 | 0 |\n",
       "| AADACL4     | 0 |\n",
       "| AADAT       | 0 |\n",
       "| AAK1        | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene2       bp.tcga.overlap\n",
       "1  1-Dec       0              \n",
       "2  5_8S_rRNA   0              \n",
       "3  5S_rRNA     0              \n",
       "4  6-Sep       0              \n",
       "5  9-Sep       0              \n",
       "6  A1BG        0              \n",
       "7  A1BG-AS1    0              \n",
       "8  A1CF        0              \n",
       "9  A2M         0              \n",
       "10 A2MP1       0              \n",
       "11 A3GALT2     0              \n",
       "12 A4GALT      0              \n",
       "13 A4GNT       0              \n",
       "14 AACS        0              \n",
       "15 AACSP1      0              \n",
       "16 AADAC       0              \n",
       "17 AADACL2-AS1 0              \n",
       "18 AADACL4     0              \n",
       "19 AADAT       0              \n",
       "20 AAK1        0              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_tcga_ol = df_all %>% filter(Gene!=\"intergenic\") %>% group_by(gene2) %>% \n",
    "summarize(bp.tcga.overlap=sum(bp.tcga.overlap))\n",
    "head(df_tcga_ol,20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 829,
   "id": "271f0c24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAYAAAD958/bAAAEDmlDQ1BrQ0dDb2xvclNwYWNl\nR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRB\nkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4\na73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PC\nv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UA\nVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXd\na8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8\nHOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojL\njVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0\nyDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5Pt\nXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEw\nQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXH\nliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vW\nc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUt\nVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJf\ncl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdd\nuwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqv\ngcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCg\nKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8A\nrD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvF\nY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAANIoAMA\nBAAAAAEAAANIAAAAAN/ryxkAAEAASURBVHgB7N0HnBT1/f/x2X53HOWAo/feEVAEpGMBsYCC\n2Bs27D1V/VsSoz81Ro0m8Scak5hEY/xp1Bg19m7sLfaGqERFqVd3/+/PsWOGdQ+ubN/X9/H4\nMLOzM9/ynF1uPjuzs45DQQABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBIJuBLtpBlCCCAQBMF+mi98YpuirDiY8VHilcUNQpKegQGqto2\n8ao/19SiKaWtVurvWfFlz3xLZ0drw+b+Lfla29jrJFkJauEQxVhFheKNeHymaXNKql6b7dVo\nX0/DKzT/ledxc2bLtfKARjao1fL1Chvn1t47g7ROmcJbPtGDL70L4vOdNO2ZsHyDHr/jWWZ9\nsr4lK1EttPXXKr5QxBTFWhL3XyreP8VqybgRQAABBBAoOIEpGtFjCjt4sgOmxLCDyFMVpQpK\n6gWeUJWu+flJqg9p2SkKO5D2lnl64G5n01QUO7D31tmU+ZuTNGwH/Fcrqhqp7+9a3lhyoae+\nLal+bV6hmr1j+u23LTV/Zk5CXd563XlLju5TjNhC9f9KUo/1M1kxU7dud/pMwor/TLKOu653\nulrr3aKwpKsYy84atNcjUIwIjBkBBBBAAAEEvitwthbVKbwHCo3N24GenVmipFZgSwnSXDVl\nZ11sn4xJaDZXE6SJ6udb8T439lqy5RsV+ykaK6l+bZaoITtb5O2TnU2xM1stKU1JkNy2LFH6\nmSLZ2blkCdLrjXQomWtLEyS3b++qrZGNtFfIi0mQCnnvMjYEPAJBzzyzCCCAwNYElmmFcxNW\nsoO1lxR2JsEOwPsq3LKjZuwT7CPcBUxTIrBStbwXr8k+1XdLH83YmZZMlvvVmD+hwRl6HIkv\n+1hTS9i8xXtpUkc9YX22qVte0MzTCktE5ivs0iYrlrBcp7DtX1N4Szpem3upgcRkqFTLDlJc\n4W28hfM2xjUK+1tsY+un6K6wYmcBv6d4W2Fj3loZrhV6KOy14ZbemhnsPmji1C7Tey6+riVn\ntm/Dim0VNnYrdibvEsU8e0BBAAEEEEAAAQSKVcAOwLyXP32qx3bwmlh20IKNCvfTZvskvEvi\nSjxOi4Alp667Te37Qd5iB7Te573PpXL+I087V22lYnve7ZO9VpYlrG+Jw88969i6v01YJ12v\nzQc87dZ75l9NaL+pDxPPIE1I2NASkn0U9n0i1+RzzXdQeIv3DJL3vXawdyXNH6pw69ngmd/S\nGaR/aL1kxRK3pxXe+mzfFFPhDFIx7W3GWtQCnEEq6t3P4BFoloAduLpnBWzDfRUP20xCeVyP\nf6S4VGFf7LazS0MVqxSJZYgWWJI1ThFV2LoPKl5UJJYDtKBrfOEtmq5QzFNMVdin5HZG4U6F\nHTw2VprTntXRVnGIwvpvn8ZbgviZ4iGFtWWXGjalHKSVKuMr/l3TNzwb9dX83vHHdvD5C4VZ\nuGVXzQyLPzAXO2g3eztbYOUpxROKuQpLTr3FDpgtkbVtkpnaujauBYpJiq8VdvD8R4UlK+ku\nw9XAMZ5GrtX8NZ7HNmvmP1Tsqeiv+EIRUlgyYV5W0vHaHKh6Z1rl8XKBpmfH50dqatb2Wk9l\nsfHcrKhW/F+8Yvtw4TTFWfHHiZMntWBWfOGOmt7oWcEeu8XWm+0+aMHUXke/UUyMb1uq6XaK\nR+OPmzLxa6WdFPa+7aOws592NtD+H0l8fXpf42/peXu/JZaxWmBJp5VaxZUNc//9pznvd29d\n9v60PtnraluF2d2haEqx95O93+31Y/tuo+IzhZ1p/ZvCfc1qtuF7Zva+tfKu4nbFCIUts/ej\nLbP37n0KCgIIIIAAAgjkmIAdkNqBqf1xt7A/9FsqdiA0QGEHsY2V4/SEHTy4dbrTei07TxFQ\neIv3U3NLqm5VuNu4U0tY7GAyWWlue5aYfKlw606cPqvnzKUp5Vdayd0+8SDue57nbJ3xCRXa\nQZK77f7x5ywhcpedH192o2eZ+5w7PSG+jh2YustsurMi2RjtgNAO7lpaPtKGbjtXbaES2yfu\nelHN997CunbgaJH4mkrHa9O68ROF27f3NG+vR++4zLu5ZY42cOu06YQtVOB9vd+VsJ73uXP0\nnNlZfSsT1rPHttzeU7aezVs8o/CWf+qB+1xjZ5Bs/ZM969n6I21hE0s7rfeUwm3HO7X+/Vxh\nSZdbztKMu84qzSd7r93mWcf+P/CW5r7fj9XGbns3ad4SE/exTe09ZO8X77LE/6MsobL/g7zr\neOf/oOfCCrccrBn3eUvADlUk294S0xIFBQEEEEAAAQRySGCw+uL+Ibfp6a3s25EJ9dXosTcB\nszYuSWjDe1BoB0y2jiVYH8fn3f7ZJ8l9Fd7S3Pa6aWM7m+LW+Ynm7SD1eYUdzLnLf6r5ppRd\ntJK7zTsJG9inw+5zNvUmeMM8z9lZBTvItPKEwt3m/IYlm84cuMsSp40lSO7B2PuqIzFZvSxe\nb0sm3kRiSwnS5arc7evnLWlI26T6tWndsANf2+du386zhSq2v91l5lVhC5tR5mhdd3ubbilB\nMjd33TcT2vC+FywRsDOv7roj4uva1F1mr1t7DbiPn4mv4062liCZx3SFd79+psfeg323rmTT\nci30vmatH973kdsvS0rcBLiH5u297D63u+a9pbMe2P8b7vP2HnNLc9/vtt2xCreuDZ55W2bv\nE/vAYOeE5ebilsmaiSrcOqo0/4riK88ye84SP7d4E6QvtdDdfrVn3q3vUncjpggggAACCCCQ\nGwKz1A33D7VNFybp1nAtm9hIDPGs317zqxRufb/VfE9FRHGEZ7kdpHRXuMV7UGjbXqxoE3/y\ncE3d+mxqB0huaUl7+2tjtz5LYLzFTXZWaKF96usmLd51EudDWuBNuAbHV7AxJx6M3enZ2JIl\ntx/eswjeg83z4+v31nSuZ33bbm/FaEUnhZV5Crc+m36qGKqwUqp4TOE+v9IWtrB4D6S3lCDZ\nmNz2nmthW6l8bbpd2M3TL+uf+/o1K7e/Nj1J0ZzSnATpbFXstmXJsfdg3PtesATpF551T9S8\nFZu621+m+RM8j7eUIFlb9v60+I/iG4U3UXHrPEDLm1rsNe1ut1bzhyosaapUXKhwEwNb5xiF\nW27TjLvdX9yF8al3fO9rmT++vCXvd9v0WIXblk1t7LbsUMX5Cis7K7zrePeJ7QN3HJZw2vvJ\nir33n1a4291jC+PlYE3d5Tb9WDEp/pwlgLau+7ztlwHx55gggAACCCCAQA4I7KU+uH+obTon\nSZ8eSVjHu/7dnvWXetazT5G7eZ6z2WcV7rbneZ7zHhQ+r+XuAZG7ykee7X7oLtS0Je0d7qnL\nDujsYHWMwv102w7Cmlv+oA3ccdnBnZXEg3t73g5IA/akih1oudvYONySLEGy5/oq3PVtasmR\ntyQmSPt4n9T8fgrv9onGCas3+tC7L7aUIL3iac/G2pKSytem2/7/acZ1eMpdGJ/aY/e51xKe\n29rDOZ5trY4JW9jgzIR1u3vW9b4XLEFa4FnXknYrtyvcfu6u+aYmSO42jU0/U10LFU0tllx6\n6/ImQG4dN3nWWeku1HQXz/IqzVd4nnvO89yPPctb8n63zRMTJHucWHbWAu9Y3Pepu15bzUxR\n9HAXxKfWP3e7ZzzPJSZIe3ues1lLktYp3G3PsIUUBBBIv0BL//ilv2e0gAACuSRgB0Xe0tf7\noJnzdsDklg80M16xqyfe0rxbBrkzCdMn9dg+rfWWTz0POnjmW9Lew9reLquxYp90n6t4SbFK\ncZNivqKdojnFPg13iyUqVuyA2S33x2esXjtwtum0+DJLJO2AN9XFEhRv+dL7QPORhMepfviJ\np0JvAuBZvNXZVL42rTFL2G3/uuV37kx8+lvP4xGan+p5nMrZMk9l9lr8wvM4cfYRLbCDaCsz\nFSXxqSYN75NHbaaJxQ7I31SsTVi/Wo8tGbWDf+9rOWG17zyc6Vlir+Plnsfu7LXujKb2OrAz\nS1bs7O37DXObXov7xudHaWr/b1hJrLMl7/dNNW3+782bP2zSIzN7Ir7mYk0vUdjZo/Piy2zS\n2HvK9t/fPOvZrO3zxz3LGvv/0LMKswggkAqBYCoqoQ4EECh4gY8TRjgg4bE9tIMw70HcTD32\nfuKrhw1lsDujqdVzl+dx4mzPxAXxx3a5WmKxT5iTlZa0964qOkJxnSLgqdQ+0d0vHjZWu8FC\nsgM+Lf5OuUdLrI/uwWup5neMr/WNphd6HtuZpV6KkMJKou2mpa3/98OEKmoSHvsSHqf6oR2I\n7xKvNPFT98S2usQXWJLqLal8bVq9hyi8fxvtDM1h9kS82H7zlqP04DHvghTNd/PU84Hmaz2P\nE2e/0oKXFWMVdhbD+uwm8C9oPtn7RYuTFjvAd/fJTM3fpLCkxQ7sf6VYrXhI0dRir2O3fKKZ\nxNeYPfeeu0J8aonnwwr7EMSSp58qrBysuEZxqMIt9v/HSveBpi15v3s2b5jdoH+9/5clPt/Y\n4930xAUK2w+NFRtTsmLtJbNZ4Vm5v2eeWQQQSKOA949AGpuhagQQyHMBOwh9XWEHLlYsSThX\n4T1o+5E94Sn2yelEz2N31nsQYAe7dmamsZJ48OuuV+3OeKaNHXi0tL3fqu6HFMco7BKmYQpv\nsWTpfxX2CfeD3icambdP5u9T7K6wJMnq3FZhxQ4GH1WsV7RRzFYMUbjlr+5Miqd2IOgt9il2\nJoslSG5prxk7K/C8uyBheqYen6J4XGEeVyrs7EEqX5uqruGSTJu6Zbg708jUzhScpLDEIZVl\nsqeytz3zjc3aa9A9MP+BZ6WHPPPNnbVt91SYeUjRRWEJySTFK4qmFG9yZslbstI+YaE3OVmu\n5+z/Gmvf2u2n2Ffhlt+4M/FpS9/v3moS3xfe5xqbP0RPeD9QMZ97FQ8pRinsAxArjf0/ZR8m\n+ZM8X24bxUuqX2NuvUwRQCBBgAQpAYSHCCDQqMCf9Mx58WftzI99Sn15/HGyiR3QJCvveRba\npUNzFY0dNHhW3Wy2OQfyrWnvQ7VqiZ8dcPZQzFLYZUZ20BhQ+OKP7eC0KeU2rWQJkpXzFVaH\nlfsVtYqHFbsqdlCsVVixsdp2LSl2wJXLxQ4g7YA2HO/kDzVdFJ/3TiwZPUhh45mmsLMZ3tde\nql6bM1TvYEVziiW7Byt+0ZyNtrKuvb7GeNa5xzPf2OxDeuLk+JOdPCs19bXp2WSz2Wf16P8p\nfhJfWqbpTYoJCtt3WysfeFawJGCg4l3PMpt1PyiweavzLZuJl881vV3hvi4u0nzP+HMfa5po\n05r3e7zaht+hcuebOrXXrvt+/qXmj/dsONozX++Z987a8dhIhSVW3uJ9Pb7jfYJ5BBBIn0Cu\n//FM38ipGQEEmitwpTb4xLPRZZq3T0Xdg1v3qYmaeUAxzl2gqSUSbnncndG0h8IOSt1iBxhP\nKJ5SXKewZKG1pSXtna9GbTv7JNtNTlZq/g+KvRX/VLgl5M40Yfo3reMeIA30rO/WZ4mSFTuL\n1K1hbtNv1qyIz29tYgmntyTuG+9zuTBvB3z2OnKL2V6l8Pa7lx7fqrCzF26xdbwlVa/NpZ5K\nv9J8b4W9RpPFB1rulqPcmVZO7bVkydbVnnpeSXjseWqz2Uf0KLrZkk2vtUcTlrXkoSUl9p50\ni50RsQ8OmlL+oZXszKhbLtWM9z1Tqcc/dp/U9H6FfVjgLb/2PNjHM79c8+77yV38uDujaUv/\nf6nz1NGU2c5aaYhnxTs98zY73fN4Sx9Mn631vP9XztRj7/+jL3jqYRYBBBBAAAEEckRgrvph\nBw92VsONKs3bp8wPKFZ7lrvP2/R6hVvsAMC+L+E+/6HmD1LMUdzoWW7Pew8s/uV57izNJ5YH\ntcCt82LPky1p70xPXVanHdRZXyYqzlBUK9y29tN8c4q3n1bHSs/GduDp1utOrS+J5QktcJ8/\n3/NkO89ye94ONi9RLFJYmadwt7NpYpmhBd7n7WxBS8pH2sitJzGZSazPksGPPevbdmsVdsD/\nmGKDwq3LprYsrEgsrX1tdlCF3rauTmwg4fE5euzt19SE55M9tNe4d5tVevxJPGzeEgPv81E9\nTlav971wnNZxy/Oa8W7/jPuEpid4nvMut1X+6XnOEppkZZgWen8ry94Dw5OtmGTZKVrm7ddz\nevxjhX3AYsm/+9w6zfdTJBZ7D7+tcNezqSVGvRWJpSXvd6vjWIVb//uJlcYf7+xZx9YNxJfb\n1PvaeVKPJ8XjOk3dem36lsItB2vG+5zN36s4SnGewt4H7vMvat6voCCAAAIIIIBADgrMVp/s\nYM79w72lqX1yfIEiqPCWbfTgI8WWtv25dwPNew8Kz0p4zh4+qHDruzjh+ea2V6Lt7dIdt77G\npv+ndeyArDnlRK3sre93CRt/mvD8oITn7WFjCZI9ZwdS3vpt/jJ7QiUXEyTrV1/Fw4rEfic+\nfl3rdFQ0Vlrz2vQeIFu7kxtrJL68j6Z2kO72MXE/Jtt8jmd9d7vGpv/Rukckq0TLvO8Fb4Jk\n+9lbn/d90NoEybpyWkL9j+hxU1//x2tdS/i8/fPOW3K0RNFYOVNPeNe/s7EVtby573eryrv/\n32+k7sYSJFs90d7bV+//dfaBUnm8fm+CZP9XvqPwbufOm830+DZMEEAAAQQQQCBHBbqqX/YJ\n8HsK94+4O7WDoGcVlsTY5TONFXvOLl/7WuFua9MPFHawkli8B4XNTZCsrua2F9E231esUHj7\nZ/OrFKcqLJFqbumjDbz1HZJQgR1ou8/bmbZkZUsJkp1xWKlw67CzEpfHK8nVBMm651ecoXhV\nUadw+29TG4+dhShVbK209LVpZzXcNu1sRVOKfdrvbmNnV7aUvFl9W0qQbHs7u3C/4gJFe0Vj\nxfte8CZIe2gDtz82tf3tllQkSLaP7BI2bxtL3QaaMN1H69yt8L7nLWGwDyMGKLZU7P1rZ63c\ntvfc0sp6rrnv99YmSPb/hSVJ3rOAG/TYPuix1+0HCrfvB2reysEKd5klxB0Utyu8ifcLejxS\nQUEAgQwKNPWTnwx2iaYQQCDPBOxApLfCDg7tj7x9Wrpa0ZzSXyv3UlgyYtvbAUI6S3Pas4PC\n7gobo5UPFJ8r7MAmV0tQHRuqsATuNYUdhOZTsX6PUtg43lXY66olJRWvzZa0yzZbFrD31GiF\nJcJvxqeapK005/3e2k60UwVDFJbMvaGwMTZWLEH6bfzJLzS116sVq8OSIkvUbTkFAQQQQAAB\nBBBAAAEEECh4gcQzSAU/YAaIQL4I2Kc4FAQQQAABBBBAAAEEEEAAAQmQIPEyQAABBBBAAAEE\nEEAAAQTiAnZ9NwUBBBBAAAEEEEAgswKfqrn7402uyWzTtIYAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIINAKAV8rtmVTBBBAAAEE0iUQUsWVSaK9lrVtCL+/XcAf7OD3\n+zv4Ar62TixW5sScUNRxgo4TCzrRaDAWiymcgB77HccX9fkcPd0wrXccf73jd+r9jq/e8Tkb\nnJhvfcyJrY3WR9fEonVrotHoWrWzXrFa8aXiiyTTKi2jIIAAAggUkAAJUgHtTIaCAAII5ImA\nJTh9FL090TcUKRmoP0q96+rqOkXr69q4Y/H5/dHS8ra1bTtURMvadXA07y8rLw9qGigpK3dK\n2rRRlDvhklInEAw6gUBA09CmeZvqsU8Rq69XzmQRdaKar1fYMpvWVG10qjesd6o3arpxg1O1\nUfMbNtRvXLeuft3XX0bXfr06tv6br/1V69eFlHQp2dpU/IHAhmAw9LkSq49qq6vf09JPFCvi\n8bGm7yosyaIggAACCOSJAAlSnuwouokAAgjkmUCZ+jvYE0NDJSUjlZgMqq+t7WBj8QeC9e07\nda7p2K2nr7J3n3Cnbj38Cqddp85O24pOio4N07K27RyfTv3kQlFy5GxYu8ZZ9/VqxVfOmi+/\ncL5a9Zmz+vNPbRr9YsXHNV99vjK25sv/hOpqa3UmS6ezQqEvlaC9V1tV9boevq14Jz59Q9ON\nCgoCCCCAQA4J5MZfnBwCoSsIIIAAAs0S0OVrziDFGItAMDxOl7uNq6uu7q7HvkhpWU2X3n3r\nuw8cEuned4C/a59+Ttc+/Z1O3Xs0JD+5kviorykva1d/5az6+APn848/bJh+9tEH0c/ee6d6\n1YoPAzoTFVaD0VAk8omSxpfq6+pe0ONX4/GWpnUKCgIIIIBAFgRIkLKATpMIIIBAngpYMjRS\nMdEiHCmZUltbMzgWjYYtEeo5aGi03/BRJb2GDHd6DRzqdO3bv+EsUJ6ONa3dXr/mG+eTd950\nVig+efct58M3Xq3S1K/L/MJKGuuC4ci/a6urHlUn/qV4VmFnn/S9KQoCCCCAQLoFSJDSLUz9\nCCCAQP4K9FLXJyu2D4VLptXX1Y7Rd3hK2nbsXDVo7PjggFHbBHsPHub0UnTUpXGU1gusXvW5\nEqd/O++//rLz3isv1r778vNRXc4X0Y0oqgPh8Cu6TO8xtfKEwpKnz1rfIjUggAACCCQKkCAl\nivAYAQQQKF6B/hr6DB2Mz/SHQjvqMrme4dLS2v4jxkYHbTMhooTIGTBqrNO+c5fiFcrCyL/5\nYpXz/msvK15y3nnp+RolTT7dVCKky/M+rq2t/afuOvGIumUJk323iYIAAggg0EoBEqRWArI5\nAgggkMcCdoZoF78/MMcf8M/RTQW66IYI1cMnTgkM225KcOiE7Z2eA4fkzA0S8tg5pV23u/Ct\nePsN563nn3XefO6pujeefbJ+w5pvIgHdDEKXO96r7zTdowbvU3ya0oapDAEEECgSARKkItnR\nDBMBBBCQQIliumIXnX3YQ7elHtSmXfvqUVNmBIduOylgCVH3fgOBykOBzz/6QAnT085rTz5a\n98oTD0c3rlsb1j5+W/v4Tg3nXoWdZdqQh0OjywgggEDGBUiQMk5OgwgggEBGBey3hhboVtN7\n6vd+pukGAIGBo8fVjZ0+JzJqynSn95ARnCHK6O5If2N2K/KP/v2a89pTjzovP/pA9TsvP28/\nmBvV70E9orOEt6oHdyjs95ooCCCAAAJJBEiQkqCwCAEEEMhzgdHq/4JQOLKktqZ6ZLtOldUT\n5swNj54ywzdsu8lOSdm3v8Ga58Ok+00RsB+/1aV4zkuP/DP67H131dpNH3R26VWdXbpZ21uy\n9FJT6mEdBBBAoFgESJCKZU8zTgQQKHQBu9vc3ro99JK6mupe3foOqJq4y24l42bt7PQdNqrQ\nx874mihgZ5d0S3HnhYfudf51391Vn37wboleM5/qNXOTqvizwm4pTkEAAQSKWoAEqah3P4NH\nAIE8F7AfZ90vGA4fXFdT033A6G1qtttpfmTcrF2cLr365PnQ6H4mBL5YuULJ0n3Ok3fdVv3B\n6y+HlSx9omTpd2rbkiXOLGViJ9AGAgjknAAJUs7tEjqEAAIIbFHA7qKwny6fO1SXzw3UbxBV\n77D73pGJu+zuVHTptsUNeRKBLQl8+eknzrP33uU8cddfq1e8/e+IXmPv6zX2W21jCdN7W9qW\n5xBAAIFCEiBBKqS9yVgQQKBQBco1sH10pugYnSnarrJXn6oddl9Usv3cPZyuffoV6pgZVxYF\nVn38ofPMP/7mPP63W6s+/+j9SDAUfrqutubX6tJfFOuy2DWaRgABBNIuQIKUdmIaQAABBFos\nMFU/2nqE4/iWhEtL/EqKQtMW7OPrM3RkiytkQwSaK2A/Uvv4HbdEH7/zr3W1VVVRbX9zNFq/\nXFO7dXisufWxPgIIIJDrAiRIub6H6B8CCBSbQKUGfHhIZ4tqa2r66kdba2fsvX94vG62oE/x\ni82C8eaQgC63c158+H7n4Vv/WPPGM48Hdev4T/QavUJdvF7xZQ51la4ggAACrRIgQWoVHxsj\ngAACKRPY3h8InKC7jO3TvnOX6KxFB0T03SKnY7ceKWuAihBIlcDqVZ87j91xs/PPP91YvXb1\nlz6f47OzSlep/qdT1Qb1IIAAAtkSIEHKljztIoAAAo5TIoQl+m7Rqfpu0Wg7W7TTAUvDY6bO\ncnRpHT4I5LxAtL7efl/Juf+P19e88eyTIZ35fE1nlS5Xx/+o2JDzA6CDCCCAQBIBEqQkKCxC\nAAEE0izQXfWfGAgGjw2EQqXTF+4bnLPPwb6uffunuVmqRyB9Ap9/9IHz4C2/jz381z/W6lbh\nVfV1dXb5nZ1V+jx9rVIzAgggkHoBEqTUm1IjAggg0JjACL8/cIYuozuwsmfv+rmHHh2ZvOsC\nJ1Ja1tj6LEcg7wRqqqqcJ+681bn7+muqv/xspV8HGr+PRqOXaCCv591g6DACCBSlAAlSUe52\nBo0AAhkWmB4Ihn9QX1ezy8Ax42vmLz0uMnbabMenb25QEChUAX0Q0HBTh7uWX13z3isvhPQe\nuE/vgYs03gcKdcyMCwEECkOAv86FsR8ZBQII5KbAnrrz3Dn6/Zhtxs3auX7Xw5YFB44el5s9\npVcIpFHg/Vdfcu6+4Vd1zz9wjz8QDL2k98RZau6uNDZJ1QgggECLBUiQWkzHhggggEBSAft/\n1RKjn+gL7EOn6neL5h1ytJ8fdE1qxcIiE1i14iPnruuuqn/sjr84gUDwjXiidLsY+D2lInst\nMFwEclmABCmX9w59QwCBfBKw/08X6o50FygxGjJjr/18upTO37Gr3Y+BggACXoEvP/3EuXP5\n1dFHb/tTTLe3f1t3cTxbz9+qsB+ipSCAAAJZFSBByio/jSOAQIEI7B1PjAbP3PsAJUbH+iu6\ndCuQoTEMBNInsHrVZ87fb/hV9MFb/hD1+X3vK1E6Q63ZGSUKAgggkDUBEqSs0dMwAggUgMAs\nXUr3c/1A5qhZiw/07XqYJUZdC2BYDAGBzAp88+V/nDuvvUqJ0u/tjNKLuvTudPXgocz2gtYQ\nQACBTQIkSLwSEEAAgeYLbBMMhi6tq6udNWX+wujC404PdOres/m1sAUCCGwm8MXKj52/XnVJ\n/VN/v92v99iDeo/ZGaXnN1uJBwgggECaBUiQ0gxM9QggUFAC/QOBwIX19fX7jN5hZt3ik38Q\n6jVoaEENkMEgkAsCK9550/nLL35W+/JjDwZ1M4e/1tfXnal+vZcLfaMPCCBQ+AIkSIW/jxkh\nAgi0XqC9qjhbv1t0Yr8Ro6NLTjsrPGTcdq2vlRoQQGCLAu+89Jzz50svqHnvtZd8sWj0Mq38\nE8XaLW7EkwgggEArBUiQWgnI5gggUNACfo3u8EAweHG7TpVl+595TmTC7LkFPWAGh0AuCjx7\n313OTf9zXvW61V+ur6+r+576uFzBHe9ycWfRJwQKQIAEqQB2IkNAAIG0CEzTDRiucXy+oXse\nfVJw5wOXOqFwJC0NUSkCCGxdoLa6yrn399c5t//mijonFntLN3I4Tls9tPUtWQMBBBBongAJ\nUvO8WBsBBApfoLe+83CZvvOwt92AYdFJPwh0qOxS+KNmhAjkicA3X6xy/nLlxfWP3/EX3fAu\neJveqyeo6yvzpPt0EwEE8kCABCkPdhJdRACBjAgE1MpJfn/gJ72HDPcf9KMLwgNGbZORhmkE\nAQSaL/DB6684N5z3vWrd0EF32q//oWq4UlHf/JrYAgEEENhcgARpcw8eIYBAcQpM1OV01+u7\nRoOXnPbj0Iy99tOVdfz3WJwvBUadTwLRaNR56JbfOzf/4sLaaF3927rsbqn6/1Q+jYG+IoBA\n7gnYJ6YUBBBAoFgF2vv9/stjsdg12+08v+Mpv7whNHTC9iRHxfpqYNx5J2AfZPQfNdaZtmBJ\n4KvPVlZ88s6bR+gscG+9px/TYKrybkB0GAEEckKAj0hzYjfQCQQQyILAYp0xuqaismv5IWdf\nFBk5aWoWukCTCCCQSoE3nn1Cl919v1rJ0jrd7e5I1X1bKuunLgQQKA4BEqTi2M+MEgEE/ivQ\nRV/s/nU0Ft19/uHH+Xc/4jhfKFLy32eZQwCBvBaoral27vjNFbG7l18T0xliu4nDMg3oP3k9\nKDqPAAIZFSBByig3jSGAQJYFluis0a+79u1fctRPfhHpM3RElrtD8wggkC6BD//9qnPtj06u\n/vyjD6p0NukYtfOndLVFvQggUFgCJEiFtT8ZDQIIJBf49qzRHked6NeZI18wFEq+JksRQKBg\nBOpqa527lv8y9rffXBn1+X33KFE6QoP7rGAGyEAQQCAtAiRIaWGlUgQQyCGBfXTW6Dfd+g4o\nOeqnv4joFt451DW6ggACmRDQrcDtbFLNyvfetrNJh6nNv2aiXdpAAIH8FOAudvm53+g1Aghs\nXaCtfkXyOsfxnbv7USdYchTsUNl161uxBgIIFJxAu46dnekL9w3o7nbht55/ZrHudDdA8/dr\noDUFN1gGhAACrRbgDFKrCakAAQRyUGBSIBi+pUNlZeWyi38Z4Qdfc3AP0SUEsiTw7svPO1d/\n7/jqNV9+saq+tmaJuvFklrpCswggkKMCnEHK0R1DtxBAoEUC9n/aj/XbKDdOnr+g7UlXLA9V\n9uzToorYCAEEClOgY9fuzoy99g2u/vzT8o/femOpRmn/b9jvJkULc8SMCgEEmivAGaTmirE+\nAgjkqkBf3Xjh5kAwNO7wcy8JbbfTrrnaT/qFAAI5IvDcP//uXHfOGbV1NdUv6YYOe6lbH+dI\n1+gGAghkUYAEKYv4NI0AAikT2N0fCP5h4OhtIkf/7MqwfUJMQQABBJoisHrV5841Zyyref/1\nl+0GDvtrm7uash3rIIBA4QrYaWUKAgggkK8CQXX8Z47Pd4Vu3R054vzLgmVt2+XrWOg3Aghk\nQaC0Tbmzwx6LArXV1eG3X3puP3WhjeIhBZfcCYGCQDEKcAapGPc6Y0agMAR66JK6W0ORkvHL\nLroqPGrKjMIYFaNAAIGsCbzy+EPOr75/Qk1tddWLuuRub3VkRdY6Q8MIIJA1ARKkrNHTMAII\ntEJgjn7b6JbeQ0a0Of6yX3NJXSsg2RQBBDYXWL3qM+eXpx9T8+Ebr27UJXeL9ex9m6/BIwQQ\nKHQBLrEr9D3M+BAoPIEf6JK663fa79ASfd8o2KZd+8IbISNCAIGsCTRccrf7okDVhvXhd195\n4UB1ZKPiiax1iIYRQCDjAiRIGSenQQQQaKFAWSAQ/KPOHC07+sIrA7scdIRPPwTbwqrYDAEE\nEGhcwO/3O6MmT/d16zvA99Ij98/STweM1g/L2s0bahvfimcQQKBQBLjErlD2JONAoLAFdAvv\n8N3lFR0HnnLl9ZHeQ4YX9mgZHQII5IzAR2++7lx+4mE161avfqeutma+OvZBznSOjiCAQFoE\nSJDSwkqlCCCQQoHpOmt0+4DR48pOuOw34fIOFSmsmqoQQACBrQus+3q1fS+p9u0Xn9sQra+z\n30t6YOtbsQYCCOSrANen5Oueo98IFIfAMl3a8udZiw8qOfrCK4IlZXb3XQoCCCCQWYFwSakz\nZf5egY3r1oTfe/XFg9T654rnMtsLWkMAgUwJkCBlSpp2EECgOQJ+x++/3O/zn3Po2RcGdjvi\neJ99J4CCAAIIZEtA/wk5+jkBn36I2v/Sow/Mc2Ixu0PM/YpYtvpEuwggkB4BLrFLjyu1IoBA\nywXKdEndzfrO0c4nXbE8NGzbSS2viS0RQACBNAi88cwTzhUnL63VbyX9Q7cCX6ImNqShGapE\nAIEsCZAgZQmeZhFAIKlAV/346z3lFZ1GnPGr34e79x+UdCUWIoAAAtkWWPne284lyw7SzRu+\nekM3b5ir/nyW7T7RPgIIpEaABCk1jtSCAAKtFxgeCIbv7zVoSOdTfnlDuF3Hzq2vkRoQQACB\nNAqs+epL5/ITDqv5+K1/f1VfV7OTmno1jc1RNQIIZEiA7yBlCJpmEEBgiwIz/YHgg2Omzmh/\nylU3hMrattviyjyJAAII5IJApLTMmbLbwsAn775Z8vlHHxwai0UfVb8+yoW+0QcEEGi5AAlS\ny+3YEgEEUiOwQF9+vmPHfQ+JHH7u/wR0iV1qaqUWBBBAIAMC+s6ks+2Ou/rXfvVF4MM3Xj1I\nPyj7kpp9KwNN0wQCCKRJgAQpTbBUiwACTRI4VLfx/v2CY08L7H3CGX7NN2kjVkIAAQRyScD+\n7xozbbbf8Tn+N5972m7aYGeRXsylPtIXBBBougAJUtOtWBMBBFIrcJrj81118A8vCOxy4BFk\nRqm1pTYEEMiCwNAJk3ztO1X6X3rswT3UfJXi8Sx0gyYRQKCVAiRIrQRkcwQQaJHAhX5/4Jyj\nf3ZFYIfdF7WoAjZCAAEEclGg34gxTs+BQ33PPXDPbF1uZ1+ovDcX+0mfEECgcQESpMZteAYB\nBFIvoK8bBa7VF42OPenK5cFxM+2mTxQEEECgsAR6DBjsDN5mO/8z9945UT8o20OJ0t2FNUJG\ng0BhC5AgFfb+ZXQI5JKAX8nRDaFI+MAzf/NHfgA2l/YMfUEAgZQLVPbs7Qzfbor/qb/fPlZ3\nt+uvJOlvaiSW8oaoEAEEUi5AgpRyUipEAIEkAgF/IHBjqCSy7/f+9+bQgFHbJFmFRQgggEBh\nCXTs1sMZMWlqQEnSKCVJQ5Qk3a4RkiQV1m5mNAUoQIJUgDuVISGQYwIBlZvCJaWLvn/dn0P9\nho/Ose7RHQQQQCB9AhVdujmjd5gReOru24fHovWjlSTdptai6WuRmhFAoLUC3DmqtYJsjwAC\nWxIIBgLBP4dLS3f7/nU3h3sPGb6ldXkOAQQQKFiBFW//2/nZ0n1qqjdsuKe+vs7uTlNbsINl\nYAjkuQAJUp7vQLqPQA4LWHL0l0hZ2bzvL78l3GvQ0BzuKl1DAAEE0i+w8r23nQsPW1yzYf3a\nu2L19YvVYn36W6UFBBBorgCX2DVXjPURQKApAvrKUeCmSFmb3X94w63hngOHNGUb1kEAAQQK\nWqBtRSdn7PTZgSfvum1gtL5+WPxyO76TVNB7ncHlowAJUj7uNfqMQG4L2K28r9MNGZb8gDNH\nub2n6B0CCGRcoF3HTg3fSXrizr8O03eS+ipJsrvbURBAIIcESJByaGfQFQQKQsDvvyIYCh1+\n5rV/4oYMBbFDGQQCCKRaoH3nLs6I7XcIPHHXX0dFY7Gu+q0kficp1cjUh0ArBEiQWoHHpggg\n8B2BnwaDwZNOu/p3ocHbbPudJ1mAAAIIILBJwO5uN3TCpMBTd902PhqNttPS+7BBAIHcECBB\nyo39QC8QKASBH+rSuh+fdOXy4IiJOxTCeBgDAgggkFaBTt17OgPHTPA/fc8dE2PRaJ0aeyyt\nDVI5Agg0SYAEqUlMrIQAAlsROMrn81163KXXBMZOm7OVVXkaAQQQQMAVqOzVx+k5aIj/2fvu\nmqVlKxQvuM8xRQCB7AiQIGXHnVYRKCSBPZQc3XjIWT8LTJq3ZyGNi7EggAACGRHo0X+Qo+8l\n+V5+9IHd1KAlSG9lpGEaQQCBpAIkSElZWIgAAk0UmKxb1t295zEnB3c+cCm/q9ZENFZDAAEE\nEgX6jRhti3xvPf/MIt204QHNf5y4Do8RQCAzAiRImXGmFQQKUWCYPxB8ePrCJZElp/zIX4gD\nZEwIIIBAJgWGbTvJ980Xq5yP3nxjSSwWvV1t/yeT7dMWAghsEiBB4pWAAAItEegRCIYfG73D\n9PZHXvDzoM4itaQOtkEAAQQQSBAYPXWW/6M3Xwv8Z8WH++jGDTfp6bUJq/AQAQTSLMAlMWkG\npnoEClCgPBgKP9Nn6PCB3/vfP4dDkZICHCJDQgABBLInUFtT7Vy0dJ8anUl6s662ZpJ6siF7\nvaFlBIpPgI99i2+fM2IEWiOgq+qCN7fvXDng5Kt+S3LUGkm2RQABBBoRCIUjzklXXB9u27HT\nUP2f+2etxvFaI1YsRiAdArzh0qFKnQgUrsDFOnu046lX3xgpb9+hcEfJyBBAAIEsC7St6Oic\nfs3vwsFQaBd15eIsd4fmESgqAb6DVFS7m8Ei0CqBw3U775+efOUNwQGjxraqIjZGAAEEENi6\ngCVJA0aNCzx51//pMrvYSm3x/Na3Yg0EEGitAAlSawXZHoHiEJih5OgvB/7g/MDEne1nOigI\nIIAAApkQsB+S7VDZ8BtJ89Xe44r3M9EubSBQzAIkSMW89xk7Ak0TGKhr4B+atc9BkT2OOokb\nuzTNjLUQQACBlAn0Gz7aqVq/znn/tZcX6vbf9p2kr1NWORUhgMB3BEiQvkPCAgQQ8Ai0CYbD\njwwdv33lUT+5nNt5e2CYRQABBDIpMGLiDr43n3va//V/Vs2LRuuvV9u1mWyfthAoJgFu0lBM\ne5uxItBMgUAgcEPbik79ll38y5A/wOcpzeRjdQQQQCBlAvZ/8LGXXBMqr+g4QP83W4JEQQCB\nNAmQIKUJlmoRKACBU/UDsAtOvmJ5uE279gUwHIaAAAII5LeA3T1U/ydH9H/zXhrJafk9GnqP\nQO4K8JFw7u4beoZANgVm6qYMv1t63qXBkZOmZbMftI0AAggg4BHQ79A5lT37+J9/8B9ztJib\nNnhsmEUgVQIkSKmSpB4ECkegV/ymDCW7HraMmzIUzn5lJAggUCACvQYPczauW+t88PorC2LR\n6E0a1jcFMjSGgUBOCHCJXU7sBjqBQM4IhPSjhLfrjklt9j3tLP5/yJndQkcQQACBzQX2OfmH\n/v4jx5bq/+xb9Uxw82d5hAACrRHgAKg1emyLQOEJXBQuLRt1/GW/DussUuGNjhEhgAACBSJg\nN21YdvHV4VCkZIyGdGGBDIthIJATAlxilxO7gU4gkBMC8x2f7xcn/vzaYO+hI3KiQ3QCAQQQ\nQKBxgdI25U7fYSMDT/799kla61+Ktxtfm2cQQKCpAiRITZViPQQKW6Cnzhj9c94hR5fMXLQ/\n3zsq7H3N6BBAoIAEuvTu59RWVzvvvfribvo+0u81tDUFNDyGgkBWBLjELivsNIpATgkEdA37\nzX2GjihdeOxpJEc5tWvoDAIIILB1Afu/u+/wUfZ9pL9obT783joZayCwRQESpC3y8CQCRSFw\nljKk7Y79n2v43lFR7G4GiQAChSZg3xm1/8OD4ch4je3cQhsf40Eg0wJ8ypBpcdpDILcEput7\nR8uXXXRVcODocbnVM3qDAAIIINBkgdLytk6PAYMDz9x751RtdL/i4yZvzIoIILCZAAnSZhw8\nQKCoBNoGguGHpi9c0kbfPeJsclHtegaLAAKFKNC930Dnq89WRj955+25sWj9tRpjTSGOkzEh\nkG4BDorSLUz9COSogN8fuLJDZWWnfU/7MR+U5Og+olsIIIBAcwX2O+OcQLvOnSvt//jmbsv6\nCCCwSYAEiVcCAsUpsGcsFj3o6AuviERKy4pTgFEjgAACBShQUtbGWfazKyP6P/5gDW9BAQ6R\nISGQdgE+OU47MQ0gkHMClfpC7/1zDz2mdOoei7lrXc7tHjqEAAIItE6gY7ceTn1dnfPuy8/P\n062/r1dt61tXI1sjUFwCnEEqrv3NaBFwlBwt79Z3QNmCY04hOeL1gAACCBSowJ5Hn+zTTRtK\n9X++JUgUBBBohgAJUjOwWBWBAhA4OBaLzTvmoqsi+r2MAhgOQ0AAAQQQSCZgt/4+6qdXhGPR\n2M56fv9k67AMAQSSC3CJXXIXliJQiAJdA4Hg3/c67vTSCbPnFuL4GBMCCCCAgEegXcdOjs/v\n97/9wrM76lK76/TUBs/TzCKAQCMCnEFqBIbFCBSagJKjX3brNyCyy0FHFtrQGA8CCCCAQCMC\nux62zOnap1+J/gZc3cgqLEYAgQQBEqQEEB4iUKACe0Sj9QuPOP+yiF12QUEAAQQQKA4B+z//\nyAsuj+hvwF4a8Z7FMWpGiUDrBLjErnV+bI1APgi0CwRD9+184NI2O+y+NzdmyIc9Rh8RQACB\nFAp0qOzi1FRVOe+/+uJOutTOfkC2KoXVUxUCBSfAGaSC26UMCIHNBfRjgZfoj2P7BctOJTna\nnIZHCCCAQNEILDjmZF/Hrt3b6m/CpUUzaAaKQAsFSJBaCMdmCOSJwDRdVrF06XmXRMIlJXnS\nZbqJAAIIIJBqgVCkxDn83EvsUrtDVffkVNdPfQgUkgAJUiHtTcaCwOYCwWA4/L+6rC42bFv+\nFm5OwyMEEECg+ASGTtjemTRvz2gwFLY72vE1i+J7CTDiJgqQIDURitUQyEOB43XXov6LT/4h\nfwTzcOfRZQQQQCAdAvueflbQHwwMUt3Hp6N+6kSgEAQ4cCqEvcgYEPiuQDd/IHD7vqf+uGTY\ndpw9+i4PSxBAAIHiFIiUljll5e0Crz75yHT9cPj1UlhXnBKMGoHGBTiD1LgNzyCQtwJKjn7e\no/+gwMzFB+btGOg4AggggEB6BOxvQ48Bg4OBQOAX6WmBWhHIbwESpPzef/QegWQC06P19UsO\nOevCiH5APdnzLEMAAQQQKGIB+9tw2NkXhevr6xeJYXYRUzB0BJIKcPSUlIWFCOStgN2Y4dod\n9lgUHThmfN4Ogo4jgAACCKRXoP+osc60hUvshg1XqyWOB9PLTe15JsAbIs92GN1FYCsCR+o3\nLvovPukHfL9wK1A8jQACCBS7wF7HnR7w+X0D5LC02C0YPwJeAQ6ivBrMI5DfAu0CweCdC489\nrc2oydPyeyT0HgEEEEAg7QIlZW0cfagWeOv5p3eIRaPXqMGatDdKAwjkgQBnkPIMac+BAABA\nAElEQVRgJ9FFBJoo8IPyik6lO+53aBNXZzUEEEAAgWIX2PmAwx397Wgjhx8WuwXjR8AVIEFy\nJZgikN8CfXx+/6n7nvqjSCgcye+R0HsEEEAAgYwJhCIlzn6n/TiivyGnqdF+GWuYhhDIYQES\npBzeOXQNgaYK6AqJi/sMGR7bfu4eTd2E9RBAAAEEEGgQmLjL7k6/4aNi+nHxSyBBAAHuWsJr\nAIFCENhOt/XeZ//vncupo0LYm4wBAQQQyIKA/obott91e6np7bLQPE0ikFMCnEHKqd1BZxBo\nvkAwFLps/Kxd6gdvs23zN2YLBBBAAAEEJDBw9Dhnmxk71gVCoYsBQaDYBUiQiv0VwPjzXWBO\nfV3dlL1POCOY7wOh/wgggAAC2RXY+4QzQ/W1tTPUi5nZ7QmtI5BdARKk7PrTOgKtEtAP/F00\nedcFse79B7WqHjZGAAEEEECg58AhzqR5e9qPx/JdJF4ORS1AglTUu5/B57nAbrpefJs9l53K\n75nl+Y6k+wgggECuCCzUj8fa3xb1Z/dc6RP9QCDTAiRImRanPQRSI+Czs0fTF+7rVPbsnZoa\nqQUBBBBAoOgF7G/KzL339wXDYfsukq/oQQAoSgESpKLc7Qy6AAQWxWLRoXscdSJnjwpgZzIE\nBBBAIJcEdj/yRL8Ti9m120tyqV/0BYFMCZAgZUqadhBInYA/EA5fOGfJwb6KLt1SVys1IYAA\nAgggIIEOlV2cWYsPCugs0rl6yFkkXhVFJ0CCVHS7nAEXgMBeTjTWd95hy3j/FsDOZAgIIIBA\nLgrMO/QYn35jb6D6tiAX+0efEEinAAdY6dSlbgTSIKDvHv2/GXvt52/fqTINtVMlAggggAAC\nm84i6W+Nfd/1PDwQKDYBEqRi2+OMN98F5unuQsPmHXo0791835P0HwEEEMhxAZ1F8utvznB1\nc16Od5XuIZBSAQ6yUspJZQikVyAYCp0zZf5Cp1P3nultiNoRQAABBIpeoHOPXs6U3fZyOItU\n9C+FogMgQSq6Xc6A81hgel1t7cRdDz+WO9fl8U6k6wgggEA+Cey29PhAfV3tePV5dj71m74i\n0BoBEqTW6LEtAhkUCAZD50yYM7e+ez/7ziwFAQQQQACB9At07dPP2Xan+VFdwfCj9LdGCwjk\nhgAJUm7sB3qBwNYExtbV1c7e/cgTgltbkecRQAABBBBIpcC8Q44O6gqGWapzZCrrpS4EclWA\nBClX9wz9QsAj4PcHTh0ybruaPkP52+RhYRYBBBBAIAMC/UaMdgaOGV/r9/tPz0BzNIFA1gVI\nkLK+C+gAAlsV6BpzYvvNPeTo8FbXZAUEEEAAAQTSILDrYcvCMcc5QFV3TUP1VIlATgmQIOXU\n7qAzCCQVWNaxa/fo2Olzkj7JQgQQQAABBNItYH+D7G+R2jk23W1RPwLZFiBByvYeoH0EtiwQ\nCQSDJ849+KiIz+fb8po8iwACCCCAQJoEdHmdo+8i2d+kk9RESZqaoVoEckKABCkndgOdQKBR\ngf0DoVD51D0XN7oCTyCAAAIIIJAJgR32WOwEw5FStXVQJtqjDQSyJUCClC152kWgCQLBcPjM\nWYsODEZKy5qwNqsggAACCCCQPoFIaakza/GBoVA4fEr6WqFmBLIvQIKU/X1ADxBoTGBqXU3N\n0Dn7HsK1dY0JsRwBBBBAIKMCMxcd4KutqRmmRidltGEaQyCDAiRIGcSmKQSaI+APBJaNnDyt\nrnOPXs3ZjHURQAABBBBIm0CXXn2c4ROn6JbfgePS1ggVI5BlARKkLO8AmkegEYGOsVhs0ex9\nDg418jyLEUAAAQQQyIrAjvseqlt+x/ZR4x2z0gEaRSDNAiRIaQamegRaKHBIm3YVsbHTZrdw\nczZDAAEEEEAgPQJ2y+/y9hX6WSTn0PS0QK0IZFeABCm7/rSOQFIBfQH2+NlLDozoMrukz7MQ\nAQQQQACBbAnY36bZSw6O6G/VieoD35PN1o6g3bQJkCCljZaKEWixwIy62tp+0xfu2+IK2BAB\nBBBAAIF0CtjfqLraut5qY1Y626FuBLIhQIKUDXXaRGALAnZzhlFTZtTrF8u3sBZPIYAAAggg\nkD2Bii5dnVGTp9XrZg2HZ68XtIxAegRIkNLjSq0ItFSgnW7OsHDG3vtxc4aWCrIdAggggEBG\nBKYuWGJ/q/ZWlGekQRpBIEMCJEgZgqYZBJoosMh+iG/M1NlNXJ3VEEAAAQQQyI7AuJk7OqGS\nEjuWXJSdHtAqAukRIEFKjyu1ItAigWAovHTK/L2CwRAnkFoEyEYIIIAAAhkT0N8sZ8r8hfqb\nFT4yY43SEAIZECBBygAyTSDQRIG+dbU1kyfvthfvyyaCsRoCCCCAQHYFpu6x2G9/u9SLftnt\nCa0jkDoBDsRSZ0lNCLRW4MDOPXpVDxw9rrX1sD0CCCCAAAIZEeg/aqzTpVefGjV2cEYapBEE\nMiBAgpQBZJpAoCkCoXDkiGkLl5Q0ZV3WQQABBBBAIFcEpi3cV7+JFFmaK/2hHwi0VoAEqbWC\nbI9AagS2q62p7jt514WpqY1aEEAAAQQQyJDAxF12d/Q3rI+a2yZDTdIMAmkVIEFKKy+VI9Bk\ngcV9h4+q0SV2Td6AFRFAAAEEEMgFgcqevZ3eQ4ZXqS9LcqE/9AGB1gqQILVWkO0RSIFAMBzZ\nf9K8PSMpqIoqEEAAAQQQyLjA5PkLS/S37MCMN0yDCKRBgAQpDahUiUAzBSbU1VT33HbHXZu5\nGasjgAACCCCQGwLb7TTfsb9l6s2E3OgRvUCg5QIkSC23Y0sEUiWwuM+wkdWdutvfFQoCCCCA\nAAL5J2B/w/oOG2V3s+Myu/zbffQ4QYAEKQGEhwhkWkCXJBwwaS6X12XanfYQQAABBFIrMGn+\ngoj9TUttrdSGQOYFSJAyb06LCHgFxumShF7b7jTPu4x5BBBAAAEE8k5gO10qrr9pPdTx8XnX\neTqMgEeABMmDwSwCWRDYq9fgYVWde/TOQtM0iQACCCCAQOoEOnbr4fQYMNjuZrd76mqlJgQy\nL0CClHlzWkTgW4FQJLK3bs7Aj8N+K8IMAggggEA+C2y747wS/Wjs4nweA31HgASJ1wAC2RPo\nVVtdPXzstNnZ6wEtI4AAAgggkEKBbWbsZD8aO1JVcuehFLpSVWYFSJAy601rCHgF5pe3r6jW\nHey8y5hHAAEEEEAgbwX0o+eO/W3TAHbL20HQ8aIXIEEq+pcAANkSCATDC8bN3iXk8/my1QXa\nRQABBBBAIKUC9jdt/Jy5oWAotHdKK6YyBDIoQIKUQWyaQsAjUBKL1s3aZvoc3oMeFGYRQAAB\nBPJfYNzMnfz19fUzNJKy/B8NIyhGAQ7OinGvM+ZcEJjt+HzBEdvvkAt9oQ8IIIAAAgikTGD4\ndlMcv98fUIWzUlYpFSGQQQESpAxi0xQCHoF5Q8dvXx8p5cM1jwmzCCCAAAIFIBAuKXEGjZ1Q\nr6HsWADDYQhFKECCVIQ7nSFnX0C399519NSZ4ez3hB4ggAACCCCQeoHRU2eF9beOGzWknpYa\nMyBAgpQBZJpAIEGgh27vPYDL6xJUeIgAAgggUDACI7ef6uhv3SANqHvBDIqBFI0ACVLR7GoG\nmkMCc0ratKnpPWREDnWJriCAAAIIIJA6AfsJi5I25TWqkcvsUsdKTRkSIEHKEDTNIOAK6Iur\nO46cNM3P7b1dEaYIIIAAAoUmYH/jRk2ebvdq2KXQxsZ4Cl+ABKnw9zEjzDEBfyAwVwlSMMe6\nRXcQQAABBBBIqYASpKAv4J+b0kqpDIEMCJAgZQCZJhDwCAypq63tMnwit/f2mDCLAAIIIFCA\nAsP1Uxb1tbWdNLShBTg8hlTAAiRIBbxzGVpOCsxqW9G5umuffjnZOTqFAAIIIIBAqgQqe/Z2\n2rTvUK36pqaqTupBIBMCJEiZUKYNBOICuhh76vDtJnF5Ha8IBBBAAIGiEBi27eSA/vZNK4rB\nMsiCESBBKphdyUDyQcAfCs0cPH6i/bo4BQEEEEAAgYIXGDph+2AgFJpd8ANlgAUlQIJUULuT\nweS4QJe66upeg8aOz/Fu0j0EEEAAAQRSIzB43Lb2e0i9VVvX1NRILQikX4AEKf3GtICAK7BD\nKByp6z14uPuYKQIIIIAAAgUtYH/zQpFIrQbJ95AKek8X1uBIkAprfzKa3BaY0n/U2Hrd5ju3\ne0nvEEAAAQQQSJGA/c0bOHp8TNWRIKXIlGrSL0CClH5jWkCgQUBnj2brWuwIHAgggAACCBST\ngP72hXUWaUYxjZmx5rcACVJ+7z96nz8Cobq62tGDxk7Inx7TUwQQQAABBFIg0G/kGEe/AThS\nVXEX1xR4UkX6BUiQ0m9MCwiYwKhYNBrqN2IMGggggAACCBSVQL8Rox39DQxr0COKauAMNm8F\nSJDydtfR8TwTGN+2Y+eqthUd86zbdBcBBBBAAIHWCbTvVOmUd6iwH4zdtnU1sTUCmREgQcqM\nM60gML7/yDHcnYHXAQIIIIBAUQoMGD3Ojjm5zrwo937+DZoEKf/2GT3OQwF9OXVy/5FjQ3nY\ndbqMAAIIIIBAqwUGjNompL+FO7S6IipAIAMCJEgZQKaJohcI1NfWjug7fFTRQwCAAAIIIFCc\nAvY9JN2owX4IkBs1FOdLIK9GTYKUV7uLzuapwLBoNBrpO4wEKU/3H91GAAEEEGilgH1IGL9R\nA7+W3kpLNk+/AAlS+o1pAYFxZW3bVVd06YoEAggggAACRSlgN2ooaVNeo8Hb7b4pCOS0AAlS\nTu8eOlcgAiN7DhxSIENhGAgggAACCLRMoOeAIVFtSYLUMj62yqAACVIGsWmqOAUCweDo3kNH\nRIpz9IwaAQQQQACBTQJ9ho8s0d/EsXggkOsCJEi5vofoX94L+APBMT0GDM77cTAABBBAAAEE\nWiPQU38L/YEACVJrENk2IwIkSBlhppEiFojUVlf37DFgUBETMHQEEEAAAQQcp4cuN6+trukl\nixI8EMhlARKkXN479K0QBPTlo5ifM0iFsCsZAwIIIIBAawR6DRqqzWN27DmsNfWwLQLpFiBB\nSrcw9Re7wIiSsjY17Tp2LnYHxo8AAgggUOQC5R0qnNLyttViGFHkFAw/xwVIkHJ8B9G9vBcY\n3r3/QLtrDwUBBBBAAIGiF+jSu29MCAOLHgKAnBYgQcrp3UPn8l3A7/cP6j5gSDjfx0H/EUAA\nAQQQSIVA9wGD7W8iX8xNBSZ1pE2ABClttFSMgOMEwuFhXXr14X3GiwEBBBBAAAEJdO3V1x8q\nKRkOBgK5LMCBWy7vHfqW9wKx+vp+nXv2zvtxMAAEEEAAAQRSIVDZu6+jv40DUlEXdSCQLgES\npHTJUi8Cuo1pXW1txy49+2CBAAIIIIAAAhLQVRWO/W3UbCkgCOSqAAlSru4Z+lUIAv01CB9n\nkAphVzIGBBBAAIFUCFQqQVLxKTiLlApQ6kiLAAlSWlipFIEGgf6BYLC+fedKOBBAAAEEEEBA\nAu07VTrBULhOsyRIvCJyVoAEKWd3DR0rAIH+Hbt2r/H57IMyCgIIIIAAAgiYQIfKLrWa9EID\ngVwVIEHK1T1DvwpBoHen7r14jxXCnmQMCCCAAAIpE6jo2t0+OeyZsgqpCIEUC3DwlmJQqkPA\nI9CjY7fu/AaSB4RZBBBAAAEEOvXoZX8bSZB4KeSsAAlSzu4aOpbvAqFIpE+Hyq5cX5fvO5L+\nI4AAAgikVKBjl25+/Y3sl9JKqQyBFAqQIKUQk6oQ8Ar4HF9PJUjeRcwjgAACCCBQ9AL2t1F/\nI/mRwKJ/JeQuAAlS7u4bepbnAvX1dV3ad+6S56Og+wgggAACCKRWoKJLV6e+ro5PEFPLSm0p\nFCBBSiEmVSHgEQjrP/92ulOPZxGzCCCAAAIIINChSzdHHyKWS4Ifi+XlkJMCJEg5uVvoVAEI\ndLcxcAapAPYkQ0AAAQQQSKlAu46d3fr4oUBXgmlOCZAg5dTuoDMFJNBw6qh9p2//CBTQ0BgK\nAggggAACLRdo0669u3End4YpArkkQIKUS3uDvhSSQIXf749GSssKaUyMBQEEEEAAgVYLlJa3\ndXz6G6mKOra6MiooJAG7829QkfU7AJMgFdLLirHkkkBFpKyN/VI4BQEEEEAAAQQ8Aj6fzynZ\n9DeSM0ibXCo0eVDxWw9Trs3uqQ69qBjWzI4FtP7/KFYovlL8SdHYpZXn6Lk1ioavKWiatWJZ\nGgUBBFIvUFHWtp19OkZBAAEEEEAAgQQBXWYX3bhuLWeQNrnYD+fOVPx708Oc+3ekevR7hd1Y\no7mXxpyvbU5TWGL0juJkxQDF9oqYwi2WLJ+quFqx0l2YrSkJUrbkabfQBSratOtQ6GNkfAgg\ngAACCLRIoE37CueLlSs4g9QivYxuNFWt/UFhyVFziyV+JyjuVewf3/hzTa9SzFb8M77MJt9T\n2KV1F9mDbBcSpGzvAdovVIGK8ooKLmEt1L3LuBBAAAEEGhWIxWKOzg5tivXrvp2v0vwGLa9a\nt85Z9/VXEVUwttFKeCLbApYQ/UxxrMLO9Hyj+PbuGppvSumnlaye5z0rvxCfH6GpmyDZJXXH\nKy5T/EeR9UKClPVdQAcKVKCivEPHUIGOjWEhgAACCBSoQPXGDc6GtUpi1q9tSGbWfb3aqamq\nsoTGqa2pdtau1rS6WknPGmf92jXRb75Y5dRs3OhsVPKz/pvVvqoNG5xYNPqdL9nrpgyxYDDo\nBALBWDAUdOpqa22ddhlitEs69lXYQbkd+76iuFvxocItczQzWPGw4g13oWfaX/O7KN5V3OdZ\n3lbz9v2cbRRtFC8prO6PFN5iCeFhCtv+X4qjFHar2zsUbykaK/Ydnl0V1vchCvuOztuKvyve\nV3jLZD2wpPOvCktMdlPY5XFPKWz9zxRNLRdrxWWKFYoDFGcrzKg5xfpupWrTpOFf9/vZ3t/A\n+pGeqVZc4lkvq7MkSFnlp/FCFQgEAh11fTVnkAp1BzMuBBBAIMcELHnZdNYmfsZGCU7i43Wr\nlexUVzlrv/oiVrVhfWzD2jXxZGidz+bramt8icPSDRVigU2JjaMEJxYKhZxwOOSUlZb6y8vL\n/ZXl5U67rp2dzp07Ox07dnQ6dOjQMN+tWzena9euzrBhwxybV3HrbpguXbrUWb58+arE9tLw\neJHqvEZhyYidCbHvB9uB+3rF6YpfKaz0Vth6NyksIUgs9v2Y4xWW2LhlB838TtFfYXXbwX9Y\nYUmMrWvPucWSJ6v/NoWdMZmksHKQYnzD3Hf/GapFNyomxp+q09Q9dt+g+SMUf4w/Z5O9Facp\nLPmwhMYS0C8V1mfr34mKqxRNKau10vcVv1Ssa8oGSdb5UMusXfNxiztviaKVvoojFecpvlbk\nRHGRc6IzdAKBQhHwBYLl4UhJoQyHcSCAAAIIpEmgvq6u4exLwxmbhjM37mVom87g/PeytIbH\nsXXfrI5tWLPG2aiERmdtdMZmvc72NBy/+hK6uCmxUXITjJ+1CYfDsdJIxN+mTRufkhtf9+5d\nnPbDBjsVFRVOp06dGhIbS2osevbs2ZDcKCny1uudT2iueQ8jkYglXGV1Gn8ayzDVbQmG9ftY\nhSU/NYr5il8rLGFZqbhDcYviSsUCRbnCmxTYFSH7KjYq/qywYknO3xWWEJ2kWK6wumcprldY\nu5YEPKHwlnl6YOtdoLA2LOoVyYolWNspfqG4VvGmYnvF4fG4VNO/KCwx85bL9OBvisMUliBZ\nAnar4ucKO8P1qGJr5UdbW6EJz1sSZ+0uUtyneE9h9Zr5PQorlsjZ5Xs2xpwpJEg5syvoSCEJ\n6BO3NiH9509BAAEEEChcgWTfs7FLzb5dbt/DscdKfNav+Sa6fs3XOmOzRt/B0fIN6322XrLL\n0fyBQENy4yY2OmsTK7HEpqysIbHp3bGd065fr4YzNnbWxpIbS2q6dOni9OjRoyGxsQTII++d\n9yzO3qwlSLraIt0JkiVAdjbFzubYvFssqbCzPP9QXKGwRMfOKN2ssOTDkqTfK9wyVzOdFX9Q\n2HZWfqpoqzhDYXW4xeq0MyJ3Kuyg3xIcb7FPT49TWELllq7ujGc6UvOjFP9WnK5wM8nHNf+i\nYj9Fd0VPxQcKb/lQD+xsUm184fOaHqx4RHG1YrQiU+UUNdRPYXZWPlPsozDvIYpDFGcq1ily\nppAg5cyuoCOFJOBzfKWcQSqkPcpYEECgUATsBgL2nZqGJMZzGZr7nRu7gUDDjQSU2GzQ92yU\n0ETX63s4lths1HN21sa2ra9r+A7NZiz2PRt9x8axS9JCIV2OFgzFIpGwzy5H01kbf5d2Smy6\nDWg4Y2OJjV2WVllZ2ZDYdO/e3Rk8eHDDvCp1Exp3ulk7hfCgpKTE0YeJliykqwRU8WTFWsW1\nSRq5V8teVoxRDFa8rrCkxRKkAxXeBMmSCyt2Zsgtu2kmpvAmOu5zVrddoratooPia4W33OF9\n0Mj8a1reRmFGbnLkrhrSzDsKS3QsSUsslgS5yZH7nJ01ekMxQlGm2KDIRFmhRiYpLNmzdi25\n26iwcq7ic4X118o2iiMVPRRPKy5XVCkyXkiQMk5Og0Uh4IuVhvSfPwUBBBBAIHUC+o6M+52Z\n/yY4OjvTcJam4UxNw2VnDc8poWm4HM0uRdt00wFdjqYbEFTrkjSVxMRDZ2xCdslXw/ds7CYC\nkXDEKS2J2Bkbf3t9z6Z3n54N36+xy9HcxMa9HM1NbtJ1OVrqBHOnJjuDpNJOYcmJPbCwP5zu\nvDt1l1lS8AdFYrKhRUlLfy21Ot5U2CVtyYqdnbEEaZjCEqTHFW8pdlTYWR07eO+g2F3xoeIB\nhRVb1llhl8ZZMpSsWH+tDFY82zC36R/r/xeex1uatQTMzoDtpbAzStbPofGpX1MrlggmFhtX\nsvKeFg5X2JifSrZCmpaZ00sJdVsfliiOU1gSNEHxsMLWteTIztAtVExRNHYJop5KTyFBSo8r\ntRa7QMzhDFKxvwYYPwIIfCsQra/flMTYpWUWSS5D23TmpuEsjRKbr/U9m/jlaO73bJTYJLsc\nze6KFggGdNYmpDM2Qd1AIOzocjQlNm189l2bbrqBQPshA7+9eYAlN+7laPY9m0GDBjllunTt\n285+N3nyPMVsqgTef/99p6qqyg6KLSHxlqgeVCeEHUDbWQc7C9LUBKmL1rWypfXXbFqlIeGJ\nzzo3aMYOzvdTXK5YrLBE67cKS1isWPJkpUZh/U1W3ogvtLMm3vKl98FW5o/X8z9T2JkkK18p\n7KzXnxX7KizZSVYaa6Ph0wFtUJFsowwvO1/tWdL5v/F2z4lPLaFcpThIcaPigPhUk8wVEqTM\nWdNSEQlEnVgJl9gV0Q5nqAgUqEA0GnXsts8NSY0uL7PL0L5NcBrO3MQfN1x6tja2bs03sQ3f\nbEpsbL2q9et1E4F1uhytzpuAmFbM77ekpuFyNN3+OWCJjf3H6S8rK234nk3PDrocrU+Pzb5n\nY9+xseTGvmfTv3//hjM5qsut250W6N4orGGNGTPG0Vmkt6urq2dqZG5CZIlQ4uVkLR24nS2x\n0nvTJOm/feJL/+N51g7K7eB9f4UlSAcqLDG6QeGWjzRjy6zfE92FKZ7aWawrFJbUHKv4h8Id\nk2a/vdNestd9D1shSekVX/ZCkucyucjM9lAcrnAvBdxB83ZWy5IjK7dvmjTclML2SUYLCVJG\nuWmsWAT0KWcJl9gVy95mnAjkpsCm79nELzn79jI03Szgm2/+m/RYwqPn9J2bqH7Dxm4k0PAj\nnhvXr/VV67dtaqo2fufga9Ntn0NKbgK6HC0UC+sGAkpufG2U2OhMjL+zfc+mS7+G79nY5Wj2\nHRs7a+PeQMAuR1Nyo6/r+F2477ThPsG0cAXsO0h6DdjlYSvTNMrPVK+dIbIkqL/ifYW3lOuB\nncGyYmdl3PKJZu5VzFNsr5imeFjh3X6jHluS1FcxXvG8wlvCemDJlX0P6TJFY2d09FSjxS4v\ns/fGRYprEtayM1oD48vMMLFYv29JWNhBj0crzNtssll+osbfVLiJj/1nYP0zL7fYd8csCe3q\nLsjklAQpk9q0lS0B9z+PzF3DGrPvnn77xz9b46ZdBBDIQwH9gOZ/z9LoJgENNwaIX5bWcPcz\nzyVq7vdsNt1AwM7mrPPZd2zsrI9uRrB54mG/Z9NwAwFLbBpuIKBP8O1ytBJdilbmb9u2rdOz\nV/fNvmfjXo5mv2NjYWdtuBwtD19UOdhlS5BU7EA/ncXOwPxYcaliicI9W6HZht/d6aTpA4oP\nbYGnXK95S5CuU9j7yB4nliu14BKFtbGzYoPCLf9PM8sUbyms/ZYUS7Ks9Gv497//WH9+o3CP\nbZIZHqPnr1Z4zzidq8dtFcsV2Swz1bidHdtP4R6XRTVviWmlwi02b2O1RDTjJZsJ0nSN1naW\nnZ7cX2HXVTZWbtMTllkuUHzT2EpFsPwUjdFOSZ6neDBHxmuvoaMVdmq0r+JDxU2KuxQxRWJp\n7vqJ27uP7RMbe/OPUXygOF/xR0VisdfN+4obFSclPpm+xzElSPa+piCAQLEI2Pds7HIyb0IT\n/60aT8LjXpKmy9G+0eVo8e/Z2HdylNjocrQNTrQ+yeVogU2Xo1lio7M2uoFAqOF7NvYdG4vK\nzhVOh0H9v/2ejXvbZ/dyNLskTWdwvP8peeeLZRcxzhwSsJs0KIl3k4B09exCVbyPws7GPKH4\nk8Iu47NjqZ0VLygWKRLL7VpgZ31GKuxMxq2KxGKJkR2/2vGP1XODwtadr9hJUaOwS+OSHQtp\n8VaLtXmE4iCFJRD3K/oprP7JivcUAxTJzrCY67MKS+I+VewWjwc0PUORzXKBGn9Z8eeETtj4\nDlTYnf+eVxynsPKPTZPM/msHq9kqdvePmfHGbQceEJ9PNrEXn2WS6X4jJWs7l5YNUWdmKn6V\nI53qpX78n2KCwj4FeEcxTmGfCtg+PVHhLc1d37utd95eOw8pvlacrbD/6G5SrFFYYuYt9h+B\nvW5+6l2YgXnyowwg0wQCqRCw2z5v+p6NJTf/vQvafxOdTUmN3URAl545+p5NtOF7NrocLX73\nNF+NLkerran+TtJht322O6PpuzYN03Ao7N72ueHuaN3tcrRe3RoSG0tq3MTGezlar169uBwt\nFTuaOnJKwBIkfcfN/j6ns9hZHTtG+R/FoQo742NlpeJ3CvvgebUisVhyc5PiBMUtivWKxFKr\nBVMU9gGtnS3yHmdYcnKm4iFFS8s92vD/t3cfcFKU5wPHZ9sdNlCKKCqgAkHEbuw9GmM0RmNs\nKcbeY2LXmMSYaNTYG4q9Ib1X6R1p0o8Od0c5uOM4uL5lZv/Pc9zxP88DruzuTfm9n8/Dtpl3\n3uf7Lrf77MzOaoGl/d5VGVooTZXQ91onS3wt8SuJQRLV21Ny40KJf0j4JfIkvpD4q4SOu6ma\n7pXT9/TXSNQsHJ+V+34mMUdC9ybpe0Yds+cKJMl5d9MKfIDE4N33cMUJAlqo6R+e8RI6h/of\nUPeZvyuhf1T0E4KPJapafZevWq/mpf6hOEji5xLfSbwpsUFC/yBUL5AOldt/kdBibatEKpv8\nFNKP3iulcvtsCwEEqgksmjbRmDFiYEVho9+z2XUCgV2nfdbv6shH2fX5D6tvOH7Q9BMRLYQ0\n9I2fhhyKpj+EWdVv1eXu9Yql4NLYvHnz7vu4goBXBEpKSoxoNLp/CvItlm1oAfOgRCcJfWO+\nSmJfTT/krflBb811wnKHFkJPSnSUaCORKZErUbNtlzt+9HegciF9j1LbY+/L/Z9KHCOxn8Ry\niTIJbcskelVc+/E/O+SuX0u0kOgosURCi6vGtEsbs3Lluu3lUgs+3UNXs+n7uKoP2dvJ9VkS\nI2oulKrbTbkHqSpHfeIeKKFPAq2K8yVo9hfQTwCulND/1NdJ6N4bbfJOo+KQu7Pk8j0JLZ4y\nJeq7vKyyx3ZC5SMLKi9jcqn/+U+rvF118bRc0T1b/6u6I2WXNT8XSdmG2RACCOxJQM4s6U87\nVPbWSNAQQKBpBbZv2Wzk5uUZcTk0NUVNN7QySdvSV/31lZHoTWgRpoVRQ9pOWWlRQ1ZM0jo9\n99GvFpH63rHJmx0KpLGioBWu7lbTPQ83S9SnHSwL3yTRTULz0TfKoySyJOrT6tJPc+lQ95SU\nSHy1h86vl/tbSfSVKKi2zBly/QKJ4yS0StbdrzrO6m+l0+X2bRJrJeZJ3C3RWmKYxHSJvbVO\n8uCFEl0ldJ11EmqhVXr1bZwtt0+SGCShhelVEsdL6J6Y0RJbJOrStMrXpnlWFUcVd8g/+kdo\nsMTfJS6X0P8Q9V1eVtljC8gjuo1ItSV0l3GzarePlOv3Srwkof/hUtvkcyCOsUstOVtDYG8C\nJ51/iaFBQwABewjMnzjGWL1gbtgeo2EUCPxQwA4Fkr55v0NC38xroTNAYqBEXdpvZSHd86QF\ngfajuw/1zbMWMI9JfCBRl1bXfoqkM92VerTEdxKrJaq3tnKjt0SOxIeVD+ib9lckHpCQt80V\nJ6XQQkjbGIlbJXQvjLYDJDQfLS4OlzhLQtsfJfQNf21N+9QxPSeh/aqB3qehbYbE5RJqou06\niUcl9pP4p4QWfbrXTosxNXxI4l2JfbUOlQus38OCaqBNC0MtkOq7vK67p6bb1HluL5FVudDR\ncrmu8rpeaHGmOb+hN1Ld5CsNcsBN1RSkeutsDwEEEEAAAXsL6ElN5K2K/kNDwHYCPzqOuolG\nqG9yH6/cdg+51IJnX62rLPClhO4FuV/iEImDJK6X0E8ktNC4WmJfrT79xKWzzys7/EMtHd8s\n9+kb9y8ktFDR9rzEgxIzJU6U0MKkncTHEr+Q0BxqtivkDt0jpus+JfEfiZhEbU2XfUkiT+JK\niZYSWlzdLrFBQg9tu02iZntd7pgmcWhlnCaXOg9aUJwvsa+WXbmAFim1tY6Vd+p4tNV3+V1r\n1f6vHnOrvm9LHCfxj8rLT+RS2zESd0jooXU1927JXclvsvcoJi35G2ILCCCAAAIIOFBAfjxY\nP0nkhTKxc9dHurtVYmpiu/Veb34bpax7GcZL6Bv29+owLi2AtNjQPUV6fadEmYTugdJCRZu+\ngQ5VXNvzP/XtR4sfLZR+X0uXf6y87/PKy85yqXtkNklcJbFEQtfVvSt3ScyT+LmEFjbVWzO5\n8bCEvvF/WWJvHjfJ49rn0xKjJNRhq8RnlSEXFadM1MvqTYsh3Zuke4+0fS9xi4TuVdQidV9t\nfuUC6lBVBFWtc6Bc+UPlDT10UVt9l9+1Vu3/quNfJa6QyJD4t4TmW+X0L7m+TaIue8JkscQ3\n+Q2kaCxa/QjAxG+DHhFAAAEEEHCqQMUeJL/fdOr4bTpufV+p71PX2HR8jhmWnQokRbtTokji\nBonfSuypBeSBsyV02Y9qWWis3LdYooOEFil7ag3pRwuLCRLHSlQdAqf9Hy9xqsR0iaonphY/\nIYmBEjskara+lXdcVvMBuT2slvtqu0uLmgMketV40Ce3MyvvO6jGY3pTi6Bojfunye3lErr3\nav8aj9W8qXvEdIyHSkyR0FwPk7hAQn2qnltVVUJ9l5cu9trekUd1T9wlEkdL3C6h+ejYtWh7\nUaJUQudYi02dgw8lzpFIRtPCUuehpUQ7qVjNaLg8GduhTwQQQAABBBwvYJoxzYECyfEz6c4E\nqt7E2iU7LT50j5A2fQPfpuLaj//RN8TpEuslqt6A11xqReUdXWs+UO12Q/vRvRXaqvaS6PU/\n6j/Sqh7T61XF2bVyXav6mnGXLiStarldt3YVU7oHpK5N95xpwaiFgBaMUyTyJD6V0KZFQs1W\n5VPz/nVyhz4v9HDAfbV7ZAHdVneJbyVyKm/r+ndIaNM9WlWtvstXrbenSzWaJJFZbYF/y/VN\nEj0r7+sll3o4oRZvV0pMlbhGor5NPRZKrJTIlNBct0toEaZ/4LU4K5bIl9gk1VHbzAzd0UVD\nAAEEEEAAgZoCugfJb/gqqqSaj3EbgaYW0E+97db0U37de3SZhBZJ10vUbIdW3lHbXpmqZQsr\nrxxcdUctlw3tR0+ioG/8b5T4q4QloXstSiT6SVS1tpVXwnKpy9Rs2sdciYIaD+ib7Lo23Ybu\nyTmjcgV9s75KYpRErsSjErW1PW1Dc9B2yK6Lvf67RR69ROI6iTMrlxwvl5Mlfl15Wwu1qlbf\n5avWq+vlqbLgbyS0EFNzva1z9IrEExJ66OJ8idckhkjUp2XLwlp8aiGkfVcP3VVU/XY4lJ4+\n8rCOx3SS+2kIIIAAAgggUEMgFpWXU98eP+SusTQ3EUitgB0LJBW4U0I/ftdC6QaJmk33cmg7\natdFrf+2r7y3+hv0mgs2tJ8y6ai3xL0SemiZ3j5S4gsJ3YtQ1TIrr7wnl29W3ZngywHSnxZH\nIyVelZgjUSqh7eZdF7WeTq1d5WM1LzQPbQt2XezzXy38+ldG9YWr9kDpeKq3+i5ffd19XX9e\nFtA5rdqLd27lClpAatNC5luJhyXaSOztuSEP/6DtkFs6j3VqPsNfHA1rzURDAAEEEEAAgZoC\n4dIS+Qa1r/p7ppqLcBuBJhPwN9mW975h/bT+scpF9E3pfjUW1z0RhRLtJfQwuZrtQLnjtMo7\nF9d8sNrtxvRT9Sb8N9Kf7qXQVnXfrlv//4NkupeltqZ7WTS/a2t7sA73tZFlzpPQgkj34kyu\nvC4XFa1b5WVth9idWflY9YuD5cYJEpsl1GZv7Rh5cLREz1oW0ufVLRJxCT0ET1t9l9+1Vt3/\n1WJIT9rwL4mYhLZWuy5+sIdOnzfa2u66SM6/cV+8PBbZ09GfydkmvSKAAAIIIOAUgXIpkOJG\nvMgp42Wc3hLQN7J2bR/JwMZKtJbQgqdme1vu0Df+erhUqMaD/5bb+uZ4okRWjcdq3mxoP7pn\nZJnE1RJa4Oiei6kS1dtAuaHfVfmVRNUhZ1WP63diPpW4X2L/qjvreZlWuXy6XB5eY91T5faf\nqz1e4+GKvV9atFRvz8mNgyR0j9C+2gZZQIuSuyVOrrHwP+S27onSk1BkSmir7/K71qr7v7r3\nKEPim2qrZFde10Kyqh1aeUXHk7QWN+Ml4bKqHXlJ2wwdI4AAAggg4EiBcGmpYZnWTkcOnkG7\nXiBo8wzvlPEtlWheyzhflPtukNDiZKZEHwk9hEoLlp9L6CFiv5XYV2tMP7rHSA9r0/auhO4x\nqd70k5G/SPSWGCTxhcQsCd2zc51ES4kxElVns5Or9Wp6MoLvJM6SGCLxtYTed76E7tXSvUAt\nJGrbW5Im98+VeEciR+KqytCi8nGJfbWoLKCF6CsSoyQ0x7USl0voHKyXqN5PfZeX1evcLpUl\nL5LQ+bYkqpruvdK9SXdLqJM66Nj0elL/KFtmNL+sWKefhgACCCCAAAI1BcrLSgzTiiX1tbjm\nNrmNQF0F7F4g6af8j0h8XEtC+vH8aRL6Bv1WiapCZbNc/0riYYkCiX21xvSjBclLEgGJL/ew\nId0bo3syPpC4VeI2CW0lEjrmZyWqDgmTq/VuWgipj57U4qTKtTVvHdfrEsslukt0lMiUqGpP\nyZULJXRvj18iT0ILuL9KaDFTl6bj13X/LqHzpE2LRM35UYmNEtVbfZevvu7err8gDy6Q0CK0\netOC7XmJf0lcLKEFqVYtD0oktVmWtb20aKcpG9HnBg0BBBBAAAEEqgmUFxebhmXxSWI1E67a\nR8Bnn6E0aiT6JrSThL451zO4NbQlqp89bf9AeaCrRLHEeolEfotf9450lNACcYPEnpoWKVq8\n3CGhh/i1kOgosUSi+t4XuVnnli5LdpbQQwVXS2iBtrdW3+X31tdR8uCTEr0lZuxhQS2ONPIl\nBkhskkh2e6nrT8/56xMffqO50hBAAAEEEECgmsBr9/8xsmzWtHfkrseq3c1VBGwhELTFKBo/\nCP2kfmXju6n4PZtE9LOnoWhhNG9PDzby/q2yvkZ9m+7eXlTflWosr4Xe0hr37e1mfZffW19a\nDD64twXksUmVsY/FEvrwztLCnQ0tOBM6EDpDAAEEEEDAbgJlugdp11Eddhsa40Gg4vAoGBBA\nIPECO0qLCxPfKz0igAACCCDgAoGiAj2ow9jmglRIwYUC+v0RGgIIJF5gZ1lRIf+/Eu9Kjwgg\ngAACLhAo3lGgr5EUSC6YSzem4JZD7Nw4N8nKqY90rN832tP3dZK1Xa/1WyC/8cD/L6/NOvki\ngAACCOxTIB6PG+UlxSFZkAJpn1os0BQCvIFrCvWm3eY82bwGLbkCW81YLKCn+t7vQP1pKRoC\nCCCAAAIIqEBpUaEhRRJ7kHg62FaAQ4BsOzUMzOECW3T8O/P17Ok0BBBAAAEEEKgSqPz+kd7k\nRbIKhUtbCVAg2Wo6GIyLBPSPfrwwn6MHXDSnpIIAAgggkACB4oLdvwbCi2QCPOki8QIUSIk3\npUcEVCAaCAaL2IPEkwEBBBBAAIEfCugeJH8gUCb3Rn74CLcQsIcABZI95oFRuFBACqQ89iC5\ncGJJCQEEEECgUQIFeVuNYDCU26hOWBmBJApQICURl669LSAn6dmycxuHV3v7WUD2CCCAAAI1\nBbZvzTHiRjyr5v3cRsAuAhRIdpkJxuE6ATMa2bBjW27cdYmREAIIIIAAAo0QkALJiobD6xrR\nBasikFQBCqSk8tK5lwUsy1qfuyGT46u9/CQgdwQQQACBHwnkbcjW18YNP3qAOxCwiQAFkk0m\ngmG4UiBr26YNliszIykEEEAAAQQaKFCwdbMeXbGxgauzGgJJF6BASjoxG/CwQKYcYpemvxhO\nQwABBBBAAIFdAnKG1zS5xh4knhC2FaBAsu3UMDAXCGRZphnYkceJelwwl6SAAAIIIJAAgaKC\n7YYZiwWkK/YgJcCTLpIjQIGUHFd6RUAFKs7Qk7+Z1wCeDggggAACCKhA3qbsKojMqitcImA3\nAQoku80I43GTQFkgFCrYlkOB5KZJJRcEEEAAgYYLbM3ONPS1UXooangvrIlAcgUokJLrS+8e\nFwj4g9nsQfL4k4D0EUAAAQR2C+RKgeQPBNbsvoMrCNhQgALJhpPCkNwjEIuGl27OXMuZ7Nwz\npWSCAAIIINAIgZzs9Va0vHxZI7pgVQSSLkCBlHRiNuBlAfktpOUbVy0Pe9mA3BFAAAEEEKgS\nyFm7Sl8TV1fd5hIBOwpQINlxVhiTmwRWbN2QGXRTQuSCAAIIIIBAQwVyN2brGewokBoKyHop\nEaBASgkzG/GwwMpIWVlIfvPBwwSkjgACCCCAgGEU79xhhEtL9DeQKJB4QthagALJ1tPD4Fwg\nIC8CPitn/VoXpEIKCCCAAAIINFwgZ33FuRn019MpkBrOyJopEKBASgEym/C0QDiYnrZ5SyYF\nkqefBSSPAAIIIGBsXL3CCKanbxKKEjgQsLMABZKdZ4exuUMgHs9gD5I7ppIsEEAAAQQaLrBx\nzUojbpoLG94DayKQGgEKpNQ4sxUPC8QikQWZGUs4k52HnwOkjgACCCBgGFkZi8vNWGwBFgjY\nXYACye4zxPjcILAge1WGzw2JkAMCCCCAAAINFdi0drW+71zS0PVZD4FUCVAgpUqa7XhZYKGe\ntWf7ls1eNiB3BBBAAAEPCxTkbjHCZaV6BjsKJA8/D5ySOgWSU2aKcTpZYLXf7w9nr8xwcg6M\nHQEEEEAAgQYL6AkafD5/VDrgDHYNVmTFVAlQIKVKmu14WcAKhEIZWSuWedmA3BFAAAEEPCyQ\nLa+BobQ0LY5MDzOQukMEKJAcMlEM09kC0XB4dtbyJfrJGQ0BBBBAAAHPCaxdsiAaCZdP81zi\nJOxIAQokR04bg3agwML1GUv41MyBE8eQEUAAAQQaL7BmyQJ9DZzT+J7oAYHkC1AgJd+YLSCg\nAvN25m1tVlSwHQ0EEEAAAQQ8JbAjL9co3p7fTJKe66nESdaxAhRIjp06Bu4wgUU+vz8ihxg4\nbNgMFwEEEEAAgcYJrF+2yPD7A+XSC2crahwla6dIgAIpRdBsxvMCsWAwbeHaxd97HgIABBBA\nAAFvCWiBFAiGFkvWHGrural3bLYUSI6dOgbuNIFopHzyyvmzw04bN+NFAAEEEECgMQJrFs6P\nyGvg9Mb0wboIpFKAAimV2mzL6wLfZWUsCVgmH6B5/YlA/ggggIBXBCzLMtYtW+iTfGd7JWfy\ndL4ABZLz55AMnCMwS85xGty4ZqVzRsxIEUAAAQQQaISA/kBspKwsJF2wB6kRjqyaWgEKpNR6\nszVvC2wJpqXnrONEDd5+FpA9Aggg4CGBVd/PMYLp6Rsl5c0eSptUHS5AgeTwCWT4zhKwzNjk\nlfO+4xg7Z00bo0UAAQQQaKDAyvnfxWLR6PgGrs5qCDSJAAVSk7CzUa8KyPePJiz9blrMq/mT\nNwIIIICAtwSWz5lpGpY1xVtZk63TBSiQnD6DjN9pApNKdu5Iz8lc67RxM14EEEAAAQTqJbAl\na51RWlSYLitNrdeKLIxAEwtQIDXxBLB5zwmsk+8hbVkxd5bnEidhBBBAAAFvCayaL98/CoXy\nJOt13sqcbJ0uQIHk9Blk/I4TsGLRsRmzp3OYneNmjgEjgAACCNRHYPncmaac5ntifdZhWQTs\nIECBZIdZYAyeEpAXiwkZs2dY8XjcU3mTLAIIIICAdwT0NW7x9MmmfPd2jHeyJlO3CFAguWUm\nycNJApPKiovSNq1d5aQxM1YEEEAAAQTqLJC9cplRVlyYJiuMq/NKLIiATQQokGwyEQzDUwIb\nQmnp2ctmTfVU0iSLAAIIIOAdgaUzpxqh9PTVkvEm72RNpm4RoEByy0ySh6MEopHw4AWTx4Ud\nNWgGiwACCCCAQB0FFk+bGI6Gw8PruDiLIWArAQokW00Hg/GQwKg1i+YHy0tLPJQyqSKAAAII\neEEgXFZqrF2yICi5fuuFfMnRfQIUSO6bUzJyhsAUIx6PydnsnDFaRokAAggggEAdBVbM+05e\n4uKmLD6tjquwGAK2EqBAstV0MBgPCYR9/uDERdMmWR7KmVQRQAABBDwgsHj6pHggENJPAMs8\nkC4pulCAAsmFk0pKzhAwY5FhCyePizpjtIwSAQQQQACBugnMGzcqEouGB9ZtaZZCwH4CFEj2\nmxNG5B2B0UUF+ekbVi33TsZkigACCCDgaoHMjCWGvrZJksNcnSjJuVqAAsnV00tyNhfIklOg\nrpKz2dl8mAwPAQQQQACBugksmDxWT++9VJbeWLc1WAoB+wlQINlvThiRhwTkFKi954wZVu6h\nlEkVAQQQQMDFAnPHjSyX17Z+Lk6R1DwgQIHkgUkmRVsLDNy8fk2z3I3Zth4kg0MAAQQQQGBf\nAnmbNhhbMtc1k+WG7GtZHkfAzgIUSHaeHcbmBYElobT07PkTRnshV3JEAAEEEHCxgB5eF0xL\n3ywpLnFxmqTmAQEKJA9MMinaWyAaCfeePWZY2N6jZHQIIIAAAgjsXWDOmOHhWCTcd+9L8SgC\n9hegQLL/HDFC9wsMzF6xLG371hz3Z0qGCCCAAAKuFNDXsHVLF6ZJcn1cmSBJeUqAAslT002y\nNhWYG0xL2/r9xG9tOjyGhQACCCCAwN4F5nw7XA6vS9PD6+bsfUkeRcD+AhRI9p8jRugBgVgk\n0mvGsAEcZueBuSZFBBBAwI0CM4cPlMPrIl+6MTdy8p4ABZL35pyM7SnwddaKpem5G7LsOTpG\nhQACCCCAwB4EtmZnGhvXrNQfh+Xwuj0YcbezBCiQnDVfjNa9AgvlbHZrZ43izKjunWIyQwAB\nBNwpoIfXyWvYOslusTszJCuvCVAgeW3Gyde2AnI2u0+mD+3HYXa2nSEGhgACCCBQm8DMEQPL\n5TXs89oe4z4EnChAgeTEWWPMbhX4Jj9nU5qcBcit+ZEXAggggIDLBLKWLzXkEDv9cdjeLkuN\ndDwsQIHk4cknddsJZMkZgGbPGjk4bruRMSAEEEAAAQRqEZAjHyx97ZKH1tTyMHch4EgBCiRH\nThuDdquAnAHoMymQomYs5tYUyQsBBBBAwCUCclidMX34wJi8dvV0SUqkgUCFAAUSTwQE7CXQ\nt6yk2Fg4ZZy9RsVoEEAAAQQQqCGwYPI4Ixout+Tu/jUe4iYCjhagQHL09DF4Fwrs9Bm+vpP6\nfx1xYW6khAACCCDgIoGpA3tHjHi8r6RU7KK0SAUBgwKJJwECNhOwLLNnxuwZITlhg81GxnAQ\nQAABBBDYJVCQu8XImDMzZFnWJ5gg4DYBCiS3zSj5uEFgRigtbd20IfqhHA0BBBBAAAH7CcwY\nNkB++yi0QUY2zX6jY0QINE6AAqlxfqyNQFIEopHIe5P69wrLJ3NJ6Z9OEUAAAQQQaKiAvjZN\n6PtlWF6rejS0D9ZDwM4CFEh2nh3G5mWBL4t3FviXzJjsZQNyRwABBBCwocCiqROMwu3b9D0k\nh9fZcH4YUuMFKJAab0gPCCRDIN/v8w2Z2PfLaDI6p08EEEAAAQQaKjC+92cRn8+nx4Fva2gf\nrIeAnQUokOw8O4zN0wKmab4te5CCuRuzPe1A8ggggAAC9hHYmrXeWK4nZzDNd+0zKkaCQGIF\nKJAS60lvCCRSYLr8OvmyCb0/jyeyU/pCAAEEEECgoQIT+30VlxMJLZX1Zze0D9ZDwO4CFEh2\nnyHG52kB+XXyV6cM6h0rLy3xtAPJI4AAAgg0vUC4rMyYOrhPTE7O8EbTj4YRIJA8AQqk5NnS\nMwKJEOhjxqLFM4YPTERf9IEAAggggECDBWaNGmxIdVQuHfRpcCesiIADBCiQHDBJDNHTAmEz\nFnv32y8/DMfjHGnn6WcCySOAAAJNKKCvQaM/+0Bfk96RYZQ14VDYNAJJF6BASjoxG0Cg0QLv\n5+dsCiydOaXRHdEBAggggAACDRFYOGW8sS1no75v1AKJhoCrBSiQXD29JOcSgRyf399/5Cc9\nIi7JhzQQQAABBBwmMPLTHhGfYfSSYW9x2NAZLgL1FqBAqjcZKyCQegE5neqLqxbMCa1fuij1\nG2eLCCCAAAKeFli3dKGxbsmCkGVZr3gaguQ9I0CB5JmpJlGHCywJBNPGDv/kXX441uETyfAR\nQAABpwmM+vT9mL4GybgznDZ2xotAQwQokBqixjoINIGAGYs8v3DyuGBO5tom2DqbRAABBBDw\nooD+WPmCyWP98hr0ohfzJ2dvClAgeXPeydqZAtODobS5oz5733Tm8Bk1AggggIDTBOS7R1Yg\nFFos4+ZMQU6bPMbbYAEKpAbTsSICqReQ35/496wRg43tW3NSv3G2iAACCCDgKQE5g6oxY2j/\nuPxo+T88lTjJel6AAsnzTwEAHCYw0h8MrPn2y48sh42b4SKAAAIIOExghOw9ktec5TLsEQ4b\nOsNFoFECFEiN4mNlBFIvIJ/k/XNiv6+sHXm5qd84W0QAAQQQ8IRAQe4WY9rgPuw98sRsk2RN\nAQqkmiLcRsD+Av19ft/aER+/y14k+88VI0QAAQQcKSDfd7X8gcAqGfxQRybAoBFohAAFUiPw\nWBWBJhLQT/SenjywV5zvIjXRDLBZBBBAwMUCeoTC5AG9qvYexV2cKqkhUKsABVKtLNyJgO0F\nhvj9gRXDP3qHM9rZfqoYIAIIIOAsgVGf9bB8fr/+psQgZ42c0SKQGAEKpMQ40gsCqRaIyxnt\nnp46uI9v2+aNqd4220MAAQQQcKmAvqZM7PuV7j16SlJk75FL55m09i5AgbR3Hx5FwM4CwwOB\n4JJhH77FXiQ7zxJjQwABBBwkMPi9V0357pH+7tFgBw2boSKQUAEKpIRy0hkCqRWQvUhPzhg2\n0Jezfk1qN8zWEEAAAQRcJ7BxzUpj1qghfnltecR1yZEQAvUQoECqBxaLImBDgW8DwcD0fm/8\nN2rDsTEkBBBAAAEHCfR/88VoMBiaJEOe7KBhM1QEEi5AgZRwUjpEILUCsWj0r4umTQys+n5O\najfM1hBAAAEEXCOweuE8Y8mMycFYLPqYa5IiQ9gF4AAALbJJREFUEQQaKECB1EA4VkPARgIL\n5Hjxvt/877mIjcbEUBBAAAEEHCTQ9/XnI4FAYIAMeYGDhs1QEUiKAAVSUljpFIHUClim+dSG\n1cuNOWNHpHbDbA0BBBBAwPEC88aPNtYvXew3TfNpxydDAggkQIACKQGIdIGADQSy45b1Zt/X\nXwjLl2ttMByGgAACCCDgBIFoJGx888pz4XjcekPGq799REPA8wIUSJ5/CgDgIoEXC/PzwhP6\nfOGilEgFAQQQQCCZAmO/+tgo2r6tVLbxfDK3Q98IOEmAAslJs8VYEdi7wA4zFntq0HuvRXdu\ny937kjyKAAIIIOB5gR15ucbQD9+OyWvHE4JR6HkQABCoFKBA4qmAgLsEPpRD7VbLab/58Vh3\nzSvZIIAAAgkXGPD2SzEjHl8pHX+a8M7pEAEHC1AgOXjyGDoCtQiY8h2ke/SH/tYsml/Lw9yF\nAAIIIICAYaxfttiYOWJQQF4z7hcPCxMEEPh/AQqk/7fgGgJuEZgup/3u8+XzfwtbFq95bplU\n8kAAAQQSJaCvDfIaEQkEg0Okz6mJ6pd+EHCLAAWSW2aSPBCoJiCn/X508/o11tRBvavdy1UE\nEEAAAQQMY8rAbwz5aQhTvnv0EB4IIPBjAQqkH5twDwJuEMiRIunZfm++GCneUeCGfMgBAQQQ\nQCABAjvz84x+b7wQldeIZ6S7jQnoki4QcJ0ABZLrppSEENgt8KYcW57V+5XnOGHDbhKuIIAA\nAt4W6PPqf+SsdeZqUXjb2xJkj8CeBSiQ9mzDIwg4XSAai0T+JCds8C37brrTc2H8CCCAAAKN\nFMiYPcOYPWaYXz48u1264sOzRnqyunsFKJDcO7dkhoAKzPL7/T0/fe6JcKS8HBEEEEAAAY8K\nRCNh4/N/PxmW14SPhWC2RxlIG4E6CQTqtBQLIYCAYwXi8fi0aHn53fLiuP/xZ53vc2wiDBwB\nBBBAoMECQz94K75kxuRC+e7Rr6QTPjFrsCQrekGAPUhemGVy9LpAofw40t3ffvlRPHtlhtct\nyB8BBBDwnED2ymXGyE/ei8tZ6+6W5Hd4DoCEEainAAVSPcFYHAGHCgz1B/wjPn32sbB8eujQ\nFBg2AggggEB9BWLRqPHh3/6ih9bpbx4NrO/6LI+AFwU4xM6Ls07OnhSIW9aUoh0F98sPA6Z3\nOfUMTxqQNAIIIOA1geEfvxNfMGlsifzo0eWSe6nX8idfBBoiwB6khqixDgLOFJDfRordO+T9\nN8wNq5Y7MwNGjQACCCBQZ4GNq1cYwz58W/70x+6RlfLqvCILIuBxAQokjz8BSN9zAt/4fL7h\nPZ/+c1gPu6AhgAACCLhTQIoiPbQuIofWjZIM+7ozS7JCIDkCHGKXHFd6RcC2AvG4NbG0qPBe\nefFs1u2MczmrnW1nioEhgAACDRcY/tHb8fkTvi2V3Uc/l15KGt4TayLgPQH2IHlvzskYgTwp\njm4f9WmP+LqlC9FAAAEEEHCZwNolC4xhPd+KS3F0h6S21WXpkQ4CSRegQEo6MRtAwJYCQ3x+\nf++eTz8UDpeV2XKADAoBBBBAoP4C5aUlRo8nHgjL3/hesvaA+vfAGgggwCF2PAcQ8KiA/IDs\nxHBp2e07tm3d/5QLL+PDEo8+D0gbAQTcJfDFf5421y5ZuMWKxX4pmUXclR3ZIJAaAd4UpcaZ\nrSBgR4EdZixy/fQh/Yy54/Q7vDQEEEAAAScLzBs/2pg5YpBhRiM3SB7FTs6FsSPQlALsQWpK\nfbaNQNMLbJAhxJdMn3jeWVdcE9j/oOZNPyJGgAACCCBQb4GC3C3Ga/f+IRqLRp6XlfXwOhoC\nCDRQgD1IDYRjNQRcJPCCnLRh7gdPPhCxTNNFaZEKAggg4A0B/dv9wZMPRqQ4WiQZa4FEQwCB\nRgiwB6kReKyKgEsE4pZlfVu4Pf8eM2amHXfGOZz62yUTSxoIIOANgUHvvRafO3ZkqXzYdbFk\nXOCNrMkSgeQJsAcpebb0jICTBDbJC+stIz55N7587kwnjZuxIoAAAp4WWDJjsjHysx5x04z9\nXiCyPI1B8ggkSIA9SAmCpBsEXCCw0vD7Wy2cNPaUs6+8NrDfAQe6ICVSQAABBNwrsH1rjvHK\n3b+LxCLh1yXLHu7NlMwQSK0Ae5BS683WELC3gGU9FikvW/juI3fLsexRe4+V0SGAAAIeFpC9\n/sZ7j94TiYbLFwjDMx6mIHUEEi7AHqSEk9IhAo4WkK8jWaOKCvLvLNmxI/3E8y7m+0iOnk4G\njwACbhXo/+aL1oLJ44qlULpIctzp1jzJC4GmEKBAagp1tomAvQWK5KwNszMzFt/Stv3R/iM7\nd7X3aBkdAggg4DGBOWNHGH1ffyEuf6t/Lakv9Fj6pItA0gUokJJOzAYQcKSAftG3dNHUiZec\nesnl/uYtWzkyCQaNAAIIuE0ge2WG8eafb43Jqb31sLov3ZYf+SBgBwEOn7HDLDAGBGwqEAgE\nhxxy6GG/eLbPyPQDmrew6SgZFgIIIOANgeIdBcY/b/xFpCh/21DTNG/wRtZkiUDqBThJQ+rN\n2SICjhGQ08b+Yce23Mx3H7knol8IpiGAAAIINI2A/g1+V07KUFxQsEaKo1ubZhRsFQFvCHCI\nnTfmmSwRaKhAxLLMETvyttxRuH1b6KTzL+FDlYZKsh4CCCDQCIE+r/7b+n7StyVmLHq+dLOt\nEV2xKgII7EOAAmkfQDyMAALGDvki8Mys5Uv/JN9F8h99/ImQIIAAAgikUGDa0H7G4Pde1bOM\n/lI2uyiFm2ZTCHhSgALJk9NO0gjUWyDbMOKb5Rfbr+xy6k/9bY44qt4dsAICCCCAQP0Fls+d\nafR4/H4zHo8/JGv3r38PrIEAAvUVoECqrxjLI+BdgQWGz9fy+/GjTzv9sisDB7Y42LsSZI4A\nAgikQCBn/Rrjf3fdHI1FI+/K5p5PwSbZBAIIiABnseNpgAAC9RGQE9uljWrRps1Fz/YannbQ\nIS3rsy7LIoAAAgjUUUC+92n86+arIkXbt42VEzTo7x1ZdVyVxRBAoJECfOG6kYCsjoDHBOTE\ndpHrCrflrXz9/lsikfJyj6VPuggggEDyBfRv6xsP3CpnrMvPkOLoRtkixVHy2dkCArsFKJB2\nU3AFAQTqKFAsh3tctnHNqrwej98XlR8rrONqLIYAAgggsC8B+a6R0fNvD8Xkb+y2WDT6C1m+\ndF/r8DgCCCRWgO8gJdaT3hDwikBJ3DJH5m3aeNvO/LzgSRf8jA9bvDLz5IkAAkkV+OZ//7K+\nGz203DJjF8qG1id1Y3SOAAK1ClAg1crCnQggUAeB/Hjcmpq1YtktwVAo0OWUn9ZhFRZBAAEE\nENiTwLAP346P+fLDmPy0gu45mren5bgfAQSSK0CBlFxfekfA7QIbjHg8Q05De33Lw9r5OnQ9\n3u35kh8CCCCQFIFJ/b82+r/5X0sOsfutbGBsUjZCpwggUCcBCqQ6MbEQAgjsRWC5PLZl0ZTx\nV7Y7povviGM772VRHkIAAQQQqCkwd9xI45NnH7fkA6c75bG+NR/nNgIIpFaAAim13mwNAbcK\nzJfEyr6fMOaSo7uf5G/bvqNb8yQvBBBAIKECy76bbrz7yD3yQ7DWU9Jxj4R2TmcIINAgAQqk\nBrGxEgII1CIwUw4NCc0dO/Kcrqef5W91+BG1LMJdCCCAAAJVAmsWzTdef+BPMTMWfUPue67q\nfi4RQKBpBSiQmtafrSPgNoFJccM4dM6YYad0P+fCwMFt2rotP/JBAAEEEiKwfuki45V7fhc1\no9HP5cOlhxLSKZ0ggEBCBCiQEsJIJwggsFsgHh8TN+KdZ48Zdpyc/jvQvGXr3Q9xBQEEEEDA\nMLJWLDVevvOmaCwS6WVZ5l2YIICAvQQokOw1H4wGAVcIyKehw+KmecKskYM7nXT+JYHmrSiS\nXDGxJIEAAo0W2LBqufHSHTdGo+HyflIc3SYdyo53GgII2EmAAslOs8FYEHCPgNRI8UEVRdKo\nIRRJ7plXMkEAgUYIbFq72njx9t9GouVlQy3T/IN0ZTWiO1ZFAIEkCVAgJQmWbhFAQE5Yu6tI\n6j5r1JDOJ553caBFqzawIIAAAp4U2LxuV3FUXloyUoqjmwTB9CQESSPgAAEKJAdMEkNEwMEC\nFEkOnjyGjgACiRHIXpkhe46uj0RKS0dLcXSD9BpLTM/0ggACyRCgQEqGKn0igEB1gaoi6Xjd\nk9T9nAsCB7c+tPrjXEcAAQRcK7Bu6UI5IcON0Ui4fJAURzdLohRHrp1tEnOLAAWSW2aSPBCw\nt4AWSYPlzcFPZg4f1LXLaWcG+J0ke08Yo0MAgcYLrPp+TsWpvOVsdV9ZlnWr9MhhdY1npQcE\nki5AgZR0YjaAAAKVAhVFUtyyDpez253SsduJ/rbtO4KDAAIIuFJg2axpxhsP/ikmraf83btP\nkuRsda6caZJyowAFkhtnlZwQsLGAVEkj5ZPUdPmdpHPbHdPJf8SxXWw8WoaGAAII1F9gwaSx\nxtsP32WasdhrcrqaR+vfA2sggEBTClAgNaU+20bAuwIT5U1Dybzxoy89pM1hvg7HdfeuBJkj\ngICrBKYM6m189PeHLdlr9KwkpkFDAAGHCVAgOWzCGC4CLhKYJblsXjhtwlWh9Ga+zief7qLU\nSAUBBLwoMLTnm/F+b/zXkj3l90j+b3nRgJwRcIMABZIbZpEcEHCuwPcy9Izlc2deW5ifZ5xw\n7kV+n8/n3GwYOQIIeFJADhs2vnzhGXNsr09iUhxdJwh9PAlB0gi4RIACySUTSRoIOFggQw63\nmyS/E3J95rJF/lMu+nkgGAo5OB2GjgACXhKIhsuNHo/fF5s7dmRp3DIvk9wneCl/ckXAjQJ8\nVOvGWSUnBJwp8JNAMG3CkZ26tHn4vS/Smrds5cwsGDUCCHhGoHhHgfHmQ7dFspZnbDdjkUsk\n8eWeSZ5EEXCxAAWSiyeX1BBwoEDbYCjt24Nate76xAe90tt2ONqBKTBkBBDwgkDO+jXGq/ff\nEi7K37Y6Fo38QnLe5IW8yREBLwhwiJ0XZpkcEXCOQIllmV9Fy8vOnDa0f4djTzw10Lrdkc4Z\nPSNFAAFPCCz7brr8AOzN0bLionFmLKrFUb4nEidJBDwiQIHkkYkmTQQcJBCV0+P2jpmx1jNH\nDDy9RavWfvlRWQcNn6EigICbBSYP6GX0fOohKxaNviV/q26XXCNuzpfcEPCiAAWSF2ednBGw\nv0BcTtwwWmLz4umTrizM3xbvfvYFfmn2HzkjRAABVwromer6vvYfa3CP16tO4/2yJBp3ZbIk\nhYDHBfgOksefAKSPgAMEzg8Eg8M6nXTaAQ+81jN0YIuDHTBkhogAAm4SKN65w3j/iQeiq+bP\nLjPN2DWS2yQ35UcuCCDwQwEKpB96cAsBBOwp0EFO3jBaTt5wzCPvfpF+xLGd7TlKRoUAAq4T\nkJ8gkDPV3R4pLti+Xk7GcKUkuNZ1SZIQAgj8QIBD7H7AwQ0EELCpwE45ecPncvKGk6cM6nNs\n2/Yd/Ud0+olNh8qwEEDALQLfjRpivPWX22Ph0pIhZix2heSV65bcyAMBBPYsQIG0ZxseQQAB\newnoyRv6WqYZnj9hzMX6+yPdzjzP5w/wZ8xe08RoEHC+gBRDFd83GvD2y/LZjPlkPB5/RLKK\nOj8zMkAAgboIcIhdXZRYBgEE7CZwkXwvaeBRXbod+ODrPdNatj3cbuNjPAgg4FCBHXm5Ro/H\n74usX7aoVAql30gafN/IoXPJsBFoqAAFUkPlWA8BBJpa4PBgKDQ4lN7slAde/SCt25nnNvV4\n2D4CCDhcYOmsqcYHTz4YiZSXLZXTeOvJGDY4PCWGjwACDRDg2JQGoLEKAgjYQqBYTrv7eSwS\nOXjmyMFnxq240eXUM3w+TgVui8lhEAg4SUAPqRv07ivxL194xohGwm/J35abZfwFTsqBsSKA\nQOIE2IOUOEt6QgCBphO4NhAIft6hW/dm9738Xlqrw49oupGwZQQQcJTA9i2bKw6py1qxrEwK\npd/J4Ec5KgEGiwACCRdgD1LCSekQAQSaQGBFPG59XVSQf+7kAV8f1rb90YEjju3SBMNgkwgg\n4CSBBZPHGa/e8/toQe6W+VIcXSRj/95J42esCCCQHAEKpOS40isCCKReoFAPuZM3Odb88aMv\n3J6zOS5nufPL95RSPxK2iAACthYIl5UaX7/4D7P/Wy8Z8ttGL8nfjj/JgHfaetAMDgEEUibA\nIXYpo2ZDCCCQQoGzA8G0AS3btm1178vvpR99/Ikp3DSbQgABOwusWTTfeP+pP4cLt23bZsYi\nN8lYp9t5vIwNAQRSL8AepNSbs0UEEEi+wMa4ZX5SXlrSaeqg3t30C9idT/kpv5mUfHe2gIBt\nBeSsdMbgHq/HP/vX4/FwaWkvy4xdJYNdY9sBMzAEEGgyAfYgNRk9G0YAgRQJXCu/mfRJ2/Yd\n97/7v2+nt/9JtxRtls0ggIBdBDatXW30fOrBcE7mWj0Rw60yrqF2GRvjQAAB+wmwB8l+c8KI\nEEAgsQIr4pb1aWlhYdfJA3t18UvrdNJphlwkdiv0hgACthPQvcejPu1hfPDUg2bxzoLxcvtS\nGSQnYrDdTDEgBOwlwB4ke80Ho0EAgeQK3OQPBHsecWznZnc9/0bakZ27Jndr9I4AAk0mkLV8\nqfHR3x8Ob81eH5bC6AEZyNdNNhg2jAACjhJgD5KjpovBIoBAIwWWyunAPy/ZueP4Sf2+7hQp\nLzM6n3y6Tw7Ba2S3rI4AAnYRiIbLjYHyo6+fPvtYvLRw51DTjP1CxjbLLuNjHAggYH8B9iDZ\nf44YIQIIJEdAvpsU+qB56zYtbn/2f+nHn3VecrZCrwggkDKBlfNnG5/889Gw/K5Roew1ulM2\nPCxlG2dDCCDgGgEKJNdMJYkggEADBJrLl5FeNCzr3rOu+LV18+PPBg86pGUDumEVBBBoSoHC\n7duMfq+/YM4cOdgn3y/8TH7X6FEZD79r1JSTwrYRcLAABZKDJ4+hI4BAwgTOCobSPg+mpR19\n0yN/Tzvvmhs4iUPCaOkIgeQJSCFkTOr/tTHgrRejZsxcKz/6qnuNZiRvi/SMAAJeEKBA8sIs\nkyMCCNRFICQLPeIPBP7V7pjO/lueeSFNz3ZHQwABewqsXbLA+OLfT4U3r19jWqb5jIzyHQnT\nnqNlVAgg4CQBCiQnzRZjRQCBVAi0kyLpNXnDdePZv7zGuuHhvwVatD40FdtlGwggUAeBwu35\nxsB3XjanDekn/1UD/UzTfFhWy6nDqiyCAAII1EmAAqlOTCyEAAIeFDhPDrv7wOf3dbn2vkdD\nl/7uNiMY0p1MNAQQaAqBaCRsjOv1qTG051tRORvl+lgkco+MY3JTjIVtIoCAuwUokNw9v2SH\nAAKNE9Bfk71LTgP+v4PbtE2/4eFn0n962S8b1yNrI4BAvQXmfDvc6P3a8+HigvxSOTvd09LB\nxxIcTldvSVZAAIG6CFAg1UWJZRBAwOsChwjA3+T0WH9p3+U4PdtdepdTz/C6CfkjkHSBNYvm\nG71efjaSvTLDiFvW67LB/0oUJX3DbAABBDwtQIHk6ekneQQQqKdAB/nSw4v6/aQTz7vYlO8n\nheSEDvXsgsURQGBfApvXrdYfe40tmDRWvmYU6C/fM3pC1sna13o8jgACCCRCgAIpEYr0gQAC\nXhM4ORAKvWHFYhec9+vr41ff/ZdAq8OP8JoB+SKQcIG8TRuMIT1eM2eNGuKX7/xNi0Wjj8tG\n5iR8Q3SIAAII7EWAAmkvODyEAAII7EPg5/LbSa9aMbPb+b+5yferOx/0t2x7+D5W4WEEEKgp\nUJC71Rj+0dvmlEG9jUAguFB+z0gLo0k1l+M2AgggkAoBCqRUKLMNBBBwu8DVcsa7Fy3L/MlF\n1/3ed+UdD/gPObSt23MmPwQaLbBzW64x6vOe8Yl9vzR9Pt9qKYz0ULoRje6YDhBAAIFGCFAg\nNQKPVRFAAIFqAvr39BrZo/Rf+TJ554uv/6PvilvvpVCqBsRVBKoEtm3eaIz+/ANr6uA+ccPv\nX29GIn+Xx/pJxKuW4RIBBBBoKgEKpKaSZ7sIIOBWAf27ep0USi/IoXfHnvOr3xhSKAUO73is\nW/MlLwTqLLAla50x8pP3zJkjB+uhdMtlj9GzsvJgCQqjOiuyIAIIJFuAAinZwvSPAAJeFdC/\nr1fJoXf/kDeBp5980WWxq+54IHRM95O96kHeHhbIWrFUCqMesXnjRwXk/8S8ysJotIdJSB0B\nBGwsQIFk48lhaAgg4BqB8wLBtGfMWOTyziefHr3qzgfTup9zoSHfuXBNgiSCQE0By7KMRVMn\nGKO/+CCyZuH8NDkr3SQ5K91zstyUmstyGwEEELCTAK/OdpoNxoIAAm4XOF5+R+lp+Y7STa3b\nHRW7/I93pp/zq+uMZvsf4Pa8yc9DAuGyUmP6sAHGmC96hrdvzfHLG41vpFh6TQiWeIiBVBFA\nwMECFEgOnjyGjgACjhVoJyO/T441eiAQDB54wW9uCl56062+Q4/q4NiEGDgC+htGk/p/HZ/U\n76uYHEJXYsZib4tKD4mt6CCAAAJOEqBActJsMVYEEHCbQLokdGMoLe2xaCTSvfvZF8Qu/d1t\nIT38zu/3uy1X8nGhgGWaFYfRTej7RSRj9oyQPJdXynP5VUm1l0S5C1MmJQQQ8IAABZIHJpkU\nEUDAEQLnBAKBv1pW/NqDWrYyL/rt79LP//UNRqvDj3DE4BmktwQKcrcYUwf1MSb2+ypcvLNA\nvk7n6yfF0vuiMNNbEmSLAAJuFKBAcuOskhMCCDhZoI0M/hb5JP4++ST+mG5nnBu96Po/pJ18\n4aWGfMndyXkxdocLyGFzFXuLpg3pF10yY7IcIRrKkufoO5LWFxLbHZ4ew0cAAQR2C1Ag7abg\nCgIIIGA7gXP9/sDdMqobmx1wgE9O6BA6+5fX+o4+/kTbDZQBuVdg3dKFxoxhAyz57aJYtLw8\nbviMgbK36GPJeJJ7syYzBBDwsgAFkpdnn9wRQMApAs1loDfLXqXb5RP7n7Zud2T43Kuvb3bW\nFb822rbv6JQcGKeDBLZt3mjMHj3UmDq4T7mcfCFdfrtoluxB0qJogESRg1JhqAgggEC9BSiQ\n6k3GCggggECTCnSUrUuxlC7FUrhT+67Hh8+7+vr00352hXHIoW2bdGBs3NkC+TmbjLnjRhrf\njR4azl6xLF2eY3IIXViLoq8kspydHaNHAAEE6i5AgVR3K5ZEAAEE7CZwkgzo94FQ6E9mNNqm\nw3HdI2defnX6qZdcbnDKcLtNlT3Ho0XRvPGjjFmjhlQURcG09E2xSPhrGW1/ifn2HDWjQgAB\nBJIrQIGUXF96RwABBFIhoH/Lz5D4jXzqf5N86t/+sI7HlEux1Oy0n/3COLJz11SMgW04QCAe\njxtZy5dWnGxh/oTR5RvXrGxGUeSAiWOICCCQUgEKpJRyszEEEEAgJQLdZStaLN0sxVLX5q3a\nhOUseKGTzrvYf9yZ5xrN9j8gJYNgI/YQCJeVGcvnzDAWTBlvLZj4bVROy62Hz2XIc0O/TzRM\ngj1F9pgqRoEAAjYRoECyyUQwDAQQQCBJAkdLv1cEgmlXx63YRYbPF+x04qmxky+8LP2E8y4y\njji2S5I2S7dNJaB7ibJXLjPkh1uNJdMnR1YvnBuwLMv0+4OTzFhksIxrhMSmphof20UAAQTs\nLkCBZPcZYnwIIIBA4gTSpasLJH4pexCukT0IHfc/qHn4uDPODXQ789xg19PPMg4/ulPitkZP\nKRPI3ZgtBdF0Y9msqbFl302zyktK0kLp6Wuj4fBIGcQ4iQkSZSkbEBtCAAEEHCxAgeTgyWPo\nCCCAQCMFOsj6F/v9/kv8gdDPY9Fw2wOatwh3O/O8wHFnnBPsdNJpRjvZwySPN3IzrJ5IAfkN\nImPD6uXG6gXzjFXfz4mtmD/bLC7Il1Nxh/LksTGyt0gLovESOYncLn0hgAACXhGgQPLKTJMn\nAgggsG8BPRyvqmC6VAsm2QsR7djtRKvLqWekH3vCKcYxJ5xsNG/Zet89sUTCBAq3bzMyM5YY\n+oOtK+fNDq9bsiAge/+CMjcb5OyFk6QgmiYb01iZsI3SEQIIIOBhAQokD08+qSOAAAL7EGgn\nj5+lEUprdoF8f+VkeTOe3qJN2/Jju58U7HDcCcGjuhxntP9JN6PlYboorbECO7flGplyljkt\niNYvWxRdt2yRWbw9v5nP74/IHqJlcsjcJNnGdIkZErmN3R7rI4AAAgj8WIAC6ccm3IMAAggg\nULtAUO4+QeJMiZNlD8aZsgfjOC2a0vfbP3pUl27m0cef2KzdsZ2NwzocI99nOpa9TbU7GoXb\n843N61Ybm9euqrjMXpFRvmndKl9ZcVF6ZTGUIcXQTFldzzCnsUwiJkFDAAEEEEiyAAVSkoHp\nHgEEEHC5gH5BqbOE/mjtybKX43T50tLxsXD4MLntT9tvv2jbozrGjux8XPrhHY/x6w/Ytjr8\nCIkjjRat28hJ9dz7MqR7g/I2bjByN2XLZbaRuyHL2pK5JpKTuc5fXlKcZhg+K5SetlG+N7TY\njMUWi5cWQUskMiRMCRoCCCCAQBMIuPeVqQkw2SQCCCCAwG4BPWOeFk4/0ZATPRwnX5o50TRj\nHcxYtIUu5Q8EzINbHxppfcRRvjZHtE9vdXg730Hy/Sb9jlOLVnKp0bKVsf9BFYvrKrZoZSXF\nRsnOAmNHXm5lbJXLrUZB3lYrf/PGaMHWHKsgd0soFo3qHre45J3vC/gyo5HIcsOy1sp9qyS0\nGNLvDIUlaAgggAACNhKgQLLRZDAUBBBAwCMC+0ue7SX0LHoVIQVUx0Ba2tG+uHGYFFGtZY/K\ngfJYxWuUPxA092/ePLr/AQfFmx14kCFn2vMf0Lx5UAqnwH5yu9kBBxpyuJ8RCqUZgVDICKXp\nZVrFZTAYkl6kG/ltICtuGXErLlc1rIr7pIgxIuFyI1JeZkTkB1V3Xd91u7y0xCranh8r2lFg\nakFUWlTol/tCccvafVq/QDBYJLFV+twoh8Rlypg3V0aWXK6TWC/B6bUFgYYAAgg4RYACySkz\nxTgRQAABbwno3pdDJdpWXraRS92VpNFcL6WoauEPhlr5/b5DpAhKl6pHwkiT0iddqqGQFEFp\nUrhIP3E5kk9KL7kiIQ/LdV/F9bh83yfqN3zlcrtUFiuV0qlU1is2TbM4bpqFsvx2iYLKS71e\ndTtfrm+RKJegIYAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCRE4P8ANu3jo+3D3uwAAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title “Genes with TCGA DMR overlap”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "w_overlap = nrow(df_tcga_ol[df_tcga_ol$bp.tcga.overlap > 0,])\n",
    "w_overlap_pct = round(nrow(df_tcga_ol[df_tcga_ol$bp.tcga.overlap > 0,])/nrow(df_tcga_ol) * 100)\n",
    "no_overlap_pct = round(((nrow(df_tcga_ol)-w_overlap)/nrow(df_tcga_ol)*100))\n",
    "\n",
    "pie(c(w_overlap, nrow(df_tcga_ol)-w_overlap), \n",
    "    labels=paste(c(\"overlap\", \"No overlap\"),c(w_overlap_pct, no_overlap_pct),c(\"%\",\"%\")),\n",
    "   cex=1.1, main=\"Genes with TCGA DMR overlap\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8952f94e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 830,
   "id": "25229300",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAYAAAD958/bAAAEDmlDQ1BrQ0dDb2xvclNwYWNl\nR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRB\nkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4\na73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PC\nv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UA\nVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXd\na8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8\nHOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojL\njVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0\nyDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5Pt\nXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEw\nQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXH\nliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vW\nc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUt\nVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJf\ncl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdd\nuwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqv\ngcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCg\nKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8A\nrD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvF\nY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAANIoAMA\nBAAAAAEAAANIAAAAAN/ryxkAAEAASURBVHgB7N0FnFTl/sfxM7GzS5ekdDcKSkiIqIiIhKDY\n3XqvXjv+drfYXq/dYAs2YrciYqCAUop0CMLWzP/7gzne49zZZWN2d+LzvF6/Pf2c53mf2d3z\nm3PmjONQEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQTiCfjizWQeAgggkACBoOrYRdFS0VCxRrFI8aPiVwWlYgRqqtq2nqq/8Yxva7S9\nVqgeXWmZhhblLTVUQbtSVvK71l9ezDb22uqo6KWop/ghGrZdvFJLM9vELCjQ9Pcx89zJrhqx\nfXjLL5r4wzujiPFYf+9q+ZrYqLB25nkXxIwnur3bOgbWlg2K1Yo/Y9qSaZOxx680vz+ZZkV/\nEUAAAQQQQKCEAjla7zrFSkUkToQ17znFTgpK4gWGq0qveyDOLrpp3mlx5n/q2fbSOMvLMmuo\nNvK2pyTjFxaxI0ve7lJsLqLOVzXfmxxqcksZpZ+x+7XX4XZbF//tZ+M469q2I/+2VtETuxex\nvXf/lpC8qbBELF5JdHsHayfe/Rc1biaWNB4cr1EZMq8kvz8ZQkE3EchcAX/mdp2eI4BABQh0\nUp2fK85RNCiifrtyPU7xkWJEEeswu2IE7GrLbYqvFbtWzC4qrNa+0XafpGF2EXux19N3ioOK\nWO6dba/DYd4Z0XFLcCq6ZGkHeyjsOFyrKMndHJXRXttHF8XjimtK2C6tRkEAAQTSSyD2FoL0\n6h29QQCByhSorZ1NVbT37NSuIn2isNvq7HaoIYqQwoqdJD6r6K2w5ZTECGxSNT8XUdVNmn9U\nEcsqa7ZdpVq3jZ39ErO8vqbt6pAN3TJTI1aXJX37KGoqrOQo7ld8o7BkqbhiScrkmBUSnSBZ\nG9cr7P+tta21oqnCiv0OnKuYq7A2b6skqr2faUdrozuzN0rtKuP2io7ReTY4T/GCwtpPQQAB\nBBBAAAEEECiDwMPaxnvrziOarhNTj53EPhGz3q0x6zBZcQIPeuwtOY0tdjLsHsNLYxeWcXqo\ntnPrtKFdCSptuUMbuHXY7WknxVRgicctnnVsXXs9usV7y5rdnme3ktk68RLJBdFlhRp6b+Ur\n6y12fVSPt/g0cYDiV4Xbp2Uar6twS6LbG3uL3QB3RzHDozTttsmGF8csz4TJ4TEGljxSEEAg\nwwTsHS0KAgggUF4B+9yG93MLMzR9eJxKN2iendwOUjRXzFdUU8Qr9m62XRnYUWEntLMUVu/X\nitjSVjPGRmcu0dCuCrRS7K0YqPhT8YXiUYWd9MYrpdmfu731Y4SipcJOcJcrflbYLUoLFSUp\nh2kle4iFFbtK8sOWsa0/rA/jo9MRDScpzMItdtLeOTphLm8rvNvYIksc7OreMQr77JFbOmjk\nDMUmxd3uzDhDO1kfomitsGNgVwltWFmli3Z0omdn92k8tr12TC9QjFG0UaxU2NUZn8LcvMUe\ntLBU0UNh69prx46ZlfYK87NiV6BaKoq6nc/WKUux9tjrM1fxQrSCRhqeqbgoOu0dVGZ7H9SO\nL1fY76aVXbcOSvWzmda2vwXdFfa6m6Mwy5cU9nvoFnN2X9s272HFKhuJKXbsq0fnva+h3cLr\nFkte9lPspLBjN09hvwfmaq/r2OKt6wEttN+HQxUFilcVbyi2VWyf+yqGKqyvOYo1itkKO66L\nFN7SVRMjojPs792LCndef43bvLcVbyooCCCAAAIIIJBGAmeoL3bi50afbfTNbjGqVcw6p2iZ\nneC49bnDQs2zEzg7SfEWO4l313lX43sq1nnmucvsRK2FIraUdn928m0nOm69sUNrp/fkL3Z/\n3ul7PPXc7l2g8XM9y2wfvWOW28mVu287KbUS7x3wtprvrhc79J6UfupZ7waNP+WZdrezE/tj\nFSUtQ7Wiu60N+5Z0w+h6dmzc7cMaj3f83CrtxNPCEiNv8b4+VmjBJIVb5/GeFU/wzLfE0mzc\n9UZ61itudHfPNrZtcb8LX3jWneapNNHtHezZj7WpqCtIfi1b7Vl3isZLUw7RypZsuGbe4U+a\nP8hTWQ2Ne39HT/Msc0d7acRbRw93gYatFB/ELHfXtVsrvetqcktZrp/uOidpPN8zvUzjQUW8\n3x/N3lK2089ZCreO2KHZDdmy5n9/HO5Z/yWNH6mIZ/Rvzbdki4IAAggggAACaSLwuPrhniz8\nUc4+Heepy+rMU6yMmXejpr3Fe0JpJynuic9CjdsJvds2G9qJiLeUZX/XqQK3Tqv/C4Wd4Hrb\naeNNFdsqe2kFt655MSvbu8ruMhue6Vne2bPM2mDv1luJd4JXlgTJPYnbqDrN0dsOu2JjVz1K\nUoZqJe+2d2jaEr+i4hgt85ZbNeFubyexZSne14clSOMUbp2TPRXauDt/tMYrOkEyC3d/P3ra\nkej2liRBqqf9W4LutseG//S0aVujB2gF9zXj1lGoee64DW16mMItd2nEXf6lO9MzvNmz/GPP\nfLuqN9+zzOr4XWEJtFvfBo03VniLN0H6UwvcdW1ofbcS7/dn65KtV5m82yzSAkv8YvfrTXS8\nCZK9ntx17aqTO+7WeZPmURBAAAEEEEAgTQTsqo37T97eYY0tdvLet5iwd5Ot1FF4T2Ie1vT2\nCjshsqsW7j7s5MabfHhPKG2dpYoeCiu279kKd9tfbWa0lHV/dlLk1jfQrUxDewf+fYUlEF8o\njlRsq2RphbUKt74O0Q2sz7EncVOjy2xgyZK7jffqQ7wTvJDWNY8XPdu8FZ3XVUO3fKoRt04b\nXqOwdlgZo/Ce0B2/Ze62fwzVKt46tzX+fUyV1jd3my9jlpV00vv6sASpvsLty0qN23HzKWzc\n9mUn8nUVq6LTNm+koiRld63ktteGfYrZ6GLPupbkBqLrJrq9gz37sTatVtjvmYWNWxLsbbON\nf6ew101JygCt5L4pYdu+puiosNe27dv7+2KJTXWFlV4K7367b5m79YddzbGE2F1+tGfZBZ75\ndjxHK+wY2psG3r9Fllx7i/XXrc+GkxX2ur5R4V6dHa5x7zruMWmh+eujy+y1s5fCLVaHd5v+\n7gIND49ZtljT7vLtNG5W7rb2GmiroCCAAAIIIIBAGgh8oz64/+Tfj9OfYZ7l7nreoSVPVo5R\nuPMLNd7EZnrK5xp3l1/ume89obTldoXAW47WhLtdnmdBWfe3yFPfDI3vp7ATaiuW7NnJXWnK\n41rZbd8/oxvu5pnnLluneYHo8ume5dYPtxR1gmfLH1S4dT3rbuAZehOkbzXf71lmoz8q3O0v\ni1lW1ORQzzbutsUNYxMkb3JrfS5L8b4+7ITaytcKtx12cryjZ9pNxCo6QTrHs09rS1OFlUS3\nd7DqdPu6raH93lniUE9R0uJ9/S7QRm5S7W7fRSPeq0vHuws0/Fjhtul6z/x9PfPtdW+/V25Z\nrBF3m7vcmdHhPp5lmzRe3bN8uWfZQo17r/S4qxX3++PTSu0U5ukt1l9vkjnSszA2QRrvWWaj\nliRtULj9OdtmUhBAoGoFSvtPvGpby94RQCBZBez2lh7RxrUqRyM7erZdoHE7cfUWeyd6p+iM\n9t4FMeMfxEwv9Uzbu9p20vSnoqz7sxP1IxVWhkbD3lX+SjFN8aJipqKk5XmteHB05b01vE2x\ne3TaBna1Zw9FbYVdkZijcE/S7ITW9pfoYlcQrE/eYgmDW+KdXLrLihverIULillhTcyyXzXd\nPTrPTSBiVinT5Dvaqld0S7P19tWWVUbxnrxbArGymJ2+o2WJau/PqstO6u0NiIDCLa9r5FBF\nce1w1/UOLZl3yyMayXUnosMfNPxQMSQ67fbDJu9V9I/OP0TD8xX2mj4iOs8GloBZAmLFzOyq\nslusrd6EJKhpu5plv+f2Gm2usL8bseVlzdgcO3Mb05bEzFfYGyTW5gGKfgp7/XiPpdnGK7a9\n7ddbrP1mY4mZlfZbB/xEAIGqFLA/JBQEEECgvAKLPRXYyUtI4b1SY+/aP+9Zx07M7OQitnTw\nzGircUs2iirek6TYddbGzCjqRKis+ztN9fdUeBM4v6YtebO4RGFtP17xm2Jb5TWtYG20E7qh\nimoKO+myYu+eX6Nwp+1k1E767ATQyvuK0p7QbtlwGz/sHfbY4j2m9m56WcrT2uizUmz4o9bd\nK7p+s21s1yi6fPk21rPF7yjsOFrZU2En5W6Z4Y5U8NB+D9yyQCN2Yl9UeUcLEtVeS4I+VthV\nGbsN7ViFFXO+SXGCoqjfGS36W7EEy9uPX/629L8TlpS5CVLX/87ecrXqFk3bFVg7vvY6/1xh\nV5Dccp87omF7hfe1d5FnWbxR+zvxU5wF8V7fcVb726yamjKviQobL6qEi1hgv6fe3yF3tSXu\niIZtPOOMIoBAFQkEq2i/7BYBBNJL4DV15+holyxROExxf3TaBrMV+3mm7SrJK55pd9R78mAn\nubPcBXGG3qTMu9jepY090SzqhKWs+1uvffRVjFFYv3dVxJ4w2a0+TyiGKrZV7BabNxV2UmhJ\n0liFe6XsXY1bEmTvoNsJ7TBFR4VbnnNHEjy0K2yxxWwru1iC5JY6GrGk9Ct3RszwHE3/S/Gh\nwlxuV3gTH03+Vd7TmL0u/IqBCrdvtr55V0bxvkkwdxs7rIj22mvqOEUDxbjo/g/X0BIQG5ak\nmJe9fmtFV3aHsdvasXOLJQpusdfZo4p/RGdY8tFSEYpOf6HhzOi4Dby/szZtiZ7tv6jiHtfY\n5fFe37HreKdra2KGwn1TxPb5RnTeOxq+rWiosBLeOvifn/U0x15vscu9fzvW/M9WzEAAgUoX\nIEGqdHJ2iEBaCkxVr+yEwf1Hf6nG7UpBUScu7tUPrfK38rNnqkDjIxSxJxOeVeKOFnVCFG/l\n8uzPTgxfUNiJuP0t7asYrjhC0VphZVeFnXyusoltlOe1fN/oOldoGIiOv6WhJXyWKI1U2Mn8\nHwor1lfbrizFTtRSodhJqJ0UuyfMF2h8QpyGb6d5hymsX4MV2Qp7t7+osloLvlHsoLArdm6x\nk/F17kQFDseo7p6e+u1NhuJKRbbXkiRL1twrQeZo7XlCUZKyQCv1iK64c5wN7LW8o2f+bM+4\njd6rcBOk0RpvZTOj5d/uSHS4QEN73VsSZ+URxT1bxkr3I7d0q2/5TjY3OdqkbbspFnnqqOMZ\nt78N8Yr9nbDtYvvfwbPyPM84owggUEUCqfIPsop42C0CCJRQwE4YLves21zjXyrcEwp3kSVQ\n5ynspMZb3JOdDz0zm2ncEgy32EnWR4pPFPcrLFkobynL/npppy8q5ijsXeg+CkvmrG2XKg5W\neIt7Yu+dF2/8Zc10T6zaeVaYHh23RMmKXUVqsmVs661qS6LjJRlYO91S0na561fVcJ52fLNn\n5+M1fofC2/7mmn5W0UjhFltnW+WdOCvEmxdntTLPytKWhyvu8tRgJ8zeac+iv42+87eprRPx\n5sVZrdhZq7T02Jg1btO0e0UkZtH/TD7nmXOQxi3Z8pbTNdE6OsOSm9ej4+7gO424v4sNND4s\numCDhk9Gx93BZo3Y3xa32P685QBNzFe8pLhG4b5po9G/Fe/vwt8WFDFhb0y45RuNLHInNOyv\n8L4eLREqqlysBe7fO1tnqMKbPFqCTkEAAQQQQACBNBHwqx/vKuwEyBt2Am8nRN8rLAHwLrNx\nO+HppLBiJw528uGus1Djhyl2VzzimW/LhyjcMkoj7ja2j9iyq2a4y21YPbpCWfZn26701Pet\nxo9SdFHsr3hT4e7rR42XpszQyu62NvzNs3H3mGW2/BzPcnd0eMx6lli6xRINt/6NGr9KYa5u\n+VQj7vJL3Zme4QzP8us984sbHerZxupervh1G/GGlnuLJYWLFW7bbPiH4j3FB4o/Fd5lNs97\nwup9fazQMrfY1QrvdjY+0l2ooSUO7nLvfM8q/zNqr1V3Gxt6+2vjdjXQuzys6UEKb0l0ewer\ncu8+B3h35hl/MGa9Rz3LihttrIX2Wnf3Yb/T9yjOVjzvmW/L71TEK/Z77m7vDmOvHrnbWZLs\nrmPDexVDFccp7LXlLntb495i/u6yI7wLPONF/f6c79nW6rCkr43iWMUChVuvDb1vkhwes8yW\n2+v7eMXlCnsdu9t+rXG/goIAAggggAACaSRQXX2xkyr3H/62hvZOsPfdU6OwW54WKYrb9hZb\n0VO8J5SlSZCsirLsz279s5PA4tqYq+X9FaUp/9TK3jpjT1CXxixvH6fyok7wbNWxMdu7+6od\nracyEiR3n8UN472L3kptfLeI9nvrskS8frQ/7sD7+vAmSPW0gr1e3O3tqkItdyMNE5EguXXH\nG1pb7AQ7tiS6vSVNkOqoIYsV3rbuEdu4IqYtSZods623Hhu3hMXrq8m/in32zutt6+/019K/\nj9gbG5MUsfV7p+13xX3jxd26PAlSS1Wytoh92mvIm5hd5e5QQ2+CZG9KzFN42+mO29WyIQoK\nAggggAACCKSpwN7q13OKPIV7AuAO7SToCcVeiqJKQy2wd55jT0gWaN7JitjiPaEsbYJkdZV2\nf7bNzooXFd4TbLePUzXfEq/SFjsJc+uw4RExFXiTT7vSFq8UlyBlaYMHFN592El6a4WVZE6Q\nrH327vrZCrtqZ8mMtx+/afpfimqK2OJ9fXgTJFvvK4Vbj/XfW7wn7CO9C4oZ313L3Ppih3Yr\n6k+KtxRXKiwhiVcS3d6SJkjWFkv+ve2eq2lLXkpSGmiluxTfKcIKt54lGv+HwhKb4srNWuhu\nY8dlW+UIrWBXrry/g/a6mKxor4gt5UmQrC67zc4ScLeNNrTX4i6KozzzLQlyy+Eacde3115d\nRezfDXtDoJuCggACSSKwrT9WSdJMmoEAAikqkK12N4+Gnbj8rligiJfEaHbc0kZzrQ47ybIr\nS6XZVquXupR2fzW1B0tsGimsfwsVdiKczMVOZDsrrK3mmorFTtq7K4KK+YrYxEezKFUoYFfx\n7DVmCeHKCm5HDdXfQ2FXbX9WrFNUVLEkvZVie8Ucxbb6ZgnSwwortq69GWOltsKSIktAt1WH\nVqEggAACCCCAAAIIIIAAAqkvEHsFKfV7RA8QyAABeyeEggACCCCAAAIIIIAAAgggIAESJF4G\nCCCAAAIIIIAAAggggEBUwO7dpiCAAAIIIIAAAggkXmCpqnwrWu36xFdPjQgggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAII\nIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAAC\nCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAA\nAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCA\nAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAggg\ngAACCCCAAAIIIIAAAggggAACCCCAAAIIIIAAAlUg4KuCfbJLBBBAAAEEtiWQpRW2UzRSNIwO\nbbq2otaWod9fO+gP1vMF/XV9Ec33OSEn4oTCjqNtI0EnHA5GIpEsRVDrOz6fr1CDiMbCPp+j\ncX/Y8Tthv+PL17YbnYhvY8SJbAgXhtdHwgXrw+HwRq3/h2K1YlWcWKl5mxQUBBBAAIE0EiBB\nSqODSVcQQACBFBGoqXa2jEaL6LBVViinnRKX5oWFBfULCwpsnS3F5/eHq9eslV+zbr1wtVq1\nneq1avs1HahWs1awWo1aTk6NGk5O9ZpOMBRygllZTiAYVGwdbp22XMtRvlToRMJhDRWFhVuG\nNl2Qn+/kbvrTyVPkbtq0ZdymN2/6M7xpwx8Ff6xaVbhh3Wpn4/p1/s0bN1jC5d9SoX74A8GN\nwWBwmRKrRfm5ub9o1q+esOmfFX8qKAgggAACKSJAgpQiB4pmIoAAAikmUE3tba/oGI1OWTk5\n3cMFhe0LC/LrWF/8gUBhnQYN8xo0be5r2KJlqEGTZv76jZs6terVd2rV327LsLaGSojs6o9t\nUuVFyZGzacN6Z8Patc4fa1c7a1csc9YuX+asWf67s3r57+GVSxbnrVn2W2TdqpVZBfl5W65c\nKUlb5QsE5udv3vyDOjAvGj9qaNObFRQEEEAAgSQSSI7/OEkEQlMQQAABBEolENDalgj1tAgE\nQ318ft+OBXm5jTXtC1Wrlt+4eauCZu07Zzdt3dbfuGVrp1HzVk79ps0cS36SJfFRWxNe1q9e\n6SxfvNBZvmSRhgucZYsWhn//ZV7u8iULA7oyFdIOw1nZOb+GCwu+1hWzWZr+Nho/aZivoCCA\nAAIIVIEACVIVoLNLBBBAIEUFLBnqruir6KeT+366StJRt6mFsqtVz9u+fadw6649clp27Oo0\nbdveadyyjZKgBina1Ypt9oZ1a51f5/24NX6e6yz84dvNv83/yb/5z40h3VKYr6tOP+iWvQ/U\nii+i8b2G9hkqCgIIIIBABQuQIFUwMNUjgAACKSxgnw/qZ6HPBw3WrXG99DmenNoNGm5u36t3\nsHWXHsEWnbo6LTp0duo3aZbC3Uyepq9budxZ9OP3zoLvZzs/z56ZP3/2zPCGtWuylTTl6TNW\ns3Wb3vtqrSVOFsuSp+W0BAEEEEgfARKk9DmW9AQBBBAor0AbVbCr3+8f6s/K2rMgN7dZKKda\nvq4KhTvsuHN22+47OG2793LqbGcPlqNUloAlTb8oYVrw/TfO3K8+z5v3zVe+/NzNWVnZ2Yvy\n8/Pf1tMmLGmymFtZbWI/CCCAQDoLkCCl89GlbwgggEDxAnaFaLjfHximp7HtXpCf21gPRMjt\n0ndgoMvOA4KddurvNGvbIa0/J1Q8T3IutafwLf7pe2fuzC+cH7/8pGDOF58Ubly3NltP7lut\n2x1f01W+19XyNxVLk7MHtAoBBBBIbgESpOQ+PrQOAQQQSKRAjiobrBihqw9j9BmXdjXq1M3t\n1m9QsPPOAwKd+vRzmrZpn8j9UVclCdjDIJQoOd9+9G7Btx+/F9bjyEM6xnN1jKeqCZYsvaew\n73WiIIAAAghsQ4AEaRtALEYAAQRSXKCV2j9aH/ofre/+GRzRl6i279k7v9eQ3bO777Kr01Kf\nIaKkl4A9inzRnO+c7z553/nmgxm582Z9aV+YGw4EAu/oO5+eU29fVtj3NVEQQAABBOIIkCDF\nQWEWAgggkOIC9qS5cVmh7In5ebndam/XKLf30D1DPQYO9XXpu4u+VLVGineP5pdGIG/zZmfO\n5x85M999K/zl9Ffz9dAHu7r0na4uTVE9Lym+Lk19rIsAAgikuwAJUrofYfqHAAKZIGB/y/sr\n9guGsg/UdxA1b9Kq7ea+e43K6T1sL10l6pYJBvSxBAJ2dUmPFHe+fvdN54s3X9n82y/zcvR0\nvKUFeXmPafPJCnusOAUBBBDIaAESpIw+/HQeAQRSXMC+nPUgneAeoRPcJm2698rbec99snsP\nG6EvY22Z4l2j+ZUhsGrpr86Xb7/mfDLthdwFP8wOKcH+VQn2o9r30wr78loKAgggkHECJEgZ\nd8jpMAIIpLhAW7X/IN0+d6Run2vfvEPn3IGjJ2T3Hb6vU69R4xTvGs2vSgFLlj5/Y5rz0bTn\ncpfMnZOt19hCvcYeVJseUfxSlW1j3wgggEBlCpAgVaY2+0IAAQTKJlBLm03UlaITdaWoT8Pt\nW2zeZd8JOf33HuM0btm6bDWyFQLFCCxfskjJ0lTn/ReeztUT8kLBrNAnBfl5/9Ymzyg2FLMp\nixBAAIGUFyBBSvlDSAcQQCCNBQbqS1uPcxzfxFC1HP/A0ftnDRo9wdeqsz2DgYJA5Qj88u0s\n54OXpoQ/mvZ8Qf7mzWHtdYq+a+kBDd9V6MGIFAQQQCC9BEiQ0ut40hsEEEh9gUbqwpFZulqU\nn5fXWk+dyx864ZDQjkP3dPQufur3jh6krIBut9PDHd5y3n/+6Xw9QlzfS5u1RK/R29Qhuw1v\ndcp2jIYjgAACMQIkSDEgTCKAAAJVJNDPHwicFglH9q+9XcPC3SYckq2rRU79Js2qqDnsFoGi\nBdauWO689/xTzvTJj+RuWLPa53N8k3VV6Q5t8WnRW7EEAQQQSA0BEqTUOE60EgEE0lMgW906\nUJ8tOkOfLerRtd/A/D0POca+r8jRrXXp2WN6lVYC+vJhZ9Z70523nnoo74fPPsrSlc/vdVVp\nkjppjw3flFadpTMIIJAxAiRIGXOo6SgCCCSRgF0WOjUQDJ6s+5SqD9nvwODuBxzua9yqTRI1\nkaYgUDoBPczBmTHlsci7zz2Rry+h3VRYUGC3392pWFa6mlgbAQQQqFoBEqSq9WfvCCCQWQJd\n/f7AOfqyzkMaNm9RuNdhx2XvMmo/J7ta9cxSoLdpLZC7aZPz4cvPOK8+dE/u6mVL/TrReCwc\nDt+kTn+X1h2ncwggkDYCJEhpcyjpCAIIJLHA4EAwdH5hQd6Idj1754869tRQz0G7OT59coOC\nQLoKKCnacvvdKw/enTv/m69CwWDW2wUF+Verv2+na5/pFwIIpIcA/53T4zjSCwQQSD4B+/s6\nRk+eu0jfH7PjDkP3LNjn6JOz2vXYMflaSosQqGCBX777xnnt4XsLvnjrFb9uK52p34mLtctX\nKni3VI8AAgiUSYAEqUxsbIQAAggUKWB/V8cqMbpKT/XqOHjMAc6Iw48P8PmiIr1YkEECyxYt\ncKbdf0fhh1Ofc/QZvO/1cBJLlF5U8H1KGfQ6oKsIJLsACVKyHyHahwACqSKwNTEKKTEqLOw4\ndPzBvn2OOcVfr1GTVGk/7USg0gRW/rbEmfbg3eH3n38qosfbz1OidIl2PkURrrRGsCMEEECg\nCAESpCJgmI0AAgiUUMD+jo7To7qvJDEqoRirIRAVWLN8md16F3578qNhn983X4nSOVr0EkAI\nIIBAVQqQIFWlPvtGAIFUFximW+lu0a103YaOP0RXjE7milGqH1HaXyUC61Yud16+7/bwjGce\ndwKBoH1G6Sw15J0qaQw7RQCBjBcgQcr4lwAACCBQBoEd9ESum/RErt30mO7wfqecFajfxL7a\niIIAAuURWPHrYuf5O28s/OTVF/3BrKx3CvLzz1Z9X5anTrZFAAEESitAglRaMdZHAIFMFmij\nz0tcrVvpJvbYZdeCA864MGv7dh0z2YO+I1AhAr/O/8l55rbr8me9Nz2ohzk8ry+dPVM7WlAh\nO6NSBBBAIEaABCkGhEkEEEAgjkAdzbvU5/Of2qpLt/CBZ14U6ti7b5zVmIUAAokUmD97pvPE\n9ZfmLfh+ti+y9ctm7XuU/kjkPqgLAQQQiBUgQYoVYRoBBBD4r4Bfo8foHezra2/XqNrBZ1+c\n3WfYiP8uZQwBBCpF4LM3pjpP3nhF7oY1qzbqatJ52un9Cp54Vyn67ASBzBMgQcq8Y06PEUCg\nZAKD9ACGexyfr9OYE04LDj/0GCcrlF2yLVkLAQQSLpCfu9l547H7nZfuuy0/Eo7M14McTtFO\n3k74jqgQAQQyXoAEKeNfAgAggECMQItAIHBTYWHhhAEjx4b3P/2CQN2GjWJWYRIBBKpKYN2q\nFc6zt19f+MGLU/SRwOCzhYUFp6ktv1VVe9gvAgiknwAJUvodU3qEAAJlEwhqs9P9/sAVzTt0\n9h924ZWhdj12LFtNbIUAAhUu8Mt33zgPXX5urh7ooCftF56vHd6hKKzwHbMDBBBIewESpLQ/\nxHQQAQRKINBPt9M9qM8atT/wrIuyhow7UHfW8eexBG6sgkCVCoTDYeedKY85kyddkx8uKJyr\n2+6OUYM+qdJGsXMEEEh5gUDK94AOIIAAAmUXqOP3+2+NRCJ37Tx8n/r/uvOhrE69+5Ecld2T\nLRGoVAF7I6NN917O4LETA6t//63er/N+PFa/0y31O/2+GrK5UhvDzhBAIG0EeIs0bQ4lHUEA\ngVIK7K8rRnfVbdi41pEXX5fdrf+gUm7O6gggkGwCc7742HnwsnNzlSz9oafdHaf2vZBsbaQ9\nCCCQ/AIkSMl/jGghAggkVqCREqP79BSsfUYefZJ/32NP9WVl5yR2D9SGAAJVJpCfl+u8eM+t\nkVcfuieizxQ+p4c4nKzGrKiyBrFjBBBIOQESpJQ7ZDQYAQTKITBRydG9TVq1rXb81ZNCLTp2\nKUdVbIoAAskssPCHb537/u/03GWLFmzW1aST1NYnk7m9tA0BBJJHgAQpeY4FLUEAgYoTaKjH\nAf87HAnvO/r4f/pHHXOqT4lSxe2NmhFAICkECvLznVcevNt56d5JhT6/73UlSvYQh9+TonE0\nAgEEklaABClpDw0NQwCBBAnYZ43ua9yqTc7xV03Kbtmpa4KqpRoEEEgVAT0K3LnvwtPzNNyk\nJOkotfv5VGk77UQAgcoX4Cl2lW/OHhFAoHIEaulbJB/UI+kuGXXsqZYcBes1alI5e2YvCCCQ\nVAK16zfY8qQ7fQF09tyZnx+gzya10ZPupquReUnVUBqDAAJJIcAVpKQ4DDQCAQQSLNA/EAxN\nqdewYcOTbrg7u023ngmunuoQQCBVBebN+tK5+7x/5K5fuXJFYUHeRPXjo1TtC+1GAIGKEeAK\nUsW4UisCCFSNgP1N+z99N8ojA0aNq3XabQ9kNdy+RdW0hL0igEBSCtRv0szZdb8Dg2tX/F5j\n0U8/HK1GZineV4STssE0CgEEKl2AK0iVTs4OEUCgggRaBrOyng4Es/ocfdkNWTvvuU8F7YZq\nEUAgXQS+mvG6c/9FZ+bp0eBf64EO49WvJenSN/qBAAJlFyBBKrsdWyKAQPIIjPEHgo+17d4r\ndOK1t4fsHWIKAgggUBKBNct/d+4868Q8PRbcHuBwsLZ5pSTbsQ4CCKSvALfYpe+xpWcIZIKA\nPav7Oj2I4dZ9j/1H6JgrbgpWr1U7E/pNHxFAIEEC1WrUdAaN3j+Qn5ubPXfWlwep2hqKdxTc\ncicECgKZKMAVpEw86vQZgfQQaKZb6p7Nys7pfdL1d4a6DxiSHr2iFwggUGUCsz98x7nnvH/k\n5edunhW95W5xlTWGHSOAQJUJkCBVGT07RgCBcggM03cbPdOiY9cap958b6h+46blqIpNEUAA\ngf8K2C13d519Ut6C72fbLXcTtOSt/y5lDAEEMkGAW+wy4SjTRwTSS+AC3VL34B4HHpFz4rV3\nBGvUrpNevaM3CCBQpQLuLXebN24IzZ898zA1ZqPi4yptFDtHAIFKFSBBqlRudoYAAuUQqB4I\nBJ8KZGWdeMI1twX2Ouw4n74IthzVsSkCCCAQX8Dn9zu6bdfXuGUb36z3pg/TVwd01xfL2sMb\n8uNvwVwEEEgnAW6xS6ejSV8QSF+BVsGs0Cs169Vvf8YdD4Wad+icvj2lZwggkFQCi3783rn1\nn0flbVizZl5Bft4oNe6XpGogjUEAgYQLkCAlnJQKEUAgwQJD9HmjF9v17F391JvuDdWsWy/B\n1VMdAgggULzAhrVr9Cjwk/Lmzfpisz6XtJ/Wnl78FixFAIFUFuD+lFQ+erQdgfQXOFG3tjw9\ndMKhObqtLphT3Z6+S0EAAQQqVyCUU83ZZZ9xAX0uKevn2TMP1d5/V3xZua1gbwggUFkCJEiV\nJc1+EECgNAJ+x++/ze/zX3LERdcE9j3uH/pIgL8027MuAgggkFAB93NJ9Ro28c/64O2RTiRi\nX7pmT7iLJHRHVIYAAlUuwC12VX4IaAACCMQIVNctdZOzQtnDT7vtgaxOffrFLGYSAQQQqFqB\n7z/90Ln9X8fm67uSXtctdxPVmj+rtkXsHQEEEilAgpRITepCAIHyCjTWl7++VrNeg65n3/t4\nqGnrduWtj+0RQACBChH47ee5zk0nH577x+pVP+rhDSO0k6UVsiMqRQCBShcgQap0cnaIAAJF\nCHQOBENvbd++Q8Mz7nw4VLv+dkWsxmwEEEAgOQTWr17lTNIT7vSkuzW6krS7WvVdcrSMViCA\nQHkE+AxSefTYFgEEEiWwqz8QnNFj4JA6Z9zxcKh6Lbu1n4IAAggkt0B2terOLqP2C/w6/6ec\nZYsWHBGJhN9Tixcnd6tpHQIIbEuABGlbQixHAIGKFhinDz+/tPvEw7KPufymoG6xq+j9UT8C\nCCCQMAF9gbWz0x4j/etXrggunPPt4Xp4wyxV/lPCdkBFCCBQ6QIkSJVOzg4RQMAjcJQe4/3o\nfqecFRz/j3M0yl2/HhtGEUAgRQTsb1evIcP0qM2I/6cvP7WHNthVpK9TpPk0EwEEYgRIkGJA\nmEQAgUoTONPx+W4//IIrA8MPPabSdsqOEEAAgYoS6LxTf1+dBg31GPAZ+2ofuYoPK2pf1IsA\nAhUnQIJUcbbUjAACRQtc7fcHLj3h2tsCA/edUPRaLEEAAQRSTKB1155O8/adfF9Of223yNbv\nSnojxbpAcxHIeAESpIx/CQCAQKUK6Pte/fcGglmnnHb7A8Edh+5ZqTtnZwgggEBlCDRr28Hp\nsMPO/s9ef7mvbrtrqkTplcrYL/tAAIHECJAgJcaRWhBAYNsCumgUeCgrO/vQs+99IqvzTgO2\nvQVrIIAAAikq0HD7Fk6nnfr7P3n1pR2cSLi1kqSp6kokRbtDsxHIKAESpIw63HQWgSoTCCg5\neiyUnXPAefdPzmrbfYcqawg7RgABBCpLoEHT7Z1u/QcFPnn1xe56BHhHJUkvat8kSZV1ANgP\nAmUUIEEqIxybIYBAiQUsOXoilFNt/HkPTM5q1bl7iTdkRQQQQCDVBeo1auL0GLhr4NNXX+oc\nKQz3VKL0nPoUTvV+0X4E0lmAZ+qm89GlbwhUvUAwEAg8FapWfV9dOQq16Nil6ltECxBAAIEq\nEFgy70fn2mMOyMv9c+PrhQUF+6kJBVXQDHaJAAIlECBBKgESqyCAQJkEgr5AYHK16jX2Oe+B\nKSE91alMlbARAgggkC4CS3+Z51x15Pi83I0bXy4sLLDvSypMl77RDwTSSYBb7NLpaNIXBJJH\nQHfVBZ7KqVFz3/MffFbJUcfkaRktQQABBKpIoFa9+k7PwcMCH019rkM4XGifSXpBTeEzSVV0\nPNgtAkUJkCAVJcN8BBAoq4BPj6t7IJSTs//5XDkqqyHbIYBAmgrUrr+d032XIYGPpz7fJRIu\nbKUk6eU07SrdQiBlBUiQUvbQ0XAEklTA778tmJV19Ln3PZXVqgsPZEjSo0SzEECgCgXqbtfI\n6dpvYOCjac931yPA7XuSplVhc9g1AgjECJAgxYAwiQAC5RK4KhAInn7m3Y9mtd+hT7kqYmME\nEEAgnQXs6Xadd+of+Hja8zvqdru66usb6dxf+oZAKgmQIKXS0aKtCCS3wHm6te7i02+7P6h3\nRpO7pbQOAQQQSAKBBk2aOe179dGXyb7QV1eRctWkD5OgWTQBgYwXIEHK+JcAAAgkROAEn893\n88k33hXYYcgeCamQShBAAIFMEGi4fQunWduO/s/femWY+rtY8XUm9Js+IpDMAiRIyXx0aBsC\nqSEwxvH5Hjnqkuv8/fcekxotppUIIIBAEgk0a9vBqdOgoe+bD2aMUrNmKn5KoubRFAQyToAE\nKeMOOR1GIKECA/TIumljT/xXYPghR/O9agmlpTIEEMgkgdZde1p3fT999dkEJxKZoXG7mkRB\nAIEqECBBqgJ0dolAmgh0CgSD7w4eOzF74r8u8KdJn+gGAgggUGUCemiDb/3KFc6iH7+fGImE\n7fHfy6usMewYgQwWIEHK4INP1xEoh0DTQDD0YfcBg+scd+UtQV1FKkdVbIoAAggg4Ar0GDTU\nv2TuD4HlixdN0PckPaH5G9xlDBFAoHIEuCWmcpzZCwLpJFAzmBX6tEXHzu3P/c/kkL4QNp36\nRl8QQACBKhfIz93sXHP0AXlL5s6ZU5CfN0AN+rPKG0UDEMggAd72zaCDTVcRSICAX7fVTanb\nsFG7f935CMlRAkCpAgEEEIgVyMrOcU6//cFQzXr1O+tv7tNazvlaLBLTCFSgAL9wFYhL1Qik\nocANunq0+xl3PZJds459ryEFAQQQQKAiBGrXb+CcdfejoWBW1l6q//qK2Ad1IoBAfAE+gxTf\nhbkIIPC/Akfru46uPv32h4JtuvX636XMQQABBBBIqECteg2ctt13DHz8ygv99GS731X5lwnd\nAZUhgEBcARKkuCzMRACBGIEhSo6eOfT8KwJ9h9vXdFAQQAABBCpDoGHzlk69Rk38s96bPlL7\n+1DxS2Xsl30gkMkCJEiZfPTpOwIlE2ire+Df2W3/w7JHn3AaD3YpmRlrIYAAAgkTaNWlu7N5\n4wZnwfffjI2Ew0+p4rUJq5yKEEDgfwRIkP6HhBkIIOARqBkMhd7vuGPfhsdffSuP8/bAMIoA\nAghUpkDXvgN9cz7/OLBu5YoR4XDhQ9p3fmXun30hkEkCPKQhk442fUWglAKBQPChWvW3a3nK\njXdl+QO8n1JKPlZHAAEEEiZgf4NPufGerJr16rUPBAIPJqxiKkIAgf8RIEH6HxJmIIBAVOAM\nn9835vRJ92dXr1UHFAQQQACBKhaoWbeec/ptD4b05dz7qSlnVXFz2D0CaSvAW8Jpe2jpGALl\nEthVD2V47JjLbwp26z+4XBWxMQIIIIBA4gTqbNfQadSitf/Lt1/bQ7V+pPg5cbVTEwIImAAJ\nEq8DBBCIFdheD2V4d7cDDssZedRJPJQhVodpBBBAoIoFmrfvpIc2bIxEH9rwmJqzvoqbxO4R\nSCsBbrFLq8NJZxAot4B9KeGLemJSjYln/B9/H8rNSQUIIIBAxQjsf9p5/tZdelTT3+xntYdg\nxeyFWhHITAFOgDLzuNNrBIoSuC67Wo0ep970b0uUilqH+QgggAACVSxgD204+ca7Q1nZOTuo\nKVdVcXPYPQJpJcAtdml1OOkMAuUS2Mfx+W49bdL9wRYdu5SrIjZGAAEEEKh4gWo1ajotO3UN\nfPLqiwO0ty8Ucyt+r+wBgfQXIEFK/2NMDxEoiYB97mj63keemLPrfgfxuaOSiLEOAgggkAQC\njVu2dvJzc52fv/16X32J7ONqEp9HSoLjQhNSW4Bb7FL7+NF6BBIhENDtdJP1LmS1cSedQXKU\nCFHqQAABBCpRYL9TzvLp80g5fB6pEtHZVVoLkCCl9eGlcwiUSOD/lCHtfPINd4d0FalEG7AS\nAggggEDyCPz1eaRQdi+16uLkaRktQSA1BbjFLjWPG61GIFECQ/S5owdPvPb2YLuevRNVJ/Ug\ngAACCFSygH0eqWmbdoHP3phmX173pmJJJTeB3SGQNgIkSGlzKOkIAqUWqB0Iht7Zdb8Da+x9\nxAlcTS41HxsggAACySXQtE17Z9XSX8O/zp+3dyRceJ9al5dcLaQ1CKSGACdFqXGcaCUCCRfw\n+wO31WvYsL6+74g3ShKuS4UIIIBA1QgcfM6lgdrbbddQf+Nvr5oWsFcEUl+ABCn1jyE9QKAs\nAmMikfBhJ1x7e3Z2tWpl2Z5tEEAAAQSSUCCneg3nJP1t19/4w9W8cUnYRJqEQNIL8M5x0h8i\nGohAwgUa6mEMb4048sRqg0ZP4Kl1CeelQgQQQKBqBeo3aeaEC8PO/G++1K124YfUmg1V2yL2\njkBqCXAFKbWOF61FoNwCSo7ub9yqTfWxJ/6L5KjcmlSAAAIIJKfA6OP/6du+Xcdq9jc/OVtI\nqxBIXgESpOQ9NrQMgYoQODwSiYw88do7svV9GRVRP3UigAACCCSBgH1tw3FX3RqKhCN7qTkH\nJUGTaAICKSPALXYpc6hoKALlFmjsDwRf0xcK5vTZfUS5K6MCBBBAAIHkFqhdv4Gjr3Lwz/36\n8z10q51dSfozuVtM6xBIDgGuICXHcaAVCFS4QCAQvKtp67bZIw4/vsL3xQ4QQAABBJJDYORR\nJzmNW7bO0f+Au5OjRbQCgeQXIEFK/mNECxFIhMDocLhw7LFX3hyyb1ynIIAAAghkhoDdTn3c\nlbdm63+APdFubGb0ml4iUD4BzpTK58fWCKSCgL4QNuvN4YceU2PgvuN5MEMqHDHaiAACCCRQ\noG7DRk5+Xp7z8+yZe+pWO/sC2c0JrJ6qEEg7Aa4gpd0hpUMI/F3A7/ffoH+OdcaedAbJ0d9p\nmEIAAQQyRmDMCaf59PjvmvY/IWM6TUcRKKMACVIZ4dgMgRQRGBwOh489+rIbskM5OSnSZJqJ\nAAIIIJBogaxQtnPUJdfpVrvwUaq7f6Lrpz4E0kmABCmdjiZ9QeDvAsFgKHT/wNETIl123uXv\nS5hCAAEEEMg4gc47DXD6jRgdDmaF7Il2fMwi414BdLikAiRIJZViPQRST+AfempR6wNOv4B/\ngql37GgxAgggUCECB551UdAfDHRQ5f+okB1QKQJpIMCJUxocRLqAQByBJnpa3YsHnnlRTued\nuJMijg+zEEAAgYwUyKlew6les3bg24/f21VfHP6gEP7ISAg6jUAxAlxBKgaHRQikqoCSo1ua\ntWkfGDrhkFTtAu1GAAEEEKgggaH7H+ps365jQHcZ3FZBu6BaBFJagAQppQ8fjUcgrsDgcGHh\nxMP/7+psPa0o7grMRAABBBDIXAH733DkxdeFCgsL9pPCsMyVoOcIxBfg7Cm+C3MRSFUBfeVR\n6D5931G4fa8+qdoH2o0AAgggUMECbbr1dAaN2d8e2HCHdsX5YAV7U31qCfALkVrHi9YisC2B\nE/Xh27b782CGbTmxHAEEEMh4gfGnnh3w+X3tBXFMxmMAgIBHgIc0eDAYRSDFBWoHgsGX9zv5\nrBrdBgxO8a7QfAQQQACBihawBzb4ff7AT199NjASDt+t/eVV9D6pH4FUEOAKUiocJdqIQMkE\nzq9Zr0G13Q86omRrsxYCCCCAQMYLDD/0GEf/O2oI4sKMxwAAgagACRIvBQTSQ6Clz+8/46Az\n/y/bvi2dggACCCCAQEkEsrJznIPOuijb/odo/TYl2YZ1EEh3ARKkdD/C9C8jBPR4mSdYAABA\nAElEQVRY7+taduwS2Xn4qIzoL51EAAEEEEicQJtuvZzsatWDPp9vauJqpSYEUleABCl1jx0t\nR8AV2Mke633wuZdl65+bO48hAggggAACxQr8+cd6Z8qkayPnj9ktEvD59L2xkS7aoHexG7EQ\ngQwQIEHKgINMF9NbIJiVdesOQ/cs6LDDTundUXqHAAIIIJAQgYL8POfNxx9wztp7l8j0Jx9y\nTjj+ON/6dWv9I0eOLMjOzr4+ITuhEgRSWIC3m1P44NF0BCSwh64avX7lc2/5m7ZuBwgCCCCA\nAALFCnz59mvOE9ddGlm/eqUzfM89fZMnT3Zq1qy5ZZvZs2c7vXr1sitJQzXjvWIrYiECaSzA\nFaQ0Prh0Lf0F9AV/1w3YZ1yE5Cj9jzU9RAABBMoj8OOXnzqXTBwZvvvskyOtmjVx5s+b53vl\nlVf+So6s7h49ejgTJ04sDIVCN5RnX2yLQKoLcAUp1Y8g7c9kgVF66tAL1770bqDh9i0y2YG+\nI4AAAggUIfDbz3OdZ267Pvz1u2/6WrRoGXn66af8AwYMKGJtx5k/f77TsWPHwnA4PEYrTSty\nRRYgkMYCXEFK44NL19JawKerR9cOGXegQ3KU1seZziGAAAJlEli3crnz2DUXhS+aMDyy4Jsv\nnKeeesq3aNHCYpMj21G7du2cY4891tFnkewqEm+kl0mfjVJdgBd+qh9B2p+pAhMCweBT10/7\nIFCvUZNMNaDfCCCAAAIxApv/3Oi8+cQDzsv33R4JZWVFLjj/fP+FF5buO2B/++03p3Xr1gX5\n+fmHqvqnY3bBJAJpLxBI+x7SQQTST8AfCIVeGLb/YfV3Hr4Pb3Kk3/GlRwgggECpBZTNOB++\n/Ixzy6lHRuZ89pFz6MEH+z766CPf0KFDS11XrVq1nDVr1vhmzpy5Q2Fh4R2lroANEEhxAU6u\nUvwA0vyMFNg/EMx64sbXPgrWadAwIwHoNAIIIIDAfwVmznjDeeqmKyKrly11dhs61DdlyhSn\nbt26/12hDGNLly51WrZsWVhQUDBBm79QhirYBIGUFeAzSCl76Gh4pgpkhUKXDJ1wsJ/kKFNf\nAfQbAQQQ2Cowf/ZM58rDxobvOPOESMM6tZzvvv3W9+abb5Y7ObLamzZtap9F8umzSJfhjUCm\nCXAFKdOOOP1NdYG99eS6l6+f+n6gQdPtU70vtB8BBBBAoAwCyxcvdJ6988bw56+/7GvStGnk\n8cce8w8bNqwMNRW/yYIFC+yhDXqgXXhvrflG8WuzFIH0EeAKUvocS3qSAQLBrKxLdtlnnENy\nlAEHmy4igAACMQJ/rFm95Va688fuFvnp0/ed//znP76lv/1WIcmR7VoPanAOPvjgiL4X6dKY\npjCJQFoLcAUprQ8vnUszgcHqz7tXPT/dxxfDptmRpTsIIIBAMQJ5mzc7b09+xHnh7psjfp/P\nOfmkk3w33XRTMVskbtGcOXOcrl27RlSGqNYPElczNSGQvALB5G0aLUMAAa9AMJh1Sc/BwwqV\nHPF764VhHAEEEEhTAd3a5nz22kvOk3oAw6Y/1jtjRo/2PfHEE46u6FRajzt37uyMHTu28JVX\nXrk0Nzd3j0rbMTtCoAoFuIJUhfjsGoFSCOygdb+65Mmpvladu5diM1ZFAAEEEEhFga/fm+5M\nvvmq8PIlC32DBw3yTZ482WnUqFGVdEWP+3Z69+5t+7b/RbOqpBHsFIFKFOAzSJWIza4QKKuA\n3x84o2PvvvkkR2UVZDsEEEAgNQQW/fidc+0xE8O3nX5spGaW3zfzq69877zzTpUlR6a24447\nOoMGDcoLBAJnpIYirUSgfAIkSOXzY2sEKkOgse79PmjEESdU3j0VldEr9oEAAggg8JfAqqW/\nOv+5+KzwpQfuE1m9+Bfn5Zde8s2dO9fXs2fPv9apypGzzz47ZP+L1IaquYxVlZ1n3xknQIKU\ncYecDqegwEn1mzQt7DV4WAo2nSYjgAACCBQn8Ocf65xnb78hct7oXSPfvPOGc9ttt/lWLF/m\nHzVqVHGbVfoya8/2228f1o5PrPSds0MEKlkgUMn7Y3cIIFA6gexAMDhl7Eln1Gzbw279piCA\nAAIIpINAQX6eM2PKY86kfx4dmT/rS+e4Y4/xffjBB75+/folZfd8enqebrELTp8+vaceHnGr\nGlmYlA2lUQgkQICHNCQAkSoQqECBo0I51e69dfoXWTnVa1TgbqgaAQQQQKAyBHSbmvPl9Nec\nJ66/NGLfazRir+G+p556yqlZs2Zl7L5c+/jjjz+cxo0b52/atOlYVfRIuSpjYwSSWIBb7JL4\n4NA0BLJCobOHTjgkSHLEawEBBBBIfYE5X3ziXHrgyPDd554Sadtie+fn+fN8U6dOTYnkyPRr\n1arlnHDCCcHs7OyzU/9o0AMEihbgClLRNixBoKoFtnwx7PXTPvBt16x5VbeF/SOAAAIIlFFg\n8U8/OFMmXRP+9qP3fO3atYvou4z8ffv2LWNtVbvZL7/84lgfVHZVS96v2tawdwQqRoArSBXj\nSq0IlFvAHwic1H3AkAKSo3JTUgECCCBQJQJrVyx3Hr3qQj2ZbmTk1x++daZMmeKbN29eyiZH\nhtimTRtnxIgRBVlZWadUCSo7RaASBEiQKgGZXSBQBoH6endu/LCJh2eVYVs2QQABBBCoQoHN\nf250Xvr3pMg5owZFPn31Befqq6/2rV69yj9hwoQqbFXidn3SSSdlFRYW7qca6yeuVmpCIHkE\ngsnTFFqCAAIegSNq1K4X6TloN88sRhFAAAEEklmgsKDA+fClZ5zJk66J5G/e7Bxy0EG+//zn\nP75gML1Ot0aOHOk0aNAgvGLFisN0PCYl8zGhbQiURYArSGVRYxsEKlhAD2c4ddjEQ7N1m10F\n74nqEUAAAQQSIfD1e9Od88cMjTx6zf9Fdum7s2/5st99Dz30kJNuyZFZ6XHf9rAGPashm9vs\nEvHioY6kE+AhDUl3SGgQAs4Qfd/EjBte+dBfv0kzOBBAAAEEklhg/jdfOU/ccFl4wfezfd26\ndnWee+45X4cOHZK4xYlp2oIFC5y2bdvawxoGqsaPE1MrtSCQHAJcQUqO40ArEPhLIPpwhkKS\no79IGEEAAQSSTmDZwl+cO88+OXzVEftF8tauct6ePt03e/bsjEiO7GC0bt3a2W233fJ1NemE\npDs4NAiBcgqQIJUTkM0RSLBAbb0bt5+++4iHMyQYluoQQACBRAjYl7s+ds3FkQv22z0y74uP\nnYcffti3ZPFi/9ChQxNRfUrVcfLJJ4fU4AMVtVOq4TQWgW0IpNenBrfRWRYjkAICE7KrVXN6\n8HCGFDhUNBEBBDJJIE8PXZj+9MPOi/fcEvH7/M6555zju+aaazL6owqjR4+2L4/1rV279iC9\nFu7NpNcDfU1vAa4gpffxpXcpJhAMhY7ZZZ/9gsEsLiCl2KGjuQggkKYC4XDY+Wjqc87ZI3eJ\nPH/nTZFxY8b41q1dY8lRmva45N3SdyE5hx9+eFZOTs5RJd+KNRFIfoGMfucj+Q8PLcwwgZbq\n74ILH3ne167HjhnWdbqLAAIIJJ/Ad5984Dx+7cXh5UsW+nYZMMD37LPPOo0aNUq+hlZhiz77\n7DOnX79+ETWhtWJRFTaFXSOQMAGuICWMkooQKLfAods1a55LclRuRypAAAEEyiWwcM63zrXH\nHBC++eTDIrVzsnyzvv7a9/7775McxVHt27ev07JlyzwtstvsKAikhQAJUlocRjqRDgJZoexj\nB4+dmJMOfaEPCCCAQCoKrPxtiXPf/50RvuygUZE1SxY4U6dO9f04Z46ve/fuqdidSmvzkUce\nma3b7I6utB2yIwQqWIBb7CoYmOoRKKHATlrvs+unve/brlmLEm7CaggggAACiRD48491zkv/\nvj3y1pMPOjVq1IxcfdWV/lNPPTURVWdEHT/99JPTqVMn62sPxbcZ0Wk6mdYCXEFK68NL51JI\nYGLrrj3ySI5S6IjRVAQQSHmBgvw8580nHnTOHLFL5J1nHnNOPeUU3/p1a0mOSnlkO3bs6PTs\n2TNXmx1Syk1ZHYGkFCBBSsrDQqMyTSAYyj64395jsjOt3/QXAQQQqAoBfd+c8/mb05xz9hkU\nmXLr1ZHhuw/zrVm1yjdp0qSqaE5a7NNus1M5Qp3h7qS0OKKZ3QlexJl9/Ol9cgj0UTO+uOGV\nD50GTbdPjhbRCgQQQCBNBX74/CPnyesvC//28zzfzjvv5OjJdL7tt+dvb3kP99KlSx05KveM\nDFRdH5e3PrZPSwHLOwKKQoU9+TBpC1eQkvbQ0LAMEpjQslPXXJKjDDridBUBBCpd4Nf5Pzm3\n/OPo8A3HHxwJFuY5n3zyse+TTz4hOUrQkWjatKnTp08fe5rd2ARVmW7V/EsdmqHYLYk6ZgnL\nkYrHFR8qnlKcoggqSlos4blBsUSxWmF1NFTEK5do5npF03gLk2keCVIyHQ3akpECenrdIf33\nHsvtdRl59Ok0AghUtMCa5cuch644P3zx/ntFlnw/y5kyZYrvl59/9u+8884VveuMq/+AAw6w\np9kdmHEdL1mHO2q1oYpk+SKtkNpiCduDioMVdhl1ouIOxReKuoqSlCu00pmK9xS27UjFNEXs\nXWoNNO8MxV2K3xRJXUiQkvrw0LgMENghPy+3RZ899s6ArtJFBBBAoPIENm3c4Lxwzy2Rc0cN\ninzx+svOtdde61u9aqV/woQJldeIDNvTuHHjnM2bN9uXnnfNsK6nYncvVqN3VXyusG+nb62w\nRxG+puiluF2xrWJJ1j8UbygsybI6z1fYuw/DFN5yriYsabrOOzNZx0mQkvXI0K5MERjfvH2n\nzQ2359HemXLA6ScCCFSsQEF+vvP20484Z++9S+TVh+5xjjrySN8f69f5zznnnIrdMbU77du3\nt9gsinFwJL2AeyvkZWrp19HW/qShJThW7BhuK09orXVqKr5SuGVmdMSbJNstdfbc/EmKFdHl\nST0ozT2GSd0RGodAKgpkZWdP2Hn4KL4cNhUPHm1GAIGkE5j5zpvOE9dfGlm7Ypmz+7Bhvqef\nftqpW7ekdwolXXdSskEHHnhgzk033TRx06ZNV1ViB+wg2619dlJu57azFa8oFircUlsjdpVj\no+JRd2bMcH9N261gTyvWeJb11fgQRRfFYoVddbH6Iwq32K3yRynmK+wWteMV2yleUnygKK60\n10K7mtNZYdv8rLA+vKjw7mOApu3qznMKS0xGKbopPlG8qvhdUdJiHlbmbB389fMHjZmRnZvU\nUPyhKKrY54+sWFLslvzoSDV3hoYXKnIVN3rmJfUoCVJSHx4al+YCzfNzczv3HLxbmneT7iGA\nAAIVKzBv1peWGIUXzvnO1717d+fT99/1tWvXrmJ3Su1xBcaOHetceeWV9oWxzRVL4q6U2Jl2\nz+TdCkssLJkIK+zE3U7yz1Lco7BiJ/p2GbGNwhKKuQpvaayJJxVLFf+OLrAk4QbFKQqfwk7y\nLRGy8priSMUyhRVLJqwdzyvsikl/hZXDFGYRr1id1qbLFFavtd3mWVixByfspbC+WBmvOFNh\nycfFCktyViksGbO+/1Nxh6Ikxdp/nOJohSUwbjlAI9YXS+qKS45s/YUK228bm4gWd9wSRSut\nFLafyxVrFSlRtnXpLCU6QSMRSFGBfWrWqZfbqnP3FG0+zUYAAQSqVmDZwl+cO846MXz1keMj\nBevXOO/MmOH7ZtYskqMqPCx6kp3TuHFjSyTGVkIzOmsfjyjsasrJinqKWgq7EmRtsIRltMKK\nncg/ZCMqh24d/O3nQZqyxOphhSUqVq5UnKr4SNFTYYlJM8V/FCMUtu/Ysrdm2JUs2/Y8xRWK\nAkW8Yuteq1ih2EdRX2HJ1dGKxYqBiqMUseVmzXhf0SgafTRcqLhFMVhRknK9VvpaYQmaXamy\n/Vjf71FYYnuuYlvlT63wrMKSVLs6Z0mhJVu/KSwBs2KJ3DrFJJtIlUKClCpHinamnUAgGBqz\n47C9stKuY3QIAQQQqGCB9atXOY9fd0nkgnG7R37+6jPn0Ucf9S1evNg/ZMiQCt4z1ZdEYPz4\n8SF9aey4kqxbznUsAbKkxa4U2bidiG9SPKOwhMfKbQr3f60lAJYoHaKILXalx8pDW346TgcN\n7YrMr4pRitkK29auMB2n+EIxXGGJjbfYVad/KS5SXKe4U1FUOVALrM7zFa8orP3LFA9GQwNn\nJ/sRUywZGq+wq0dWvlIcrggq7lKUpMzTSpbkWXI2WvGAwq1jX41bUliSYn2do3hc8bGikcL6\nZVe9OiqOUFgSuEGRMoUEKWUOFQ1NM4GcSLhg2A6Dh/E7mGYHlu4ggEDFCeRu2uRMe+Au5+yR\nu0Q+fHGKc9555/rWrF7lP/TQeBcEKq4d1Fy8wIgRI3wFBQWDtJYlCxVV7GrPAIXdBnZfnJ28\noXnfKFopLNmxYonFdEU7RX+FW7pppLfiA4UlDlYs+clS2BWSeLeGPa35VvbcOvjbz5f+NlX0\nhCUkNRSWXHiLTxMLojNqeRdExy0Jyo+Zb1eUflDY1avqMcviTVqCNSu67uUaWlJjV39WK95R\nWNtKUuxqk1nuoNhF0VZhbbFymcISPjdps3UsYbTbEM9TVOTrQ9WXvVimSUEAgcoX2M3x+YJd\n+9v/DwoCCCCAQHEC4XDY+Xjqc87Tt1wVyf1zozNh/HjfQw895IRCoeI2Y1kVCey22262ZzvH\ntNu93rSJCihtVGe24kdFXhH1z9H8norOiu+j69jVmT0UllV/Ep3nXj2yZW7pEB0Zp+FAd6Zn\n6CYu7nruIkumVroTJRjaFS9LLPopLLmxqy6WsDVQWLFEMLZYv+KVnzWzi8L67PYt3no1NdOS\nMkuydlS4SaFGnXsVMxUPK+wqmeum0SJLoZZYsuUt1oaJilMUmxV9FO8qbN1PFVcrzNb6bvOS\nqvDudVIdDhqTQQL7dOrdrzC7Wkne5MkgFbqKAAIIxAh8/d5058Jxw8IPXn5eZKcdevl++/VX\n3xNPPEFyFOOUTJM1a9Z0+vXrZyffwyuwXY2idce7uuPudn10pK47Q0O7erFOYSfvlsTZufAh\nCrslbLLCLY2jI7kahuOE1fG5Yo3CW1Z5J7YxbvuwZOFDxc2KoxQNFa8oblIUVYrah/XBSr2t\ngyJ/2nGx5PJ+hTc5sg2sbkterOy7dVCmn1doq4WK/0S3viQ6tITS9n+Eoq/C7JOucAUp6Q4J\nDcoEAT3eex89vS6UCX2ljwgggEBZBBZ8P9t58sbLw/O+/sLXqVMn3/RZX/u6dbM31impIDBq\n1Kjsr776al99cezZJWivJSk7KqopPijB+raKXS2x0mLrIO7PltG5KzxLN2n8ScWJCjtRt+nm\nCrti4v2czAJNW7lTceuWscT/eEZVWpIwTXGj4jPFnworB20d/PVEu+jklkEz74Rn3Pphxa4A\nFVfsqo2VX7YO/uene4VqsJZc9z9Ltz3D+jRacbTCEmUrAxV2VWu5Tai8uHWw5crZI9HxpBnY\nC5KCAAKVK9BMj/du3aWv/a2gIIAAAgh4BVb+tti594LTwpcfsm9kvcanTZvm++GHH0iOvEgp\nMD58+HBHyVEnNbVpnOb6NK+H4jTFCwq7amG3c52pKGn5XSvaFSJLgtrE2aim5tltXVa+2Tr4\n66d7K91+mjMxOted5670Y3RkmDsjZjhG05Y8jYuZX9LJhlpxkMISovGKd6LjGmwpXaPDeLfY\n9Ysu8w7sKpmZ/qYwm+LKvOhC1yd2Xfe2waISqNj1Y6ev0gzzcxMfyzesfd6rbX9oOqJorEi6\nQoKUdIeEBmWAwO45NWrmtejYJQO6ShcRQACBkglsXL/OmXzL1ZHzRg+NfP/hDOeuu+7y/f77\nUv/ee+9dsgpYK6kEevfu7dSuXTtPjdoz2jBLluyqjd3GtkxhScup0fGTNGyiKG2ycZu2sQTi\nJkWWwlsu10QDxdsKu9XLW+xKzXcKu8ph+7SrUe8pvOVZTSxV7KuwZMhbrK0PKE5WVPcuKMW4\nexdJtraJTSJ7a94/onXZ8thijm1jZl6m6VqKKTHz402+pZm5CkvMhsSsUE/T50fnTY1ZVpLJ\noVrJPuN1qcL9bFFY478qLCl0i41borzInZFMQ26xS6ajQVsyQsDv9+/erf8gv89nfxcoCCCA\nQGYL5OflOjOmPOY8f9dNEUcPYzj1lFN8kyZN4g9kir8s7H/cwIED/a+++qqduF+jsNvC7GR4\nhuIsxduKJYryFKv3AIUlOR8pnlLYAwEs8RmumKmYoIhX7IqR3dZm5Q6FXc3wFrvCcZriScVz\niocVHyu6KiyxqK94TfG0oizFEoZPFP0VdhXtMYXNs9va7KrW74o6inhXWCy5ss8/3a5YqhgV\nDTM9W7GtYleQLAm6WTFNYX20vtkxsqTPhv9RvK4obblSG1jyG+tiSdmhip0UXylOUVgpyz62\nblmBP0mQKhCXqhGIJ+APZI3o1n8wv3vxcJiHAAIZIxCJRJzPXn/ZeeqmKyMb161xxowe7Xvk\nkUec6tXL+oZ8xtClTEd79eoVfP3115vqKYR2pcgSo/kJbrzdntZHcYPiSIWb8Pym8UcV/1Ks\nUcQrlpBcq7ArUI/EW0Hz7GqMJXX3KI5UHKWwslFh+7pEUaAoa7FEyBKRPRW9opVYe61dlrz8\noOiuaK1YoHDLeRrZVXGRwq9YobAE7nSF+5kfjRZbbtHSXxS3Ko6LhgZbHvNtyYv1ubRlb21g\nnx8Yq4jEbGxWuys+U1gi2FxhbU7KBMmnhlEQQKDyBDpoVz9d88IMp3GrNpW3V/aEAAIIJJHA\nD5995Dxx/aXhpb/M9+28807OM88842ve3M6XKOkkMHfuXKdjx47WpfaKRCdHsVSW6Nh+7MT8\np9iFCZiuqTo6KzYoLLHIVSSq2FWi1gpL7BYriiqWlJ2pOEbxgKKOorVitsJuYytrsf23VVji\nYvuPTW40q0TlBK3VUnFhEWvbVbeDFHaFyq5YTVUkZeFd7KQ8LDQqjQWG1aq3Xa6So+w07iNd\nQwABBOIKLPrxe2fKpGvC3338vq9d+/bOp59+4ttpJ7vjhpKOAh06dHDq1q2bu3bt2kHqX0Un\nSIXax48V6GiJ0RcVVP8y1WtR2rJOG8wq7UZx1i/r/mOrujd2Rsz0ak3fGTMvKSftshwFAQQq\nSUCfPxrcte8A3pioJG92gwACySGwZvnvzgOXnRO+9MCRkd/n/rDlitG8uXP9JEfJcXwqshVD\nhw4N6H/fkIrcB3UjkGgBEqREi1IfAsUI+LOydm2/4852GwAFAQQQSHuBTRs36OELN0fO2Wdw\nZOZbrzjXX3+9b+WKFf7x48enfd/p4FYBJUjBUCi0Ox4IpJIA72Sn0tGiraku0LggN7d5+172\n9E4KAgggkL4CBfn5znvPP+U8e/v1kcL8POfoo4703XPPPT5dSUjfTtOzuAKDBw+270NqpYWN\nFMvjrsTMkgo8pRXt80YflnQD1iubAAlS2dzYCoGyCAwIhrILWnTowu9dWfTYBgEEUkLgqxmv\nO09ef1lk7coVzp577O57+umn7ftwUqLtNDLxAnqSnZOTk5OvJMk+h/Rc4veQUTXaZ6Aq6nNQ\nGQW5rc7yVs62hFiOQOIEBrbp1rPQH+AOu8SRUhMCCCSLwNyvv3AuO3hU+M6zToq0aNLImfvT\njz59Bw7JUbIcoCpqR0D/83bZZRd7KtrgKmoCu0Wg1AIkSKUmYwMEyiaQFcrZrfNO/Xl6Xdn4\n2AoBBJJU4PeFPzu3n3F8+JqjJkTCG9Y77737rm/mzJm+Nm34KoMkPWSV3qwhQ4aEqlWrxoMa\nKl2eHZZVgFt9yirHdgiUTiBUUJDXs32vPqXbirURQACBJBVYv3ql88Ldt0TeffZJp269us6j\njz7qO/TQQ31J2lyaVYUCffr0cfLy8rqrCXYLhT2Om4JAUguQICX14aFxaSTQPRIOZ7Xp5n5R\ndhr1jK4ggEBGCeRu2uS8+fj9zsv/uT2i26ciF154gf+KK64gMcqoV0HpOmsJUmFhYUhbdVF8\nW7qtWRuByhcgQap8c/aYmQI71qq/3eaadevlZGb36TUCCKS6QLiw0Plw6rPOlFuuieRu2ugc\nsP/+vgceeMCnRzinetdofwULNG3a1GnQoEHuqlWr7FuBSZAq2Jvqyy/AZ5DKb0gNCJREoHfb\nbj15OkNJpFgHAQSSTuCbD2Y4F4zbPfzwFRdEdv7/9u4DTqrqeuD4fTOzRaVIR1CagIqIWGLD\nxBKNJUZNjEmMiX+jicaYponGaIoaW+wVBbsiIFVQQJEmVSnSe186LOwCy+5Oe/M/B3bNsiyw\nuzOz88rvfj6HmXnz3n33fO+yM2ffmzend7c2bthg9enTx1AcOW6qHDugb33rW/qek/PMHTtD\nDKyiAAVSRQ3uI5AmgaycnHPbnXxqVpq6p1sEEEAgLQKrF84zj918nf3CH29JNKl/pDVv7hxr\n3LhxpmnTpmnZH516V+Dss8/Okgs1nOfdDMnMSwKcYuel2SQXpwoE49Fol7YnnuzU8TEuBBBA\nYD+BbRvWmUEvP2nP+PRjq6WcHiWX67Yuu+yy/dbhAQI1ETjzzDP1Qg36QsiFGmoCx7oZEaBA\nygg7O/WZwAm2bee0PUkv4ENDAAEEnCtQtLPQfPLGy4kx/d429erVM6+99pp1++23cwEG506Z\na0ZWdqEG/aoLLtTgmlnz70ApkPw792RedwKnHVGvfrhR85Z8B1LdmbMnBBCogUA0EjZj+r1j\nhvd6PmHkKz3vvusu66mnnqIwqoEhqx5aQC/U0LBhw/DOnTv1cq5cqOHQXDybYQEKpAxPALv3\nhUDX4zrrH8xoCCCAgLMEEomE+WrUMNP/2UcTxbsKzTVXX7334gu5uVxw01kz5Y3RnHzyyYmp\nU6d28UY2ZOFlAQokL88uuTlCIBgKdZMCiaNHjpgNBoEAAuUCi76aYvo++aC9ee0q66xvfcsM\nHjzYatWqVfnT3CKQcoHu3bvnzpw58xT5LFLK+6ZDBFIpQIGUSk36QqAKgUAwdMox7TtW8QyL\nEEAAgboXWL98ien/zCP2oulTrI7HH29mTJ9unX766XU/EPboO4EuXboY+XLhbr5LnIRdJ0CB\n5LopY8AuE8iJhsOtW3WgQHLZvDFcBDwnsGPLJjP01WftKcMGWk2bNTND5IjRD3/4Qz5n5LmZ\ndm5CJ510kiktLT1WRqjfLsxhJOdOle9HRoHk+x8BANIs0MmYRIAjSGlWpnsEEDioQEnRbjPy\nndcSn77b2+hni5555hnr7rvvpjA6qBhPpEtAjyDJ5970Mt/y2mgWpms/9ItAsgIUSMkKsj0C\nhxboknPkUZGGTZrpX8toCCCAQJ0JxKJR88WQfmbIy08l4tGI+c2vb7VeeeUVKxDgO+LrbBLY\n0X4CLVu2NEcddVR0z549euUiCqT9dHjgJAEKJCfNBmPxosBJx7TrYHsxMXJCAAHnCswcM0o+\nZ/SfxM78rebyyy6z+vbtaxo0aODcATMy3wjIaXZxuVADV7LzzYy7M1EKJHfOG6N2iYD8pbZz\n644ncgU7l8wXw0TA7QLLZs8wff/7oL1u+WLr9NNOM4O/nGq1bdvW7Wkxfg8JdO3aNWf27Nkn\nxuNxD2VFKl4ToEDy2oySj6MEQlk5JzY/tg3n+jtqVhgMAt4T2LRmpRnw7OP23EljrDZSEE2a\nONHq0aOH9xIlI9cLdOjQwcrJyTmhuLjY9bmQgHcFKJC8O7dk5gAB2461bdr6OAeMhCEggIAX\nBXZu32aGv/aCPWFwX+voRkebPn36WDfeeCN/lPHiZHskp3bt2hk5etTGI+mQhkcFKJA8OrGk\n5QiBXPmQdONmrSiQHDEbDAIBDwmES4rN6D5vmk/eeDkRCgXNv/71T+vBBx+kMPLQHHs1lfbt\n25twONxE8tPTz8NezZO83C1AgeTu+WP0zhZoJ8OzOILk7ElidAi4ScCWz21MHjbQDHzxiURE\niqSf33CD9cYbb1ihEC/nbppHP49VjyBJ02JePxy3TB/QEHCaAL9RnTYjjMdLAu2DoVC8YdNm\n+p0PNAQQQCApgbmTxpl+Tz6U2L5pg7nwwgusAQMGmMaNGyfVJxsjUNcCrVq1MsFgUM6yi7eT\nfVMg1fUEsL9qCVAgVYuJlRColUD7Rs1bRizLOqJWW7MRAgggIAKrF8w1fZ96yF41f7alX7Q5\nYfQ8Sy6VjA0CrhTQ7+E65phjouvXr2/nygQYtC8EKJB8Mc0kmSEBuUBDG76RMUP47BYBtwts\nXZ9nBr7whD1rzEirVevWZvTo0dYll1zi9rQYPwLm+OOPtyiQ+EFwsgBv3pw8O4zN7QKtGrc8\nJtvtSTB+BBCoW4GiwgLT76mHEvdfc1Fi+Ywppnfv3taG9esDFEd1Ow/sLX0CUiBly2l27dO3\nB3pGIDkBjiAl58fWCBxUICsnp83RzVpwVamDCvEEAghUFIiGS82Y/u+a4b2eT5iEMX/5y93W\nk08+ye+Qikjc94SAnGJnZWdnH1tSUuKJfEjCewIUSN6bUzJyiIBlrFZHN23ukNEwDAQQcKqA\nbdvmy1HDzIDnHk0U79ppfnjttdZ7771ncnNznTpkxoVAUgItWrTQ7Vsm1QkbI5BGAQqkNOLS\ntb8F4vFY84YUSP7+ISB7BA4jsPDLyabvkw/aW9auts499xwjV6az9CpfNAS8LKAFklzFrpmX\ncyQ3dwtQILl7/hi9cwWy47FYfbnEt3NHyMgQQCBjAuuWLTb9nn7YXjLzS6tTp05m5KyZVvfu\n3TM2HnaMQF0KaIEUiUTqyz71azDidblv9oVAdQQokKqjxDoI1FxATx2wOMWu5nBsgYCXBXZs\n2WQGv/yUPe2TIVaz5s3NsI8+sq6++mo+Z+TlSSe3AwTKTrHTC4XpXxE3H7ACCxDIsAAFUoYn\ngN17VmDvCdYcQfLs/JIYAjUSKN69y4x4q2didJ83TO4RR5jnnnvO+vOf/0xhVCNFVvaKQFmB\npOnoayUFklcm1kN5UCB5aDJJxVECjQLyTeE5Rxyppw/QEEDApwKxaNRMGPSBGfLK0wk7FjO3\n33ab9dJLL1n6ZZk0BPwq0KhRIyOX+ZaPIcX3/jHRrw7k7VwBCiTnzg0jc7dAo9wjj4pJChRI\n7p5HRo9ArQVmfD7C9H/mkcSu7fnm+1deYX3wwQemXr16te6PDRHwkoAUSdH8/Hwu9eqlSfVQ\nLhRIHppMUnGUQKMj6tW3HTUiBoMAAnUisOzr6XJluofsdcsXW2ecfroZ9NU0q23btnWyb3aC\ngFsEGjZsmJACqaFbxss4/SVAgeSv+SbbuhNodFSDo+tub+wJAQQyLrBp9Qrz4bOP2fMmj7Pa\ntW9vpkyeLJfuPjfj42IACDhRoEGDBjqsvf84cXyMyd8CFEj+nn+yT5/A0fUaNeJDBunzpWcE\nHCOwM3+rGfrqc4lJQ/ubRo0bm759+1o33HADF2BwzAwxECcKHH300foaSYHkxMlhTIYCiR8C\nBNIj0Khew0b8/0qPLb0i4AiBcEmx+fTd3nuvTpeVnZV4+OGHA//4xz8ojBwxOwzC6QJSIOln\ndPW7kGgIOE6AN3COmxIG5AUBuTpPk6Ma7v3l74V0yAEBBCoI2PG4mShHiwa/9FQiUlpibvrl\nL6xevXpZoRAvqRWYuIvAIQXkIg0huZpjQ9vm47qHhOLJjAjw2zwj7OzU6wJWINQgJ/cIr6dJ\nfgj4TmDOF2NMv6cfTuzYtNFcfPFFVv/+/U1jOa2OhgACNROoX7++kT8qNIpEIjXbkLURqAMB\nCqQ6QGYX/hOwAtYRWTk5/kucjBHwqMCqBXNMP7kyndxaXU85xUwaM9o64YQTPJotaSGQfgG9\nSIOcbdEo/XtiDwjUXIACqeZmbIHAYQUsYx2ZnZN72PVYAQEEnC2wdd1aM/DF/9qzxoy0Wh97\nrPn888+t7373u84eNKNDwAUCWiBZlsVFGlwwV34cIgWSH2ednNMvYCVyQxxBSr8ze0AgTQJF\nhQVmWK8XEuMHvG/qNahv3njjDevWW2/lAgxp8qZb/wnk5uaaRCLBXxL9N/WuyJgCyRXTxCBd\nJ5AwR3AEyXWzxoARMNFwqfm879tmeO8XE5aUQ/fee4/1+OOPUxjxs4FAigXKLmrC+9AUu9Jd\nagT4wUyNI70gsJ+AbRI5fAZpPxIeIOBoAb2S1rRPhpiBLzyRKN69y1z/4+ust956y+hfuWkI\nIJB6gaysLO107z+p750eEUhOgAIpOT+2RqBKgYRt53IEqUoaFiLgOIEF0yaafk89ZG/JW2P1\nOO88M2DAAKtly5aOGycDQsBLAnoESU6x0+9CoiHgOAEKJMdNCQPyhID80g8E+e/libkkCc8K\n5C1dKIXRf+ylX39lndC5s/n066+tbt26eTZfEkPASQIcQXLSbDCWygK8g6sswmMEEEAAAU8L\n7Ni80Qx66Un7y5EfWS1atDQfDx9uXXXVVXzOyNOzTnJOE9AjSHJqK0eQnDYxjGevAAUSPwgI\npEdArl6ano7pFQEEaiegny365M1XEp9/8KY54ogjzPPPP2/96U9/4n9q7TjZCoGkBPQIkpxi\nx/vQpBTZOF0CvDCkS5Z+fS0QDIV2//a/L9c74+LLfe1A8gg4RWDOF2NM7wf+nCjdUyR/vLBM\n+/bt9TtYnDI8xoGA7wT27NljtmzZEpUiKdt3yZOw4wWo3B0/RQwQAQQQQCBZgUAwaE6/+DIq\nomQh2R6BFAnoqa5bt20ziXg8RT3SDQKpE6BASp0lPSGAAAIIOFSg2/kXGQ0aAgg4Q2DWuE/N\n8tkzws4YDaNAYH+BwP4PeYQAAqkS4PSdVEnSDwIIIICA1wTsvUeOLA4feW1iPZIPBZJHJpI0\nHCeQkO9CctygGBACCCCAAAJOEIjHYvI5QBNzwlgYAwKVBSiQKovwGIEUCFhWIBINc+ZACijp\nAgEEEEDAgwJ7jyAFAhxB8uDceiElCiQvzCI5OE8gYIWj0YjzxsWIEEAAAQQQcIBAPB4zAcMp\ndg6YCoZQhQAFUhUoLEIgWQH5pR/mCFKyimyPAAIIIOBVgX2fQeIUO6/Or9vzokBy+wwyfqcK\nhKMRTrFz6uQwLgQQQACBzArEolFjLMOpFpmdBvZ+EAEKpIPAsBiB5ASs0liE3/vJGbI1Aggg\ngIBXBcLFe4xJWEVezY+83C1AgeTu+WP0DhVIWIkSjiA5dHIYFgIIIIBAxgVKpUBKmMTujA+E\nASBQhQAFUhUoLEIgWYFEPFEcLS1Nthu2RwABBBBAwJMC4eJiY8ftnZ5MjqRcL0CB5PopJAEn\nCtjx2PbiIv4w5sS5YUwIIIAAApkXKC3ZY+J2jAIp81PBCKoQoECqAoVFCCQrYNvxguJdhXy/\nQ7KQbI8AAggg4EmB0qKiuLFt/pLoydl1f1IUSO6fQzJwpkDh7sJCviHcmXPDqBBAAAEEMixQ\nXLRL/4jIRRoyPA/svmoBCqSqXViKQLIChcW7dtrJdsL2CCCAAAIIeFGgRI8gGcMRJC9Orgdy\nokDywCSSgiMFdhbv5tRqR84Mg0IAAQQQyLjA7oLtOob8jA+EASBQhQAFUhUoLEIgBQKFcpEG\n/n+lAJIuEEAAAQS8J1BUWKCvkRRI3ptaT2TEGzhPTCNJOFCgUL4EL+TAcTEkBBBAAAEEMiqQ\nSCRM6Z6iLBkEBVJGZ4KdH0yAAulgMixHIDmBrfFYLFjCpb6TU2RrBBBAAAHPCRTv3mWkSOII\nkudm1jsJUSB5Zy7JxFkCm3U4u7bzxzFnTQujQQABBBDItEDZ5490GNsyPRb2j0BVAhRIVamw\nDIHkBbZKF4md2/ndnzwlPSCAAAIIeEmgqKCgPB3+ilguwa2jBCiQHDUdDMZDAtFgKLSbAslD\nM0oqCCCAAAIpEdAjSIFgsEQ6i6SkQzpBIMUCFEgpBqU7BMoFgsHQtp35HEEq9+AWAQQQQAAB\nFSjYutmEQllb0EDAqQIUSE6dGcbleoGEMZv4DJLrp5EEEEAAAQRSLLBDCqSESeSluFu6QyBl\nAhRIKaOkIwT2F4hHI3mF+VulTqIhgAACCCCAQLnAji2b7Gg4vKr8MbcIOE2AAslpM8J4PCNg\n2/a6bevzOL/aMzNKIggggAACqRDYtm7va+O6VPRFHwikQ4ACKR2q9InAPoE12zbk2WAggAAC\nCCCAwP8ECrZs1LMr1v9vCfcQcJYABZKz5oPReEtgTeG2Ldn6jeE0BBBAAAEEENgnIFd4zZZ7\nHEHiB8KxAhRIjp0aBuYBgTV2PB4s3KZfiURDAAEEEEAAgd0FO0w8FguKBEeQ+HFwrAAFkmOn\nhoF5QGCt5rB9I68BHphLUkAAAQQQSIGAnHpe3sua8jvcIuA0AQokp80I4/GSQEkwK6sgfyNn\nEXhpUskFAQQQQKD2Alvy1hh9bZQedte+F7ZEIL0CFEjp9aV3nwsEAqG8fI4g+fyngPQRQAAB\nBMoFtkqBFAgGV5Y/5hYBJwpQIDlxVhiTZwTi0fCiLevWciU7z8woiSCAAAIIJCOwKW+1HS0t\nXZBMH2yLQLoFKJDSLUz/vhaQ70JatG7porCvEUgeAQQQQACBMoFNK5fpa+JyQBBwsgAFkpNn\nh7F5QWDJ5rxVIS8kQg4IIIAAAggkK7B1fZ5ewY4CKVlItk+rAAVSWnnpHAGzJFJSkiXf+QAF\nAggggAACvhYo2llowsV79DuQVvgaguQdL0CB5PgpYoAuF5C/kln2ptV8HtXl88jwEUAAAQSS\nFNi0em9dpN+evizJrtgcgbQKUCCllZfOETDhUE7OxrIXBTgQQAABBBDwrcD65UuMvCZuEIA9\nvkUgcVcIUCC5YpoYpJsFEnF74aY1HEFy8xwydgQQQACB5AXWr1hqEvH4nOR7ogcE0itAgZRe\nX3pHwMRjkbl5ixdyJTt+FhBAAAEEfC2wdtG80ngsNtvXCCTvCgEKJFdME4N0ucDsvKULLZfn\nwPARQAABBBBISmDDyuX6vnN+Up2wMQJ1IECBVAfI7ML3AnNK5ao9OzZv9D0EAAgggAAC/hQo\n2LrZhEuK9Qp2FEj+/BFwVdYUSK6aLgbrUoFlgUAgnLd0kUuHz7ARQAABBBBITkAv0GBZgaj0\nwncgJUfJ1nUgQIFUB8jswvcCdjAra9HaJQt9DwEAAggggIA/BfLkNTArO1uLo7g/BcjaTQIU\nSG6aLcbqWoFoOPzl2sXz9S9nNAQQQAABBHwnsHL+7GgkXDrJd4mTsCsFKJBcOW0M2oUCc1Yv\nnMdfzVw4cQwZAQQQQCB5gRXzZ+tr4Izke6IHBNIvQIGUfmP2gIAKfL0zf2tuUWEBGggggAAC\nCPhKoHDbVlO0Y3uuJD3dV4mTrGsFKJBcO3UM3GUCc61AICKnGLhs2AwXAQQQQACB5ARWL5xr\n9GJF0gtXK0qOkq3rSIACqY6g2Y3vBaKhUPaclXO/9j0EAAgggAAC/hLQAikYyp4nWXOqub+m\n3rXZUiC5duoYuNsEopHSCUtnfaV/QaMhgAACCCDgG4EVc2ZF5DWQCzT4ZsbdnygFkvvnkAzc\nIzBt9aJ5QTvOH9DcM2WMFAEEEEAgGQHbts2qhXMs6YPPHyUDybZ1KkCBVKfc7MznAl/GIuHQ\n+hVLfc5A+ggggAACfhHQL4iNlJRkSb4cQfLLpHsgTwokD0wiKSQtEJQeNNLdNodycjaunMfn\nkNINTf8IIIAAAs4QWD57hpHXvvUymo3OGBGjQODwAk4ukN6U4Y+XuPMwaVxTtt6fDrOeH56+\nq8ziIgcl203G8qzEWInPJB6WaClxsJYtT+icfyAxRWKIxIMSTSRq0k6Xlb+UKJbQq+bcIFFV\nO1oW5kvoGNPe7Fhs4rKvp3OOXdql2QECCCCAgBMElsycFotFo2OcMBbGgEB1BZxcIH1LkrhQ\nQt+4dpU4WGstT1wo0flgK/houRpcKNHcITn/UcahXwqnhdspEjq+f0powdJRonI7Thbocy9L\n/FBCi6JrJf4toeelVbfwayrrTpBoJfEvCS2A+kp8X6Jyu0cWZEs8VvmJdDyWzx+NXThtUiwd\nfdMnAggggAACThNYPH1q3Nj2F04bF+NB4FACTi6Qysetb17fkQiVL+DWFQJnyyifk9CjJTdK\naNHWXuJ8iYjEcIkGEhWbHjU6XuI1CS2OTpTQbbR40cf6fGOJw7XfyAr1JX4i8bTExRKbJe6T\nqNh0THrk8SWJLRWfSOP9cUU7C3I2rVmZxl3QNQIIIIAAApkX2Lx2lSnevStHRjIx86NhBAhU\nX8ANBVKRpHOGROU3t9XPkjUzIfCI7FR/vh6S6FthAFPkvhYlJ0lo8VLe9PG3JfQI0u8lSiS0\nrZV4QGKoxDESl0kcrunRKm2z990YPWIzX6JL2ePym7/LHS3gnixfkOZbLf4utqxAycQh/dO8\nK7pHAAEEEEAgswLLZk03oaysbTKKVZkdCXtHoGYCbjgqo4WR/oX/nxLDJPSNbk3aBbLydyQ6\nSOib7ZkSIyQSEjVp+lmVn0nom2x103GMlNA+y9t35U4niS8kFpcvrHDbXu7rG3w9fPB5heV6\ntOMaie4SR0nMldC+8yQqNv0rzK8kdHvN4zYJPZ1Mj8ZMljhY0wsQXCmhY+8ssUtiucQoidUS\nFdu58uBUiSES9SSukjhZQj/Po+vrkZjqtNPKVupdxcpa7NgSP6jwnNoskdB9aNFSuc2SBXra\n3QmVn6jisearfeiRqvIWlTu55Q/k9liJ30o8IbFDIh1N93GxxEVlt23kdmMiYW9av2KJ3nfD\n/z8ZJg0BBBBAAIGaCyyeMTUul/keV/Mt2QKBzAq44Q3aF0Kkn0n5g8S7EmdJ6BGBwzU9fet9\niavLVtQ3yFll98fL7c0SlQuQsqcPuPmxLHlVQosRLaz0zb2+Cd8j8VeJ1yS0HSeh6+kRkxsl\nKre7ZcHvJbSwKW895I6OUwsE7VvHmS2hRYyuq8+VNy2etH8tMPRoyjkS2n4poW/Gq2paULwn\noW7a1K583ovl/q8l+kmUt+vkzl8kjpD4l4Q6bpfQMev4/iih83GopuNsIrFToqCKFbVw0T5b\nSqjZOgktCDUO1rqWPbH8YCtUWL5a7uv8aBGytmy5+q4qu683/5DQ+XtOH6SotZB+LioLLYw6\nSmyTmCDxuMR4iaUSN62aP+f1RCJhLMuShzQEEEAAAQS8JaCvcfMmT4jLZ28/9VZmZOMHgYBL\nkrxPxrlSQo9K3F/NMT8v62lxNEniFAk9etBBYqCEvokdLFGd/E+U9d6TqCfxO4lGEvUlrpcI\nS2jBUl6Ead9FEnphAV2/YtPi7GcSJRIflj2hRc4oiVYSf5LQYkT7vlxC37zrfs+TqNyukAV6\nNOgRCbX5j8TBisb35Tktjl6Q0CJDC5/zJd6SOFLiGYnywlHuftOelXtq17wszpDbtRJaUHxb\n4lBNx75DomFZVF5XC6hmZQsbV36yiscnybIfS6jdBInDtQ9kBVviRQnd9p9lt2/KrTb9ObhV\n4kkJLURT0YZKJ5sl9OehhYQWkd3K7v9Ebl+T0OJI27iSot3ZG/g+pH0a/IsAAggg4DmBvKUL\nTUnRLv2D72jPJUdCnheoToHgBIRiGcQtEnoEQ//yf6rEodoF8uSvJNZLfE9igYS+YdYjC/pm\ndZrEmRL6JvlwTd/walHxVwm9r0dFSiQGSdwgoU3fiGuRoYXBAAktPLRIqti06GkqMUSi/E35\nY3JfCyLNSfsokohIfCbxGwltWthUbrmy4C4JfeP/X4lXJKpqJ8tCLYqWSOj4F0poITVF4o8S\nmocWaa0lKjcthvRo0vayJ76W25skQhI9y5Yd6mZW2ZN/qGKl2yssO7rC/aruaiH1kYTu9wGJ\nDRKHa/NlhT9LXCGxSOJhibclyp0elPv5Ei9LpKo9Ix2dKaFHznTudd50HPozW7mtz8rJyVv4\n1eTKy3mMAAIIIICAJwQWTJ1o5LVOz/rY6ImESMJXAm4pkHRSJkq8JKGFyDtlt3JTZdMjJNr0\nDXHp3nv7//NU2cNL9198wKOgLDlXYrfE6wc8u++vIvNkeVuJTmXP65EZbb/Yd/PNv1pcaNM3\n6uXtKrmjb6DLtylfrrf6F5cCCX3TXVURMVyWH65pQaRHa06X0MKoYlPHFWUL6ld8ouy+FkHR\nSsv1iNJiiS4SWgQeqt0vT9oSD0r8R6KzRAeJv0k8KrFVQpsWhAdrLeUJddBtP5aoqliUxVU2\n/VlpJXGxRHuJWyQ0Hx37jRJ6yluxhM6xFpuDJXpLnCdRm6bVziwJzfmwLRoOfzRnwueRw67I\nCggggAACCLhQYN6kcWF5rdPXbhoCrhMIuGzEf5fx6pv67hIPHGLselqVtqX7bg74V4+oaDtx\n381B/9U31jkSqyUO9ma2cl9TZN1lEpdItJDQpgXODyT0qMw4CW26rKmEvqEeLTGzUuhRLi1i\ntHXad/PNv4VyT4+AVKdpAaZHwLQo0CNWegRLC6ftEqdIaNMioXIrz6vy8lWyQH9uulV+otJj\nzec3EmGJf0joXKyUeFTidxKzJbTt3HdzwL86N2qgcz1M4scS1So+ZL3ypkbjJdaUL5DbhyU2\nSPQqW/aB3D4rocXY9yUmSlwrke42cvmcmcHS4j3p3g/9I4AAAgggUKcC4ZJis3L+7JDs9LM6\n3TE7QyBFAm4rkPQv/rdI6Jv++yX0zXNVrXnZQi0kqmq7yhYeXdWTFZYdrh9dtaq+3pHlWnTc\nIKHtegkttN6V0LFrKy+etPDSN/5VhR6tmSFR+WiNFjfVbb+XFfMk+kj8XeICCT1685CE9n+w\ndrB9lL+jb3SwDSssf0vu6+mQf5V4XeJOCS183pZoK6Ft276b/f49Wx5NlWgn0VPiOgl1Srbp\nkbQfSegRrbCEPv6pxFMSPSSOl1gq8YxEutsEk0jEFnGaXbqd6R8BBBBAoI4Flsz8Ul7iEnHZ\n7aQ63jW7QyAlAm4rkDRp/c/2okSWxDtlt3KzX1tV9ui4/Zb+70GbsrtVvTn/31r/u+rZwfrR\ndavq6z1Zrr8Yfq4rSPuFhBZG70iUNy1adJm+UT/rMPGFPF+bpkex1Er3o0dttABoInGRxMMS\nQQlt1r6b/f5ttd+j/z04tuxu+RGg/z1T9T094qcFx20SWuzo4/oS7SXWSFSeAx3bGAktXu+V\nuFNCLVPRHpFO9Gfj7bLOtCjSNnzfzd7TMT+T+x0kmpUtS9dN2AqExs2dNN5O1w7oFwEEEEAA\ngUwIzJ00LhEMBifLvksysX/2iUCyAoFkO8jQ9vfLfvWNth6d+HMVY9CjANou2HdzwL8Xli2Z\ne8Az+y/YLA93SWgRpG/oK7d6suCMsoXzKjy5Qe6PlviWhB4N+bbEFxKrJcqb/tLQIkkLgdPL\nF1a4zZb7WlA8KqFFTW3aD2UjS0Iv5PCqhBYH5U2PaGnBpK28UNr3aN+/Ou7KTcd6isRGCbU5\nVHtAnpwooZ+hqtxukAW6//GVnrhYHo+U0Nx/KqFHdlLVtBjSizY8KBGT0FbuWrDv4d5/db61\ntdh3k75/47HI8NnjP4umbw/0jAACCCCAQN0LzBozKhKLRgfX/Z7ZIwKpEXBrgVQs6f9KQv/6\n3k6icntfFuib3pskzqn0ZHt5rKd86bbvVnquqocvykItIPQoSFalFR6Wx/ome5zE2krPlR+l\neFOWWxLljyuu9lLZA91H5dPoHpRld0joZ292SNSmaaGhrd3ef//3j46nt0R5YaTFSuX2W1nQ\nodLCh+SxHv0ZWGl5VQ/1qI8WhlooVWxt5IEWuOr/dIUn9HNSb0jkSvxKojr7kNWq3fTo0SKJ\nvhW2yCu7X/FoUfOyZesqrJeuuyOLCgty1i1bnK7+6RcBBBBAAIE6FVizaL7ZXbBd31cMr9Md\nszMEUigQSmFfdd2VHrp9QeKuKnasH86/V0KL5L0b9gAAOCVJREFUgC8keknMkegooUVHQ4lb\nJfS5w7XHZYWfSOjRmKkS/SVKJa6W+J7EbAktYiq3YbJgu8TJErslqvpLihZGP5fQoxvazzsS\nuu73JS6ViEjoqXEJido03eevJX4poQWJnrrWTkL7P1dilYQWQVUdLdHiaoaEFnGbJK4qi3Fy\ne4/E4Zraq/U1EqMlhkq0k7hRorWEFk5asJS3v8id9hJaWD1UFnJzQHtflmhhWpN2iax8oYTO\nkzqUty/kjh5Nuk3iSwl10HnV+zsl0t3y5BKoy+Z8MabzcZ1PSve+6B8BBBBAAIG0C8yeMFov\n771ArmC3Pu07YwcIpEkgkKZ+66pbfZO9/CA706MR+sZ4tcQfJN6U0CNH+qZcC4aqjujI4gNa\nsSw5Q+I1ia4SetTj5bL7+mZdC5kCicpNi5u+ZQsHyu2eyivI46jEeRJPSbSSeExCC5LLJb6W\nuExirERt26eyoRZYmsNvJD6U0IJPi4TTJP4loe0H+272+/c+eTRZ4p8SmvvZEu9KXCeh4z5c\n06NemsdECTXqKXGvhBZ7WpxqrhWbzpW2oETHQ0TFoz26fnXao7KSFqBDKq28Uh4/IqGn/OnP\nyTKJXInfS9RJkxeQfl+NGlZaJztjJwgggAACCKRZYMbnI0rltW1AmndD9wikVcBKa+/O6byR\nDOV4CT2XqapCpbojLX/zrm/y9c10KpvORTsJLQDWSGyVSFXLkY46SBwhoQYlEgdrWgDq0Rwt\nYt6SaCjRTmK+RMWjL/Kw2q2lrKn715y0EIlL1FU7Tnb0N4l+ElMOstOLZLnGdolBEhsk6qqd\nIjua98THE03zY9vU1T7ZDwIIIIAAAikX2LZhnfnbVd/WfrtJ6PsGGgKuFAi5ctQ1H7Qe4ZlZ\n880O2ELf2C89YGlqFmjRpcWDRqpbWDrUwqg2TU81m1ubDStss1nua2SirZOdHu6I0HhZRyMT\nbX5Wdk7erLGj2lzxf7dnYv/sEwEEEEAAgZQI6Ol1oeycjbFImOIoJaJ0kikBt59ilyk39otA\nygSikXDfrz4drkUsDQEEEEAAAdcK6GuZFEd6Oj8NAVcLUCC5evoYvEcEBuctWZi9Y8smj6RD\nGggggAACfhPQ17DVC+ZmS979/ZY7+XpPgALJe3OabEb6i+1miYnJdsT21RaYKackbJ419tNq\nb8CKCCCAAAIIOElg+mcfy+l12RtlTNOdNC7GgkBtBCiQaqPm7W30s1rvSqzwdprOyk5OSeg3\n7ZMhnGbnrGlhNAgggAAC1RSY+smQ0lgk8l41V2c1BBwtQIHk6OlhcD4S6LNm8fycrevzfJQy\nqSKAAAIIeEFgS94as375klzJhdPrvDCh5GAokPghQMAZArPlanYrp40Y6ozRMAoEEEAAAQSq\nKaCn18lr2CpZfV41N2E1BBwtQIHk6OlhcH4SkKvZvTll+AC+NNZPk06uCCCAgAcEpn4yuFRe\nw97xQCqkgMBeAQokfhAQcI5A3/yNG3JWLZjjnBExEgQQQAABBA4hsHbxAiOn2OnpdfqF7DQE\nPCFAgeSJaSQJjwislSsAfSUfdE14JB/SQAABBBDwuMDkYQNsfe2SNLm4k8fn2k/pUSD5abbJ\n1fECcgWgt+RzSNF4LOb4sTJABBBAAAF/C8hpdWbyx4Nj8trVy98SZO81AQokr80o+bhdYGBp\n8R4zZ+IYt+fB+BFAAAEEPC4we8LnJhoujUuaAz2eKun5TIACyWcTTrqOFyi0LGvA+AF9Io4f\nKQNEAAEEEPC1wMTB/SImkRggCEW+hiB5zwlQIHluSknI7QJ2PN5r0VeTs7Zv2uD2VBg/Aggg\ngIBHBQq2bjaLpk/Nsm37TY+mSFo+FqBA8vHkk7pjBSZnZWevmvTRh44dIANDAAEEEPC3wJTh\ng+S7j7LWicIkf0uQvRcFKJC8OKvk5HqBaCTyyviBH4TlaJLrcyEBBBBAAAFvCchRIzP2w/fC\n+lrlrczIBoF9AhRI/CQg4EyB94p2FgTmT/3CmaNjVAgggAACvhWYO3Gs2bUjX99DvuVbBBL3\ntAAFkqenl+RcLLA9YFlDx334XtTFOTB0BBBAAAEPCozp93ZELiik54HnezA9UkLAUCDxQ4CA\nQwXi8fhL86dMCG1dn+fQETIsBBBAAAG/CWxZu9os1oszyGuU33InX/8IUCD5Z67J1H0CerGG\nhWP7vZNw39AZMQIIIICAFwXGDXg/Ia9NCyS36V7Mj5wQUAEKJH4OEHCwgHwA9ukvhvSL6ZfH\n0hBAAAEEEMikQLikxEwc2j8mr03PZXIc7BuBdAtQIKVbmP4RSE6gfzwWLdLLqdIQQAABBBDI\npMC0kUONVEelMob+mRwH+0Yg3QIUSOkWpn8EkhMIx2Oxlz57//VwIsGZdslRsjUCCCCAQG0F\n9DVo1Nuv7n1Nkj5KatsP2yHgBgEKJDfMEmP0u8Cr2zdtDMoFG/zuQP4IIIAAAhkSmPPFGJO/\naYO+b+TiDBmaA3ZbdwIUSHVnzZ4QqK3AZitgDRz59quR2nbAdggggAACCCQjMOKtnhHLmD7S\nx+Zk+mFbBNwgQIHkhllijL4XkMupPrHs6+lZqxfO870FAAgggAACdSuwasEcs2r+7Czbtp+u\n2z2zNwQyI0CBlBl39opATQXmBUPZn3/8xkt8cWxN5VgfAQQQQCApgZFvvRqT16DR0smipDpi\nYwRcIkCB5JKJYpgIxGORh+dM+Dy0ac1KMBBAAAEEEKgTAf2y8tkTRgfkNeiJOtkhO0HAAQIU\nSA6YBIaAQDUFpoSysqaPfKtnvJrrsxoCCCCAAAJJCXzy1it2MCtLz++ekFRHbIyAiwQokFw0\nWQwVgVg0+vDUEUOtHVs2gYEAAggggEBaBbZv2mCmDhuUiEUi/0zrjugcAYcJUCA5bEIYDgKH\nERgZDIWWffbe6/Zh1uNpBBBAAAEEkhL45K2ediAYXCydfJJUR2yMgMsEKJBcNmEMFwH5S96D\n4wa8bxdu2woGAggggAACaREo2LrZTBraPxGLcvQoLcB06mgBCiRHTw+DQ6BKgQHyvUgrR8h5\n4VU+y0IEEEAAAQSSFJDv3tOjR8ukm2FJdsXmCLhOgALJdVPGgBEwej74/eMH9knwWSR+GhBA\nAAEEUi2gZyhMGPRB+WePEqnun/4QcLoABZLTZ4jxIVC1wNBAILjk49df4op2VfuwFAEEEECg\nlgIj3+5pW4GAfqfEkFp2wWYIuFqAAsnV08fgfSyg54XfN3Fofyt/4zofM5A6AggggEAqBfI3\nrjfjPnxfjx7dJ/1y9CiVuPTlGgEKJNdMFQNF4ACBT4LB0LzhvV/kKNIBNCxAAAEEEKiNwNBX\nno7LZ4/0e4+G1mZ7tkHACwIUSF6YRXLwrYAeRZoyfLC1aY2eCUFDAAEEEECg9gLrVyw100Z+\nFJDXlrtr3wtbIuB+AQok988hGfhb4LNgKDhlwLOPRv3NQPYIIIAAAskKDHz+8WgolDVe+pmQ\nbF9sj4CbBSiQ3Dx7jB0BEYhFo3+aO2lccNnX0/FAAAEEEECgVgIr5swy86dMCMVi0b/WqgM2\nQsBDAhRIHppMUvGtwGw5X/zDvk8+FPGtAIkjgAACCCQl0P/Z/0SCweAg6WR2Uh2xMQIeEKBA\n8sAkkgICdjx+37rli830zz4GAwEEEEAAgRoJzBwzyqxeMC8Qj8f/XqMNWRkBjwpQIHl0YknL\ndwJ5Cdt+/sPnHgvLh2t9lzwJI4AAAgjUTiAaCZu+Tz0UTiTs56QHrvhTO0a28pgABZLHJpR0\nfC3w2K7t28Jj+7/rawSSRwABBBCovsDo998wu3fkF8sWj1R/K9ZEwNsCFEjenl+y85fAzngs\n9vchrzwT3bl9m78yJ1sEEEAAgRoLFG7baob1fjEmrx33ysa7atwBGyDgUQEKJI9OLGn5VqCX\nnGq3fOBzj/Hlsb79ESBxBBBAoHoCg158ImYSiaWy9lvV24K1EPCHAAWSP+aZLP0jEJfPIN0+\ndcTQwIq5s/yTNZkigAACCNRIYPXCeWbqJ0OC8prxO9nQrtHGrIyAxwUokDw+waTnS4HJctnv\n/u89cn/YtnnN8+VPAEkjgAAChxDQ1wZ5jYgEQ6GPZLWJh1iVpxDwpQAFki+nnaS9LiCX/f7L\nxtUr7IlD+nk9VfJDAAEEEKihwBeD+xr5aoi4fPbojzXclNUR8IUABZIvppkkfSiwSYqkfw14\n/vFIUWGBD9MnZQQQQACBqgT0Ij4Dnns0Kq8R98vz66tah2UI+F2AAsnvPwHk72WBF+Tc8rX9\nn36YCzZ4eZbJDQEEEKiBQP+n/yNXrYsvl01eqsFmrIqArwQokHw13STrM4FoLBK5WS7YYC36\naorPUiddBBBAAIHKAvpa8NWnwwPyx7Nb5Dn+eFYZiMcIlAlQIPGjgIC3BaYGAoHebz54TzhS\nWurtTMkOAQQQQOCgAtFI2Lzz8N/C8prwhqz01UFX5AkEEDBBDBBAwNsCiURiUrS09DZ5cTzy\n5HO+bXk7W7JDAAEEEKhKYNhrLyTmT5mwSz579AN5nr+YVYXEMgTKBDiCxI8CAt4X2CVfjnTb\nZ+/1TuQtXej9bMkQAQQQQGA/Af3dP+LNVxJy1brb5InC/Z7kAQIIHCBAgXQACQsQ8KTAsEAg\nOPzNf/01In899GSCJIUAAgggcKBALBo1ve//k55ap995NPjANViCAAKVBTjFrrIIjxHwqEAi\nYU8sKiy8I5SVldP59LM8miVpIYAAAghUFPj4jZcSs8eP3iNfenSZLC+u+Bz3EUCgagGOIFXt\nwlIEvCgg340U++3Qns/G1y1b7MX8yAkBBBBAoILA+uVLzPDeL8qv/tjtsnhbhae4iwAChxCg\nQDoEDk8h4EGBvpZlfdzr738I62kXNAQQQAABbwpIUaSn1kXk1LqRkuGH3sySrBBIjwCn2KXH\nlV4RcKyAnGo3tnj3rt/Go7HcLmf34Kp2jp0pBoYAAgjUXuDj119MzBr7WbEcPvqe9LKn9j2x\nJQL+E+AIkv/mnIwRyJe/LP5q5Ns9Eyvnz0YDAQQQQMBjAvq7fXivFxJSHN0qqW3xWHqkg0Da\nBSiQ0k7MDhBwpMAwKxDo2/vvf+QLZB05PQwKAQQQqJ1AafEe0/PeO8OWFegjPQyqXS9shYC/\nBTjFzt/zT/Y+FpAvkB0XLim9dWf+liO7X3AJfyzx8c8CqSOAgHcE3v3P3+Mr58/ZLEePvi9Z\nRbyTGZkgUHcCvCmqO2v2hIDTBHbGY5HrJ370oTVzzCinjY3xIIAAAgjUUEB/l0/9ZIiJRyM/\nkU2Larg5qyOAQJkAR5D4UUDA3wLrJP3EvMnjzj/nimuDR9Zv4G8NskcAAQRcKlCwdbN55re/\niMaikUckhQ9cmgbDRsARAhxBcsQ0MAgEMirwqFy0Ycar9/4uYsfjGR0IO0cAAQQQqLmA/u5+\n7W+/j0hxNFe21gKJhgACSQhwBCkJPDZFwCMCCdu2P9tdsP12KZSyTzrrPC797ZGJJQ0EEPCH\nwJBXnknMGD2iWH6HXygZF/oja7JEIH0CHEFKny09I+AmgQ3ywvrLT958JbF4xlQ3jZuxIoAA\nAr4WmD9lghkhX9sQj8duFIg8X2OQPAIpEuAIUoog6QYBDwgsNYFAkznjR5927vd/FDziqHoe\nSIkUEEAAAe8K7NiyyTx1288jsUj4Wcmyp3czJTME6laAI0h1683eEHC2gG3/NVJaMuflu38j\n57JHnT1WRocAAgj4WECO+ptX/nJ7JBou1W/8fsDHFKSOQMoFOIKUclI6RMDVAvJxJHukfB7p\n13sKC3O6nX8Rn0dy9XQyeAQQ8KrAwOcft2dP+Lyo7HNHO72aJ3khkAkBCqRMqLNPBJwtsFuu\n2vDVmkXzbmrRpn3g2E4nOnu0jA4BBBDwmcD00Z+YD599NCG/q6+R1Of4LH3SRSDtAhRIaSdm\nBwi4UmCtjLp47sRxF5920fcCDRo3dWUSDBoBBBDwmkDe0kXm+T/cHJNLe98vub3vtfzIBwEn\nCHD6jBNmgTEg4FCBYDA0tFHzllf8u/+InKMaNHToKBkWAggg4A+BosIC86+fXh7ZvT1/WDwe\n/4k/siZLBOpegIs01L05e0TANQJy2dibCvO3rn35L7dH9QPBNAQQQACBzAjo72D5XRwpKihY\nIcXRzZkZBXtFwB8CnGLnj3kmSwRqKxCx7fgnhVs33yoXbsjq9u2L+aNKbSXZDgEEEEhCoP/T\nD9tfj/9sTzwW/bZ0k59EV2yKAAKHEaBAOgwQTyOAgCmUDwJPWbNowf81aNwk0P7kbpAggAAC\nCNShwKRhA8zQV57Wq4xeKbudW4e7ZlcI+FKAAsmX007SCNRYQL6dPbFh/uQJP+h8+resZq2P\nq3EHbIAAAgggUHOBxTOmmp73/C6eSCT+IFsPqnkPbIEAAjUVoECqqRjrI+BfAfkywkSjr8d+\nesaZl34/WK/h0f6VIHMEEECgDgQ2rlpunrrthqh8cfdLsrtH62CX7AIBBESAq9jxY4AAAjUR\nkAvbZY9s1Kz5Bf/8YHhO/UaNa7It6yKAAAIIVFNg14588+ANV0V278gfLRdo0O87squ5Kash\ngECSAnzgOklANkfAZwJyYbvIdXJlu2XP3vl/kUhpqc/SJ10EEEAg/QL6u/W5O2+WK9ZtXyTF\n0U9ljxRH6WdnDwh8I0CB9A0FdxBAoJoCRbFo5Hvrly/d1vOeO6LyZYXV3IzVEEAAAQQOJyCf\nNTK9H/hTbP2KZflyat0Vsn7x4bbheQQQSK0An0FKrSe9IeAXgaKEHR+Rv3H9r3bmbwud+p3v\n8scWv8w8eSKAQFoF+v73QfvLUR+V2vHYBbKj1WndGZ0jgECVAhRIVbKwEAEEqiGwXS7/PXHt\nkoU3hUKhYOfTz6rGJqyCAAIIIHAwgeG9X0h8+l7vmPxuvVzWmXmw9ViOAALpFaBASq8vvSPg\ndYF1JpFYuHjGtOsbt2xltT3xZK/nS34IIIBAWgTGD+xjBj7/uC2n2P1YdjA6LTuhUwQQqJYA\nBVK1mFgJAQQOIbBEntsy94sxV7bq0NlqfXynQ6zKUwgggAAClQVmfD7CvPnve2z5g9Ov5bkP\nKz/PYwQQqFsBCqS69WZvCHhVYJYkViLfkXRx+66nBlq0aefVPMkLAQQQSKnAwi8nm5fvvl2+\nCNa+TzrumdLO6QwBBGolQIFUKzY2QgCBKgSmyqkhWTNGjzjvxDPPCTQ5pnUVq7AIAQQQQKBc\nYMXcWUa+MiEWj0Wfk2UPlS/nFgEEMitAgZRZf/aOgNcExieMaT790+GndT3vguDRzVp4LT/y\nQQABBFIisHrBXPPU7T+PxqORd+SPS39MSad0ggACKRGgQEoJI50ggMA3AonEp/Ji3/HLT4d1\nkct/Bxs0bvrNU9xBAAEEEDBm7ZIF5r+//lk0Fol8YNv2bzBBAAFnCVAgOWs+GA0CnhCQAulj\nuUxtt2kjhnakSPLElJIEAgikSGDdssXmiVt/Go2GSwfYdvxX0q0ceKchgICTBCiQnDQbjAUB\n7whIjWQPScTjp+wtkr59cbBBE44keWd6yQQBBGojsGHlcvP4LT+OREtLhtnx+C+kD7s2/bAN\nAgikV4ACKb2+9I6AnwX0+zy0SOo6beRHnbqdf1GwYZNmfvYgdwQQ8LHAxlXLzRO3XB8JFxeP\njMfjPxOKuI85SB0BRwtQIDl6ehgcAq4XkBppb5F0shZJp/S4MNiwKUWS62eVBBBAoEYCeUsX\nypGj8uIo9hPZOFajDlgZAQTqVIACqU652RkCvhTQImmonE7SRYqkE07pcYEUSc19CUHSCCDg\nP4FVC+bsvSBDJFw6RH4P6pEjiiP//RiQscsEKJBcNmEMFwGXCuwrkuz4CVM+HnziCWeeE2zS\nspVLU2HYCCCAQPUEln09fe+lvOVqde/L1epulq34zFH16FgLgYwKUCBllJ+dI+Argb1Fklzd\n7phpI4ac1q5Lt0CLNu18BUCyCCDgH4GFX042z915U0xaL/m9d4dkztXq/DP9ZOpyAQokl08g\nw0fAbQJSJY2Qv6TmfPXp8B6tOnQMtD6+s9tSYLwIIIDAIQVmjx9tXrzr1/F4LPaMSST+csiV\neRIBBBwnQIHkuClhQAj4QmCcvGnYM3PMqEsaNWtptT2pqy+SJkkEEPC+wBdD+pnX/3GXLUeN\n/i3ZatAQQMBlAhRILpswhouAhwSmSS4b5kwae1VWTq7VqfuZHkqNVBBAwI8Cw3o9nxjw3GP6\nFQe3S/4v+NGAnBHwggAFkhdmkRwQcK/AbBn6osUzpv5w945807XHhQHLstybDSNHAAFfCshp\nw+a9Rx+Ij/7gzZgUR9cJQn9fQpA0Ah4RoEDyyESSBgIuFlgkp9uNX7tk0fVrFs4NnHbh94Kh\nrCwXp8PQEUDATwLRcKnpec8dsRmjRxQn7PilkvtYP+VPrgh4UYA/1XpxVskJAXcKnBAMZY89\ntmPnZne98m52g8ZN3JkFo0YAAd8IFBUWmOf/+KvI2sWLdsRjkYsl8cW+SZ5EEfCwAAWShyeX\n1BBwoUALOXr0Wf0mzU6897UPclq0be/CFBgyAgj4QWDT6hXm6d/dFN69PX95LBq5XHLe4Ie8\nyREBPwhwip0fZpkcEXCPwB45l//9aGnJWZOGDWzb8dQzgk1bHeue0TNSBBDwhYB+x9FTt98Q\nLSna/Xk8FtXiaLsvEidJBHwiQIHkk4kmTQRcJBCVy+P2j8VjzaZ+PPiMhk2bBdp1OcVFw2eo\nCCDgZYEJgz4wve77ox2LRl+Q31W3SK4RL+dLbgj4UYACyY+zTs4IOF8gIRduGCWxad6kcVfu\n2p6f6HrudwLSnD9yRogAAp4U0CvVffjMf+yhPZ8tv4z3fyXRhCeTJSkEfC7AZ5B8/gNA+gi4\nQODbwVBo+PHdTj/y98/2zq7X8GgXDJkhIoCAlwSKdhaaV++9M7p01leldjx2jeQ23kv5kQsC\nCOwvQIG0vwePEEDAmQJtQ1nZo+o3adrh7pffzWl9fCdnjpJRIYCA5wTyli6SK9XdEikq2LFK\nLsZwlSS40nNJkhACCOwnwCl2+3HwAAEEHCqw07bj70RLSk6dOLR/xxZt2gVadzzBoUNlWAgg\n4BWBL0d+ZF740y2xcPGej+Kx2JWS11av5EYeCCBwcAEKpIPb8AwCCDhLIJpI2B/a8Xhk5thP\nL5bvH0l0Oft8KxDk15izponRIOB+ASmG9n7eaNCL/5W/zcT/lkgk7pasou7PjAwQQKA6Apxi\nVx0l1kEAAacJXCifSxp8XOcu9X7/bK/sxi2Ocdr4GA8CCLhUoHDbVtPznjsiqxfOLZZC6UeS\nBp83culcMmwEaitAgVRbObZDAIFMCxwjXyo7NCsn97TfPfVq9snnnJ/p8bB/BBBwucCCaRPN\na3/7fSRSWjJfLuP9Q0lnnctTYvgIIFALAc5NqQUamyCAgCMEiuSyu+/EIpGjp40ceracAmM6\nn3aWZXEpcEdMDoNAwE0CekrdkJefSrz36AMmGgm/IL9bbpDxF7gpB8aKAAKpE+AIUuos6QkB\nBDIn8MNAMPRuuy5dc+747yvZTY5pnbmRsGcEEHCVwI7NG/eeUrd2ycISKZR+LoMf6aoEGCwC\nCKRcgCNIKSelQwQQyIDAErmAQ5/dBdt7TBjUp2WLNu2DrY/vnIFhsEsEEHCTwOwJn5unb78x\nWrB189dSHF0oY//aTeNnrAggkB4BCqT0uNIrAgjUvcAuOS3mbXmTY88aM+qC7Zs26FXuAvI5\npbofCXtEAAFHC4RLik2fx/8ZH/jCE0a+2+gJ+d1xkwx4p6MHzeAQQKDOBDjFrs6o2RECCNSh\nwLnBUPbAxi1aNL3jyZ457bqcUoe7ZlcIIOBkgRVzZ5lX7/tDeFd+fn48FvmZjHWyk8fL2BBA\noO4FOIJU9+bsEQEE0i+wPmHH35Qvd+z4xeC+XeKxuOl02rf4zqT0u7MHBBwrIFelM0NffTbx\n9oP3JMLFxR/Y8dhVMtgVjh0wA0MAgYwJcAQpY/TsGAEE6kjgR/KdSW+0aNPuyNseezGnzQld\n6mi37AYBBJwisHHV8r2X7960ZqV+t9HNMq5hThkb40AAAecJcATJeXPCiBBAILUCixO2/Vbx\nrl0nThj8QeeAtI6nnmHkJrV7oTcEEHCcgBRDZuRbPc2rf/t9vGhnwRh5fIkMkgsxOG6mGBAC\nzhLgCJKz5oPRIIBAegV+JkeTerXq0Cn3N488l31spxPTuzd6RwCBjAmsXbzAvP6Pu8Jb8laH\npTC6UwbSJ2ODYccIIOAqAY4guWq6GCwCCCQpsECOJr2zZ2fhyeMH9jk+Gi61Op56piVFU5Ld\nsjkCCDhFQP5fm8Hypa9v/fuvieJdO4fF47HLZWzTnDI+xoEAAs4X4AiS8+eIESKAQHoE9LNJ\nrzVo2rzBLf9+Mufkc85Pz17oFQEE6kxg6ayvzJv/+ktYvtdolxw1+rXseHid7ZwdIYCAZwQo\nkDwzlSSCAAK1EGggH0Z63Nj2b8++/Gr7hnv+HWrQuEktumETBBDIpMCuHflmwLOPxqeOGGrJ\n5wvfke81ulvGw/caZXJS2DcCLhagQHLx5DF0BBBImcDZoezsd0JZ2R1+dvc/ss+/9idcxCFl\ntHSEQPoEpBAycrqsGfTC41G5nP9K+dJXPWo0JX17pGcEEPCDAAWSH2aZHBFAoDoC+kGkuwPB\n4EOtj+9s/fL+R3L0anc0BBBwpsDK+bPNuw/fF964ekXcjscfkFG+JBF35mgZFQIIuEmAAslN\ns8VYEUCgLgRaS5H0jLzh+sm5V15rX//n+4NHN2teF/tlHwggUA0BPZ1OLsIQnzT0Q/mvGhwQ\nj8fvks02VWNTVkEAAQSqJUCBVC0mVkIAAR8K9JBT7npZAavztXfcnXXpz39l5LEPGUgZAWcI\nRCNh8/kHb5lhvV6IJhL26lgk8lsZ2XhnjI5RIICAlwQokLw0m+SCAAKpFtBvk/2NXO3uv0c3\na5H7k7seyPnWpVemeh/0hwAChxGY/tnHpt8z/wkXFewolqvT/V1Wf0OC0+kO48bTCCBQOwEK\npNq5sRUCCPhLoJGk+4BcHuuPbTqfpFe7y+l8+ln+EiBbBDIgsHLe16bPE/+K5C1dZOQ7zJ6V\nITwmsTsDQ2GXCCDgIwEKJB9NNqkigEDSAm3lQw+Py+eTfnpKjwvjckQpq/XxnZLulA4QQGB/\ngQ0rl5shLz8VnT1hdEg+ZzRIPmd0j6yxdv+1eIQAAgikR4ACKT2u9IoAAt4W6B7MynrOjsW+\nc/411yeuuf3PwcYtW3k7Y7JDoA4Etm1YZz7q+Ux82siPAqGsrEmxaFQLo+l1sGt2gQACCHwj\nQIH0DQV3EEAAgRoLXCrfn/SMHYt3ueC6G6zv33pnoHGLY2rcCRsg4HeBgq1bzMevvxj/Ykg/\nEwyG5sj3GWlhxAUY/P6DQf4IZEiAAilD8OwWAQQ8JXCNXOHuMduOn3DhdTfuLZQaNW/hqQRJ\nBoF0COzM32pGvtMrMe7D9+KWZS2Xwuhe2c8n6dgXfSKAAALVFaBAqq4U6yGAAAKHFtDfp9fK\nEaXH5MPknS66/hfWFTffEaBQOjQaz/pTIH/jejPqndfsiUP7J0wgsDoeifxDJAZIJPwpQtYI\nIOAkAQokJ80GY0EAAS8I6O/VH0mh9Kicetexx9XXmStu/m2wZdsOXsiNHBBISmDL2tXmkzdf\njk8dMVRPpVssR4z+LR0OlaAwSkqWjRFAIJUCFEip1KQvBBBA4H8C+vv1B3Lq3T/kTeCZ3S+8\nNHbVrXdmdeja/X9rcA8BnwisXbLAjHzr1diMz0cE5f/EzLLCaJRP0idNBBBwmQAFkssmjOEi\ngIArBc4PhrIfiMcil3Xsfkb0B7/+Q7ZcJtyViTBoBKorYNu2mTtxrBn17muRFXNmZctV6cbL\nVekelu0nVLcP1kMAAQQyIUCBlAl19okAAn4VOFm+R+k++YzSDc1at4l97xe35pz3g+tM7pFH\n+dWDvD0oEC4pNpOHDzKfvtsrvGPLpoC80egrxdIzkup8D6ZLSggg4EEBCiQPTiopIYCA4wVa\nywjvCIZCd8r5Rkd950c/C333p/9ntWjTzvEDZ4AIHExAv8No/MA+ifED3o/JKXR74rHYi7Ju\nT4ktB9uG5QgggIATBSiQnDgrjAkBBPwikCOJ/jQrO/uv0Uika9dzvxO79MZbsk4+9ztyYa+A\nXwzI08UCdjy+9zS6sR++G1n01ZQs+VleKj/LerSoj0Spi1Nj6Agg4GMBCiQfTz6pI4CAowTO\nDQaDf5ZTkX5Uv3HT+EU/vjHn/GuuN02O0YNNNAScJVCwdbOZOKS/GTfg/XDRzgL5CiNroBRL\nerRoqrNGymgQQACBmgtQINXcjC0QQACBdAo0k85vkr/E/07+Et/+pLPOi174419kn3bhpUY+\n5J7O/dI3AocUkNPm9h4tmvTRgOj8KRPk7NCsPPkZfUk2ekdixyE35kkEEEDARQIUSC6aLIaK\nAAK+E+gRCARvk6+I+WnOkUdZPa7+cdZ53/+R1a7LKb6DIOHMCaxaMMdMGT7Ilu8uikVLSxLG\nsgbL0aI3ZETjMzcq9owAAgikT4ACKX229IwAAgikSqCBdHSDfPnsrbFI5MymrY4N97j6+txz\nrrjGcGGHVBHTT0WB/I3rzVejhpmJQ/uXysUXcuS7i6bJESQtigZJ7K64LvcRQAABrwlQIHlt\nRskHAQS8LtBOEvx5VnbOr6KRcMc2J3TRYinnzEuuNI2at/B67uSXRgEtimaOGWm+HPlROG/p\nohz5GZNT6MJaFL0nsTaNu6ZrBBBAwFECFEiOmg4GgwACCNRI4FRZ+0b5LMhN8gWczdue1DVy\n9mVX55x+8WWm+XFta9QRK/tTYPumDXuLomlaFC1ZmBPKztkQi4T1CnQDJL72pwpZI4CA3wUo\nkPz+E0D+CCDgBQH9XX6WxI/kr/43yF/9j2vZrkOpFEu5Z3z3cnNspxO9kCM5pEAgkUiYtYsX\n7L3Ywqyxo0rXr1iaW1YUfSDda1E0KwW7oQsEEEDA1QIUSK6ePgaPAAIIVCnQVZaWF0snNmjS\nLCxXwcvq1uPCwEln9zC5Rx5V5UYs9KZApLTULPpqspkzcaz99dhPo3JZbj19bpEU0vp5ouES\nFEXenHqyQgCBWgpQINUSjs0QQAABlwi0l3FeGQxlX52wYxfIFchCHbudHut+waU5XXtcYI7t\neIJL0mCY1RXQo0R5SxdKUTTFzJ88IbJ8zsyAbcftQCA0IR6LDJF+PpHYUN3+WA8BBBDwmwAF\nkt9mnHwRQMDPAjmS/HckrsjKybk2Gg63P7J+g3CXs88PyvcthU488xxzTPuOfvZxbe5b1+eZ\nxdOnmAVTJ8YWfjnRLt2zJ1vmeKXM8QhJ6nOJsRIlrk2QgSOAAAJ1KECBVIfY7AoBBBBwmIBe\nyeHiQCBwcSCYdWksGm5xVIOG3xRMHbufaVp16GTkeYcN29/Dke8gMuuWLzbLZ880y76eHlsy\nc1q8qLBALsWdtU2e+8y27dEiNEZik7+lyB4BBBConQAFUu3c2AoBBBDwooCejndRWcF0iRZM\nchQi2q5LN7vz6WflHH/KaabDKd1Ng8ZNvZi7Y3PatSPfrFk036xeMNcsmflleNWC2QE5MpQl\nc7MuHo2Ok4JosgxeY4ljk2BgCCCAgIsEKJBcNFkMFQEEEKhjgVayv3M0srJzvyOfX+kub8Zz\nGjZrUXr8Kd1DbU/sGjqu80lGvovJNG6pq9KSFdiZv9WskavM7SuI5kRXLZoXL9qxPdcKBCJy\nhGiRFEbjZB9TymJLsvtjewQQQACBAwUokA40YQkCCCCAQNUCIVl8isTZEqfJEYxz5AjGCVo0\n5RxxZFSKpXi7k7vlHnv8CUYuM743ONpUNeSuHdvNxlXLzcaVy/be5i1ZVLph1bJASdHu7ArF\n0FTZWq8wp7FQIiZBQwABBBBIswAFUpqB6R4BBBDwuIB+QKmThH5pbXc5ynGmfGjp5Fg43FIe\nB7Jzj4i2aNMudmynk3JatmsfaNGmvWkiR5uaHHOsadi0mVxUz7svQ3o0aNv6dWbrhjyzbd1a\nIxdSsDevWRHZvHb13kLIGMvOysleL58bmhePxeaK1yKJ+RKLJSiGBIGGAAIIZELAu69MmdBk\nnwgggAAC5QK5ckcviafXET9BiqYT5UMz3ex4rF08Fm2oKwWCwfjRTZtHmrQ61mp+bNucJse0\nsvSIk3xvk3zOqYncyn25PbL+3tV1E0c0Ocpj9uwqNIXbtpbFFrndYgq2bbG3b1wfLdiyyS7Y\nujkrFo3qEbdEKCtnuxW01kQjkcXGtlfKsmUSWgzpZ4bCEjQEEEAAAQcJUCA5aDIYCgIIIOAT\ngSMlzzYSehW9vSEXhmgbzM7uYCVMy3g81lSOqNST5/a+RmkhdWSDhtEjj6qfOKJ+AymYGgSO\natAgJIVT8Ih69U3uUfVMdm6uCYWyTDAry2RlZ8tttpGjWSZLbuWUNZOwbWMnbLlNGP2eoMTe\n+7aJyxXhIqUl+6JEb0tNJCwhy0qL99i7d2yPyRXi4vLlqqZ4966ALMuSvr65rF8wFCoKBkNb\npMd18vmgNTLmjWWxVm5XSayW4PLagkBDAAEE3CJAgeSWmWKcCCCAgL8EsiTdZhLNJVqU3ddD\nSRoN9FaKqoaBUKixFECN5FS9I6TqyTYJk20bkyPVUFbC2FLMJKSfhBQ0li1n8yVkO3layjC5\nLy+Acj8QDwSsEnlcLEuKpdApluKpSE57260h6xdI7DhIbJblpRI0BBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAA\nAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBA\nAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQ\nQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEE\nEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAAB\nBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAAQQQQAABBBBAAAEEEEAAgZQI/D+FBE4kcF4PBAAA\nAABJRU5ErkJggg==",
      "text/plain": [
       "Plot with title “Genes with ECM DMR overlap”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_ecm_ol = df_all %>% filter(Gene!=\"intergenic\") %>% group_by(gene2) %>% \n",
    "summarize(bp.ecm.overlap=sum(bp.ecm.overlap)) \n",
    "w_overlap = nrow(df_ecm_ol[df_ecm_ol$bp.ecm.overlap > 0,])\n",
    "w_overlap_pct = round(nrow(df_ecm_ol[df_ecm_ol$bp.ecm.overlap > 0,])/nrow(df_ecm_ol) * 100)\n",
    "no_overlap_pct = round(((nrow(df_ecm_ol)-w_overlap)/nrow(df_ecm_ol)*100))\n",
    "\n",
    "pie(c(w_overlap, nrow(df_ecm_ol)-w_overlap), \n",
    "    labels=paste(c(\"overlap\", \"No overlap\"),c(w_overlap_pct, no_overlap_pct),c(\"%\",\"%\")),\n",
    "   cex=1.1, main=\"Genes with ECM DMR overlap\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2609cb7",
   "metadata": {},
   "source": [
    "## Select panel region subsets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbe140be",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Select panel regions\n",
    "# reasons to select a region\n",
    "# —high number of pubs\n",
    "# —has a paper with recent functional evidence or review mention and at least one other evidence type from our work\n",
    "# —extract promoter regions from these genes\n",
    "# —count number of CpGs in the region\n",
    "# —filter by overlaps with TCGA DMRs or ECM DMRs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "1de923b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 23</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr_hg19</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>chr_hg38</th><th scope=col>start_hg38</th><th scope=col>end_hg38</th><th scope=col>pub.strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>⋯</th><th scope=col>in.gencode.gene</th><th scope=col>gencode.gene.name</th><th scope=col>in.gencode.promoter</th><th scope=col>is.ingergenic</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>region</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>ecm.methyl.status</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106067032</td><td>106200973</td><td>chr4 </td><td>105145875</td><td>105279816</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>TET2-AS1</td><td>TRUE</td><td>FALSE</td><td>TET2  </td><td>4</td><td>4 : 105145875 - 105279816</td><td>373</td><td> 174</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31350191</td><td> 31397165</td><td>chr20</td><td> 32762385</td><td> 32809359</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene</td><td>⋯</td><td>TRUE</td><td>DNMT3B  </td><td>TRUE</td><td>FALSE</td><td>DNMT3B</td><td>2</td><td>20 : 32762385 - 32809359 </td><td>  0</td><td> 806</td><td>hyper</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209130798</td><td>chr2 </td><td>208236229</td><td>208266074</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>IDH1-AS1</td><td>TRUE</td><td>FALSE</td><td>IDH1  </td><td>1</td><td>2 : 208236229 - 208266074</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140624729</td><td>chr7 </td><td>140719327</td><td>140924929</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene</td><td>⋯</td><td>TRUE</td><td>RNU6-85P</td><td>TRUE</td><td>FALSE</td><td>BRAF  </td><td>1</td><td>7 : 140719327 - 140924929</td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66763863</td><td> 66950461</td><td>chrX </td><td> 67544021</td><td> 67730619</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene</td><td>⋯</td><td>TRUE</td><td>AR      </td><td>TRUE</td><td>FALSE</td><td>AR    </td><td>2</td><td>X : 67544021 - 67730619  </td><td>  0</td><td>   0</td><td>.    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128747680</td><td>128755197</td><td>chr8 </td><td>127735434</td><td>127742951</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene</td><td>⋯</td><td>TRUE</td><td>MYC     </td><td>TRUE</td><td>FALSE</td><td>MYC   </td><td>3</td><td>8 : 127735434 - 127742951</td><td>  0</td><td>1954</td><td>hypo </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 23\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr\\_hg19 & start\\_hg19 & end\\_hg19 & chr\\_hg38 & start\\_hg38 & end\\_hg38 & pub.strand & source & pub.gene.name & pub.reg.type & ⋯ & in.gencode.gene & gencode.gene.name & in.gencode.promoter & is.ingergenic & hybrid.gene.name & no.total.pubs & region & bp.tcga.overlap & bp.ecm.overlap & ecm.methyl.status\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <chr> & <lgl> & <lgl> & <chr> & <int> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106067032 & 106200973 & chr4  & 105145875 & 105279816 & + & Kukkonen et al. 2021 & TET2   & gene & ⋯ & TRUE & TET2-AS1 & TRUE & FALSE & TET2   & 4 & 4 : 105145875 - 105279816 & 373 &  174 & hyper\\\\\n",
       "\t2 & 20 &  31350191 &  31397165 & chr20 &  32762385 &  32809359 & + & Kukkonen et al. 2021 & DNMT3B & gene & ⋯ & TRUE & DNMT3B   & TRUE & FALSE & DNMT3B & 2 & 20 : 32762385 - 32809359  &   0 &  806 & hyper\\\\\n",
       "\t3 & 2  & 209100953 & 209130798 & chr2  & 208236229 & 208266074 & - & Kukkonen et al. 2021 & IDH1   & gene & ⋯ & TRUE & IDH1-AS1 & TRUE & FALSE & IDH1   & 1 & 2 : 208236229 - 208266074 &   0 &    0 & .    \\\\\n",
       "\t4 & 7  & 140419127 & 140624729 & chr7  & 140719327 & 140924929 & - & Kukkonen et al. 2021 & BRAF   & gene & ⋯ & TRUE & RNU6-85P & TRUE & FALSE & BRAF   & 1 & 7 : 140719327 - 140924929 &   0 &    0 & .    \\\\\n",
       "\t5 & X  &  66763863 &  66950461 & chrX  &  67544021 &  67730619 & + & Kukkonen et al. 2021 & AR     & gene & ⋯ & TRUE & AR       & TRUE & FALSE & AR     & 2 & X : 67544021 - 67730619   &   0 &    0 & .    \\\\\n",
       "\t6 & 8  & 128747680 & 128755197 & chr8  & 127735434 & 127742951 & + & Kukkonen et al. 2021 & MYC    & gene & ⋯ & TRUE & MYC      & TRUE & FALSE & MYC    & 3 & 8 : 127735434 - 127742951 &   0 & 1954 & hypo \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 23\n",
       "\n",
       "| <!--/--> | chr_hg19 &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | chr_hg38 &lt;chr&gt; | start_hg38 &lt;int&gt; | end_hg38 &lt;int&gt; | pub.strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | ⋯ ⋯ | in.gencode.gene &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | in.gencode.promoter &lt;lgl&gt; | is.ingergenic &lt;lgl&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | region &lt;chr&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | ecm.methyl.status &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106067032 | 106200973 | chr4  | 105145875 | 105279816 | + | Kukkonen et al. 2021 | TET2   | gene | ⋯ | TRUE | TET2-AS1 | TRUE | FALSE | TET2   | 4 | 4 : 105145875 - 105279816 | 373 |  174 | hyper |\n",
       "| 2 | 20 |  31350191 |  31397165 | chr20 |  32762385 |  32809359 | + | Kukkonen et al. 2021 | DNMT3B | gene | ⋯ | TRUE | DNMT3B   | TRUE | FALSE | DNMT3B | 2 | 20 : 32762385 - 32809359  |   0 |  806 | hyper |\n",
       "| 3 | 2  | 209100953 | 209130798 | chr2  | 208236229 | 208266074 | - | Kukkonen et al. 2021 | IDH1   | gene | ⋯ | TRUE | IDH1-AS1 | TRUE | FALSE | IDH1   | 1 | 2 : 208236229 - 208266074 |   0 |    0 | .     |\n",
       "| 4 | 7  | 140419127 | 140624729 | chr7  | 140719327 | 140924929 | - | Kukkonen et al. 2021 | BRAF   | gene | ⋯ | TRUE | RNU6-85P | TRUE | FALSE | BRAF   | 1 | 7 : 140719327 - 140924929 |   0 |    0 | .     |\n",
       "| 5 | X  |  66763863 |  66950461 | chrX  |  67544021 |  67730619 | + | Kukkonen et al. 2021 | AR     | gene | ⋯ | TRUE | AR       | TRUE | FALSE | AR     | 2 | X : 67544021 - 67730619   |   0 |    0 | .     |\n",
       "| 6 | 8  | 128747680 | 128755197 | chr8  | 127735434 | 127742951 | + | Kukkonen et al. 2021 | MYC    | gene | ⋯ | TRUE | MYC      | TRUE | FALSE | MYC    | 3 | 8 : 127735434 - 127742951 |   0 | 1954 | hypo  |\n",
       "\n"
      ],
      "text/plain": [
       "  chr_hg19 start_hg19 end_hg19  chr_hg38 start_hg38 end_hg38  pub.strand\n",
       "1 4        106067032  106200973 chr4     105145875  105279816 +         \n",
       "2 20        31350191   31397165 chr20     32762385   32809359 +         \n",
       "3 2        209100953  209130798 chr2     208236229  208266074 -         \n",
       "4 7        140419127  140624729 chr7     140719327  140924929 -         \n",
       "5 X         66763863   66950461 chrX      67544021   67730619 +         \n",
       "6 8        128747680  128755197 chr8     127735434  127742951 +         \n",
       "  source               pub.gene.name pub.reg.type ⋯ in.gencode.gene\n",
       "1 Kukkonen et al. 2021 TET2          gene         ⋯ TRUE           \n",
       "2 Kukkonen et al. 2021 DNMT3B        gene         ⋯ TRUE           \n",
       "3 Kukkonen et al. 2021 IDH1          gene         ⋯ TRUE           \n",
       "4 Kukkonen et al. 2021 BRAF          gene         ⋯ TRUE           \n",
       "5 Kukkonen et al. 2021 AR            gene         ⋯ TRUE           \n",
       "6 Kukkonen et al. 2021 MYC           gene         ⋯ TRUE           \n",
       "  gencode.gene.name in.gencode.promoter is.ingergenic hybrid.gene.name\n",
       "1 TET2-AS1          TRUE                FALSE         TET2            \n",
       "2 DNMT3B            TRUE                FALSE         DNMT3B          \n",
       "3 IDH1-AS1          TRUE                FALSE         IDH1            \n",
       "4 RNU6-85P          TRUE                FALSE         BRAF            \n",
       "5 AR                TRUE                FALSE         AR              \n",
       "6 MYC               TRUE                FALSE         MYC             \n",
       "  no.total.pubs region                    bp.tcga.overlap bp.ecm.overlap\n",
       "1 4             4 : 105145875 - 105279816 373              174          \n",
       "2 2             20 : 32762385 - 32809359    0              806          \n",
       "3 1             2 : 208236229 - 208266074   0                0          \n",
       "4 1             7 : 140719327 - 140924929   0                0          \n",
       "5 2             X : 67544021 - 67730619     0                0          \n",
       "6 3             8 : 127735434 - 127742951   0             1954          \n",
       "  ecm.methyl.status\n",
       "1 hyper            \n",
       "2 hyper            \n",
       "3 .                \n",
       "4 .                \n",
       "5 .                \n",
       "6 hypo             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all = read.table(file=\"./marker_collation/all_lit_regions_annotated_overlaps_05202022.tsv\", header=TRUE, sep=\"\\t\")\n",
    "head(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "0174e45f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "   1    2    3    4    5    6    7    8   15 \n",
       "4401 1055  248   70   19    5    3    4    1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# pub count was originally on hybrid names, so there should be a unique count per hybrid name\n",
    "df_pub_count = df_all %>% filter(!(is.ingergenic)) %>% select(hybrid.gene.name,no.total.pubs) %>% distinct()\n",
    "\n",
    "table(df_pub_count$no.total.pubs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "cbcaced6",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_pub_count, file=\"./marker_collation/gene_pub_count_table.csv\", sep=\",\", row.names=FALSE, col.names=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "d8bd0420",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TET2  </td><td>4</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>DNMT3B</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>IDH1  </td><td>1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BRAF  </td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AR    </td><td>2</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MYC   </td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & hybrid.gene.name & no.total.pubs\\\\\n",
       "  & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & TET2   & 4\\\\\n",
       "\t2 & DNMT3B & 2\\\\\n",
       "\t3 & IDH1   & 1\\\\\n",
       "\t4 & BRAF   & 1\\\\\n",
       "\t5 & AR     & 2\\\\\n",
       "\t6 & MYC    & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | TET2   | 4 |\n",
       "| 2 | DNMT3B | 2 |\n",
       "| 3 | IDH1   | 1 |\n",
       "| 4 | BRAF   | 1 |\n",
       "| 5 | AR     | 2 |\n",
       "| 6 | MYC    | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "  hybrid.gene.name no.total.pubs\n",
       "1 TET2             4            \n",
       "2 DNMT3B           2            \n",
       "3 IDH1             1            \n",
       "4 BRAF             1            \n",
       "5 AR               2            \n",
       "6 MYC              3            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pub_count = read.table(file=\"./marker_collation/gene_pub_count_table.csv\", sep=\",\",header=TRUE)\n",
    "df_pub_count = df_pub_count[,c(1,2)]\n",
    "head(df_pub_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "58f45dd3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Input</th><th scope=col>Match.type</th><th scope=col>Approved.symbol</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TET2  </td><td>Approved symbol</td><td>TET2  </td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>DNMT3B</td><td>Approved symbol</td><td>DNMT3B</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>IDH1  </td><td>Approved symbol</td><td>IDH1  </td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BRAF  </td><td>Approved symbol</td><td>BRAF  </td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AR    </td><td>Approved symbol</td><td>AR    </td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MYC   </td><td>Approved symbol</td><td>MYC   </td><td>MYC proto-oncogene, bHLH transcription factor</td><td>HGNC:7553 </td><td>8q24.21 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Input & Match.type & Approved.symbol & Approved.name & HGNC.ID & Location\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & TET2   & Approved symbol & TET2   & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24    \\\\\n",
       "\t2 & DNMT3B & Approved symbol & DNMT3B & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21\\\\\n",
       "\t3 & IDH1   & Approved symbol & IDH1   & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34    \\\\\n",
       "\t4 & BRAF   & Approved symbol & BRAF   & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34    \\\\\n",
       "\t5 & AR     & Approved symbol & AR     & androgen receptor                             & HGNC:644   & Xq12    \\\\\n",
       "\t6 & MYC    & Approved symbol & MYC    & MYC proto-oncogene, bHLH transcription factor & HGNC:7553  & 8q24.21 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Input &lt;chr&gt; | Match.type &lt;chr&gt; | Approved.symbol &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | TET2   | Approved symbol | TET2   | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     |\n",
       "| 2 | DNMT3B | Approved symbol | DNMT3B | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 |\n",
       "| 3 | IDH1   | Approved symbol | IDH1   | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     |\n",
       "| 4 | BRAF   | Approved symbol | BRAF   | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     |\n",
       "| 5 | AR     | Approved symbol | AR     | androgen receptor                             | HGNC:644   | Xq12     |\n",
       "| 6 | MYC    | Approved symbol | MYC    | MYC proto-oncogene, bHLH transcription factor | HGNC:7553  | 8q24.21  |\n",
       "\n"
      ],
      "text/plain": [
       "  Input  Match.type      Approved.symbol\n",
       "1 TET2   Approved symbol TET2           \n",
       "2 DNMT3B Approved symbol DNMT3B         \n",
       "3 IDH1   Approved symbol IDH1           \n",
       "4 BRAF   Approved symbol BRAF           \n",
       "5 AR     Approved symbol AR             \n",
       "6 MYC    Approved symbol MYC            \n",
       "  Approved.name                                 HGNC.ID    Location\n",
       "1 tet methylcytosine dioxygenase 2              HGNC:25941 4q24    \n",
       "2 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21\n",
       "3 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34    \n",
       "4 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34    \n",
       "5 androgen receptor                             HGNC:644   Xq12    \n",
       "6 MYC proto-oncogene, bHLH transcription factor HGNC:7553  8q24.21 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# look up ensemble ids from hgcn database for each gene name because the pub gene name might not be in the gencode file\n",
    "# look up promoter region based on ensemble id. \n",
    "\n",
    "# GENCODE name = ENSEMBLE name, HUGO agrees with these, but occasionally not\n",
    "# look up synonyms here: https://www.genenames.org/tools/multi-symbol-checker/\n",
    "\n",
    "\n",
    "# these are hybrid gene names matched with their approved hugo gene symbols\n",
    "df_hugo = read.csv(file=\"./marker_collation/pub_count_table_hgnc-symbol-check2.csv\", header=TRUE)\n",
    "head(df_hugo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "03603cf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>gene_name</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>10869</td><td>12869</td><td>DDX11L1    </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>13404</td><td>15404</td><td>WASH7P     </td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>28554</td><td>30554</td><td>MIR1302-2HG</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>33554</td><td>35554</td><td>FAM138A    </td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>51473</td><td>53473</td><td>OR4G4P     </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>56598</td><td>58598</td><td>OR4G11P    </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & gene\\_name & strand & dot2 & type\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 10869 & 12869 & DDX11L1     & + & . & promoter\\\\\n",
       "\t2 & 1 & 13404 & 15404 & WASH7P      & - & . & promoter\\\\\n",
       "\t3 & 1 & 28554 & 30554 & MIR1302-2HG & + & . & promoter\\\\\n",
       "\t4 & 1 & 33554 & 35554 & FAM138A     & - & . & promoter\\\\\n",
       "\t5 & 1 & 51473 & 53473 & OR4G4P      & + & . & promoter\\\\\n",
       "\t6 & 1 & 56598 & 58598 & OR4G11P     & + & . & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | gene_name &lt;chr&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 10869 | 12869 | DDX11L1     | + | . | promoter |\n",
       "| 2 | 1 | 13404 | 15404 | WASH7P      | - | . | promoter |\n",
       "| 3 | 1 | 28554 | 30554 | MIR1302-2HG | + | . | promoter |\n",
       "| 4 | 1 | 33554 | 35554 | FAM138A     | - | . | promoter |\n",
       "| 5 | 1 | 51473 | 53473 | OR4G4P      | + | . | promoter |\n",
       "| 6 | 1 | 56598 | 58598 | OR4G11P     | + | . | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 gene_name   strand dot2 type    \n",
       "1 1   10869      12869    DDX11L1     +      .    promoter\n",
       "2 1   13404      15404    WASH7P      -      .    promoter\n",
       "3 1   28554      30554    MIR1302-2HG +      .    promoter\n",
       "4 1   33554      35554    FAM138A     -      .    promoter\n",
       "5 1   51473      53473    OR4G4P      +      .    promoter\n",
       "6 1   56598      58598    OR4G11P     +      .    promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>63331</li><li>7</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 63331\n",
       "\\item 7\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 63331\n",
       "2. 7\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 63331     7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_promoters = read.table(file=\"../Translational_Science/gencode.v40lift37.promoters.bed\", header=TRUE, sep=\"\\t\")\n",
    "head(df_promoters)\n",
    "dim(df_promoters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "90403f48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5706</li><li>2</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5706\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5706\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5706    2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5706</li><li>7</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5706\n",
       "\\item 7\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5706\n",
       "2. 7\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5706    7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>Match.type</th><th scope=col>Approved.symbol</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TET2  </td><td>4</td><td>Approved symbol</td><td>TET2  </td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td><td>4 </td><td>106066032</td><td>106068032</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>DNMT3B</td><td>2</td><td>Approved symbol</td><td>DNMT3B</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td><td>20</td><td> 31349191</td><td> 31351191</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>IDH1  </td><td>1</td><td>Approved symbol</td><td>IDH1  </td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td><td>2 </td><td>209099953</td><td>209101953</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BRAF  </td><td>1</td><td>Approved symbol</td><td>BRAF  </td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td><td>7 </td><td>140418127</td><td>140420127</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AR    </td><td>2</td><td>Approved symbol</td><td>AR    </td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td><td>X </td><td> 66762863</td><td> 66764863</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MYC   </td><td>3</td><td>Approved symbol</td><td>MYC   </td><td>MYC proto-oncogene, bHLH transcription factor</td><td>HGNC:7553 </td><td>8q24.21 </td><td>8 </td><td>128746680</td><td>128748680</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & hybrid.gene.name & no.total.pubs & Match.type & Approved.symbol & Approved.name & HGNC.ID & Location & chr & start\\_hg19 & end\\_hg19 & strand & dot2 & type\\\\\n",
       "  & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & TET2   & 4 & Approved symbol & TET2   & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24     & 4  & 106066032 & 106068032 & + & . & promoter\\\\\n",
       "\t2 & DNMT3B & 2 & Approved symbol & DNMT3B & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21 & 20 &  31349191 &  31351191 & + & . & promoter\\\\\n",
       "\t3 & IDH1   & 1 & Approved symbol & IDH1   & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34     & 2  & 209099953 & 209101953 & - & . & promoter\\\\\n",
       "\t4 & BRAF   & 1 & Approved symbol & BRAF   & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34     & 7  & 140418127 & 140420127 & - & . & promoter\\\\\n",
       "\t5 & AR     & 2 & Approved symbol & AR     & androgen receptor                             & HGNC:644   & Xq12     & X  &  66762863 &  66764863 & + & . & promoter\\\\\n",
       "\t6 & MYC    & 3 & Approved symbol & MYC    & MYC proto-oncogene, bHLH transcription factor & HGNC:7553  & 8q24.21  & 8  & 128746680 & 128748680 & + & . & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | Match.type &lt;chr&gt; | Approved.symbol &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TET2   | 4 | Approved symbol | TET2   | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     | 4  | 106066032 | 106068032 | + | . | promoter |\n",
       "| 2 | DNMT3B | 2 | Approved symbol | DNMT3B | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 | 20 |  31349191 |  31351191 | + | . | promoter |\n",
       "| 3 | IDH1   | 1 | Approved symbol | IDH1   | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     | 2  | 209099953 | 209101953 | - | . | promoter |\n",
       "| 4 | BRAF   | 1 | Approved symbol | BRAF   | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     | 7  | 140418127 | 140420127 | - | . | promoter |\n",
       "| 5 | AR     | 2 | Approved symbol | AR     | androgen receptor                             | HGNC:644   | Xq12     | X  |  66762863 |  66764863 | + | . | promoter |\n",
       "| 6 | MYC    | 3 | Approved symbol | MYC    | MYC proto-oncogene, bHLH transcription factor | HGNC:7553  | 8q24.21  | 8  | 128746680 | 128748680 | + | . | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  hybrid.gene.name no.total.pubs Match.type      Approved.symbol\n",
       "1 TET2             4             Approved symbol TET2           \n",
       "2 DNMT3B           2             Approved symbol DNMT3B         \n",
       "3 IDH1             1             Approved symbol IDH1           \n",
       "4 BRAF             1             Approved symbol BRAF           \n",
       "5 AR               2             Approved symbol AR             \n",
       "6 MYC              3             Approved symbol MYC            \n",
       "  Approved.name                                 HGNC.ID    Location chr\n",
       "1 tet methylcytosine dioxygenase 2              HGNC:25941 4q24     4  \n",
       "2 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21 20 \n",
       "3 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34     2  \n",
       "4 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34     7  \n",
       "5 androgen receptor                             HGNC:644   Xq12     X  \n",
       "6 MYC proto-oncogene, bHLH transcription factor HGNC:7553  8q24.21  8  \n",
       "  start_hg19 end_hg19  strand dot2 type    \n",
       "1 106066032  106068032 +      .    promoter\n",
       "2  31349191   31351191 +      .    promoter\n",
       "3 209099953  209101953 -      .    promoter\n",
       "4 140418127  140420127 -      .    promoter\n",
       "5  66762863   66764863 +      .    promoter\n",
       "6 128746680  128748680 +      .    promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5724</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5724\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5724\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5724   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df_pub_count)\n",
    "\n",
    "# get gene hugo name\n",
    "df_pub_count2 = df_pub_count %>% inner_join(df_hugo, by=c(hybrid.gene.name=\"Input\"))\n",
    "dim(df_pub_count2)\n",
    "\n",
    "# match hugo name with genecode promoter\n",
    "df_pub_count3 = df_pub_count2 %>% inner_join(df_promoters, by=c(Approved.symbol=\"gene_name\"))\n",
    "head(df_pub_count3)\n",
    "dim(df_pub_count3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "6e30df5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>Match.type</th><th scope=col>Approved.symbol</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th><th scope=col>size</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TET2  </td><td>4</td><td>Approved symbol</td><td>TET2  </td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td><td>4 </td><td>106066032</td><td>106068032</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>DNMT3B</td><td>2</td><td>Approved symbol</td><td>DNMT3B</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td><td>20</td><td> 31349191</td><td> 31351191</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>IDH1  </td><td>1</td><td>Approved symbol</td><td>IDH1  </td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td><td>2 </td><td>209099953</td><td>209101953</td><td>-</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BRAF  </td><td>1</td><td>Approved symbol</td><td>BRAF  </td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td><td>7 </td><td>140418127</td><td>140420127</td><td>-</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AR    </td><td>2</td><td>Approved symbol</td><td>AR    </td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td><td>X </td><td> 66762863</td><td> 66764863</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MYC   </td><td>3</td><td>Approved symbol</td><td>MYC   </td><td>MYC proto-oncogene, bHLH transcription factor</td><td>HGNC:7553 </td><td>8q24.21 </td><td>8 </td><td>128746680</td><td>128748680</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & hybrid.gene.name & no.total.pubs & Match.type & Approved.symbol & Approved.name & HGNC.ID & Location & chr & start\\_hg19 & end\\_hg19 & strand & dot2 & type & size\\\\\n",
       "  & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & TET2   & 4 & Approved symbol & TET2   & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24     & 4  & 106066032 & 106068032 & + & . & promoter & 2000\\\\\n",
       "\t2 & DNMT3B & 2 & Approved symbol & DNMT3B & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21 & 20 &  31349191 &  31351191 & + & . & promoter & 2000\\\\\n",
       "\t3 & IDH1   & 1 & Approved symbol & IDH1   & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34     & 2  & 209099953 & 209101953 & - & . & promoter & 2000\\\\\n",
       "\t4 & BRAF   & 1 & Approved symbol & BRAF   & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34     & 7  & 140418127 & 140420127 & - & . & promoter & 2000\\\\\n",
       "\t5 & AR     & 2 & Approved symbol & AR     & androgen receptor                             & HGNC:644   & Xq12     & X  &  66762863 &  66764863 & + & . & promoter & 2000\\\\\n",
       "\t6 & MYC    & 3 & Approved symbol & MYC    & MYC proto-oncogene, bHLH transcription factor & HGNC:7553  & 8q24.21  & 8  & 128746680 & 128748680 & + & . & promoter & 2000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | Match.type &lt;chr&gt; | Approved.symbol &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TET2   | 4 | Approved symbol | TET2   | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     | 4  | 106066032 | 106068032 | + | . | promoter | 2000 |\n",
       "| 2 | DNMT3B | 2 | Approved symbol | DNMT3B | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 | 20 |  31349191 |  31351191 | + | . | promoter | 2000 |\n",
       "| 3 | IDH1   | 1 | Approved symbol | IDH1   | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     | 2  | 209099953 | 209101953 | - | . | promoter | 2000 |\n",
       "| 4 | BRAF   | 1 | Approved symbol | BRAF   | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     | 7  | 140418127 | 140420127 | - | . | promoter | 2000 |\n",
       "| 5 | AR     | 2 | Approved symbol | AR     | androgen receptor                             | HGNC:644   | Xq12     | X  |  66762863 |  66764863 | + | . | promoter | 2000 |\n",
       "| 6 | MYC    | 3 | Approved symbol | MYC    | MYC proto-oncogene, bHLH transcription factor | HGNC:7553  | 8q24.21  | 8  | 128746680 | 128748680 | + | . | promoter | 2000 |\n",
       "\n"
      ],
      "text/plain": [
       "  hybrid.gene.name no.total.pubs Match.type      Approved.symbol\n",
       "1 TET2             4             Approved symbol TET2           \n",
       "2 DNMT3B           2             Approved symbol DNMT3B         \n",
       "3 IDH1             1             Approved symbol IDH1           \n",
       "4 BRAF             1             Approved symbol BRAF           \n",
       "5 AR               2             Approved symbol AR             \n",
       "6 MYC              3             Approved symbol MYC            \n",
       "  Approved.name                                 HGNC.ID    Location chr\n",
       "1 tet methylcytosine dioxygenase 2              HGNC:25941 4q24     4  \n",
       "2 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21 20 \n",
       "3 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34     2  \n",
       "4 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34     7  \n",
       "5 androgen receptor                             HGNC:644   Xq12     X  \n",
       "6 MYC proto-oncogene, bHLH transcription factor HGNC:7553  8q24.21  8  \n",
       "  start_hg19 end_hg19  strand dot2 type     size\n",
       "1 106066032  106068032 +      .    promoter 2000\n",
       "2  31349191   31351191 +      .    promoter 2000\n",
       "3 209099953  209101953 -      .    promoter 2000\n",
       "4 140418127  140420127 -      .    promoter 2000\n",
       "5  66762863   66764863 +      .    promoter 2000\n",
       "6 128746680  128748680 +      .    promoter 2000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pub_count3 = df_pub_count3 %>% mutate(size = end_hg19 - start_hg19)\n",
    "head(df_pub_count3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "161ddb7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_pub_count3, file=\"./marker_collation/gene_pub_count_table.csv\",sep=\",\",row.names=FALSE, col.names=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "bd4f90fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "   1    2    3    4    5    6    7    8   15 \n",
       "4318 1057  247   70   19    5    3    4    1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts = table(df_pub_count3$no.total.pubs)\n",
    "counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "b01f55c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sum(is.na(df_pub_count3$Approved.symbol))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "69da8963",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1\n",
      "[1] 2\n",
      "[1] 3\n",
      "[1] 4\n",
      "[1] 5\n",
      "[1] 6\n",
      "[1] 7\n",
      "[1] 8\n",
      "[1] 15\n"
     ]
    }
   ],
   "source": [
    "panel_size <- function(no.pubs){\n",
    "    return(sum(df_pub_count3$size[df_pub_count3$no.total.pubs >= no.pubs]))\n",
    "}\n",
    "\n",
    "sizes = vector()\n",
    "for (s in names(counts)){\n",
    "    print(as.numeric(s))\n",
    "    sizes = c(sizes, panel_size(as.numeric(s)))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "aa81d31c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11448000</li><li>2812000</li><li>698000</li><li>204000</li><li>64000</li><li>26000</li><li>16000</li><li>10000</li><li>2000</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11448000\n",
       "\\item 2812000\n",
       "\\item 698000\n",
       "\\item 204000\n",
       "\\item 64000\n",
       "\\item 26000\n",
       "\\item 16000\n",
       "\\item 10000\n",
       "\\item 2000\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11448000\n",
       "2. 2812000\n",
       "3. 698000\n",
       "4. 204000\n",
       "5. 64000\n",
       "6. 26000\n",
       "7. 16000\n",
       "8. 10000\n",
       "9. 2000\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11448000  2812000   698000   204000    64000    26000    16000    10000\n",
       "[9]     2000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAYAAAD958/bAAAEDmlDQ1BrQ0dDb2xvclNwYWNl\nR2VuZXJpY1JHQgAAOI2NVV1oHFUUPpu5syskzoPUpqaSDv41lLRsUtGE2uj+ZbNt3CyTbLRB\nkMns3Z1pJjPj/KRpKT4UQRDBqOCT4P9bwSchaqvtiy2itFCiBIMo+ND6R6HSFwnruTOzu5O4\na73L3PnmnO9+595z7t4LkLgsW5beJQIsGq4t5dPis8fmxMQ6dMF90A190C0rjpUqlSYBG+PC\nv9rt7yDG3tf2t/f/Z+uuUEcBiN2F2Kw4yiLiZQD+FcWyXYAEQfvICddi+AnEO2ycIOISw7UA\nVxieD/Cyz5mRMohfRSwoqoz+xNuIB+cj9loEB3Pw2448NaitKSLLRck2q5pOI9O9g/t/tkXd\na8Tbg0+PszB9FN8DuPaXKnKW4YcQn1Xk3HSIry5ps8UQ/2W5aQnxIwBdu7yFcgrxPsRjVXu8\nHOh0qao30cArp9SZZxDfg3h1wTzKxu5E/LUxX5wKdX5SnAzmDx4A4OIqLbB69yMesE1pKojL\njVdoNsfyiPi45hZmAn3uLWdpOtfQOaVmikEs7ovj8hFWpz7EV6mel0L9Xy23FMYlPYZenAx0\nyDB1/PX6dledmQjikjkXCxqMJS9WtfFCyH9XtSekEF+2dH+P4tzITduTygGfv58a5VCTH5Pt\nXD7EFZiNyUDBhHnsFTBgE0SQIA9pfFtgo6cKGuhooeilaKH41eDs38Ip+f4At1Rq/sjr6NEw\nQqb/I/DQqsLvaFUjvAx+eWirddAJZnAj1DFJL0mSg/gcIpPkMBkhoyCSJ8lTZIxk0TpKDjXH\nliJzZPO50dR5ASNSnzeLvIvod0HG/mdkmOC0z8VKnzcQ2M/Yz2vKldduXjp9bleLu0ZWn7vW\nc+l0JGcaai10yNrUnXLP/8Jf59ewX+c3Wgz+B34Df+vbVrc16zTMVgp9um9bxEfzPU5kPqUt\nVWxhs6OiWTVW+gIfywB9uXi7CGcGW/zk98k/kmvJ95IfJn/j3uQ+4c5zn3Kfcd+AyF3gLnJf\ncl9xH3OfR2rUee80a+6vo7EK5mmXUdyfQlrYLTwoZIU9wsPCZEtP6BWGhAlhL3p2N6sTjRdd\nuwbHsG9kq32sgBepc+xurLPW4T9URpYGJ3ym4+8zA05u44QjST8ZIoVtu3qE7fWmdn5LPdqv\ngcZz8Ww8BWJ8X3w0PhQ/wnCDGd+LvlHs8dRy6bLLDuKMaZ20tZrqisPJ5ONiCq8yKhYM5cCg\nKOu66Lsc0aYOtZdo5QCwezI4wm9J/v0X23mlZXOfBjj8Jzv3WrY5D+CsA9D7aMs2gGfjve8A\nrD6mePZSeCfEYt8CONWDw8FXTxrPqx/r9Vt4biXeANh8vV7/+/16ffMD1N8AuKD/A/8leAvF\nY9bLAAAAOGVYSWZNTQAqAAAACAABh2kABAAAAAEAAAAaAAAAAAACoAIABAAAAAEAAANIoAMA\nBAAAAAEAAANIAAAAAN/ryxkAAEAASURBVHgB7N0L1GRleSf6bpsGBNLQIpeG2BCCRnFARRQx\nAWNAziQ5BCKHTDRrcDwSwUDIUZzJQRwNKzCzdGE8gEaJRpGAGhIJmotEIlFUjBNFxchlCNDc\nGkRuzU0bGjj/R6tYVW3111Vt0Xy139+z1p+qvWt/9dX7e5vLw9717gULFAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAEC\nBAgQIDDvBRbO+0/oA05T4Bl5s1+Y5hvOwHvdnc94zQx8Th+RAAECBAgQIECAAIGNLPBn+X2P\nN5ZHNrKxX0eAAAECBAgQIDDDApvM8Gf30ScX2PSXf/mXFxx77LGT/+QM/sSVV1654J3vfKc/\n4zM4dz4yAQIECBAgQOCpEvAfj0+V/FP0excuXLhg8eLFT9Fv37i/dtGiRRv3F/ptBAgQIECA\nAAECMy/wtJkfgQEQIECAAAECBAgQIEBgSgIapClBehsCBAgQIECAAAECBGZfQIM0+3NoBAQI\nECBAgAABAgQITElAgzQlSG9DgAABAgQIECBAgMDsC2iQZn8OjYAAAQIECBAgQIAAgSkJaJCm\nBOltCBAgQIAAAQIECBCYfQEN0uzPoREQIECAAAECBAgQIDAlAQ3SlCC9DQECBAgQIECAAAEC\nsy+gQZr9OTQCAgQIECBAgAABAgSmJKBBmhKktyFAgAABAgQIECBAYPYFNEizP4dGQIAAAQIE\nCBAgQIDAlAQ0SFOC9DYECBAgQIAAAQIECMy+gAZp9ufQCAgQIECAAAECBAgQmJKABmlKkN6G\nAAECBAgQIECAAIHZF9Agzf4cGgEBAgQIECBAgAABAlMS0CBNCdLbECBAgAABAgQIECAw+wIa\npNmfQyMgQIAAAQIECBAgQGBKAhqkKUF6GwIECBAgQIAAAQIEZl9AgzT7c2gEBAgQIECAAAEC\nBAhMSUCDNCVIb0OAAAECBAgQIECAwOwLaJBmfw6NgAABAgQIECBAgACBKQlokKYE6W0IECBA\ngAABAgQIEJh9AQ3S7M+hERAgQIAAAQIECBAgMCUBDdKUIL0NAQIECBAgQIAAAQKzL6BBmv05\nNAICBAgQIECAAAECBKYkoEGaEqS3IUCAAAECBAgQIEBg9gU0SLM/h0ZAgAABAgQIECBAgMCU\nBDRIU4L0NgQIECBAgAABAgQIzL6ABmn259AICBAgQIAAAQIECBCYkoAGaUqQ3oYAAQIECBAg\nQIAAgdkX0CDN/hwaAQECBAgQIECAAAECUxLQIE0J0tsQIECAAAECBAgQIDD7Ahqk2Z9DIyBA\ngAABAgQIECBAYEoCGqQpQXobAgQIECBAgAABAgRmX0CDNPtzaAQECBAgQIAAAQIECExJQIM0\nJUhvQ4AAAQIECBAgQIDA7AtokGZ/Do2AAAECBAgQIECAAIEpCWiQpgTpbQgQIECAAAECBAgQ\nmH0BDdLsz6ERECBAgAABAgQIECAwJQEN0pQgvQ0BAgQIECBAgAABArMvoEGa/Tk0AgIECBAg\nQIAAAQIEpiSgQZoSpLchQIAAAQIECBAgQGD2BTRIsz+HRkCAAAECBAgQIECAwJQENEhTgvQ2\nBAgQIECAAAECBAjMvoAGafbn0AgIECBAgAABAgQIEJiSgAZpSpDehgABAgQIECBAgACB2RfQ\nIM3+HBoBAQIECBAgQIAAAQJTEuh6g7RbnF4wJStvQ4AAAQIECBAgQIBAxwW63iC9M/P3reSs\nZNOOz6XhESBAgAABAgQIECDwUwp0vUHq87wxT76avLi/wyMBAgQIECBAgAABAgTWFmipQVqe\nwX8teX/y82tD2CZAgAABAgQIECBAgEArDdL5mer/kPxt8nvJtck/JL+etGKQoSoCBAgQIECA\nAAECBOYSaKk5+F4gfjP5neTq5FeTv0uqWXpbclCyLFEECBAgQIAAAQIECDQq0FKD1J/ij+fJ\nHsn+yTlJNUWnJhcnK5O7kvrOkiJAgAABAgQIECBAoDGBTRob7+Bwv5yNyvHJq5NaDvx5STVP\nziQFQREgQIAAAQIECBBoTaDlBqk/16vy5KP9jd5ji2fW1iKwSYAAAQIECBAgQKA9AY3A6Dl/\nbPRuewkQIECAAAECBAgQ6LJA1xukozN5S5I6S6QIECBAgAABAgQIECAwp0DXL7H7YUZfUQQI\nECBAgAABAgQIEFivQNcbpFEAS7Nz62Sz5IHk3uTBRBEgQIAAAQIECBAg0LhA1y+x60/vi/Lk\nw8kdyd3JDUndC+mWpJqk65Kzku0SRYAAAQIECBAgQIBAowItnEF6R+b25N783pTHrybVJFVj\nVGeSnpEsT+reR4cntex33StJESBAgAABAgQIECDQmEDXG6QjMp/VHF2UnJRcnoyqhdlZN459\nT3JesiK5LFEECBAgQIAAAQIECDQk0PVL7A7LXF6f1OO6mqOa7seTS5ODk/uTIxNFgAABAgQI\nECBAgEBjAl1vkPbKfNYldavHnNd7ctwVyc5jHu8wAgQIECBAgAABAgQ6JND1Bum2zNWLk8Vj\nzlmtcFdNVS3goAgQIECAAAECBAgQaEyg6w3SxzKfz00+lew7x9z2v4NU31XaIrlwjmO9RIAA\nAQIECBAgQIBARwW6vkhDrUa3fXJKckhya3JLcldyX7IkqVXsdkmWJWuSE5KvJIoAAQIECBAg\nQIAAgcYEut4g1eIL700+nZyaHJCsfSbpoexbmdQKdqcnNyeKAAECBAgQIECAAIEGBbreIPWn\ntFaye01vo84a1f2PNk/qxrGrEkWAAAECBAgQIECAAIEFXf8O0qgprkvrdkqOS+ryO0WAAAEC\nBAgQIECAAIEfCbTYINXA90iOT55VG4oAAQIECBAgQIAAAQIl0GqDZPYJECBAgAABAgQIECDw\nEwIapJ8gsYMAAQIECBAgQIAAgVYFWlmk4QWZ4LcPTPKuvefvzOObBva/Nc9vHNje0Kfb5QfP\nTDYd8w3quF2T/zDm8Q4jQIAAAQIECBAgQOBJEGilQdohdq8e8Ksbw1btn9RS4P06JU+m0SA9\nnPe5JVncf+P1PO6Y15+fVKNUP6sIECBAgAABAgQIEHgKBFppkD4X20UDvq/P848kByWXDOyf\n1tNaOrzORo1b++XA3xr3YMcRIECAAAECBAgQIPDkCPgO0pPj6l0JECBAgAABAgQIEJhBAQ3S\nDE6aj0yAAAECBAgQIECAwJMj0GqD9GA4b05WPzms3pUAAQIECBAgQIAAgVkUaOU7SGvPzfnZ\nUVEECBAgQIAAAQIECBB4QqDVM0gF8PRkz2Tf2kht+eMHfyVAgAABAgQIECBAoFWBFhuk5Zns\nOntUl9ldkZyWVJ2bnJJsVhuKAAECBAgQIECAAIH2BFq7xG5ZpvjyZNvkqmSLpF91b6STksOS\nfZIfJooAAQIECBAgQIAAgYYEWjuDdEbmti6tqxvE7pFUs9Svw/Pk1KRu2Pq6/k6PBAgQIECA\nAAECBAi0I9Bag3Rgpvb9yZdHTPGj2XdyUjd5fdmI1+0iQIAAAQIECBAgQKDjAi01SEsyl0uT\na+aY00fy2nd7x81xmJcIECBAgAABAgQIEOiiQEsN0n2ZwNuTl8wxkdVE1SV2V89xjJcIECBA\ngAABAgQIEOioQEsNUk3hZ5OjkuOSrZLB2iYb5yRbJxcPvuA5AQIECBAgQIAAAQJtCLTWIL0l\n07oyOTO5NXl5sltyYXJdcmhydvL5RBEgQIAAAQIECBAg0JhAaw3SvZnfvZOzks2THZKdkmqM\nqo5P6gyTIkCAAAECBAgQIECgQYHW7oNUU3xnckxybLJLsmOyIqkzS4oAAQIECBAgQIAAgYYF\nWmyQ+tNdy3pf30t/n0cCBAgQIECAAAECBBoWaO0Su4an2tAJECBAgAABAgQIEFifgAZpfUJe\nJ0CAAAECBAgQIECgGQENUjNTbaAECBAgQIAAAQIECKxPoOvfQfrdANTNXyety/IDX530hxxP\ngAABAgQIECBAgMBsC3S9Qfq9TM8LN2CK/ig/o0HaADg/QoAAAQIECBAgQGCWBbreIP1qJueC\nZL/k08lHknHqmnEOcgwBAgQIECBAgAABAt0S6HqDdHum65XJF5Nqlk5OvpkoAgQIECBAgAAB\nAgQI/IRAC4s0rM6o39Ab+Zk/IWAHAQIECBAgQIAAAQIEegItNEg11O8mb0tqwYY9E0WAAAEC\nBAgQIECAAIGfEGilQaqBvyfZK/lObSgCBAgQIECAAAECBAisLdBSg7T22G0TIECAAAECBAgQ\nIEBgSECDNMRhgwABAgQIECBAgACBlgU0SC3PvrETIECAAAECBAgQIDAkoEEa4rBBgAABAgQI\nECBAgEDLAhqklmff2AkQIECAAAECBAgQGBLQIA1x2CBAgAABAgQIECBAoGUBDVLLs2/sBAgQ\nIECAAAECBAgMCWiQhjhsECBAgAABAgQIECDQsoAGqeXZN3YCBAgQIECAAAECBIYENEhDHDYI\nECBAgAABAgQIEGhZQIPU8uwbOwECBAgQIECAAAECQwIapCEOGwQIECBAgAABAgQItCygQWp5\n9o2dAAECBAgQIECAAIEhAQ3SEIcNAgQIECBAgAABAgRaFtAgtTz7xk6AAAECBAgQIECAwJCA\nBmmIwwYBAgQIECBAgAABAi0LaJBann1jJ0CAAAECBAgQIEBgSECDNMRhgwABAgQIECBAgACB\nlgU0SC3PvrETIECAAAECBAgQIDAkoEEa4rBBgAABAgQIECBAgEDLAhqklmff2AkQIECAAAEC\nBAgQGBLQIA1x2CBAgAABAgQIECBAoGUBDVLLs2/sBAgQIECAAAECBAgMCWiQhjhsECBAgAAB\nAgQIECDQsoAGqeXZN3YCBAgQIECAAAECBIYENEhDHDYIECBAgAABAgQIEGhZQIPU8uwbOwEC\nBAgQIECAAAECQwIapCEOGwQIECBAgAABAgQItCygQWp59o2dAAECBAgQIECAAIEhAQ3SEIcN\nAgQIECBAgAABAgRaFtAgtTz7xk6AAAECBAgQIECAwJCABmmIwwYBAgQIECBAgAABAi0LaJBa\nnn1jJ0CAAAECBAgQIEBgSECDNMRhgwABAgQIECBAgACBlgU0SC3PvrETIECAAAECBAgQIDAk\noEEa4rBBgAABAgQIECBAgEDLAhqklmff2AkQIECAAAECBAgQGBLQIA1x2CBAgAABAgQIECBA\noGUBDVLLs2/sBAgQIECAAAECBAgMCWiQhjhsECBAgAABAgQIECDQsoAGqeXZN3YCBAgQIECA\nAAECBIYENEhDHDYIECBAgAABAgQIEGhZQIPU8uwbOwECBAgQIECAAAECQwIapCEOGwQIECBA\ngAABAgQItCygQWp59o2dAAECBAgQIECAAIEhAQ3SEIcNAgQIECBAgAABAgRaFtAgtTz7xk6A\nAAECBAgQIECAwJCABmmIwwYBAgQIECBAgAABAi0LaJBann1jJ0CAAAECBAgQIEBgSECDNMRh\ngwABAgQIECBAgACBlgU0SC3PvrETIECAAAECBAgQIDAkoEEa4rBBgAABAgQIECBAgEDLAq03\nSIsy+bsn27T8h8DYCRAgQIAAAQIECBD4sUALDdL2GeoHk48OTPrWef6B5MHk2uSu5IrkhEQR\nIECAAAECBAgQINCowCYdH/czM77Lk52TS3tjXZzHS5K9k8eSLyR3Ji9NTkvqjNKxSb2mCBAg\nQIAAAQIECBBoSKDrZ5Delrms5ujE5ODevB6Xx2qOPpTUa69MjkienZyRHJMcmCgCBAgQIECA\nAAECBBoT6HqDtF/m84bk3cnq3tzun8d7kzpLdHtvXz08nLw5uTk5KFEECBAgQIAAAQIECDQm\n0PUGqS4h/GYyeLnco9m+KXkkWbvquJVJnU1SBAgQIECAAAECBAg0JtD1Bukbmc9XJdsOzOul\nef6cZLuBff2nO+bJPsm3+zs8EiBAgAABAgQIECDQjkDXG6QPZyo3S76V1KV1VX+eVON0frJT\n0q8X5kk1T2uSC/o7PRIgQIAAAQIECBAg0I5A11ex+3qmshZd+NPki8l3kmqOrkrekKxI/j2p\nM0y1HPjjydFJHacIECBAgAABAgQIEGhMoOtnkGo66/5HuybvSpYmRyZHJQuTWvL7ecmWySeT\nvZJa3U4RIECAAAECBAgQINCgQNfPIPWn9Ht5Ukt9VxYl9V2jWuL7oeSWpFa1UwQIECBAgAAB\nAgQINC7QSoM0OM1LslFnjlYlDySjVrPLbkWAAAECBAgQIECAQGsCLVxiV3P6oqQWbLgjuTup\neyNdndTZo2qSrkvOSkatbJfdigABAgQIECBAgACBFgRaOIP0jkzkyb3JvCmPX02qSarGaOvk\nGcny5I3J4cnxyccTRYAAAQIECBAgQIBAYwJdb5COyHxWc3RRclJyeTKqasGGWgb8Pcl5yYrk\nskQRIECAAAECBAgQINCQQNcvsTssc3l9Uo/rao5qumt570uTg5P7k1rpThEgQIAAAQIECBAg\n0JhA1xukvTKfdUnd6jHn9Z4cd0VSK9wpAgQIECBAgAABAgQaE+h6g3Rb5vPFSa1aN07VfZKq\nqaoFHBQBAgQIECBAgAABAo0JdL1B+ljm87nJp5J955jb/neQ6rtKWyQXznGslwgQIECAAAEC\nBAgQ6KhA1xdpqNXotk9OSQ5Jbk1uSe5K7kuWJLWK3S7JsmRNckLylUQRIECAAAECBAgQINCY\nQNcbpFp84b3Jp5NTkwOStc8kPZR9K5Nawe705OZEESBAgAABAgQIECDQoEDXG6T+lNZKdq/p\nbdRZo7r/0eZJ3Th2VaIIECBAgAABAgQIECCwoOvfQRo1xXVp3U7JcUldfqcIECBAgAABAgQI\nECDwI4EWG6Qa+B7J8cmzakMRIECAAAECBAgQIECgBFptkMw+AQIECBAgQIAAAQIEfkJAg/QT\nJHYQIECAAAECBAgQINCqQCuLNLwgE/z2gUnetff8nXl808D+t+b5jQPbG/q0FoF4R7LpmG+w\n45jHOYwAAQIECBAgQIAAgSdRoJUGaYcYvnrAsW4MW7V/UkuB9+uUPJlGg1SNUd1XadwGqe7F\npAgQIECAAAECBAgQeIoFWmmQPhfnRQPWr8/zjyQHJZcM7J/W0+/njV47wZvtl2Mvm+B4hxIg\nQIAAAQIECBAg8CQI+A7Sk4DqLQkQIECAAAECBAgQmE0BDdJszptPTYAAAQIECBAgQIDAkyDQ\naoP0YCxvTlY/CabekgABAgQIECBAgACBGRVo5TtIa0/P+dlRUQQIECBAgAABAgQIEHhCoNUz\nSE8AeEKAAAECBAgQIECAAIG+QMsN0tODsGeybw9jyz6KRwIECBAgQIAAAQIE2hRosUFanqmu\ny+vqe0hXJKclVecmpySb1YYiQIAAAQIECBAgQKA9gda+g1Q3b7082Ta5Ktki6VfdPPak5LBk\nn+SHiSJAgAABAgQIECBAoCGB1s4gnZG5rUvr9k/2SKpZ6tfheXJq8vzkdf2dHgkQIECAAAEC\nBAgQaEegtQbpwEzt+5Mvj5jiR7Pv5GRV8rIRr9tFgAABAgQIECBAgEDHBVpqkJZkLpcm18wx\np4/kte/2jpvjMC8RIECAAAECBAgQINBFgZYapPsygbcnL5ljIquJqkvsrp7jGC8RIECAAAEC\nBAgQINBRgZYapJrCzyZHJcclWyWDtU02zkm2Ti4efMFzAgQIECBAgAABAgTaEGitQXpLpnVl\ncmZya/LyZLfkwuS65NDk7OTziSJAgAABAgQIECBAoDGB1hqkezO/eydnJZsnOyQ7JdUYVR2f\n1BkmRYAAAQIECBAgQIBAgwKt3QeppvjO5Jjk2GSXZMdkRVJnlhQBAgQIECBAgAABAg0LtHYG\naXCqa1nv65OvJ9slWyaKAAECBAgQIECAAIGGBVppkP5T5vh9yR8mu/fmuxZp+Mukzih9K6lV\n7s5JapEGRYAAAQIECBAgQIBAgwJdv8SuGsC/SX5jYG7/3zx/QfLfk99KLkn+PXlR8p+Tn0sO\nSB5PFAECBAgQIECAAAECDQl0/QzS72Yuqzn6p95jfe/onqRWqXtDckRyYHJ08tKkmqZfSn47\nUQQIECBAgAABAgQINCbQ9TNIh2Y+704OSX7Ym9tb8vjp5B+Sv+7t6z/8jzx5Y1JN0if6Oz0S\nIECAAAECBAgQINCGQNfPINUqdXUJXb85qlmts0ePJVfWxlpV+29Ilq+13yYBAgQIECBAgAAB\nAg0IdL1BuilzWJfQ1T2P+vWreVLj3qO/Y+CxzqjVfZJWDOzzlAABAgQIECBAgACBRgS63iDV\npXRLk7qc7jeTE5P/L6lV66pRem3Sr7L4UFKr230hUQQIECBAgAABAgQINCbQ9e8g/Vnm8z8m\n9V2kV/bm9vu9fX+cx/OSP0jqe0kvS3ZKLk4+lSgCBAgQIECAAAECBBoT6HqDVN8pOiyps0cv\nT65PPpN8L/lvyabJrycvTX6QnJnUvZIUAQIECBAgQIAAAQINCnS9QepP6d/kSWWw7s3Gf0nq\n0rpazOGm5NFEESBAgAABAgQIECDQqEArDdJc09tfuW6uY7xGgAABAgQIECBAgEADAl1fpKGB\nKTREAgQIECBAgAABAgSmJaBBmpak9yFAgAABAgQIECBAYOYFNEgzP4UGQIAAAQIECBAgQIDA\ntAQ0SNOS9D4ECBAgQIAAAQIECMy8gAZp5qfQAAgQIECAAAECBAgQmJaABmlakt6HAAECBAgQ\nIECAAIGZF9AgzfwUGgABAgQIECBAgAABAtMS0CBNS9L7ECBAgAABAgQIECAw8wIapJmfQgMg\nQIAAAQIECBAgQGBaAhqkaUl6HwIECBAgQIAAAQIEZl5AgzTzU2gABAgQIECAAAECBAhMS0CD\nNC1J70OAAAECBAgQIECAwMwLaJBmfgoNgAABAgQIECBAgACBaQlokKYl6X0IECBAgAABAgQI\nEJh5AQ3SzE+hARAgQIAAAQIECBAgMC0BDdK0JL0PAQIECBAgQIAAAQIzL6BBmvkpNAACBAgQ\nIECAAAECBKYloEGalqT3IUCAAAECBAgQIEBg5gU0SDM/hQZAgAABAgQIECBAgMC0BDRI05L0\nPgQIECBAgAABAgQIzLyABmnmp9AACBAgQIAAAQIECBCYloAGaVqS3ocAAQIECBAgQIAAgZkX\n0CDN/BQaAAECBAgQIECAAAEC0xLQIE1L0vsQIECAAAECBAgQIDDzAhqkmZ9CAyBAgAABAgQI\nECBAYFoCGqRpSXofAgQIECBAgAABAgRmXkCDNPNTaAAECBAgQIAAAQIECExLQIM0LUnvQ4AA\nAQIECBAgQIDAzAtokGZ+Cg2AAAECBAgQIECAAIFpCWiQpiXpfQgQIECAAAECBAgQmHkBDdLM\nT6EBECBAgAABAgQIECAwLQEN0rQkvQ8BAgQIECBAgAABAjMvoEGa+Sk0AAIECBAgQIAAAQIE\npiWgQZqWpPchQIAAAQIECBAgQGDmBTRIMz+FBkCAAAECBAgQIECAwLQENEjTkvQ+BAgQIECA\nAAECBAjMvIAGaean0AAIECBAgAABAgQIEJiWgAZpWpLehwABAgQIECBAgACBmRfQIM38FBoA\nAQIECBAgQIAAAQLTEtAgTUvS+xAgQIAAAQIECBAgMPMCGqSZn0IDIECAAAECBAgQIEBgWgIa\npGlJeh8CBAgQIECAAAECBGZeQIM081NoAAQIECBAgAABAgQITEtAgzQtSe9DgAABAgQIECBA\ngMDMC2iQZn4KDYAAAQIECBAgQIAAgWkJaJCmJel9CBAgQIAAAQIECBCYeQEN0sxPoQEQIECA\nAAECBAgQIDAtAQ3StCS9DwECBAgQIECAAAECMy+gQZr5KTQAAgQIECBAgAABAgSmJaBBmpak\n9yFAgAABAgQIECBAYOYFWm+QFmUGd0+2mfmZNAACBAgQIECAAAECBH5qgZ+mQXp6fvueyb69\nT7HlT/1pnpw32D5v+8HkowNvv3WefyB5MLk2uSu5IjkhUQQIECBAgAABAgQINCqwIQ3S8lid\nn1RzUU3FaUnVuckpyWa1MU/qmfkclydHJ7v1PtPiPF6SHJPU8y8kFyTVNNVYqnHaEJf8mCJA\ngAABAgQIECBAYJYFJm0ElmWw1XAckVyd3Jj0a2GenJR8I9m8v/Mpfnxbfv/OyYnJwb3Pclwe\n904+lNRrr0xqPM9OzkiqcTowUQQIECBAgAABAgQINCYwaYNUDURdWrd/skdSzVK/Ds+TU5Pn\nJ6/r73yKH/fL778heXeyuvdZ6rPfmxyb3N7bVw8PJ29Obk4OShQBAgQIECBAgAABAo0JTNog\n1ZmV9ydfHuH0aPadnKxKXjbi9adi1yb5pd9MHhv45fU5b0oeGdjXf1rHrUzqbJIiQIAAAQIE\nCBAgQKAxgUkapCWxWZpcM4dRNR3f7R03x2Eb7aW63O9VybYDv/HSPH9Ost3Avv7THfNkn+Tb\n/R0eCRAgQIAAAQIECBBoR2CSBum+sNQlaS+Zg6eaqLrErr6fNB/qw/kQtWjEt5K6tK7qz5Nq\nnGqhiZ2Sfr0wT6p5WpNc0N/pkQABAgQIECBAgACBdgQmaZBK5bPJUUktdLBVMlh1L6FzkloN\n7uLBF57C51/P765FF2o1uy8mdWbofclVySuSFcmVyfeSuhRv9+T3k+8kigABAgQIECBAgACB\nxgQmbZDeEp/6js6Zya3Jy5PdkguT65JDk7OTzyfzper+R7sm70rqEsEjk2ryFiaLk+cldQ+n\nTyZ7JR9KFAECBAgQIECAAAECDQpM2iDdG6O9k7OSzZMdkrpMrRqjquOTaj7mW9UZohOT5Uld\ncvezyb7Jnkk1TXU27DXJvyWKAAECBAgQIECAAIFGBTbZgHHfmZ+py9ZqmexdklrYYEVSZ5Zm\noep7UnXmaFXyQDJqNbvsVgQIECBAgAABAgQItCYw6Rmk/QL0rB7So3m8Prks6TdHddla3Svp\nV5P5VC/Kh6kFG+5I7k7q3ki1kMQtSTVJdXlgnRUbtbJddisCBAgQIECAAAECBFoQmLRBOi8o\n30pqgYNRVe9Xixy8dNSLT9G+d+T3Xp68IflB8tXk75O/TC5K/leyRfLGpBZveG2iCBAgQIAA\nAQIECBBoUGBDLrF7Rpw+lxydnJ3M5zoiH+7kpBqhk5JqlEZVnfnaP3lPUk3giqTOjCkCBAgQ\nIECAAAECBBoSmPQMUtGcnVyT1OpwpybVXMzXOiwfrC4DrMd1NUf12R9PLk0OTu5PjkwUAQIE\nCBAgQIAAAQKNCWxIg/T9GP1S8k/J25JaHrtWtJuPtVc+VF1St3rMD3dPjrsi2XnM4x1GgAAB\nAgQIECBAgECHBDakQarL8u5Lfi05O/mt5J+T7ZPHkvlUt+XDvDhZPOaHqiW/q6mqBRwUAQIE\nCBAgQIAAAQKNCWxIg9QnquWxX5/8UfKy5GvJc5JaCGG+XHb3sXyW5yafSuq+R+uq+rz1HaT6\nrlIt2HBhoggQIECAAAECBAgQaExgQxZpWJvo5Oy4IalltL+SLErmS308H6TObJ2SHJLcmtyS\n3JXUWbAlSS06sUuyLFmTnJDUOBQBAgQIECBAgAABAo0JTKNBKrJzkmo8Lkg2TeZL1eIL700+\nnZyaHJCsfSbpoexbmdQKdqcnNyeKAAECBAgQIECAAIEGBSZtkF4do1rIYFRdkp2/mPw/yeWj\nDngK912f3/2a3u+vs0ZbJ7WwxB3JqkQRIECAAAECBAgQIEBgwfoapFq0YHFyd1KXn9XZlbqE\nri5bG1W1wl3db+jBUS/Ok311ad3zkroh7PsSDVIQFAECBAgQIECAAAECCxasb5GGWp3ue8kL\ne1j/2tuufXPlrb3j5+vDHvlgxyfPmq8f0OciQIAAAQIECBAgQGDjC6zvDFLd6+japH9Z3Wfz\nfF1njwY//ZWDG54TIECAAAECBAgQIEBgFgTW1yCtfSbo2FkYlM9IgAABAgQIECBAgACBDRFY\nX4M07nvW+/x88r+TWjluvtUL8oHePvChdu09f2ce3zSwvxrCGwe2N/Rp3Uup3re+vzVOLR/n\nIMcQIECAAAECBAgQIPDkCmxIg3R4PtLBydG9j1b3Fzo3qdXhbkvekNSlePOpdsiHqRX4+lU3\nhq2qm8MONnSnZHsaDdI2eZ/fSMZd8nyrHKsIECBAgAABAgQIEHiKBSZtkA7L5/3r5IfJMUk1\nRX+R/EzyueRlySeSFyfXJfOl6rPV6nv9en2efCQ5KKnlyadddV+lV0zwpvvl2MsmON6hBAgQ\nIECAAAECBAg8CQLrW8Vu7V9Zl6TdkNTNVuvMy6FJ3VPotOT/SF7U265GShEgQIAAAQIECBAg\nQGCmBCZpkOrY5yZ1huiK3ih/rff4qd5j3ZD1qmTv3rYHAgQIECBAgAABAgQIzIzAJA1SXUa3\neXJ7b3R1yVp9F+nupO6P1K86Ztzv3vR/ZmM/1o1sb05Wb+xf7PcRIECAAAECBAgQIDB/BSZp\nkFZlGNUM1cIGVa9KliYXJY8lVXWJ3c8ldSZpPtf5+XDLk6/M5w/psxEgQIAAAQIECBAgsHEF\nJmmQ6pPVanVHJF/oPa/vIf1ZUvXfky8n1SydnSgCBAgQIECAAAECBAjMlMCkq9j9YUZXZ42q\nSbo/+f3ki0nVK5KFyeuS+h7SfK+n5wPuntQ9i76WbJnUpXeKAAECBAgQIECAAIFGBSZtkGp5\n7yOTo5JHkjqD1K+35MkNSTVO87nq0rpade//Sqqh+3JSlw2em3w3+ePEd5OCoAgQIECAAAEC\nBAi0JjBpg9T3ebj/ZOCxv7LdwK5593RZPtHlybZJneWqs0f9qmbppKSWKN8nqWZQESBAgAAB\nAgQIECDQkMCk30GadZozMoC6tK7OGO2RVLPUr8Pz5NTk+UldJqgIECBAgAABAgQIEGhMoLUG\n6cDM7/uTuqxu7Xo0O05OarW+l639om0CBAgQIECAAAECBLov0FKDtCTTWQtMXDPHtNb3qup7\nSHWcIkCAAAECBAgQIECgMYGWGqT7Mrd1k9uXzDHH1UTVJXZXz3GMlwgQIECAAAECBAgQ6KhA\nSw1STeFnk6OS45KtksHaJhvnJFsnFw++4DkBAgQIECBAgAABAm0IrG8Vu83CsCFNVF2qtmYe\nEtZS5AclZya1IMMPkvru0YVJLdzwjOTs5POJIkCAAAECBAgQIECgMYH1NT91A9WHNiC1XPZ8\nrHvzofZOzko2T3ZIdkoOTaqOT+oMkyJAgAABAgQIECBAoEGB9Z1Buiwm9b2dSeu6SX9gIx5/\nZ37XMcmxyS7JjsmKZGWiCBAgQIAAAQIECBBoWGB9DdLvddimLq27MamzaHWWTBEgQIAAAQIE\nCBAg0LjA+i6xm4unbri6Z7Jv76At5zr4KXxt+/zuDyYfHfgMtRDDB5IHk2uTu5IrkhMSRYAA\nAQIECBAgQIBAowIb0iAtj9X5STUX1VScllSdm5yS1MIO86WemQ9yeXJ0slvvQy3O4yVJXWZX\nz7+QXJBU01RjqcZpQ1zyY4oAAQIECBAgQIAAgVkWmLQRWJbBVsNxRFL3CqpL1Pq1ME9qcYZv\nJLUAwnyot+VD7JycmBzc+0DH5bEWavhQUq+9MqnxPDs5I6nG6cBEESBAgAABAgQIECDQmMCk\nDVI1EHVpXS2JvUdSzVK/Ds+TU5O60err+juf4sf98vtvSN6drO59lvrstZpdLdJwe29fPTyc\nvDm5OTkoUQQIECBAgAABAgQINCYwaYNUZ1ben3x5hFMtenBysip52YjXn4pdm+SXfjN5bOCX\n1+e8KXlkYF//aR1Xq9nV2SRFgAABAgQIECBAgEBjApM0SEtiszS5Zg6jajq+2ztujsM22kt1\nud+rkm0HfuOlef6cZLuBff2nteT3Psm3+zs8EiBAgAABAgQIECDQjsAkDdJ9YalL0l4yB081\nUXWJXX0/aT7Uh/MhatGIbyV1aV3VnyfVONVCE3WT2H69ME+qeVqTXNDf6ZEAAQIECBAgQIAA\ngXYEJmmQSuWzyVFJLXSwVTJY22TjnKRWg7t48IWn8PnX87tr0YVaze6LSZ0Zel9yVfKKZEVy\nZfK9pC7F2z35/eQ7iSJAgAABAgQIECBAoDGBSRukt8SnvqNzZnJr8vJkt+TC5Lrk0OTs5PPJ\nfKm6/9GuybuSukTwyKSavIXJ4uR5Sd3D6ZPJXsmHEkWAAAECBAgQIECAQIMCkzZI98Zo7+Ss\nZPNkh6QuU6vGqOr4pJqP+VZ1hujEZHlSl9z9bLJvsmdSTVOdDXtN8m+JIkCAAAECBAgQIECg\nUYFNNmDcd+Zn6rK1WiZ7l6QWNliR1JmlWahH8yHr7FdFESBAgAABAgQIECBA4AmBDWmQ+j9c\njcb1vfT3eSRAgAABAgQIECBAgMDMCmxIg/QrGe1/TrZPnp7Ud3nWrrOz42Nr77RNgAABAgQI\nECBAgACB+SwwaYP0WxnMX44xoC+OcYxDCBAgQIAAAQIECBAgMK8EJm2Q/jif/sHkjck/J3ck\no+qxUTvtI0CAAAECBAgQIECAwHwWmKRBqqWwn53UCnYfn8+D8tkIECBAgAABAgQIECCwIQKT\nLPP9g/yC+5I6g6QIECBAgAABAgQIECDQOYFJGqS6bK6+W1T3C5rk5zqHZkAECBAgQIAAAQIE\nCHRTYNJG53fD8FDy18kByfJk2xGp1e0UAQIECBAgQIAAAQIEZkpg0gbpMxldLe/9m0mdTbox\nqRvHrp0/zD5FgAABAgQIECBAgACBmRKYZJGGGtg3k5VjjPCqMY5xCAECBAgQIECAAAECBOaV\nwKQN0pvm1af3YQgQIECAAAECBAgQIDBFgUkvsRvnVy/KQTuMc6BjCBAgQIAAAQIECBAgMJ8E\nJj2DVJ/9sOTwZOtkcVK1MKn3qsUZdk8+kPxRoggQIECAAAECBAgQIDAzApM2SK/PyD6yntFd\nm9e/tZ5jvEyAAAECBAgQIECAAIF5JzDpJXb/LSOom8UemeycPJCcmPxC8trknuSfkgsTRYAA\nAQIECBAgQIAAgZkSmKRBqu8W/XxyUfIXSa1m97Xk5cn/Tj6RHJgcnbwkUQQIECBAgAABAgQI\nEJgpgUkapK0ysvrO0aUDI7w6z18wsF3LgFezdOjAPk8JECBAgAABAgQIECAwEwKTNEirMqK6\nIexzB0ZWDdLyZHDVupuyvcfAMZ4SIECAAAECBAgQIEBgJgQmaZBqQLX4Qq1it29tpL7z44cf\n7aunP5Psn9T3lBQBAgQIECBAgAABAgRmSmDSBum/ZnR1tuiryS8mX0quT05PamGG65Ja6vuS\nRBEgQIAAAQIECBAgQGCmBCZtkOoM0n9MPpd8P3ksOSK5O6nvHW2XnJecmygCBAgQIECAAAEC\nBAjMlMCk90GqwdXZocEzRJdn+1lJLdZwb3J9oggQIECAAAECBAgQIDBzAhvSII0a5KPZWY2S\nIkCAAAECBAgQIECAwMwK/DQNUi35/bzkmUldeleX2SkCBAgQIECAAAECBAjMrMCk30GqgS5L\n/jF5IPl28vnkruSG5JhEESBAgAABAgQIECBAYCYFJj2DtHdG+fdJrWR3cXJlUo3SzyYHJR9I\n6j5Jb04eTxQBAgQIECBAgAABAgRmRmDSBumEjGzz5CXJN9Ya5abZ/pPkD5K/Sr6SKAIECBAg\nQIAAAQIECMyMwCSX2C3KqGqJ71OTtZujGvDDSTVHtyW/nigCBAgQIECAAAECBAjMlMAkDVKd\nbdoquXWOEdZqdiuSn5vjGC8RIECAAAECBAgQIEBgXgpM0iCtzgj+V/JfknX93C557YXJpYki\nQIAAAQIECBAgQIDATAmsq9FZ1yDekBeqAfrbpL6HVN87qtoi+Y3kc0kt3PA3ybYDeXqeKwIE\nCBAgQIAAAQIECMxrgUkXafhERvMzya/18lgeaxW7Jclg1feQButt2fifgzs8J0CAAAECBAgQ\nIECAwHwTmLRBqkvsbtyAQVyzAT/jRwgQIECAAAECBAgQILBRBSZtkN60UT+dX0aAAAECBAgQ\nIECAAIGNKDDpd5A24kfzqwgQIECAAAECBAgQILBxBTRIG9fbbyNAgAABAgQIECBAYB4LaJDm\n8eT4aAQIECBAgAABAgQIbFwBDdLG9fbbCBAgQIAAAQIECBCYxwIapHk8OT4aAQIECBAgQIAA\nAQIbV0CDtHG9/TYCBAgQIECAAAECBOaxgAZpHk+Oj0aAAAECBAgQIECAwMYV0CBtXG+/jQAB\nAgQIECBAgACBeSygQZrHk+OjESBAgAABAgQIECCwcQU0SBvX228jQIAAAQIECBAgQGAeC2iQ\n5vHk+GgECBAgQIAAAQIECGxcAQ3SxvX22wgQIECAAAECBAgQmMcCGqR5PDk+GgECBAgQIECA\nAAECG1dAg7Rxvf02AgQIECBAgAABAgTmsYAGaR5Pjo9GgAABAgQIECBAgMDGFdAgbVxvv40A\nAQIECBAgQIAAgXks0HqDtChzs3uyzTyeIx+NAAECBAgQIECAAIGNJNBCg7R9LD+YfHTAdOs8\n/0DyYHJtcldyRXJCoggQIECAAAECBAgQaFRgk46P+5kZ3+XJzsmlvbEuzuMlyd7JY8kXkjuT\nlyanJXVG6dikXlMECBAgQIAAAQIECDQk0PUzSG/LXFZzdGJycG9ej8tjNUcfSuq1VyZHJM9O\nzkiOSQ5MFAECBAgQIECAAAECjQl0vUHaL/N5Q/LuZHVvbvfP471JnSW6vbevHh5O3pzcnByU\nKAIECBAgQIAAAQIEGhPoeoNUlxB+Mxm8XO7RbN+UPJKsXXXcyqTOJikCBAgQIECAAAECBBoT\n6HqD9I3M56uSbQfm9dI8f06y3cC+/tMd82Sf5Nv9HR4JECBAgAABAgQIEGhHoOsN0oczlZsl\n30rq0rqqP0+qcTo/2Snp1wvzpJqnNckF/Z0eCRAgQIAAAQIECBBoR6Drq9h9PVNZiy78afLF\n5DtJNUdXJW9IViT/ntQZploO/PHk6KSOUwQIECBAgAABAgQINCbQ9TNINZ11/6Ndk3clS5Mj\nk6OShUkt+f28ZMvkk8leSa1upwgQIECAAAECBAgQaFCg62eQ+lP6vTyppb4ri5L6rlEt8f1Q\ncktSq9opAgQIECBAgAABAgQaF2ilQRqc5iXZqDNHq5IHklGr2WW3IkCAAAECBAgQIECgNYEW\nLrGrOX1RUgs23JHcndS9ka5O6uxRNUnXJWclo1a2y25FgAABAgQIECBAgEALAi2cQXpHJvLk\n3mTelMevJtUkVWO0dfKMZHnyxuTw5Pjk44kiQIAAAQIECBAgQKAxga43SEdkPqs5uig5Kbk8\nGVW1YEMtA/6e5LxkRXJZoggQIECAAAECBAgQaEig65fYHZa5vD6px3U1RzXdtbz3pcnByf1J\nrXSnCBAgQIAAAQIECBBoTKDrDdJemc+6pG71mPN6T467IqkV7hQBAgQIECBAgAABAo0JdL1B\nui3z+eKkVq0bp+o+SdVU1QIOigABAgQIECBAgACBxgS63iB9LPP53ORTyb5zzG3/O0j1XaUt\nkgvnONZLBAgQIECAAAECBAh0VKDrizTUanTbJ6ckhyS3JrckdyX3JUuSWsVul2RZsiY5IflK\noggQIECAAAECBAgQaEyg6w1SLb7w3uTTyanJAcnaZ5Ieyr6VSa1gd3pyc6IIECBAgAABAgQI\nEGhQoOsNUn9KayW71/Q26qxR3f9o86RuHLsqUQQIECBAgAABAgQIEFjQSoM0ONV1aV1FESBA\ngAABAgQIECBAYEig64s0DA12YKMus6vL6Z49sM9TAgQIECBAgAABAgQaF2i1Qdoj83588qzG\n59/wCRAgQIAAAQIECBAYEGi1QRog8JQAAQIECBAgQIAAAQI/FmjxO0gbY+4X55ccntTjOLX7\nOAc5hgABAgQIECBAgACBJ1eglQbpBWF8+wDlrr3n78zjmwb2vzXPbxzY3tCnO+UHT0k2HfMN\nxj1uzLdzGAECBAgQIECAAAECGyLQSoO0Q3BePQC0sPd8/zzWvZL6VU3NNBqkeo/d+286xuN+\nOeayMY5zCAECBAgQIECAAAECT6JAK99B+lwMFw3kDT3Tgwb21evf7u33QIAAAQIECBAgQIBA\ngwKtNEgNTq0hEyBAgAABAgQIECAwqYAGaVIxxxMgQIAAAQIECBAg0FmBVhukBzOjNyerOzuz\nBkaAAAECBAgQIECAwMQCrSzSsDbM+dlRUQQIECBAgAABAgQIEHhCoNUzSAXw9GTPZN/aSG35\n4wd/JUCAAAECBAgQIECgVYEWG6Tlmew6e1SX2V2RnJZUnZuckmxWG4oAAQIECBAgQIAAgfYE\nWrvEblmm+PJk2+SqZIukX3VvpJOSw5J9kh8migABAgQIECBAgACBhgRaO4N0Rua2Lq2rG8Tu\nkVSz1K/D8+TU5PnJ6/o7PRIgQIAAAQIECBAg0I5Aaw3SgZna9ydfHjHFj2bfycmq5GUjXreL\nAAECBAgQIECAAIGOC7TUIC3JXC5NrpljTh/Ja9/tHTfHYV4iQIAAAQIECBAgQKCLAi01SPdl\nAm9PXjLHRFYTVZfYXT3HMV4iQIAAAQIECBAgQKCjAi01SDWFn02OSo5LtkoGa5tsnJNsnVw8\n+ILnBAgQIECAAAECBAi0IdBag/SWTOvK5Mzk1uTlyW7Jhcl1yaHJ2cnnE0WAAAECBAgQIECA\nQGMCrTVI92Z+907OSjZPdkh2Sqoxqjo+qTNMigABAgQIECBAgACBBgVauw9STfGdyTHJscku\nyY7JiqTOLCkCBAgQIECAAAECBBoWaO0M0uBU17Le1yfXJrU4Q8sWGb4iQIAAAQIECBAgQEBT\nsGDBW/PH4KqkFmlQBAgQIECAAAECBAg0LND1S+z2ytxuuZ753bn3ei3/XUuBV92c3PKjZ/5C\ngAABAgQIECBAgEAzAl1vkM7JTL5gzNm8aOC4P8rzkwe2PSVAgAABAgQIECBAoAGBrjdIH8wc\nvjepFes+k9SldGvXK7PjpckZyQ96L36l9+iBAAECBAgQIECAAIGGBFpokL6U+fx48qrkn5L3\nJY8n/XpXnlSDVGeM7u7v9EiAAAECBAgQIECAQHsCLSzS8N1MazVAf5qcnvxj0v/eUZ4qAgQI\nECBAgAABAgQI/FighQapRro6qdXqDkqel3wn+e1EESBAgAABAgQIECBA4AmBVhqk/oAvyZNa\n2e7i5BNJXXq3NFEECBAgQIAAAQIECBBY0PXvII2a4nuy8z8lf5fU95HqJrGKAAECBAgQIECA\nAAECC1o7gzQ45X+RjVoC/K+TLySPJIoAAQIECBAgQIAAgYYFWjyDNDjdK7JxxOAOzwkQIECA\nAAECBAgQaFeg5TNI7c66kRMgQIAAAQIECBAgMFJAgzSSxU4CBAgQIECAAAECBFoU0CC1OOvG\nTIAAAQIECBAgQIDASAEN0kgWOwkQIECAAAECBAgQaFFAg9TirBszAQIECBAgQIAAAQIjBTRI\nI1nsJECAAAECBAgQIECgRQENUouzbswECBAgQIAAAQIECIwU0CCNZLGTAAECBAgQIECAAIEW\nBTRILc66MRMgQIAAAQIECBAgMFJAgzSSxU4CBAgQIECAAAECBFoU0CC1OOvGTIAAAQIECBAg\nQIDASAEN0kgWOwkQIECAAAECBAgQaFFAg9TirBszAQIECBAgQIAAAQIjBTRII1nsJECAAAEC\nBAgQIECgRQENUouzbswECBAgQIAAAQIECIwU0CCNZLGTAAECBAgQIECAAIEWBTRILc66MRMg\nQIAAAQIECBAgMFJAgzSSxU4CBAgQIECAAAECBFoU0CC1OOvGTIAAAQIECBAgQIDASAEN0kgW\nOwkQIECAAAECBAgQaFFAg9TirBszAQIECBAgQIAAAQIjBTRII1nsJECAAAECBAgQIECgRQEN\nUouzbswECBAgQIAAAQIECIwU0CCNZLGTAAECBAgQIECAAIEWBTRILc66MRMgQIAAAQIECBAg\nMFJAgzSSxU4CBAgQIECAAAECBFoU0CC1OOvGTIAAAQIECBAgQIDASAEN0kgWOwksLZDeAAAx\nC0lEQVQQIECAAAECBAgQaFFAg9TirBszAQIECBAgQIAAAQIjBTRII1nsJECAAAECBAgQIECg\nRQENUouzbswECBAgQIAAAQIECIwU0CCNZLGTAAECBAgQIECAAIEWBTRILc66MRMgQIAAAQIE\nCBAgMFJgk5F77STQpsDSTTbZ5PynPe1pW7Qy/MdTjzzyyNsz3i+0MmbjJECAAAECBAjMJaBB\nmkvHa60J7LxmzZqDDjvssAWbbrppE2O/6KKL0h89sk8G+4UmBmyQBAgQIECAAIH1CGiQ1gPk\n5fYEXv3qVy9YsmRJEwP/l3/5lzX33XdfE2M1SAIECBAgQIDAOAK+gzSOkmMIECBAgAABAgQI\nEGhCQIPUxDQbJAECBAgQIECAAAEC4whokMZRcgwBAgQIECBAgAABAk0IaJCamGaDJECAAAEC\nBAgQIEBgHAEN0jhKjiFAgAABAgQIECBAoAkBDVIT02yQBAgQIECAAAECBAiMI6BBGkfJMQQI\nECBAgAABAgQINCGgQWpimg2SAAECBAgQIECAAIFxBDRI4yg5hgABAgQIECBAgACBJgQ0SE1M\ns0ESIECAAAECBAgQIDCOgAZpHCXHECBAgAABAgQIECDQhIAGqYlpNkgCBAgQIECAAAECBMYR\n0CCNo+QYAgQIECBAgAABAgSaENAgNTHNBkmAAAECBAgQIECAwDgCrTdIi4K0e7LNOFiOIUCA\nAAECBAgQIECg2wItNEjbZwo/mHx0YCq3zvMPJA8m1yZ3JVckJySKAAECBAgQIECAAIFGBTbp\n+LifmfFdnuycXNob6+I8XpLsnTyWfCG5M3lpclpSZ5SOTeo1RYAAAQIECBAgQIBAQwJdP4P0\ntsxlNUcnJgf35vW4PFZz9KGkXntlckTy7OSM5JjkwEQRIECAAAECBAgQINCYQNcbpP0ynzck\n705W9+Z2/zzem9RZott7++rh4eTNyc3JQYkiQIAAAQIECBAgQKAxga43SHUJ4TeTwcvlHs32\nTckjydpVx61M6mySIkCAAAECBAgQIECgMYGuN0jfyHy+Ktl2YF4vzfPnJNsN7Os/3TFP9km+\n3d/hkQABAgQIECBAgACBdgS63iB9OFO5WfKtpC6tq/rzpBqn85Odkn69ME+qeVqTXNDf6ZEA\nAQIECBAgQIAAgXYEur6K3dczlbXowp8mX0y+k1RzdFXyhmRF8u9JnWGq5cAfT45O6jhFgAAB\nAgQIECBAgEBjAl0/g1TTWfc/2jV5V7I0OTI5KlmY1JLfz0u2TD6Z7JXU6naKAAECBAgQIECA\nAIEGBbp+Bqk/pd/Lk1rqu7Ioqe8a1RLfDyW3JLWqnSJAgAABAgQIECBAoHGBVhqkwWleko06\nc7QqeSAZtZpddisCBAgQIECAAAECBFoTaOESu5rTFyW1YMMdyd1J3Rvp6qTOHlWTdF1yVjJq\nZbvsVgQIECBAgAABAgQItCDQwhmkd2QiT+5N5k15/GpSTVI1Rlsnz0iWJ29MDk+OTz6eKAIE\nCBAgQIAAAQIEGhPoeoN0ROazmqOLkpOSy5NRVQs21DLg70nOS1YklyWKAAECBAgQIECAAIGG\nBLp+id1hmcvrk3pcV3NU013Le1+aHJzcn9RKd4oAAQIECBAgQIAAgcYEut4g7ZX5rEvqVo85\nr/fkuCuSWuFOESBAgAABAgQIECDQmEDXG6TbMp8vTmrVunGq7pNUTVUt4KAIECBAgAABAgQI\nEGhMoOsN0scyn89NPpXsO8fc9r+DVN9V2iK5cI5jvUSAAAECBAgQIECAQEcFur5IQ61Gt31y\nSnJIcmtyS3JXcl+yJKlV7HZJliVrkhOSrySKAAECBAgQIECAAIHGBLreINXiC+9NPp2cmhyQ\nrH0m6aHsW5nUCnanJzcnigABAgQIECBAgACBBgW63iD1p7RWsntNb6POGtX9jzZP6saxqxJF\ngAABAgQIECBAgACBBa00SINTXZfWVRQBAgQIECBAgAABAgSGBLq+SMPQYAc26jK7upzu2QP7\nPCVAgAABAgQIECBAoHGBVhukPTLvxyfPanz+DZ8AAQIECBAgQIAAgQGBVhukAQJPCRAgQIAA\nAQIECBAg8GOBFr+DtDHmvu6r9EvJ4jF/WZ3RUgQIECBAgAABAgQIPMUCrTRIL4jz2wesd+09\nf2ce3zSw/615fuPA9oY+3S0/+Llk0w19Az9HgAABAgQIECBAgMDGF2ilQdohtK8e4K0zPFX7\nJ3WvpH6dkifTaJCuy/s8vf+mYzzul2MuG+M4hxAgQIAAAQIECBAg8CQKtPIdpDqbs2ggb+iZ\nHjSwr17/dm+/BwIECBAgQIAAAQIEGhRopUFqcGoNmQABAgQIECBAgACBSQU0SJOKOZ4AAQIE\nCBAgQIAAgc4KtNogPZgZvTlZ3dmZNTACBAgQIECAAAECBCYWaGWRhrVhzs+OiiJAgAABAgQI\nECBAgMATAq2eQSqAWmVuz2Tf2kht+eMHfyVAgAABAgQIECBAoFWBFhuk5ZnsOntUl9ldkZyW\nVJ2b1DLfm9WGIkCAAAECBAgQIECgPYHWLrFblim+PNk2uSrZIulX3RvppOSwZJ/kh4kiQIAA\nAQIECBAgQKAhgdbOIJ2Rua1L6+oGsXsk1Sz16/A8OTV5fvK6/k6PBAgQIECAAAECBAi0I9Ba\ng3Rgpvb9yZdHTPGj2Xdysip52YjX7SJAgAABAgQIECBAoOMCLTVISzKXS5Nr5pjTR/Lad3vH\nzXGYlwgQIECAAAECBAgQ6KJASw3SfZnA25OXzDGR1UTVJXZXz3GMlwgQIECAAAECBAgQ6KhA\nSw1STeFnk6OS45KtksHaJhvnJFsnFw++4DkBAgQIECBAgAABAm0ItNYgvSXTujI5M7k1eXmy\nW3Jhcl1yaHJ28vlEESBAgAABAgQIECDQmEBrDdK9md+9k7OSzZMdkp2Saoyqjk/qDJMiQIAA\nAQIECBAgQKBBgdbug1RTfGdyTHJsskuyY7IiqTNLigABAgQIECBAgACBhgVaa5DqjNljvfmu\nZb2v76W3a8GiPKmFGn6QuFFsX8UjAQIECBAgQIAAgUYEWrjEri6j+8vk7qRWsvvn5BeTUbVn\ndtZxfzjqRfsIECBAgAABAgQIEOi2QNcbpFqp7l+T30rq7NAtySuSS5NTE0WAAAECBAgQIECA\nAIEnBLreIP3XjPRZycnJzybPTeo+SP+WvC35k0QRIECAAAECBAgQIEDgRwJdb5BqGe87klOS\n+3804gULvpHHA5IvJW9OqolSBAgQIECAAAECBAgQWND1BmnnzHE1QmvWmutV2f4/kyuSdyV1\nCZ4iQIAAAQIECBAgQKBxga43SDdmfg9K6p5Ha1ct2PBrSX0v6WPJuhZuyEuKAAECBAgQIECA\nAIEWBLreIH0+k7h18j+SuiHs2nVrdrwqqcvv/iH59UQRIECAAAECBAgQINCoQNcbpPdlXq9M\n6rtGNye/naxd12THwUndH6m+q1S18McP/kqAAAECBAgQIECAQEsCXW+Q6mav+yZnJDclDyej\n6lvZuU9y0agX7SNAgAABAgQIECBAoA2BTRoY5gMZ4x/0MldDeF2O+dWklgGvxkoRIECAAAEC\nBAgQINCYQAsN0uCU1mV066u6sawiQIAAAQIECBAgQKBBgbnOqDTIYcgECBAgQIAAAQIECLQs\noEFqefaNnQABAgQIECBAgACBIQEN0hCHDQIECBAgQIAAAQIEWhbQILU8+8ZOgAABAgQIECBA\ngMCQgAZpiMMGAQIECBAgQIAAAQItC2iQWp59YydAgAABAgQIECBAYEhAgzTEYYMAAQIECBAg\nQIAAgZYFNEgtz76xEyBAgAABAgQIECAwJKBBGuKwQYAAAQIECBAgQIBAywIapJZn39gJECBA\ngAABAgQIEBgS0CANcdggQIAAAQIECBAgQKBlAQ1Sy7Nv7AQIECBAgAABAgQIDAlokIY4bBAg\nQIAAAQIECBAg0LKABqnl2Td2AgQIECBAgAABAgSGBDRIQxw2CBAgQIAAAQIECBBoWUCD1PLs\nGzsBAgQIECBAgAABAkMCGqQhDhsECBAgQIAAAQIECLQsoEFqefaNnQABAgQIECBAgACBIQEN\n0hCHDQIECBAgQIAAAQIEWhbQILU8+8ZOgAABAgQIECBAgMCQgAZpiMMGAQIECBAgQIAAAQIt\nC2iQWp59YydAgAABAgQIECBAYEhAgzTEYYMAAQIECBAgQIAAgZYFNEgtz76xEyBAgAABAgQI\nECAwJKBBGuKwQYAAAQIECBAgQIBAywIapJZn39gJECBAgAABAgQIEBgS0CANcdggQIAAAQIE\nCBAgQKBlAQ1Sy7Nv7AQIECBAgAABAgQIDAlokIY4bBAgQIAAAQIECBAg0LKABqnl2Td2AgQI\nECBAgAABAgSGBDRIQxw2CBAgQIAAAQIECBBoWUCD1PLsGzsBAgQIECBAgAABAkMCGqQhDhsE\nCBAgQIAAAQIECLQsoEFqefaNnQABAgQIECBAgACBIQEN0hCHDQIECBAgQIAAAQIEWhbQILU8\n+8ZOgAABAgQIECBAgMCQgAZpiMMGAQIECBAgQIAAAQItC2iQWp59YydAgAABAgQIECBAYEhA\ngzTEYYMAAQIECBAgQIAAgZYFNml58MZOgMDEAr+enzhg4p+a7R/493z8D832EHx6AgQIECBA\nYFwBDdK4Uo4jQGDBJpts8palS5e+YtmyZWta4Fi1atXTbrnllvsfffRRDVILE26MBAgQIEAg\nAhokfwwIEBhbYGFq//33X3TkkUcuGvuHZvjAL33pSwtOP/30hTM8BB+dAAECBAgQmFDAd5Am\nBHM4AQIECBAgQIAAAQLdFdAgdXdujYwAAQIECBAgQIAAgQkFNEgTgjmcAAECBAgQIECAAIHu\nCmiQuju3RkaAAAECBAgQIECAwIQCGqQJwRxOgAABAgQIECBAgEB3BTRI3Z1bIyNAgAABAgQI\nECBAYEIBDdKEYA4nQIAAAQIECBAgQKC7Ahqk7s6tkREgQIAAAQIECBAgMKFAizeKXRqjrZPN\nkgeSe5MHE0WAAAECBAgQIECAQOMCrZxBelHm+cPJHcndyQ3J1cktSTVJ1yVnJdsligABAgQI\nECBAgACBRgVaOIP0jsztyb35vSmPX02qSarGqM4kPSNZnrwxOTw5Pvl4oggQIECAAAECBAgQ\naEyg6w3SEZnPao4uSk5KLk9G1cLs3D95T3JesiK5LFEECBAgQIAAAQIECDQk0PVL7A7LXF6f\n1OO6mqOa7seTS5ODk/uTIxNFgAABAgQIECBAgEBjAl1vkPbKfNYldavHnNd7ctwVyc5jHu8w\nAgQIECBAgAABAgQ6JND1Bum2zNWLk8VjzlmtcFdNVS3goAgQIECAAAECBAgQaEyg6w3SxzKf\nz00+lew7x9z2v4NU31XaIrlwjmO9RIAAAQIECBAgQIBARwW6vkhDrUa3fXJKckhya3JLcldy\nX7IkqVXsdkmWJWuSE5KvJIoAAQIECBAgQIAAgcYEut4g1eIL700+nZyaHJCsfSbpoexbmdQK\ndqcnNyeKAAECBAgQIECAAIEGBbreIPWntFaye01vo84a1f2PNk/qxrGrEkWAAAECBAgQIECA\nAIEFXf8O0qgpXpSdlRr7VsmWiSJAgAABAgQIECBAgEAzDdKLMtcfTuqM0d3JDUmtVFffR3og\nuS45K9kuUQQIECBAgAABAgQINCrQwiV278jcntyb35vyWPdFqiapGqO61K4WaVievDE5PDk+\nqcUdFAECBAgQIECAAAECjQl0vUE6IvNZzVEt331ScnkyqvrLfNdCDeclK5LLEkWAAAECBAgQ\nIECAQEMCXW+QDstc1gIN9bh6jnmt1e4uTQ5ObkyOTDRIQVAECEwssE1+ouv/bB1EeSwbdVZe\nESBAgACBTgh0/V/ie2WW6pK6uZqjwYm8JxtXJDsP7vScAAECYwq8NMd9bcxju3TYoRnMZ7o0\nIGMhQIAAgXYFut4g3ZapfXGyOHlkjGlemmOqqaoFGxQBAgQmFViycOHCx04//fRmVgg98cQT\nVz/44INLJoVyPAECBAgQmK8CXW+QPhb4c5NPJXWj2HX9n936DtIvJaclWyQXJooAAQIbJLB8\nea370kYtWrSoLlFWBAgQIECgMwJdb5BqNbrtk1OSQ5Jbk1uSu5L7kvq/nrWK3S7JsmRNckLy\nlUQRIECAAAECBAgQINCYQNcbpPo/m+9NPp3UGaQDkn2TwXooGyuTWsHu9OTmRBEgQIAAAQIE\nCBAg0KBA1xuk/pRenyev6W3UWaO6/9HmSd04dlWiCBAgQIAAAQIECBAg0NRStP3prkvrKooA\nAQIECBAgQIAAAQJDAq2cQRoadDbqMrvXJu9Lrk0UAQIECGwcgf87v6ZWF22pvpHBfqSlARsr\nAQIEZlmg1QZpj0za8Ul9N0mDNMt/gn12AgRmSmDTTTf9b8uWLXv2Tjvt9NhMffAN/LArV658\n2m233Xbtww8/rEHaQEM/RoAAgY0t0GqDtLGd/T4CBAgQiEDuE7XgoIMOetohhxzSxL2i/vZv\n/3bBueeeO8ncPysH/37ShE8PplaQfXdyd297fQ91a46WfMrj0fWheJ0AgekJaJCmZ7n2O+2W\nHYvX3rmO7eXr2D/13Q899NCCW2+t1c67X9///vc3aJD5v70L7r///g362Vn7oTVr1tR/aExU\nZdPKn6F77rlnIpv+wa341HgfffTRif8M3Xvvvc38GaqxTlivSBN5wnbbbdfM/aXuuOOORTH6\np17Wy7XJJptcmX92PXe9B3bogNxv7OP5e+13xhzSy3P82+p/RrRSjz322EPJURnvuN8xf2GO\n3bYVn944v5PHWpxMEXhC4AV59lcD+dc8r3/5fHFgX71e90OaRu2eN6n3nyR1uUn9S+LJrDPz\n5pN8ppk/Nv+CeHAC0F1zbM3DzI97wjEcnePHqvxL94Ic2JRPxnzDWDg/Pqi+39iUT2+8h45r\nlP+4/ZfWjHpjHpeoLFv8M7T2LTjW6dX7e7Ipo94/e9dpstYL9c/0pnwy3vp3967JuFX/96s1\noz8ZF8dxudqhEYSDM87PDoy1xl3p/83Rf2nvPPl2f+OnfKz/MzHuGaT6VfU395Pd2dcZw2fW\nL2uoqkG6f4LxLs2xm01w/KwfWn8P1J+7ehynyqaMWqr68zNJo11/j7V0dn7Sf3Y9PT5bt/QH\nKGNdlfxggjHXDc5buoSsLrG7cwKfLXPsz0xwfBcOrf+gXz3mQOq/b+rPUCv/jVcsZVNG41b9\n+ak/Ry1V/T1Wf68pAusUeH1eqf8g/JV1HuEFAgQIECBAgAABAgSaE2jp/1A1N7kGTIAAAQIE\nCBAgQIDAZAIapMm8HE2AAAECBAgQIECAQIcFWm2Q6vsENyfjXs/b4T8ChkaAAAECBAgQIECA\nAAECBAgQIECAAAECBAgQIPCEQK2ktGfSX1q0tdVMnoDwhAABAgQIECBAgACBdgWWZ+jnJ7U0\nba1k96Wk6m+SU5KWlniucSsCBAgQIECAAAECBBoVWJZx1zrw1RhdmaxI+g3ShXle+/8t2TxR\nBAgQIECAAAECBAgQ6LTAX2V0tUDDL/VGeUEe+w3SojyvM0jVJB2dKAIECBAgQIAAAQIECHRa\n4O6M7t0DIxxskGr34uTe5KO1oQgQIECAAAECBAgQaEugpWW+l2RqlybXzDHFj+S17/aOm+Mw\nLxEgQIAAAQIECBAg0EWBlhqk+zKBtycvmWMiq4l6fnL1HMd4iQABAgQIECBAgACBjgq01CDV\nFH42OSo5LtkqGaxtsnFOsnVy8eALnhMgQIAAAQIECBAgQKCLAtUE3ZTUQgyrkjqjdGtSK9jd\nldR+3z8KgiJAgAABAgQIECBAoA2BZ2aYH0xWJ9UQ9VMN0u8ntZqdIkCAAAECBAgQIECgQYGF\nDY65P+RqhHZJdkxWJCsTRYAAAQIECBAgQIBAwwItNki1kl19z2iz5IGklvWueyMpAgQIECBA\ngAABAgQINCHwoozyw8kdSf+SusHH67L/rGS7RBEgQIAAAQIECBAgQKCzAu/IyPrN0I15flny\nd8knk1rV7mvJbUkdc2fy2kQRIECAAAECBAgQIECgcwJHZETV+FQjtPcco6tLDQ9I/jWp41+e\nKAIECBAgQIAAAQIECHRK4LyMpi6fq+8bjVP1/aS6oWytcqcIECBAgAABAgQIEGhMoOs3it0r\n8/nVpJb0HqfuyUFXJDuPc7BjCBAgQIAAAQIECBDolkDXG6T6btGLk8VjTludQaqm6uoxj3cY\nAQIECBAgQIAAAQIEZkbgd/JJ6ztFn0n2neNT13eQ9k9qwYY1yS8migABAgQIECBAgAABAp0S\nqMbnzUnd56gapVuSf0n+PvlE77EuwaubxNbrjyR/kCgCBAgQIECAAAECBAh0VmC3jKwaoluT\naoQGU83TtclpybMSRYAAAQIECBAgQIBAowJ1hqW1WpIBb51sntSNY1cligABAgQIECBAgAAB\nAgQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgAAB\nAgQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgMAEAi0u8z0Bj0OnKLBr3usXk/Om\n+J5deKstMog9k12Suk/XvyWWng/CQP1Cnu+R1A2dL0/qhs5qtMCvZPcPk8tGv9zc3rqtw9J1\njLrugXfnOl5rbffiDLj+OVT3DLwh+WbyWNJy1X8f7ToGQP1zafUYx3X9kOUZYP0ZqvpOctOP\nnrX9l10z/Ln+u6fM1vXf4fXfA2sSRYBAhwXqP1KuTO7v8Bg3ZGhH5oe+lwzeuPi+bB+/IW/W\nwZ95Rsb0mWTQ56Fsv7GDY53GkH6tZ/WP03izjrzHn/ZMBv8M/f/t3Qe4JWV9x/GH3hcXREFC\npLiwSweRLDzSu9JWDIFgAgFJIgqJSolKExVEDUjEaIIgSJG6NIFFmoAUpQSkl9BC7yu95/vj\nzrvMHc7ZPXcbe+98/8/z25l5Z87MeT/33HPnPTMHyvxJQ6SPU9qNzdnBcw2n61keMaU7HuSP\nn69hUl43zenqg7yfU/r052AHvyAZUBebzP83yf9vsq01qfOejwBTvDpNl24r3IzS71lnlCfi\n8xiyAvn09tdkFHlxyPZy4B3biIccSx4k3yTnknz6vxs5guSE5XjS5jqZzsfpKJI/wIuQPcl/\nkbfI0cTqE1iIyTFivE9gZVryvpPXT7NuaDa0cHkL+nw2uY3sTB4g+QBiVzKWrEraesX2dfp+\nOOlUObndgTxG7uu0QYvavk9fdyEXkHhlcPQ1ktdQrma38QO/Xs578t6UupjkzpFm5RzAUkCB\nISowhn49SvLpSG5B8AoSCFVdxjQuG5eGavqpqj0nLG2u1eh8fK5rICzBcv4AX9Vob/tiTnKf\nJDHzClLfq2FmJhkc5XfN6iyQ368/kxGN1aewnNfSuo12F/sEzmCSv2lrtBwkt4fl9ZO/7fPX\nLHL1Le2vkLZ9EN/rec8+2OR3bB1iKaBAiwQ2o6/55c89/luSG4kDJBConLj9kWQQNAtp1p00\nvEk6rWtuO1SXl6VjB5ENO3Twf2l7tkN7W5vyiX9+17aupuPaCtHo9zKVxw8b7S72CeTELK+b\nf+sAshhtG5BcKbH6C2zPYtwO7N/cyqV56XX+VuXve7OuoCFObXoNDeS8J3fW5MO+DCYtBRRo\nkcBG9PU7ZIGqzw6Qevvh557t8eTe3jZv3Var0OPcXnda63reucP55D9XSY4kee3khMQBEgjU\n35B4bEfWJLuTHUkGTlbfbVDxWbXCyBWAfKF8oWrZyfsFFqbpGXIXmeP9q1vZciW9zutoxVrv\nl2I+79M31draMDuQ8547ALmTLEgy6P4q2YTMRSwFFGiRgAOk3n7YB7BZ/tgc2tvmrdgqt3Hs\nRPKJW27b+BNZnLS9cutKrkTmj+zcxAESCLU6hPn8Lt1dTTOf5MTtMNK2W3/ocr+KQTxy2+q5\nJC7F6Azmc+Jm9Rc4jsUYbdu/udVLK9D7W8jL5ARyNHmB5EO+3DLe5up23pP36/y+PU7yN638\n3mWa96vViaWAAi0R6PZG0ZLu99TN/NHNm2beIP0U6T2yjzFb/wPyXZZne291a+dyhTZfoC8n\nIQ6Q+r8UxrGY101u9cmtL39RTW9lmvZ9SZvrFDofh7w330x2IbnqdhZJ+1UkH05YfQLDmbxC\n8r1a33/6TPJvbhnfk+Q1U8+BLGddm6vbec9oUGL1GtmLjCK5rfxg8iZ5gixALAUUaIFAtzeK\nFnS9py7uxFavk3yilDdL6z2BuZhdjKxGfk4yKLiN5P73ttaadDx/SA+oAThAqmEwuxbZmcSl\nXguz8Dx5lcxTX9Gy+Qvob07SbidNowwqsy4DJqtPYA8mMTmwb9F/EZidXE1yop9bxD5S5V+Z\n5vfrd6TNv2PdznvitB3JLa3NOpSGvM7yQaClgAItEOj2RtGCrk+yi/uzRd4Q7yNLT3JrN8j3\nj+K1TUsp5qPfea3kd2oYye0aST5xjMtF1XJOXqzOAuU1VK6+dd5qaLceQ/fyevlyh27uVq37\ncYd1bW3KbWT5cCZXtK0+gXxnJq+hAzqA5MpI1o3psK4tTZNz3rNc5XZeW5Bm1H62/fLnjPpz\n8Xm1QyC3rxxBvk2uI2uQu4k1cYGjq9WfnfhmQ3btKvRsCZLpePJSlXx5PLUhSdtxWbA6CjxV\ntWaA2dZ6uOp4budp1sVVw0LNFS1dHk2/lydnkkdbatCp25tXjbkts1mnVw1bNFe4PFEB35sm\nyjP9VjpAmn7WHkmBukB+9/IJ7h4kf1zWJZ1OVGhuZe1Fr/M/ylu/Q+/frtpe7LCuDU05QftJ\nh/ys6vxD1boLq+U2TnKV7QZyNcnvWrNGVg13NVe0aPmOqq+rdujzIlVbPrix+v6T53E4W4x+\nAuW9OLeMNatcwZ6lucLld29HzHvP9h0sfG/qgGKTAkNZYHIuNQ9ljy/Rudx+MJb4B+T9P+l8\n6hiffGLbrPNoyLqtmitavpzvkcRlXMsdSvdzS1Q8ti0N1TT3/efE7pJGe9sWcwKbwfQjZNFG\n509jOXafbLS3dfGEymOltgJ06fdfVy65WtT8IOJH1bpduzy2Dc3dznu2qWzyH4zJnSSlMp/3\n7/zurV0anSqgwNAW6PZGMbR73bl3C9KcqyN5E8xJWq4gdcq8tLe18ofifBKj35K/JVuT8sfj\nVOat/gIOkPp7bMDiW+Rp8u8ktx7uTV4gz5AVSdtrRwAyWLyd/DPZmJxI8nv3Q2L1CeTvV/6j\nKHMI0k8g79MXkrxeziH5j3psSn5B0pYruG3+ALDbeU9MLq2MLmP6d2QMyd+6uB1FLAUUaIlA\ntzeKlnS/Xzdz5SNvgpPK8H6Pat/CMLr8HyQnJsXqJeb3Jf5ndkFolAOkBgiLnyH5Xl95/eS1\ndCVZglh9AjF6kBSjR5n/AcnJr9V3ZeRlIMotiZr0F5iHxQymXyPlNfQ68z8l85M218TOe/L3\nPbdF1/++5cOcvdoMZt8VUEABBXoXmItNVyZLkzZ/Gtm7mFs2BRahId+1mbu5wuUJAgszN3LC\nkjMKDExgVjbP62dZ4gdYvdvlg60VyOK9P8QtFVBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQ\nQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUU\nUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEF\nFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEAB\nBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBA\nAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQ\nQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUU\nUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEF\nFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEAB\nBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBAAQUUUEABBRRQQAEFFFBA\nAQUUUEABBRSYcoFZpnwX7kEBBRRQYIgIzE8/PkNmI08Osj7NxfPdlHyBrE7uIS+R6VXDOdBm\nJH9Xn6oOmueR3Efeqtqm1+SDPPb06qPHUUABBRRQQAEFFFBgmgqszN7fIT+epkeZ+jufiV2e\nRfLcSxaZ+oeZ6B5HV8c+tLbVaVXbQrW2aTG7PDs9sbHj6XXsxmFdVEABBQa/wKyDvwv2QAEF\nFFCg5QJL0v+tyB1ke5IrOI+RD7pu5Ankytbr0/iJnMn+52wcY3odu3FYFxVQQIHBL+AAafD/\nDO2BAgoo0HaBRSuA85jePANhHPIBPpcP8tgfYLc9tAIKKDDlAg6QptzQPSiggAIDEch3VbYk\n15DHySbkU+QJchH5E6nXhixkAHACqX+PZXaWc7XkIXIZSa1LPkxOJ9nnBiTv81l/FUktQ/Jd\nncXI9eQUktvSmpXbwj5HRpBbyPmkfLeG2QmV/X+W5Pa8OchN5FzyCqnX+iyk71m3M1mQnE1u\nJROr5Vi5Efk4uZ9cQXKMUlsw8+lqYUmmO5IMkurbVKvfnRT/a1kaTzYn6ePl5DJSf96xTN/y\nHG8g9cp3fEaRc8hz9RW1+XWYX5zE+FVSKmarkfXIh0h+5qeSN0i9cvz8DPMzy3b3kt+T8hoZ\nxvwYkulsJH2/n8So27HzusnPdQWSitVvyMtZqKoYXcNyL6/RPCy3NH6eLEXy/bXbyHnkTWIp\noIACCiiggAIKKNBVYEXWvEMOIjmJzHxuwco0J8j/SOo1joWsm6veyHxOYtOe26tKjWXm/8h+\npL7fzH+J5GT6NVKOl/Zfk1IrM5O2i0n2k+2eIGl7jKxF6pUByR9I1o8nT1fztzNNP+uVgdGD\n5EiS7ZMMCCdWh7EyJ9hvk0eraQaJac8gI/U/pOwv02z7bdKt8ryy3X+SR8ifSW7Ny+OeJyNJ\nqU7fKyrrjmAm+1mpaui07WnVNhlslspA5xaSx+bYL1TzMZuTlNqKmadItottBliZT//3JqkM\n7PK8055kvvw8Ox17FdbfU22bY2e/eVzaVielilGvr9ENeWB5fnkN5DWW/V5HFiWWAgoooIAC\nCiiggAJdBcrJZ05mc9VgDZKT/ZwQ58TyWTI3KTWOmZxs9jpAyn5z0r0JyVWFjUlOhMu+d2c+\nJ+mLk7tJ9j2KpMoAKW0nkXnSSOUqxnMkVxPKc5uJ+ZwA54T9CyTLqY3IM+QuMjsplQFSBjvZ\nzxfJ9mRt0q12ZkWeR/r/0WqjXFE5h6T961VbJuuTtO2bhUlU8c/2x5PyHFdjPlc+7iTDSGo0\nyXaHZqFRR7CcdStV7Z22bQ5S5mDbK8irpJjl57oPyb6+RVI5fgYwcczzmoXkZ7E1yc/xZTI/\nKZUBTga09WoeO8dJ314k25GZSX5mnyM5VgaLpd/FqNfX6H089imyLEnNS75H0qfvE0sBBRRQ\nQAEFFFBAga4C5eTzAbYoJ+dl41OZyUlltik1jpm05QS3XsNZSPuZtcaxVduXa22Z/U3VflCj\nPQOK7GNM1V4GSDnZzsl8vfZiIdtmgJXKSXaWM/BpVo6Tdf9UW5Ht0vaVWlu32Zy4ZyCVAUJ9\nIJDtM1B4jLxQzTOZrAHSozyufsUm+8lzy3PcNQvUaJLlqTVAyuAx+zuc1Cv9vYT8lsxGctz8\n3HchzSqOy9dW9DJA+irb59j71R5XZr9WrTugahjIazSGGST/jpRBMrPvvn6+wXSzLFgKKKDA\nYBLIJ0iWAgoooMD0F7iBQ+YWtno9UC0MqzdOxnz2Xa9bqoU/1huZz0AjNV/fZMK/ZzGXKxX1\nKgOxXNFI5SQ+dSnJCXU9uV0sVbbtW+r79w/1hS7zH6f9QyTfYRnf2OYllvNccpViZGPdQBbH\nsvGrjQecUy2v0mifWotlvzl2vTJw2YBsTN4g15JNydEklUFT+rodKVfT5mZ+ILVStfGJHR50\nQtXW/Hn18hqN4VVkHXI1yUBsFMnr5xByAbEUUECBQSWQ2zosBRRQQIHpL/B4h0OWE/Yp/fDq\nwca+8wl/6vm+yYR/S/uEhmqm+fg0P1ytW7Kajqimh1XTTpNPdGi8v0Nbs2mZqqHT88iq0p79\nNweD1UMnOSn7qG/4CAsZrKxQb5yK82WQ0rwdrtMh0revk3VJ5mclueUtV85SM/VNev43pulb\np2M/SfsrJMepV6+v0c/zoJPJemQ0yWsiP+dfkYNJ84MAmiwFFFBgxhXIG66lgAIKKDD9BXKy\nO5BqnhA3r/rU9zWlJ6Sz13dWzefWttTTfZMJV192YPmJqq05yXdbmtXLc8tVolQ5Zt/Se/+W\nvpcB5Xtrep/r1MfsN87NgUHTPkfJFayBVq4OpSZ19Wck2+RqTJ7PheR4chO5jhxIdiMDrZim\nH3OR8jzKPmIxJ2l69voazQBrfbI02YxsStYlB5A1yCbEUkABBQaNgAOkQfOj8okqoEBLBcpJ\na05g8+X8UkuVmWkwbV5JyCGWrY5zbzW9u5pmEHRJNV8muT0ut2s9XhoGOL2n2r4cs/nw0v5g\nc8UAlssVsPpDRlYLN1TTun19u8x3Mmpu01wudvnZldsQyzZ7MzOa7En2IMPJTuQ4Uq8MQlKz\n9E16/jemG5DYXdt4VPqdwdPkeGYQm1sHnyJ3kbwujiALkpvJxuRj5FFiKaCAAoNCYOZB8Sx9\nkgoooEB7BcoVm3xCX6+dqoWc2E7t2podLtbYaU7gc4vWqVV7vq+T5W+S5sn6kbRdRHL1YHIq\nV6RyEp+T6wy06rU8C1uS3MKVE/DJrW15YE7c65XBSfp0edVY7Ndmud7HFVj+ZLXNQPzP5zHZ\n/+7VY8tkGDPfIhuS3Oa3BEmlj/XKQKSYzlZb8Qbz3a62lc3Oqma+wbT5nPMzTJXvmfUt9fZv\nBopXkhMamz/D8oMkt3GWgWZjExcVUECBGVNg1hnzafmsFFBAAQUqgdOY7kJ+SvI9kpxAZwDz\nV6R8H4XZqVo5+b6UHEoeIzn+5uRYch1JXUOOJf9ALic/Ia+TPLcdSAZQZTDF7IDrKzwix7iY\nfIfcRDI42p+MJ2NIBhuTW7nV7PfkQJKB0I5kG7IfyXFTD5MM1HJl5wxyOhlBdiWPkHI1h9me\n6ka2OobE82xyNFmEfJFkkPQv5DUS+03J4eRHJAONtUgGcFmfwdACpNSTzIwivyRXkhyjWRfS\nkAFQ3M4lOfbbJP1O21HkV2SgdTMPuIysRzIIyzHyfabNyZpkLHmWWAoooIACCiiggAIKdBRY\nkdZ3SAYUzTqIhqxbu7EiVxxyK1vW5aT2RrIUeYjkhLRUTkazzYKloZp+t2r/dKN9p6r976v2\nlavlfZieSN6qlnPC+z0yC6nXzCzsRZ4nOW55fhlILEzqlZPyrB9Wb5zE/Cqsv56Ufb/MfAZM\nq5J6rc9Cttm33thlvvj/kvUnk9LHB5iPf7M+TsMVpDyH8czvT3au2lZimsogKttkUFkqg9u0\nLVQamMYwz/NFUvb5HPP1q0rZ5mfkzdo2jzOfgVmuqOVxPyel1mHmCZL2W6vGTseejXXpY/3Y\n97J8MJmJlCpGvb5G83o7idSfb16vR5Ic01JAAQUUUEABBRRQYKoLzMwec5Xgw1N9z913OB+r\nliezdt9kwpq/ZC6DhYEMgCY8eBIz2WdO2mefxHa9rG6e/GffvVwJGs52y5HmILGXY3baJj/P\nT5BlSDffPLcMWpu3AtLUsT5K65wd1/RvzGBoKbJY/+YpXpqXPcRoBKkPuKZ4x+5AAQUUUEAB\nBRRQQAEFpo1Ac4A0bY7iXhVQQAEFBq1APsGyFFBAAQUUUEABBRRQQAEFEHCA5MtAAQUUUKBN\nAvmeTv5jEvm+jKWAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIK\nKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIAC\nCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCA\nAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiig\ngAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoo\noIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIK\nKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIAC\nCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCAAgoooIACCiiggAIKKKCA\nAgoooIACCiiggAIKKKCAAgoooIACCigwtAT+HyIl1CppcEijAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "sizes\n",
    "barplot(sizes, names.arg=names(counts), xlab=\"number of pulications\", ylab=\"panel size\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "cb21665f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Input</th><th scope=col>Match.type</th><th scope=col>Approved.symbol</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CD24  </td><td>Approved symbol</td><td>CD24  </td><td>CD24 molecule                                </td><td>HGNC:1645 </td><td>6q21    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TET2  </td><td>Approved symbol</td><td>TET2  </td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>DNMT3B</td><td>Approved symbol</td><td>DNMT3B</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>IDH1  </td><td>Approved symbol</td><td>IDH1  </td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>BRAF  </td><td>Approved symbol</td><td>BRAF  </td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AR    </td><td>Approved symbol</td><td>AR    </td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Input & Match.type & Approved.symbol & Approved.name & HGNC.ID & Location\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & CD24   & Approved symbol & CD24   & CD24 molecule                                 & HGNC:1645  & 6q21    \\\\\n",
       "\t2 & TET2   & Approved symbol & TET2   & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24    \\\\\n",
       "\t3 & DNMT3B & Approved symbol & DNMT3B & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21\\\\\n",
       "\t4 & IDH1   & Approved symbol & IDH1   & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34    \\\\\n",
       "\t5 & BRAF   & Approved symbol & BRAF   & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34    \\\\\n",
       "\t6 & AR     & Approved symbol & AR     & androgen receptor                             & HGNC:644   & Xq12    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Input &lt;chr&gt; | Match.type &lt;chr&gt; | Approved.symbol &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | CD24   | Approved symbol | CD24   | CD24 molecule                                 | HGNC:1645  | 6q21     |\n",
       "| 2 | TET2   | Approved symbol | TET2   | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     |\n",
       "| 3 | DNMT3B | Approved symbol | DNMT3B | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 |\n",
       "| 4 | IDH1   | Approved symbol | IDH1   | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     |\n",
       "| 5 | BRAF   | Approved symbol | BRAF   | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     |\n",
       "| 6 | AR     | Approved symbol | AR     | androgen receptor                             | HGNC:644   | Xq12     |\n",
       "\n"
      ],
      "text/plain": [
       "  Input  Match.type      Approved.symbol\n",
       "1 CD24   Approved symbol CD24           \n",
       "2 TET2   Approved symbol TET2           \n",
       "3 DNMT3B Approved symbol DNMT3B         \n",
       "4 IDH1   Approved symbol IDH1           \n",
       "5 BRAF   Approved symbol BRAF           \n",
       "6 AR     Approved symbol AR             \n",
       "  Approved.name                                 HGNC.ID    Location\n",
       "1 CD24 molecule                                 HGNC:1645  6q21    \n",
       "2 tet methylcytosine dioxygenase 2              HGNC:25941 4q24    \n",
       "3 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21\n",
       "4 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34    \n",
       "5 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34    \n",
       "6 androgen receptor                             HGNC:644   Xq12    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# genes to add to whitelist\n",
    "# note I manually looked up gene synonyms for a few of these that werent matching\n",
    "df_white = read.table(file=\"./marker_collation/whitelist_hgnc-symbol-check.csv\",header=TRUE, sep=\",\")\n",
    "head(df_white)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "id": "61a794ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Input</th><th scope=col>Match.type</th><th scope=col>Approved.symbol</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CD24  </td><td>Approved symbol</td><td>CD24  </td><td>CD24 molecule                                </td><td>HGNC:1645 </td><td>6q21    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TET2  </td><td>Approved symbol</td><td>TET2  </td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>DNMT3B</td><td>Approved symbol</td><td>DNMT3B</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>IDH1  </td><td>Approved symbol</td><td>IDH1  </td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>BRAF  </td><td>Approved symbol</td><td>BRAF  </td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AR    </td><td>Approved symbol</td><td>AR    </td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Input & Match.type & Approved.symbol & Approved.name & HGNC.ID & Location\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & CD24   & Approved symbol & CD24   & CD24 molecule                                 & HGNC:1645  & 6q21    \\\\\n",
       "\t2 & TET2   & Approved symbol & TET2   & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24    \\\\\n",
       "\t3 & DNMT3B & Approved symbol & DNMT3B & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21\\\\\n",
       "\t4 & IDH1   & Approved symbol & IDH1   & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34    \\\\\n",
       "\t5 & BRAF   & Approved symbol & BRAF   & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34    \\\\\n",
       "\t6 & AR     & Approved symbol & AR     & androgen receptor                             & HGNC:644   & Xq12    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Input &lt;chr&gt; | Match.type &lt;chr&gt; | Approved.symbol &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | CD24   | Approved symbol | CD24   | CD24 molecule                                 | HGNC:1645  | 6q21     |\n",
       "| 2 | TET2   | Approved symbol | TET2   | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     |\n",
       "| 3 | DNMT3B | Approved symbol | DNMT3B | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 |\n",
       "| 4 | IDH1   | Approved symbol | IDH1   | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     |\n",
       "| 5 | BRAF   | Approved symbol | BRAF   | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     |\n",
       "| 6 | AR     | Approved symbol | AR     | androgen receptor                             | HGNC:644   | Xq12     |\n",
       "\n"
      ],
      "text/plain": [
       "  Input  Match.type      Approved.symbol\n",
       "1 CD24   Approved symbol CD24           \n",
       "2 TET2   Approved symbol TET2           \n",
       "3 DNMT3B Approved symbol DNMT3B         \n",
       "4 IDH1   Approved symbol IDH1           \n",
       "5 BRAF   Approved symbol BRAF           \n",
       "6 AR     Approved symbol AR             \n",
       "  Approved.name                                 HGNC.ID    Location\n",
       "1 CD24 molecule                                 HGNC:1645  6q21    \n",
       "2 tet methylcytosine dioxygenase 2              HGNC:25941 4q24    \n",
       "3 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21\n",
       "4 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34    \n",
       "5 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34    \n",
       "6 androgen receptor                             HGNC:644   Xq12    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_white = df_white %>% distinct()\n",
    "head(df_white)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "3e681490",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Approved.symbol</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>Match.type</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th><th scope=col>size</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TET2  </td><td>TET2  </td><td>4</td><td>Approved symbol</td><td>tet methylcytosine dioxygenase 2             </td><td>HGNC:25941</td><td>4q24    </td><td>4 </td><td>106066032</td><td>106068032</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>DNMT3B</td><td>DNMT3B</td><td>2</td><td>Approved symbol</td><td>DNA methyltransferase 3 beta                 </td><td>HGNC:2979 </td><td>20q11.21</td><td>20</td><td> 31349191</td><td> 31351191</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>IDH1  </td><td>IDH1  </td><td>1</td><td>Approved symbol</td><td>isocitrate dehydrogenase (NADP(+)) 1         </td><td>HGNC:5382 </td><td>2q34    </td><td>2 </td><td>209099953</td><td>209101953</td><td>-</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BRAF  </td><td>BRAF  </td><td>1</td><td>Approved symbol</td><td>B-Raf proto-oncogene, serine/threonine kinase</td><td>HGNC:1097 </td><td>7q34    </td><td>7 </td><td>140418127</td><td>140420127</td><td>-</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AR    </td><td>AR    </td><td>2</td><td>Approved symbol</td><td>androgen receptor                            </td><td>HGNC:644  </td><td>Xq12    </td><td>X </td><td> 66762863</td><td> 66764863</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MYC   </td><td>MYC   </td><td>3</td><td>Approved symbol</td><td>MYC proto-oncogene, bHLH transcription factor</td><td>HGNC:7553 </td><td>8q24.21 </td><td>8 </td><td>128746680</td><td>128748680</td><td>+</td><td>.</td><td>promoter</td><td>2000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & Approved.symbol & hybrid.gene.name & no.total.pubs & Match.type & Approved.name & HGNC.ID & Location & chr & start\\_hg19 & end\\_hg19 & strand & dot2 & type & size\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & TET2   & TET2   & 4 & Approved symbol & tet methylcytosine dioxygenase 2              & HGNC:25941 & 4q24     & 4  & 106066032 & 106068032 & + & . & promoter & 2000\\\\\n",
       "\t2 & DNMT3B & DNMT3B & 2 & Approved symbol & DNA methyltransferase 3 beta                  & HGNC:2979  & 20q11.21 & 20 &  31349191 &  31351191 & + & . & promoter & 2000\\\\\n",
       "\t3 & IDH1   & IDH1   & 1 & Approved symbol & isocitrate dehydrogenase (NADP(+)) 1          & HGNC:5382  & 2q34     & 2  & 209099953 & 209101953 & - & . & promoter & 2000\\\\\n",
       "\t4 & BRAF   & BRAF   & 1 & Approved symbol & B-Raf proto-oncogene, serine/threonine kinase & HGNC:1097  & 7q34     & 7  & 140418127 & 140420127 & - & . & promoter & 2000\\\\\n",
       "\t5 & AR     & AR     & 2 & Approved symbol & androgen receptor                             & HGNC:644   & Xq12     & X  &  66762863 &  66764863 & + & . & promoter & 2000\\\\\n",
       "\t6 & MYC    & MYC    & 3 & Approved symbol & MYC proto-oncogene, bHLH transcription factor & HGNC:7553  & 8q24.21  & 8  & 128746680 & 128748680 & + & . & promoter & 2000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | Approved.symbol &lt;chr&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | Match.type &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TET2   | TET2   | 4 | Approved symbol | tet methylcytosine dioxygenase 2              | HGNC:25941 | 4q24     | 4  | 106066032 | 106068032 | + | . | promoter | 2000 |\n",
       "| 2 | DNMT3B | DNMT3B | 2 | Approved symbol | DNA methyltransferase 3 beta                  | HGNC:2979  | 20q11.21 | 20 |  31349191 |  31351191 | + | . | promoter | 2000 |\n",
       "| 3 | IDH1   | IDH1   | 1 | Approved symbol | isocitrate dehydrogenase (NADP(+)) 1          | HGNC:5382  | 2q34     | 2  | 209099953 | 209101953 | - | . | promoter | 2000 |\n",
       "| 4 | BRAF   | BRAF   | 1 | Approved symbol | B-Raf proto-oncogene, serine/threonine kinase | HGNC:1097  | 7q34     | 7  | 140418127 | 140420127 | - | . | promoter | 2000 |\n",
       "| 5 | AR     | AR     | 2 | Approved symbol | androgen receptor                             | HGNC:644   | Xq12     | X  |  66762863 |  66764863 | + | . | promoter | 2000 |\n",
       "| 6 | MYC    | MYC    | 3 | Approved symbol | MYC proto-oncogene, bHLH transcription factor | HGNC:7553  | 8q24.21  | 8  | 128746680 | 128748680 | + | . | promoter | 2000 |\n",
       "\n"
      ],
      "text/plain": [
       "  Approved.symbol hybrid.gene.name no.total.pubs Match.type     \n",
       "1 TET2            TET2             4             Approved symbol\n",
       "2 DNMT3B          DNMT3B           2             Approved symbol\n",
       "3 IDH1            IDH1             1             Approved symbol\n",
       "4 BRAF            BRAF             1             Approved symbol\n",
       "5 AR              AR               2             Approved symbol\n",
       "6 MYC             MYC              3             Approved symbol\n",
       "  Approved.name                                 HGNC.ID    Location chr\n",
       "1 tet methylcytosine dioxygenase 2              HGNC:25941 4q24     4  \n",
       "2 DNA methyltransferase 3 beta                  HGNC:2979  20q11.21 20 \n",
       "3 isocitrate dehydrogenase (NADP(+)) 1          HGNC:5382  2q34     2  \n",
       "4 B-Raf proto-oncogene, serine/threonine kinase HGNC:1097  7q34     7  \n",
       "5 androgen receptor                             HGNC:644   Xq12     X  \n",
       "6 MYC proto-oncogene, bHLH transcription factor HGNC:7553  8q24.21  8  \n",
       "  start_hg19 end_hg19  strand dot2 type     size\n",
       "1 106066032  106068032 +      .    promoter 2000\n",
       "2  31349191   31351191 +      .    promoter 2000\n",
       "3 209099953  209101953 -      .    promoter 2000\n",
       "4 140418127  140420127 -      .    promoter 2000\n",
       "5  66762863   66764863 +      .    promoter 2000\n",
       "6 128746680  128748680 +      .    promoter 2000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_white = df_white %>% select(\"Approved.symbol\") %>% inner_join(df_pub_count3, by=\"Approved.symbol\")\n",
    "head(df_white)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "760f0673",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_panel = rbind(df_pub_count3, df_white) %>% distinct() %>% arrange(chr, start_hg19, end_hg19)\n",
    "df_panel$is.whitelist = df_panel$Approved.symbol %in% df_white$Approved.symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "id": "839a2bdd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out = df_panel %>% select(-c(\"dot2\",\"Match.type\")) %>% rename(hugo_gene_name=\"Approved.symbol\") %>%\n",
    "select(chr, start_hg19, end_hg19, everything())\n",
    "write.table(df_out, file=\"./marker_collation/Prostate_lit_regions_annotated_05252022.tsv\",\n",
    "           row.names=FALSE, col.names=TRUE, quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "e1c1a01a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# calc overlaps with TCGA and ECM\n",
    "# bedtools intersect -a Prostate_lit_regions_annotated_05252022.tsv -b ../PRAD_450k_DMR.deepFilter_0.33_0.67_minProbes3_fmt.bed -wao > lit_promoters_tcga_overlap.out\n",
    "# bedtools intersect -a Prostate_lit_regions_annotated_05252022.tsv -b ../ECM_DMRs_raw_Hang_05172022_sorted.bed -wao > lit_promoters_ecm_overlap.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "07c8cd04",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'V1', 'V2'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "5708"
      ],
      "text/latex": [
       "5708"
      ],
      "text/markdown": [
       "5708"
      ],
      "text/plain": [
       "[1] 5708"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_tcga_ol = read.csv(file=\"./marker_collation/lit_promoters_tcga_overlap.out\", sep=\"\\t\", header=FALSE)\n",
    "# head(df_tcga_ol)\n",
    "df_tcga_ol = df_tcga_ol %>% select(V1,V2,V3,V17) %>% group_by(V1,V2,V3) %>% summarize(bp.tcga.overlap=sum(V17, na.rm=TRUE)) \n",
    "nrow(df_tcga_ol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "b9e46702",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'V1', 'V2'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "5708"
      ],
      "text/latex": [
       "5708"
      ],
      "text/markdown": [
       "5708"
      ],
      "text/plain": [
       "[1] 5708"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_ecm_ol = read.csv(file=\"./marker_collation/lit_promoters_ecm_overlap.out\", sep=\"\\t\", header=FALSE)\n",
    "df_ecm_ol = df_ecm_ol %>% select(V1,V2,V3,V34) %>% group_by(V1,V2,V3) %>% summarize(bp.ecm.overlap=sum(V34, na.rm=TRUE)) \n",
    "nrow(df_ecm_ol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "78193f27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 751751</td><td> 753751</td><td>FAM87B  </td><td>1</td><td>FAM87B  </td><td>family with sequence similarity 87 member B             </td><td>HGNC:32236</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 878583</td><td> 880583</td><td>NOC2L   </td><td>1</td><td>NOC2L   </td><td>NOC2 like nucleolar associated transcriptional repressor</td><td>HGNC:24517</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td> 935518</td><td> 937518</td><td>ISG15   </td><td>1</td><td>ISG15   </td><td>ISG15 ubiquitin like modifier                           </td><td>HGNC:4053 </td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1108260</td><td>1110260</td><td>TTLL10  </td><td>2</td><td>TTLL10  </td><td>tubulin tyrosine ligase like 10                         </td><td>HGNC:26693</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>1269656</td><td>1271656</td><td>DVL1    </td><td>1</td><td>DVL1    </td><td>dishevelled segment polarity protein 1                  </td><td>HGNC:3084 </td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>1308110</td><td>1310110</td><td>AURKAIP1</td><td>1</td><td>AURKAIP1</td><td>aurora kinase A interacting protein 1                   </td><td>HGNC:24114</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  751751 &  753751 & FAM87B   & 1 & FAM87B   & family with sequence similarity 87 member B              & HGNC:32236 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t2 & 1 &  878583 &  880583 & NOC2L    & 1 & NOC2L    & NOC2 like nucleolar associated transcriptional repressor & HGNC:24517 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t3 & 1 &  935518 &  937518 & ISG15    & 1 & ISG15    & ISG15 ubiquitin like modifier                            & HGNC:4053  & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t4 & 1 & 1108260 & 1110260 & TTLL10   & 2 & TTLL10   & tubulin tyrosine ligase like 10                          & HGNC:26693 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t5 & 1 & 1269656 & 1271656 & DVL1     & 1 & DVL1     & dishevelled segment polarity protein 1                   & HGNC:3084  & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t6 & 1 & 1308110 & 1310110 & AURKAIP1 & 1 & AURKAIP1 & aurora kinase A interacting protein 1                    & HGNC:24114 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  751751 |  753751 | FAM87B   | 1 | FAM87B   | family with sequence similarity 87 member B              | HGNC:32236 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 2 | 1 |  878583 |  880583 | NOC2L    | 1 | NOC2L    | NOC2 like nucleolar associated transcriptional repressor | HGNC:24517 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 3 | 1 |  935518 |  937518 | ISG15    | 1 | ISG15    | ISG15 ubiquitin like modifier                            | HGNC:4053  | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 4 | 1 | 1108260 | 1110260 | TTLL10   | 2 | TTLL10   | tubulin tyrosine ligase like 10                          | HGNC:26693 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 5 | 1 | 1269656 | 1271656 | DVL1     | 1 | DVL1     | dishevelled segment polarity protein 1                   | HGNC:3084  | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 6 | 1 | 1308110 | 1310110 | AURKAIP1 | 1 | AURKAIP1 | aurora kinase A interacting protein 1                    | HGNC:24114 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 1    751751     753751  FAM87B           1             FAM87B        \n",
       "2 1    878583     880583  NOC2L            1             NOC2L         \n",
       "3 1    935518     937518  ISG15            1             ISG15         \n",
       "4 1   1108260    1110260  TTLL10           2             TTLL10        \n",
       "5 1   1269656    1271656  DVL1             1             DVL1          \n",
       "6 1   1308110    1310110  AURKAIP1         1             AURKAIP1      \n",
       "  Approved.name                                            HGNC.ID    Location\n",
       "1 family with sequence similarity 87 member B              HGNC:32236 1p36.33 \n",
       "2 NOC2 like nucleolar associated transcriptional repressor HGNC:24517 1p36.33 \n",
       "3 ISG15 ubiquitin like modifier                            HGNC:4053  1p36.33 \n",
       "4 tubulin tyrosine ligase like 10                          HGNC:26693 1p36.33 \n",
       "5 dishevelled segment polarity protein 1                   HGNC:3084  1p36.33 \n",
       "6 aurora kinase A interacting protein 1                    HGNC:24114 1p36.33 \n",
       "  strand type     size is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 +      promoter 2000 FALSE        0               0             \n",
       "2 -      promoter 2000 FALSE        0               0             \n",
       "3 +      promoter 2000 FALSE        0               0             \n",
       "4 +      promoter 2000 FALSE        0               0             \n",
       "5 -      promoter 2000 FALSE        0               0             \n",
       "6 -      promoter 2000 FALSE        0               0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_out = df_out %>% left_join(df_tcga_ol, by=c(chr=\"V1\",start_hg19=\"V2\",end_hg19=\"V3\"))\n",
    "df_out = df_out %>% left_join(df_ecm_ol, by=c(chr=\"V1\",start_hg19=\"V2\",end_hg19=\"V3\"))\n",
    "head(df_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 423,
   "id": "849f24e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_out, file=\"./marker_collation/Prostate_lit_regions_annotated_05272022.tsv\", sep=\"\\t\",\n",
    "           row.names=FALSE, col.names=TRUE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "2231f9c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1061</th><td>11</td><td>67350213</td><td>67352213</td><td>GSTP1</td><td>15</td><td>GSTP1</td><td>glutathione S-transferase pi 1</td><td>HGNC:4638</td><td>11q13.2</td><td>+</td><td>promoter</td><td>2000</td><td>TRUE</td><td>1065</td><td>695</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1061 & 11 & 67350213 & 67352213 & GSTP1 & 15 & GSTP1 & glutathione S-transferase pi 1 & HGNC:4638 & 11q13.2 & + & promoter & 2000 & TRUE & 1065 & 695\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 15\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1061 | 11 | 67350213 | 67352213 | GSTP1 | 15 | GSTP1 | glutathione S-transferase pi 1 | HGNC:4638 | 11q13.2 | + | promoter | 2000 | TRUE | 1065 | 695 |\n",
       "\n"
      ],
      "text/plain": [
       "     chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1061 11  67350213   67352213 GSTP1            15            GSTP1         \n",
       "     Approved.name                  HGNC.ID   Location strand type     size\n",
       "1061 glutathione S-transferase pi 1 HGNC:4638 11q13.2  +      promoter 2000\n",
       "     is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1061 TRUE         1065            695           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_out[df_out$hugo_gene_name==\"GSTP1\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "5ea962a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 751751</td><td> 753751</td><td>FAM87B  </td><td>1</td><td>FAM87B  </td><td>family with sequence similarity 87 member B             </td><td>HGNC:32236</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 878583</td><td> 880583</td><td>NOC2L   </td><td>1</td><td>NOC2L   </td><td>NOC2 like nucleolar associated transcriptional repressor</td><td>HGNC:24517</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td> 935518</td><td> 937518</td><td>ISG15   </td><td>1</td><td>ISG15   </td><td>ISG15 ubiquitin like modifier                           </td><td>HGNC:4053 </td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1108260</td><td>1110260</td><td>TTLL10  </td><td>2</td><td>TTLL10  </td><td>tubulin tyrosine ligase like 10                         </td><td>HGNC:26693</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>1269656</td><td>1271656</td><td>DVL1    </td><td>1</td><td>DVL1    </td><td>dishevelled segment polarity protein 1                  </td><td>HGNC:3084 </td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>1308110</td><td>1310110</td><td>AURKAIP1</td><td>1</td><td>AURKAIP1</td><td>aurora kinase A interacting protein 1                   </td><td>HGNC:24114</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  751751 &  753751 & FAM87B   & 1 & FAM87B   & family with sequence similarity 87 member B              & HGNC:32236 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t2 & 1 &  878583 &  880583 & NOC2L    & 1 & NOC2L    & NOC2 like nucleolar associated transcriptional repressor & HGNC:24517 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t3 & 1 &  935518 &  937518 & ISG15    & 1 & ISG15    & ISG15 ubiquitin like modifier                            & HGNC:4053  & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t4 & 1 & 1108260 & 1110260 & TTLL10   & 2 & TTLL10   & tubulin tyrosine ligase like 10                          & HGNC:26693 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t5 & 1 & 1269656 & 1271656 & DVL1     & 1 & DVL1     & dishevelled segment polarity protein 1                   & HGNC:3084  & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t6 & 1 & 1308110 & 1310110 & AURKAIP1 & 1 & AURKAIP1 & aurora kinase A interacting protein 1                    & HGNC:24114 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  751751 |  753751 | FAM87B   | 1 | FAM87B   | family with sequence similarity 87 member B              | HGNC:32236 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 2 | 1 |  878583 |  880583 | NOC2L    | 1 | NOC2L    | NOC2 like nucleolar associated transcriptional repressor | HGNC:24517 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 3 | 1 |  935518 |  937518 | ISG15    | 1 | ISG15    | ISG15 ubiquitin like modifier                            | HGNC:4053  | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 4 | 1 | 1108260 | 1110260 | TTLL10   | 2 | TTLL10   | tubulin tyrosine ligase like 10                          | HGNC:26693 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 5 | 1 | 1269656 | 1271656 | DVL1     | 1 | DVL1     | dishevelled segment polarity protein 1                   | HGNC:3084  | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 6 | 1 | 1308110 | 1310110 | AURKAIP1 | 1 | AURKAIP1 | aurora kinase A interacting protein 1                    | HGNC:24114 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 1    751751     753751  FAM87B           1             FAM87B        \n",
       "2 1    878583     880583  NOC2L            1             NOC2L         \n",
       "3 1    935518     937518  ISG15            1             ISG15         \n",
       "4 1   1108260    1110260  TTLL10           2             TTLL10        \n",
       "5 1   1269656    1271656  DVL1             1             DVL1          \n",
       "6 1   1308110    1310110  AURKAIP1         1             AURKAIP1      \n",
       "  Approved.name                                            HGNC.ID    Location\n",
       "1 family with sequence similarity 87 member B              HGNC:32236 1p36.33 \n",
       "2 NOC2 like nucleolar associated transcriptional repressor HGNC:24517 1p36.33 \n",
       "3 ISG15 ubiquitin like modifier                            HGNC:4053  1p36.33 \n",
       "4 tubulin tyrosine ligase like 10                          HGNC:26693 1p36.33 \n",
       "5 dishevelled segment polarity protein 1                   HGNC:3084  1p36.33 \n",
       "6 aurora kinase A interacting protein 1                    HGNC:24114 1p36.33 \n",
       "  strand type     size is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 +      promoter 2000 FALSE        0               0             \n",
       "2 -      promoter 2000 FALSE        0               0             \n",
       "3 +      promoter 2000 FALSE        0               0             \n",
       "4 +      promoter 2000 FALSE        0               0             \n",
       "5 -      promoter 2000 FALSE        0               0             \n",
       "6 -      promoter 2000 FALSE        0               0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df = read.csv(\"./marker_collation/Prostate_lit_regions_annotated_05252022.tsv\", sep=\"\\t\", header=TRUE)\n",
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "ab2892c3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "154"
      ],
      "text/latex": [
       "154"
      ],
      "text/markdown": [
       "154"
      ],
      "text/plain": [
       "[1] 154"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sum(df$bp.tcga.overlap>0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "12418648",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "558"
      ],
      "text/latex": [
       "558"
      ],
      "text/markdown": [
       "558"
      ],
      "text/plain": [
       "[1] 558"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sum(df$bp.ecm.overlap>0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "5da2ee18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "5708"
      ],
      "text/latex": [
       "5708"
      ],
      "text/markdown": [
       "5708"
      ],
      "text/plain": [
       "[1] 5708"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "737457bb",
   "metadata": {},
   "source": [
    "# reattach publication annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "95f13bf6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr_hg19</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>chr_hg38</th><th scope=col>start_hg38</th><th scope=col>end_hg38</th><th scope=col>pub.strand</th><th scope=col>source</th><th scope=col>pub.gene.name</th><th scope=col>pub.reg.type</th><th scope=col>probe</th><th scope=col>notes</th><th scope=col>no.pubs</th><th scope=col>in.gencode.gene</th><th scope=col>gencode.gene.name</th><th scope=col>in.gencode.promoter</th><th scope=col>is.ingergenic</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>region</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>4 </td><td>106067032</td><td>106200973</td><td>chr4 </td><td>105145875</td><td>105279816</td><td>+</td><td>Kukkonen et al. 2021</td><td>TET2  </td><td>gene</td><td>NA</td><td>hypermethylated subtype            </td><td>1</td><td>TRUE</td><td>TET2-AS1</td><td>TRUE</td><td>FALSE</td><td>TET2  </td><td>4</td><td>4 : 105145875 - 105279816</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20</td><td> 31350191</td><td> 31397165</td><td>chr20</td><td> 32762385</td><td> 32809359</td><td>+</td><td>Kukkonen et al. 2021</td><td>DNMT3B</td><td>gene</td><td>NA</td><td>hypermethylated subtype            </td><td>1</td><td>TRUE</td><td>DNMT3B  </td><td>TRUE</td><td>FALSE</td><td>DNMT3B</td><td>2</td><td>20 : 32762385 - 32809359 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>2 </td><td>209100953</td><td>209130798</td><td>chr2 </td><td>208236229</td><td>208266074</td><td>-</td><td>Kukkonen et al. 2021</td><td>IDH1  </td><td>gene</td><td>NA</td><td>hypermethylated subtype            </td><td>1</td><td>TRUE</td><td>IDH1-AS1</td><td>TRUE</td><td>FALSE</td><td>IDH1  </td><td>1</td><td>2 : 208236229 - 208266074</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7 </td><td>140419127</td><td>140624729</td><td>chr7 </td><td>140719327</td><td>140924929</td><td>-</td><td>Kukkonen et al. 2021</td><td>BRAF  </td><td>gene</td><td>NA</td><td>hypermethylated subtype            </td><td>1</td><td>TRUE</td><td>RNU6-85P</td><td>TRUE</td><td>FALSE</td><td>BRAF  </td><td>1</td><td>7 : 140719327 - 140924929</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X </td><td> 66763863</td><td> 66950461</td><td>chrX </td><td> 67544021</td><td> 67730619</td><td>+</td><td>Kukkonen et al. 2021</td><td>AR    </td><td>gene</td><td>NA</td><td>hypermethylated intergenic regions </td><td>1</td><td>TRUE</td><td>AR      </td><td>TRUE</td><td>FALSE</td><td>AR    </td><td>2</td><td>X : 67544021 - 67730619  </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>8 </td><td>128747680</td><td>128755197</td><td>chr8 </td><td>127735434</td><td>127742951</td><td>+</td><td>Kukkonen et al. 2021</td><td>MYC   </td><td>gene</td><td>NA</td><td>hypermethylated intergenic regions </td><td>1</td><td>TRUE</td><td>MYC     </td><td>TRUE</td><td>FALSE</td><td>MYC   </td><td>3</td><td>8 : 127735434 - 127742951</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 20\n",
       "\\begin{tabular}{r|llllllllllllllllllll}\n",
       "  & chr\\_hg19 & start\\_hg19 & end\\_hg19 & chr\\_hg38 & start\\_hg38 & end\\_hg38 & pub.strand & source & pub.gene.name & pub.reg.type & probe & notes & no.pubs & in.gencode.gene & gencode.gene.name & in.gencode.promoter & is.ingergenic & hybrid.gene.name & no.total.pubs & region\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <chr> & <lgl> & <lgl> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 4  & 106067032 & 106200973 & chr4  & 105145875 & 105279816 & + & Kukkonen et al. 2021 & TET2   & gene & NA & hypermethylated subtype             & 1 & TRUE & TET2-AS1 & TRUE & FALSE & TET2   & 4 & 4 : 105145875 - 105279816\\\\\n",
       "\t2 & 20 &  31350191 &  31397165 & chr20 &  32762385 &  32809359 & + & Kukkonen et al. 2021 & DNMT3B & gene & NA & hypermethylated subtype             & 1 & TRUE & DNMT3B   & TRUE & FALSE & DNMT3B & 2 & 20 : 32762385 - 32809359 \\\\\n",
       "\t3 & 2  & 209100953 & 209130798 & chr2  & 208236229 & 208266074 & - & Kukkonen et al. 2021 & IDH1   & gene & NA & hypermethylated subtype             & 1 & TRUE & IDH1-AS1 & TRUE & FALSE & IDH1   & 1 & 2 : 208236229 - 208266074\\\\\n",
       "\t4 & 7  & 140419127 & 140624729 & chr7  & 140719327 & 140924929 & - & Kukkonen et al. 2021 & BRAF   & gene & NA & hypermethylated subtype             & 1 & TRUE & RNU6-85P & TRUE & FALSE & BRAF   & 1 & 7 : 140719327 - 140924929\\\\\n",
       "\t5 & X  &  66763863 &  66950461 & chrX  &  67544021 &  67730619 & + & Kukkonen et al. 2021 & AR     & gene & NA & hypermethylated intergenic regions  & 1 & TRUE & AR       & TRUE & FALSE & AR     & 2 & X : 67544021 - 67730619  \\\\\n",
       "\t6 & 8  & 128747680 & 128755197 & chr8  & 127735434 & 127742951 & + & Kukkonen et al. 2021 & MYC    & gene & NA & hypermethylated intergenic regions  & 1 & TRUE & MYC      & TRUE & FALSE & MYC    & 3 & 8 : 127735434 - 127742951\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 20\n",
       "\n",
       "| <!--/--> | chr_hg19 &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | chr_hg38 &lt;chr&gt; | start_hg38 &lt;int&gt; | end_hg38 &lt;int&gt; | pub.strand &lt;chr&gt; | source &lt;chr&gt; | pub.gene.name &lt;chr&gt; | pub.reg.type &lt;chr&gt; | probe &lt;chr&gt; | notes &lt;chr&gt; | no.pubs &lt;int&gt; | in.gencode.gene &lt;lgl&gt; | gencode.gene.name &lt;chr&gt; | in.gencode.promoter &lt;lgl&gt; | is.ingergenic &lt;lgl&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | region &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 4  | 106067032 | 106200973 | chr4  | 105145875 | 105279816 | + | Kukkonen et al. 2021 | TET2   | gene | NA | hypermethylated subtype             | 1 | TRUE | TET2-AS1 | TRUE | FALSE | TET2   | 4 | 4 : 105145875 - 105279816 |\n",
       "| 2 | 20 |  31350191 |  31397165 | chr20 |  32762385 |  32809359 | + | Kukkonen et al. 2021 | DNMT3B | gene | NA | hypermethylated subtype             | 1 | TRUE | DNMT3B   | TRUE | FALSE | DNMT3B | 2 | 20 : 32762385 - 32809359  |\n",
       "| 3 | 2  | 209100953 | 209130798 | chr2  | 208236229 | 208266074 | - | Kukkonen et al. 2021 | IDH1   | gene | NA | hypermethylated subtype             | 1 | TRUE | IDH1-AS1 | TRUE | FALSE | IDH1   | 1 | 2 : 208236229 - 208266074 |\n",
       "| 4 | 7  | 140419127 | 140624729 | chr7  | 140719327 | 140924929 | - | Kukkonen et al. 2021 | BRAF   | gene | NA | hypermethylated subtype             | 1 | TRUE | RNU6-85P | TRUE | FALSE | BRAF   | 1 | 7 : 140719327 - 140924929 |\n",
       "| 5 | X  |  66763863 |  66950461 | chrX  |  67544021 |  67730619 | + | Kukkonen et al. 2021 | AR     | gene | NA | hypermethylated intergenic regions  | 1 | TRUE | AR       | TRUE | FALSE | AR     | 2 | X : 67544021 - 67730619   |\n",
       "| 6 | 8  | 128747680 | 128755197 | chr8  | 127735434 | 127742951 | + | Kukkonen et al. 2021 | MYC    | gene | NA | hypermethylated intergenic regions  | 1 | TRUE | MYC      | TRUE | FALSE | MYC    | 3 | 8 : 127735434 - 127742951 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr_hg19 start_hg19 end_hg19  chr_hg38 start_hg38 end_hg38  pub.strand\n",
       "1 4        106067032  106200973 chr4     105145875  105279816 +         \n",
       "2 20        31350191   31397165 chr20     32762385   32809359 +         \n",
       "3 2        209100953  209130798 chr2     208236229  208266074 -         \n",
       "4 7        140419127  140624729 chr7     140719327  140924929 -         \n",
       "5 X         66763863   66950461 chrX      67544021   67730619 +         \n",
       "6 8        128747680  128755197 chr8     127735434  127742951 +         \n",
       "  source               pub.gene.name pub.reg.type probe\n",
       "1 Kukkonen et al. 2021 TET2          gene         NA   \n",
       "2 Kukkonen et al. 2021 DNMT3B        gene         NA   \n",
       "3 Kukkonen et al. 2021 IDH1          gene         NA   \n",
       "4 Kukkonen et al. 2021 BRAF          gene         NA   \n",
       "5 Kukkonen et al. 2021 AR            gene         NA   \n",
       "6 Kukkonen et al. 2021 MYC           gene         NA   \n",
       "  notes                               no.pubs in.gencode.gene gencode.gene.name\n",
       "1 hypermethylated subtype             1       TRUE            TET2-AS1         \n",
       "2 hypermethylated subtype             1       TRUE            DNMT3B           \n",
       "3 hypermethylated subtype             1       TRUE            IDH1-AS1         \n",
       "4 hypermethylated subtype             1       TRUE            RNU6-85P         \n",
       "5 hypermethylated intergenic regions  1       TRUE            AR               \n",
       "6 hypermethylated intergenic regions  1       TRUE            MYC              \n",
       "  in.gencode.promoter is.ingergenic hybrid.gene.name no.total.pubs\n",
       "1 TRUE                FALSE         TET2             4            \n",
       "2 TRUE                FALSE         DNMT3B           2            \n",
       "3 TRUE                FALSE         IDH1             1            \n",
       "4 TRUE                FALSE         BRAF             1            \n",
       "5 TRUE                FALSE         AR               2            \n",
       "6 TRUE                FALSE         MYC              3            \n",
       "  region                   \n",
       "1 4 : 105145875 - 105279816\n",
       "2 20 : 32762385 - 32809359 \n",
       "3 2 : 208236229 - 208266074\n",
       "4 7 : 140719327 - 140924929\n",
       "5 X : 67544021 - 67730619  \n",
       "6 8 : 127735434 - 127742951"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_regions = read.csv(\"./marker_collation/all_lit_regions_wcoords_annot_as_hg19.tsv\", \n",
    "                     header=TRUE, sep=\"\\t\")\n",
    "head(df_regions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "10ae1d27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "120417"
      ],
      "text/latex": [
       "120417"
      ],
      "text/markdown": [
       "120417"
      ],
      "text/plain": [
       "[1] 120417"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5944"
      ],
      "text/latex": [
       "5944"
      ],
      "text/markdown": [
       "5944"
      ],
      "text/plain": [
       "[1] 5944"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hybrid.gene.name</th><th scope=col>sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GSTP1</td><td>Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " hybrid.gene.name & sources\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t GSTP1 & Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 2\n",
       "\n",
       "| hybrid.gene.name &lt;chr&gt; | sources &lt;chr&gt; |\n",
       "|---|---|\n",
       "| GSTP1 | Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021 |\n",
       "\n"
      ],
      "text/plain": [
       "  hybrid.gene.name\n",
       "1 GSTP1           \n",
       "  sources                                                                        \n",
       "1 Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(df_regions)\n",
    "pubs_key = df_regions %>% group_by(hybrid.gene.name) %>% summarize(sources=paste(unique(source), collapse=\";\"))\n",
    "nrow(pubs_key)\n",
    "pubs_key %>% filter(hybrid.gene.name==\"GSTP1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fa34becd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 751751</td><td> 753751</td><td>FAM87B  </td><td>1</td><td>FAM87B  </td><td>family with sequence similarity 87 member B             </td><td>HGNC:32236</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 878583</td><td> 880583</td><td>NOC2L   </td><td>1</td><td>NOC2L   </td><td>NOC2 like nucleolar associated transcriptional repressor</td><td>HGNC:24517</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td> 935518</td><td> 937518</td><td>ISG15   </td><td>1</td><td>ISG15   </td><td>ISG15 ubiquitin like modifier                           </td><td>HGNC:4053 </td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1108260</td><td>1110260</td><td>TTLL10  </td><td>2</td><td>TTLL10  </td><td>tubulin tyrosine ligase like 10                         </td><td>HGNC:26693</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>1269656</td><td>1271656</td><td>DVL1    </td><td>1</td><td>DVL1    </td><td>dishevelled segment polarity protein 1                  </td><td>HGNC:3084 </td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>1308110</td><td>1310110</td><td>AURKAIP1</td><td>1</td><td>AURKAIP1</td><td>aurora kinase A interacting protein 1                   </td><td>HGNC:24114</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  751751 &  753751 & FAM87B   & 1 & FAM87B   & family with sequence similarity 87 member B              & HGNC:32236 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t2 & 1 &  878583 &  880583 & NOC2L    & 1 & NOC2L    & NOC2 like nucleolar associated transcriptional repressor & HGNC:24517 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t3 & 1 &  935518 &  937518 & ISG15    & 1 & ISG15    & ISG15 ubiquitin like modifier                            & HGNC:4053  & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t4 & 1 & 1108260 & 1110260 & TTLL10   & 2 & TTLL10   & tubulin tyrosine ligase like 10                          & HGNC:26693 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t5 & 1 & 1269656 & 1271656 & DVL1     & 1 & DVL1     & dishevelled segment polarity protein 1                   & HGNC:3084  & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\t6 & 1 & 1308110 & 1310110 & AURKAIP1 & 1 & AURKAIP1 & aurora kinase A interacting protein 1                    & HGNC:24114 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  751751 |  753751 | FAM87B   | 1 | FAM87B   | family with sequence similarity 87 member B              | HGNC:32236 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 2 | 1 |  878583 |  880583 | NOC2L    | 1 | NOC2L    | NOC2 like nucleolar associated transcriptional repressor | HGNC:24517 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 3 | 1 |  935518 |  937518 | ISG15    | 1 | ISG15    | ISG15 ubiquitin like modifier                            | HGNC:4053  | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 4 | 1 | 1108260 | 1110260 | TTLL10   | 2 | TTLL10   | tubulin tyrosine ligase like 10                          | HGNC:26693 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 5 | 1 | 1269656 | 1271656 | DVL1     | 1 | DVL1     | dishevelled segment polarity protein 1                   | HGNC:3084  | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "| 6 | 1 | 1308110 | 1310110 | AURKAIP1 | 1 | AURKAIP1 | aurora kinase A interacting protein 1                    | HGNC:24114 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 1    751751     753751  FAM87B           1             FAM87B        \n",
       "2 1    878583     880583  NOC2L            1             NOC2L         \n",
       "3 1    935518     937518  ISG15            1             ISG15         \n",
       "4 1   1108260    1110260  TTLL10           2             TTLL10        \n",
       "5 1   1269656    1271656  DVL1             1             DVL1          \n",
       "6 1   1308110    1310110  AURKAIP1         1             AURKAIP1      \n",
       "  Approved.name                                            HGNC.ID    Location\n",
       "1 family with sequence similarity 87 member B              HGNC:32236 1p36.33 \n",
       "2 NOC2 like nucleolar associated transcriptional repressor HGNC:24517 1p36.33 \n",
       "3 ISG15 ubiquitin like modifier                            HGNC:4053  1p36.33 \n",
       "4 tubulin tyrosine ligase like 10                          HGNC:26693 1p36.33 \n",
       "5 dishevelled segment polarity protein 1                   HGNC:3084  1p36.33 \n",
       "6 aurora kinase A interacting protein 1                    HGNC:24114 1p36.33 \n",
       "  strand type     size is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 +      promoter 2000 FALSE        0               0             \n",
       "2 -      promoter 2000 FALSE        0               0             \n",
       "3 +      promoter 2000 FALSE        0               0             \n",
       "4 +      promoter 2000 FALSE        0               0             \n",
       "5 -      promoter 2000 FALSE        0               0             \n",
       "6 -      promoter 2000 FALSE        0               0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_regions_outfile = read.csv(\"./marker_collation/Prostate_lit_regions_annotated_05272022.tsv\",\n",
    "                             header=TRUE, sep=\"\\t\")\n",
    "\n",
    "head(df_regions_outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "80301f03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "5708"
      ],
      "text/latex": [
       "5708"
      ],
      "text/markdown": [
       "5708"
      ],
      "text/plain": [
       "[1] 5708"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_updated_outfile = df_regions_outfile %>% \n",
    "    left_join(pubs_key %>% select(hybrid.gene.name, sources), by=\"hybrid.gene.name\")\n",
    "nrow(df_updated_outfile)\n",
    "sum(is.na(df_updated_outfile$sources))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "24665a84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 16</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>67350213</td><td>67352213</td><td>GSTP1</td><td>15</td><td>GSTP1</td><td>glutathione S-transferase pi 1</td><td>HGNC:4638</td><td>11q13.2</td><td>+</td><td>promoter</td><td>2000</td><td>TRUE</td><td>1065</td><td>695</td><td>Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 16\n",
       "\\begin{tabular}{llllllllllllllll}\n",
       " chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap & sources\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 11 & 67350213 & 67352213 & GSTP1 & 15 & GSTP1 & glutathione S-transferase pi 1 & HGNC:4638 & 11q13.2 & + & promoter & 2000 & TRUE & 1065 & 695 & Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 16\n",
       "\n",
       "| chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | sources &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 67350213 | 67352213 | GSTP1 | 15 | GSTP1 | glutathione S-transferase pi 1 | HGNC:4638 | 11q13.2 | + | promoter | 2000 | TRUE | 1065 | 695 | Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 11  67350213   67352213 GSTP1            15            GSTP1         \n",
       "  Approved.name                  HGNC.ID   Location strand type     size\n",
       "1 glutathione S-transferase pi 1 HGNC:4638 11q13.2  +      promoter 2000\n",
       "  is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 TRUE         1065            695           \n",
       "  sources                                                                        \n",
       "1 Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_updated_outfile %>% filter(hybrid.gene.name==\"GSTP1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3dc60448",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_updated_outfile, \"./marker_collation/Prostate_lit_regions_annotated_08022022.tsv\",\n",
    "         row.names=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "08ae9d94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 16</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>sources</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 751751</td><td> 753751</td><td>FAM87B  </td><td>1</td><td>FAM87B  </td><td>family with sequence similarity 87 member B             </td><td>HGNC:32236</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Ylitalo et al 2021</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 878583</td><td> 880583</td><td>NOC2L   </td><td>1</td><td>NOC2L   </td><td>NOC2 like nucleolar associated transcriptional repressor</td><td>HGNC:24517</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td> 935518</td><td> 937518</td><td>ISG15   </td><td>1</td><td>ISG15   </td><td>ISG15 ubiquitin like modifier                           </td><td>HGNC:4053 </td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1108260</td><td>1110260</td><td>TTLL10  </td><td>2</td><td>TTLL10  </td><td>tubulin tyrosine ligase like 10                         </td><td>HGNC:26693</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>1269656</td><td>1271656</td><td>DVL1    </td><td>1</td><td>DVL1    </td><td>dishevelled segment polarity protein 1                  </td><td>HGNC:3084 </td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>1308110</td><td>1310110</td><td>AURKAIP1</td><td>1</td><td>AURKAIP1</td><td>aurora kinase A interacting protein 1                   </td><td>HGNC:24114</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 16\n",
       "\\begin{tabular}{r|llllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap & sources\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  751751 &  753751 & FAM87B   & 1 & FAM87B   & family with sequence similarity 87 member B              & HGNC:32236 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Ylitalo et al 2021\\\\\n",
       "\t2 & 1 &  878583 &  880583 & NOC2L    & 1 & NOC2L    & NOC2 like nucleolar associated transcriptional repressor & HGNC:24517 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017\\\\\n",
       "\t3 & 1 &  935518 &  937518 & ISG15    & 1 & ISG15    & ISG15 ubiquitin like modifier                            & HGNC:4053  & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017\\\\\n",
       "\t4 & 1 & 1108260 & 1110260 & TTLL10   & 2 & TTLL10   & tubulin tyrosine ligase like 10                          & HGNC:26693 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017\\\\\n",
       "\t5 & 1 & 1269656 & 1271656 & DVL1     & 1 & DVL1     & dishevelled segment polarity protein 1                   & HGNC:3084  & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017\\\\\n",
       "\t6 & 1 & 1308110 & 1310110 & AURKAIP1 & 1 & AURKAIP1 & aurora kinase A interacting protein 1                    & HGNC:24114 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 16\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | sources &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  751751 |  753751 | FAM87B   | 1 | FAM87B   | family with sequence similarity 87 member B              | HGNC:32236 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Ylitalo et al 2021 |\n",
       "| 2 | 1 |  878583 |  880583 | NOC2L    | 1 | NOC2L    | NOC2 like nucleolar associated transcriptional repressor | HGNC:24517 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 |\n",
       "| 3 | 1 |  935518 |  937518 | ISG15    | 1 | ISG15    | ISG15 ubiquitin like modifier                            | HGNC:4053  | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 |\n",
       "| 4 | 1 | 1108260 | 1110260 | TTLL10   | 2 | TTLL10   | tubulin tyrosine ligase like 10                          | HGNC:26693 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 |\n",
       "| 5 | 1 | 1269656 | 1271656 | DVL1     | 1 | DVL1     | dishevelled segment polarity protein 1                   | HGNC:3084  | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 |\n",
       "| 6 | 1 | 1308110 | 1310110 | AURKAIP1 | 1 | AURKAIP1 | aurora kinase A interacting protein 1                    | HGNC:24114 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 1    751751     753751  FAM87B           1             FAM87B        \n",
       "2 1    878583     880583  NOC2L            1             NOC2L         \n",
       "3 1    935518     937518  ISG15            1             ISG15         \n",
       "4 1   1108260    1110260  TTLL10           2             TTLL10        \n",
       "5 1   1269656    1271656  DVL1             1             DVL1          \n",
       "6 1   1308110    1310110  AURKAIP1         1             AURKAIP1      \n",
       "  Approved.name                                            HGNC.ID    Location\n",
       "1 family with sequence similarity 87 member B              HGNC:32236 1p36.33 \n",
       "2 NOC2 like nucleolar associated transcriptional repressor HGNC:24517 1p36.33 \n",
       "3 ISG15 ubiquitin like modifier                            HGNC:4053  1p36.33 \n",
       "4 tubulin tyrosine ligase like 10                          HGNC:26693 1p36.33 \n",
       "5 dishevelled segment polarity protein 1                   HGNC:3084  1p36.33 \n",
       "6 aurora kinase A interacting protein 1                    HGNC:24114 1p36.33 \n",
       "  strand type     size is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 +      promoter 2000 FALSE        0               0             \n",
       "2 -      promoter 2000 FALSE        0               0             \n",
       "3 +      promoter 2000 FALSE        0               0             \n",
       "4 +      promoter 2000 FALSE        0               0             \n",
       "5 -      promoter 2000 FALSE        0               0             \n",
       "6 -      promoter 2000 FALSE        0               0             \n",
       "  sources           \n",
       "1 Ylitalo et al 2021\n",
       "2 Massie et al. 2017\n",
       "3 Massie et al. 2017\n",
       "4 Massie et al. 2017\n",
       "5 Massie et al. 2017\n",
       "6 Massie et al. 2017"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>First.Author.Name</th><th scope=col>YEAR</th><th scope=col>DOI</th><th scope=col>DOI.short</th><th scope=col>PMID</th><th scope=col>PMCID</th><th scope=col>X7</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Aryee     </td><td>2013</td><td>https://doi.org/10.1038/s41467-021-21867-0   </td><td>10.1038/s41467-021-21867-0   </td><td>33741908</td><td>PMC7979745</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Constancio</td><td>2019</td><td>https://doi.org/10.1080/14737159.2019.1604224</td><td>10.1080/14737159.2019.1604224</td><td>30961397</td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Massie    </td><td>2017</td><td>https://doi.org/10.1016/j.jsbmb.2016.04.009  </td><td>10.1016/j.jsbmb.2016.04.009  </td><td>27117390</td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Beltran   </td><td>2020</td><td>https://doi.org/10.1172/JCI131041            </td><td>10.1172/JCI131041            </td><td>32091413</td><td>PMC7108892</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Borno     </td><td>2012</td><td>https://doi.org/10.1158/2159-8290.CD-12-0041 </td><td>10.1158/2159-8290.CD-12-0041 </td><td>22930729</td><td>NA        </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Cai       </td><td>2021</td><td>https://doi.org/10.2147/PGPM.S312024         </td><td>10.2147/PGPM.S312024         </td><td>34285549</td><td>PMC8285280</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & First.Author.Name & YEAR & DOI & DOI.short & PMID & PMCID & X7\\\\\n",
       "  & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Aryee      & 2013 & https://doi.org/10.1038/s41467-021-21867-0    & 10.1038/s41467-021-21867-0    & 33741908 & PMC7979745 & NA\\\\\n",
       "\t2 & Constancio & 2019 & https://doi.org/10.1080/14737159.2019.1604224 & 10.1080/14737159.2019.1604224 & 30961397 & NA         & NA\\\\\n",
       "\t3 & Massie     & 2017 & https://doi.org/10.1016/j.jsbmb.2016.04.009   & 10.1016/j.jsbmb.2016.04.009   & 27117390 & NA         & NA\\\\\n",
       "\t4 & Beltran    & 2020 & https://doi.org/10.1172/JCI131041             & 10.1172/JCI131041             & 32091413 & PMC7108892 & NA\\\\\n",
       "\t5 & Borno      & 2012 & https://doi.org/10.1158/2159-8290.CD-12-0041  & 10.1158/2159-8290.CD-12-0041  & 22930729 & NA         & NA\\\\\n",
       "\t6 & Cai        & 2021 & https://doi.org/10.2147/PGPM.S312024          & 10.2147/PGPM.S312024          & 34285549 & PMC8285280 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | First.Author.Name &lt;chr&gt; | YEAR &lt;dbl&gt; | DOI &lt;chr&gt; | DOI.short &lt;chr&gt; | PMID &lt;chr&gt; | PMCID &lt;chr&gt; | X7 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | Aryee      | 2013 | https://doi.org/10.1038/s41467-021-21867-0    | 10.1038/s41467-021-21867-0    | 33741908 | PMC7979745 | NA |\n",
       "| 2 | Constancio | 2019 | https://doi.org/10.1080/14737159.2019.1604224 | 10.1080/14737159.2019.1604224 | 30961397 | NA         | NA |\n",
       "| 3 | Massie     | 2017 | https://doi.org/10.1016/j.jsbmb.2016.04.009   | 10.1016/j.jsbmb.2016.04.009   | 27117390 | NA         | NA |\n",
       "| 4 | Beltran    | 2020 | https://doi.org/10.1172/JCI131041             | 10.1172/JCI131041             | 32091413 | PMC7108892 | NA |\n",
       "| 5 | Borno      | 2012 | https://doi.org/10.1158/2159-8290.CD-12-0041  | 10.1158/2159-8290.CD-12-0041  | 22930729 | NA         | NA |\n",
       "| 6 | Cai        | 2021 | https://doi.org/10.2147/PGPM.S312024          | 10.2147/PGPM.S312024          | 34285549 | PMC8285280 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  First.Author.Name YEAR DOI                                          \n",
       "1 Aryee             2013 https://doi.org/10.1038/s41467-021-21867-0   \n",
       "2 Constancio        2019 https://doi.org/10.1080/14737159.2019.1604224\n",
       "3 Massie            2017 https://doi.org/10.1016/j.jsbmb.2016.04.009  \n",
       "4 Beltran           2020 https://doi.org/10.1172/JCI131041            \n",
       "5 Borno             2012 https://doi.org/10.1158/2159-8290.CD-12-0041 \n",
       "6 Cai               2021 https://doi.org/10.2147/PGPM.S312024         \n",
       "  DOI.short                     PMID     PMCID      X7\n",
       "1 10.1038/s41467-021-21867-0    33741908 PMC7979745 NA\n",
       "2 10.1080/14737159.2019.1604224 30961397 NA         NA\n",
       "3 10.1016/j.jsbmb.2016.04.009   27117390 NA         NA\n",
       "4 10.1172/JCI131041             32091413 PMC7108892 NA\n",
       "5 10.1158/2159-8290.CD-12-0041  22930729 NA         NA\n",
       "6 10.2147/PGPM.S312024          34285549 PMC8285280 NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# attach DOIs/PMIDs\n",
    "df_outfile = read.csv(\"./marker_collation/Prostate_lit_regions_annotated_08022022.tsv\",sep=\"\\t\")\n",
    "head(df_outfile)\n",
    "df_DOIs = read.xlsx(\"./prostate_key_to PMIDS.xlsx\",\n",
    "                   sheet=\"Sheet1\")\n",
    "head(df_DOIs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "76138ad5",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_outfile$first.author = str_split(df_outfile$sources, \" \")\n",
    "\n",
    "i = 0\n",
    "doi_vec = list()\n",
    "pmids_vec = list()\n",
    "for (sources in df_outfile$sources){\n",
    "    #print(sources)\n",
    "    i = i + 1\n",
    "    dois = list()\n",
    "    pmids = list()\n",
    "    sources = gsub(\";\",\" \",sources)\n",
    "    authors = strsplit(sources,\" \")\n",
    "    for (auth in authors[[1]]){\n",
    "        #print(auth)\n",
    "        if (auth %in% df_DOIs$First.Author.Name){\n",
    "            dois = append(dois, df_DOIs[df_DOIs$First.Author.Name==auth,\"DOI\"])\n",
    "            pmids = append(pmids, df_DOIs[df_DOIs$First.Author.Name==auth,\"PMID\"])\n",
    "            }\n",
    "        }\n",
    "    doi_vec[[i]] <- paste(shQuote(dois), collapse=\", \")\n",
    "    #print(length(doi_vec))\n",
    "    pmids_vec[[i]] <- paste(shQuote(pmids[!is.na(pmids)]), collapse=\", \")\n",
    "    }\n",
    "stopifnot(length(doi_vec)==nrow(df_outfile))  #theres no check for authors missing in the sheet TODO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6aa60115",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>sources</th><th scope=col>first.author</th><th scope=col>DOIs</th><th scope=col>PMIDs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1061</th><td>11</td><td>67350213</td><td>67352213</td><td>GSTP1</td><td>15</td><td>GSTP1</td><td>glutathione S-transferase pi 1</td><td>HGNC:4638</td><td>11q13.2</td><td>+</td><td>promoter</td><td>2000</td><td>TRUE</td><td>1065</td><td>695</td><td>Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021</td><td>Constancio  , et          , al.         , 2019;Massie , et          , al.         , 2017;Ylitalo, et          , al          , 2021;Sugiura, et          , al          , 2021        </td><td>'https://doi.org/10.1080/14737159.2019.1604224', 'https://doi.org/10.1016/j.jsbmb.2016.04.009', 'https://doi.org/10.1186/s13148-021-01119-0', 'https://doi.org/10.1111/iju.14406'</td><td>'30961397', '27117390', '34193246', '33111429'</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 19\n",
       "\\begin{tabular}{r|lllllllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap & sources & first.author & DOIs & PMIDs\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int> & <chr> & <list> & <list> & <list>\\\\\n",
       "\\hline\n",
       "\t1061 & 11 & 67350213 & 67352213 & GSTP1 & 15 & GSTP1 & glutathione S-transferase pi 1 & HGNC:4638 & 11q13.2 & + & promoter & 2000 & TRUE & 1065 & 695 & Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021 & Constancio  , et          , al.         , 2019;Massie , et          , al.         , 2017;Ylitalo, et          , al          , 2021;Sugiura, et          , al          , 2021         & 'https://doi.org/10.1080/14737159.2019.1604224', 'https://doi.org/10.1016/j.jsbmb.2016.04.009', 'https://doi.org/10.1186/s13148-021-01119-0', 'https://doi.org/10.1111/iju.14406' & '30961397', '27117390', '34193246', '33111429'\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 19\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | sources &lt;chr&gt; | first.author &lt;list&gt; | DOIs &lt;list&gt; | PMIDs &lt;list&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1061 | 11 | 67350213 | 67352213 | GSTP1 | 15 | GSTP1 | glutathione S-transferase pi 1 | HGNC:4638 | 11q13.2 | + | promoter | 2000 | TRUE | 1065 | 695 | Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021 | Constancio  , et          , al.         , 2019;Massie , et          , al.         , 2017;Ylitalo, et          , al          , 2021;Sugiura, et          , al          , 2021         | 'https://doi.org/10.1080/14737159.2019.1604224', 'https://doi.org/10.1016/j.jsbmb.2016.04.009', 'https://doi.org/10.1186/s13148-021-01119-0', 'https://doi.org/10.1111/iju.14406' | '30961397', '27117390', '34193246', '33111429' |\n",
       "\n"
      ],
      "text/plain": [
       "     chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1061 11  67350213   67352213 GSTP1            15            GSTP1         \n",
       "     Approved.name                  HGNC.ID   Location strand type     size\n",
       "1061 glutathione S-transferase pi 1 HGNC:4638 11q13.2  +      promoter 2000\n",
       "     is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1061 TRUE         1065            695           \n",
       "     sources                                                                        \n",
       "1061 Constancio et al. 2019;Massie et al. 2017;Ylitalo et al 2021;Sugiura et al 2021\n",
       "     first.author                                                                                                                                                                        \n",
       "1061 Constancio  , et          , al.         , 2019;Massie , et          , al.         , 2017;Ylitalo, et          , al          , 2021;Sugiura, et          , al          , 2021        \n",
       "     DOIs                                                                                                                                                                             \n",
       "1061 'https://doi.org/10.1080/14737159.2019.1604224', 'https://doi.org/10.1016/j.jsbmb.2016.04.009', 'https://doi.org/10.1186/s13148-021-01119-0', 'https://doi.org/10.1111/iju.14406'\n",
       "     PMIDs                                         \n",
       "1061 '30961397', '27117390', '34193246', '33111429'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>hybrid.gene.name</th><th scope=col>no.total.pubs</th><th scope=col>hugo_gene_name</th><th scope=col>Approved.name</th><th scope=col>HGNC.ID</th><th scope=col>Location</th><th scope=col>strand</th><th scope=col>type</th><th scope=col>size</th><th scope=col>is.whitelist</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>sources</th><th scope=col>first.author</th><th scope=col>DOIs</th><th scope=col>PMIDs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td> 751751</td><td> 753751</td><td>FAM87B  </td><td>1</td><td>FAM87B  </td><td>family with sequence similarity 87 member B             </td><td>HGNC:32236</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Ylitalo et al 2021</td><td>Ylitalo, et     , al     , 2021   </td><td>'https://doi.org/10.1186/s13148-021-01119-0'</td><td>'34193246'</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td> 878583</td><td> 880583</td><td>NOC2L   </td><td>1</td><td>NOC2L   </td><td>NOC2 like nucleolar associated transcriptional repressor</td><td>HGNC:24517</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td><td>Massie, et    , al.   , 2017  </td><td>'https://doi.org/10.1016/j.jsbmb.2016.04.009'</td><td>'27117390'</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td> 935518</td><td> 937518</td><td>ISG15   </td><td>1</td><td>ISG15   </td><td>ISG15 ubiquitin like modifier                           </td><td>HGNC:4053 </td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td><td>Massie, et    , al.   , 2017  </td><td>'https://doi.org/10.1016/j.jsbmb.2016.04.009'</td><td>'27117390'</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1108260</td><td>1110260</td><td>TTLL10  </td><td>2</td><td>TTLL10  </td><td>tubulin tyrosine ligase like 10                         </td><td>HGNC:26693</td><td>1p36.33</td><td>+</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td><td>Massie, et    , al.   , 2017  </td><td>'https://doi.org/10.1016/j.jsbmb.2016.04.009'</td><td>'27117390'</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>1269656</td><td>1271656</td><td>DVL1    </td><td>1</td><td>DVL1    </td><td>dishevelled segment polarity protein 1                  </td><td>HGNC:3084 </td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td><td>Massie, et    , al.   , 2017  </td><td>'https://doi.org/10.1016/j.jsbmb.2016.04.009'</td><td>'27117390'</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>1308110</td><td>1310110</td><td>AURKAIP1</td><td>1</td><td>AURKAIP1</td><td>aurora kinase A interacting protein 1                   </td><td>HGNC:24114</td><td>1p36.33</td><td>-</td><td>promoter</td><td>2000</td><td>FALSE</td><td>0</td><td>0</td><td>Massie et al. 2017</td><td>Massie, et    , al.   , 2017  </td><td>'https://doi.org/10.1016/j.jsbmb.2016.04.009'</td><td>'27117390'</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 19\n",
       "\\begin{tabular}{r|lllllllllllllllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & hybrid.gene.name & no.total.pubs & hugo\\_gene\\_name & Approved.name & HGNC.ID & Location & strand & type & size & is.whitelist & bp.tcga.overlap & bp.ecm.overlap & sources & first.author & DOIs & PMIDs\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <int> & <chr> & <list> & <list> & <list>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &  751751 &  753751 & FAM87B   & 1 & FAM87B   & family with sequence similarity 87 member B              & HGNC:32236 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Ylitalo et al 2021 & Ylitalo, et     , al     , 2021    & 'https://doi.org/10.1186/s13148-021-01119-0' & '34193246'\\\\\n",
       "\t2 & 1 &  878583 &  880583 & NOC2L    & 1 & NOC2L    & NOC2 like nucleolar associated transcriptional repressor & HGNC:24517 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017 & Massie, et    , al.   , 2017   & 'https://doi.org/10.1016/j.jsbmb.2016.04.009' & '27117390'\\\\\n",
       "\t3 & 1 &  935518 &  937518 & ISG15    & 1 & ISG15    & ISG15 ubiquitin like modifier                            & HGNC:4053  & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017 & Massie, et    , al.   , 2017   & 'https://doi.org/10.1016/j.jsbmb.2016.04.009' & '27117390'\\\\\n",
       "\t4 & 1 & 1108260 & 1110260 & TTLL10   & 2 & TTLL10   & tubulin tyrosine ligase like 10                          & HGNC:26693 & 1p36.33 & + & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017 & Massie, et    , al.   , 2017   & 'https://doi.org/10.1016/j.jsbmb.2016.04.009' & '27117390'\\\\\n",
       "\t5 & 1 & 1269656 & 1271656 & DVL1     & 1 & DVL1     & dishevelled segment polarity protein 1                   & HGNC:3084  & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017 & Massie, et    , al.   , 2017   & 'https://doi.org/10.1016/j.jsbmb.2016.04.009' & '27117390'\\\\\n",
       "\t6 & 1 & 1308110 & 1310110 & AURKAIP1 & 1 & AURKAIP1 & aurora kinase A interacting protein 1                    & HGNC:24114 & 1p36.33 & - & promoter & 2000 & FALSE & 0 & 0 & Massie et al. 2017 & Massie, et    , al.   , 2017   & 'https://doi.org/10.1016/j.jsbmb.2016.04.009' & '27117390'\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 19\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.total.pubs &lt;int&gt; | hugo_gene_name &lt;chr&gt; | Approved.name &lt;chr&gt; | HGNC.ID &lt;chr&gt; | Location &lt;chr&gt; | strand &lt;chr&gt; | type &lt;chr&gt; | size &lt;int&gt; | is.whitelist &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | sources &lt;chr&gt; | first.author &lt;list&gt; | DOIs &lt;list&gt; | PMIDs &lt;list&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |  751751 |  753751 | FAM87B   | 1 | FAM87B   | family with sequence similarity 87 member B              | HGNC:32236 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Ylitalo et al 2021 | Ylitalo, et     , al     , 2021    | 'https://doi.org/10.1186/s13148-021-01119-0' | '34193246' |\n",
       "| 2 | 1 |  878583 |  880583 | NOC2L    | 1 | NOC2L    | NOC2 like nucleolar associated transcriptional repressor | HGNC:24517 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 | Massie, et    , al.   , 2017   | 'https://doi.org/10.1016/j.jsbmb.2016.04.009' | '27117390' |\n",
       "| 3 | 1 |  935518 |  937518 | ISG15    | 1 | ISG15    | ISG15 ubiquitin like modifier                            | HGNC:4053  | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 | Massie, et    , al.   , 2017   | 'https://doi.org/10.1016/j.jsbmb.2016.04.009' | '27117390' |\n",
       "| 4 | 1 | 1108260 | 1110260 | TTLL10   | 2 | TTLL10   | tubulin tyrosine ligase like 10                          | HGNC:26693 | 1p36.33 | + | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 | Massie, et    , al.   , 2017   | 'https://doi.org/10.1016/j.jsbmb.2016.04.009' | '27117390' |\n",
       "| 5 | 1 | 1269656 | 1271656 | DVL1     | 1 | DVL1     | dishevelled segment polarity protein 1                   | HGNC:3084  | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 | Massie, et    , al.   , 2017   | 'https://doi.org/10.1016/j.jsbmb.2016.04.009' | '27117390' |\n",
       "| 6 | 1 | 1308110 | 1310110 | AURKAIP1 | 1 | AURKAIP1 | aurora kinase A interacting protein 1                    | HGNC:24114 | 1p36.33 | - | promoter | 2000 | FALSE | 0 | 0 | Massie et al. 2017 | Massie, et    , al.   , 2017   | 'https://doi.org/10.1016/j.jsbmb.2016.04.009' | '27117390' |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 hybrid.gene.name no.total.pubs hugo_gene_name\n",
       "1 1    751751     753751  FAM87B           1             FAM87B        \n",
       "2 1    878583     880583  NOC2L            1             NOC2L         \n",
       "3 1    935518     937518  ISG15            1             ISG15         \n",
       "4 1   1108260    1110260  TTLL10           2             TTLL10        \n",
       "5 1   1269656    1271656  DVL1             1             DVL1          \n",
       "6 1   1308110    1310110  AURKAIP1         1             AURKAIP1      \n",
       "  Approved.name                                            HGNC.ID    Location\n",
       "1 family with sequence similarity 87 member B              HGNC:32236 1p36.33 \n",
       "2 NOC2 like nucleolar associated transcriptional repressor HGNC:24517 1p36.33 \n",
       "3 ISG15 ubiquitin like modifier                            HGNC:4053  1p36.33 \n",
       "4 tubulin tyrosine ligase like 10                          HGNC:26693 1p36.33 \n",
       "5 dishevelled segment polarity protein 1                   HGNC:3084  1p36.33 \n",
       "6 aurora kinase A interacting protein 1                    HGNC:24114 1p36.33 \n",
       "  strand type     size is.whitelist bp.tcga.overlap bp.ecm.overlap\n",
       "1 +      promoter 2000 FALSE        0               0             \n",
       "2 -      promoter 2000 FALSE        0               0             \n",
       "3 +      promoter 2000 FALSE        0               0             \n",
       "4 +      promoter 2000 FALSE        0               0             \n",
       "5 -      promoter 2000 FALSE        0               0             \n",
       "6 -      promoter 2000 FALSE        0               0             \n",
       "  sources            first.author                      \n",
       "1 Ylitalo et al 2021 Ylitalo, et     , al     , 2021   \n",
       "2 Massie et al. 2017 Massie, et    , al.   , 2017      \n",
       "3 Massie et al. 2017 Massie, et    , al.   , 2017      \n",
       "4 Massie et al. 2017 Massie, et    , al.   , 2017      \n",
       "5 Massie et al. 2017 Massie, et    , al.   , 2017      \n",
       "6 Massie et al. 2017 Massie, et    , al.   , 2017      \n",
       "  DOIs                                          PMIDs     \n",
       "1 'https://doi.org/10.1186/s13148-021-01119-0'  '34193246'\n",
       "2 'https://doi.org/10.1016/j.jsbmb.2016.04.009' '27117390'\n",
       "3 'https://doi.org/10.1016/j.jsbmb.2016.04.009' '27117390'\n",
       "4 'https://doi.org/10.1016/j.jsbmb.2016.04.009' '27117390'\n",
       "5 'https://doi.org/10.1016/j.jsbmb.2016.04.009' '27117390'\n",
       "6 'https://doi.org/10.1016/j.jsbmb.2016.04.009' '27117390'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_outfile$DOIs = doi_vec\n",
    "df_outfile$PMIDs = pmids_vec\n",
    "df_outfile[df_outfile$hybrid.gene.name==\"GSTP1\",]\n",
    "head(df_outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e2c1e3b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_outfile <- apply(df_outfile,2,as.character)\n",
    "\n",
    "write.table(df_outfile, \"./marker_collation/Prostate_lit_regions_annotated_10032022.tsv\",\n",
    "         row.names=FALSE, quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b680cfdf",
   "metadata": {},
   "source": [
    "## Process regions associated with disease progression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ce850186",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>hg18.location</th><th scope=col>hg19.location</th><th scope=col>source</th><th scope=col>association</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GSTP1 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CDH17 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>PCDH10</td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>PCDH8 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>RARB  </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MEG3  </td><td></td><td></td><td></td><td>Cai et al. 2021       </td><td>hypomethylation, assoc. with recurrence free survival</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & gene & probe & hg18.location & hg19.location & source & association\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & GSTP1  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                        \\\\\n",
       "\t2 & CDH17  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                        \\\\\n",
       "\t3 & PCDH10 &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                        \\\\\n",
       "\t4 & PCDH8  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                        \\\\\n",
       "\t5 & RARB   &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                        \\\\\n",
       "\t6 & MEG3   &  &  &  & Cai et al. 2021        & hypomethylation, assoc. with recurrence free survival\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | probe &lt;chr&gt; | hg18.location &lt;chr&gt; | hg19.location &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | GSTP1  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         |\n",
       "| 2 | CDH17  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         |\n",
       "| 3 | PCDH10 | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         |\n",
       "| 4 | PCDH8  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         |\n",
       "| 5 | RARB   | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         |\n",
       "| 6 | MEG3   | <!----> | <!----> | <!----> | Cai et al. 2021        | hypomethylation, assoc. with recurrence free survival |\n",
       "\n"
      ],
      "text/plain": [
       "  gene   probe hg18.location hg19.location source                \n",
       "1 GSTP1                                    Constancio et al. 2019\n",
       "2 CDH17                                    Constancio et al. 2019\n",
       "3 PCDH10                                   Constancio et al. 2019\n",
       "4 PCDH8                                    Constancio et al. 2019\n",
       "5 RARB                                     Constancio et al. 2019\n",
       "6 MEG3                                     Cai et al. 2021       \n",
       "  association                                          \n",
       "1 staging, survival, recurrence                        \n",
       "2 staging, survival, recurrence                        \n",
       "3 staging, survival, recurrence                        \n",
       "4 staging, survival, recurrence                        \n",
       "5 staging, survival, recurrence                        \n",
       "6 hypomethylation, assoc. with recurrence free survival"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_regions = read.csv(\"./marker_collation/disease_progression_loci.csv\", header=TRUE, sep=',')\n",
    "head(df_regions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "80618164",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "151"
      ],
      "text/latex": [
       "151"
      ],
      "text/markdown": [
       "151"
      ],
      "text/plain": [
       "[1] 151"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>hg18.location</th><th scope=col>hg19.location</th><th scope=col>source</th><th scope=col>association</th><th scope=col>hg19_chr</th><th scope=col>hg19_pos</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>C11orf80</td><td>cg19932343</td><td></td><td></td><td>Panja et al. 2018</td><td>hypomethylated, ADT resistance </td><td>11</td><td>66610763</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>RCE1    </td><td>cg19932343</td><td></td><td></td><td>Panja et al. 2018</td><td>hypomethylated, ADT resistance </td><td>11</td><td>66610763</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>CLDN17  </td><td>cg07201638</td><td></td><td></td><td>Panja et al. 2018</td><td>hypermethylated, ADT resistance</td><td>21</td><td>31538714</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SLC46A1 </td><td>cg25014117</td><td></td><td></td><td>Panja et al. 2018</td><td>hypomethylated, ADT resistance </td><td>17</td><td>26733068</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>STMN1   </td><td>cg23079732</td><td></td><td></td><td>Panja et al. 2018</td><td>hypomethylated, ADT resistance </td><td>1 </td><td>26232465</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>NOTCH4  </td><td>cg08592230</td><td></td><td></td><td>Panja et al. 2018</td><td>hypomethylated, ADT resistance </td><td>6 </td><td>32163562</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & probe & hg18.location & hg19.location & source & association & hg19\\_chr & hg19\\_pos\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & C11orf80 & cg19932343 &  &  & Panja et al. 2018 & hypomethylated, ADT resistance  & 11 & 66610763\\\\\n",
       "\t2 & RCE1     & cg19932343 &  &  & Panja et al. 2018 & hypomethylated, ADT resistance  & 11 & 66610763\\\\\n",
       "\t3 & CLDN17   & cg07201638 &  &  & Panja et al. 2018 & hypermethylated, ADT resistance & 21 & 31538714\\\\\n",
       "\t4 & SLC46A1  & cg25014117 &  &  & Panja et al. 2018 & hypomethylated, ADT resistance  & 17 & 26733068\\\\\n",
       "\t5 & STMN1    & cg23079732 &  &  & Panja et al. 2018 & hypomethylated, ADT resistance  & 1  & 26232465\\\\\n",
       "\t6 & NOTCH4   & cg08592230 &  &  & Panja et al. 2018 & hypomethylated, ADT resistance  & 6  & 32163562\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | probe &lt;chr&gt; | hg18.location &lt;chr&gt; | hg19.location &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | hg19_chr &lt;chr&gt; | hg19_pos &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | C11orf80 | cg19932343 | <!----> | <!----> | Panja et al. 2018 | hypomethylated, ADT resistance  | 11 | 66610763 |\n",
       "| 2 | RCE1     | cg19932343 | <!----> | <!----> | Panja et al. 2018 | hypomethylated, ADT resistance  | 11 | 66610763 |\n",
       "| 3 | CLDN17   | cg07201638 | <!----> | <!----> | Panja et al. 2018 | hypermethylated, ADT resistance | 21 | 31538714 |\n",
       "| 4 | SLC46A1  | cg25014117 | <!----> | <!----> | Panja et al. 2018 | hypomethylated, ADT resistance  | 17 | 26733068 |\n",
       "| 5 | STMN1    | cg23079732 | <!----> | <!----> | Panja et al. 2018 | hypomethylated, ADT resistance  | 1  | 26232465 |\n",
       "| 6 | NOTCH4   | cg08592230 | <!----> | <!----> | Panja et al. 2018 | hypomethylated, ADT resistance  | 6  | 32163562 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene     probe      hg18.location hg19.location source           \n",
       "1 C11orf80 cg19932343                             Panja et al. 2018\n",
       "2 RCE1     cg19932343                             Panja et al. 2018\n",
       "3 CLDN17   cg07201638                             Panja et al. 2018\n",
       "4 SLC46A1  cg25014117                             Panja et al. 2018\n",
       "5 STMN1    cg23079732                             Panja et al. 2018\n",
       "6 NOTCH4   cg08592230                             Panja et al. 2018\n",
       "  association                     hg19_chr hg19_pos\n",
       "1 hypomethylated, ADT resistance  11       66610763\n",
       "2 hypomethylated, ADT resistance  11       66610763\n",
       "3 hypermethylated, ADT resistance 21       31538714\n",
       "4 hypomethylated, ADT resistance  17       26733068\n",
       "5 hypomethylated, ADT resistance  1        26232465\n",
       "6 hypomethylated, ADT resistance  6        32163562"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# this file has some weird formatting \n",
    "# grep cg humanmethylation450_15017482_v1-2.csv > humanmethylation450_15017482_v1-2_cg.csv   \n",
    "#  tail -n +8 humanmethylation450_15017482_v1-2_cg.csv| cut -d \",\" -f1,12-13 > humanmethylation450_15017482_v1-2_coords.csv\n",
    "\n",
    "# look up probe hg19 locations\n",
    "df_450k = read.table(\"./../Translational_Science/humanmethylation450_15017482_v1-2_coords.csv\", header=FALSE, sep=\",\")\n",
    "colnames(df_450k) = c(\"probe\", \"hg19_chr\", \"hg19_pos\")\n",
    "\n",
    "df_probes = df_regions %>% filter(probe!=\"\") %>% left_join(df_450k, by=\"probe\")\n",
    "nrow(df_probes)\n",
    "head(df_probes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "408e43e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_probes$hg19_start = df_probes$hg19_pos - 10\n",
    "df_probes$hg19_end = df_probes$hg19_pos + 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "96ec429f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start_hg19</th><th scope=col>end_hg19</th><th scope=col>gene_name</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>10869</td><td>12869</td><td>DDX11L1    </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>13404</td><td>15404</td><td>WASH7P     </td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>28554</td><td>30554</td><td>MIR1302-2HG</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>33554</td><td>35554</td><td>FAM138A    </td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>51473</td><td>53473</td><td>OR4G4P     </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>56598</td><td>58598</td><td>OR4G11P    </td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & chr & start\\_hg19 & end\\_hg19 & gene\\_name & strand & dot2 & type\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 10869 & 12869 & DDX11L1     & + & . & promoter\\\\\n",
       "\t2 & 1 & 13404 & 15404 & WASH7P      & - & . & promoter\\\\\n",
       "\t3 & 1 & 28554 & 30554 & MIR1302-2HG & + & . & promoter\\\\\n",
       "\t4 & 1 & 33554 & 35554 & FAM138A     & - & . & promoter\\\\\n",
       "\t5 & 1 & 51473 & 53473 & OR4G4P      & + & . & promoter\\\\\n",
       "\t6 & 1 & 56598 & 58598 & OR4G11P     & + & . & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start_hg19 &lt;int&gt; | end_hg19 &lt;int&gt; | gene_name &lt;chr&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 10869 | 12869 | DDX11L1     | + | . | promoter |\n",
       "| 2 | 1 | 13404 | 15404 | WASH7P      | - | . | promoter |\n",
       "| 3 | 1 | 28554 | 30554 | MIR1302-2HG | + | . | promoter |\n",
       "| 4 | 1 | 33554 | 35554 | FAM138A     | - | . | promoter |\n",
       "| 5 | 1 | 51473 | 53473 | OR4G4P      | + | . | promoter |\n",
       "| 6 | 1 | 56598 | 58598 | OR4G11P     | + | . | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start_hg19 end_hg19 gene_name   strand dot2 type    \n",
       "1 1   10869      12869    DDX11L1     +      .    promoter\n",
       "2 1   13404      15404    WASH7P      -      .    promoter\n",
       "3 1   28554      30554    MIR1302-2HG +      .    promoter\n",
       "4 1   33554      35554    FAM138A     -      .    promoter\n",
       "5 1   51473      53473    OR4G4P      +      .    promoter\n",
       "6 1   56598      58598    OR4G11P     +      .    promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_promoters = read.csv(\"../Translational_Science/gencode.v40lift37.promoters.bed\", sep=\"\\t\", header=TRUE)\n",
    "head(df_promoters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "d17e5b20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "26"
      ],
      "text/latex": [
       "26"
      ],
      "text/markdown": [
       "26"
      ],
      "text/plain": [
       "[1] 26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>hg18.location</th><th scope=col>hg19.location</th><th scope=col>source</th><th scope=col>association</th><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>strand</th><th scope=col>dot2</th><th scope=col>type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GSTP1 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>11</td><td> 67350213</td><td> 67352213</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CDH17 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>8 </td><td> 95138390</td><td> 95140390</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>PCDH10</td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>4 </td><td>134069449</td><td>134071449</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>PCDH8 </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>13</td><td> 53416024</td><td> 53418024</td><td>-</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>RARB  </td><td></td><td></td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>3 </td><td> 24728378</td><td> 24730378</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MEG3  </td><td></td><td></td><td></td><td>Cai et al. 2021       </td><td>hypomethylation, assoc. with recurrence free survival</td><td>14</td><td>101244747</td><td>101246747</td><td>+</td><td>.</td><td>promoter</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & gene & probe & hg18.location & hg19.location & source & association & hg19\\_chr & hg19\\_start & hg19\\_end & strand & dot2 & type\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & GSTP1  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                         & 11 &  67350213 &  67352213 & + & . & promoter\\\\\n",
       "\t2 & CDH17  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                         & 8  &  95138390 &  95140390 & - & . & promoter\\\\\n",
       "\t3 & PCDH10 &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                         & 4  & 134069449 & 134071449 & + & . & promoter\\\\\n",
       "\t4 & PCDH8  &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                         & 13 &  53416024 &  53418024 & - & . & promoter\\\\\n",
       "\t5 & RARB   &  &  &  & Constancio et al. 2019 & staging, survival, recurrence                         & 3  &  24728378 &  24730378 & + & . & promoter\\\\\n",
       "\t6 & MEG3   &  &  &  & Cai et al. 2021        & hypomethylation, assoc. with recurrence free survival & 14 & 101244747 & 101246747 & + & . & promoter\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | probe &lt;chr&gt; | hg18.location &lt;chr&gt; | hg19.location &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | hg19_chr &lt;chr&gt; | hg19_start &lt;int&gt; | hg19_end &lt;int&gt; | strand &lt;chr&gt; | dot2 &lt;chr&gt; | type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | GSTP1  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | 11 |  67350213 |  67352213 | + | . | promoter |\n",
       "| 2 | CDH17  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | 8  |  95138390 |  95140390 | - | . | promoter |\n",
       "| 3 | PCDH10 | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | 4  | 134069449 | 134071449 | + | . | promoter |\n",
       "| 4 | PCDH8  | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | 13 |  53416024 |  53418024 | - | . | promoter |\n",
       "| 5 | RARB   | <!----> | <!----> | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | 3  |  24728378 |  24730378 | + | . | promoter |\n",
       "| 6 | MEG3   | <!----> | <!----> | <!----> | Cai et al. 2021        | hypomethylation, assoc. with recurrence free survival | 14 | 101244747 | 101246747 | + | . | promoter |\n",
       "\n"
      ],
      "text/plain": [
       "  gene   probe hg18.location hg19.location source                \n",
       "1 GSTP1                                    Constancio et al. 2019\n",
       "2 CDH17                                    Constancio et al. 2019\n",
       "3 PCDH10                                   Constancio et al. 2019\n",
       "4 PCDH8                                    Constancio et al. 2019\n",
       "5 RARB                                     Constancio et al. 2019\n",
       "6 MEG3                                     Cai et al. 2021       \n",
       "  association                                           hg19_chr hg19_start\n",
       "1 staging, survival, recurrence                         11        67350213 \n",
       "2 staging, survival, recurrence                         8         95138390 \n",
       "3 staging, survival, recurrence                         4        134069449 \n",
       "4 staging, survival, recurrence                         13        53416024 \n",
       "5 staging, survival, recurrence                         3         24728378 \n",
       "6 hypomethylation, assoc. with recurrence free survival 14       101244747 \n",
       "  hg19_end  strand dot2 type    \n",
       "1  67352213 +      .    promoter\n",
       "2  95140390 -      .    promoter\n",
       "3 134071449 +      .    promoter\n",
       "4  53418024 -      .    promoter\n",
       "5  24730378 +      .    promoter\n",
       "6 101246747 +      .    promoter"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_genes = df_regions %>% filter(probe==\"\", hg18.location==\"\", hg19.location==\"\") %>% left_join(df_promoters, by=c(\"gene\"=\"gene_name\"))\n",
    "df_genes = df_genes %>% rename(\"hg19_chr\"=\"chr\", \"hg19_end\"=\"end_hg19\", \"hg19_start\"=\"start_hg19\")\n",
    "nrow(df_genes)\n",
    "head(df_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "fce93af4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1:17022023-17022067   </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr13:111156571-111156615</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr13:111290973-111291032</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr17:78988130-78988189  </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr18:77662518-77662577  </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr20:63278124-63278183  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & V1\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1:17022023-17022067   \\\\\n",
       "\t2 & chr13:111156571-111156615\\\\\n",
       "\t3 & chr13:111290973-111291032\\\\\n",
       "\t4 & chr17:78988130-78988189  \\\\\n",
       "\t5 & chr18:77662518-77662577  \\\\\n",
       "\t6 & chr20:63278124-63278183  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1 | chr1:17022023-17022067    |\n",
       "| 2 | chr13:111156571-111156615 |\n",
       "| 3 | chr13:111290973-111291032 |\n",
       "| 4 | chr17:78988130-78988189   |\n",
       "| 5 | chr18:77662518-77662577   |\n",
       "| 6 | chr20:63278124-63278183   |\n",
       "\n"
      ],
      "text/plain": [
       "  V1                       \n",
       "1 chr1:17022023-17022067   \n",
       "2 chr13:111156571-111156615\n",
       "3 chr13:111290973-111291032\n",
       "4 chr17:78988130-78988189  \n",
       "5 chr18:77662518-77662577  \n",
       "6 chr20:63278124-63278183  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1:100540281-100540325</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1:108199715-108199774</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1:1174388-1174447    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1:118984317-118984367</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1:118989514-118989558</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1:118993109-118993153</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & V1\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1:100540281-100540325\\\\\n",
       "\t2 & chr1:108199715-108199774\\\\\n",
       "\t3 & chr1:1174388-1174447    \\\\\n",
       "\t4 & chr1:118984317-118984367\\\\\n",
       "\t5 & chr1:118989514-118989558\\\\\n",
       "\t6 & chr1:118993109-118993153\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1 | chr1:100540281-100540325 |\n",
       "| 2 | chr1:108199715-108199774 |\n",
       "| 3 | chr1:1174388-1174447     |\n",
       "| 4 | chr1:118984317-118984367 |\n",
       "| 5 | chr1:118989514-118989558 |\n",
       "| 6 | chr1:118993109-118993153 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1                      \n",
       "1 chr1:100540281-100540325\n",
       "2 chr1:108199715-108199774\n",
       "3 chr1:1174388-1174447    \n",
       "4 chr1:118984317-118984367\n",
       "5 chr1:118989514-118989558\n",
       "6 chr1:118993109-118993153"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# lift over hg18 coords for Kron et al. \n",
    "failed.hits = read.table(\"./marker_collation/Kron_disease_prog_hglft_genome_343ef_a4cd40.err.txt\", sep=\"\\t\", header=FALSE)\n",
    "head(failed.hits)\n",
    "conversions = read.table(\"./marker_collation/Kron_disease_prog_hglft_genome_343ef_a4cd40.bed\", sep=\"\\t\", header=FALSE)\n",
    "head(conversions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "8d66c77a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "421"
      ],
      "text/latex": [
       "421"
      ],
      "text/markdown": [
       "421"
      ],
      "text/plain": [
       "[1] 421"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "415"
      ],
      "text/latex": [
       "415"
      ],
      "text/markdown": [
       "415"
      ],
      "text/plain": [
       "[1] 415"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>hg18.location</th><th scope=col>hg19.location</th><th scope=col>source</th><th scope=col>association</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>410</th><td>chrX:039752897-039752946</td><td></td><td>chrX:39893643-39893692</td><td>chrX:40008699-40008748</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "\t<tr><th scope=row>411</th><td>BCOR                    </td><td></td><td>chrX:39976744-39976788</td><td>chrX:40091800-40091844</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "\t<tr><th scope=row>412</th><td>chrX:064544802-064544856</td><td></td><td>chrX:65324922-65324976</td><td>chrX:65408197-65408251</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "\t<tr><th scope=row>413</th><td>OGT                     </td><td></td><td>chrX:71449418-71449477</td><td>chrX:71532693-71532752</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "\t<tr><th scope=row>414</th><td>chrY:027182977-027183036</td><td></td><td>chrY:25036830-25036889</td><td>chrY:26627442-26627501</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "\t<tr><th scope=row>415</th><td>PPP2R3B                 </td><td></td><td>chrY:322187-322245    </td><td>chrY:352187-352245    </td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & gene & probe & hg18.location & hg19.location & source & association\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t410 & chrX:039752897-039752946 &  & chrX:39893643-39893692 & chrX:40008699-40008748 & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\t411 & BCOR                     &  & chrX:39976744-39976788 & chrX:40091800-40091844 & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\t412 & chrX:064544802-064544856 &  & chrX:65324922-65324976 & chrX:65408197-65408251 & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\t413 & OGT                      &  & chrX:71449418-71449477 & chrX:71532693-71532752 & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\t414 & chrY:027182977-027183036 &  & chrY:25036830-25036889 & chrY:26627442-26627501 & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\t415 & PPP2R3B                  &  & chrY:322187-322245     & chrY:352187-352245     & Kron et al. 2013 & hypermethylation assoc w/Gleason\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | probe &lt;chr&gt; | hg18.location &lt;chr&gt; | hg19.location &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 410 | chrX:039752897-039752946 | <!----> | chrX:39893643-39893692 | chrX:40008699-40008748 | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "| 411 | BCOR                     | <!----> | chrX:39976744-39976788 | chrX:40091800-40091844 | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "| 412 | chrX:064544802-064544856 | <!----> | chrX:65324922-65324976 | chrX:65408197-65408251 | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "| 413 | OGT                      | <!----> | chrX:71449418-71449477 | chrX:71532693-71532752 | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "| 414 | chrY:027182977-027183036 | <!----> | chrY:25036830-25036889 | chrY:26627442-26627501 | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "| 415 | PPP2R3B                  | <!----> | chrY:322187-322245     | chrY:352187-352245     | Kron et al. 2013 | hypermethylation assoc w/Gleason |\n",
       "\n"
      ],
      "text/plain": [
       "    gene                     probe hg18.location         \n",
       "410 chrX:039752897-039752946       chrX:39893643-39893692\n",
       "411 BCOR                           chrX:39976744-39976788\n",
       "412 chrX:064544802-064544856       chrX:65324922-65324976\n",
       "413 OGT                            chrX:71449418-71449477\n",
       "414 chrY:027182977-027183036       chrY:25036830-25036889\n",
       "415 PPP2R3B                        chrY:322187-322245    \n",
       "    hg19.location          source           association                     \n",
       "410 chrX:40008699-40008748 Kron et al. 2013 hypermethylation assoc w/Gleason\n",
       "411 chrX:40091800-40091844 Kron et al. 2013 hypermethylation assoc w/Gleason\n",
       "412 chrX:65408197-65408251 Kron et al. 2013 hypermethylation assoc w/Gleason\n",
       "413 chrX:71532693-71532752 Kron et al. 2013 hypermethylation assoc w/Gleason\n",
       "414 chrY:26627442-26627501 Kron et al. 2013 hypermethylation assoc w/Gleason\n",
       "415 chrY:352187-352245     Kron et al. 2013 hypermethylation assoc w/Gleason"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#remove failed hits but keep in same order\n",
    "df_kron = df_regions %>% filter(grepl(\"Kron\",source), hg18.location!=\"\") \n",
    "nrow(df_kron)\n",
    "for (loc in failed.hits$V1){\n",
    "    df_kron = df_kron %>% filter(hg18.location!=loc)\n",
    "}\n",
    "\n",
    "# add lift over coords\n",
    "df_kron$hg19.location <- conversions$V1\n",
    "nrow(df_kron)\n",
    "tail(df_kron)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "633e8bbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kron$hg19_chr = sapply(strsplit(df_kron$hg19.location,\":|-\"), \"[\",1)\n",
    "df_kron$hg19_start = sapply(strsplit(df_kron$hg19.location,\":|-\"), \"[\",2)\n",
    "df_kron$hg19_end = sapply(strsplit(df_kron$hg19.location,\":|-\"), \"[\",3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "160044bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# these are all 2 bp probe locations\n",
    "df_beltran = df_regions %>% filter(grepl(\"Beltran\", source), hg19.location!=\"\")\n",
    "df_beltran$hg19_chr = sapply(strsplit(df_beltran$hg19.location,\":|-\"), \"[\",1)\n",
    "df_beltran$hg19_start = sapply(strsplit(df_beltran$hg19.location,\":|-\"), \"[\",2)\n",
    "df_beltran$hg19_end = sapply(strsplit(df_beltran$hg19.location,\":|-\"), \"[\",3)\n",
    "\n",
    "df_beltran$hg19_start = as.numeric(df_beltran$hg19_start) - 10\n",
    "df_beltran$hg19_end = as.numeric(df_beltran$hg19_start) + 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "0cba5b18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>hg18.location</th><th scope=col>hg19.location</th><th scope=col>source</th><th scope=col>association</th><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GPR88   </td><td></td><td>chr1:100312869-100312913</td><td>chr1:100540281-100540325</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>1</td><td>100540281</td><td>100540325</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SLC25A24</td><td></td><td>chr1:108001238-108001297</td><td>chr1:108199715-108199774</td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>1</td><td>108199715</td><td>108199774</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>TTLL10  </td><td></td><td>chr1:1164251-1164310    </td><td>chr1:1174388-1174447    </td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>1</td><td>1174388  </td><td>1174447  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>TBX15   </td><td></td><td>chr1:118785840-118785890</td><td>chr1:118984317-118984367</td><td>Kron et al. 2013</td><td>hypomethylation assoc w/Gleason </td><td>1</td><td>118984317</td><td>118984367</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TBX15   </td><td></td><td>chr1:118791037-118791081</td><td>chr1:118989514-118989558</td><td>Kron et al. 2013</td><td>hypomethylation assoc w/Gleason </td><td>1</td><td>118989514</td><td>118989558</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>TBX15   </td><td></td><td>chr1:118794632-118794676</td><td>chr1:118993109-118993153</td><td>Kron et al. 2013</td><td>hypomethylation assoc w/Gleason </td><td>1</td><td>118993109</td><td>118993153</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & gene & probe & hg18.location & hg19.location & source & association & hg19\\_chr & hg19\\_start & hg19\\_end\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & GPR88    &  & chr1:100312869-100312913 & chr1:100540281-100540325 & Kron et al. 2013 & hypermethylation assoc w/Gleason & 1 & 100540281 & 100540325\\\\\n",
       "\t2 & SLC25A24 &  & chr1:108001238-108001297 & chr1:108199715-108199774 & Kron et al. 2013 & hypermethylation assoc w/Gleason & 1 & 108199715 & 108199774\\\\\n",
       "\t3 & TTLL10   &  & chr1:1164251-1164310     & chr1:1174388-1174447     & Kron et al. 2013 & hypermethylation assoc w/Gleason & 1 & 1174388   & 1174447  \\\\\n",
       "\t4 & TBX15    &  & chr1:118785840-118785890 & chr1:118984317-118984367 & Kron et al. 2013 & hypomethylation assoc w/Gleason  & 1 & 118984317 & 118984367\\\\\n",
       "\t5 & TBX15    &  & chr1:118791037-118791081 & chr1:118989514-118989558 & Kron et al. 2013 & hypomethylation assoc w/Gleason  & 1 & 118989514 & 118989558\\\\\n",
       "\t6 & TBX15    &  & chr1:118794632-118794676 & chr1:118993109-118993153 & Kron et al. 2013 & hypomethylation assoc w/Gleason  & 1 & 118993109 & 118993153\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | probe &lt;chr&gt; | hg18.location &lt;chr&gt; | hg19.location &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | hg19_chr &lt;chr&gt; | hg19_start &lt;chr&gt; | hg19_end &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | GPR88    | <!----> | chr1:100312869-100312913 | chr1:100540281-100540325 | Kron et al. 2013 | hypermethylation assoc w/Gleason | 1 | 100540281 | 100540325 |\n",
       "| 2 | SLC25A24 | <!----> | chr1:108001238-108001297 | chr1:108199715-108199774 | Kron et al. 2013 | hypermethylation assoc w/Gleason | 1 | 108199715 | 108199774 |\n",
       "| 3 | TTLL10   | <!----> | chr1:1164251-1164310     | chr1:1174388-1174447     | Kron et al. 2013 | hypermethylation assoc w/Gleason | 1 | 1174388   | 1174447   |\n",
       "| 4 | TBX15    | <!----> | chr1:118785840-118785890 | chr1:118984317-118984367 | Kron et al. 2013 | hypomethylation assoc w/Gleason  | 1 | 118984317 | 118984367 |\n",
       "| 5 | TBX15    | <!----> | chr1:118791037-118791081 | chr1:118989514-118989558 | Kron et al. 2013 | hypomethylation assoc w/Gleason  | 1 | 118989514 | 118989558 |\n",
       "| 6 | TBX15    | <!----> | chr1:118794632-118794676 | chr1:118993109-118993153 | Kron et al. 2013 | hypomethylation assoc w/Gleason  | 1 | 118993109 | 118993153 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene     probe hg18.location            hg19.location           \n",
       "1 GPR88          chr1:100312869-100312913 chr1:100540281-100540325\n",
       "2 SLC25A24       chr1:108001238-108001297 chr1:108199715-108199774\n",
       "3 TTLL10         chr1:1164251-1164310     chr1:1174388-1174447    \n",
       "4 TBX15          chr1:118785840-118785890 chr1:118984317-118984367\n",
       "5 TBX15          chr1:118791037-118791081 chr1:118989514-118989558\n",
       "6 TBX15          chr1:118794632-118794676 chr1:118993109-118993153\n",
       "  source           association                      hg19_chr hg19_start\n",
       "1 Kron et al. 2013 hypermethylation assoc w/Gleason 1        100540281 \n",
       "2 Kron et al. 2013 hypermethylation assoc w/Gleason 1        108199715 \n",
       "3 Kron et al. 2013 hypermethylation assoc w/Gleason 1        1174388   \n",
       "4 Kron et al. 2013 hypomethylation assoc w/Gleason  1        118984317 \n",
       "5 Kron et al. 2013 hypomethylation assoc w/Gleason  1        118989514 \n",
       "6 Kron et al. 2013 hypomethylation assoc w/Gleason  1        118993109 \n",
       "  hg19_end \n",
       "1 100540325\n",
       "2 108199774\n",
       "3 1174447  \n",
       "4 118984367\n",
       "5 118989558\n",
       "6 118993153"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_kron_beltran = rbind(df_kron, df_beltran)\n",
    "df_kron_beltran$hg19_chr = sapply(strsplit(df_kron_beltran$hg19_chr,\"chr\"),\"[\",2)\n",
    "head(df_kron_beltran)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "08e979de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'gene'</li><li>'probe'</li><li>'hg18.location'</li><li>'hg19.location'</li><li>'source'</li><li>'association'</li><li>'hg19_chr'</li><li>'hg19_start'</li><li>'hg19_end'</li><li>'strand'</li><li>'dot2'</li><li>'type'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'gene'\n",
       "\\item 'probe'\n",
       "\\item 'hg18.location'\n",
       "\\item 'hg19.location'\n",
       "\\item 'source'\n",
       "\\item 'association'\n",
       "\\item 'hg19\\_chr'\n",
       "\\item 'hg19\\_start'\n",
       "\\item 'hg19\\_end'\n",
       "\\item 'strand'\n",
       "\\item 'dot2'\n",
       "\\item 'type'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'gene'\n",
       "2. 'probe'\n",
       "3. 'hg18.location'\n",
       "4. 'hg19.location'\n",
       "5. 'source'\n",
       "6. 'association'\n",
       "7. 'hg19_chr'\n",
       "8. 'hg19_start'\n",
       "9. 'hg19_end'\n",
       "10. 'strand'\n",
       "11. 'dot2'\n",
       "12. 'type'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"gene\"          \"probe\"         \"hg18.location\" \"hg19.location\"\n",
       " [5] \"source\"        \"association\"   \"hg19_chr\"      \"hg19_start\"   \n",
       " [9] \"hg19_end\"      \"strand\"        \"dot2\"          \"type\"         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'gene'</li><li>'probe'</li><li>'hg18.location'</li><li>'hg19.location'</li><li>'source'</li><li>'association'</li><li>'hg19_chr'</li><li>'hg19_pos'</li><li>'hg19_start'</li><li>'hg19_end'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'gene'\n",
       "\\item 'probe'\n",
       "\\item 'hg18.location'\n",
       "\\item 'hg19.location'\n",
       "\\item 'source'\n",
       "\\item 'association'\n",
       "\\item 'hg19\\_chr'\n",
       "\\item 'hg19\\_pos'\n",
       "\\item 'hg19\\_start'\n",
       "\\item 'hg19\\_end'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'gene'\n",
       "2. 'probe'\n",
       "3. 'hg18.location'\n",
       "4. 'hg19.location'\n",
       "5. 'source'\n",
       "6. 'association'\n",
       "7. 'hg19_chr'\n",
       "8. 'hg19_pos'\n",
       "9. 'hg19_start'\n",
       "10. 'hg19_end'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"gene\"          \"probe\"         \"hg18.location\" \"hg19.location\"\n",
       " [5] \"source\"        \"association\"   \"hg19_chr\"      \"hg19_pos\"     \n",
       " [9] \"hg19_start\"    \"hg19_end\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'gene'</li><li>'probe'</li><li>'hg18.location'</li><li>'hg19.location'</li><li>'source'</li><li>'association'</li><li>'hg19_chr'</li><li>'hg19_start'</li><li>'hg19_end'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'gene'\n",
       "\\item 'probe'\n",
       "\\item 'hg18.location'\n",
       "\\item 'hg19.location'\n",
       "\\item 'source'\n",
       "\\item 'association'\n",
       "\\item 'hg19\\_chr'\n",
       "\\item 'hg19\\_start'\n",
       "\\item 'hg19\\_end'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'gene'\n",
       "2. 'probe'\n",
       "3. 'hg18.location'\n",
       "4. 'hg19.location'\n",
       "5. 'source'\n",
       "6. 'association'\n",
       "7. 'hg19_chr'\n",
       "8. 'hg19_start'\n",
       "9. 'hg19_end'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"gene\"          \"probe\"         \"hg18.location\" \"hg19.location\"\n",
       "[5] \"source\"        \"association\"   \"hg19_chr\"      \"hg19_start\"   \n",
       "[9] \"hg19_end\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(df_genes)\n",
    "colnames(df_probes)\n",
    "colnames(df_kron_beltran)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "ca943651",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "631"
      ],
      "text/latex": [
       "631"
      ],
      "text/markdown": [
       "631"
      ],
      "text/plain": [
       "[1] 631"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "629"
      ],
      "text/latex": [
       "629"
      ],
      "text/markdown": [
       "629"
      ],
      "text/plain": [
       "[1] 629"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols = c(\"hg19_chr\",\"hg19_start\",\"hg19_end\",\"gene\",\"probe\",\"source\",\"association\")\n",
    "df_regions2 = rbind(df_genes[,cols],df_probes[,cols],df_kron_beltran[,cols])\n",
    "nrow(df_regions2)\n",
    "df_regions2 = df_regions2[complete.cases(df_regions2),]\n",
    "nrow(df_regions2)\n",
    "write.table(df_regions2, file=\"./marker_collation/disease_prog_loci_hg19.bed\", col.names=TRUE, sep=\"\\t\", row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "47eda78a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>source</th><th scope=col>association</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " hg19\\_chr & hg19\\_start & hg19\\_end & gene & probe & source & association\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 7\n",
       "\n",
       "| hg19_chr &lt;chr&gt; | hg19_start &lt;chr&gt; | hg19_end &lt;chr&gt; | gene &lt;chr&gt; | probe &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     hg19_chr hg19_start hg19_end gene probe source association"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# look for overlaps with promoters, tcga, and ecm\n",
    "\n",
    "# vim disease_prog_loci_hg19.bed  # comment out header\n",
    "# sort -k1,1 -k2,2 -n disease_prog_loci_hg19.bed >disease_prog_loci_hg19_sorted.bed\n",
    "# bedtools intersect -a disease_prog_loci_hg19_sorted.bed -b ../../Translational_Science/gencode.v40lift37.promoters.bed -wao > disease_prog_loci_prom_overlap.out\n",
    "# bedtools intersect -a disease_prog_loci_hg19_sorted.bed -b ../PRAD_450k_DMR.deepFilter_0.33_0.67_minProbes3_fmt.bed -wao > disease_prog_loci_tcga_overlap.out\n",
    "# bedtools intersect -a disease_prog_loci_hg19_sorted.bed -b ../ECM_DMRs_raw_Hang_05172022_sorted.bed -wao > disease_prog_loci_ecm_overlap.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "6201ceeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_prom = read.table(\"./marker_collation/disease_prog_loci_prom_overlap.out\", header=FALSE, sep=\"\\t\")\n",
    "\n",
    "df_tcga = read.table(\"./marker_collation/disease_prog_loci_tcga_overlap.out\", header=FALSE, sep=\"\\t\")\n",
    "\n",
    "df_ecm = read.table(\"./marker_collation/disease_prog_loci_ecm_overlap.out\", header=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "5bfed233",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Y</td><td>  352187</td><td>  352245</td><td>PPP2R3B                 </td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>X</td><td>25003756</td><td>25003815</td><td>ARX                     </td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Y</td><td>26627442</td><td>26627501</td><td>chrY:027182977-027183036</td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>X</td><td>40008699</td><td>40008748</td><td>chrX:039752897-039752946</td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>X</td><td>40091800</td><td>40091844</td><td>BCOR                    </td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>X</td><td>65408197</td><td>65408251</td><td>chrX:064544802-064544856</td><td></td><td>Kron et al. 2013</td><td>hypermethylation assoc w/Gleason</td><td>.</td><td>-1</td><td>-1</td><td>.</td><td>.</td><td>.</td><td>.</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13 & V14 & V15\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Y &   352187 &   352245 & PPP2R3B                  &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\t2 & X & 25003756 & 25003815 & ARX                      &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\t3 & Y & 26627442 & 26627501 & chrY:027182977-027183036 &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\t4 & X & 40008699 & 40008748 & chrX:039752897-039752946 &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\t5 & X & 40091800 & 40091844 & BCOR                     &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\t6 & X & 65408197 & 65408251 & chrX:064544802-064544856 &  & Kron et al. 2013 & hypermethylation assoc w/Gleason & . & -1 & -1 & . & . & . & . & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;int&gt; | V10 &lt;int&gt; | V11 &lt;chr&gt; | V12 &lt;chr&gt; | V13 &lt;chr&gt; | V14 &lt;chr&gt; | V15 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | Y |   352187 |   352245 | PPP2R3B                  | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "| 2 | X | 25003756 | 25003815 | ARX                      | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "| 3 | Y | 26627442 | 26627501 | chrY:027182977-027183036 | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "| 4 | X | 40008699 | 40008748 | chrX:039752897-039752946 | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "| 5 | X | 40091800 | 40091844 | BCOR                     | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "| 6 | X | 65408197 | 65408251 | chrX:064544802-064544856 | <!----> | Kron et al. 2013 | hypermethylation assoc w/Gleason | . | -1 | -1 | . | . | . | . | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2       V3       V4                       V5 V6              \n",
       "1 Y    352187   352245 PPP2R3B                     Kron et al. 2013\n",
       "2 X  25003756 25003815 ARX                         Kron et al. 2013\n",
       "3 Y  26627442 26627501 chrY:027182977-027183036    Kron et al. 2013\n",
       "4 X  40008699 40008748 chrX:039752897-039752946    Kron et al. 2013\n",
       "5 X  40091800 40091844 BCOR                        Kron et al. 2013\n",
       "6 X  65408197 65408251 chrX:064544802-064544856    Kron et al. 2013\n",
       "  V7                               V8 V9 V10 V11 V12 V13 V14 V15\n",
       "1 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  \n",
       "2 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  \n",
       "3 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  \n",
       "4 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  \n",
       "5 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  \n",
       "6 hypermethylation assoc w/Gleason .  -1 -1  .   .   .   .   0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df_prom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "4af3e7e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_regions2$hg19_start = as.integer(df_regions2$hg19_start)\n",
    "df_regions2$hg19_end = as.integer(df_regions2$hg19_end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "53a56204",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>source</th><th scope=col>association</th><th scope=col>promoter_name</th><th scope=col>is.promoter</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>11</td><td> 67350213</td><td> 67352213</td><td>GSTP1 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>GSTP1 </td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>8 </td><td> 95138390</td><td> 95140390</td><td>CDH17 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>CDH17 </td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>4 </td><td>134069449</td><td>134071449</td><td>PCDH10</td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH10</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>13</td><td> 53416024</td><td> 53418024</td><td>PCDH8 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH8 </td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>3 </td><td> 24728378</td><td> 24730378</td><td>RARB  </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>RARB  </td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>14</td><td>101244747</td><td>101246747</td><td>MEG3  </td><td></td><td>Cai et al. 2021       </td><td>hypomethylation, assoc. with recurrence free survival</td><td>MEG3  </td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & hg19\\_chr & hg19\\_start & hg19\\_end & gene & probe & source & association & promoter\\_name & is.promoter\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & 11 &  67350213 &  67352213 & GSTP1  &  & Constancio et al. 2019 & staging, survival, recurrence                         & GSTP1  & TRUE\\\\\n",
       "\t2 & 8  &  95138390 &  95140390 & CDH17  &  & Constancio et al. 2019 & staging, survival, recurrence                         & CDH17  & TRUE\\\\\n",
       "\t3 & 4  & 134069449 & 134071449 & PCDH10 &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH10 & TRUE\\\\\n",
       "\t4 & 13 &  53416024 &  53418024 & PCDH8  &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH8  & TRUE\\\\\n",
       "\t5 & 3  &  24728378 &  24730378 & RARB   &  & Constancio et al. 2019 & staging, survival, recurrence                         & RARB   & TRUE\\\\\n",
       "\t6 & 14 & 101244747 & 101246747 & MEG3   &  & Cai et al. 2021        & hypomethylation, assoc. with recurrence free survival & MEG3   & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | hg19_chr &lt;chr&gt; | hg19_start &lt;int&gt; | hg19_end &lt;int&gt; | gene &lt;chr&gt; | probe &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | promoter_name &lt;chr&gt; | is.promoter &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 11 |  67350213 |  67352213 | GSTP1  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | GSTP1  | TRUE |\n",
       "| 2 | 8  |  95138390 |  95140390 | CDH17  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | CDH17  | TRUE |\n",
       "| 3 | 4  | 134069449 | 134071449 | PCDH10 | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH10 | TRUE |\n",
       "| 4 | 13 |  53416024 |  53418024 | PCDH8  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH8  | TRUE |\n",
       "| 5 | 3  |  24728378 |  24730378 | RARB   | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | RARB   | TRUE |\n",
       "| 6 | 14 | 101244747 | 101246747 | MEG3   | <!----> | Cai et al. 2021        | hypomethylation, assoc. with recurrence free survival | MEG3   | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "  hg19_chr hg19_start hg19_end  gene   probe source                \n",
       "1 11        67350213   67352213 GSTP1        Constancio et al. 2019\n",
       "2 8         95138390   95140390 CDH17        Constancio et al. 2019\n",
       "3 4        134069449  134071449 PCDH10       Constancio et al. 2019\n",
       "4 13        53416024   53418024 PCDH8        Constancio et al. 2019\n",
       "5 3         24728378   24730378 RARB         Constancio et al. 2019\n",
       "6 14       101244747  101246747 MEG3         Cai et al. 2021       \n",
       "  association                                           promoter_name\n",
       "1 staging, survival, recurrence                         GSTP1        \n",
       "2 staging, survival, recurrence                         CDH17        \n",
       "3 staging, survival, recurrence                         PCDH10       \n",
       "4 staging, survival, recurrence                         PCDH8        \n",
       "5 staging, survival, recurrence                         RARB         \n",
       "6 hypomethylation, assoc. with recurrence free survival MEG3         \n",
       "  is.promoter\n",
       "1 TRUE       \n",
       "2 TRUE       \n",
       "3 TRUE       \n",
       "4 TRUE       \n",
       "5 TRUE       \n",
       "6 TRUE       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_regions3 = df_regions2 %>% left_join(df_prom %>% select(V1,V2,V3,V6,V11,V15), by=c(\"hg19_chr\"=\"V1\",\n",
    "                                                                                      \"hg19_start\"=\"V2\",\n",
    "                                                                                      \"hg19_end\"=\"V3\",\n",
    "                                                                                      \"source\"=\"V6\")) %>%\n",
    "  rename(\"promoter_name\"=\"V11\",\"is.promoter\"=\"V15\")\n",
    "df_regions3$is.promoter = ifelse(df_regions3$is.promoter>0,TRUE,FALSE)\n",
    "head(df_regions3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "c59c76a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'V1', 'V2', 'V3'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>source</th><th scope=col>association</th><th scope=col>promoter_name</th><th scope=col>is.promoter</th><th scope=col>bp.tcga.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>11</td><td> 67350213</td><td> 67352213</td><td>GSTP1 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>GSTP1 </td><td>TRUE</td><td>1065</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>8 </td><td> 95138390</td><td> 95140390</td><td>CDH17 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>CDH17 </td><td>TRUE</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>4 </td><td>134069449</td><td>134071449</td><td>PCDH10</td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH10</td><td>TRUE</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>13</td><td> 53416024</td><td> 53418024</td><td>PCDH8 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH8 </td><td>TRUE</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>3 </td><td> 24728378</td><td> 24730378</td><td>RARB  </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>RARB  </td><td>TRUE</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>14</td><td>101244747</td><td>101246747</td><td>MEG3  </td><td></td><td>Cai et al. 2021       </td><td>hypomethylation, assoc. with recurrence free survival</td><td>MEG3  </td><td>TRUE</td><td>   0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & hg19\\_chr & hg19\\_start & hg19\\_end & gene & probe & source & association & promoter\\_name & is.promoter & bp.tcga.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 11 &  67350213 &  67352213 & GSTP1  &  & Constancio et al. 2019 & staging, survival, recurrence                         & GSTP1  & TRUE & 1065\\\\\n",
       "\t2 & 8  &  95138390 &  95140390 & CDH17  &  & Constancio et al. 2019 & staging, survival, recurrence                         & CDH17  & TRUE &    0\\\\\n",
       "\t3 & 4  & 134069449 & 134071449 & PCDH10 &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH10 & TRUE &    0\\\\\n",
       "\t4 & 13 &  53416024 &  53418024 & PCDH8  &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH8  & TRUE &    0\\\\\n",
       "\t5 & 3  &  24728378 &  24730378 & RARB   &  & Constancio et al. 2019 & staging, survival, recurrence                         & RARB   & TRUE &    0\\\\\n",
       "\t6 & 14 & 101244747 & 101246747 & MEG3   &  & Cai et al. 2021        & hypomethylation, assoc. with recurrence free survival & MEG3   & TRUE &    0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | hg19_chr &lt;chr&gt; | hg19_start &lt;int&gt; | hg19_end &lt;int&gt; | gene &lt;chr&gt; | probe &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | promoter_name &lt;chr&gt; | is.promoter &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 11 |  67350213 |  67352213 | GSTP1  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | GSTP1  | TRUE | 1065 |\n",
       "| 2 | 8  |  95138390 |  95140390 | CDH17  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | CDH17  | TRUE |    0 |\n",
       "| 3 | 4  | 134069449 | 134071449 | PCDH10 | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH10 | TRUE |    0 |\n",
       "| 4 | 13 |  53416024 |  53418024 | PCDH8  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH8  | TRUE |    0 |\n",
       "| 5 | 3  |  24728378 |  24730378 | RARB   | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | RARB   | TRUE |    0 |\n",
       "| 6 | 14 | 101244747 | 101246747 | MEG3   | <!----> | Cai et al. 2021        | hypomethylation, assoc. with recurrence free survival | MEG3   | TRUE |    0 |\n",
       "\n"
      ],
      "text/plain": [
       "  hg19_chr hg19_start hg19_end  gene   probe source                \n",
       "1 11        67350213   67352213 GSTP1        Constancio et al. 2019\n",
       "2 8         95138390   95140390 CDH17        Constancio et al. 2019\n",
       "3 4        134069449  134071449 PCDH10       Constancio et al. 2019\n",
       "4 13        53416024   53418024 PCDH8        Constancio et al. 2019\n",
       "5 3         24728378   24730378 RARB         Constancio et al. 2019\n",
       "6 14       101244747  101246747 MEG3         Cai et al. 2021       \n",
       "  association                                           promoter_name\n",
       "1 staging, survival, recurrence                         GSTP1        \n",
       "2 staging, survival, recurrence                         CDH17        \n",
       "3 staging, survival, recurrence                         PCDH10       \n",
       "4 staging, survival, recurrence                         PCDH8        \n",
       "5 staging, survival, recurrence                         RARB         \n",
       "6 hypomethylation, assoc. with recurrence free survival MEG3         \n",
       "  is.promoter bp.tcga.overlap\n",
       "1 TRUE        1065           \n",
       "2 TRUE           0           \n",
       "3 TRUE           0           \n",
       "4 TRUE           0           \n",
       "5 TRUE           0           \n",
       "6 TRUE           0           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_tcga2 = df_tcga %>% group_by(V1,V2,V3,V6) %>% summarize(bp.tcga.overlap=sum(V11))\n",
    "\n",
    "df_regions4 = df_regions3 %>% left_join(df_tcga2, by=c(\"hg19_chr\"=\"V1\",\"hg19_start\"=\"V2\",\"hg19_end\"=\"V3\",\"source\"=\"V6\"))\n",
    "\n",
    "head(df_regions4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "f4130547",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'V1', 'V2', 'V3'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>source</th><th scope=col>association</th><th scope=col>promoter_name</th><th scope=col>is.promoter</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>11</td><td> 67350213</td><td> 67352213</td><td>GSTP1 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>GSTP1 </td><td>TRUE</td><td>1065</td><td>695</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>8 </td><td> 95138390</td><td> 95140390</td><td>CDH17 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>CDH17 </td><td>TRUE</td><td>   0</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>4 </td><td>134069449</td><td>134071449</td><td>PCDH10</td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH10</td><td>TRUE</td><td>   0</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>13</td><td> 53416024</td><td> 53418024</td><td>PCDH8 </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>PCDH8 </td><td>TRUE</td><td>   0</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>3 </td><td> 24728378</td><td> 24730378</td><td>RARB  </td><td></td><td>Constancio et al. 2019</td><td>staging, survival, recurrence                        </td><td>RARB  </td><td>TRUE</td><td>   0</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>14</td><td>101244747</td><td>101246747</td><td>MEG3  </td><td></td><td>Cai et al. 2021       </td><td>hypomethylation, assoc. with recurrence free survival</td><td>MEG3  </td><td>TRUE</td><td>   0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & hg19\\_chr & hg19\\_start & hg19\\_end & gene & probe & source & association & promoter\\_name & is.promoter & bp.tcga.overlap & bp.ecm.overlap\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 11 &  67350213 &  67352213 & GSTP1  &  & Constancio et al. 2019 & staging, survival, recurrence                         & GSTP1  & TRUE & 1065 & 695\\\\\n",
       "\t2 & 8  &  95138390 &  95140390 & CDH17  &  & Constancio et al. 2019 & staging, survival, recurrence                         & CDH17  & TRUE &    0 &   0\\\\\n",
       "\t3 & 4  & 134069449 & 134071449 & PCDH10 &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH10 & TRUE &    0 &   0\\\\\n",
       "\t4 & 13 &  53416024 &  53418024 & PCDH8  &  & Constancio et al. 2019 & staging, survival, recurrence                         & PCDH8  & TRUE &    0 &   0\\\\\n",
       "\t5 & 3  &  24728378 &  24730378 & RARB   &  & Constancio et al. 2019 & staging, survival, recurrence                         & RARB   & TRUE &    0 &   0\\\\\n",
       "\t6 & 14 & 101244747 & 101246747 & MEG3   &  & Cai et al. 2021        & hypomethylation, assoc. with recurrence free survival & MEG3   & TRUE &    0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | hg19_chr &lt;chr&gt; | hg19_start &lt;int&gt; | hg19_end &lt;int&gt; | gene &lt;chr&gt; | probe &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | promoter_name &lt;chr&gt; | is.promoter &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 11 |  67350213 |  67352213 | GSTP1  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | GSTP1  | TRUE | 1065 | 695 |\n",
       "| 2 | 8  |  95138390 |  95140390 | CDH17  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | CDH17  | TRUE |    0 |   0 |\n",
       "| 3 | 4  | 134069449 | 134071449 | PCDH10 | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH10 | TRUE |    0 |   0 |\n",
       "| 4 | 13 |  53416024 |  53418024 | PCDH8  | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | PCDH8  | TRUE |    0 |   0 |\n",
       "| 5 | 3  |  24728378 |  24730378 | RARB   | <!----> | Constancio et al. 2019 | staging, survival, recurrence                         | RARB   | TRUE |    0 |   0 |\n",
       "| 6 | 14 | 101244747 | 101246747 | MEG3   | <!----> | Cai et al. 2021        | hypomethylation, assoc. with recurrence free survival | MEG3   | TRUE |    0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  hg19_chr hg19_start hg19_end  gene   probe source                \n",
       "1 11        67350213   67352213 GSTP1        Constancio et al. 2019\n",
       "2 8         95138390   95140390 CDH17        Constancio et al. 2019\n",
       "3 4        134069449  134071449 PCDH10       Constancio et al. 2019\n",
       "4 13        53416024   53418024 PCDH8        Constancio et al. 2019\n",
       "5 3         24728378   24730378 RARB         Constancio et al. 2019\n",
       "6 14       101244747  101246747 MEG3         Cai et al. 2021       \n",
       "  association                                           promoter_name\n",
       "1 staging, survival, recurrence                         GSTP1        \n",
       "2 staging, survival, recurrence                         CDH17        \n",
       "3 staging, survival, recurrence                         PCDH10       \n",
       "4 staging, survival, recurrence                         PCDH8        \n",
       "5 staging, survival, recurrence                         RARB         \n",
       "6 hypomethylation, assoc. with recurrence free survival MEG3         \n",
       "  is.promoter bp.tcga.overlap bp.ecm.overlap\n",
       "1 TRUE        1065            695           \n",
       "2 TRUE           0              0           \n",
       "3 TRUE           0              0           \n",
       "4 TRUE           0              0           \n",
       "5 TRUE           0              0           \n",
       "6 TRUE           0              0           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "df_ecm2 = df_ecm %>% group_by(V1,V2,V3,V6) %>% summarize(bp.ecm.overlap=sum(V28))\n",
    "#head(df_ecm2)\n",
    "df_regions5 = df_regions4 %>% left_join(df_ecm2, by=c(\"hg19_chr\"=\"V1\",\"hg19_start\"=\"V2\",\"hg19_end\"=\"V3\",\"source\"=\"V6\"))\n",
    "head(df_regions5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "7d737099",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_regions5$hybrid.gene.name = ifelse(df_regions5$gene==\"\",df_regions5$promoter_name,df_regions5$gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "297ffda7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>hg19_chr</th><th scope=col>hg19_start</th><th scope=col>hg19_end</th><th scope=col>gene</th><th scope=col>probe</th><th scope=col>source</th><th scope=col>association</th><th scope=col>promoter_name</th><th scope=col>is.promoter</th><th scope=col>bp.tcga.overlap</th><th scope=col>bp.ecm.overlap</th><th scope=col>hybrid.gene.name</th><th scope=col>no.pubs</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " hg19\\_chr & hg19\\_start & hg19\\_end & gene & probe & source & association & promoter\\_name & is.promoter & bp.tcga.overlap & bp.ecm.overlap & hybrid.gene.name & no.pubs\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 13\n",
       "\n",
       "| hg19_chr &lt;chr&gt; | hg19_start &lt;int&gt; | hg19_end &lt;int&gt; | gene &lt;chr&gt; | probe &lt;chr&gt; | source &lt;chr&gt; | association &lt;chr&gt; | promoter_name &lt;chr&gt; | is.promoter &lt;lgl&gt; | bp.tcga.overlap &lt;int&gt; | bp.ecm.overlap &lt;int&gt; | hybrid.gene.name &lt;chr&gt; | no.pubs &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     hg19_chr hg19_start hg19_end gene probe source association promoter_name\n",
       "     is.promoter bp.tcga.overlap bp.ecm.overlap hybrid.gene.name no.pubs"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_pub_count = df_regions5 %>% group_by(hybrid.gene.name) %>% summarize(no.pubs=length(unique(source)))\n",
    "\n",
    "df_regions6 = df_regions5 %>% left_join(df_pub_count, by=\"hybrid.gene.name\")\n",
    "df_regions6[!complete.cases(df_regions6),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "1cbad80d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "  1   2 \n",
       "674   4 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(df_regions6$no.pubs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "b94623d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(df_regions6, file=\"./marker_collation/PRAD_disease_progression_methyl_loci_annotated_06032022.tsv\",\n",
    "           row.names=FALSE, col.names=TRUE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e675e354",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
