{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "479a5995",
   "metadata": {},
   "source": [
    "# Moffit Data Delivery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d047306a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.2     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.16.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite either one:\n",
      "- Gu, Z. Complex Heatmap Visualization. iMeta 2022.\n",
      "- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "    genomic data. Bioinformatics 2016.\n",
      "\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "Type 'citation(\"pROC\")' for a citation.\n",
      "\n",
      "\n",
      "Attaching package: ‘pROC’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    cov, smooth, var\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2023-11-08 11:40:53 PST\"\n",
      "R version 4.3.0 (2023-04-21)\n",
      "Platform: aarch64-apple-darwin20 (64-bit)\n",
      "Running under: macOS Ventura 13.5.1\n",
      "\n",
      "Matrix products: default\n",
      "BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib \n",
      "LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0\n",
      "\n",
      "locale:\n",
      "[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8\n",
      "\n",
      "time zone: America/Los_Angeles\n",
      "tzcode source: internal\n",
      "\n",
      "attached base packages:\n",
      "[1] grid      stats     graphics  grDevices utils     datasets  methods  \n",
      "[8] base     \n",
      "\n",
      "other attached packages:\n",
      " [1] ggsignif_0.6.4        glue_1.6.2            pROC_1.18.4          \n",
      " [4] readxl_1.4.3          RColorBrewer_1.1-3    maftools_2.16.0      \n",
      " [7] ComplexHeatmap_2.16.0 lubridate_1.9.2       forcats_1.0.0        \n",
      "[10] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.1          \n",
      "[13] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         \n",
      "[16] tidyverse_2.0.0       ggplot2_3.4.2        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] gtable_0.3.3        circlize_0.4.15     shape_1.4.6        \n",
      " [4] rjson_0.2.21        GlobalOptions_0.1.2 lattice_0.21-8     \n",
      " [7] tzdb_0.4.0          vctrs_0.6.3         tools_4.3.0        \n",
      "[10] generics_0.1.3      stats4_4.3.0        parallel_4.3.0     \n",
      "[13] fansi_1.0.4         cluster_2.1.4       pkgconfig_2.0.3    \n",
      "[16] Matrix_1.6-0        data.table_1.14.8   S4Vectors_0.38.1   \n",
      "[19] uuid_1.1-0          lifecycle_1.0.3     compiler_4.3.0     \n",
      "[22] munsell_0.5.0       repr_1.1.6          codetools_0.2-19   \n",
      "[25] clue_0.3-64         htmltools_0.5.5     pillar_1.9.0       \n",
      "[28] crayon_1.5.2        iterators_1.0.14    foreach_1.5.2      \n",
      "[31] tidyselect_1.2.0    digest_0.6.33       stringi_1.7.12     \n",
      "[34] splines_4.3.0       fastmap_1.1.1       colorspace_2.1-0   \n",
      "[37] cli_3.6.1           magrittr_2.0.3      base64enc_0.1-3    \n",
      "[40] survival_3.5-5      utf8_1.2.3          IRdisplay_1.1      \n",
      "[43] withr_2.5.0         scales_1.2.1        IRkernel_1.3.2     \n",
      "[46] timechange_0.2.0    matrixStats_1.0.0   pbdZMQ_0.3-9       \n",
      "[49] cellranger_1.1.0    png_0.1-8           GetoptLong_1.0.5   \n",
      "[52] hms_1.1.3           DNAcopy_1.74.1      evaluate_0.21      \n",
      "[55] IRanges_2.34.1      doParallel_1.0.17   rlang_1.1.1        \n",
      "[58] Rcpp_1.0.11         BiocGenerics_0.46.0 jsonlite_1.8.7     \n",
      "[61] plyr_1.8.8          R6_2.5.1           \n"
     ]
    }
   ],
   "source": [
    "library(ggplot2)\n",
    "library(tidyverse)\n",
    "library(ComplexHeatmap)\n",
    "library(maftools)\n",
    "library(RColorBrewer)\n",
    "library(readxl)\n",
    "library(stats)\n",
    "library(pROC)\n",
    "library(stringr)\n",
    "source(\"~/Desktop/puffin/R/helper_functions.R\")\n",
    "options(stringsAsFactors = FALSE)\n",
    "options(repr.matrix.max.cols=75, repr.matrix.max.rows=50)\n",
    "source(\"../fix_PIDs.R\")\n",
    "\n",
    "# set some defaults\n",
    "options(stringsAsFactors = FALSE)\n",
    "options(repr.matrix.max.cols=75, repr.matrix.max.rows=50)\n",
    "formals(table)$useNA <- \"always\"\n",
    "formals(write.csv)$row.names <- FALSE\n",
    "formals(write.csv)$as.is <- TRUE\n",
    "\n",
    "print(Sys.time())\n",
    "print(sessionInfo())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ac35d7f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"152237\" \"161922\" \"169558\" \"172530\" \"173254\" \"173912\" \"174091\" \"174872\"\n",
      " [9] \"175648\" \"175733\" \"175980\" \"176349\" \"176388\" \"180264\" \"182022\" \"116381\"\n",
      "[17] \"165028\" \"167162\" \"171374\" \"171769\" \"171821\" \"172153\" \"172229\" \"172586\"\n",
      "[25] \"172602\" \"172749\" \"172851\" \"173075\" \"173281\" \"173334\" \"173350\" \"173359\"\n",
      "[33] \"173362\" \"173509\" \"173975\" \"173983\" \"175610\" \"174083\" \"174947\" \"175325\"\n",
      "[41] \"175387\" \"175425\" \"175492\" \"175526\" \"175672\" \"175725\" \"175854\" \"176828\"\n",
      "[49] \"176992\" \"177967\" \"178886\" \"175831\" \"179692\" \"181439\" \"181568\" \"182435\"\n",
      "[57] \"182910\" \"184289\" \"184485\" \"184615\" \"184957\" \"185014\" \"185212\" \"185382\"\n",
      "[65] \"185581\" \"186478\" \"186652\" \"187204\" \"187807\" \"188867\" \"189083\" \"189342\"\n",
      "[73] \"190233\" \"191632\" \"191692\" \"192682\" \"193186\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'PatientID'</li><li>'has.IndexTURBT'</li><li>'Age'</li><li>'Sex'</li><li>'Race'</li><li>'Smoking.Status'</li><li>'ECOG.scoreAt.Repeat.TURBT'</li><li>'Previous.pelvic.XRT'</li><li>'Previous.history.of.NMIBC'</li><li>'Progression'</li><li>'Previous.Intravesical.Therapy..choice.None.'</li><li>'Previous.Intravesical.Therapy..choice.BCG.'</li><li>'Previous.Intravesical.Therapy..choice.Gemcitabine.....Docetaxel.'</li><li>'Previous.Intravesical.Therapy..choice.Mitomycin.C.'</li><li>'Other.Intravesical.Therapy'</li><li>'Prior.Treatment.Free.Text..1.2.sentence.summary.of.treatment.prior.to.Index.TURBT.inclusive.of.NMIBC.Pathology.'</li><li>'Index.TURBT.Date'</li><li>'IndexHistology'</li><li>'Variant.Histology'</li><li>'Index.Stage'</li><li>'IndexGrade'</li><li>'Index.CIS'</li><li>'LVI...35'</li><li>'Immediate.Post.Op.Intravesical.Therapy.Index'</li><li>'Repeat.TURBT.Date'</li><li>'RepeatHistology'</li><li>'Variant.Histology.Repeat'</li><li>'Repeat.Stage'</li><li>'RepeatGrade'</li><li>'Repeat.CIS'</li><li>'LVI...43'</li><li>'Immediate.Post.Op.Intravesical.Therapy.Repeat'</li><li>'Induction.Intravesical.Therapy.Administered.'</li><li>'Induction.Intravesical.Therapy.Type.'</li><li>'Induction.Cycles'</li><li>'Maintenance.Intravesical.Therapy.'</li><li>'Maintenance.Intravesical.Therapy.Type.'</li><li>'Maintenance.Duration..Pick.Closest.Time.Point.'</li><li>'Complete.'</li><li>'Clinical.Recurrence.'</li><li>'CytoscopyDate'</li><li>'Date.of.Subsequent.TURBT'</li><li>'TURBT.Recurrence.Pathology'</li><li>'disease.positive'</li><li>'PatientID.old'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'PatientID'\n",
       "\\item 'has.IndexTURBT'\n",
       "\\item 'Age'\n",
       "\\item 'Sex'\n",
       "\\item 'Race'\n",
       "\\item 'Smoking.Status'\n",
       "\\item 'ECOG.scoreAt.Repeat.TURBT'\n",
       "\\item 'Previous.pelvic.XRT'\n",
       "\\item 'Previous.history.of.NMIBC'\n",
       "\\item 'Progression'\n",
       "\\item 'Previous.Intravesical.Therapy..choice.None.'\n",
       "\\item 'Previous.Intravesical.Therapy..choice.BCG.'\n",
       "\\item 'Previous.Intravesical.Therapy..choice.Gemcitabine.....Docetaxel.'\n",
       "\\item 'Previous.Intravesical.Therapy..choice.Mitomycin.C.'\n",
       "\\item 'Other.Intravesical.Therapy'\n",
       "\\item 'Prior.Treatment.Free.Text..1.2.sentence.summary.of.treatment.prior.to.Index.TURBT.inclusive.of.NMIBC.Pathology.'\n",
       "\\item 'Index.TURBT.Date'\n",
       "\\item 'IndexHistology'\n",
       "\\item 'Variant.Histology'\n",
       "\\item 'Index.Stage'\n",
       "\\item 'IndexGrade'\n",
       "\\item 'Index.CIS'\n",
       "\\item 'LVI...35'\n",
       "\\item 'Immediate.Post.Op.Intravesical.Therapy.Index'\n",
       "\\item 'Repeat.TURBT.Date'\n",
       "\\item 'RepeatHistology'\n",
       "\\item 'Variant.Histology.Repeat'\n",
       "\\item 'Repeat.Stage'\n",
       "\\item 'RepeatGrade'\n",
       "\\item 'Repeat.CIS'\n",
       "\\item 'LVI...43'\n",
       "\\item 'Immediate.Post.Op.Intravesical.Therapy.Repeat'\n",
       "\\item 'Induction.Intravesical.Therapy.Administered.'\n",
       "\\item 'Induction.Intravesical.Therapy.Type.'\n",
       "\\item 'Induction.Cycles'\n",
       "\\item 'Maintenance.Intravesical.Therapy.'\n",
       "\\item 'Maintenance.Intravesical.Therapy.Type.'\n",
       "\\item 'Maintenance.Duration..Pick.Closest.Time.Point.'\n",
       "\\item 'Complete.'\n",
       "\\item 'Clinical.Recurrence.'\n",
       "\\item 'CytoscopyDate'\n",
       "\\item 'Date.of.Subsequent.TURBT'\n",
       "\\item 'TURBT.Recurrence.Pathology'\n",
       "\\item 'disease.positive'\n",
       "\\item 'PatientID.old'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'PatientID'\n",
       "2. 'has.IndexTURBT'\n",
       "3. 'Age'\n",
       "4. 'Sex'\n",
       "5. 'Race'\n",
       "6. 'Smoking.Status'\n",
       "7. 'ECOG.scoreAt.Repeat.TURBT'\n",
       "8. 'Previous.pelvic.XRT'\n",
       "9. 'Previous.history.of.NMIBC'\n",
       "10. 'Progression'\n",
       "11. 'Previous.Intravesical.Therapy..choice.None.'\n",
       "12. 'Previous.Intravesical.Therapy..choice.BCG.'\n",
       "13. 'Previous.Intravesical.Therapy..choice.Gemcitabine.....Docetaxel.'\n",
       "14. 'Previous.Intravesical.Therapy..choice.Mitomycin.C.'\n",
       "15. 'Other.Intravesical.Therapy'\n",
       "16. 'Prior.Treatment.Free.Text..1.2.sentence.summary.of.treatment.prior.to.Index.TURBT.inclusive.of.NMIBC.Pathology.'\n",
       "17. 'Index.TURBT.Date'\n",
       "18. 'IndexHistology'\n",
       "19. 'Variant.Histology'\n",
       "20. 'Index.Stage'\n",
       "21. 'IndexGrade'\n",
       "22. 'Index.CIS'\n",
       "23. 'LVI...35'\n",
       "24. 'Immediate.Post.Op.Intravesical.Therapy.Index'\n",
       "25. 'Repeat.TURBT.Date'\n",
       "26. 'RepeatHistology'\n",
       "27. 'Variant.Histology.Repeat'\n",
       "28. 'Repeat.Stage'\n",
       "29. 'RepeatGrade'\n",
       "30. 'Repeat.CIS'\n",
       "31. 'LVI...43'\n",
       "32. 'Immediate.Post.Op.Intravesical.Therapy.Repeat'\n",
       "33. 'Induction.Intravesical.Therapy.Administered.'\n",
       "34. 'Induction.Intravesical.Therapy.Type.'\n",
       "35. 'Induction.Cycles'\n",
       "36. 'Maintenance.Intravesical.Therapy.'\n",
       "37. 'Maintenance.Intravesical.Therapy.Type.'\n",
       "38. 'Maintenance.Duration..Pick.Closest.Time.Point.'\n",
       "39. 'Complete.'\n",
       "40. 'Clinical.Recurrence.'\n",
       "41. 'CytoscopyDate'\n",
       "42. 'Date.of.Subsequent.TURBT'\n",
       "43. 'TURBT.Recurrence.Pathology'\n",
       "44. 'disease.positive'\n",
       "45. 'PatientID.old'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"PatientID\"                                                                                                      \n",
       " [2] \"has.IndexTURBT\"                                                                                                 \n",
       " [3] \"Age\"                                                                                                            \n",
       " [4] \"Sex\"                                                                                                            \n",
       " [5] \"Race\"                                                                                                           \n",
       " [6] \"Smoking.Status\"                                                                                                 \n",
       " [7] \"ECOG.scoreAt.Repeat.TURBT\"                                                                                      \n",
       " [8] \"Previous.pelvic.XRT\"                                                                                            \n",
       " [9] \"Previous.history.of.NMIBC\"                                                                                      \n",
       "[10] \"Progression\"                                                                                                    \n",
       "[11] \"Previous.Intravesical.Therapy..choice.None.\"                                                                    \n",
       "[12] \"Previous.Intravesical.Therapy..choice.BCG.\"                                                                     \n",
       "[13] \"Previous.Intravesical.Therapy..choice.Gemcitabine.....Docetaxel.\"                                               \n",
       "[14] \"Previous.Intravesical.Therapy..choice.Mitomycin.C.\"                                                             \n",
       "[15] \"Other.Intravesical.Therapy\"                                                                                     \n",
       "[16] \"Prior.Treatment.Free.Text..1.2.sentence.summary.of.treatment.prior.to.Index.TURBT.inclusive.of.NMIBC.Pathology.\"\n",
       "[17] \"Index.TURBT.Date\"                                                                                               \n",
       "[18] \"IndexHistology\"                                                                                                 \n",
       "[19] \"Variant.Histology\"                                                                                              \n",
       "[20] \"Index.Stage\"                                                                                                    \n",
       "[21] \"IndexGrade\"                                                                                                     \n",
       "[22] \"Index.CIS\"                                                                                                      \n",
       "[23] \"LVI...35\"                                                                                                       \n",
       "[24] \"Immediate.Post.Op.Intravesical.Therapy.Index\"                                                                   \n",
       "[25] \"Repeat.TURBT.Date\"                                                                                              \n",
       "[26] \"RepeatHistology\"                                                                                                \n",
       "[27] \"Variant.Histology.Repeat\"                                                                                       \n",
       "[28] \"Repeat.Stage\"                                                                                                   \n",
       "[29] \"RepeatGrade\"                                                                                                    \n",
       "[30] \"Repeat.CIS\"                                                                                                     \n",
       "[31] \"LVI...43\"                                                                                                       \n",
       "[32] \"Immediate.Post.Op.Intravesical.Therapy.Repeat\"                                                                  \n",
       "[33] \"Induction.Intravesical.Therapy.Administered.\"                                                                   \n",
       "[34] \"Induction.Intravesical.Therapy.Type.\"                                                                           \n",
       "[35] \"Induction.Cycles\"                                                                                               \n",
       "[36] \"Maintenance.Intravesical.Therapy.\"                                                                              \n",
       "[37] \"Maintenance.Intravesical.Therapy.Type.\"                                                                         \n",
       "[38] \"Maintenance.Duration..Pick.Closest.Time.Point.\"                                                                 \n",
       "[39] \"Complete.\"                                                                                                      \n",
       "[40] \"Clinical.Recurrence.\"                                                                                           \n",
       "[41] \"CytoscopyDate\"                                                                                                  \n",
       "[42] \"Date.of.Subsequent.TURBT\"                                                                                       \n",
       "[43] \"TURBT.Recurrence.Pathology\"                                                                                     \n",
       "[44] \"disease.positive\"                                                                                               \n",
       "[45] \"PatientID.old\"                                                                                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE  <NA> \n",
       "   15    35    27 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read in clinical data\n",
    "\n",
    "## functions for harmonizing input sheets\n",
    "source(\"../fix_PIDs.R\")\n",
    "\n",
    "fix_sampleTypes <- function(col){\n",
    "    #print(table(col, useNA=\"always\"))\n",
    "    col.new <- gsub(\"FFPE\", \"Tissue\", col)\n",
    "    col.new <- gsub(\"Urine_Supernatant|Supernatant\", \"Urine\", col.new)\n",
    "    col.new <- gsub(\"Buffy Coat|PBMC\", \"Buffy_Coat\", col.new)\n",
    "    print(table(col.new, useNA=\"always\"))\n",
    "    return(col.new)\n",
    "}\n",
    "fix_StudyVisit <- function(col){\n",
    "    warn_na(col)\n",
    "    col.new <- sapply(str_split(col, pattern=\"_\"), \"[\", 2)\n",
    "    return(col.new)\n",
    "}\n",
    "\n",
    "\n",
    "df_clin = read.csv(\"./clinical_data/NMIBC_clinical_data_merged_2023-10-11.csv\", as.is=T) %>%\n",
    "                mutate(PatientID=as.character(PatientID))\n",
    "stopifnot(all(!duplicated(df_clin$PatientID)))\n",
    "print(df_clin$PatientID)\n",
    "names(df_clin)\n",
    "table(df_clin$disease.positive) # no clin data on validation cohort samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "10eb10ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SampleID.short format: remove.suffix\n",
      " [1] \"116381\" \"152237\" \"161922\" \"165028\" \"167162\" \"169558\" \"171374\" \"171769\"\n",
      " [9] \"171821\" \"172153\" \"172229\" \"172530\" \"172586\" \"172602\" \"172749\" \"172851\"\n",
      "[17] \"173075\" \"173254\" \"173281\" \"173334\" \"173350\" \"173359\" \"173362\" \"173509\"\n",
      "[25] \"173912\" \"173975\" \"173983\" \"174083\" \"174091\" \"174872\" \"174947\" \"175325\"\n",
      "[33] \"175387\" \"175425\" \"175492\" \"175526\" \"175610\" \"175648\" \"175672\" \"175725\"\n",
      "[41] \"175733\" \"175831\" \"175854\" \"175980\" \"176349\" \"176388\" \"176828\" \"176992\"\n",
      "[49] \"177967\" \"178886\" \"179692\" \"180264\" \"181439\" \"181568\" \"182022\" \"182435\"\n",
      "[57] \"182910\" \"184289\" \"184485\" \"184615\" \"184957\" \"185014\" \"185212\" \"185382\"\n",
      "[65] \"185581\" \"186478\" \"186652\" \"187204\" \"187807\" \"188867\" \"189083\" \"189342\"\n",
      "[73] \"190233\" \"191632\" \"191692\" \"192682\" \"193186\"\n",
      "col.new\n",
      "        Buffycoat            Plasma            Tissue             Urine \n",
      "               73                73               129               142 \n",
      "Urine Cell Pellet              <NA> \n",
      "              141                 0 \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                   \n",
       "                    BenignUrothelium FollowUp1 FollowUp2 FollowUp3 Followup4\n",
       "  Buffycoat                        0         0         0         0         0\n",
       "  Plasma                           0         0         0         0         0\n",
       "  Tissue                           9         0         0         0         0\n",
       "  Urine                            0        18         9         4         1\n",
       "  Urine Cell Pellet                0        18         9         4         1\n",
       "  <NA>                             0         0         0         0         0\n",
       "                   \n",
       "                    IndexTURBT PostTURBT Pre-RepeatTURBT RepeatTURBT\n",
       "  Buffycoat                  0         0              73           0\n",
       "  Plasma                     0         0              73           0\n",
       "  Tissue                    72         0               0          48\n",
       "  Urine                      0        30              77           0\n",
       "  Urine Cell Pellet          0        29              77           0\n",
       "  <NA>                       0         0               0           0\n",
       "                   \n",
       "                    Surveillance1 <NA>\n",
       "  Buffycoat                     0    0\n",
       "  Plasma                        0    0\n",
       "  Tissue                        0    0\n",
       "  Urine                         3    0\n",
       "  Urine Cell Pellet             3    0\n",
       "  <NA>                          0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "558"
      ],
      "text/latex": [
       "558"
      ],
      "text/markdown": [
       "558"
      ],
      "text/plain": [
       "[1] 558"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Master sample list from manifests\n",
    "df_samples_all = make_names(read_excel(\"Manifest_data_all_10112023.xlsx\", sheet=\"Sample information real time\")) %>%\n",
    "                    mutate(SampleID.short = make_SIDshort(SampleID, \"remove.suffix\"),\n",
    "                           PatientID = as.character(fix_PIDs(PatientID)),\n",
    "                      StudyVisit2 = fix_StudyVisit(StudyVisit),\n",
    "                      StudyVisit = gsub(\" \",\"\", ifelse(!is.na(StudyVisit2), StudyVisit2, StudyVisit)),\n",
    "                      SampleType = fix_sampleTypes(SampleType))\n",
    "table(df_samples_all$SampleType, df_samples_all$StudyVisit, useNA=\"always\")\n",
    "nrow(df_samples_all) #558\n",
    "stopifnot(all(!(duplicated(df_samples_all$SampleID.short))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f90a84f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SampleID.short format: remove.suffix\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Standardize.names WARN:Fill all with NA for SampleType\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"116381\" \"152237\" \"161922\" \"165028\" \"167162\" \"169558\" \"171374\" \"171769\"\n",
      " [9] \"171821\" \"172153\" \"172229\" \"172530\" \"172586\" \"172602\" \"172749\" \"172851\"\n",
      "[17] \"173281\" \"173334\" \"173350\" \"173362\" \"173509\" \"173912\" \"173975\" \"173983\"\n",
      "[25] \"174083\" \"174091\" \"174872\" \"174947\" \"175325\" \"175387\" \"175425\" \"175492\"\n",
      "[33] \"175526\" \"175610\" \"175648\" \"175672\" \"175725\" \"175733\" \"175854\" \"175980\"\n",
      "[41] \"176349\" \"176388\" \"176828\" \"176992\" \"177967\" \"178886\" \"180264\" \"182022\"\n",
      "[49] \"173359\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SubjectID'</li><li>'StudyVisit'</li><li>'ExternalID'</li><li>'Sample.Name'</li><li>'total_abnormal_regions_per_sample'</li><li>'total_abnormal_frags_per_sample'</li><li>'filtered_total_abnormal_regions_per_sample'</li><li>'filtered_total_abnormal_frags_per_sample'</li><li>'SampleID'</li><li>'SampleID.short'</li><li>'PatientID'</li><li>'SampleType'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SubjectID'\n",
       "\\item 'StudyVisit'\n",
       "\\item 'ExternalID'\n",
       "\\item 'Sample.Name'\n",
       "\\item 'total\\_abnormal\\_regions\\_per\\_sample'\n",
       "\\item 'total\\_abnormal\\_frags\\_per\\_sample'\n",
       "\\item 'filtered\\_total\\_abnormal\\_regions\\_per\\_sample'\n",
       "\\item 'filtered\\_total\\_abnormal\\_frags\\_per\\_sample'\n",
       "\\item 'SampleID'\n",
       "\\item 'SampleID.short'\n",
       "\\item 'PatientID'\n",
       "\\item 'SampleType'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SubjectID'\n",
       "2. 'StudyVisit'\n",
       "3. 'ExternalID'\n",
       "4. 'Sample.Name'\n",
       "5. 'total_abnormal_regions_per_sample'\n",
       "6. 'total_abnormal_frags_per_sample'\n",
       "7. 'filtered_total_abnormal_regions_per_sample'\n",
       "8. 'filtered_total_abnormal_frags_per_sample'\n",
       "9. 'SampleID'\n",
       "10. 'SampleID.short'\n",
       "11. 'PatientID'\n",
       "12. 'SampleType'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SubjectID\"                                 \n",
       " [2] \"StudyVisit\"                                \n",
       " [3] \"ExternalID\"                                \n",
       " [4] \"Sample.Name\"                               \n",
       " [5] \"total_abnormal_regions_per_sample\"         \n",
       " [6] \"total_abnormal_frags_per_sample\"           \n",
       " [7] \"filtered_total_abnormal_regions_per_sample\"\n",
       " [8] \"filtered_total_abnormal_frags_per_sample\"  \n",
       " [9] \"SampleID\"                                  \n",
       "[10] \"SampleID.short\"                            \n",
       "[11] \"PatientID\"                                 \n",
       "[12] \"SampleType\"                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# methylation data\n",
    "\n",
    "methyl.data = read.csv(\"./Methylation_data/Nov_6_2023/PRDC-MOFFITT-NMIBC-22001_165_samples_Methylation_summary_info_11082023.csv\") # missing batch 3 and 4 urine samples\n",
    "methyl.data = standardize_names(methyl.data %>% mutate(SampleID=Sample.Name) %>% slice_head(n=165),\n",
    "                                sid.format = \"remove.suffix\", input.type=\"samples\") %>%\n",
    "                select(-SampleType) %>%\n",
    "                mutate(PatientID=fix_PIDs(SubjectID)) %>%\n",
    "                left_join(df_samples_all %>% select(SampleID.short, SampleType), by=\"SampleID.short\")\n",
    "names(methyl.data)\n",
    "#write.csv(methyl.data, \"./Methylation_data/Nov_6_2023/region_summary_for_client_formatted.csv\", row.names=F)\n",
    "write.csv(methyl.data, glue(\"./data_delivery/Methylation/region_summary_for_client_formatted_{Sys.Date()}.csv\"), row.names=F)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e7a315c",
   "metadata": {},
   "source": [
    "## WES+ data merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "44a0e3d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 72</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SampleID</th><th scope=col>Total.reads..M.</th><th scope=col>Mapped.reads..M.</th><th scope=col>Mapping.rate....</th><th scope=col>Consensus.reads..M.</th><th scope=col>Duplication.rate....</th><th scope=col>Target.unique.fragment.coverage</th><th scope=col>Panel2.Target.unique.fragment.coverage</th><th scope=col>Target.read.coverage</th><th scope=col>Mean.target.read.coverage</th><th scope=col>Panel2.Target.read.coverage</th><th scope=col>Percentage.of.target...20..Mean.coverage</th><th scope=col>Within.2.fold.....after.dedup</th><th scope=col>Within.2.fold.....before.dedup</th><th scope=col>Panel2.Within.2.fold.....after.dedup</th><th scope=col>Panel2.Within.2.fold.....before.dedup</th><th scope=col>onTarget.rate....</th><th scope=col>onTarget.rate.ext200....</th><th scope=col>gcRatio.low</th><th scope=col>gcRatio.high</th><th scope=col>Mean.error.rate.....after.dedup</th><th scope=col>background.50..after.dedup</th><th scope=col>background.99..after.dedup</th><th scope=col>Fragment_Size_Mode</th><th scope=col>Fragment_Size_Bandwidth</th><th scope=col>Softclip.ratio</th><th scope=col>Double.strand.ratio</th><th scope=col>BC_totalCount</th><th scope=col>SNP_BC</th><th scope=col>tmsi_score</th><th scope=col>msi_status</th><th scope=col>msi_QC</th><th scope=col>Specimen_type</th><th scope=col>Gender</th><th scope=col>Panel</th><th scope=col>ProjectDir</th><th scope=col>analysis_type</th><th scope=col>lane</th><th scope=col>work_order</th><th scope=col>original_adapter</th><th scope=col>total_dna_input</th><th scope=col>total_cfdna_input</th><th scope=col>library_dna_input_ug</th><th scope=col>enriched_library_fa_or_ba_nm</th><th scope=col>qubit_ng_ul</th><th scope=col>pversion</th><th scope=col>category</th><th scope=col>sampleFolder</th><th scope=col>PatientID</th><th scope=col>externalSampleID</th><th scope=col>trialVisitNum</th><th scope=col>Duplicated</th><th scope=col>Coverage...160....</th><th scope=col>Coverage...80....</th><th scope=col>Coverage...40....</th><th scope=col>Coverage...400....</th><th scope=col>Coverage...200....</th><th scope=col>Coverage...100....</th><th scope=col>QC_Status</th><th scope=col>Action</th><th scope=col>Load</th><th scope=col>SNV_Num</th><th scope=col>Indel_Num</th><th scope=col>pTMB</th><th scope=col>pTMB_norm</th><th scope=col>pTMB.weighted</th><th scope=col>pTMB_norm.weighted</th><th scope=col>pTMB.adj</th><th scope=col>pTMB_norm.adj</th><th scope=col>maxAF_TMB</th><th scope=col>tumorFraction</th><th scope=col>tumorStatus</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>P220085</td><td>260</td><td>260</td><td>99.9</td><td>127</td><td>51.0</td><td>1795</td><td>200</td><td>3200</td><td>3714</td><td>341</td><td>99.8</td><td>53.1</td><td>52.3</td><td>58.8</td><td>59.6</td><td>65.9</td><td>89.4</td><td>0.757</td><td>1.80</td><td>0.0220</td><td>0.382</td><td>2.19</td><td>183.3</td><td>135</td><td>1.63</td><td>28.9</td><td>NA</td><td>'00011101101220010020101110102022021'</td><td>0</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>female</td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>2</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC56_v2</td><td>246.25</td><td>227.83</td><td>2</td><td>30.2</td><td>8.76</td><td>1.7.0</td><td>RUO</td><td>P220085_220907</td><td>173983</td><td>06S22036059</td><td>IndexTURBT </td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>99.8</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>25</td><td>4</td><td>12</td><td> 8.9365</td><td>11.5410</td><td> 8.5947</td><td>12</td><td> 8.9365</td><td>21.0</td><td>0.348</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>P220076</td><td>257</td><td>256</td><td>99.8</td><td>135</td><td>47.3</td><td>1840</td><td>198</td><td>3150</td><td>3714</td><td>317</td><td>99.2</td><td>51.5</td><td>50.4</td><td>57.5</td><td>58.4</td><td>64.6</td><td>89.0</td><td>0.819</td><td>1.99</td><td>0.0394</td><td>0.410</td><td>2.44</td><td>193.7</td><td>151</td><td>1.72</td><td>25.6</td><td>NA</td><td>'21101000111021002011220001111021021'</td><td>2</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>male  </td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220902_A01832_0037_BHCKWTDSX5//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>3</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC24_v2</td><td>283.25</td><td>216.09</td><td>2</td><td>31.0</td><td>7.33</td><td>1.7.0</td><td>RUO</td><td>P220076_220902</td><td>173334</td><td>06S22068063</td><td>IndexTURBT </td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>99.4</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>88</td><td>0</td><td>91</td><td>67.7690</td><td>83.5810</td><td>62.2440</td><td>91</td><td>67.7690</td><td>59.8</td><td>0.748</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>P220070</td><td>251</td><td>251</td><td>99.8</td><td>128</td><td>49.1</td><td>1568</td><td>169</td><td>2730</td><td>3650</td><td>280</td><td>98.7</td><td>43.1</td><td>42.8</td><td>47.1</td><td>47.3</td><td>65.1</td><td>89.4</td><td>0.733</td><td>2.43</td><td>0.0262</td><td>0.345</td><td>2.33</td><td>188.5</td><td>145</td><td>1.74</td><td>32.2</td><td>NA</td><td>'11102001200000001010210010000112021'</td><td>0</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>male  </td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>1</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC46_v2</td><td>179.50</td><td>160.22</td><td>2</td><td>33.4</td><td>8.70</td><td>1.7.0</td><td>RUO</td><td>P220070_220907</td><td>172586</td><td>06S22052895</td><td>RepeatTURBT</td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>98.8</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>15</td><td>4</td><td> 5</td><td> 3.7236</td><td> 4.4055</td><td> 3.2808</td><td> 5</td><td> 3.7236</td><td>39.6</td><td>0.568</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>P220079</td><td>246</td><td>245</td><td>99.8</td><td>119</td><td>51.6</td><td>1821</td><td>206</td><td>3450</td><td>3491</td><td>355</td><td>99.9</td><td>69.0</td><td>66.2</td><td>73.6</td><td>73.3</td><td>66.0</td><td>89.2</td><td>0.837</td><td>1.18</td><td>0.0243</td><td>0.461</td><td>2.33</td><td>185.7</td><td>154</td><td>1.65</td><td>21.2</td><td>NA</td><td>'11102200200111201111211000001111111'</td><td>0</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>male  </td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>1</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC52_v2</td><td>184.50</td><td>136.75</td><td>2</td><td>14.3</td><td>4.62</td><td>1.7.0</td><td>RUO</td><td>P220079_220907</td><td>172530</td><td>06S22036103</td><td>IndexTURBT </td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>99.8</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>23</td><td>2</td><td>19</td><td>14.1500</td><td>17.1400</td><td>12.7640</td><td>19</td><td>14.1500</td><td>15.0</td><td>0.260</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>P220069</td><td>245</td><td>245</td><td>99.9</td><td>109</td><td>55.5</td><td>1349</td><td>180</td><td>3100</td><td>3535</td><td>339</td><td>99.8</td><td>59.2</td><td>56.5</td><td>61.5</td><td>61.0</td><td>68.7</td><td>89.4</td><td>0.935</td><td>1.61</td><td>0.0338</td><td>0.474</td><td>2.38</td><td>156.7</td><td>126</td><td>2.70</td><td>15.6</td><td>NA</td><td>'11102001200000001010210010000112021'</td><td>2</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>male  </td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>1</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC45_v2</td><td>279.00</td><td>211.04</td><td>2</td><td>13.2</td><td>3.72</td><td>1.7.0</td><td>RUO</td><td>P220069_220907</td><td>172586</td><td>06S22035975</td><td>IndexTURBT </td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>99.5</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>15</td><td>4</td><td> 4</td><td> 2.9788</td><td> 3.4055</td><td> 2.5361</td><td> 4</td><td> 2.9788</td><td>57.3</td><td>0.729</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>P220086</td><td>245</td><td>244</td><td>99.8</td><td>124</td><td>49.1</td><td>1160</td><td>134</td><td>2110</td><td>3322</td><td>226</td><td>99.1</td><td>49.3</td><td>48.7</td><td>50.8</td><td>50.6</td><td>63.4</td><td>87.6</td><td>0.842</td><td>3.43</td><td>0.0358</td><td>0.350</td><td>2.99</td><td>192.6</td><td>155</td><td>2.75</td><td>22.3</td><td>NA</td><td>'00011101101220010020101110102022021'</td><td>1</td><td>MSS</td><td>PASS</td><td>Tissue</td><td>female</td><td>PREDICINEATLAS_V2</td><td>/prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>tissue_analysis</td><td>2</td><td>WOP00249_PRDC-MOFFITT-NMIBC-22001</td><td>HBC57_v2</td><td>244.75</td><td>203.00</td><td>2</td><td>20.5</td><td>6.04</td><td>1.7.0</td><td>RUO</td><td>P220086_220907</td><td>173983</td><td>06S22052900</td><td>RepeatTURBT</td><td>FALSE</td><td>NA</td><td>NA</td><td>NA</td><td>98.4</td><td>99.9</td><td>100</td><td>Pass</td><td>NA</td><td>NA</td><td>22</td><td>3</td><td>10</td><td> 7.4471</td><td> 9.6252</td><td> 7.1680</td><td>10</td><td> 7.4471</td><td>17.1</td><td>0.292</td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 72\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & SampleID & Total.reads..M. & Mapped.reads..M. & Mapping.rate.... & Consensus.reads..M. & Duplication.rate.... & Target.unique.fragment.coverage & Panel2.Target.unique.fragment.coverage & Target.read.coverage & Mean.target.read.coverage & Panel2.Target.read.coverage & Percentage.of.target...20..Mean.coverage & Within.2.fold.....after.dedup & Within.2.fold.....before.dedup & Panel2.Within.2.fold.....after.dedup & Panel2.Within.2.fold.....before.dedup & onTarget.rate.... & onTarget.rate.ext200.... & gcRatio.low & gcRatio.high & Mean.error.rate.....after.dedup & background.50..after.dedup & background.99..after.dedup & Fragment\\_Size\\_Mode & Fragment\\_Size\\_Bandwidth & Softclip.ratio & Double.strand.ratio & BC\\_totalCount & SNP\\_BC & tmsi\\_score & msi\\_status & msi\\_QC & Specimen\\_type & Gender & Panel & ProjectDir & analysis\\_type & lane & work\\_order & original\\_adapter & total\\_dna\\_input & total\\_cfdna\\_input & library\\_dna\\_input\\_ug & enriched\\_library\\_fa\\_or\\_ba\\_nm & qubit\\_ng\\_ul & pversion & category & sampleFolder & PatientID & externalSampleID & trialVisitNum & Duplicated & Coverage...160.... & Coverage...80.... & Coverage...40.... & Coverage...400.... & Coverage...200.... & Coverage...100.... & QC\\_Status & Action & Load & SNV\\_Num & Indel\\_Num & pTMB & pTMB\\_norm & pTMB.weighted & pTMB\\_norm.weighted & pTMB.adj & pTMB\\_norm.adj & maxAF\\_TMB & tumorFraction & tumorStatus\\\\\n",
       "  & <chr> & <int> & <int> & <dbl> & <int> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <dbl> & <dbl> & <lgl> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <dbl> & <dbl> & <int> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <lgl> & <dbl> & <dbl> & <int> & <chr> & <lgl> & <lgl> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & P220085 & 260 & 260 & 99.9 & 127 & 51.0 & 1795 & 200 & 3200 & 3714 & 341 & 99.8 & 53.1 & 52.3 & 58.8 & 59.6 & 65.9 & 89.4 & 0.757 & 1.80 & 0.0220 & 0.382 & 2.19 & 183.3 & 135 & 1.63 & 28.9 & NA & '00011101101220010020101110102022021' & 0 & MSS & PASS & Tissue & female & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220907\\_A00934\\_0209\\_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 2 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC56\\_v2 & 246.25 & 227.83 & 2 & 30.2 & 8.76 & 1.7.0 & RUO & P220085\\_220907 & 173983 & 06S22036059 & IndexTURBT  & FALSE & NA & NA & NA & 99.8 & 99.9 & 100 & Pass & NA & NA & 25 & 4 & 12 &  8.9365 & 11.5410 &  8.5947 & 12 &  8.9365 & 21.0 & 0.348 & TRUE\\\\\n",
       "\t2 & P220076 & 257 & 256 & 99.8 & 135 & 47.3 & 1840 & 198 & 3150 & 3714 & 317 & 99.2 & 51.5 & 50.4 & 57.5 & 58.4 & 64.6 & 89.0 & 0.819 & 1.99 & 0.0394 & 0.410 & 2.44 & 193.7 & 151 & 1.72 & 25.6 & NA & '21101000111021002011220001111021021' & 2 & MSS & PASS & Tissue & male   & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220902\\_A01832\\_0037\\_BHCKWTDSX5//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 3 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC24\\_v2 & 283.25 & 216.09 & 2 & 31.0 & 7.33 & 1.7.0 & RUO & P220076\\_220902 & 173334 & 06S22068063 & IndexTURBT  & FALSE & NA & NA & NA & 99.4 & 99.9 & 100 & Pass & NA & NA & 88 & 0 & 91 & 67.7690 & 83.5810 & 62.2440 & 91 & 67.7690 & 59.8 & 0.748 & TRUE\\\\\n",
       "\t3 & P220070 & 251 & 251 & 99.8 & 128 & 49.1 & 1568 & 169 & 2730 & 3650 & 280 & 98.7 & 43.1 & 42.8 & 47.1 & 47.3 & 65.1 & 89.4 & 0.733 & 2.43 & 0.0262 & 0.345 & 2.33 & 188.5 & 145 & 1.74 & 32.2 & NA & '11102001200000001010210010000112021' & 0 & MSS & PASS & Tissue & male   & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220907\\_A00934\\_0209\\_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 1 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC46\\_v2 & 179.50 & 160.22 & 2 & 33.4 & 8.70 & 1.7.0 & RUO & P220070\\_220907 & 172586 & 06S22052895 & RepeatTURBT & FALSE & NA & NA & NA & 98.8 & 99.9 & 100 & Pass & NA & NA & 15 & 4 &  5 &  3.7236 &  4.4055 &  3.2808 &  5 &  3.7236 & 39.6 & 0.568 & TRUE\\\\\n",
       "\t4 & P220079 & 246 & 245 & 99.8 & 119 & 51.6 & 1821 & 206 & 3450 & 3491 & 355 & 99.9 & 69.0 & 66.2 & 73.6 & 73.3 & 66.0 & 89.2 & 0.837 & 1.18 & 0.0243 & 0.461 & 2.33 & 185.7 & 154 & 1.65 & 21.2 & NA & '11102200200111201111211000001111111' & 0 & MSS & PASS & Tissue & male   & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220907\\_A00934\\_0209\\_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 1 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC52\\_v2 & 184.50 & 136.75 & 2 & 14.3 & 4.62 & 1.7.0 & RUO & P220079\\_220907 & 172530 & 06S22036103 & IndexTURBT  & FALSE & NA & NA & NA & 99.8 & 99.9 & 100 & Pass & NA & NA & 23 & 2 & 19 & 14.1500 & 17.1400 & 12.7640 & 19 & 14.1500 & 15.0 & 0.260 & TRUE\\\\\n",
       "\t5 & P220069 & 245 & 245 & 99.9 & 109 & 55.5 & 1349 & 180 & 3100 & 3535 & 339 & 99.8 & 59.2 & 56.5 & 61.5 & 61.0 & 68.7 & 89.4 & 0.935 & 1.61 & 0.0338 & 0.474 & 2.38 & 156.7 & 126 & 2.70 & 15.6 & NA & '11102001200000001010210010000112021' & 2 & MSS & PASS & Tissue & male   & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220907\\_A00934\\_0209\\_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 1 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC45\\_v2 & 279.00 & 211.04 & 2 & 13.2 & 3.72 & 1.7.0 & RUO & P220069\\_220907 & 172586 & 06S22035975 & IndexTURBT  & FALSE & NA & NA & NA & 99.5 & 99.9 & 100 & Pass & NA & NA & 15 & 4 &  4 &  2.9788 &  3.4055 &  2.5361 &  4 &  2.9788 & 57.3 & 0.729 & TRUE\\\\\n",
       "\t6 & P220086 & 245 & 244 & 99.8 & 124 & 49.1 & 1160 & 134 & 2110 & 3322 & 226 & 99.1 & 49.3 & 48.7 & 50.8 & 50.6 & 63.4 & 87.6 & 0.842 & 3.43 & 0.0358 & 0.350 & 2.99 & 192.6 & 155 & 2.75 & 22.3 & NA & '00011101101220010020101110102022021' & 1 & MSS & PASS & Tissue & female & PREDICINEATLAS\\_V2 & /prednet/data22/OutputByRun22/220907\\_A00934\\_0209\\_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & tissue\\_analysis & 2 & WOP00249\\_PRDC-MOFFITT-NMIBC-22001 & HBC57\\_v2 & 244.75 & 203.00 & 2 & 20.5 & 6.04 & 1.7.0 & RUO & P220086\\_220907 & 173983 & 06S22052900 & RepeatTURBT & FALSE & NA & NA & NA & 98.4 & 99.9 & 100 & Pass & NA & NA & 22 & 3 & 10 &  7.4471 &  9.6252 &  7.1680 & 10 &  7.4471 & 17.1 & 0.292 & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 72\n",
       "\n",
       "| <!--/--> | SampleID &lt;chr&gt; | Total.reads..M. &lt;int&gt; | Mapped.reads..M. &lt;int&gt; | Mapping.rate.... &lt;dbl&gt; | Consensus.reads..M. &lt;int&gt; | Duplication.rate.... &lt;dbl&gt; | Target.unique.fragment.coverage &lt;int&gt; | Panel2.Target.unique.fragment.coverage &lt;int&gt; | Target.read.coverage &lt;int&gt; | Mean.target.read.coverage &lt;int&gt; | Panel2.Target.read.coverage &lt;int&gt; | Percentage.of.target...20..Mean.coverage &lt;dbl&gt; | Within.2.fold.....after.dedup &lt;dbl&gt; | Within.2.fold.....before.dedup &lt;dbl&gt; | Panel2.Within.2.fold.....after.dedup &lt;dbl&gt; | Panel2.Within.2.fold.....before.dedup &lt;dbl&gt; | onTarget.rate.... &lt;dbl&gt; | onTarget.rate.ext200.... &lt;dbl&gt; | gcRatio.low &lt;dbl&gt; | gcRatio.high &lt;dbl&gt; | Mean.error.rate.....after.dedup &lt;dbl&gt; | background.50..after.dedup &lt;dbl&gt; | background.99..after.dedup &lt;dbl&gt; | Fragment_Size_Mode &lt;dbl&gt; | Fragment_Size_Bandwidth &lt;int&gt; | Softclip.ratio &lt;dbl&gt; | Double.strand.ratio &lt;dbl&gt; | BC_totalCount &lt;lgl&gt; | SNP_BC &lt;chr&gt; | tmsi_score &lt;int&gt; | msi_status &lt;chr&gt; | msi_QC &lt;chr&gt; | Specimen_type &lt;chr&gt; | Gender &lt;chr&gt; | Panel &lt;chr&gt; | ProjectDir &lt;chr&gt; | analysis_type &lt;chr&gt; | lane &lt;int&gt; | work_order &lt;chr&gt; | original_adapter &lt;chr&gt; | total_dna_input &lt;dbl&gt; | total_cfdna_input &lt;dbl&gt; | library_dna_input_ug &lt;int&gt; | enriched_library_fa_or_ba_nm &lt;dbl&gt; | qubit_ng_ul &lt;dbl&gt; | pversion &lt;chr&gt; | category &lt;chr&gt; | sampleFolder &lt;chr&gt; | PatientID &lt;chr&gt; | externalSampleID &lt;chr&gt; | trialVisitNum &lt;chr&gt; | Duplicated &lt;lgl&gt; | Coverage...160.... &lt;lgl&gt; | Coverage...80.... &lt;lgl&gt; | Coverage...40.... &lt;lgl&gt; | Coverage...400.... &lt;dbl&gt; | Coverage...200.... &lt;dbl&gt; | Coverage...100.... &lt;int&gt; | QC_Status &lt;chr&gt; | Action &lt;lgl&gt; | Load &lt;lgl&gt; | SNV_Num &lt;int&gt; | Indel_Num &lt;int&gt; | pTMB &lt;int&gt; | pTMB_norm &lt;dbl&gt; | pTMB.weighted &lt;dbl&gt; | pTMB_norm.weighted &lt;dbl&gt; | pTMB.adj &lt;int&gt; | pTMB_norm.adj &lt;dbl&gt; | maxAF_TMB &lt;dbl&gt; | tumorFraction &lt;dbl&gt; | tumorStatus &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | P220085 | 260 | 260 | 99.9 | 127 | 51.0 | 1795 | 200 | 3200 | 3714 | 341 | 99.8 | 53.1 | 52.3 | 58.8 | 59.6 | 65.9 | 89.4 | 0.757 | 1.80 | 0.0220 | 0.382 | 2.19 | 183.3 | 135 | 1.63 | 28.9 | NA | '00011101101220010020101110102022021' | 0 | MSS | PASS | Tissue | female | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 2 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC56_v2 | 246.25 | 227.83 | 2 | 30.2 | 8.76 | 1.7.0 | RUO | P220085_220907 | 173983 | 06S22036059 | IndexTURBT  | FALSE | NA | NA | NA | 99.8 | 99.9 | 100 | Pass | NA | NA | 25 | 4 | 12 |  8.9365 | 11.5410 |  8.5947 | 12 |  8.9365 | 21.0 | 0.348 | TRUE |\n",
       "| 2 | P220076 | 257 | 256 | 99.8 | 135 | 47.3 | 1840 | 198 | 3150 | 3714 | 317 | 99.2 | 51.5 | 50.4 | 57.5 | 58.4 | 64.6 | 89.0 | 0.819 | 1.99 | 0.0394 | 0.410 | 2.44 | 193.7 | 151 | 1.72 | 25.6 | NA | '21101000111021002011220001111021021' | 2 | MSS | PASS | Tissue | male   | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220902_A01832_0037_BHCKWTDSX5//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 3 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC24_v2 | 283.25 | 216.09 | 2 | 31.0 | 7.33 | 1.7.0 | RUO | P220076_220902 | 173334 | 06S22068063 | IndexTURBT  | FALSE | NA | NA | NA | 99.4 | 99.9 | 100 | Pass | NA | NA | 88 | 0 | 91 | 67.7690 | 83.5810 | 62.2440 | 91 | 67.7690 | 59.8 | 0.748 | TRUE |\n",
       "| 3 | P220070 | 251 | 251 | 99.8 | 128 | 49.1 | 1568 | 169 | 2730 | 3650 | 280 | 98.7 | 43.1 | 42.8 | 47.1 | 47.3 | 65.1 | 89.4 | 0.733 | 2.43 | 0.0262 | 0.345 | 2.33 | 188.5 | 145 | 1.74 | 32.2 | NA | '11102001200000001010210010000112021' | 0 | MSS | PASS | Tissue | male   | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 1 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC46_v2 | 179.50 | 160.22 | 2 | 33.4 | 8.70 | 1.7.0 | RUO | P220070_220907 | 172586 | 06S22052895 | RepeatTURBT | FALSE | NA | NA | NA | 98.8 | 99.9 | 100 | Pass | NA | NA | 15 | 4 |  5 |  3.7236 |  4.4055 |  3.2808 |  5 |  3.7236 | 39.6 | 0.568 | TRUE |\n",
       "| 4 | P220079 | 246 | 245 | 99.8 | 119 | 51.6 | 1821 | 206 | 3450 | 3491 | 355 | 99.9 | 69.0 | 66.2 | 73.6 | 73.3 | 66.0 | 89.2 | 0.837 | 1.18 | 0.0243 | 0.461 | 2.33 | 185.7 | 154 | 1.65 | 21.2 | NA | '11102200200111201111211000001111111' | 0 | MSS | PASS | Tissue | male   | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 1 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC52_v2 | 184.50 | 136.75 | 2 | 14.3 | 4.62 | 1.7.0 | RUO | P220079_220907 | 172530 | 06S22036103 | IndexTURBT  | FALSE | NA | NA | NA | 99.8 | 99.9 | 100 | Pass | NA | NA | 23 | 2 | 19 | 14.1500 | 17.1400 | 12.7640 | 19 | 14.1500 | 15.0 | 0.260 | TRUE |\n",
       "| 5 | P220069 | 245 | 245 | 99.9 | 109 | 55.5 | 1349 | 180 | 3100 | 3535 | 339 | 99.8 | 59.2 | 56.5 | 61.5 | 61.0 | 68.7 | 89.4 | 0.935 | 1.61 | 0.0338 | 0.474 | 2.38 | 156.7 | 126 | 2.70 | 15.6 | NA | '11102001200000001010210010000112021' | 2 | MSS | PASS | Tissue | male   | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 1 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC45_v2 | 279.00 | 211.04 | 2 | 13.2 | 3.72 | 1.7.0 | RUO | P220069_220907 | 172586 | 06S22035975 | IndexTURBT  | FALSE | NA | NA | NA | 99.5 | 99.9 | 100 | Pass | NA | NA | 15 | 4 |  4 |  2.9788 |  3.4055 |  2.5361 |  4 |  2.9788 | 57.3 | 0.729 | TRUE |\n",
       "| 6 | P220086 | 245 | 244 | 99.8 | 124 | 49.1 | 1160 | 134 | 2110 | 3322 | 226 | 99.1 | 49.3 | 48.7 | 50.8 | 50.6 | 63.4 | 87.6 | 0.842 | 3.43 | 0.0358 | 0.350 | 2.99 | 192.6 | 155 | 2.75 | 22.3 | NA | '00011101101220010020101110102022021' | 1 | MSS | PASS | Tissue | female | PREDICINEATLAS_V2 | /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001 | tissue_analysis | 2 | WOP00249_PRDC-MOFFITT-NMIBC-22001 | HBC57_v2 | 244.75 | 203.00 | 2 | 20.5 | 6.04 | 1.7.0 | RUO | P220086_220907 | 173983 | 06S22052900 | RepeatTURBT | FALSE | NA | NA | NA | 98.4 | 99.9 | 100 | Pass | NA | NA | 22 | 3 | 10 |  7.4471 |  9.6252 |  7.1680 | 10 |  7.4471 | 17.1 | 0.292 | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "  SampleID Total.reads..M. Mapped.reads..M. Mapping.rate....\n",
       "1 P220085  260             260              99.9            \n",
       "2 P220076  257             256              99.8            \n",
       "3 P220070  251             251              99.8            \n",
       "4 P220079  246             245              99.8            \n",
       "5 P220069  245             245              99.9            \n",
       "6 P220086  245             244              99.8            \n",
       "  Consensus.reads..M. Duplication.rate.... Target.unique.fragment.coverage\n",
       "1 127                 51.0                 1795                           \n",
       "2 135                 47.3                 1840                           \n",
       "3 128                 49.1                 1568                           \n",
       "4 119                 51.6                 1821                           \n",
       "5 109                 55.5                 1349                           \n",
       "6 124                 49.1                 1160                           \n",
       "  Panel2.Target.unique.fragment.coverage Target.read.coverage\n",
       "1 200                                    3200                \n",
       "2 198                                    3150                \n",
       "3 169                                    2730                \n",
       "4 206                                    3450                \n",
       "5 180                                    3100                \n",
       "6 134                                    2110                \n",
       "  Mean.target.read.coverage Panel2.Target.read.coverage\n",
       "1 3714                      341                        \n",
       "2 3714                      317                        \n",
       "3 3650                      280                        \n",
       "4 3491                      355                        \n",
       "5 3535                      339                        \n",
       "6 3322                      226                        \n",
       "  Percentage.of.target...20..Mean.coverage Within.2.fold.....after.dedup\n",
       "1 99.8                                     53.1                         \n",
       "2 99.2                                     51.5                         \n",
       "3 98.7                                     43.1                         \n",
       "4 99.9                                     69.0                         \n",
       "5 99.8                                     59.2                         \n",
       "6 99.1                                     49.3                         \n",
       "  Within.2.fold.....before.dedup Panel2.Within.2.fold.....after.dedup\n",
       "1 52.3                           58.8                                \n",
       "2 50.4                           57.5                                \n",
       "3 42.8                           47.1                                \n",
       "4 66.2                           73.6                                \n",
       "5 56.5                           61.5                                \n",
       "6 48.7                           50.8                                \n",
       "  Panel2.Within.2.fold.....before.dedup onTarget.rate....\n",
       "1 59.6                                  65.9             \n",
       "2 58.4                                  64.6             \n",
       "3 47.3                                  65.1             \n",
       "4 73.3                                  66.0             \n",
       "5 61.0                                  68.7             \n",
       "6 50.6                                  63.4             \n",
       "  onTarget.rate.ext200.... gcRatio.low gcRatio.high\n",
       "1 89.4                     0.757       1.80        \n",
       "2 89.0                     0.819       1.99        \n",
       "3 89.4                     0.733       2.43        \n",
       "4 89.2                     0.837       1.18        \n",
       "5 89.4                     0.935       1.61        \n",
       "6 87.6                     0.842       3.43        \n",
       "  Mean.error.rate.....after.dedup background.50..after.dedup\n",
       "1 0.0220                          0.382                     \n",
       "2 0.0394                          0.410                     \n",
       "3 0.0262                          0.345                     \n",
       "4 0.0243                          0.461                     \n",
       "5 0.0338                          0.474                     \n",
       "6 0.0358                          0.350                     \n",
       "  background.99..after.dedup Fragment_Size_Mode Fragment_Size_Bandwidth\n",
       "1 2.19                       183.3              135                    \n",
       "2 2.44                       193.7              151                    \n",
       "3 2.33                       188.5              145                    \n",
       "4 2.33                       185.7              154                    \n",
       "5 2.38                       156.7              126                    \n",
       "6 2.99                       192.6              155                    \n",
       "  Softclip.ratio Double.strand.ratio BC_totalCount\n",
       "1 1.63           28.9                NA           \n",
       "2 1.72           25.6                NA           \n",
       "3 1.74           32.2                NA           \n",
       "4 1.65           21.2                NA           \n",
       "5 2.70           15.6                NA           \n",
       "6 2.75           22.3                NA           \n",
       "  SNP_BC                                tmsi_score msi_status msi_QC\n",
       "1 '00011101101220010020101110102022021' 0          MSS        PASS  \n",
       "2 '21101000111021002011220001111021021' 2          MSS        PASS  \n",
       "3 '11102001200000001010210010000112021' 0          MSS        PASS  \n",
       "4 '11102200200111201111211000001111111' 0          MSS        PASS  \n",
       "5 '11102001200000001010210010000112021' 2          MSS        PASS  \n",
       "6 '00011101101220010020101110102022021' 1          MSS        PASS  \n",
       "  Specimen_type Gender Panel            \n",
       "1 Tissue        female PREDICINEATLAS_V2\n",
       "2 Tissue        male   PREDICINEATLAS_V2\n",
       "3 Tissue        male   PREDICINEATLAS_V2\n",
       "4 Tissue        male   PREDICINEATLAS_V2\n",
       "5 Tissue        male   PREDICINEATLAS_V2\n",
       "6 Tissue        female PREDICINEATLAS_V2\n",
       "  ProjectDir                                                                                                               \n",
       "1 /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "2 /prednet/data22/OutputByRun22/220902_A01832_0037_BHCKWTDSX5//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "3 /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "4 /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "5 /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "6 /prednet/data22/OutputByRun22/220907_A00934_0209_BHCV2YDMXY//dsrun1.7.0/lbwfresult1.7.0/WOP00249_PRDC-MOFFITT-NMIBC-22001\n",
       "  analysis_type   lane work_order                        original_adapter\n",
       "1 tissue_analysis 2    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC56_v2        \n",
       "2 tissue_analysis 3    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC24_v2        \n",
       "3 tissue_analysis 1    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC46_v2        \n",
       "4 tissue_analysis 1    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC52_v2        \n",
       "5 tissue_analysis 1    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC45_v2        \n",
       "6 tissue_analysis 2    WOP00249_PRDC-MOFFITT-NMIBC-22001 HBC57_v2        \n",
       "  total_dna_input total_cfdna_input library_dna_input_ug\n",
       "1 246.25          227.83            2                   \n",
       "2 283.25          216.09            2                   \n",
       "3 179.50          160.22            2                   \n",
       "4 184.50          136.75            2                   \n",
       "5 279.00          211.04            2                   \n",
       "6 244.75          203.00            2                   \n",
       "  enriched_library_fa_or_ba_nm qubit_ng_ul pversion category sampleFolder  \n",
       "1 30.2                         8.76        1.7.0    RUO      P220085_220907\n",
       "2 31.0                         7.33        1.7.0    RUO      P220076_220902\n",
       "3 33.4                         8.70        1.7.0    RUO      P220070_220907\n",
       "4 14.3                         4.62        1.7.0    RUO      P220079_220907\n",
       "5 13.2                         3.72        1.7.0    RUO      P220069_220907\n",
       "6 20.5                         6.04        1.7.0    RUO      P220086_220907\n",
       "  PatientID externalSampleID trialVisitNum Duplicated Coverage...160....\n",
       "1 173983    06S22036059      IndexTURBT    FALSE      NA                \n",
       "2 173334    06S22068063      IndexTURBT    FALSE      NA                \n",
       "3 172586    06S22052895      RepeatTURBT   FALSE      NA                \n",
       "4 172530    06S22036103      IndexTURBT    FALSE      NA                \n",
       "5 172586    06S22035975      IndexTURBT    FALSE      NA                \n",
       "6 173983    06S22052900      RepeatTURBT   FALSE      NA                \n",
       "  Coverage...80.... Coverage...40.... Coverage...400.... Coverage...200....\n",
       "1 NA                NA                99.8               99.9              \n",
       "2 NA                NA                99.4               99.9              \n",
       "3 NA                NA                98.8               99.9              \n",
       "4 NA                NA                99.8               99.9              \n",
       "5 NA                NA                99.5               99.9              \n",
       "6 NA                NA                98.4               99.9              \n",
       "  Coverage...100.... QC_Status Action Load SNV_Num Indel_Num pTMB pTMB_norm\n",
       "1 100                Pass      NA     NA   25      4         12    8.9365  \n",
       "2 100                Pass      NA     NA   88      0         91   67.7690  \n",
       "3 100                Pass      NA     NA   15      4          5    3.7236  \n",
       "4 100                Pass      NA     NA   23      2         19   14.1500  \n",
       "5 100                Pass      NA     NA   15      4          4    2.9788  \n",
       "6 100                Pass      NA     NA   22      3         10    7.4471  \n",
       "  pTMB.weighted pTMB_norm.weighted pTMB.adj pTMB_norm.adj maxAF_TMB\n",
       "1 11.5410        8.5947            12        8.9365       21.0     \n",
       "2 83.5810       62.2440            91       67.7690       59.8     \n",
       "3  4.4055        3.2808             5        3.7236       39.6     \n",
       "4 17.1400       12.7640            19       14.1500       15.0     \n",
       "5  3.4055        2.5361             4        2.9788       57.3     \n",
       "6  9.6252        7.1680            10        7.4471       17.1     \n",
       "  tumorFraction tumorStatus\n",
       "1 0.348         TRUE       \n",
       "2 0.748         TRUE       \n",
       "3 0.568         TRUE       \n",
       "4 0.260         TRUE       \n",
       "5 0.729         TRUE       \n",
       "6 0.292         TRUE       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): \n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): \nTraceback:\n",
      "1. stop()"
     ]
    }
   ],
   "source": [
    "## NGSQC\n",
    "source(\"~/Desktop/puffin/R/helper_functions.R\")\n",
    "\n",
    "df1.ffpe = read.csv(\"./data_batch1_and_2/FFPE_WES/bg_rerun_09132023/PRDC-MOFFITT-NMIBC-22001_NGSQC_all_2023-09-13_tissue_5_updated.csv\")\n",
    "df1.urine = read.csv(\"./data_batch1_and_2/Urine-UCP/Urine_post_analysis/urine_WES_only//WOP00253_WOP00254_PRDC-MOFFITT-NMIBC-22001_NGSQC_all_2023-09-13_urine_supernatant_0.35_updated.csv\")\n",
    "df3.ffpe =read.csv(\"./data_batch3_02222023/PRDC-MOFFITT-NMIBC-22001-B_NGSQC_all_2023-07-30_ffpe_5_updated.csv\")\n",
    "df.combo = read.csv(\"./batch1_2_3_urine_WES_baselines/WOP00873_WOP00875_MCC-NMIBC_WES_urine_NGSQC_all_2023-07-30_urine_supernatant_0.35_updated.csv\")\n",
    "df.3.5.ffpe = read.csv(\"./benign_urothelium/WOP00981_PRDC-MOFFITT-NMIBC-22001_NGSQC_all_2023-09-01_ffpe_5_updated.csv\")\n",
    "df4.ffpe = read.csv(\"./Validation_cohort/Validation_FFPE/\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "6e27726e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3323708"
      ],
      "text/latex": [
       "3323708"
      ],
      "text/markdown": [
       "3323708"
      ],
      "text/plain": [
       "[1] 3323708"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "        \n",
       "         Tissue Urine  <NA>\n",
       "  116381  42685     0     0\n",
       "  152237  56605     0     0\n",
       "  161922  16096     0     0\n",
       "  165028  21336     0     0\n",
       "  167162  53029     0     0\n",
       "  169558  16353     0     0\n",
       "  171374      0 28545     0\n",
       "  171769  33265 34606     0\n",
       "  171821  55762 38216     0\n",
       "  172229  41776 36921     0\n",
       "  172530  35003     0     0\n",
       "  172586  33410 34957     0\n",
       "  172602  16105 28583     0\n",
       "  172749  49454 35508     0\n",
       "  172851  33158 35076     0\n",
       "  173075  19476 36112     0\n",
       "  173254      0 35884     0\n",
       "  173281  27560 37554     0\n",
       "  173334  40631     0     0\n",
       "  173350  57523     0     0\n",
       "  173359      0 33924     0\n",
       "  173362  16325 35120     0\n",
       "  173509  33262     0     0\n",
       "  173912  48446     0     0\n",
       "  173975  58235     0     0\n",
       "  173983  34580     0     0\n",
       "  174083  54690 36000     0\n",
       "  174091  30682     0     0\n",
       "  174872  29942     0     0\n",
       "  174947  31342 39030     0\n",
       "  175325  59089     0     0\n",
       "  175387  56408     0     0\n",
       "  175425  50623 35645     0\n",
       "  175492  54989     0     0\n",
       "  175526  54080     0     0\n",
       "  175610  53023     0     0\n",
       "  175648  54631 36529     0\n",
       "  175672  66742 43263     0\n",
       "  175725  50044 37570     0\n",
       "  175733  52016     0     0\n",
       "  175831  35533     0     0\n",
       "  175854  62561     0     0\n",
       "  175980  51684     0     0\n",
       "  176349  56557     0     0\n",
       "  176388  54288 36835     0\n",
       "  176828  52382 37903     0\n",
       "  176992  43420     0     0\n",
       "  177967  62514     0     0\n",
       "  178886  50909     0     0\n",
       "  179692  19237     0     0\n",
       "  180264  48488     0     0\n",
       "  181439  15830     0     0\n",
       "  181568  17667     0     0\n",
       "  182022  58894     0     0\n",
       "  182435  27221     0     0\n",
       "  182910  24334     0     0\n",
       "  184289  24281     0     0\n",
       "  184485  27258     0     0\n",
       "  184615  28336     0     0\n",
       "  184957  16145     0     0\n",
       "  185212  19132     0     0\n",
       "  185382  17294     0     0\n",
       "  185581  17583     0     0\n",
       "  186478  15373     0     0\n",
       "  186652  23009     0     0\n",
       "  187204  19726     0     0\n",
       "  187807  16492     0     0\n",
       "  188867  17740     0     0\n",
       "  189083  22533     0     0\n",
       "  189342  25015     0     0\n",
       "  190233  22185     0     0\n",
       "  191632  21121     0     0\n",
       "  193186  16809     0     0\n",
       "  <NA>        0     0     0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "source(\"~/Desktop/puffin/R/helper_functions.R\")\n",
    "\n",
    "## All Variant Marked\n",
    "# df1.ffpe = read.csv(\"./data_batch1_and_2/FFPE_WES/bg_rerun_09132023/PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-09-13_tissue_marked.csv\") \n",
    "# df1.urine = read.csv(\"./data_batch1_and_2/Urine-UCP/Urine_post_analysis/urine_WES_only/WOP00253_WOP00254_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-09-13_urine_marked.csv\")\n",
    "# df3.ffpe =read.csv(\"./data_batch3_02222023/PRDC-MOFFITT-NMIBC-22001-B_Variant_all_2023-05-03_ffpe_marked.csv\")\n",
    "# df.combo = read.csv(\"./batch1_2_3_urine_WES_baselines/WOP00873_WOP00875_MCC-NMIBC_WES_urine_Variant_all_2023-07-26_urine_supernatant_marked.csv\")\n",
    "# df.3.5.ffpe = read.csv(\"./benign_urothelium/WOP00981_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-09-01_ffpe_marked.csv\")\n",
    "# df4.ffpe = read.csv(\"./Validation_cohort/Validation_FFPE/WOP01194_WOP01039_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-10-24_ffpe_marked.csv\")\n",
    "\n",
    "# cols = intersect(intersect(intersect(names(df4.ffpe), names(df1.urine)), names(df.combo)), names(df1.ffpe))\n",
    "# df.all.marked = rbind(df1.ffpe[,cols], df1.urine[,cols])\n",
    "# df.all.marked = rbind(df.all.marked, df3.ffpe[,cols])\n",
    "# df.all.marked = rbind(df.all.marked, df.combo[,cols])\n",
    "# df.all.marked = rbind(df.all.marked, df.3.5.ffpe[,cols])\n",
    "# df.all.marked = rbind(df.all.marked, df4.ffpe[,cols])\n",
    "\n",
    "# df_samples_all = df_samples_all %>% mutate(SampleID.short = make_SIDshort(SampleID, \"strict\"))\n",
    "\n",
    "# df.all.marked_ = df.all.marked %>% select(-PatientID) %>% mutate(SampleID.short=make_SIDshort(sampleNames, \"strict\")) %>%\n",
    "#                 left_join(df_samples_all %>% select(PatientID, StudyVisit, SampleType, SampleID.short), by=\"SampleID.short\")\n",
    "# warn_na(df.all.marked_$PatientID)\n",
    "# warn_na(df.all.marked_$StudyVisit)\n",
    "# warn_na(df.all.marked_$SampleType)\n",
    "\n",
    "# df.all.marked.f = df.all.marked_ %>% filter(SampleType != \"Buffy_Coat\") %>% \n",
    "#                                     distinct()\n",
    "nrow(df.all.marked.f)\n",
    "table(df.all.marked.f$PatientID, df.all.marked.f$SampleType)\n",
    "\n",
    "write.csv(df.all.marked.f, glue(\"./data_delivery/WES/PRDC-MOFFITT-NMIBC-22001_Variant_all_combined_{Sys.Date()}.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "97bd1bc7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in cbind(parts$left, chars$ellip_h, parts$right, deparse.level = 0L):\n",
      "“number of rows of result is not a multiple of vector length (arg 2)”\n",
      "Warning message in cbind(parts$left, chars$ellip_h, parts$right, deparse.level = 0L):\n",
      "“number of rows of result is not a multiple of vector length (arg 2)”\n",
      "Warning message in cbind(parts$left, chars$ellip_h, parts$right, deparse.level = 0L):\n",
      "“number of rows of result is not a multiple of vector length (arg 2)”\n",
      "Warning message in cbind(parts$left, chars$ellip_h, parts$right, deparse.level = 0L):\n",
      "“number of rows of result is not a multiple of vector length (arg 2)”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 99</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>start</th><th scope=col>end</th><th scope=col>width</th><th scope=col>strand</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>totalDepth</th><th scope=col>refDepth</th><th scope=col>altDepth</th><th scope=col>sampleNames</th><th scope=col>VariantFreq</th><th scope=col>SYMBOL</th><th scope=col>GeneID</th><th scope=col>Feature</th><th scope=col>HGVSc</th><th scope=col>HGVSp</th><th scope=col>Amino_acids</th><th scope=col>Codons</th><th scope=col>BIOTYPE</th><th scope=col>EXON</th><th scope=col>INTRON</th><th scope=col>Consequence</th><th scope=col>DISTANCE</th><th scope=col>Existing_variation</th><th scope=col>IMPACT</th><th scope=col>VARIANT_CLASS</th><th scope=col>CLIN_SIG</th><th scope=col>Clinvar</th><th scope=col>dbSNP</th><th scope=col>COSMIC</th><th scope=col>COSMIC.CNT</th><th scope=col>genome1000</th><th scope=col>AF</th><th scope=col>CANONICAL</th><th scope=col>filteredCnt</th><th scope=col>DSCnt</th><th scope=col>filteredDSCnt</th><th scope=col>⋯</th><th scope=col>fc.ds.ratio</th><th scope=col>CopyNumber</th><th scope=col>zScore.cnv</th><th scope=col>coVariant</th><th scope=col>ol.coVariant</th><th scope=col>coVariantFilter</th><th scope=col>sideVariant</th><th scope=col>sideVariant.AF.ratio</th><th scope=col>prevalenceInternal</th><th scope=col>filterPrevalence</th><th scope=col>filterScore</th><th scope=col>finalKeep</th><th scope=col>VariantID</th><th scope=col>sampleDir</th><th scope=col>VariantType</th><th scope=col>VariantType.old</th><th scope=col>FilterType</th><th scope=col>sampleFolder</th><th scope=col>externalSampleID</th><th scope=col>ID</th><th scope=col>highFrequent.inbatch</th><th scope=col>concordant</th><th scope=col>totalDepth.Baseline</th><th scope=col>dscnt.Baseline.filtered</th><th scope=col>altDepth.Baseline</th><th scope=col>altDepth.Baseline.filtered</th><th scope=col>AF.Baseline</th><th scope=col>odds.ratio</th><th scope=col>pvalue</th><th scope=col>MAF.diff</th><th scope=col>finalKeep.old</th><th scope=col>SampleID.short</th><th scope=col>StudyVisit.x</th><th scope=col>SampleType.x</th><th scope=col>PatientID</th><th scope=col>StudyVisit.y</th><th scope=col>SampleType.y</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 99\n",
       "\\begin{tabular}{llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       " seqnames & start & end & width & strand & ref & alt & totalDepth & refDepth & altDepth & sampleNames & VariantFreq & SYMBOL & GeneID & Feature & HGVSc & HGVSp & Amino\\_acids & Codons & BIOTYPE & EXON & INTRON & Consequence & DISTANCE & Existing\\_variation & IMPACT & VARIANT\\_CLASS & CLIN\\_SIG & Clinvar & dbSNP & COSMIC & COSMIC.CNT & genome1000 & AF & CANONICAL & filteredCnt & DSCnt & filteredDSCnt & ⋯ & fc.ds.ratio & CopyNumber & zScore.cnv & coVariant & ol.coVariant & coVariantFilter & sideVariant & sideVariant.AF.ratio & prevalenceInternal & filterPrevalence & filterScore & finalKeep & VariantID & sampleDir & VariantType & VariantType.old & FilterType & sampleFolder & externalSampleID & ID & highFrequent.inbatch & concordant & totalDepth.Baseline & dscnt.Baseline.filtered & altDepth.Baseline & altDepth.Baseline.filtered & AF.Baseline & odds.ratio & pvalue & MAF.diff & finalKeep.old & SampleID.short & StudyVisit.x & SampleType.x & PatientID & StudyVisit.y & SampleType.y\\\\\n",
       " <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <int> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <dbl> & <dbl> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <dbl> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 99\n",
       "\n",
       "| seqnames &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | width &lt;int&gt; | strand &lt;chr&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | totalDepth &lt;int&gt; | refDepth &lt;int&gt; | altDepth &lt;int&gt; | sampleNames &lt;chr&gt; | VariantFreq &lt;dbl&gt; | SYMBOL &lt;chr&gt; | GeneID &lt;int&gt; | Feature &lt;chr&gt; | HGVSc &lt;chr&gt; | HGVSp &lt;chr&gt; | Amino_acids &lt;chr&gt; | Codons &lt;chr&gt; | BIOTYPE &lt;chr&gt; | EXON &lt;chr&gt; | INTRON &lt;chr&gt; | Consequence &lt;chr&gt; | DISTANCE &lt;int&gt; | Existing_variation &lt;chr&gt; | IMPACT &lt;chr&gt; | VARIANT_CLASS &lt;chr&gt; | CLIN_SIG &lt;chr&gt; | Clinvar &lt;chr&gt; | dbSNP &lt;chr&gt; | COSMIC &lt;chr&gt; | COSMIC.CNT &lt;chr&gt; | genome1000 &lt;chr&gt; | AF &lt;dbl&gt; | CANONICAL &lt;chr&gt; | filteredCnt &lt;dbl&gt; | DSCnt &lt;int&gt; | filteredDSCnt &lt;dbl&gt; | ⋯ ⋯ | fc.ds.ratio &lt;dbl&gt; | CopyNumber &lt;dbl&gt; | zScore.cnv &lt;dbl&gt; | coVariant &lt;chr&gt; | ol.coVariant &lt;chr&gt; | coVariantFilter &lt;lgl&gt; | sideVariant &lt;chr&gt; | sideVariant.AF.ratio &lt;dbl&gt; | prevalenceInternal &lt;dbl&gt; | filterPrevalence &lt;lgl&gt; | filterScore &lt;dbl&gt; | finalKeep &lt;chr&gt; | VariantID &lt;chr&gt; | sampleDir &lt;chr&gt; | VariantType &lt;chr&gt; | VariantType.old &lt;chr&gt; | FilterType &lt;chr&gt; | sampleFolder &lt;chr&gt; | externalSampleID &lt;chr&gt; | ID &lt;chr&gt; | highFrequent.inbatch &lt;int&gt; | concordant &lt;lgl&gt; | totalDepth.Baseline &lt;dbl&gt; | dscnt.Baseline.filtered &lt;dbl&gt; | altDepth.Baseline &lt;int&gt; | altDepth.Baseline.filtered &lt;dbl&gt; | AF.Baseline &lt;dbl&gt; | odds.ratio &lt;dbl&gt; | pvalue &lt;dbl&gt; | MAF.diff &lt;dbl&gt; | finalKeep.old &lt;chr&gt; | SampleID.short &lt;chr&gt; | StudyVisit.x &lt;chr&gt; | SampleType.x &lt;chr&gt; | PatientID &lt;chr&gt; | StudyVisit.y &lt;chr&gt; | SampleType.y &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     seqnames start end width strand ref alt totalDepth refDepth altDepth\n",
       "     sampleNames VariantFreq SYMBOL GeneID Feature HGVSc HGVSp Amino_acids\n",
       "     Codons BIOTYPE EXON INTRON Consequence DISTANCE Existing_variation IMPACT\n",
       "     VARIANT_CLASS CLIN_SIG Clinvar dbSNP COSMIC COSMIC.CNT genome1000 AF\n",
       "     CANONICAL filteredCnt DSCnt filteredDSCnt ⋯ fc.ds.ratio CopyNumber\n",
       "     zScore.cnv coVariant ol.coVariant coVariantFilter sideVariant\n",
       "     sideVariant.AF.ratio prevalenceInternal filterPrevalence filterScore\n",
       "     finalKeep VariantID sampleDir VariantType VariantType.old FilterType\n",
       "     sampleFolder externalSampleID ID highFrequent.inbatch concordant\n",
       "     totalDepth.Baseline dscnt.Baseline.filtered altDepth.Baseline\n",
       "     altDepth.Baseline.filtered AF.Baseline odds.ratio pvalue MAF.diff\n",
       "     finalKeep.old SampleID.short StudyVisit.x SampleType.x PatientID\n",
       "     StudyVisit.y SampleType.y"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'seqnames'</li><li>'start'</li><li>'end'</li><li>'width'</li><li>'strand'</li><li>'ref'</li><li>'alt'</li><li>'totalDepth'</li><li>'refDepth'</li><li>'altDepth'</li><li>'sampleNames'</li><li>'VariantFreq'</li><li>'SYMBOL'</li><li>'GeneID'</li><li>'Feature'</li><li>'HGVSc'</li><li>'HGVSp'</li><li>'Amino_acids'</li><li>'Codons'</li><li>'BIOTYPE'</li><li>'EXON'</li><li>'INTRON'</li><li>'Consequence'</li><li>'DISTANCE'</li><li>'Existing_variation'</li><li>'IMPACT'</li><li>'VARIANT_CLASS'</li><li>'CLIN_SIG'</li><li>'Clinvar'</li><li>'dbSNP'</li><li>'COSMIC'</li><li>'COSMIC.CNT'</li><li>'genome1000'</li><li>'AF'</li><li>'CANONICAL'</li><li>'filteredCnt'</li><li>'DSCnt'</li><li>'filteredDSCnt'</li><li>'pseudoDSCnt'</li><li>'altLOD'</li><li>'Qscore'</li><li>'entropy'</li><li>'avgEntropy'</li><li>'filtered_tlen_mean'</li><li>'tlen.ratio'</li><li>'filtered_toend_numunique'</li><li>'filtered_total_base_count'</li><li>'filterVarType'</li><li>'repeatRegion'</li><li>'white.list'</li><li>'wl.tier'</li><li>'MNV'</li><li>'GC.Percent'</li><li>'tandemRepeat'</li><li>'AF.th'</li><li>'AF.diff'</li><li>'bgFilter'</li><li>'generalFilter'</li><li>'localVariantFilter'</li><li>'ReadEndFilter'</li><li>'DS.filter'</li><li>'highQ.ratio'</li><li>'fc.ds.ratio'</li><li>'CopyNumber'</li><li>'zScore.cnv'</li><li>'coVariant'</li><li>'ol.coVariant'</li><li>'coVariantFilter'</li><li>'sideVariant'</li><li>'sideVariant.AF.ratio'</li><li>'prevalenceInternal'</li><li>'filterPrevalence'</li><li>'filterScore'</li><li>'finalKeep'</li><li>'VariantID'</li><li>'sampleDir'</li><li>'VariantType'</li><li>'VariantType.old'</li><li>'FilterType'</li><li>'sampleFolder'</li><li>'externalSampleID'</li><li>'ID'</li><li>'highFrequent.inbatch'</li><li>'concordant'</li><li>'totalDepth.Baseline'</li><li>'dscnt.Baseline.filtered'</li><li>'altDepth.Baseline'</li><li>'altDepth.Baseline.filtered'</li><li>'AF.Baseline'</li><li>'odds.ratio'</li><li>'pvalue'</li><li>'MAF.diff'</li><li>'finalKeep.old'</li><li>'SampleID.short'</li><li>'StudyVisit.x'</li><li>'SampleType.x'</li><li>'PatientID'</li><li>'StudyVisit.y'</li><li>'SampleType.y'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'seqnames'\n",
       "\\item 'start'\n",
       "\\item 'end'\n",
       "\\item 'width'\n",
       "\\item 'strand'\n",
       "\\item 'ref'\n",
       "\\item 'alt'\n",
       "\\item 'totalDepth'\n",
       "\\item 'refDepth'\n",
       "\\item 'altDepth'\n",
       "\\item 'sampleNames'\n",
       "\\item 'VariantFreq'\n",
       "\\item 'SYMBOL'\n",
       "\\item 'GeneID'\n",
       "\\item 'Feature'\n",
       "\\item 'HGVSc'\n",
       "\\item 'HGVSp'\n",
       "\\item 'Amino\\_acids'\n",
       "\\item 'Codons'\n",
       "\\item 'BIOTYPE'\n",
       "\\item 'EXON'\n",
       "\\item 'INTRON'\n",
       "\\item 'Consequence'\n",
       "\\item 'DISTANCE'\n",
       "\\item 'Existing\\_variation'\n",
       "\\item 'IMPACT'\n",
       "\\item 'VARIANT\\_CLASS'\n",
       "\\item 'CLIN\\_SIG'\n",
       "\\item 'Clinvar'\n",
       "\\item 'dbSNP'\n",
       "\\item 'COSMIC'\n",
       "\\item 'COSMIC.CNT'\n",
       "\\item 'genome1000'\n",
       "\\item 'AF'\n",
       "\\item 'CANONICAL'\n",
       "\\item 'filteredCnt'\n",
       "\\item 'DSCnt'\n",
       "\\item 'filteredDSCnt'\n",
       "\\item 'pseudoDSCnt'\n",
       "\\item 'altLOD'\n",
       "\\item 'Qscore'\n",
       "\\item 'entropy'\n",
       "\\item 'avgEntropy'\n",
       "\\item 'filtered\\_tlen\\_mean'\n",
       "\\item 'tlen.ratio'\n",
       "\\item 'filtered\\_toend\\_numunique'\n",
       "\\item 'filtered\\_total\\_base\\_count'\n",
       "\\item 'filterVarType'\n",
       "\\item 'repeatRegion'\n",
       "\\item 'white.list'\n",
       "\\item 'wl.tier'\n",
       "\\item 'MNV'\n",
       "\\item 'GC.Percent'\n",
       "\\item 'tandemRepeat'\n",
       "\\item 'AF.th'\n",
       "\\item 'AF.diff'\n",
       "\\item 'bgFilter'\n",
       "\\item 'generalFilter'\n",
       "\\item 'localVariantFilter'\n",
       "\\item 'ReadEndFilter'\n",
       "\\item 'DS.filter'\n",
       "\\item 'highQ.ratio'\n",
       "\\item 'fc.ds.ratio'\n",
       "\\item 'CopyNumber'\n",
       "\\item 'zScore.cnv'\n",
       "\\item 'coVariant'\n",
       "\\item 'ol.coVariant'\n",
       "\\item 'coVariantFilter'\n",
       "\\item 'sideVariant'\n",
       "\\item 'sideVariant.AF.ratio'\n",
       "\\item 'prevalenceInternal'\n",
       "\\item 'filterPrevalence'\n",
       "\\item 'filterScore'\n",
       "\\item 'finalKeep'\n",
       "\\item 'VariantID'\n",
       "\\item 'sampleDir'\n",
       "\\item 'VariantType'\n",
       "\\item 'VariantType.old'\n",
       "\\item 'FilterType'\n",
       "\\item 'sampleFolder'\n",
       "\\item 'externalSampleID'\n",
       "\\item 'ID'\n",
       "\\item 'highFrequent.inbatch'\n",
       "\\item 'concordant'\n",
       "\\item 'totalDepth.Baseline'\n",
       "\\item 'dscnt.Baseline.filtered'\n",
       "\\item 'altDepth.Baseline'\n",
       "\\item 'altDepth.Baseline.filtered'\n",
       "\\item 'AF.Baseline'\n",
       "\\item 'odds.ratio'\n",
       "\\item 'pvalue'\n",
       "\\item 'MAF.diff'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'SampleID.short'\n",
       "\\item 'StudyVisit.x'\n",
       "\\item 'SampleType.x'\n",
       "\\item 'PatientID'\n",
       "\\item 'StudyVisit.y'\n",
       "\\item 'SampleType.y'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'seqnames'\n",
       "2. 'start'\n",
       "3. 'end'\n",
       "4. 'width'\n",
       "5. 'strand'\n",
       "6. 'ref'\n",
       "7. 'alt'\n",
       "8. 'totalDepth'\n",
       "9. 'refDepth'\n",
       "10. 'altDepth'\n",
       "11. 'sampleNames'\n",
       "12. 'VariantFreq'\n",
       "13. 'SYMBOL'\n",
       "14. 'GeneID'\n",
       "15. 'Feature'\n",
       "16. 'HGVSc'\n",
       "17. 'HGVSp'\n",
       "18. 'Amino_acids'\n",
       "19. 'Codons'\n",
       "20. 'BIOTYPE'\n",
       "21. 'EXON'\n",
       "22. 'INTRON'\n",
       "23. 'Consequence'\n",
       "24. 'DISTANCE'\n",
       "25. 'Existing_variation'\n",
       "26. 'IMPACT'\n",
       "27. 'VARIANT_CLASS'\n",
       "28. 'CLIN_SIG'\n",
       "29. 'Clinvar'\n",
       "30. 'dbSNP'\n",
       "31. 'COSMIC'\n",
       "32. 'COSMIC.CNT'\n",
       "33. 'genome1000'\n",
       "34. 'AF'\n",
       "35. 'CANONICAL'\n",
       "36. 'filteredCnt'\n",
       "37. 'DSCnt'\n",
       "38. 'filteredDSCnt'\n",
       "39. 'pseudoDSCnt'\n",
       "40. 'altLOD'\n",
       "41. 'Qscore'\n",
       "42. 'entropy'\n",
       "43. 'avgEntropy'\n",
       "44. 'filtered_tlen_mean'\n",
       "45. 'tlen.ratio'\n",
       "46. 'filtered_toend_numunique'\n",
       "47. 'filtered_total_base_count'\n",
       "48. 'filterVarType'\n",
       "49. 'repeatRegion'\n",
       "50. 'white.list'\n",
       "51. 'wl.tier'\n",
       "52. 'MNV'\n",
       "53. 'GC.Percent'\n",
       "54. 'tandemRepeat'\n",
       "55. 'AF.th'\n",
       "56. 'AF.diff'\n",
       "57. 'bgFilter'\n",
       "58. 'generalFilter'\n",
       "59. 'localVariantFilter'\n",
       "60. 'ReadEndFilter'\n",
       "61. 'DS.filter'\n",
       "62. 'highQ.ratio'\n",
       "63. 'fc.ds.ratio'\n",
       "64. 'CopyNumber'\n",
       "65. 'zScore.cnv'\n",
       "66. 'coVariant'\n",
       "67. 'ol.coVariant'\n",
       "68. 'coVariantFilter'\n",
       "69. 'sideVariant'\n",
       "70. 'sideVariant.AF.ratio'\n",
       "71. 'prevalenceInternal'\n",
       "72. 'filterPrevalence'\n",
       "73. 'filterScore'\n",
       "74. 'finalKeep'\n",
       "75. 'VariantID'\n",
       "76. 'sampleDir'\n",
       "77. 'VariantType'\n",
       "78. 'VariantType.old'\n",
       "79. 'FilterType'\n",
       "80. 'sampleFolder'\n",
       "81. 'externalSampleID'\n",
       "82. 'ID'\n",
       "83. 'highFrequent.inbatch'\n",
       "84. 'concordant'\n",
       "85. 'totalDepth.Baseline'\n",
       "86. 'dscnt.Baseline.filtered'\n",
       "87. 'altDepth.Baseline'\n",
       "88. 'altDepth.Baseline.filtered'\n",
       "89. 'AF.Baseline'\n",
       "90. 'odds.ratio'\n",
       "91. 'pvalue'\n",
       "92. 'MAF.diff'\n",
       "93. 'finalKeep.old'\n",
       "94. 'SampleID.short'\n",
       "95. 'StudyVisit.x'\n",
       "96. 'SampleType.x'\n",
       "97. 'PatientID'\n",
       "98. 'StudyVisit.y'\n",
       "99. 'SampleType.y'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"seqnames\"                   \"start\"                     \n",
       " [3] \"end\"                        \"width\"                     \n",
       " [5] \"strand\"                     \"ref\"                       \n",
       " [7] \"alt\"                        \"totalDepth\"                \n",
       " [9] \"refDepth\"                   \"altDepth\"                  \n",
       "[11] \"sampleNames\"                \"VariantFreq\"               \n",
       "[13] \"SYMBOL\"                     \"GeneID\"                    \n",
       "[15] \"Feature\"                    \"HGVSc\"                     \n",
       "[17] \"HGVSp\"                      \"Amino_acids\"               \n",
       "[19] \"Codons\"                     \"BIOTYPE\"                   \n",
       "[21] \"EXON\"                       \"INTRON\"                    \n",
       "[23] \"Consequence\"                \"DISTANCE\"                  \n",
       "[25] \"Existing_variation\"         \"IMPACT\"                    \n",
       "[27] \"VARIANT_CLASS\"              \"CLIN_SIG\"                  \n",
       "[29] \"Clinvar\"                    \"dbSNP\"                     \n",
       "[31] \"COSMIC\"                     \"COSMIC.CNT\"                \n",
       "[33] \"genome1000\"                 \"AF\"                        \n",
       "[35] \"CANONICAL\"                  \"filteredCnt\"               \n",
       "[37] \"DSCnt\"                      \"filteredDSCnt\"             \n",
       "[39] \"pseudoDSCnt\"                \"altLOD\"                    \n",
       "[41] \"Qscore\"                     \"entropy\"                   \n",
       "[43] \"avgEntropy\"                 \"filtered_tlen_mean\"        \n",
       "[45] \"tlen.ratio\"                 \"filtered_toend_numunique\"  \n",
       "[47] \"filtered_total_base_count\"  \"filterVarType\"             \n",
       "[49] \"repeatRegion\"               \"white.list\"                \n",
       "[51] \"wl.tier\"                    \"MNV\"                       \n",
       "[53] \"GC.Percent\"                 \"tandemRepeat\"              \n",
       "[55] \"AF.th\"                      \"AF.diff\"                   \n",
       "[57] \"bgFilter\"                   \"generalFilter\"             \n",
       "[59] \"localVariantFilter\"         \"ReadEndFilter\"             \n",
       "[61] \"DS.filter\"                  \"highQ.ratio\"               \n",
       "[63] \"fc.ds.ratio\"                \"CopyNumber\"                \n",
       "[65] \"zScore.cnv\"                 \"coVariant\"                 \n",
       "[67] \"ol.coVariant\"               \"coVariantFilter\"           \n",
       "[69] \"sideVariant\"                \"sideVariant.AF.ratio\"      \n",
       "[71] \"prevalenceInternal\"         \"filterPrevalence\"          \n",
       "[73] \"filterScore\"                \"finalKeep\"                 \n",
       "[75] \"VariantID\"                  \"sampleDir\"                 \n",
       "[77] \"VariantType\"                \"VariantType.old\"           \n",
       "[79] \"FilterType\"                 \"sampleFolder\"              \n",
       "[81] \"externalSampleID\"           \"ID\"                        \n",
       "[83] \"highFrequent.inbatch\"       \"concordant\"                \n",
       "[85] \"totalDepth.Baseline\"        \"dscnt.Baseline.filtered\"   \n",
       "[87] \"altDepth.Baseline\"          \"altDepth.Baseline.filtered\"\n",
       "[89] \"AF.Baseline\"                \"odds.ratio\"                \n",
       "[91] \"pvalue\"                     \"MAF.diff\"                  \n",
       "[93] \"finalKeep.old\"              \"SampleID.short\"            \n",
       "[95] \"StudyVisit.x\"               \"SampleType.x\"              \n",
       "[97] \"PatientID\"                  \"StudyVisit.y\"              \n",
       "[99] \"SampleType.y\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#table(df.all.marked$)\n",
    "head(df.all.marked %>% filter(is.na(PatientID)))\n",
    "names(df.all.marked)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c6cc9c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All Variant Short Marked\n",
    "df1.ffpe = read.csv(\"./data_batch1_and_2/FFPE_WES/bg_rerun_09132023/\") \n",
    "df1.urine = read.csv(\"./data_batch1_and_2/Urine-UCP/Urine_post_analysis/urine_WES_only/WOP00253_WOP00254_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-09-13_urine_marked.csv\")\n",
    "df3.ffpe =read.csv(\"./data_batch3_02222023/PRDC-MOFFITT-NMIBC-22001-B_Variant_all_2023-05-03_ffpe_marked.csv\")\n",
    "df.combo = read.csv(\"./batch1_2_3_urine_WES_baselines/WOP00873_WOP00875_MCC-NMIBC_WES_urine_Variant_all_2023-07-26_urine_supernatant_marked.csv\")\n",
    "df.3.5.ffpe = read.csv(\"./benign_urothelium/WOP00981_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-09-01_ffpe_marked.csv\")\n",
    "df4.ffpe = read.csv(\"./Validation_cohort/Validation_FFPE/WOP01194_WOP01039_PRDC-MOFFITT-NMIBC-22001_Variant_all_2023-10-24_ffpe_marked.csv\")\n",
    "\n",
    "## TMB Marked\n",
    "\n",
    "## CNV\n",
    "\n",
    "## NGSQC updated\n",
    "\n",
    "## Fusion all\n",
    "\n",
    "## CNV all\n",
    "\n",
    "stop()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "########\n",
    "## unpatched data\n",
    "# batch 1+2\n",
    "all.snv_urine = read.csv(\"./data_batch1_and_2/Urine-UCP/Urine_post_analysis/urine_WES_only/WOP00253_WOP00254_PRDC-MOFFITT-NMIBC-22001_Variant_short_all_2023-09-13_urine_marked.csv\", as.is=T)\n",
    "# see also a marked file for batch 1 ucp\n",
    "all.snv_tissue = read.csv(file=\"./data_batch1_and_2/FFPE_WES/bg_rerun_09132023/PRDC-MOFFITT-NMIBC-22001_Variant_short_all_2023-09-13_tissue_marked.csv\",\n",
    "                       as.is=T) \n",
    "all.snv_a = merge.combine(standardize_names(rbind.common(all.snv_urine, all.snv_tissue), sid.format=\"none\"), df_wes_ngsqc.updated, join.type = \"left\", join.cols.left = \"SampleID.short\",\n",
    "                            join.cols.right=\"SampleID.short\", priority=\"right\") %>%\n",
    "                    mutate(PatientID = fix_PIDs(PatientID))\n",
    "# batch 3 ffpe\n",
    "all.ffpe_snv_b = read.csv(\"./data_batch3_02222023/PRDC-MOFFITT-NMIBC-22001-B_Variant_short_all_2023-05-03_ffpe_marked.csv\",\n",
    "                         as.is=T) %>% filter(SpecimenType==\"FFPE\") \n",
    "all.ffpe_snv_b = merge.combine(standardize_names(all.ffpe_snv_b, sid.format=\"none\"), df_wes_ngsqc.updated, join.type = \"left\", join.cols.left = \"SampleID.short\",\n",
    "                            join.cols.right=\"SampleID.short\", priority=\"right\") %>%\n",
    "                mutate(SampleType=fix_sampleTypes(SampleType)) %>%\n",
    "                    mutate(PatientID = fix_PIDs(PatientID))\n",
    "nrow(all.ffpe_snv_b)\n",
    "# additional urine WES+\n",
    "all.urine_wes_snv = read.csv(\"./batch1_2_3_urine_WES_baselines/WOP00873_WOP00875_MCC-NMIBC_WES_urine_Variant_short_all_2023-06-29_urine_supernatant_marked.csv\",\n",
    "                         as.is=T)\n",
    "all.urine_wes_snv = merge.combine(standardize_names(all.urine_wes_snv, sid.format=\"none\"), df_wes_ngsqc.updated, join.type = \"left\", join.cols.left = \"SampleID.short\",\n",
    "                            join.cols.right=\"SampleID.short\", priority=\"right\") %>%\n",
    "                    mutate(SampleType=fix_sampleTypes(SampleType)) %>%\n",
    "                    mutate(PatientID = fix_PIDs(PatientID))\n",
    "nrow(all.urine_wes_snv)\n",
    "# benign samples\n",
    "all.snv_benign = read.csv(\"./benign_urothelium/WOP00981_PRDC-MOFFITT-NMIBC-22001_Variant_short_all_2023-09-01_ffpe_marked.csv\", \n",
    "                          as.is=T)\n",
    "all.snv_benign = merge.combine(standardize_names(all.snv_benign, sid.format=\"none\"), df_wes_ngsqc.updated, join.type = \"left\", join.cols.left = \"SampleID.short\",\n",
    "                            join.cols.right=\"SampleID.short\", priority=\"right\") %>%\n",
    "                    mutate(SampleType=fix_sampleTypes(SampleType)) %>%\n",
    "                    mutate(PatientID = fix_PIDs(PatientID))\n",
    "\n",
    "all.snv = rbind.common(all.snv_a, all.ffpe_snv_b) \n",
    "all.snv = rbind.common(all.snv, all.urine_wes_snv)\n",
    "all.snv = rbind.common(all.snv, all.snv_benign)\n",
    "all.snv.wes =  all.snv %>%\n",
    "                filter(SampleID.short %in% df_wes_ngsqc.updated$SampleID.short) %>% distinct()\n",
    "nrow(all.snv.wes) #20654 all, \n",
    "all.snv.wes.somatic = all.snv.wes %>% filter(grepl(\"somatic\", VariantType, ignore.case=T))\n",
    "nrow(all.snv.wes.somatic) #9147 somatic\n",
    "\n",
    "print(\"wes short som unpatched\")\n",
    "table(all.snv.wes.somatic$PatientID, all.snv.wes.somatic$StudyVisit)\n",
    "\n",
    "#########\n",
    "#### patched wes data\n",
    "print(\"wes short germ/som patched\")\n",
    "all.snv.wes.patched = standardize_names(read.csv(\"hybrid_probe_mrd/NMIBC_All_baseline_variants_short_marked_patched_09-13-2023.csv\", as.is=T, check.names=F),\n",
    "                                        sid.format=\"none\")\n",
    "#print(unique(all.snv.wes.patched$SampleID.short[all.snv.wes.patched$SampleID.short %!in% df_wes_ngsqc.all$SampleID.short]))\n",
    "stopifnot(all(all.snv.wes.patched$SampleID.short %in% df_wes_ngsqc.updated$SampleID.short))\n",
    "all.snv.wes.patched =  merge.combine(all.snv.wes.patched, df_wes_ngsqc.updated, priority=\"right\", warn=TRUE) %>%\n",
    "                    mutate(PatientID = fix_PIDs(PatientID))\n",
    "all.snv.wes.patched = all.snv.wes.patched %>% \n",
    "                mutate(VariantID.p = paste(PatientID, VariantID, sep=\":\")) %>% distinct()\n",
    "nrow(all.snv.wes.patched) # 26726\n",
    "table(all.snv.wes.patched$SampleType, all.snv.wes.patched$StudyVisit)\n",
    "                       \n",
    "# filter to variants only in the marked as somatic in the original short file of WES+ variants\n",
    "print(\"wes short som patched\")\n",
    "orig.somatic.vars = all.snv.wes.patched %>% filter(grepl(\"somatic\", VariantType, ignore.case=T) & Patched==\"FALSE\") %>%\n",
    "                                            pull(VariantID.p)\n",
    "all.snv.wes.patched.somatic = all.snv.wes.patched %>% filter(VariantID.p %in% unique(orig.somatic.vars))\n",
    "table(all.snv.wes.patched.somatic$PatientID, all.snv.wes.patched.somatic$StudyVisit, useNA=\"always\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c3a81ea",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
