{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "13ae3801",
   "metadata": {},
   "source": [
    "## Combined Somatic and CHIP variant calling models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "5ae54225",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2024-02-01 10:11:46 PST\"\n",
      "R version 4.3.0 (2023-04-21)\n",
      "Platform: aarch64-apple-darwin20 (64-bit)\n",
      "Running under: macOS 14.1.1\n",
      "\n",
      "Matrix products: default\n",
      "BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib \n",
      "LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0\n",
      "\n",
      "locale:\n",
      "[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8\n",
      "\n",
      "time zone: America/Los_Angeles\n",
      "tzcode source: internal\n",
      "\n",
      "attached base packages:\n",
      "[1] grid      stats     graphics  grDevices utils     datasets  methods  \n",
      "[8] base     \n",
      "\n",
      "other attached packages:\n",
      " [1] ComplexHeatmap_2.16.0 maftools_2.16.0       eulerr_7.0.0         \n",
      " [4] glue_1.6.2            stringr_1.5.0         pROC_1.18.4          \n",
      " [7] caret_6.0-94          lattice_0.21-8        tidyr_1.3.0          \n",
      "[10] RColorBrewer_1.1-3    ggplot2_3.4.2         dplyr_1.1.2          \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] tidyselect_1.2.0     timeDate_4022.108    IRdisplay_1.1       \n",
      " [4] fastmap_1.1.1        digest_0.6.33        rpart_4.1.19        \n",
      " [7] timechange_0.2.0     lifecycle_1.0.3      cluster_2.1.4       \n",
      "[10] survival_3.5-5       magrittr_2.0.3       compiler_4.3.0      \n",
      "[13] rlang_1.1.1          tools_4.3.0          utf8_1.2.3          \n",
      "[16] data.table_1.14.8    plyr_1.8.8           repr_1.1.6          \n",
      "[19] pbdZMQ_0.3-9         withr_2.5.0          purrr_1.0.1         \n",
      "[22] BiocGenerics_0.46.0  nnet_7.3-19          stats4_4.3.0        \n",
      "[25] fansi_1.0.4          colorspace_2.1-0     future_1.33.0       \n",
      "[28] globals_0.16.2       scales_1.2.1         iterators_1.0.14    \n",
      "[31] MASS_7.3-60          cli_3.6.1            crayon_1.5.2        \n",
      "[34] generics_0.1.3       future.apply_1.11.0  reshape2_1.4.4      \n",
      "[37] rjson_0.2.21         DNAcopy_1.74.1       splines_4.3.0       \n",
      "[40] parallel_4.3.0       matrixStats_1.0.0    base64enc_0.1-3     \n",
      "[43] vctrs_0.6.3          hardhat_1.3.0        Matrix_1.6-0        \n",
      "[46] jsonlite_1.8.7       S4Vectors_0.38.1     IRanges_2.34.1      \n",
      "[49] GetoptLong_1.0.5     clue_0.3-64          listenv_0.9.0       \n",
      "[52] foreach_1.5.2        gower_1.0.1          recipes_1.0.6       \n",
      "[55] parallelly_1.36.0    codetools_0.2-19     shape_1.4.6         \n",
      "[58] lubridate_1.9.2      stringi_1.7.12       gtable_0.3.3        \n",
      "[61] munsell_0.5.0        tibble_3.2.1         pillar_1.9.0        \n",
      "[64] htmltools_0.5.5      ipred_0.9-14         IRkernel_1.3.2      \n",
      "[67] circlize_0.4.15      lava_1.7.2.1         R6_2.5.1            \n",
      "[70] doParallel_1.0.17    evaluate_0.21        png_0.1-8           \n",
      "[73] class_7.3-22         Rcpp_1.0.11          uuid_1.1-0          \n",
      "[76] nlme_3.1-162         prodlim_2023.03.31   ModelMetrics_1.2.2.2\n",
      "[79] pkgconfig_2.0.3      GlobalOptions_0.1.2 \n"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(RColorBrewer)\n",
    "#library(gridExtra)\n",
    "library(tidyr)\n",
    "library(caret)\n",
    "#library(ROCR)\n",
    "library(pROC)\n",
    "#library(knitr)\n",
    "\n",
    "source(\"~/Desktop/puffin/R/helper_functions.R\")\n",
    "source(\"~/Desktop/Moffitt_2022/fix_PIDs.R\")\n",
    "options(stringsAsFactors = FALSE)\n",
    "# set some defaults\n",
    "options(stringsAsFactors = FALSE)\n",
    "options(repr.matrix.max.cols=75, repr.matrix.max.rows=100)\n",
    "formals(table)$useNA <- \"always\"\n",
    "formals(write.csv)$row.names <- FALSE\n",
    "formals(write.csv)$as.is <- TRUE\n",
    "\n",
    "print(Sys.time())\n",
    "print(sessionInfo())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "bfbce239",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"173736\" \"172592\" \"175702\" \"89200\"  \"174438\" \"172956\" \"174280\" \"172109\"\n",
      " [9] \"170762\" \"170594\" \"172163\" \"173129\" \"171660\" \"175253\" \"171351\" \"175743\"\n",
      "[17] \"172422\" \"171126\" \"173657\" \"174688\" \"175827\" \"173149\" \"173983\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " CHIP mutations        Germline Likely germline  Likely somatic         Somatic \n",
       "             56           47018            1928            6629              13 \n",
       "           <NA> \n",
       "              0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "       CHIP mutations              Germline     Likely background \n",
       "                   33                 50450                  1216 \n",
       "Likely CHIP mutations       Likely germline        Likely somatic \n",
       "                   36                  1101                   600 \n",
       "              Somatic                  <NA> \n",
       "                 2208                     0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "55644"
      ],
      "text/latex": [
       "55644"
      ],
      "text/markdown": [
       "55644"
      ],
      "text/plain": [
       "[1] 55644"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Plasma   <NA> \n",
       " 55644      0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>PatientID</th><th scope=col>n_mutations</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>170594</td><td>2634</td></tr>\n",
       "\t<tr><td>170762</td><td>2448</td></tr>\n",
       "\t<tr><td>171126</td><td>2288</td></tr>\n",
       "\t<tr><td>171351</td><td>2515</td></tr>\n",
       "\t<tr><td>171660</td><td>2478</td></tr>\n",
       "\t<tr><td>172109</td><td>2359</td></tr>\n",
       "\t<tr><td>172163</td><td>2467</td></tr>\n",
       "\t<tr><td>172422</td><td>2379</td></tr>\n",
       "\t<tr><td>172592</td><td>2337</td></tr>\n",
       "\t<tr><td>172956</td><td>2461</td></tr>\n",
       "\t<tr><td>173129</td><td>2332</td></tr>\n",
       "\t<tr><td>173149</td><td>2296</td></tr>\n",
       "\t<tr><td>173657</td><td>2360</td></tr>\n",
       "\t<tr><td>173736</td><td>2438</td></tr>\n",
       "\t<tr><td>173983</td><td>2581</td></tr>\n",
       "\t<tr><td>174280</td><td>2405</td></tr>\n",
       "\t<tr><td>174438</td><td>2398</td></tr>\n",
       "\t<tr><td>174688</td><td>2552</td></tr>\n",
       "\t<tr><td>175253</td><td>2402</td></tr>\n",
       "\t<tr><td>175702</td><td>2401</td></tr>\n",
       "\t<tr><td>175743</td><td>2410</td></tr>\n",
       "\t<tr><td>175827</td><td>2276</td></tr>\n",
       "\t<tr><td>89200 </td><td>2427</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " PatientID & n\\_mutations\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 170594 & 2634\\\\\n",
       "\t 170762 & 2448\\\\\n",
       "\t 171126 & 2288\\\\\n",
       "\t 171351 & 2515\\\\\n",
       "\t 171660 & 2478\\\\\n",
       "\t 172109 & 2359\\\\\n",
       "\t 172163 & 2467\\\\\n",
       "\t 172422 & 2379\\\\\n",
       "\t 172592 & 2337\\\\\n",
       "\t 172956 & 2461\\\\\n",
       "\t 173129 & 2332\\\\\n",
       "\t 173149 & 2296\\\\\n",
       "\t 173657 & 2360\\\\\n",
       "\t 173736 & 2438\\\\\n",
       "\t 173983 & 2581\\\\\n",
       "\t 174280 & 2405\\\\\n",
       "\t 174438 & 2398\\\\\n",
       "\t 174688 & 2552\\\\\n",
       "\t 175253 & 2402\\\\\n",
       "\t 175702 & 2401\\\\\n",
       "\t 175743 & 2410\\\\\n",
       "\t 175827 & 2276\\\\\n",
       "\t 89200  & 2427\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 23 × 2\n",
       "\n",
       "| PatientID &lt;chr&gt; | n_mutations &lt;int&gt; |\n",
       "|---|---|\n",
       "| 170594 | 2634 |\n",
       "| 170762 | 2448 |\n",
       "| 171126 | 2288 |\n",
       "| 171351 | 2515 |\n",
       "| 171660 | 2478 |\n",
       "| 172109 | 2359 |\n",
       "| 172163 | 2467 |\n",
       "| 172422 | 2379 |\n",
       "| 172592 | 2337 |\n",
       "| 172956 | 2461 |\n",
       "| 173129 | 2332 |\n",
       "| 173149 | 2296 |\n",
       "| 173657 | 2360 |\n",
       "| 173736 | 2438 |\n",
       "| 173983 | 2581 |\n",
       "| 174280 | 2405 |\n",
       "| 174438 | 2398 |\n",
       "| 174688 | 2552 |\n",
       "| 175253 | 2402 |\n",
       "| 175702 | 2401 |\n",
       "| 175743 | 2410 |\n",
       "| 175827 | 2276 |\n",
       "| 89200  | 2427 |\n",
       "\n"
      ],
      "text/plain": [
       "   PatientID n_mutations\n",
       "1  170594    2634       \n",
       "2  170762    2448       \n",
       "3  171126    2288       \n",
       "4  171351    2515       \n",
       "5  171660    2478       \n",
       "6  172109    2359       \n",
       "7  172163    2467       \n",
       "8  172422    2379       \n",
       "9  172592    2337       \n",
       "10 172956    2461       \n",
       "11 173129    2332       \n",
       "12 173149    2296       \n",
       "13 173657    2360       \n",
       "14 173736    2438       \n",
       "15 173983    2581       \n",
       "16 174280    2405       \n",
       "17 174438    2398       \n",
       "18 174688    2552       \n",
       "19 175253    2402       \n",
       "20 175702    2401       \n",
       "21 175743    2410       \n",
       "22 175827    2276       \n",
       "23 89200     2427       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Buffy Coat       <NA> \n",
       "    591299          0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"173983\" \"170594\" \"172163\" \"172401\" \"172592\" \"172422\" \"172403\" \"173149\"\n",
      " [9] \"174438\" \"174688\" \"170762\" \"171351\" \"172109\" \"172956\" \"173129\" \"171660\"\n",
      "[17] \"173736\" \"174280\" \"173657\" \"175253\" \"89200\"  \"175702\" \"175743\" \"171126\"\n",
      "[25] \"175827\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55644</li><li>101</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55644\n",
       "\\item 101\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55644\n",
       "2. 101\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55644   101"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 101</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>seqnames</th><th scope=col>start</th><th scope=col>end</th><th scope=col>width</th><th scope=col>strand</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>totalDepth</th><th scope=col>refDepth</th><th scope=col>altDepth</th><th scope=col>sampleNames</th><th scope=col>VariantFreq</th><th scope=col>SYMBOL</th><th scope=col>GeneID</th><th scope=col>Feature</th><th scope=col>HGVSc</th><th scope=col>HGVSp</th><th scope=col>Amino_acids</th><th scope=col>Codons</th><th scope=col>BIOTYPE</th><th scope=col>EXON</th><th scope=col>INTRON</th><th scope=col>Consequence</th><th scope=col>DISTANCE</th><th scope=col>Existing_variation</th><th scope=col>IMPACT</th><th scope=col>VARIANT_CLASS</th><th scope=col>CLIN_SIG</th><th scope=col>Clinvar</th><th scope=col>dbSNP</th><th scope=col>COSMIC</th><th scope=col>COSMIC.CNT</th><th scope=col>genome1000</th><th scope=col>AF</th><th scope=col>CANONICAL</th><th scope=col>filteredCnt</th><th scope=col>DSCnt</th><th scope=col>⋯</th><th scope=col>CopyNumber</th><th scope=col>zScore.cnv</th><th scope=col>coVariant</th><th scope=col>ol.coVariant</th><th scope=col>snvmember</th><th scope=col>mergedVariant</th><th scope=col>coVariantFilter</th><th scope=col>sideVariant</th><th scope=col>sideVariant.AF.ratio</th><th scope=col>prevalenceInternal</th><th scope=col>filterPrevalence</th><th scope=col>filterScore</th><th scope=col>finalKeep</th><th scope=col>VariantID</th><th scope=col>sampleDir</th><th scope=col>VariantType</th><th scope=col>depth</th><th scope=col>FilterType</th><th scope=col>sampleFolder</th><th scope=col>PatientID</th><th scope=col>externalSampleID</th><th scope=col>trialVisitNum</th><th scope=col>SpecimenType</th><th scope=col>ID</th><th scope=col>highFrequent.inbatch</th><th scope=col>concordant</th><th scope=col>PatientID.old</th><th scope=col>totalDepth.Baseline</th><th scope=col>dscnt.Baseline.filtered</th><th scope=col>altDepth.Baseline</th><th scope=col>altDepth.Baseline.filtered</th><th scope=col>AF.Baseline</th><th scope=col>odds.ratio</th><th scope=col>pvalue</th><th scope=col>MAF.diff</th><th scope=col>VariantType.old</th><th scope=col>finalKeep.old</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>chr1</td><td> 11187893</td><td> 11187893</td><td>1</td><td>*</td><td>T</td><td><span style=white-space:pre-wrap>C  </span></td><td>3769</td><td><span style=white-space:pre-wrap>   4</span></td><td>3765</td><td>P224641</td><td>99.89387</td><td>MTOR </td><td><span style=white-space:pre-wrap>  2475</span></td><td><span style=white-space:pre-wrap>NM_004958.4   </span></td><td><span style=white-space:pre-wrap>c.6034-30A&gt;G          </span></td><td><span style=white-space:pre-wrap>p.(=)     </span></td><td><span style=white-space:pre-wrap>NA      </span></td><td><span style=white-space:pre-wrap>NA     </span></td><td>protein_coding</td><td><span style=white-space:pre-wrap>NA     </span></td><td>'43/57'</td><td><span style=white-space:pre-wrap>intron_variant     </span></td><td>NA</td><td><span style=white-space:pre-wrap>rs1770344             </span></td><td>MODIFIER</td><td><span style=white-space:pre-wrap>SNV      </span></td><td><span style=white-space:pre-wrap>NA    </span></td><td><span style=white-space:pre-wrap>NA      </span></td><td><span style=white-space:pre-wrap>NA       </span></td><td><span style=white-space:pre-wrap>NA         </span></td><td>NA</td><td>rs1770344 </td><td>1.0000</td><td>yes</td><td>2991</td><td>2500</td><td>⋯</td><td>2.28</td><td>0.184</td><td><span style=white-space:pre-wrap>                                                                                   </span></td><td><span style=white-space:pre-wrap>                     </span></td><td>NA</td><td></td><td>TRUE</td><td><span style=white-space:pre-wrap>                          </span></td><td><span style=white-space:pre-wrap>  NA</span></td><td>0.0584</td><td>TRUE</td><td>8.475</td><td>FALSE</td><td><span style=white-space:pre-wrap>chr1:11187893:T:C   </span></td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td><span style=white-space:pre-wrap>P224641:chr1:11187893:T:C   </span></td><td>23</td><td>FALSE</td><td>173736</td><td> 922</td><td>28</td><td> 922</td><td>736</td><td>100.0</td><td>0.9989388</td><td>0.5185111</td><td>-0.10612895</td><td>Germline</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>chr1</td><td> 11205058</td><td> 11205058</td><td>1</td><td>*</td><td>C</td><td><span style=white-space:pre-wrap>T  </span></td><td>3506</td><td><span style=white-space:pre-wrap>  26</span></td><td>3480</td><td>P224641</td><td>99.25841</td><td>MTOR </td><td><span style=white-space:pre-wrap>  2475</span></td><td><span style=white-space:pre-wrap>NM_004958.4   </span></td><td><span style=white-space:pre-wrap>c.4731G&gt;A             </span></td><td>p.Ala1577=</td><td>p.A1577A</td><td>gcG/gcA</td><td>protein_coding</td><td>'33/58'</td><td><span style=white-space:pre-wrap>NA     </span></td><td>synonymous_variant </td><td>NA</td><td>rs1057079&amp;COSV63875443</td><td><span style=white-space:pre-wrap>LOW     </span></td><td><span style=white-space:pre-wrap>SNV      </span></td><td>benign</td><td>Benign:1</td><td>rs1057079</td><td>COSM4142146</td><td>3 </td><td>rs1057079 </td><td>0.8165</td><td>yes</td><td>3001</td><td>1943</td><td>⋯</td><td>2.12</td><td>0.265</td><td><span style=white-space:pre-wrap>                                                                                   </span></td><td><span style=white-space:pre-wrap>                     </span></td><td>NA</td><td></td><td>TRUE</td><td><span style=white-space:pre-wrap>                          </span></td><td><span style=white-space:pre-wrap>  NA</span></td><td>0.9180</td><td>TRUE</td><td>9.800</td><td>FALSE</td><td><span style=white-space:pre-wrap>chr1:11205058:C:T   </span></td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td><span style=white-space:pre-wrap>P224641:chr1:11205058:C:T   </span></td><td>21</td><td>FALSE</td><td>173736</td><td> 512</td><td>13</td><td> 508</td><td>408</td><td> 99.2</td><td>1.0004027</td><td>0.5110042</td><td> 0.05841415</td><td>Germline</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>chr1</td><td> 11288758</td><td> 11288758</td><td>1</td><td>*</td><td>G</td><td><span style=white-space:pre-wrap>A  </span></td><td>3818</td><td><span style=white-space:pre-wrap>  27</span></td><td>3791</td><td>P224641</td><td>99.29282</td><td>MTOR </td><td><span style=white-space:pre-wrap>  2475</span></td><td><span style=white-space:pre-wrap>NM_004958.4   </span></td><td><span style=white-space:pre-wrap>c.2997C&gt;T             </span></td><td>p.Asn999= </td><td>p.N999N </td><td>aaC/aaT</td><td>protein_coding</td><td>'19/58'</td><td><span style=white-space:pre-wrap>NA     </span></td><td>synonymous_variant </td><td>NA</td><td>rs1064261&amp;COSV63873449</td><td><span style=white-space:pre-wrap>LOW     </span></td><td><span style=white-space:pre-wrap>SNV      </span></td><td><span style=white-space:pre-wrap>NA    </span></td><td><span style=white-space:pre-wrap>NA      </span></td><td><span style=white-space:pre-wrap>NA       </span></td><td>COSM4142152</td><td>2 </td><td>rs1064261 </td><td>0.9191</td><td>yes</td><td>3171</td><td>2524</td><td>⋯</td><td>2.30</td><td>0.846</td><td>chr1:11288618:G14G124G:A14C124A;chr1:11288633:G124G:C124A;chr1:11288618:G139G:A139A</td><td>0.17;0.00422;0.000315</td><td>NA</td><td></td><td>TRUE</td><td><span style=white-space:pre-wrap>                          </span></td><td><span style=white-space:pre-wrap>  NA</span></td><td>0.9290</td><td>TRUE</td><td>9.050</td><td>FALSE</td><td><span style=white-space:pre-wrap>chr1:11288758:G:A   </span></td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td><span style=white-space:pre-wrap>P224641:chr1:11288758:G:A   </span></td><td>20</td><td>FALSE</td><td>173736</td><td>1013</td><td>31</td><td>1008</td><td>844</td><td> 99.5</td><td>0.9978536</td><td>0.5270965</td><td>-0.20717653</td><td>Germline</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>chr1</td><td> 11301714</td><td> 11301714</td><td>1</td><td>*</td><td>A</td><td><span style=white-space:pre-wrap>G  </span></td><td>2724</td><td><span style=white-space:pre-wrap>   7</span></td><td>2717</td><td>P224641</td><td>99.74302</td><td>MTOR </td><td><span style=white-space:pre-wrap>  2475</span></td><td><span style=white-space:pre-wrap>NM_004958.4   </span></td><td><span style=white-space:pre-wrap>c.1437T&gt;C             </span></td><td>p.Asp479= </td><td>p.D479D </td><td>gaT/gaC</td><td>protein_coding</td><td>'10/58'</td><td><span style=white-space:pre-wrap>NA     </span></td><td>synonymous_variant </td><td>NA</td><td>rs1135172&amp;COSV63873456</td><td><span style=white-space:pre-wrap>LOW     </span></td><td><span style=white-space:pre-wrap>SNV      </span></td><td>benign</td><td>Benign:1</td><td>rs1135172</td><td>COSM4142157</td><td>2 </td><td>rs1135172 </td><td>0.8789</td><td>yes</td><td>1917</td><td>1688</td><td>⋯</td><td>2.07</td><td>1.050</td><td><span style=white-space:pre-wrap>                                                                                   </span></td><td><span style=white-space:pre-wrap>                     </span></td><td>NA</td><td></td><td>TRUE</td><td><span style=white-space:pre-wrap>                          </span></td><td><span style=white-space:pre-wrap>  NA</span></td><td>0.9360</td><td>TRUE</td><td>8.975</td><td>FALSE</td><td><span style=white-space:pre-wrap>chr1:11301714:A:G   </span></td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td><span style=white-space:pre-wrap>P224641:chr1:11301714:A:G   </span></td><td>20</td><td>FALSE</td><td>173736</td><td> 752</td><td>16</td><td> 752</td><td>648</td><td>100.0</td><td>0.9974305</td><td>0.5292151</td><td>-0.25697504</td><td>Germline</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>chr1</td><td>114224782</td><td>114224782</td><td>1</td><td>*</td><td>A</td><td>ATT</td><td>3391</td><td>2276</td><td>1115</td><td>P224641</td><td>32.88116</td><td>MAGI3</td><td>260425</td><td>NM_001142782.2</td><td>c.3329-722_3329-721dup</td><td>p.(=)     </td><td>NA      </td><td>NA     </td><td>protein_coding</td><td>NA     </td><td>'20/20'</td><td>intron_variant     </td><td>NA</td><td>rs11438160            </td><td>MODIFIER</td><td>insertion</td><td>NA    </td><td>NA      </td><td>NA       </td><td>NA         </td><td>NA</td><td>rs11438160</td><td>0.1770</td><td>YES</td><td> 892</td><td> 394</td><td>⋯</td><td>3.12</td><td>5.430</td><td>                                                                                   </td><td>                     </td><td>NA</td><td></td><td>TRUE</td><td>adj:chr1:114224782:A:AT:12</td><td>2.74</td><td>0.0231</td><td>TRUE</td><td>3.030</td><td>FALSE</td><td>chr1:114224782:A:ATT</td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td>P224641:chr1:114224782:A:ATT</td><td> 4</td><td>FALSE</td><td>173736</td><td> 396</td><td> 2</td><td> 134</td><td>118</td><td> 33.8</td><td>0.9717162</td><td>0.6296152</td><td>-0.91884400</td><td>Germline</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>chr1</td><td>117690272</td><td>117690272</td><td>1</td><td>*</td><td>G</td><td><span style=white-space:pre-wrap>A  </span></td><td>2978</td><td><span style=white-space:pre-wrap>  11</span></td><td>2967</td><td>P224641</td><td>99.63062</td><td>VTCN1</td><td> 79679</td><td><span style=white-space:pre-wrap>NM_024626.4   </span></td><td><span style=white-space:pre-wrap>c.*8C&gt;T               </span></td><td><span style=white-space:pre-wrap>p.(=)     </span></td><td><span style=white-space:pre-wrap>NA      </span></td><td><span style=white-space:pre-wrap>NA     </span></td><td>protein_coding</td><td><span style=white-space:pre-wrap>'5/6'  </span></td><td><span style=white-space:pre-wrap>NA     </span></td><td>3_prime_UTR_variant</td><td>NA</td><td><span style=white-space:pre-wrap>rs10754339            </span></td><td>MODIFIER</td><td><span style=white-space:pre-wrap>SNV      </span></td><td><span style=white-space:pre-wrap>NA    </span></td><td><span style=white-space:pre-wrap>NA      </span></td><td><span style=white-space:pre-wrap>NA       </span></td><td><span style=white-space:pre-wrap>NA         </span></td><td>NA</td><td>rs10754339</td><td>0.9586</td><td>yes</td><td>2735</td><td>1500</td><td>⋯</td><td>2.04</td><td>0.532</td><td><span style=white-space:pre-wrap>                                                                                   </span></td><td><span style=white-space:pre-wrap>                     </span></td><td>NA</td><td></td><td>TRUE</td><td><span style=white-space:pre-wrap>                          </span></td><td><span style=white-space:pre-wrap>  NA</span></td><td>0.9800</td><td>TRUE</td><td>9.050</td><td>FALSE</td><td><span style=white-space:pre-wrap>chr1:117690272:G:A  </span></td><td>/prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3</td><td>Germline</td><td>high</td><td>otherFinal</td><td>P224641_221123</td><td>173736</td><td>06S22025850</td><td>Pre Op</td><td>Plasma</td><td><span style=white-space:pre-wrap>P224641:chr1:117690272:G:A  </span></td><td>23</td><td>FALSE</td><td>173736</td><td> 502</td><td> 8</td><td> 500</td><td>411</td><td> 99.6</td><td>1.0003092</td><td>0.5119251</td><td> 0.03062458</td><td>Germline</td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 101\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & X & seqnames & start & end & width & strand & ref & alt & totalDepth & refDepth & altDepth & sampleNames & VariantFreq & SYMBOL & GeneID & Feature & HGVSc & HGVSp & Amino\\_acids & Codons & BIOTYPE & EXON & INTRON & Consequence & DISTANCE & Existing\\_variation & IMPACT & VARIANT\\_CLASS & CLIN\\_SIG & Clinvar & dbSNP & COSMIC & COSMIC.CNT & genome1000 & AF & CANONICAL & filteredCnt & DSCnt & ⋯ & CopyNumber & zScore.cnv & coVariant & ol.coVariant & snvmember & mergedVariant & coVariantFilter & sideVariant & sideVariant.AF.ratio & prevalenceInternal & filterPrevalence & filterScore & finalKeep & VariantID & sampleDir & VariantType & depth & FilterType & sampleFolder & PatientID & externalSampleID & trialVisitNum & SpecimenType & ID & highFrequent.inbatch & concordant & PatientID.old & totalDepth.Baseline & dscnt.Baseline.filtered & altDepth.Baseline & altDepth.Baseline.filtered & AF.Baseline & odds.ratio & pvalue & MAF.diff & VariantType.old & finalKeep.old\\\\\n",
       "  & <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <int> & ⋯ & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <lgl> & <chr> & <dbl> & <dbl> & <lgl> & <dbl> & <lgl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <lgl> & <int> & <dbl> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & chr1 &  11187893 &  11187893 & 1 & * & T & C   & 3769 &    4 & 3765 & P224641 & 99.89387 & MTOR  &   2475 & NM\\_004958.4    & c.6034-30A>G           & p.(=)      & NA       & NA      & protein\\_coding & NA      & '43/57' & intron\\_variant      & NA & rs1770344              & MODIFIER & SNV       & NA     & NA       & NA        & NA          & NA & rs1770344  & 1.0000 & yes & 2991 & 2500 & ⋯ & 2.28 & 0.184 &                                                                                     &                       & NA &  & TRUE &                            &   NA & 0.0584 & TRUE & 8.475 & FALSE & chr1:11187893:T:C    & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:11187893:T:C    & 23 & FALSE & 173736 &  922 & 28 &  922 & 736 & 100.0 & 0.9989388 & 0.5185111 & -0.10612895 & Germline & TRUE\\\\\n",
       "\t2 & 2 & chr1 &  11205058 &  11205058 & 1 & * & C & T   & 3506 &   26 & 3480 & P224641 & 99.25841 & MTOR  &   2475 & NM\\_004958.4    & c.4731G>A              & p.Ala1577= & p.A1577A & gcG/gcA & protein\\_coding & '33/58' & NA      & synonymous\\_variant  & NA & rs1057079\\&COSV63875443 & LOW      & SNV       & benign & Benign:1 & rs1057079 & COSM4142146 & 3  & rs1057079  & 0.8165 & yes & 3001 & 1943 & ⋯ & 2.12 & 0.265 &                                                                                     &                       & NA &  & TRUE &                            &   NA & 0.9180 & TRUE & 9.800 & FALSE & chr1:11205058:C:T    & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:11205058:C:T    & 21 & FALSE & 173736 &  512 & 13 &  508 & 408 &  99.2 & 1.0004027 & 0.5110042 &  0.05841415 & Germline & TRUE\\\\\n",
       "\t3 & 3 & chr1 &  11288758 &  11288758 & 1 & * & G & A   & 3818 &   27 & 3791 & P224641 & 99.29282 & MTOR  &   2475 & NM\\_004958.4    & c.2997C>T              & p.Asn999=  & p.N999N  & aaC/aaT & protein\\_coding & '19/58' & NA      & synonymous\\_variant  & NA & rs1064261\\&COSV63873449 & LOW      & SNV       & NA     & NA       & NA        & COSM4142152 & 2  & rs1064261  & 0.9191 & yes & 3171 & 2524 & ⋯ & 2.30 & 0.846 & chr1:11288618:G14G124G:A14C124A;chr1:11288633:G124G:C124A;chr1:11288618:G139G:A139A & 0.17;0.00422;0.000315 & NA &  & TRUE &                            &   NA & 0.9290 & TRUE & 9.050 & FALSE & chr1:11288758:G:A    & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:11288758:G:A    & 20 & FALSE & 173736 & 1013 & 31 & 1008 & 844 &  99.5 & 0.9978536 & 0.5270965 & -0.20717653 & Germline & TRUE\\\\\n",
       "\t4 & 4 & chr1 &  11301714 &  11301714 & 1 & * & A & G   & 2724 &    7 & 2717 & P224641 & 99.74302 & MTOR  &   2475 & NM\\_004958.4    & c.1437T>C              & p.Asp479=  & p.D479D  & gaT/gaC & protein\\_coding & '10/58' & NA      & synonymous\\_variant  & NA & rs1135172\\&COSV63873456 & LOW      & SNV       & benign & Benign:1 & rs1135172 & COSM4142157 & 2  & rs1135172  & 0.8789 & yes & 1917 & 1688 & ⋯ & 2.07 & 1.050 &                                                                                     &                       & NA &  & TRUE &                            &   NA & 0.9360 & TRUE & 8.975 & FALSE & chr1:11301714:A:G    & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:11301714:A:G    & 20 & FALSE & 173736 &  752 & 16 &  752 & 648 & 100.0 & 0.9974305 & 0.5292151 & -0.25697504 & Germline & TRUE\\\\\n",
       "\t5 & 5 & chr1 & 114224782 & 114224782 & 1 & * & A & ATT & 3391 & 2276 & 1115 & P224641 & 32.88116 & MAGI3 & 260425 & NM\\_001142782.2 & c.3329-722\\_3329-721dup & p.(=)      & NA       & NA      & protein\\_coding & NA      & '20/20' & intron\\_variant      & NA & rs11438160             & MODIFIER & insertion & NA     & NA       & NA        & NA          & NA & rs11438160 & 0.1770 & YES &  892 &  394 & ⋯ & 3.12 & 5.430 &                                                                                     &                       & NA &  & TRUE & adj:chr1:114224782:A:AT:12 & 2.74 & 0.0231 & TRUE & 3.030 & FALSE & chr1:114224782:A:ATT & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:114224782:A:ATT &  4 & FALSE & 173736 &  396 &  2 &  134 & 118 &  33.8 & 0.9717162 & 0.6296152 & -0.91884400 & Germline & TRUE\\\\\n",
       "\t6 & 6 & chr1 & 117690272 & 117690272 & 1 & * & G & A   & 2978 &   11 & 2967 & P224641 & 99.63062 & VTCN1 &  79679 & NM\\_024626.4    & c.*8C>T                & p.(=)      & NA       & NA      & protein\\_coding & '5/6'   & NA      & 3\\_prime\\_UTR\\_variant & NA & rs10754339             & MODIFIER & SNV       & NA     & NA       & NA        & NA          & NA & rs10754339 & 0.9586 & yes & 2735 & 1500 & ⋯ & 2.04 & 0.532 &                                                                                     &                       & NA &  & TRUE &                            &   NA & 0.9800 & TRUE & 9.050 & FALSE & chr1:117690272:G:A   & /prednet/bdata21/OutputByRunB21/221123\\_A00934\\_0231\\_BHLL7KDSX5\\_runtask\\_221126000739\\_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 & Germline & high & otherFinal & P224641\\_221123 & 173736 & 06S22025850 & Pre Op & Plasma & P224641:chr1:117690272:G:A   & 23 & FALSE & 173736 &  502 &  8 &  500 & 411 &  99.6 & 1.0003092 & 0.5119251 &  0.03062458 & Germline & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 101\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | seqnames &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | width &lt;int&gt; | strand &lt;chr&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | totalDepth &lt;int&gt; | refDepth &lt;int&gt; | altDepth &lt;int&gt; | sampleNames &lt;chr&gt; | VariantFreq &lt;dbl&gt; | SYMBOL &lt;chr&gt; | GeneID &lt;int&gt; | Feature &lt;chr&gt; | HGVSc &lt;chr&gt; | HGVSp &lt;chr&gt; | Amino_acids &lt;chr&gt; | Codons &lt;chr&gt; | BIOTYPE &lt;chr&gt; | EXON &lt;chr&gt; | INTRON &lt;chr&gt; | Consequence &lt;chr&gt; | DISTANCE &lt;int&gt; | Existing_variation &lt;chr&gt; | IMPACT &lt;chr&gt; | VARIANT_CLASS &lt;chr&gt; | CLIN_SIG &lt;chr&gt; | Clinvar &lt;chr&gt; | dbSNP &lt;chr&gt; | COSMIC &lt;chr&gt; | COSMIC.CNT &lt;chr&gt; | genome1000 &lt;chr&gt; | AF &lt;dbl&gt; | CANONICAL &lt;chr&gt; | filteredCnt &lt;dbl&gt; | DSCnt &lt;int&gt; | ⋯ ⋯ | CopyNumber &lt;dbl&gt; | zScore.cnv &lt;dbl&gt; | coVariant &lt;chr&gt; | ol.coVariant &lt;chr&gt; | snvmember &lt;chr&gt; | mergedVariant &lt;chr&gt; | coVariantFilter &lt;lgl&gt; | sideVariant &lt;chr&gt; | sideVariant.AF.ratio &lt;dbl&gt; | prevalenceInternal &lt;dbl&gt; | filterPrevalence &lt;lgl&gt; | filterScore &lt;dbl&gt; | finalKeep &lt;lgl&gt; | VariantID &lt;chr&gt; | sampleDir &lt;chr&gt; | VariantType &lt;chr&gt; | depth &lt;chr&gt; | FilterType &lt;chr&gt; | sampleFolder &lt;chr&gt; | PatientID &lt;chr&gt; | externalSampleID &lt;chr&gt; | trialVisitNum &lt;chr&gt; | SpecimenType &lt;chr&gt; | ID &lt;chr&gt; | highFrequent.inbatch &lt;int&gt; | concordant &lt;lgl&gt; | PatientID.old &lt;int&gt; | totalDepth.Baseline &lt;dbl&gt; | dscnt.Baseline.filtered &lt;int&gt; | altDepth.Baseline &lt;int&gt; | altDepth.Baseline.filtered &lt;dbl&gt; | AF.Baseline &lt;dbl&gt; | odds.ratio &lt;dbl&gt; | pvalue &lt;dbl&gt; | MAF.diff &lt;dbl&gt; | VariantType.old &lt;chr&gt; | finalKeep.old &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | chr1 |  11187893 |  11187893 | 1 | * | T | C   | 3769 |    4 | 3765 | P224641 | 99.89387 | MTOR  |   2475 | NM_004958.4    | c.6034-30A&gt;G           | p.(=)      | NA       | NA      | protein_coding | NA      | '43/57' | intron_variant      | NA | rs1770344              | MODIFIER | SNV       | NA     | NA       | NA        | NA          | NA | rs1770344  | 1.0000 | yes | 2991 | 2500 | ⋯ | 2.28 | 0.184 | <!----> | <!----> | NA | <!----> | TRUE | <!----> |   NA | 0.0584 | TRUE | 8.475 | FALSE | chr1:11187893:T:C    | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:11187893:T:C    | 23 | FALSE | 173736 |  922 | 28 |  922 | 736 | 100.0 | 0.9989388 | 0.5185111 | -0.10612895 | Germline | TRUE |\n",
       "| 2 | 2 | chr1 |  11205058 |  11205058 | 1 | * | C | T   | 3506 |   26 | 3480 | P224641 | 99.25841 | MTOR  |   2475 | NM_004958.4    | c.4731G&gt;A              | p.Ala1577= | p.A1577A | gcG/gcA | protein_coding | '33/58' | NA      | synonymous_variant  | NA | rs1057079&amp;COSV63875443 | LOW      | SNV       | benign | Benign:1 | rs1057079 | COSM4142146 | 3  | rs1057079  | 0.8165 | yes | 3001 | 1943 | ⋯ | 2.12 | 0.265 | <!----> | <!----> | NA | <!----> | TRUE | <!----> |   NA | 0.9180 | TRUE | 9.800 | FALSE | chr1:11205058:C:T    | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:11205058:C:T    | 21 | FALSE | 173736 |  512 | 13 |  508 | 408 |  99.2 | 1.0004027 | 0.5110042 |  0.05841415 | Germline | TRUE |\n",
       "| 3 | 3 | chr1 |  11288758 |  11288758 | 1 | * | G | A   | 3818 |   27 | 3791 | P224641 | 99.29282 | MTOR  |   2475 | NM_004958.4    | c.2997C&gt;T              | p.Asn999=  | p.N999N  | aaC/aaT | protein_coding | '19/58' | NA      | synonymous_variant  | NA | rs1064261&amp;COSV63873449 | LOW      | SNV       | NA     | NA       | NA        | COSM4142152 | 2  | rs1064261  | 0.9191 | yes | 3171 | 2524 | ⋯ | 2.30 | 0.846 | chr1:11288618:G14G124G:A14C124A;chr1:11288633:G124G:C124A;chr1:11288618:G139G:A139A | 0.17;0.00422;0.000315 | NA | <!----> | TRUE | <!----> |   NA | 0.9290 | TRUE | 9.050 | FALSE | chr1:11288758:G:A    | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:11288758:G:A    | 20 | FALSE | 173736 | 1013 | 31 | 1008 | 844 |  99.5 | 0.9978536 | 0.5270965 | -0.20717653 | Germline | TRUE |\n",
       "| 4 | 4 | chr1 |  11301714 |  11301714 | 1 | * | A | G   | 2724 |    7 | 2717 | P224641 | 99.74302 | MTOR  |   2475 | NM_004958.4    | c.1437T&gt;C              | p.Asp479=  | p.D479D  | gaT/gaC | protein_coding | '10/58' | NA      | synonymous_variant  | NA | rs1135172&amp;COSV63873456 | LOW      | SNV       | benign | Benign:1 | rs1135172 | COSM4142157 | 2  | rs1135172  | 0.8789 | yes | 1917 | 1688 | ⋯ | 2.07 | 1.050 | <!----> | <!----> | NA | <!----> | TRUE | <!----> |   NA | 0.9360 | TRUE | 8.975 | FALSE | chr1:11301714:A:G    | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:11301714:A:G    | 20 | FALSE | 173736 |  752 | 16 |  752 | 648 | 100.0 | 0.9974305 | 0.5292151 | -0.25697504 | Germline | TRUE |\n",
       "| 5 | 5 | chr1 | 114224782 | 114224782 | 1 | * | A | ATT | 3391 | 2276 | 1115 | P224641 | 32.88116 | MAGI3 | 260425 | NM_001142782.2 | c.3329-722_3329-721dup | p.(=)      | NA       | NA      | protein_coding | NA      | '20/20' | intron_variant      | NA | rs11438160             | MODIFIER | insertion | NA     | NA       | NA        | NA          | NA | rs11438160 | 0.1770 | YES |  892 |  394 | ⋯ | 3.12 | 5.430 | <!----> | <!----> | NA | <!----> | TRUE | adj:chr1:114224782:A:AT:12 | 2.74 | 0.0231 | TRUE | 3.030 | FALSE | chr1:114224782:A:ATT | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:114224782:A:ATT |  4 | FALSE | 173736 |  396 |  2 |  134 | 118 |  33.8 | 0.9717162 | 0.6296152 | -0.91884400 | Germline | TRUE |\n",
       "| 6 | 6 | chr1 | 117690272 | 117690272 | 1 | * | G | A   | 2978 |   11 | 2967 | P224641 | 99.63062 | VTCN1 |  79679 | NM_024626.4    | c.*8C&gt;T                | p.(=)      | NA       | NA      | protein_coding | '5/6'   | NA      | 3_prime_UTR_variant | NA | rs10754339             | MODIFIER | SNV       | NA     | NA       | NA        | NA          | NA | rs10754339 | 0.9586 | yes | 2735 | 1500 | ⋯ | 2.04 | 0.532 | <!----> | <!----> | NA | <!----> | TRUE | <!----> |   NA | 0.9800 | TRUE | 9.050 | FALSE | chr1:117690272:G:A   | /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3 | Germline | high | otherFinal | P224641_221123 | 173736 | 06S22025850 | Pre Op | Plasma | P224641:chr1:117690272:G:A   | 23 | FALSE | 173736 |  502 |  8 |  500 | 411 |  99.6 | 1.0003092 | 0.5119251 |  0.03062458 | Germline | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "  X seqnames start     end       width strand ref alt totalDepth refDepth\n",
       "1 1 chr1      11187893  11187893 1     *      T   C   3769          4    \n",
       "2 2 chr1      11205058  11205058 1     *      C   T   3506         26    \n",
       "3 3 chr1      11288758  11288758 1     *      G   A   3818         27    \n",
       "4 4 chr1      11301714  11301714 1     *      A   G   2724          7    \n",
       "5 5 chr1     114224782 114224782 1     *      A   ATT 3391       2276    \n",
       "6 6 chr1     117690272 117690272 1     *      G   A   2978         11    \n",
       "  altDepth sampleNames VariantFreq SYMBOL GeneID Feature       \n",
       "1 3765     P224641     99.89387    MTOR     2475 NM_004958.4   \n",
       "2 3480     P224641     99.25841    MTOR     2475 NM_004958.4   \n",
       "3 3791     P224641     99.29282    MTOR     2475 NM_004958.4   \n",
       "4 2717     P224641     99.74302    MTOR     2475 NM_004958.4   \n",
       "5 1115     P224641     32.88116    MAGI3  260425 NM_001142782.2\n",
       "6 2967     P224641     99.63062    VTCN1   79679 NM_024626.4   \n",
       "  HGVSc                  HGVSp      Amino_acids Codons  BIOTYPE        EXON   \n",
       "1 c.6034-30A>G           p.(=)      NA          NA      protein_coding NA     \n",
       "2 c.4731G>A              p.Ala1577= p.A1577A    gcG/gcA protein_coding '33/58'\n",
       "3 c.2997C>T              p.Asn999=  p.N999N     aaC/aaT protein_coding '19/58'\n",
       "4 c.1437T>C              p.Asp479=  p.D479D     gaT/gaC protein_coding '10/58'\n",
       "5 c.3329-722_3329-721dup p.(=)      NA          NA      protein_coding NA     \n",
       "6 c.*8C>T                p.(=)      NA          NA      protein_coding '5/6'  \n",
       "  INTRON  Consequence         DISTANCE Existing_variation     IMPACT  \n",
       "1 '43/57' intron_variant      NA       rs1770344              MODIFIER\n",
       "2 NA      synonymous_variant  NA       rs1057079&COSV63875443 LOW     \n",
       "3 NA      synonymous_variant  NA       rs1064261&COSV63873449 LOW     \n",
       "4 NA      synonymous_variant  NA       rs1135172&COSV63873456 LOW     \n",
       "5 '20/20' intron_variant      NA       rs11438160             MODIFIER\n",
       "6 NA      3_prime_UTR_variant NA       rs10754339             MODIFIER\n",
       "  VARIANT_CLASS CLIN_SIG Clinvar  dbSNP     COSMIC      COSMIC.CNT genome1000\n",
       "1 SNV           NA       NA       NA        NA          NA         rs1770344 \n",
       "2 SNV           benign   Benign:1 rs1057079 COSM4142146 3          rs1057079 \n",
       "3 SNV           NA       NA       NA        COSM4142152 2          rs1064261 \n",
       "4 SNV           benign   Benign:1 rs1135172 COSM4142157 2          rs1135172 \n",
       "5 insertion     NA       NA       NA        NA          NA         rs11438160\n",
       "6 SNV           NA       NA       NA        NA          NA         rs10754339\n",
       "  AF     CANONICAL filteredCnt DSCnt ⋯ CopyNumber zScore.cnv\n",
       "1 1.0000 yes       2991        2500  ⋯ 2.28       0.184     \n",
       "2 0.8165 yes       3001        1943  ⋯ 2.12       0.265     \n",
       "3 0.9191 yes       3171        2524  ⋯ 2.30       0.846     \n",
       "4 0.8789 yes       1917        1688  ⋯ 2.07       1.050     \n",
       "5 0.1770 YES        892         394  ⋯ 3.12       5.430     \n",
       "6 0.9586 yes       2735        1500  ⋯ 2.04       0.532     \n",
       "  coVariant                                                                          \n",
       "1                                                                                    \n",
       "2                                                                                    \n",
       "3 chr1:11288618:G14G124G:A14C124A;chr1:11288633:G124G:C124A;chr1:11288618:G139G:A139A\n",
       "4                                                                                    \n",
       "5                                                                                    \n",
       "6                                                                                    \n",
       "  ol.coVariant          snvmember mergedVariant coVariantFilter\n",
       "1                       NA                      TRUE           \n",
       "2                       NA                      TRUE           \n",
       "3 0.17;0.00422;0.000315 NA                      TRUE           \n",
       "4                       NA                      TRUE           \n",
       "5                       NA                      TRUE           \n",
       "6                       NA                      TRUE           \n",
       "  sideVariant                sideVariant.AF.ratio prevalenceInternal\n",
       "1                              NA                 0.0584            \n",
       "2                              NA                 0.9180            \n",
       "3                              NA                 0.9290            \n",
       "4                              NA                 0.9360            \n",
       "5 adj:chr1:114224782:A:AT:12 2.74                 0.0231            \n",
       "6                              NA                 0.9800            \n",
       "  filterPrevalence filterScore finalKeep VariantID           \n",
       "1 TRUE             8.475       FALSE     chr1:11187893:T:C   \n",
       "2 TRUE             9.800       FALSE     chr1:11205058:C:T   \n",
       "3 TRUE             9.050       FALSE     chr1:11288758:G:A   \n",
       "4 TRUE             8.975       FALSE     chr1:11301714:A:G   \n",
       "5 TRUE             3.030       FALSE     chr1:114224782:A:ATT\n",
       "6 TRUE             9.050       FALSE     chr1:117690272:G:A  \n",
       "  sampleDir                                                                                                                                    \n",
       "1 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "2 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "3 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "4 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "5 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "6 /prednet/bdata21/OutputByRunB21/221123_A00934_0231_BHLL7KDSX5_runtask_221126000739_c84c4879/dsrun1.7.1/work/f5/fd5aef3e8a495f2a82f720c5c885d3\n",
       "  VariantType depth FilterType sampleFolder   PatientID externalSampleID\n",
       "1 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "2 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "3 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "4 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "5 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "6 Germline    high  otherFinal P224641_221123 173736    06S22025850     \n",
       "  trialVisitNum SpecimenType ID                           highFrequent.inbatch\n",
       "1 Pre Op        Plasma       P224641:chr1:11187893:T:C    23                  \n",
       "2 Pre Op        Plasma       P224641:chr1:11205058:C:T    21                  \n",
       "3 Pre Op        Plasma       P224641:chr1:11288758:G:A    20                  \n",
       "4 Pre Op        Plasma       P224641:chr1:11301714:A:G    20                  \n",
       "5 Pre Op        Plasma       P224641:chr1:114224782:A:ATT  4                  \n",
       "6 Pre Op        Plasma       P224641:chr1:117690272:G:A   23                  \n",
       "  concordant PatientID.old totalDepth.Baseline dscnt.Baseline.filtered\n",
       "1 FALSE      173736         922                28                     \n",
       "2 FALSE      173736         512                13                     \n",
       "3 FALSE      173736        1013                31                     \n",
       "4 FALSE      173736         752                16                     \n",
       "5 FALSE      173736         396                 2                     \n",
       "6 FALSE      173736         502                 8                     \n",
       "  altDepth.Baseline altDepth.Baseline.filtered AF.Baseline odds.ratio pvalue   \n",
       "1  922              736                        100.0       0.9989388  0.5185111\n",
       "2  508              408                         99.2       1.0004027  0.5110042\n",
       "3 1008              844                         99.5       0.9978536  0.5270965\n",
       "4  752              648                        100.0       0.9974305  0.5292151\n",
       "5  134              118                         33.8       0.9717162  0.6296152\n",
       "6  500              411                         99.6       1.0003092  0.5119251\n",
       "  MAF.diff    VariantType.old finalKeep.old\n",
       "1 -0.10612895 Germline        TRUE         \n",
       "2  0.05841415 Germline        TRUE         \n",
       "3 -0.20717653 Germline        TRUE         \n",
       "4 -0.25697504 Germline        TRUE         \n",
       "5 -0.91884400 Germline        TRUE         \n",
       "6  0.03062458 Germline        TRUE         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "55644"
      ],
      "text/latex": [
       "55644"
      ],
      "text/markdown": [
       "55644"
      ],
      "text/plain": [
       "[1] 55644"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Moffitt test data sets\n",
    "all.snv.plasma <- read.csv(\"../Moffitt_2022/MIBC/data/ATLAS/PRDC-MOFFITT-MIBC-22002/07_27_23_update/PRDC-MOFFITT-MIBC-22002-ATLAS_Variant_all_07-27-2023_2023-07-27_plasma_marked.csv\")\n",
    "\n",
    "all.snv.plasma = all.snv.plasma %>% mutate(PatientID.old = PatientID,\n",
    "                                           PatientID = fix_PIDs(PatientID.old))\n",
    "table(all.snv.plasma$VariantType.old)\n",
    "table(all.snv.plasma$VariantType)\n",
    "nrow(all.snv.plasma)\n",
    "table(all.snv.plasma$SpecimenType)\n",
    "all.snv.plasma %>% group_by(PatientID) %>% summarize(n_mutations=n()) \n",
    "\n",
    "### Matched normal pbmc \n",
    "all.pbmc = read.csv(\"../CHIP/PRDC-MOFFITT-MIBC-22002-WES_Variant_all_2022-12-28_pbmc.csv\", as.is=T)\n",
    "table(all.pbmc$SpecimenType)\n",
    "# fix old patient id numbers in this file\n",
    "all.pbmc = all.pbmc %>% mutate(PatientID.old = PatientID,\n",
    "                               PatientID = fix_PIDs(PatientID.old))\n",
    "# any patients without germline data?\n",
    "stopifnot(all(unique(all.snv.plasma$PatientID) %in% unique(all.pbmc$PatientID)))\n",
    "dim(all.snv.plasma)\n",
    "head(all.snv.plasma)\n",
    "# Add patient ID to VariantID for more unique ID\n",
    "all.snv.plasma <- all.snv.plasma %>% mutate(VariantID.2=paste0(PatientID, \":\", VariantID))\n",
    "length(unique(all.snv.plasma$VariantID.2)) # make sure each variants get unique ID. matches number of rows (n=55644)\n",
    "\n",
    "\n",
    "# Read in tumor only ffpe\n",
    "#all.snv.tumor <- read.csv(\"data/PRDC-MOFFITT-MIBC-22002-WES_Variant_all_2022-12-28_ffpe_marked.csv\", as.is=T)\n",
    "#all.snv.tumor = all.snv.tumor %>% mutate(PatientID.old = PatientID,\n",
    "#                                         PatientID = fix_PIDs(PatientID.old))\n",
    "#all.snv.tumor %>% group_by(PatientID) %>% summarize(n_mutations=n())\n",
    "\n",
    "# any patients without tumor data?\n",
    "# stopifnot(all(unique(all.snv.plasma$PatientID) %in% unique(all.snv.tumor$PatientID))) \n",
    "\n",
    "# some patients don't have tumor data, filter these out\n",
    "#all.snv.plasma = all.snv.plasma %>% filter(PatientID %in% unique(all.snv.tumor$PatientID))\n",
    "#dim(all.snv.plasma) #should be 48097 total variants\n",
    "\n",
    "\n",
    "### Whether variant exists in public databases = binary classification\n",
    "all.snv.plasma <- all.snv.plasma %>% mutate(in.dbSNP=ifelse(is.na(dbSNP), as.character(0), \"in.dbSNP\"),\n",
    "                                            in.genome1000=ifelse(genome1000==\"\", as.character(0), \"in.genome1000\"),\n",
    "                                            in.cosmic=ifelse(is.na(COSMIC), as.character(0), \"in.cosmic\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39de5b59",
   "metadata": {},
   "outputs": [],
   "source": [
    "## implement Tram's best somatic calling model using predicine raw call as a feature and checkGerm paired as truth\n",
    "## improve Tram's model?\n",
    "## basic stack Trams model and Kevin's model stepwise, threshold diagram\n",
    "## combine both models into one big logistic model\n",
    "# other model architectures, boosting, feature importance\n",
    "## compare agaisn raw calls, Mutect2, random forest"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47fd3255",
   "metadata": {},
   "source": [
    "## Tram's model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "1df8c68e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "99.9368288060644"
      ],
      "text/latex": [
       "99.9368288060644"
      ],
      "text/markdown": [
       "99.9368288060644"
      ],
      "text/plain": [
       "[1] 99.93683"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in eval(expr, envir, enclos):\n",
      "“NAs introduced by coercion”\n"
     ]
    }
   ],
   "source": [
    "# Add features\n",
    "\n",
    "# Across ALL patient samples at each variant locus. Find the max VAF across the dataset (t_maj)\n",
    "all.snv.plasma.f <- as.data.frame(all.snv.plasma %>% group_by(VariantID) %>% mutate(max.VaraintFreq.group = max(VariantFreq)))\n",
    "unique(all.snv.plasma.f$max.VaraintFreq.group[all.snv.plasma.f$seqnames == \"chr16\" & all.snv.plasma.f$start == \"136888\"]) \n",
    "# Check that for the same position across patients, should have the same t_maj\n",
    "\n",
    "\n",
    "########## 3. t_alt_freq: ##########\n",
    "# The fraction of reads corresponding to the alternate allele\n",
    "# I think this is just VAF (VariantFreq) which is always the alt allele\n",
    "\n",
    "\n",
    "########## 4. max_cosmic_count:\t##########\n",
    "# The number of times this allele has been observed in the COSMIC\n",
    "# Use the COSMIC.CNT column\n",
    "\n",
    "# First, we need to change the COSMIC.CNT column to integer (it is now character)\n",
    "all.snv.plasma.f$COSMIC.CNT <- as.numeric(all.snv.plasma.f$COSMIC.CNT)\n",
    "#class(all.snv.plasma$COSMIC.CNT)\n",
    "\n",
    "all.snv.plasma.f <- all.snv.plasma.f %>% group_by(VariantID) %>% # added alt here since sometimes multiple alleles at a site\n",
    "  mutate(max_COSMIC_CNT.group = ifelse(!(all(is.na(COSMIC.CNT))), max(COSMIC.CNT, na.rm = TRUE), 0))\n",
    "# If not in COSMIC = NA = set to zero"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "72fba414",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "P217576 P224597 P224601 P224606 P224608 P224615 P224619 P224623 P224624 P224625 \n",
       "   2581    2634    2467    2337    2379    2296    2398    2552    2448    2515 \n",
       "P224628 P224630 P224633 P224637 P224641 P224642 P224644 P224646 P224648 P224651 \n",
       "   2359    2461    2332    2478    2438    2405    2360    2402    2427    2401 \n",
       "P224652 P224653 P224655    <NA> \n",
       "   2410    2288    2276       0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "P217576 P224597 P224601 P224606 P224608 P224615 P224619 P224623 P224624 P224625 \n",
       "   2581    2634    2467    2337    2379    2296    2398    2552    2448    2515 \n",
       "P224628 P224630 P224633 P224637 P224641 P224642 P224644 P224646 P224648 P224651 \n",
       "   2359    2461    2332    2478    2438    2405    2360    2402    2427    2401 \n",
       "P224652 P224653 P224655    <NA> \n",
       "   2410    2288    2276       0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AF'</li><li>'AF.diff'</li><li>'AF.th'</li><li>'AFR_AF'</li><li>'Allele'</li><li>'alt'</li><li>'altDepth'</li><li>'altLOD'</li><li>'Amino_acids'</li><li>'AMR_AF'</li><li>'avgEntropy'</li><li>'BAM_EDIT'</li><li>'bgFilter'</li><li>'BIOTYPE'</li><li>'CANONICAL'</li><li>'cDNA_position'</li><li>'CDS_position'</li><li>'CLIN_SIG'</li><li>'Clinvar'</li><li>'Codons'</li><li>'Consequence'</li><li>'CopyNumber'</li><li>'COSMIC'</li><li>'COSMIC.CNT'</li><li>'coVariant'</li><li>'coVariantFilter'</li><li>'dbSNP'</li><li>'DISTANCE'</li><li>'DS.filter'</li><li>'DSCnt'</li><li>'EAS_AF'</li><li>'end'</li><li>'entropy'</li><li>'EUR_AF'</li><li>'Existing_variation'</li><li>'EXON'</li><li>'fc.ds.ratio'</li><li>'Feature'</li><li>'Feature_type'</li><li>'filtered_tlen_mean'</li><li>'filtered_tlen_N'</li><li>'filtered_tlen_std'</li><li>'filtered_toend_numunique'</li><li>'filtered_total_base_count'</li><li>'filtered.consensus'</li><li>'filteredCnt'</li><li>'filteredDSCnt'</li><li>'filterPrevalence'</li><li>'filterScore'</li><li>'FilterType'</li><li>'filterVarType'</li><li>'finalKeep'</li><li>'FLAGS'</li><li>'GC.Percent'</li><li>'Gene'</li><li>'GENE_PHENO'</li><li>'GeneID'</li><li>'generalFilter'</li><li>'genome1000'</li><li>'GIVEN_REF'</li><li>'heteroLOD'</li><li>'HGNC_ID'</li><li>'HGVS_OFFSET'</li><li>'HGVSc'</li><li>'HGVSp'</li><li>'highFrequent'</li><li>'highQ.ratio'</li><li>'homoLOD'</li><li>'ID'</li><li>'IMPACT'</li><li>'importantVariant'</li><li>'INTRON'</li><li>'KeepStatus'</li><li>'localVariantFilter'</li><li>'mapQ_mean'</li><li>'mapQ_std'</li><li>'MAX_AF'</li><li>'MAX_AF_POPS'</li><li>'mergedVariant'</li><li>'MNV'</li><li>'ol.coVariant'</li><li>'onTarget'</li><li>'partial_strand_cnt'</li><li>'partial_strand_mean'</li><li>'partial_strand_std'</li><li>'PHENO'</li><li>'prevalenceInternal'</li><li>'Protein_position'</li><li>'pseudoDSCnt'</li><li>'Qscore'</li><li>'ratio.fragLen'</li><li>'ReadEndFilter'</li><li>'ref'</li><li>'ref.avgEntropy'</li><li>'ref.DSCnt'</li><li>'ref.entropy'</li><li>'ref.filtered_tlen_mean'</li><li>'ref.filtered_tlen_N'</li><li>'ref.filtered_tlen_std'</li><li>'ref.filtered_toend_numunique'</li><li>'ref.filteredCnt'</li><li>'ref.filteredDSCnt'</li><li>'ref.mapQ_mean'</li><li>'ref.score_mean'</li><li>'ref.strand_both_cnt'</li><li>'ref.strand_forward_cnt'</li><li>'ref.strand_reverse_cnt'</li><li>'ref.toend_mean'</li><li>'ref.toend_std'</li><li>'ref.tomend_mean'</li><li>'ref.tomend_median'</li><li>'ref.tomend_numunique'</li><li>'ref.tomend_std'</li><li>'ref.width'</li><li>'refDepth'</li><li>'REFSEQ_MATCH'</li><li>'REFSEQ_OFFSET'</li><li>'repeatRegion'</li><li>'sampleDir'</li><li>'SampleDir'</li><li>'SampleID'</li><li>'sampleNames'</li><li>'SAS_AF'</li><li>'score_mean'</li><li>'score_std'</li><li>'seqnames'</li><li>'sideVariant'</li><li>'sideVariant.AF.ratio'</li><li>'snvmember'</li><li>'SOMATIC'</li><li>'start'</li><li>'strand'</li><li>'STRAND'</li><li>'strand_both_cnt'</li><li>'strand_forward_cnt'</li><li>'strand_reverse_cnt'</li><li>'SYMBOL'</li><li>'SYMBOL_SOURCE'</li><li>'tandemRepeat'</li><li>'tlen.ratio'</li><li>'toend_mean'</li><li>'toend_std'</li><li>'tomend_mean'</li><li>'tomend_median'</li><li>'tomend_numunique'</li><li>'tomend_std'</li><li>'totalDepth'</li><li>'USED_REF'</li><li>'VARIANT_CLASS'</li><li>'VariantFreq'</li><li>'VariantID'</li><li>'VariantID.old'</li><li>'VariantID2'</li><li>'VariantType'</li><li>'VEP.ID'</li><li>'white.list'</li><li>'width'</li><li>'wl.tier'</li><li>'zScore.cnv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AF'\n",
       "\\item 'AF.diff'\n",
       "\\item 'AF.th'\n",
       "\\item 'AFR\\_AF'\n",
       "\\item 'Allele'\n",
       "\\item 'alt'\n",
       "\\item 'altDepth'\n",
       "\\item 'altLOD'\n",
       "\\item 'Amino\\_acids'\n",
       "\\item 'AMR\\_AF'\n",
       "\\item 'avgEntropy'\n",
       "\\item 'BAM\\_EDIT'\n",
       "\\item 'bgFilter'\n",
       "\\item 'BIOTYPE'\n",
       "\\item 'CANONICAL'\n",
       "\\item 'cDNA\\_position'\n",
       "\\item 'CDS\\_position'\n",
       "\\item 'CLIN\\_SIG'\n",
       "\\item 'Clinvar'\n",
       "\\item 'Codons'\n",
       "\\item 'Consequence'\n",
       "\\item 'CopyNumber'\n",
       "\\item 'COSMIC'\n",
       "\\item 'COSMIC.CNT'\n",
       "\\item 'coVariant'\n",
       "\\item 'coVariantFilter'\n",
       "\\item 'dbSNP'\n",
       "\\item 'DISTANCE'\n",
       "\\item 'DS.filter'\n",
       "\\item 'DSCnt'\n",
       "\\item 'EAS\\_AF'\n",
       "\\item 'end'\n",
       "\\item 'entropy'\n",
       "\\item 'EUR\\_AF'\n",
       "\\item 'Existing\\_variation'\n",
       "\\item 'EXON'\n",
       "\\item 'fc.ds.ratio'\n",
       "\\item 'Feature'\n",
       "\\item 'Feature\\_type'\n",
       "\\item 'filtered\\_tlen\\_mean'\n",
       "\\item 'filtered\\_tlen\\_N'\n",
       "\\item 'filtered\\_tlen\\_std'\n",
       "\\item 'filtered\\_toend\\_numunique'\n",
       "\\item 'filtered\\_total\\_base\\_count'\n",
       "\\item 'filtered.consensus'\n",
       "\\item 'filteredCnt'\n",
       "\\item 'filteredDSCnt'\n",
       "\\item 'filterPrevalence'\n",
       "\\item 'filterScore'\n",
       "\\item 'FilterType'\n",
       "\\item 'filterVarType'\n",
       "\\item 'finalKeep'\n",
       "\\item 'FLAGS'\n",
       "\\item 'GC.Percent'\n",
       "\\item 'Gene'\n",
       "\\item 'GENE\\_PHENO'\n",
       "\\item 'GeneID'\n",
       "\\item 'generalFilter'\n",
       "\\item 'genome1000'\n",
       "\\item 'GIVEN\\_REF'\n",
       "\\item 'heteroLOD'\n",
       "\\item 'HGNC\\_ID'\n",
       "\\item 'HGVS\\_OFFSET'\n",
       "\\item 'HGVSc'\n",
       "\\item 'HGVSp'\n",
       "\\item 'highFrequent'\n",
       "\\item 'highQ.ratio'\n",
       "\\item 'homoLOD'\n",
       "\\item 'ID'\n",
       "\\item 'IMPACT'\n",
       "\\item 'importantVariant'\n",
       "\\item 'INTRON'\n",
       "\\item 'KeepStatus'\n",
       "\\item 'localVariantFilter'\n",
       "\\item 'mapQ\\_mean'\n",
       "\\item 'mapQ\\_std'\n",
       "\\item 'MAX\\_AF'\n",
       "\\item 'MAX\\_AF\\_POPS'\n",
       "\\item 'mergedVariant'\n",
       "\\item 'MNV'\n",
       "\\item 'ol.coVariant'\n",
       "\\item 'onTarget'\n",
       "\\item 'partial\\_strand\\_cnt'\n",
       "\\item 'partial\\_strand\\_mean'\n",
       "\\item 'partial\\_strand\\_std'\n",
       "\\item 'PHENO'\n",
       "\\item 'prevalenceInternal'\n",
       "\\item 'Protein\\_position'\n",
       "\\item 'pseudoDSCnt'\n",
       "\\item 'Qscore'\n",
       "\\item 'ratio.fragLen'\n",
       "\\item 'ReadEndFilter'\n",
       "\\item 'ref'\n",
       "\\item 'ref.avgEntropy'\n",
       "\\item 'ref.DSCnt'\n",
       "\\item 'ref.entropy'\n",
       "\\item 'ref.filtered\\_tlen\\_mean'\n",
       "\\item 'ref.filtered\\_tlen\\_N'\n",
       "\\item 'ref.filtered\\_tlen\\_std'\n",
       "\\item 'ref.filtered\\_toend\\_numunique'\n",
       "\\item 'ref.filteredCnt'\n",
       "\\item 'ref.filteredDSCnt'\n",
       "\\item 'ref.mapQ\\_mean'\n",
       "\\item 'ref.score\\_mean'\n",
       "\\item 'ref.strand\\_both\\_cnt'\n",
       "\\item 'ref.strand\\_forward\\_cnt'\n",
       "\\item 'ref.strand\\_reverse\\_cnt'\n",
       "\\item 'ref.toend\\_mean'\n",
       "\\item 'ref.toend\\_std'\n",
       "\\item 'ref.tomend\\_mean'\n",
       "\\item 'ref.tomend\\_median'\n",
       "\\item 'ref.tomend\\_numunique'\n",
       "\\item 'ref.tomend\\_std'\n",
       "\\item 'ref.width'\n",
       "\\item 'refDepth'\n",
       "\\item 'REFSEQ\\_MATCH'\n",
       "\\item 'REFSEQ\\_OFFSET'\n",
       "\\item 'repeatRegion'\n",
       "\\item 'sampleDir'\n",
       "\\item 'SampleDir'\n",
       "\\item 'SampleID'\n",
       "\\item 'sampleNames'\n",
       "\\item 'SAS\\_AF'\n",
       "\\item 'score\\_mean'\n",
       "\\item 'score\\_std'\n",
       "\\item 'seqnames'\n",
       "\\item 'sideVariant'\n",
       "\\item 'sideVariant.AF.ratio'\n",
       "\\item 'snvmember'\n",
       "\\item 'SOMATIC'\n",
       "\\item 'start'\n",
       "\\item 'strand'\n",
       "\\item 'STRAND'\n",
       "\\item 'strand\\_both\\_cnt'\n",
       "\\item 'strand\\_forward\\_cnt'\n",
       "\\item 'strand\\_reverse\\_cnt'\n",
       "\\item 'SYMBOL'\n",
       "\\item 'SYMBOL\\_SOURCE'\n",
       "\\item 'tandemRepeat'\n",
       "\\item 'tlen.ratio'\n",
       "\\item 'toend\\_mean'\n",
       "\\item 'toend\\_std'\n",
       "\\item 'tomend\\_mean'\n",
       "\\item 'tomend\\_median'\n",
       "\\item 'tomend\\_numunique'\n",
       "\\item 'tomend\\_std'\n",
       "\\item 'totalDepth'\n",
       "\\item 'USED\\_REF'\n",
       "\\item 'VARIANT\\_CLASS'\n",
       "\\item 'VariantFreq'\n",
       "\\item 'VariantID'\n",
       "\\item 'VariantID.old'\n",
       "\\item 'VariantID2'\n",
       "\\item 'VariantType'\n",
       "\\item 'VEP.ID'\n",
       "\\item 'white.list'\n",
       "\\item 'width'\n",
       "\\item 'wl.tier'\n",
       "\\item 'zScore.cnv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AF'\n",
       "2. 'AF.diff'\n",
       "3. 'AF.th'\n",
       "4. 'AFR_AF'\n",
       "5. 'Allele'\n",
       "6. 'alt'\n",
       "7. 'altDepth'\n",
       "8. 'altLOD'\n",
       "9. 'Amino_acids'\n",
       "10. 'AMR_AF'\n",
       "11. 'avgEntropy'\n",
       "12. 'BAM_EDIT'\n",
       "13. 'bgFilter'\n",
       "14. 'BIOTYPE'\n",
       "15. 'CANONICAL'\n",
       "16. 'cDNA_position'\n",
       "17. 'CDS_position'\n",
       "18. 'CLIN_SIG'\n",
       "19. 'Clinvar'\n",
       "20. 'Codons'\n",
       "21. 'Consequence'\n",
       "22. 'CopyNumber'\n",
       "23. 'COSMIC'\n",
       "24. 'COSMIC.CNT'\n",
       "25. 'coVariant'\n",
       "26. 'coVariantFilter'\n",
       "27. 'dbSNP'\n",
       "28. 'DISTANCE'\n",
       "29. 'DS.filter'\n",
       "30. 'DSCnt'\n",
       "31. 'EAS_AF'\n",
       "32. 'end'\n",
       "33. 'entropy'\n",
       "34. 'EUR_AF'\n",
       "35. 'Existing_variation'\n",
       "36. 'EXON'\n",
       "37. 'fc.ds.ratio'\n",
       "38. 'Feature'\n",
       "39. 'Feature_type'\n",
       "40. 'filtered_tlen_mean'\n",
       "41. 'filtered_tlen_N'\n",
       "42. 'filtered_tlen_std'\n",
       "43. 'filtered_toend_numunique'\n",
       "44. 'filtered_total_base_count'\n",
       "45. 'filtered.consensus'\n",
       "46. 'filteredCnt'\n",
       "47. 'filteredDSCnt'\n",
       "48. 'filterPrevalence'\n",
       "49. 'filterScore'\n",
       "50. 'FilterType'\n",
       "51. 'filterVarType'\n",
       "52. 'finalKeep'\n",
       "53. 'FLAGS'\n",
       "54. 'GC.Percent'\n",
       "55. 'Gene'\n",
       "56. 'GENE_PHENO'\n",
       "57. 'GeneID'\n",
       "58. 'generalFilter'\n",
       "59. 'genome1000'\n",
       "60. 'GIVEN_REF'\n",
       "61. 'heteroLOD'\n",
       "62. 'HGNC_ID'\n",
       "63. 'HGVS_OFFSET'\n",
       "64. 'HGVSc'\n",
       "65. 'HGVSp'\n",
       "66. 'highFrequent'\n",
       "67. 'highQ.ratio'\n",
       "68. 'homoLOD'\n",
       "69. 'ID'\n",
       "70. 'IMPACT'\n",
       "71. 'importantVariant'\n",
       "72. 'INTRON'\n",
       "73. 'KeepStatus'\n",
       "74. 'localVariantFilter'\n",
       "75. 'mapQ_mean'\n",
       "76. 'mapQ_std'\n",
       "77. 'MAX_AF'\n",
       "78. 'MAX_AF_POPS'\n",
       "79. 'mergedVariant'\n",
       "80. 'MNV'\n",
       "81. 'ol.coVariant'\n",
       "82. 'onTarget'\n",
       "83. 'partial_strand_cnt'\n",
       "84. 'partial_strand_mean'\n",
       "85. 'partial_strand_std'\n",
       "86. 'PHENO'\n",
       "87. 'prevalenceInternal'\n",
       "88. 'Protein_position'\n",
       "89. 'pseudoDSCnt'\n",
       "90. 'Qscore'\n",
       "91. 'ratio.fragLen'\n",
       "92. 'ReadEndFilter'\n",
       "93. 'ref'\n",
       "94. 'ref.avgEntropy'\n",
       "95. 'ref.DSCnt'\n",
       "96. 'ref.entropy'\n",
       "97. 'ref.filtered_tlen_mean'\n",
       "98. 'ref.filtered_tlen_N'\n",
       "99. 'ref.filtered_tlen_std'\n",
       "100. 'ref.filtered_toend_numunique'\n",
       "101. 'ref.filteredCnt'\n",
       "102. 'ref.filteredDSCnt'\n",
       "103. 'ref.mapQ_mean'\n",
       "104. 'ref.score_mean'\n",
       "105. 'ref.strand_both_cnt'\n",
       "106. 'ref.strand_forward_cnt'\n",
       "107. 'ref.strand_reverse_cnt'\n",
       "108. 'ref.toend_mean'\n",
       "109. 'ref.toend_std'\n",
       "110. 'ref.tomend_mean'\n",
       "111. 'ref.tomend_median'\n",
       "112. 'ref.tomend_numunique'\n",
       "113. 'ref.tomend_std'\n",
       "114. 'ref.width'\n",
       "115. 'refDepth'\n",
       "116. 'REFSEQ_MATCH'\n",
       "117. 'REFSEQ_OFFSET'\n",
       "118. 'repeatRegion'\n",
       "119. 'sampleDir'\n",
       "120. 'SampleDir'\n",
       "121. 'SampleID'\n",
       "122. 'sampleNames'\n",
       "123. 'SAS_AF'\n",
       "124. 'score_mean'\n",
       "125. 'score_std'\n",
       "126. 'seqnames'\n",
       "127. 'sideVariant'\n",
       "128. 'sideVariant.AF.ratio'\n",
       "129. 'snvmember'\n",
       "130. 'SOMATIC'\n",
       "131. 'start'\n",
       "132. 'strand'\n",
       "133. 'STRAND'\n",
       "134. 'strand_both_cnt'\n",
       "135. 'strand_forward_cnt'\n",
       "136. 'strand_reverse_cnt'\n",
       "137. 'SYMBOL'\n",
       "138. 'SYMBOL_SOURCE'\n",
       "139. 'tandemRepeat'\n",
       "140. 'tlen.ratio'\n",
       "141. 'toend_mean'\n",
       "142. 'toend_std'\n",
       "143. 'tomend_mean'\n",
       "144. 'tomend_median'\n",
       "145. 'tomend_numunique'\n",
       "146. 'tomend_std'\n",
       "147. 'totalDepth'\n",
       "148. 'USED_REF'\n",
       "149. 'VARIANT_CLASS'\n",
       "150. 'VariantFreq'\n",
       "151. 'VariantID'\n",
       "152. 'VariantID.old'\n",
       "153. 'VariantID2'\n",
       "154. 'VariantType'\n",
       "155. 'VEP.ID'\n",
       "156. 'white.list'\n",
       "157. 'width'\n",
       "158. 'wl.tier'\n",
       "159. 'zScore.cnv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"AF\"                           \"AF.diff\"                     \n",
       "  [3] \"AF.th\"                        \"AFR_AF\"                      \n",
       "  [5] \"Allele\"                       \"alt\"                         \n",
       "  [7] \"altDepth\"                     \"altLOD\"                      \n",
       "  [9] \"Amino_acids\"                  \"AMR_AF\"                      \n",
       " [11] \"avgEntropy\"                   \"BAM_EDIT\"                    \n",
       " [13] \"bgFilter\"                     \"BIOTYPE\"                     \n",
       " [15] \"CANONICAL\"                    \"cDNA_position\"               \n",
       " [17] \"CDS_position\"                 \"CLIN_SIG\"                    \n",
       " [19] \"Clinvar\"                      \"Codons\"                      \n",
       " [21] \"Consequence\"                  \"CopyNumber\"                  \n",
       " [23] \"COSMIC\"                       \"COSMIC.CNT\"                  \n",
       " [25] \"coVariant\"                    \"coVariantFilter\"             \n",
       " [27] \"dbSNP\"                        \"DISTANCE\"                    \n",
       " [29] \"DS.filter\"                    \"DSCnt\"                       \n",
       " [31] \"EAS_AF\"                       \"end\"                         \n",
       " [33] \"entropy\"                      \"EUR_AF\"                      \n",
       " [35] \"Existing_variation\"           \"EXON\"                        \n",
       " [37] \"fc.ds.ratio\"                  \"Feature\"                     \n",
       " [39] \"Feature_type\"                 \"filtered_tlen_mean\"          \n",
       " [41] \"filtered_tlen_N\"              \"filtered_tlen_std\"           \n",
       " [43] \"filtered_toend_numunique\"     \"filtered_total_base_count\"   \n",
       " [45] \"filtered.consensus\"           \"filteredCnt\"                 \n",
       " [47] \"filteredDSCnt\"                \"filterPrevalence\"            \n",
       " [49] \"filterScore\"                  \"FilterType\"                  \n",
       " [51] \"filterVarType\"                \"finalKeep\"                   \n",
       " [53] \"FLAGS\"                        \"GC.Percent\"                  \n",
       " [55] \"Gene\"                         \"GENE_PHENO\"                  \n",
       " [57] \"GeneID\"                       \"generalFilter\"               \n",
       " [59] \"genome1000\"                   \"GIVEN_REF\"                   \n",
       " [61] \"heteroLOD\"                    \"HGNC_ID\"                     \n",
       " [63] \"HGVS_OFFSET\"                  \"HGVSc\"                       \n",
       " [65] \"HGVSp\"                        \"highFrequent\"                \n",
       " [67] \"highQ.ratio\"                  \"homoLOD\"                     \n",
       " [69] \"ID\"                           \"IMPACT\"                      \n",
       " [71] \"importantVariant\"             \"INTRON\"                      \n",
       " [73] \"KeepStatus\"                   \"localVariantFilter\"          \n",
       " [75] \"mapQ_mean\"                    \"mapQ_std\"                    \n",
       " [77] \"MAX_AF\"                       \"MAX_AF_POPS\"                 \n",
       " [79] \"mergedVariant\"                \"MNV\"                         \n",
       " [81] \"ol.coVariant\"                 \"onTarget\"                    \n",
       " [83] \"partial_strand_cnt\"           \"partial_strand_mean\"         \n",
       " [85] \"partial_strand_std\"           \"PHENO\"                       \n",
       " [87] \"prevalenceInternal\"           \"Protein_position\"            \n",
       " [89] \"pseudoDSCnt\"                  \"Qscore\"                      \n",
       " [91] \"ratio.fragLen\"                \"ReadEndFilter\"               \n",
       " [93] \"ref\"                          \"ref.avgEntropy\"              \n",
       " [95] \"ref.DSCnt\"                    \"ref.entropy\"                 \n",
       " [97] \"ref.filtered_tlen_mean\"       \"ref.filtered_tlen_N\"         \n",
       " [99] \"ref.filtered_tlen_std\"        \"ref.filtered_toend_numunique\"\n",
       "[101] \"ref.filteredCnt\"              \"ref.filteredDSCnt\"           \n",
       "[103] \"ref.mapQ_mean\"                \"ref.score_mean\"              \n",
       "[105] \"ref.strand_both_cnt\"          \"ref.strand_forward_cnt\"      \n",
       "[107] \"ref.strand_reverse_cnt\"       \"ref.toend_mean\"              \n",
       "[109] \"ref.toend_std\"                \"ref.tomend_mean\"             \n",
       "[111] \"ref.tomend_median\"            \"ref.tomend_numunique\"        \n",
       "[113] \"ref.tomend_std\"               \"ref.width\"                   \n",
       "[115] \"refDepth\"                     \"REFSEQ_MATCH\"                \n",
       "[117] \"REFSEQ_OFFSET\"                \"repeatRegion\"                \n",
       "[119] \"sampleDir\"                    \"SampleDir\"                   \n",
       "[121] \"SampleID\"                     \"sampleNames\"                 \n",
       "[123] \"SAS_AF\"                       \"score_mean\"                  \n",
       "[125] \"score_std\"                    \"seqnames\"                    \n",
       "[127] \"sideVariant\"                  \"sideVariant.AF.ratio\"        \n",
       "[129] \"snvmember\"                    \"SOMATIC\"                     \n",
       "[131] \"start\"                        \"strand\"                      \n",
       "[133] \"STRAND\"                       \"strand_both_cnt\"             \n",
       "[135] \"strand_forward_cnt\"           \"strand_reverse_cnt\"          \n",
       "[137] \"SYMBOL\"                       \"SYMBOL_SOURCE\"               \n",
       "[139] \"tandemRepeat\"                 \"tlen.ratio\"                  \n",
       "[141] \"toend_mean\"                   \"toend_std\"                   \n",
       "[143] \"tomend_mean\"                  \"tomend_median\"               \n",
       "[145] \"tomend_numunique\"             \"tomend_std\"                  \n",
       "[147] \"totalDepth\"                   \"USED_REF\"                    \n",
       "[149] \"VARIANT_CLASS\"                \"VariantFreq\"                 \n",
       "[151] \"VariantID\"                    \"VariantID.old\"               \n",
       "[153] \"VariantID2\"                   \"VariantType\"                 \n",
       "[155] \"VEP.ID\"                       \"white.list\"                  \n",
       "[157] \"width\"                        \"wl.tier\"                     \n",
       "[159] \"zScore.cnv\"                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>VEP.ID</th><th scope=col>AFR_AF</th><th scope=col>AMR_AF</th><th scope=col>EUR_AF</th><th scope=col>EAS_AF</th><th scope=col>SAS_AF</th><th scope=col>MAX_AF</th><th scope=col>SOMATIC</th><th scope=col>PHENO</th><th scope=col>mapQ_mean</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>P224641:chr1:3644349:A:G  </td><td>0.7753</td><td>0.6758</td><td>0.825</td><td>0.3353</td><td>0.6063</td><td>0.8250</td><td>NA</td><td>NA</td><td>59.9787</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>P224641:chr1:3644374:A:G  </td><td>0.7723</td><td>0.6758</td><td>0.828</td><td>0.3353</td><td>0.6074</td><td>0.8280</td><td>NA</td><td>NA</td><td>59.9843</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>P224641:chr1:9770690:C:CAG</td><td>    NA</td><td>    NA</td><td>   NA</td><td>    NA</td><td>    NA</td><td>0.4378</td><td>NA</td><td>NA</td><td>59.9154</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>P224641:chr1:9775888:A:C  </td><td>    NA</td><td>    NA</td><td>   NA</td><td>    NA</td><td>    NA</td><td>    NA</td><td>NA</td><td>NA</td><td>58.4577</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>P224641:chr1:9775898:A:C  </td><td>    NA</td><td>    NA</td><td>   NA</td><td>    NA</td><td>    NA</td><td>    NA</td><td>NA</td><td>NA</td><td>59.2995</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>P224641:chr1:9775903:G:C  </td><td>    NA</td><td>    NA</td><td>   NA</td><td>    NA</td><td>    NA</td><td>    NA</td><td>NA</td><td>NA</td><td>58.5718</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & VEP.ID & AFR\\_AF & AMR\\_AF & EUR\\_AF & EAS\\_AF & SAS\\_AF & MAX\\_AF & SOMATIC & PHENO & mapQ\\_mean\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & P224641:chr1:3644349:A:G   & 0.7753 & 0.6758 & 0.825 & 0.3353 & 0.6063 & 0.8250 & NA & NA & 59.9787\\\\\n",
       "\t2 & P224641:chr1:3644374:A:G   & 0.7723 & 0.6758 & 0.828 & 0.3353 & 0.6074 & 0.8280 & NA & NA & 59.9843\\\\\n",
       "\t3 & P224641:chr1:9770690:C:CAG &     NA &     NA &    NA &     NA &     NA & 0.4378 & NA & NA & 59.9154\\\\\n",
       "\t4 & P224641:chr1:9775888:A:C   &     NA &     NA &    NA &     NA &     NA &     NA & NA & NA & 58.4577\\\\\n",
       "\t5 & P224641:chr1:9775898:A:C   &     NA &     NA &    NA &     NA &     NA &     NA & NA & NA & 59.2995\\\\\n",
       "\t6 & P224641:chr1:9775903:G:C   &     NA &     NA &    NA &     NA &     NA &     NA & NA & NA & 58.5718\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | VEP.ID &lt;chr&gt; | AFR_AF &lt;dbl&gt; | AMR_AF &lt;dbl&gt; | EUR_AF &lt;dbl&gt; | EAS_AF &lt;dbl&gt; | SAS_AF &lt;dbl&gt; | MAX_AF &lt;dbl&gt; | SOMATIC &lt;chr&gt; | PHENO &lt;chr&gt; | mapQ_mean &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | P224641:chr1:3644349:A:G   | 0.7753 | 0.6758 | 0.825 | 0.3353 | 0.6063 | 0.8250 | NA | NA | 59.9787 |\n",
       "| 2 | P224641:chr1:3644374:A:G   | 0.7723 | 0.6758 | 0.828 | 0.3353 | 0.6074 | 0.8280 | NA | NA | 59.9843 |\n",
       "| 3 | P224641:chr1:9770690:C:CAG |     NA |     NA |    NA |     NA |     NA | 0.4378 | NA | NA | 59.9154 |\n",
       "| 4 | P224641:chr1:9775888:A:C   |     NA |     NA |    NA |     NA |     NA |     NA | NA | NA | 58.4577 |\n",
       "| 5 | P224641:chr1:9775898:A:C   |     NA |     NA |    NA |     NA |     NA |     NA | NA | NA | 59.2995 |\n",
       "| 6 | P224641:chr1:9775903:G:C   |     NA |     NA |    NA |     NA |     NA |     NA | NA | NA | 58.5718 |\n",
       "\n"
      ],
      "text/plain": [
       "  VEP.ID                     AFR_AF AMR_AF EUR_AF EAS_AF SAS_AF MAX_AF SOMATIC\n",
       "1 P224641:chr1:3644349:A:G   0.7753 0.6758 0.825  0.3353 0.6063 0.8250 NA     \n",
       "2 P224641:chr1:3644374:A:G   0.7723 0.6758 0.828  0.3353 0.6074 0.8280 NA     \n",
       "3 P224641:chr1:9770690:C:CAG     NA     NA    NA      NA     NA 0.4378 NA     \n",
       "4 P224641:chr1:9775888:A:C       NA     NA    NA      NA     NA     NA NA     \n",
       "5 P224641:chr1:9775898:A:C       NA     NA    NA      NA     NA     NA NA     \n",
       "6 P224641:chr1:9775903:G:C       NA     NA    NA      NA     NA     NA NA     \n",
       "  PHENO mapQ_mean\n",
       "1 NA    59.9787  \n",
       "2 NA    59.9843  \n",
       "3 NA    59.9154  \n",
       "4 NA    58.4577  \n",
       "5 NA    59.2995  \n",
       "6 NA    58.5718  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "55644"
      ],
      "text/latex": [
       "55644"
      ],
      "text/markdown": [
       "55644"
      ],
      "text/plain": [
       "[1] 55644"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "55644"
      ],
      "text/latex": [
       "55644"
      ],
      "text/markdown": [
       "55644"
      ],
      "text/plain": [
       "[1] 55644"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>PatientID</th><th scope=col>count</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>170594</td><td>307</td></tr>\n",
       "\t<tr><td>170762</td><td>333</td></tr>\n",
       "\t<tr><td>171126</td><td>304</td></tr>\n",
       "\t<tr><td>171351</td><td>280</td></tr>\n",
       "\t<tr><td>171660</td><td>325</td></tr>\n",
       "\t<tr><td>172109</td><td>239</td></tr>\n",
       "\t<tr><td>172163</td><td>273</td></tr>\n",
       "\t<tr><td>172422</td><td>261</td></tr>\n",
       "\t<tr><td>172592</td><td>239</td></tr>\n",
       "\t<tr><td>172956</td><td>299</td></tr>\n",
       "\t<tr><td>173129</td><td>285</td></tr>\n",
       "\t<tr><td>173149</td><td>284</td></tr>\n",
       "\t<tr><td>173657</td><td>352</td></tr>\n",
       "\t<tr><td>173736</td><td>299</td></tr>\n",
       "\t<tr><td>173983</td><td>354</td></tr>\n",
       "\t<tr><td>174280</td><td>347</td></tr>\n",
       "\t<tr><td>174438</td><td>289</td></tr>\n",
       "\t<tr><td>174688</td><td>280</td></tr>\n",
       "\t<tr><td>175253</td><td>284</td></tr>\n",
       "\t<tr><td>175702</td><td>303</td></tr>\n",
       "\t<tr><td>175743</td><td>233</td></tr>\n",
       "\t<tr><td>175827</td><td>247</td></tr>\n",
       "\t<tr><td>89200 </td><td>281</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " PatientID & count\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 170594 & 307\\\\\n",
       "\t 170762 & 333\\\\\n",
       "\t 171126 & 304\\\\\n",
       "\t 171351 & 280\\\\\n",
       "\t 171660 & 325\\\\\n",
       "\t 172109 & 239\\\\\n",
       "\t 172163 & 273\\\\\n",
       "\t 172422 & 261\\\\\n",
       "\t 172592 & 239\\\\\n",
       "\t 172956 & 299\\\\\n",
       "\t 173129 & 285\\\\\n",
       "\t 173149 & 284\\\\\n",
       "\t 173657 & 352\\\\\n",
       "\t 173736 & 299\\\\\n",
       "\t 173983 & 354\\\\\n",
       "\t 174280 & 347\\\\\n",
       "\t 174438 & 289\\\\\n",
       "\t 174688 & 280\\\\\n",
       "\t 175253 & 284\\\\\n",
       "\t 175702 & 303\\\\\n",
       "\t 175743 & 233\\\\\n",
       "\t 175827 & 247\\\\\n",
       "\t 89200  & 281\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 23 × 2\n",
       "\n",
       "| PatientID &lt;chr&gt; | count &lt;int&gt; |\n",
       "|---|---|\n",
       "| 170594 | 307 |\n",
       "| 170762 | 333 |\n",
       "| 171126 | 304 |\n",
       "| 171351 | 280 |\n",
       "| 171660 | 325 |\n",
       "| 172109 | 239 |\n",
       "| 172163 | 273 |\n",
       "| 172422 | 261 |\n",
       "| 172592 | 239 |\n",
       "| 172956 | 299 |\n",
       "| 173129 | 285 |\n",
       "| 173149 | 284 |\n",
       "| 173657 | 352 |\n",
       "| 173736 | 299 |\n",
       "| 173983 | 354 |\n",
       "| 174280 | 347 |\n",
       "| 174438 | 289 |\n",
       "| 174688 | 280 |\n",
       "| 175253 | 284 |\n",
       "| 175702 | 303 |\n",
       "| 175743 | 233 |\n",
       "| 175827 | 247 |\n",
       "| 89200  | 281 |\n",
       "\n"
      ],
      "text/plain": [
       "   PatientID count\n",
       "1  170594    307  \n",
       "2  170762    333  \n",
       "3  171126    304  \n",
       "4  171351    280  \n",
       "5  171660    325  \n",
       "6  172109    239  \n",
       "7  172163    273  \n",
       "8  172422    261  \n",
       "9  172592    239  \n",
       "10 172956    299  \n",
       "11 173129    285  \n",
       "12 173149    284  \n",
       "13 173657    352  \n",
       "14 173736    299  \n",
       "15 173983    354  \n",
       "16 174280    347  \n",
       "17 174438    289  \n",
       "18 174688    280  \n",
       "19 175253    284  \n",
       "20 175702    303  \n",
       "21 175743    233  \n",
       "22 175827    247  \n",
       "23 89200     281  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "########## 5. pop_max: ########## \n",
    "# The maximum population frequency of the allele across multiple germline databases\n",
    "## CHECK dbNSNP?\n",
    "## https://www.internationalgenome.org/faq/how-do-i-find-out-information-about-a-single-variant/\n",
    "\n",
    "\n",
    "########################### ADD VEP POP ALLELE FREQS  #######################\n",
    "\n",
    "# Instead of PatientID we can use SampleName as patient identifier because VEP only includes sample ID and no patient.\n",
    "all.snv.plasma.f = all.snv.plasma.f %>% \n",
    "  mutate(VEP.ID =paste0(sampleNames, \":\", VariantID)) \n",
    "\n",
    "# Load VEP and create VEP.ID to match atlas variants\n",
    "vep <- read.delim(\"./Tram_working_files/NMIBC_plasma_ATLAS_variant_all_marked_dist2end_sel_wVEP_09272023.tsv\", \n",
    "                  sep = \",\",header = TRUE,as.is = T)\n",
    "vep <- vep %>% \n",
    "  mutate(VEP.ID =paste0(vep$sampleNames, \":\", VariantID)) \n",
    "table(vep$sampleNames)\n",
    "table(all.snv.plasma.f$sampleNames)\n",
    "\n",
    "#length(intersect(unique(vep$sampleNames), unique(all.snv.plasma$sampleNames))) # how many patients are in common for both (n=20)\n",
    "#length(intersect(all.snv.plasma$VEP.ID, vep$VEP.ID)) # How many variants in common = 48097\n",
    "\n",
    "# Grab columns we want from VEP\n",
    "sort(names(vep))\n",
    "# AF = Frequency of existing variant in 1000 Genomes\n",
    "# gnomADe_AF - Frequency of existing variant in gnomAD exomes combined population -- (Not available for our data)\n",
    "# MAX_AF - Maximum observed allele frequency in 1000 Genomes, ESP and gnomAD\n",
    "# MAX_AF_POPS - Populations in which maximum allele frequency was observed\n",
    "# SOMATIC - Somatic status of existing variant(s); multiple values correspond to multiple values in the Existing_variation field\n",
    "# PHENO - Indicates if existing variant is associated with a phenotype, disease or trait; multiple values correspond to multiple values in the Existing_variation field\n",
    "# IMPACT - consequence of variant\n",
    "\n",
    "# NOTE: missing ASN_AF population in this output?? Wonder why...might want to check for future\n",
    "# AF.diff, AF.th, AF already in the ATLAS database, don't need to include\n",
    "sel_col <- c(\"VEP.ID\", \"AFR_AF\", \"AMR_AF\" ,\"EUR_AF\", \"EAS_AF\" ,\"SAS_AF\", \"MAX_AF\", \"SOMATIC\", \"PHENO\", \"mapQ_mean\")\n",
    "vep.sel = vep %>% select(sel_col)\n",
    "head(vep.sel)\n",
    "\n",
    "# join VEP columns\n",
    "all.snv.plasma.f <- all.snv.plasma.f %>% left_join(vep.sel, by=\"VEP.ID\") %>% \n",
    "        mutate(\"mapQ_mean\" = as.numeric(mapQ_mean))\n",
    "\n",
    "\n",
    "########## 6. snp_vaf_bin_00: ##########\n",
    "# The number of informative SNPs in the local copy number neighborhood with VAF between 0 and 0.05\n",
    "\n",
    "# Need to get copy number neighborhoods\n",
    "# Filter for VAF bin\n",
    "# Count how many SNPs are there\n",
    "# germline variant databases and copy-number segments, we identify neighboring heterozygous germline SNPs of similar copy number, and create a histogram of variant counts with 20 non- overlapping VAF bins \n",
    "# the copy number for each variant is represented by features derived from copy-number segmentation data and variant calls. Briefly, using germline variant databases and copy-number segments, we identify neighboring heterozygous germline SNPs of similar copy number, and create a histogram of variant counts with 20 non- overlapping VAF bins (see “Methods”).\n",
    "\n",
    "# possible features to use\n",
    "# CopyNumber column\n",
    "# average CopyNumber for this snv and other snvs within ~10 MB (?could use average human genome LD MB value here) for the patient\n",
    "\n",
    "# local 1MB bin CN estimate and/or coverage zscore from additional LP-WGS data\n",
    "\n",
    "\n",
    "########## 7. Somatic Count: ##########\n",
    "# Count total number of somatic variants per patient\n",
    "nrow(all.snv.plasma.f)\n",
    "df.counts <- all.snv.plasma.f %>% filter(grepl('somatic|chip', VariantType.old, ignore.case=T)) %>% \n",
    "    group_by(PatientID) %>%  summarize(\"allSomaticCount\" = n())\n",
    "all.snv.plasma.f <- all.snv.plasma.f %>% left_join(df.counts %>% select(PatientID, allSomaticCount), by=\"PatientID\")\n",
    "nrow(all.snv.plasma.f)\n",
    "\n",
    "## RATIONALE FROM THE MCLAUHLIN PAPER: \n",
    "# \"Interestingly, in XGBoost and LightGBM, the most important feature is count \n",
    "#(the total number of variants to classify in the sample) rather than pop_max, which appears as third most important. \n",
    "# The lower dependency on population databases likely underlies the elimination of racially biased TMB inflation in these\"models. # Or perhaps knowing count, the total number of mutations to classify—which depends largely on the number of rare germline variants absent from the biased databases—allows LightGBM and XGBoost to recognize and make better decisions with samples from patients in underrepresented groups.\"\n",
    "\n",
    "# Check that each patient has unique total Variant count\n",
    "all.snv.plasma.f %>%  group_by(PatientID) %>% summarise(count = unique(allSomaticCount))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "8215ac2d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Original annotations:\"\n",
      "annotations\n",
      "                                    3_prime_UTR_variant \n",
      "                                                   1299 \n",
      "                                    5_prime_UTR_variant \n",
      "                                                    586 \n",
      "                                coding_sequence_variant \n",
      "                                                     26 \n",
      "                 coding_sequence_variant&intron_variant \n",
      "                                                      1 \n",
      "                                downstream_gene_variant \n",
      "                                                    415 \n",
      "                                     frameshift_variant \n",
      "                                                     88 \n",
      "               frameshift_variant&splice_region_variant \n",
      "                                                      2 \n",
      "frameshift_variant&splice_region_variant&intron_variant \n",
      "                                                    115 \n",
      "                                       inframe_deletion \n",
      "                                                     69 \n",
      "                                      inframe_insertion \n",
      "                                                     66 \n",
      "                                     intergenic_variant \n",
      "                                                   2113 \n",
      "                                         intron_variant \n",
      "                                                  28800 \n",
      "           intron_variant&non_coding_transcript_variant \n",
      "                                                    329 \n",
      "                                       missense_variant \n",
      "                                                   8066 \n",
      "                 missense_variant&splice_region_variant \n",
      "                                                    160 \n",
      "                     non_coding_transcript_exon_variant \n",
      "                                                     19 \n",
      "                               protein_altering_variant \n",
      "                                                      1 \n",
      "                                splice_acceptor_variant \n",
      "                                                     32 \n",
      "        splice_acceptor_variant&coding_sequence_variant \n",
      "                                                     16 \n",
      "                 splice_acceptor_variant&intron_variant \n",
      "                                                      1 \n",
      "                                   splice_donor_variant \n",
      "                                                     14 \n",
      "              splice_region_variant&5_prime_UTR_variant \n",
      "                                                      1 \n",
      "                   splice_region_variant&intron_variant \n",
      "                                                   1920 \n",
      "               splice_region_variant&synonymous_variant \n",
      "                                                    232 \n",
      "                                             start_lost \n",
      "                                                      2 \n",
      "                                            stop_gained \n",
      "                                                     94 \n",
      "                         stop_gained&frameshift_variant \n",
      "                                                     24 \n",
      "                      stop_gained&splice_region_variant \n",
      "                                                      3 \n",
      "                                              stop_lost \n",
      "                                                      1 \n",
      "                                     synonymous_variant \n",
      "                                                  10378 \n",
      "                                  upstream_gene_variant \n",
      "                                                    771 \n",
      "                                                   <NA> \n",
      "                                                      0 \n",
      "[1] \"New annotations:\"\n",
      "variant_classes_new\n",
      "  In_Frame   Missense       N.A.   Promoter Truncating     UTR_3p     UTR_5p \n",
      "       135       8067      42313        771       2473       1299        586 \n",
      "      <NA> \n",
      "         0 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55644</li><li>119</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55644\n",
       "\\item 119\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55644\n",
       "2. 119\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55644   119"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'VariantType.old'</li><li>'finalKeep.old'</li><li>'VariantID.2'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.cosmic'</li><li>'max.VaraintFreq.group'</li><li>'max_COSMIC_CNT.group'</li><li>'VEP.ID'</li><li>'AFR_AF'</li><li>'AMR_AF'</li><li>'EUR_AF'</li><li>'EAS_AF'</li><li>'SAS_AF'</li><li>'MAX_AF'</li><li>'SOMATIC'</li><li>'PHENO'</li><li>'mapQ_mean'</li><li>'allSomaticCount'</li><li>'Consequence.short'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'VariantType.old'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'max.VaraintFreq.group'\n",
       "\\item 'max\\_COSMIC\\_CNT.group'\n",
       "\\item 'VEP.ID'\n",
       "\\item 'AFR\\_AF'\n",
       "\\item 'AMR\\_AF'\n",
       "\\item 'EUR\\_AF'\n",
       "\\item 'EAS\\_AF'\n",
       "\\item 'SAS\\_AF'\n",
       "\\item 'MAX\\_AF'\n",
       "\\item 'SOMATIC'\n",
       "\\item 'PHENO'\n",
       "\\item 'mapQ\\_mean'\n",
       "\\item 'allSomaticCount'\n",
       "\\item 'Consequence.short'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'VariantType.old'\n",
       "2. 'finalKeep.old'\n",
       "3. 'VariantID.2'\n",
       "4. 'in.dbSNP'\n",
       "5. 'in.genome1000'\n",
       "6. 'in.cosmic'\n",
       "7. 'max.VaraintFreq.group'\n",
       "8. 'max_COSMIC_CNT.group'\n",
       "9. 'VEP.ID'\n",
       "10. 'AFR_AF'\n",
       "11. 'AMR_AF'\n",
       "12. 'EUR_AF'\n",
       "13. 'EAS_AF'\n",
       "14. 'SAS_AF'\n",
       "15. 'MAX_AF'\n",
       "16. 'SOMATIC'\n",
       "17. 'PHENO'\n",
       "18. 'mapQ_mean'\n",
       "19. 'allSomaticCount'\n",
       "20. 'Consequence.short'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"VariantType.old\"       \"finalKeep.old\"         \"VariantID.2\"          \n",
       " [4] \"in.dbSNP\"              \"in.genome1000\"         \"in.cosmic\"            \n",
       " [7] \"max.VaraintFreq.group\" \"max_COSMIC_CNT.group\"  \"VEP.ID\"               \n",
       "[10] \"AFR_AF\"                \"AMR_AF\"                \"EUR_AF\"               \n",
       "[13] \"EAS_AF\"                \"SAS_AF\"                \"MAX_AF\"               \n",
       "[16] \"SOMATIC\"               \"PHENO\"                 \"mapQ_mean\"            \n",
       "[19] \"allSomaticCount\"       \"Consequence.short\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "########## 8. Ontology: ##########\n",
    "# Categorical Coding mutation subclassification. McLaughlin's categories: inframe_indel, missense, nonsense, frame_shift_indel\n",
    "# Looks like we can use the \"Consequence\" column of the data. Maybe in the future we bin these?\n",
    "\n",
    "# here is a previously designed binning function for this. you can also create your own reclassification and \n",
    "#insert as var.reduc.set: \n",
    "source(\"~/Desktop/puffin/R/concordance_oncoprint.R\")\n",
    "all.snv.plasma.f$Consequence.short <- recode.variants(all.snv.plasma.f$Consequence, var.reduc.set = \"consequence_reduced\")\n",
    "dim(all.snv.plasma.f)\n",
    "names(all.snv.plasma.f)[100:length(all.snv.plasma.f)] #Check new features added"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "0378c688",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>49912</li><li>9</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 49912\n",
       "\\item 9\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 49912\n",
       "2. 9\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 49912     9"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>VariantID.2</th><th scope=col>context</th><th scope=col>mutType</th><th scope=col>mutMotif</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>173736:chr1:11187893:T:C </td><td>ATC</td><td>A[T&gt;C]C</td><td>T&gt;C</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>173736:chr1:11205058:C:T </td><td>TCG</td><td>T[C&gt;T]G</td><td>C&gt;T</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>173736:chr1:11288758:G:A </td><td>CGT</td><td>A[C&gt;T]G</td><td>C&gt;T</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>173736:chr1:11301714:A:G </td><td>CAT</td><td>A[T&gt;C]G</td><td>T&gt;C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>173736:chr1:117690272:G:A</td><td>CGA</td><td>T[C&gt;T]G</td><td>C&gt;T</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>173736:chr1:118165577:C:G</td><td>TCA</td><td>T[C&gt;G]A</td><td>C&gt;G</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & VariantID.2 & context & mutType & mutMotif\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 173736:chr1:11187893:T:C  & ATC & A{[}T>C{]}C & T>C\\\\\n",
       "\t2 & 173736:chr1:11205058:C:T  & TCG & T{[}C>T{]}G & C>T\\\\\n",
       "\t3 & 173736:chr1:11288758:G:A  & CGT & A{[}C>T{]}G & C>T\\\\\n",
       "\t4 & 173736:chr1:11301714:A:G  & CAT & A{[}T>C{]}G & T>C\\\\\n",
       "\t5 & 173736:chr1:117690272:G:A & CGA & T{[}C>T{]}G & C>T\\\\\n",
       "\t6 & 173736:chr1:118165577:C:G & TCA & T{[}C>G{]}A & C>G\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | VariantID.2 &lt;chr&gt; | context &lt;chr&gt; | mutType &lt;chr&gt; | mutMotif &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 173736:chr1:11187893:T:C  | ATC | A[T&gt;C]C | T&gt;C |\n",
       "| 2 | 173736:chr1:11205058:C:T  | TCG | T[C&gt;T]G | C&gt;T |\n",
       "| 3 | 173736:chr1:11288758:G:A  | CGT | A[C&gt;T]G | C&gt;T |\n",
       "| 4 | 173736:chr1:11301714:A:G  | CAT | A[T&gt;C]G | T&gt;C |\n",
       "| 5 | 173736:chr1:117690272:G:A | CGA | T[C&gt;T]G | C&gt;T |\n",
       "| 6 | 173736:chr1:118165577:C:G | TCA | T[C&gt;G]A | C&gt;G |\n",
       "\n"
      ],
      "text/plain": [
       "  VariantID.2               context mutType mutMotif\n",
       "1 173736:chr1:11187893:T:C  ATC     A[T>C]C T>C     \n",
       "2 173736:chr1:11205058:C:T  TCG     T[C>T]G C>T     \n",
       "3 173736:chr1:11288758:G:A  CGT     A[C>T]G C>T     \n",
       "4 173736:chr1:11301714:A:G  CAT     A[T>C]G T>C     \n",
       "5 173736:chr1:117690272:G:A CGA     T[C>T]G C>T     \n",
       "6 173736:chr1:118165577:C:G TCA     T[C>G]A C>G     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>52128</li><li>119</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 52128\n",
       "\\item 119\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 52128\n",
       "2. 119\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 52128   119"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'VariantType.old'</li><li>'finalKeep.old'</li><li>'VariantID.2'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.cosmic'</li><li>'max.VaraintFreq.group'</li><li>'max_COSMIC_CNT.group'</li><li>'VEP.ID'</li><li>'AFR_AF'</li><li>'AMR_AF'</li><li>'EUR_AF'</li><li>'EAS_AF'</li><li>'SAS_AF'</li><li>'MAX_AF'</li><li>'SOMATIC'</li><li>'PHENO'</li><li>'mapQ_mean'</li><li>'allSomaticCount'</li><li>'Consequence.short'</li><li>'context'</li><li>'mutType'</li><li>'mutMotif'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'VariantType.old'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'max.VaraintFreq.group'\n",
       "\\item 'max\\_COSMIC\\_CNT.group'\n",
       "\\item 'VEP.ID'\n",
       "\\item 'AFR\\_AF'\n",
       "\\item 'AMR\\_AF'\n",
       "\\item 'EUR\\_AF'\n",
       "\\item 'EAS\\_AF'\n",
       "\\item 'SAS\\_AF'\n",
       "\\item 'MAX\\_AF'\n",
       "\\item 'SOMATIC'\n",
       "\\item 'PHENO'\n",
       "\\item 'mapQ\\_mean'\n",
       "\\item 'allSomaticCount'\n",
       "\\item 'Consequence.short'\n",
       "\\item 'context'\n",
       "\\item 'mutType'\n",
       "\\item 'mutMotif'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'VariantType.old'\n",
       "2. 'finalKeep.old'\n",
       "3. 'VariantID.2'\n",
       "4. 'in.dbSNP'\n",
       "5. 'in.genome1000'\n",
       "6. 'in.cosmic'\n",
       "7. 'max.VaraintFreq.group'\n",
       "8. 'max_COSMIC_CNT.group'\n",
       "9. 'VEP.ID'\n",
       "10. 'AFR_AF'\n",
       "11. 'AMR_AF'\n",
       "12. 'EUR_AF'\n",
       "13. 'EAS_AF'\n",
       "14. 'SAS_AF'\n",
       "15. 'MAX_AF'\n",
       "16. 'SOMATIC'\n",
       "17. 'PHENO'\n",
       "18. 'mapQ_mean'\n",
       "19. 'allSomaticCount'\n",
       "20. 'Consequence.short'\n",
       "21. 'context'\n",
       "22. 'mutType'\n",
       "23. 'mutMotif'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"VariantType.old\"       \"finalKeep.old\"         \"VariantID.2\"          \n",
       " [4] \"in.dbSNP\"              \"in.genome1000\"         \"in.cosmic\"            \n",
       " [7] \"max.VaraintFreq.group\" \"max_COSMIC_CNT.group\"  \"VEP.ID\"               \n",
       "[10] \"AFR_AF\"                \"AMR_AF\"                \"EUR_AF\"               \n",
       "[13] \"EAS_AF\"                \"SAS_AF\"                \"MAX_AF\"               \n",
       "[16] \"SOMATIC\"               \"PHENO\"                 \"mapQ_mean\"            \n",
       "[19] \"allSomaticCount\"       \"Consequence.short\"     \"context\"              \n",
       "[22] \"mutType\"               \"mutMotif\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "A[C>A]A A[C>A]C A[C>A]G A[C>A]T A[C>G]A A[C>G]C A[C>G]G A[C>G]T A[C>T]A A[C>T]C \n",
       "    400     252     201     185     541     231     277     342    1201     779 \n",
       "A[C>T]G A[C>T]T A[T>A]A A[T>A]C A[T>A]G A[T>A]T A[T>C]A A[T>C]C A[T>C]G A[T>C]T \n",
       "   2324    1095     148     288     252     163    1097     552    2325    1067 \n",
       "A[T>G]A A[T>G]C A[T>G]G A[T>G]T C[C>A]A C[C>A]C C[C>A]G C[C>A]T C[C>G]A C[C>G]C \n",
       "    189     171     172     126     277     278     276     199     248     331 \n",
       "C[C>G]G C[C>G]T C[C>T]A C[C>T]C C[C>T]G C[C>T]T C[T>A]A C[T>A]C C[T>A]G C[T>A]T \n",
       "    563     410     645    1105    2261    1027     102     321     235     309 \n",
       "C[T>C]A C[T>C]C C[T>C]G C[T>C]T C[T>G]A C[T>G]C C[T>G]G C[T>G]T G[C>A]A G[C>A]C \n",
       "    637     801    2118     740     111     361     505     264     327     326 \n",
       "G[C>A]G G[C>A]T G[C>G]A G[C>G]C G[C>G]G G[C>G]T G[C>T]A G[C>T]C G[C>T]G G[C>T]T \n",
       "    154     111     372     288     618     276     615     827    1654     681 \n",
       "G[T>A]A G[T>A]C G[T>A]G G[T>A]T G[T>C]A G[T>C]C G[T>C]G G[T>C]T G[T>G]A G[T>G]C \n",
       "    101     145     262      40     607     547    1593     635     220     256 \n",
       "G[T>G]G G[T>G]T T[C>A]A T[C>A]C T[C>A]G T[C>A]T T[C>G]A T[C>G]C T[C>G]G T[C>G]T \n",
       "    723     283     226     215      99     187     241     362     344     537 \n",
       "T[C>T]A T[C>T]C T[C>T]G T[C>T]T T[T>A]A T[T>A]C T[T>A]G T[T>A]T T[T>C]A T[T>C]C \n",
       "    763     693    1104     811     131      79     150     215     591     611 \n",
       "T[T>C]G T[T>C]T T[T>G]A T[T>G]C T[T>G]G T[T>G]T    <NA> \n",
       "    857     724     106     115     234     354    2216 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  C>A   C>G   C>T   T>A   T>C   T>G  <NA> \n",
       " 3713  5981 17585  2941 15502  4190  2216 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>52128</li><li>122</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 52128\n",
       "\\item 122\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 52128\n",
       "2. 122\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 52128   122"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "########## COSMIC MUTATION SBS SIGNATURE: ##########\n",
    "\n",
    "########## 9. trinucleotide_context: ##########\n",
    "# The three surrounding nucleotides (1 bp upstream, the locus itself, and 1 bp downstream) in the reference genome for the SNV (non_SBS = non-single base-pair substitution, i.e, indels)\n",
    "\n",
    "######### From mutationSignature workflow:\n",
    "# We will add the tri \"context\" column\n",
    "# R script: \"get-trinucleotide.R\"\n",
    "\n",
    "# Load in our trinucleotide context and mutation motifs\n",
    "mut <- read.delim(\"./Tram_working_files/tri-mut-context_SEP.txt\", sep = \"\\t\", header = T, as.is=T)\n",
    "dim(mut)\n",
    "\n",
    "mut <- mut %>% mutate(VariantID.2=paste0(PatientID, \":\", VariantID))\n",
    "mut <- mut %>%  select(VariantID.2, context, mutType, mutMotif) # Just grab relevant info to join new columns\n",
    "mut$VariantID.2 <- as.character(mut$VariantID.2)\n",
    "head(mut)\n",
    "\n",
    "# Merge tri and mut context to ALL PLASMA variants\n",
    "### REMOVE INDELS\n",
    "all.snv.plasma.f = all.snv.plasma.f %>% filter(width<2)\n",
    "dim(all.snv.plasma.f)\n",
    "all.snv.plasma.f <- all.snv.plasma.f %>% left_join(mut, by=\"VariantID.2\")\n",
    "\n",
    "names(all.snv.plasma.f)[100:length(all.snv.plasma.f)] # Should see the tri-nucle, mutation Type, and signature columns added\n",
    "\n",
    "########## substitution_type: ##########\t\n",
    "# Which of the 6 unique transition and transversion types characterizes the single-base-substitution (non-SBS for indels)\n",
    "# C>G, C>T, T>A, T>C, T>G, non-SBS, C>A\n",
    "\n",
    "#mut <- read.delim(\"tri-mut-context.txt\", sep = \"\\t\", header = T, as.is=T)\n",
    "# Merge on the new trinucleotide/motif columns. NA for if not available for variants.\n",
    "#all.snv.plasma <- all.snv.plasma  %>% left_join(mut, by=\"ID\", suffix = c(\"\", \".drop\")) %>%\n",
    "#  select(-ends_with(\".drop\"))\n",
    "#head(all.snv.plasma)\n",
    "\n",
    "## For variants with mutType/mutMotif (indels) = set NA to nonSBS\n",
    "# all.snv.plasma = all.snv.plasma %>% mutate(mutType=ifelse(nchar(ref)>1 | nchar(alt)>1, \"nonSBS\", mutType))\n",
    "#all.snv.plasma$mutMotif[is.na(all.snv.plasma$mutMotif)] <- \"nonSBS\"\n",
    "\n",
    "# NOTE: MAYBE LEAVE THIS NA?\n",
    "table(all.snv.plasma.f$mutType)\n",
    "table(all.snv.plasma.f$mutMotif)\n",
    "\n",
    "dim(all.snv.plasma.f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "339c3553",
   "metadata": {},
   "outputs": [],
   "source": [
    "### FROM SIGPROFILERASSIGNMENT_R PROGRAM : \n",
    "### MORE DETAILED CODE TO GENERATE THESE OUTPUTS IN: \"SigProfileExtractor-workflow.R\" script. Uploaded into the test_data OneDrive\n",
    "#https://github.com/AlexandrovLab/SigProfilerAssignment\n",
    "#The script to match the program's outputs with the variants is pretty rough and I limited to only single-base substitutions but I think the code could definitely use another look through\n",
    "\n",
    "\n",
    "# Read in outputs -- which are currently separate for each patient\n",
    "mydir = (\"./Tram_working_files/Assignment_Solution/Activities/Decomposed_Mutation_Probabilities\")\n",
    "myfiles = list.files(path=mydir, pattern = \".txt\", full.names = T)\n",
    "mytitles = basename(myfiles)\n",
    "mytitles\n",
    "names <- sub(\"Decomposed_Mutation_Probabilities_\", \"\", mytitles)\n",
    "names <- sub(\".txt\", \"\", names)\n",
    "names <- sub(\"P\", \"\", names) # remove the P from the patentID to match plasma\n",
    "\n",
    "tables <- lapply(myfiles, read.delim, as.is=T, sep=\"\\t\", header=T)\n",
    "names(tables) <- names\n",
    "\n",
    "# Load in the plasma data - lets do a test df in case it messes up\n",
    "test.plasma <- all.snv.plasma.f\n",
    "\n",
    "#create empty list with length of zero\n",
    "sig_list <- list()\n",
    "\n",
    "#create empty list of length 10\n",
    "sig_list <- vector(mode='list', length=23)\n",
    "names(sig_list) <- names\n",
    "\n",
    "\n",
    "# Grab a list of PatientIDs. Loop through each one and grab the SBS probabilities.\n",
    "# This is a bit complicated because we cannot get VariantID.2 from just the outputs. We need to compare it to the VCF files that were provided as input, and then take the ref and alt alleles from those to create a VariantID.2 column that will match plasma. \n",
    "\n",
    "for (i in 1:length(names)){\n",
    "   \n",
    "   c.patient <- names[i]\n",
    "   \n",
    "   c.df <- tables[[c.patient]]\n",
    "   \n",
    "   # Remove the P from the sampleName\n",
    "   c.df$Sample.Names <- gsub(\"P\", \"\", c.df$Sample.Names)\n",
    "   \n",
    "   # Load in the VCF coords inputs for the SigProfiler\n",
    "   vcfname <- paste0(\"./Tram_working_files//Documents/Code/sigExtractor/atlas_vcfs/input/\", \"P\", c.patient, \".vcf\")\n",
    "   vcf <- read.delim(vcfname, as.is=T, sep = \"\\t\", header = F)\n",
    "   \n",
    "   nrow(c.df)\n",
    "   nrow(vcf) # Usually more VCF variants because these still include indels\n",
    "   \n",
    "   \n",
    "   # Limit VCF to only SBS -- remove indels from VCF coords \n",
    "   vcf_SBS <- vcf %>%\n",
    "     filter(nchar(V4) == 1 & nchar(V5) == 1)\n",
    "   \n",
    "   # In the VCF, sometimes there are positions that have two calls... remove those rows\n",
    "   vcf_SBS <- vcf_SBS %>%\n",
    "     distinct(V1, V2, .keep_all = TRUE)\n",
    "   \n",
    "   # Check set intersect again\n",
    "   c.df <- c.df %>% mutate(VariantID.2=paste0(Sample.Names,\":\", \"chr\", Chr, \":\", Pos))\n",
    "   vcf <- vcf_SBS %>% mutate(VariantID.2=paste0(V3,\":\", \"chr\", V1, \":\", V2))\n",
    "   \n",
    "   # There will be more c.df now because of the two-call positions\n",
    "   nrow(c.df)\n",
    "   nrow(vcf)\n",
    "   \n",
    "   vcf$VariantID.2[which(duplicated(vcf$VariantID.2))] # should be none becuase we removed the duplicated coords\n",
    "   dups <- c.df$VariantID.2[which(duplicated(c.df$VariantID.2))]\n",
    "   dups\n",
    "   \n",
    "   # remove the duplicates 2bp entries from SigProfiler\n",
    "   c.df <- c.df %>% filter(!c.df$VariantID.2 %in% dups)\n",
    "   \n",
    "   nrow(c.df)\n",
    "   nrow(vcf)\n",
    "   \n",
    "   # Just grab the necessary mutSigs from the unique variants in VCF coordinates\n",
    "   merged_df <- merge(c.df, vcf, by = \"VariantID.2\", all = FALSE) # only what's in common\n",
    "   \n",
    "   # Replace the Variant2 column ID to match plasma dataset format (tag on ref and alt allele)\n",
    "   matched_rows <- merged_df %>% mutate(VariantID.2=paste0(VariantID.2,\":\", V4, \":\", V5))\n",
    "   \n",
    "   # Add SigProfile info to the plasma dataset\n",
    "   matched_rows <- matched_rows %>% select(starts_with(\"SBS\"), \"VariantID.2\", \"MutationType\") # Grab necessary columns\n",
    "   \n",
    "   #Check how many variants have SigProfile info for\n",
    "   length(intersect(matched_rows$VariantID.2, test.plasma$VariantID.2))\n",
    "   \n",
    "   \n",
    "   # add the new data set to the sig_list\n",
    "   \n",
    "   sig_list[[c.patient]] <- matched_rows\n",
    "\n",
    "}\n",
    "\n",
    "# Combine into one dataframe\n",
    "c.df <- bind_rows(sig_list)\n",
    "\n",
    "# Remove any SBS signatures columns without any probabilities assigned\n",
    "c.df <- c.df[, !apply(c.df, 2, function(x) all(x == 0))]\n",
    "column_names <- colnames(c.df)\n",
    "column_names[grepl(\"^SBS\", column_names)] # 10 signatures\n",
    "\n",
    "# For SBS signatures without probs, set to zero\n",
    "c.df[is.na(c.df)] <- 0\n",
    "\n",
    "# left_join the SBS signatures to our plasma data\n",
    "test.plasma <- test.plasma %>% left_join(c.df, by=\"VariantID.2\")\n",
    "\n",
    "dim(test.plasma) # 133 columns\n",
    "names(test.plasma)[100:length(test.plasma)] # Should see all the SBS signature columns now.\n",
    "head(test.plasma[,110:length(test.plasma)])\n",
    "\n",
    "#all.snv.plasma <- test.plasma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23e66af4",
   "metadata": {},
   "outputs": [],
   "source": [
    "all.snv.plasma <- test.plasma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "448d8a3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'VariantType.old'</li><li>'finalKeep.old'</li><li>'VariantID.2'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.cosmic'</li><li>'SBS_Count'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'VariantType.old'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'SBS\\_Count'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'VariantType.old'\n",
       "2. 'finalKeep.old'\n",
       "3. 'VariantID.2'\n",
       "4. 'in.dbSNP'\n",
       "5. 'in.genome1000'\n",
       "6. 'in.cosmic'\n",
       "7. 'SBS_Count'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"VariantType.old\" \"finalKeep.old\"   \"VariantID.2\"     \"in.dbSNP\"       \n",
       "[5] \"in.genome1000\"   \"in.cosmic\"       \"SBS_Count\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"173736\" \"172592\" \"175702\" \"89200\"  \"174438\" \"172956\" \"174280\" \"172109\"\n",
      " [9] \"170762\" \"170594\" \"172163\" \"173129\" \"171660\" \"175253\" \"171351\" \"175743\"\n",
      "[17] \"172422\" \"171126\" \"173657\" \"174688\" \"175827\" \"173149\" \"173983\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'VariantType.old'</li><li>'finalKeep.old'</li><li>'VariantID.2'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.cosmic'</li><li>'SBS_Count'</li><li>'pTMB_norm.adj'</li><li>'pTMB_norm.weighted'</li><li>'Fragment_Size_Bandwidth'</li><li>'Fragment_Size_Mode'</li><li>'maxAF_TMB'</li><li>'tumorFraction'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'VariantType.old'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'SBS\\_Count'\n",
       "\\item 'pTMB\\_norm.adj'\n",
       "\\item 'pTMB\\_norm.weighted'\n",
       "\\item 'Fragment\\_Size\\_Bandwidth'\n",
       "\\item 'Fragment\\_Size\\_Mode'\n",
       "\\item 'maxAF\\_TMB'\n",
       "\\item 'tumorFraction'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'VariantType.old'\n",
       "2. 'finalKeep.old'\n",
       "3. 'VariantID.2'\n",
       "4. 'in.dbSNP'\n",
       "5. 'in.genome1000'\n",
       "6. 'in.cosmic'\n",
       "7. 'SBS_Count'\n",
       "8. 'pTMB_norm.adj'\n",
       "9. 'pTMB_norm.weighted'\n",
       "10. 'Fragment_Size_Bandwidth'\n",
       "11. 'Fragment_Size_Mode'\n",
       "12. 'maxAF_TMB'\n",
       "13. 'tumorFraction'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"VariantType.old\"         \"finalKeep.old\"          \n",
       " [3] \"VariantID.2\"             \"in.dbSNP\"               \n",
       " [5] \"in.genome1000\"           \"in.cosmic\"              \n",
       " [7] \"SBS_Count\"               \"pTMB_norm.adj\"          \n",
       " [9] \"pTMB_norm.weighted\"      \"Fragment_Size_Bandwidth\"\n",
       "[11] \"Fragment_Size_Mode\"      \"maxAF_TMB\"              \n",
       "[13] \"tumorFraction\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "######################################################\n",
    "########## ADD PATIENT LEVEL RELEVANT INFO: ##########\n",
    "######################################################\n",
    "\n",
    "#### NUMBER OF COSMIC SIGNATURES PER INDIVIDUAL ####\n",
    "\n",
    "SigPat <- read.delim(\"./Tram_working_files/Assignment_Solution/Activities/Assignment_Solution_Activities.txt\", as.is=T, header = T, sep=\"\\t\")\n",
    "\n",
    "# remove the patient column\n",
    "SigPat$Samples <- gsub(\"P\", \"\", SigPat$Samples)\n",
    "\n",
    "SigPat <- as.data.frame(SigPat)\n",
    "\n",
    "# Get column names with counts greater than zero\n",
    "SBS_names <- apply(SigPat[, -1] > 0, 1, function(row) {\n",
    "  colnames(SigPat)[-1][row]\n",
    "})\n",
    "\n",
    "\n",
    "SBS_count <- apply(SigPat[, -1] > 0, 1, sum)\n",
    "SBS_count <- as.data.frame(cbind(unique(SigPat$Samples), SBS_count))\n",
    "\n",
    "colnames(SBS_count) <- c(\"PatientID\", \"SBS_Count\")\n",
    "SBS_count$SBS_Count <- as.integer(SBS_count$SBS_Count)\n",
    "\n",
    "# Left_join number of mut sigs to all plasma data\n",
    "all.snv.plasma <- all.snv.plasma %>% left_join(SBS_count, by=\"PatientID\")\n",
    "names(all.snv.plasma)[100:length(all.snv.plasma)]\n",
    "\n",
    "\n",
    "#### PATIENT CLINICAL DATA ####\n",
    "NGSQC <- read.csv(\"~/Desktop/Moffitt_2022/MIBC/data/ATLAS/PRDC-MOFFITT-MIBC-22002/07_27_23_update/PRDC-MOFFITT-MIBC-22002-ATLAS_NGSQC_plasma_updated_2023-07-28.csv\", header = T)\n",
    "NGSQC = NGSQC %>% mutate(PatientID.old = PatientID,\n",
    "                                         PatientID = fix_PIDs(PatientID.old))\n",
    "\n",
    "# TMB\n",
    "patient_cols <- c(\"PatientID\", \"pTMB_norm.adj\", \"pTMB_norm.weighted\", \"Fragment_Size_Bandwidth\", \"Fragment_Size_Mode\", \"maxAF_TMB\", \"tumorFraction\")\n",
    "\n",
    "NGSQC <- NGSQC %>% select(all_of(patient_cols))\n",
    "all.snv.plasma <- all.snv.plasma %>% left_join(NGSQC, by=\"PatientID\")\n",
    "names(all.snv.plasma)[100:length(all.snv.plasma)]\n",
    "\n",
    "## Save new plasma variants \n",
    "#all.snv.plasma <- as.data.frame(all.snv.plasma)\n",
    "#save(all.snv.plasma, file = \"McL_plasma.RData\")\n",
    "#load(\"McL_plasma.RData\")\n",
    "\n",
    "\n",
    "#############################################################################\n",
    "################### COMPLETED ADDING MCLAUGHLIN FEATURES  ###################\n",
    "#############################################################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "b8a30c00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE  <NA> \n",
       " 9800 45844     0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55644</li><li>116</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55644\n",
       "\\item 116\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55644\n",
       "2. 116\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55644   116"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'VariantType.old'</li><li>'finalKeep.old'</li><li>'VariantID.2'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.cosmic'</li><li>'SBS_Count'</li><li>'pTMB_norm.adj'</li><li>'pTMB_norm.weighted'</li><li>'Fragment_Size_Bandwidth'</li><li>'Fragment_Size_Mode'</li><li>'maxAF_TMB'</li><li>'tumorFraction'</li><li>'Plasma.VariantFreq'</li><li>'Germ.VariantFreq'</li><li>'Germ.VariantType'</li><li>'in.pbmc'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'VariantType.old'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'SBS\\_Count'\n",
       "\\item 'pTMB\\_norm.adj'\n",
       "\\item 'pTMB\\_norm.weighted'\n",
       "\\item 'Fragment\\_Size\\_Bandwidth'\n",
       "\\item 'Fragment\\_Size\\_Mode'\n",
       "\\item 'maxAF\\_TMB'\n",
       "\\item 'tumorFraction'\n",
       "\\item 'Plasma.VariantFreq'\n",
       "\\item 'Germ.VariantFreq'\n",
       "\\item 'Germ.VariantType'\n",
       "\\item 'in.pbmc'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'VariantType.old'\n",
       "2. 'finalKeep.old'\n",
       "3. 'VariantID.2'\n",
       "4. 'in.dbSNP'\n",
       "5. 'in.genome1000'\n",
       "6. 'in.cosmic'\n",
       "7. 'SBS_Count'\n",
       "8. 'pTMB_norm.adj'\n",
       "9. 'pTMB_norm.weighted'\n",
       "10. 'Fragment_Size_Bandwidth'\n",
       "11. 'Fragment_Size_Mode'\n",
       "12. 'maxAF_TMB'\n",
       "13. 'tumorFraction'\n",
       "14. 'Plasma.VariantFreq'\n",
       "15. 'Germ.VariantFreq'\n",
       "16. 'Germ.VariantType'\n",
       "17. 'in.pbmc'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"VariantType.old\"         \"finalKeep.old\"          \n",
       " [3] \"VariantID.2\"             \"in.dbSNP\"               \n",
       " [5] \"in.genome1000\"           \"in.cosmic\"              \n",
       " [7] \"SBS_Count\"               \"pTMB_norm.adj\"          \n",
       " [9] \"pTMB_norm.weighted\"      \"Fragment_Size_Bandwidth\"\n",
       "[11] \"Fragment_Size_Mode\"      \"maxAF_TMB\"              \n",
       "[13] \"tumorFraction\"           \"Plasma.VariantFreq\"     \n",
       "[15] \"Germ.VariantFreq\"        \"Germ.VariantType\"       \n",
       "[17] \"in.pbmc\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "        \n",
       "         CHIP mutations Germline Likely background Likely CHIP mutations\n",
       "  170594              1     2390                47                     1\n",
       "  170762              3     2201                62                     2\n",
       "  171126              1     2077                75                     2\n",
       "  171351              4     2295                60                     2\n",
       "  171660              3     2234                60                     2\n",
       "  172109              2     2178                61                     1\n",
       "  172163              0     2263                61                     1\n",
       "  172422              0     2166                52                     2\n",
       "  172592              1     2146                32                     2\n",
       "  172956              2     2240                62                     1\n",
       "  173129              0     2109                51                     1\n",
       "  173149              0     2100                67                     2\n",
       "  173657              1     2090                47                     2\n",
       "  173736              1     2202                59                     1\n",
       "  173983              0     2146                10                     1\n",
       "  174280              3     2167                73                     1\n",
       "  174438              2     2181                50                     2\n",
       "  174688              2     2347                56                     2\n",
       "  175253              1     2192                53                     2\n",
       "  175702              0     2190                56                     1\n",
       "  175743              0     2244                32                     2\n",
       "  175827              3     2078                44                     3\n",
       "  89200               3     2214                46                     0\n",
       "  <NA>                0        0                 0                     0\n",
       "        \n",
       "         Likely germline Likely somatic Somatic <NA>\n",
       "  170594              66             23     106    0\n",
       "  170762              56             17     107    0\n",
       "  171126              41             24      68    0\n",
       "  171351              34             20     100    0\n",
       "  171660              49             28     102    0\n",
       "  172109              28             17      72    0\n",
       "  172163              34             26      82    0\n",
       "  172422              39             25      95    0\n",
       "  172592              36             17     103    0\n",
       "  172956              27             40      89    0\n",
       "  173129              51             22      98    0\n",
       "  173149              44             20      63    0\n",
       "  173657              40             85      95    0\n",
       "  173736              44             24     107    0\n",
       "  173983             207             29     188    0\n",
       "  174280              36             36      89    0\n",
       "  174438              54             16      93    0\n",
       "  174688              35             21      89    0\n",
       "  175253              41             25      88    0\n",
       "  175702              31             20     103    0\n",
       "  175743              35             18      79    0\n",
       "  175827              36             23      89    0\n",
       "  89200               37             24     103    0\n",
       "  <NA>                 0              0       0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "       CHIP mutations              Germline     Likely background \n",
       "                   33                 50450                  1216 \n",
       "Likely CHIP mutations       Likely germline        Likely somatic \n",
       "                   36                  1101                   600 \n",
       "              Somatic                  <NA> \n",
       "                 2208                     0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AF'</li><li>'AF.Baseline'</li><li>'AF.diff'</li><li>'AF.th'</li><li>'alt'</li><li>'altDepth'</li><li>'altDepth.Baseline'</li><li>'altDepth.Baseline.filtered'</li><li>'altLOD'</li><li>'Amino_acids'</li><li>'avgEntropy'</li><li>'bgFilter'</li><li>'BIOTYPE'</li><li>'CANONICAL'</li><li>'CLIN_SIG'</li><li>'Clinvar'</li><li>'Codons'</li><li>'concordant'</li><li>'Consequence'</li><li>'CopyNumber'</li><li>'COSMIC'</li><li>'COSMIC.CNT'</li><li>'coVariant'</li><li>'coVariantFilter'</li><li>'dbSNP'</li><li>'depth'</li><li>'DISTANCE'</li><li>'DS.filter'</li><li>'DSCnt'</li><li>'dscnt.Baseline.filtered'</li><li>'end'</li><li>'entropy'</li><li>'Existing_variation'</li><li>'EXON'</li><li>'externalSampleID'</li><li>'fc.ds.ratio'</li><li>'Feature'</li><li>'filtered_tlen_mean'</li><li>'filtered_toend_numunique'</li><li>'filtered_total_base_count'</li><li>'filteredCnt'</li><li>'filteredDSCnt'</li><li>'filterPrevalence'</li><li>'filterScore'</li><li>'FilterType'</li><li>'filterVarType'</li><li>'finalKeep'</li><li>'finalKeep.old'</li><li>'Fragment_Size_Bandwidth'</li><li>'Fragment_Size_Mode'</li><li>'GC.Percent'</li><li>'GeneID'</li><li>'generalFilter'</li><li>'genome1000'</li><li>'Germ.VariantFreq'</li><li>'Germ.VariantType'</li><li>'HGVSc'</li><li>'HGVSp'</li><li>'highFrequent.inbatch'</li><li>'highQ.ratio'</li><li>'ID'</li><li>'IMPACT'</li><li>'in.cosmic'</li><li>'in.dbSNP'</li><li>'in.genome1000'</li><li>'in.pbmc'</li><li>'INTRON'</li><li>'localVariantFilter'</li><li>'MAF.diff'</li><li>'maxAF_TMB'</li><li>'mergedVariant'</li><li>'MNV'</li><li>'odds.ratio'</li><li>'ol.coVariant'</li><li>'PatientID'</li><li>'PatientID.old'</li><li>'Plasma.VariantFreq'</li><li>'prevalenceInternal'</li><li>'pseudoDSCnt'</li><li>'pTMB_norm.adj'</li><li>'pTMB_norm.weighted'</li><li>'pvalue'</li><li>'Qscore'</li><li>'ReadEndFilter'</li><li>'ref'</li><li>'refDepth'</li><li>'repeatRegion'</li><li>'sampleDir'</li><li>'sampleFolder'</li><li>'sampleNames'</li><li>'SBS_Count'</li><li>'seqnames'</li><li>'sideVariant'</li><li>'sideVariant.AF.ratio'</li><li>'snvmember'</li><li>'SpecimenType'</li><li>'start'</li><li>'strand'</li><li>'SYMBOL'</li><li>'tandemRepeat'</li><li>'tlen.ratio'</li><li>'totalDepth'</li><li>'totalDepth.Baseline'</li><li>'trialVisitNum'</li><li>'tumorFraction'</li><li>'VARIANT_CLASS'</li><li>'VariantFreq'</li><li>'VariantID'</li><li>'VariantID.2'</li><li>'VariantType'</li><li>'VariantType.old'</li><li>'white.list'</li><li>'width'</li><li>'wl.tier'</li><li>'X'</li><li>'zScore.cnv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AF'\n",
       "\\item 'AF.Baseline'\n",
       "\\item 'AF.diff'\n",
       "\\item 'AF.th'\n",
       "\\item 'alt'\n",
       "\\item 'altDepth'\n",
       "\\item 'altDepth.Baseline'\n",
       "\\item 'altDepth.Baseline.filtered'\n",
       "\\item 'altLOD'\n",
       "\\item 'Amino\\_acids'\n",
       "\\item 'avgEntropy'\n",
       "\\item 'bgFilter'\n",
       "\\item 'BIOTYPE'\n",
       "\\item 'CANONICAL'\n",
       "\\item 'CLIN\\_SIG'\n",
       "\\item 'Clinvar'\n",
       "\\item 'Codons'\n",
       "\\item 'concordant'\n",
       "\\item 'Consequence'\n",
       "\\item 'CopyNumber'\n",
       "\\item 'COSMIC'\n",
       "\\item 'COSMIC.CNT'\n",
       "\\item 'coVariant'\n",
       "\\item 'coVariantFilter'\n",
       "\\item 'dbSNP'\n",
       "\\item 'depth'\n",
       "\\item 'DISTANCE'\n",
       "\\item 'DS.filter'\n",
       "\\item 'DSCnt'\n",
       "\\item 'dscnt.Baseline.filtered'\n",
       "\\item 'end'\n",
       "\\item 'entropy'\n",
       "\\item 'Existing\\_variation'\n",
       "\\item 'EXON'\n",
       "\\item 'externalSampleID'\n",
       "\\item 'fc.ds.ratio'\n",
       "\\item 'Feature'\n",
       "\\item 'filtered\\_tlen\\_mean'\n",
       "\\item 'filtered\\_toend\\_numunique'\n",
       "\\item 'filtered\\_total\\_base\\_count'\n",
       "\\item 'filteredCnt'\n",
       "\\item 'filteredDSCnt'\n",
       "\\item 'filterPrevalence'\n",
       "\\item 'filterScore'\n",
       "\\item 'FilterType'\n",
       "\\item 'filterVarType'\n",
       "\\item 'finalKeep'\n",
       "\\item 'finalKeep.old'\n",
       "\\item 'Fragment\\_Size\\_Bandwidth'\n",
       "\\item 'Fragment\\_Size\\_Mode'\n",
       "\\item 'GC.Percent'\n",
       "\\item 'GeneID'\n",
       "\\item 'generalFilter'\n",
       "\\item 'genome1000'\n",
       "\\item 'Germ.VariantFreq'\n",
       "\\item 'Germ.VariantType'\n",
       "\\item 'HGVSc'\n",
       "\\item 'HGVSp'\n",
       "\\item 'highFrequent.inbatch'\n",
       "\\item 'highQ.ratio'\n",
       "\\item 'ID'\n",
       "\\item 'IMPACT'\n",
       "\\item 'in.cosmic'\n",
       "\\item 'in.dbSNP'\n",
       "\\item 'in.genome1000'\n",
       "\\item 'in.pbmc'\n",
       "\\item 'INTRON'\n",
       "\\item 'localVariantFilter'\n",
       "\\item 'MAF.diff'\n",
       "\\item 'maxAF\\_TMB'\n",
       "\\item 'mergedVariant'\n",
       "\\item 'MNV'\n",
       "\\item 'odds.ratio'\n",
       "\\item 'ol.coVariant'\n",
       "\\item 'PatientID'\n",
       "\\item 'PatientID.old'\n",
       "\\item 'Plasma.VariantFreq'\n",
       "\\item 'prevalenceInternal'\n",
       "\\item 'pseudoDSCnt'\n",
       "\\item 'pTMB\\_norm.adj'\n",
       "\\item 'pTMB\\_norm.weighted'\n",
       "\\item 'pvalue'\n",
       "\\item 'Qscore'\n",
       "\\item 'ReadEndFilter'\n",
       "\\item 'ref'\n",
       "\\item 'refDepth'\n",
       "\\item 'repeatRegion'\n",
       "\\item 'sampleDir'\n",
       "\\item 'sampleFolder'\n",
       "\\item 'sampleNames'\n",
       "\\item 'SBS\\_Count'\n",
       "\\item 'seqnames'\n",
       "\\item 'sideVariant'\n",
       "\\item 'sideVariant.AF.ratio'\n",
       "\\item 'snvmember'\n",
       "\\item 'SpecimenType'\n",
       "\\item 'start'\n",
       "\\item 'strand'\n",
       "\\item 'SYMBOL'\n",
       "\\item 'tandemRepeat'\n",
       "\\item 'tlen.ratio'\n",
       "\\item 'totalDepth'\n",
       "\\item 'totalDepth.Baseline'\n",
       "\\item 'trialVisitNum'\n",
       "\\item 'tumorFraction'\n",
       "\\item 'VARIANT\\_CLASS'\n",
       "\\item 'VariantFreq'\n",
       "\\item 'VariantID'\n",
       "\\item 'VariantID.2'\n",
       "\\item 'VariantType'\n",
       "\\item 'VariantType.old'\n",
       "\\item 'white.list'\n",
       "\\item 'width'\n",
       "\\item 'wl.tier'\n",
       "\\item 'X'\n",
       "\\item 'zScore.cnv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AF'\n",
       "2. 'AF.Baseline'\n",
       "3. 'AF.diff'\n",
       "4. 'AF.th'\n",
       "5. 'alt'\n",
       "6. 'altDepth'\n",
       "7. 'altDepth.Baseline'\n",
       "8. 'altDepth.Baseline.filtered'\n",
       "9. 'altLOD'\n",
       "10. 'Amino_acids'\n",
       "11. 'avgEntropy'\n",
       "12. 'bgFilter'\n",
       "13. 'BIOTYPE'\n",
       "14. 'CANONICAL'\n",
       "15. 'CLIN_SIG'\n",
       "16. 'Clinvar'\n",
       "17. 'Codons'\n",
       "18. 'concordant'\n",
       "19. 'Consequence'\n",
       "20. 'CopyNumber'\n",
       "21. 'COSMIC'\n",
       "22. 'COSMIC.CNT'\n",
       "23. 'coVariant'\n",
       "24. 'coVariantFilter'\n",
       "25. 'dbSNP'\n",
       "26. 'depth'\n",
       "27. 'DISTANCE'\n",
       "28. 'DS.filter'\n",
       "29. 'DSCnt'\n",
       "30. 'dscnt.Baseline.filtered'\n",
       "31. 'end'\n",
       "32. 'entropy'\n",
       "33. 'Existing_variation'\n",
       "34. 'EXON'\n",
       "35. 'externalSampleID'\n",
       "36. 'fc.ds.ratio'\n",
       "37. 'Feature'\n",
       "38. 'filtered_tlen_mean'\n",
       "39. 'filtered_toend_numunique'\n",
       "40. 'filtered_total_base_count'\n",
       "41. 'filteredCnt'\n",
       "42. 'filteredDSCnt'\n",
       "43. 'filterPrevalence'\n",
       "44. 'filterScore'\n",
       "45. 'FilterType'\n",
       "46. 'filterVarType'\n",
       "47. 'finalKeep'\n",
       "48. 'finalKeep.old'\n",
       "49. 'Fragment_Size_Bandwidth'\n",
       "50. 'Fragment_Size_Mode'\n",
       "51. 'GC.Percent'\n",
       "52. 'GeneID'\n",
       "53. 'generalFilter'\n",
       "54. 'genome1000'\n",
       "55. 'Germ.VariantFreq'\n",
       "56. 'Germ.VariantType'\n",
       "57. 'HGVSc'\n",
       "58. 'HGVSp'\n",
       "59. 'highFrequent.inbatch'\n",
       "60. 'highQ.ratio'\n",
       "61. 'ID'\n",
       "62. 'IMPACT'\n",
       "63. 'in.cosmic'\n",
       "64. 'in.dbSNP'\n",
       "65. 'in.genome1000'\n",
       "66. 'in.pbmc'\n",
       "67. 'INTRON'\n",
       "68. 'localVariantFilter'\n",
       "69. 'MAF.diff'\n",
       "70. 'maxAF_TMB'\n",
       "71. 'mergedVariant'\n",
       "72. 'MNV'\n",
       "73. 'odds.ratio'\n",
       "74. 'ol.coVariant'\n",
       "75. 'PatientID'\n",
       "76. 'PatientID.old'\n",
       "77. 'Plasma.VariantFreq'\n",
       "78. 'prevalenceInternal'\n",
       "79. 'pseudoDSCnt'\n",
       "80. 'pTMB_norm.adj'\n",
       "81. 'pTMB_norm.weighted'\n",
       "82. 'pvalue'\n",
       "83. 'Qscore'\n",
       "84. 'ReadEndFilter'\n",
       "85. 'ref'\n",
       "86. 'refDepth'\n",
       "87. 'repeatRegion'\n",
       "88. 'sampleDir'\n",
       "89. 'sampleFolder'\n",
       "90. 'sampleNames'\n",
       "91. 'SBS_Count'\n",
       "92. 'seqnames'\n",
       "93. 'sideVariant'\n",
       "94. 'sideVariant.AF.ratio'\n",
       "95. 'snvmember'\n",
       "96. 'SpecimenType'\n",
       "97. 'start'\n",
       "98. 'strand'\n",
       "99. 'SYMBOL'\n",
       "100. 'tandemRepeat'\n",
       "101. 'tlen.ratio'\n",
       "102. 'totalDepth'\n",
       "103. 'totalDepth.Baseline'\n",
       "104. 'trialVisitNum'\n",
       "105. 'tumorFraction'\n",
       "106. 'VARIANT_CLASS'\n",
       "107. 'VariantFreq'\n",
       "108. 'VariantID'\n",
       "109. 'VariantID.2'\n",
       "110. 'VariantType'\n",
       "111. 'VariantType.old'\n",
       "112. 'white.list'\n",
       "113. 'width'\n",
       "114. 'wl.tier'\n",
       "115. 'X'\n",
       "116. 'zScore.cnv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"AF\"                         \"AF.Baseline\"               \n",
       "  [3] \"AF.diff\"                    \"AF.th\"                     \n",
       "  [5] \"alt\"                        \"altDepth\"                  \n",
       "  [7] \"altDepth.Baseline\"          \"altDepth.Baseline.filtered\"\n",
       "  [9] \"altLOD\"                     \"Amino_acids\"               \n",
       " [11] \"avgEntropy\"                 \"bgFilter\"                  \n",
       " [13] \"BIOTYPE\"                    \"CANONICAL\"                 \n",
       " [15] \"CLIN_SIG\"                   \"Clinvar\"                   \n",
       " [17] \"Codons\"                     \"concordant\"                \n",
       " [19] \"Consequence\"                \"CopyNumber\"                \n",
       " [21] \"COSMIC\"                     \"COSMIC.CNT\"                \n",
       " [23] \"coVariant\"                  \"coVariantFilter\"           \n",
       " [25] \"dbSNP\"                      \"depth\"                     \n",
       " [27] \"DISTANCE\"                   \"DS.filter\"                 \n",
       " [29] \"DSCnt\"                      \"dscnt.Baseline.filtered\"   \n",
       " [31] \"end\"                        \"entropy\"                   \n",
       " [33] \"Existing_variation\"         \"EXON\"                      \n",
       " [35] \"externalSampleID\"           \"fc.ds.ratio\"               \n",
       " [37] \"Feature\"                    \"filtered_tlen_mean\"        \n",
       " [39] \"filtered_toend_numunique\"   \"filtered_total_base_count\" \n",
       " [41] \"filteredCnt\"                \"filteredDSCnt\"             \n",
       " [43] \"filterPrevalence\"           \"filterScore\"               \n",
       " [45] \"FilterType\"                 \"filterVarType\"             \n",
       " [47] \"finalKeep\"                  \"finalKeep.old\"             \n",
       " [49] \"Fragment_Size_Bandwidth\"    \"Fragment_Size_Mode\"        \n",
       " [51] \"GC.Percent\"                 \"GeneID\"                    \n",
       " [53] \"generalFilter\"              \"genome1000\"                \n",
       " [55] \"Germ.VariantFreq\"           \"Germ.VariantType\"          \n",
       " [57] \"HGVSc\"                      \"HGVSp\"                     \n",
       " [59] \"highFrequent.inbatch\"       \"highQ.ratio\"               \n",
       " [61] \"ID\"                         \"IMPACT\"                    \n",
       " [63] \"in.cosmic\"                  \"in.dbSNP\"                  \n",
       " [65] \"in.genome1000\"              \"in.pbmc\"                   \n",
       " [67] \"INTRON\"                     \"localVariantFilter\"        \n",
       " [69] \"MAF.diff\"                   \"maxAF_TMB\"                 \n",
       " [71] \"mergedVariant\"              \"MNV\"                       \n",
       " [73] \"odds.ratio\"                 \"ol.coVariant\"              \n",
       " [75] \"PatientID\"                  \"PatientID.old\"             \n",
       " [77] \"Plasma.VariantFreq\"         \"prevalenceInternal\"        \n",
       " [79] \"pseudoDSCnt\"                \"pTMB_norm.adj\"             \n",
       " [81] \"pTMB_norm.weighted\"         \"pvalue\"                    \n",
       " [83] \"Qscore\"                     \"ReadEndFilter\"             \n",
       " [85] \"ref\"                        \"refDepth\"                  \n",
       " [87] \"repeatRegion\"               \"sampleDir\"                 \n",
       " [89] \"sampleFolder\"               \"sampleNames\"               \n",
       " [91] \"SBS_Count\"                  \"seqnames\"                  \n",
       " [93] \"sideVariant\"                \"sideVariant.AF.ratio\"      \n",
       " [95] \"snvmember\"                  \"SpecimenType\"              \n",
       " [97] \"start\"                      \"strand\"                    \n",
       " [99] \"SYMBOL\"                     \"tandemRepeat\"              \n",
       "[101] \"tlen.ratio\"                 \"totalDepth\"                \n",
       "[103] \"totalDepth.Baseline\"        \"trialVisitNum\"             \n",
       "[105] \"tumorFraction\"              \"VARIANT_CLASS\"             \n",
       "[107] \"VariantFreq\"                \"VariantID\"                 \n",
       "[109] \"VariantID.2\"                \"VariantType\"               \n",
       "[111] \"VariantType.old\"            \"white.list\"                \n",
       "[113] \"width\"                      \"wl.tier\"                   \n",
       "[115] \"X\"                          \"zScore.cnv\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# create variant identifier columns\n",
    "# In the CHIP data, it made sense to check within just somatic calls -- but for us, we want to check ALL plasma variants\n",
    "all.snv.plasma.McL = all.snv.plasma %>% \n",
    "  mutate(VariantID.2=paste0(PatientID, \":\", VariantID),\n",
    "         Plasma.VariantFreq = VariantFreq) \n",
    "\n",
    "# these features can correlate with truth labels, but will not be used for training as they will not be available for\n",
    "# new/holdout data\n",
    "all.pbmc_ = all.pbmc %>% mutate(VariantID.2=paste0(PatientID, \":\", VariantID), Germ.VariantFreq = VariantFreq,\n",
    "                                Germ.VariantType = VariantType) %>% # might need this info, mostly Germline, some might be CHIP\n",
    "  select(VariantID.2, Germ.VariantFreq, Germ.VariantType)\n",
    "\n",
    "# for variants with Germ.VariantFreq values that are NA (missing from germline sample), set Germ.VariantFreq to 0.0%\n",
    "all.snv.plasma.McL = all.snv.plasma.McL %>% left_join(all.pbmc_, by=\"VariantID.2\") %>%\n",
    "  mutate(Germ.VariantFreq=ifelse(is.na(Germ.VariantFreq), 0, Germ.VariantFreq),\n",
    "         in.pbmc = factor(Germ.VariantFreq>0)) # Create another column for updated germline freq for somatic calls \n",
    "\n",
    "# all.snv.tumor_ = all.snv.tumor %>% \n",
    "#   mutate(VariantID.2=paste0(PatientID, \":\", VariantID), Tumor.VariantFreq = VariantFreq, Tumor.VariantType = VariantType) %>%\n",
    "#   select(VariantID.2, Tumor.VariantFreq, Tumor.VariantType) # Add the tumor variant type and freq\n",
    "# all.snv.plasma.McL = all.snv.plasma.McL %>% left_join(all.snv.tumor_, by=\"VariantID.2\") %>%\n",
    "#   mutate(Tumor.VariantFreq=ifelse(is.na(Tumor.VariantFreq), 0, Tumor.VariantFreq),\n",
    "#          in.tumor = factor(Tumor.VariantFreq > 0) #,\n",
    "#          #Tumor.VariantType=ifelse(is.na(Tumor.VariantType), \"Germline\", Tumor.VariantType)\n",
    "#   )   \n",
    "## I don't think we need this last feature in the model, it is just for reference later, so maybe leave as NA\n",
    "\n",
    "\n",
    "table(all.snv.plasma.McL$in.pbmc) # 39585 variants overlap PBMC + plasma ATLAS\n",
    "#table(all.snv.plasma.McL$in.tumor) # 42340 variants overlap FFPE + plasma ATLAS . . . these are mostly germline mutations\n",
    "dim(all.snv.plasma.McL)\n",
    "\n",
    "# View all of the new columns we've added onto the plasma ATLAS dataset\n",
    "names(all.snv.plasma.McL)[100:length(all.snv.plasma.McL)]\n",
    "\n",
    "### \n",
    "# somatic Tumor.VariantType counts are low because most of the tumor variants are not detectable in the plasma;\n",
    "# or Tumor.VariantType==NA.\n",
    "#table(all.snv.plasma.McL$PatientID, all.snv.plasma.McL$Tumor.VariantType, useNA=\"always\") \n",
    "# here we have about ~120 somatic variants in the plasma per patient\n",
    "table(all.snv.plasma.McL$PatientID, all.snv.plasma.McL$VariantType, useNA=\"always\")\n",
    "\n",
    "table(all.snv.plasma.McL$VariantType) # about 2400 variants labeled as somatic/likely somatic/chip by the existing pipeline\n",
    "sort(names(all.snv.plasma.McL))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "8ab736f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Plasma   <NA> \n",
       " 55644      0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " CHIP mutations        Germline Likely germline  Likely somatic         Somatic \n",
       "             56           47018            1928            6629              13 \n",
       "           <NA> \n",
       "              0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): \n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): \nTraceback:\n",
      "1. stop()"
     ]
    }
   ],
   "source": [
    "# Ground truth = VariantType is somatic/CHIP after germline matched normal pipeline\n",
    "# We are treating CHIP in same category as somatic here.\n",
    "\n",
    "print(\"Raw labels\")\n",
    "table(all.snv.plasma.McL$VariantType.old) # raw pipeline labels\n",
    "\n",
    "all.snv.plasma.McL <- all.snv.plasma.McL %>% \n",
    "  mutate(raw.label = factor(ifelse(grepl(\"somatic|chip\", VariantType.old, ignore.case=T), \"somatic\",\"germline\"), levels=c(\"somatic\",\"germline\")),\n",
    "         matched.label=factor(ifelse(grepl(\"somatic|chip\", VariantType, ignore.case=T), \"somatic\",\"germline\"), levels=c(\"somatic\",\"germline\")),\n",
    "         # truth.label = factor(ifelse(grepl(\"somatic|chip\", Tumor.VariantType, ignore.case=T), \"somatic\",\"germline\"), levels=c(\"somatic\",\"germline\"))\n",
    "         # prevents weird treatment of NA values:\n",
    "         truth.label = factor(ifelse(!is.na(Tumor.VariantType) & grepl(\"somatic|chip\", Tumor.VariantType, ignore.case=T), \"somatic\",\"germline\"), levels=c(\"somatic\",\"germline\")))\n",
    "\n",
    "# Truth set will be: variantType = somatic | chip in FFPE (Tumor.VariantType column)\n",
    "\n",
    "table(all.snv.plasma.McL$raw.label) \n",
    "table(all.snv.plasma.McL$matched.label)\n",
    "table(all.snv.plasma.McL$truth.label)\n",
    "\n",
    "# variantType = somatic | chip in FFPE\n",
    "# match to plasma data set patientID\n",
    "\n",
    "\n",
    "## wow, a very small subset of the plasma somatic variants are also called as somatic in the tumor\n",
    "## maybe our pipeline somatic calling for tumor tissue isn't that good. might explore a less conservative truth label?\n",
    "# all.snv.plasma.McL <- all.snv.plasma.McL %>% \n",
    "#                 mutate(truth.label.2 = factor(ifelse(matched.label==\"somatic\" & as.logical(in.tumor), \"somatic\", \"germline\"), levels=c(\"somatic\",\"germline\")))\n",
    "# table(all.snv.plasma.McL$truth.label.2) # doesn't help much :-(\n",
    "# somatic germline \n",
    "#      158    47939 \n",
    "\n",
    "\n",
    "## this would be a good place to produce a venn diagram of how the three types of labels overlap for any variant with a somatic call\n",
    "## some code for this in previous notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7695fdc8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
